{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c8df45c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os\n",
    "import math\n",
    "from itertools import chain\n",
    "import pickle\n",
    "import sqlalchemy\n",
    "from sqlalchemy.orm import sessionmaker\n",
    "from sqlalchemy.engine import Engine\n",
    "from sqlalchemy import event\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0ecdc3b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "@event.listens_for(Engine, \"connect\")\n",
    "def set_sqlite_pragma(dbapi_connection, connection_record):\n",
    "    cursor = dbapi_connection.cursor()\n",
    "    cursor.execute(\"PRAGMA foreign_keys=ON\")\n",
    "    cursor.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8dd010d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# @event.listens_for(Engine, \"connect\")\n",
    "# def set_sqlite_pragma(dbapi_connection, connection_reconajmd):\n",
    "#     cursor = dbapi_connection.cursor()\n",
    "#     #cursor.execute(\"PRAGMA foreign_keys=OFF\")\n",
    "#     cursor.execute(\"PRAGMA foreign_keys=ON\")\n",
    "\n",
    "#     invalid_foreign_keys = cursor.execute(\"PRAGMA foreign_key_check\").fetchall()\n",
    "#     if invalid_foreign_keys:\n",
    "#         print('A')\n",
    "#         msg = \"\\n\".join(\n",
    "#             f\"{invalid_key.table} (rowid: {invalid_key.rowid}) -> {invalid_key.parent}\"\n",
    "#             for invalid_key in invalid_foreign_keys\n",
    "#         )\n",
    "#         msg = 'A'\n",
    "#         raise ValueError(f\"Invalid Foreign Keys detected: {msg}\")\n",
    "        \n",
    "#     cursor.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f3a69c2",
   "metadata": {},
   "source": [
    "## Create a SL Database with Wet-Lab Experiments"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47e2c471",
   "metadata": {},
   "source": [
    "all SL data:\n",
    "\n",
    "* Diehl\n",
    "* Han\n",
    "* Horlbeck (Done)\n",
    "* Ito\n",
    "* Laufer (Done)\n",
    "* Parrish\n",
    "* Shen\n",
    "* Thompson\n",
    "* Wong\n",
    "* Zhao (Done)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c058f760",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF = pd.read_csv('../Python/original_GI_110422-Copy1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a0670acd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load in the datasets\n",
    "data_locs = \"/users/PAS1376/bg12/SyntheticLethality/SyntheticLethalityReview/Project/ml_inputs\"\n",
    "learning_goals_loc_general =  os.path.join(data_locs, \"learning_goals\")\n",
    "learning_goals_loc_general = '/users/PAS1376/bg12/SyntheticLethality - NewDB/data'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4630a40a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read the database\n",
    "SLKB_engine = sqlalchemy.create_engine('sqlite:///SLKB_sqlite3')\n",
    "#SLKB_engine = sqlalchemy.create_engine('sqlite:///shan_new_SL')\n",
    "#SLKB_engine = sqlalchemy.create_engine('sqlite:///shuai_SL')\n",
    "#SLKB_engine_session = sessionmaker(bind=SLKB_engine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "62140407",
   "metadata": {},
   "outputs": [],
   "source": [
    "db_metadata = sqlalchemy.MetaData(bind=SLKB_engine)\n",
    "db_metadata.reflect(SLKB_engine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c7e6efcb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FacadeDict({'CDKO_EXPERIMENT_DESIGN': Table('CDKO_EXPERIMENT_DESIGN', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('sgRNA_id', INTEGER(), table=<CDKO_EXPERIMENT_DESIGN>, primary_key=True), Column('sgRNA_guide_name', TEXT(), table=<CDKO_EXPERIMENT_DESIGN>, nullable=False), Column('sgRNA_guide_seq', TEXT(), table=<CDKO_EXPERIMENT_DESIGN>, nullable=False), Column('sgRNA_target_name', TEXT(), table=<CDKO_EXPERIMENT_DESIGN>, nullable=False), Column('study_origin', TEXT(), table=<CDKO_EXPERIMENT_DESIGN>, nullable=False), schema=None), 'CDKO_ORIGINAL_SL_RESULTS': Table('CDKO_ORIGINAL_SL_RESULTS', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<CDKO_ORIGINAL_SL_RESULTS>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<CDKO_ORIGINAL_SL_RESULTS>), Column('gene_pair', TEXT(), table=<CDKO_ORIGINAL_SL_RESULTS>, nullable=False), Column('study_origin', TEXT(), table=<CDKO_ORIGINAL_SL_RESULTS>, nullable=False), Column('cell_line_origin', TEXT(), table=<CDKO_ORIGINAL_SL_RESULTS>, nullable=False), Column('gene_1', TEXT(), table=<CDKO_ORIGINAL_SL_RESULTS>, nullable=False), Column('gene_2', TEXT(), table=<CDKO_ORIGINAL_SL_RESULTS>, nullable=False), Column('SL_or_not', TEXT(), table=<CDKO_ORIGINAL_SL_RESULTS>, nullable=False), Column('SL_score', REAL(), table=<CDKO_ORIGINAL_SL_RESULTS>), Column('statistical_score', REAL(), table=<CDKO_ORIGINAL_SL_RESULTS>), Column('SL_score_cutoff', REAL(), table=<CDKO_ORIGINAL_SL_RESULTS>), Column('statistical_score_cutoff', REAL(), table=<CDKO_ORIGINAL_SL_RESULTS>), schema=None), 'CDKO_SGRNA_COUNTS': Table('CDKO_SGRNA_COUNTS', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('sgRNA_pair_id', INTEGER(), table=<CDKO_SGRNA_COUNTS>, primary_key=True), Column('guide_1_id', INTEGER(), ForeignKey('CDKO_EXPERIMENT_DESIGN.sgRNA_id'), table=<CDKO_SGRNA_COUNTS>), Column('guide_2_id', INTEGER(), ForeignKey('CDKO_EXPERIMENT_DESIGN.sgRNA_id'), table=<CDKO_SGRNA_COUNTS>), Column('gene_pair_id', INTEGER(), table=<CDKO_SGRNA_COUNTS>), Column('gene_pair_orientation', TEXT(), table=<CDKO_SGRNA_COUNTS>), Column('T0_counts', TEXT(), table=<CDKO_SGRNA_COUNTS>), Column('T0_replicate_names', TEXT(), table=<CDKO_SGRNA_COUNTS>), Column('TEnd_counts', TEXT(), table=<CDKO_SGRNA_COUNTS>), Column('TEnd_replicate_names', TEXT(), table=<CDKO_SGRNA_COUNTS>), Column('target_type', TEXT(), table=<CDKO_SGRNA_COUNTS>), Column('study_origin', TEXT(), table=<CDKO_SGRNA_COUNTS>, nullable=False), Column('cell_line_origin', TEXT(), table=<CDKO_SGRNA_COUNTS>, nullable=False), schema=None), 'GEMINI_SCORE': Table('GEMINI_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<GEMINI_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<GEMINI_SCORE>), Column('SL_score_Strong', REAL(), table=<GEMINI_SCORE>), Column('SL_score_SensitiveLethality', REAL(), table=<GEMINI_SCORE>), Column('SL_score_SensitiveRecovery', REAL(), table=<GEMINI_SCORE>), schema=None), 'HORLBECK_SCORE': Table('HORLBECK_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<HORLBECK_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<HORLBECK_SCORE>), Column('SL_score', REAL(), table=<HORLBECK_SCORE>), Column('standard_error', REAL(), table=<HORLBECK_SCORE>), schema=None), 'MAGECK_SCORE': Table('MAGECK_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<MAGECK_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<MAGECK_SCORE>), Column('SL_score', REAL(), table=<MAGECK_SCORE>), Column('standard_error', REAL(), table=<MAGECK_SCORE>), Column('Z_SL_score', REAL(), table=<MAGECK_SCORE>), schema=None), 'MEDIAN_B_SCORE': Table('MEDIAN_B_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<MEDIAN_B_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<MEDIAN_B_SCORE>), Column('SL_score', REAL(), table=<MEDIAN_B_SCORE>), Column('standard_error', REAL(), table=<MEDIAN_B_SCORE>), Column('Z_SL_score', REAL(), table=<MEDIAN_B_SCORE>), schema=None), 'MEDIAN_NB_SCORE': Table('MEDIAN_NB_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<MEDIAN_NB_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<MEDIAN_NB_SCORE>), Column('SL_score', REAL(), table=<MEDIAN_NB_SCORE>), Column('standard_error', REAL(), table=<MEDIAN_NB_SCORE>), Column('Z_SL_score', REAL(), table=<MEDIAN_NB_SCORE>), schema=None), 'SGRA_DERIVED_B_SCORE': Table('SGRA_DERIVED_B_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<SGRA_DERIVED_B_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<SGRA_DERIVED_B_SCORE>), Column('SL_score', REAL(), table=<SGRA_DERIVED_B_SCORE>), schema=None), 'SGRA_DERIVED_NB_SCORE': Table('SGRA_DERIVED_NB_SCORE', MetaData(bind=Engine(sqlite:///SLKB_sqlite3)), Column('id', INTEGER(), table=<SGRA_DERIVED_NB_SCORE>, primary_key=True), Column('gene_pair_id', INTEGER(), table=<SGRA_DERIVED_NB_SCORE>), Column('SL_score', REAL(), table=<SGRA_DERIVED_NB_SCORE>), schema=None)})"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_metadata.tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1f2238a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# To delete all\n",
    "#db_metadata.drop_all()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f6a6e0d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_cell_line_matrix(input_matrix, cell_line = \"temp\"):\n",
    "    input_matrix_res = input_matrix.copy(deep = True)\n",
    "    input_matrix_res = input_matrix_res.astype(object)\n",
    "    input_matrix_res.values[np.nonzero(input_matrix_res.values)] = cell_line\n",
    "    return(input_matrix_res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0e16ae72",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_matrix_from_df(input_data):\n",
    "    all_genes = sorted(set(input_data.loc[:, 'Gene_A'].tolist() + input_data.loc[:, 'Gene_B'].tolist()))\n",
    "    print(\"Total Number of genes: \" + str(len(all_genes)))\n",
    "    curr_GI_matrix = pd.DataFrame(columns = all_genes, index = all_genes)\n",
    "    for i in range(len(input_data.index)):\n",
    "        gene_a = input_data.loc[i, \"Gene_A\"]\n",
    "        gene_b = input_data.loc[i, \"Gene_B\"]\n",
    "        curr_GI_matrix.loc[gene_a, gene_b] = input_data.loc[i, \"GI_Score\"]\n",
    "        curr_GI_matrix.loc[gene_b, gene_a] = input_data.loc[i, \"GI_Score\"]\n",
    "        if i % 1000 == 0:\n",
    "            print(str(i) + \"/\" + str(len(input_data.index)))\n",
    "    return(curr_GI_matrix.fillna(0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "64eed579",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_repeated_constructs(x, index_loc):\n",
    "    if len(x) < max(index_loc):\n",
    "        sub = index_loc[index_loc < len(x)]\n",
    "        return(x[sub])\n",
    "    else:\n",
    "        return(x[index_loc])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e90e3f6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "## store pubmed IDs\n",
    "study_name_to_pubmed_id = {}\n",
    "\n",
    "study_name_to_pubmed_id['diehl_data'] = '33956155'\n",
    "\n",
    "study_name_to_pubmed_id['han_data'] = '28319085'\n",
    "\n",
    "study_name_to_pubmed_id['horlbeck_data'] = '30033366'\n",
    "\n",
    "study_name_to_pubmed_id['ito_data'] = '34857952'\n",
    "\n",
    "study_name_to_pubmed_id['parrish_data'] = '34469736'\n",
    "\n",
    "study_name_to_pubmed_id['shen_data'] = '28319113'\n",
    "\n",
    "study_name_to_pubmed_id['thompson_data'] = '33637726'\n",
    "\n",
    "study_name_to_pubmed_id['wong_data'] = '26864203'\n",
    "\n",
    "study_name_to_pubmed_id['zhao_data'] = '29452643'\n",
    "\n",
    "study_name_to_pubmed_id['shantang_data'] = '36060092'\n",
    "\n",
    "study_name_to_pubmed_id['najm_data'] = '29251726'\n",
    "\n",
    "study_name_to_pubmed_id['shantang_2CL_data'] = 'Shan_2CL'\n",
    "\n",
    "study_name_to_pubmed_id['shuai_data'] = 'Shuai'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2bcc5ada",
   "metadata": {},
   "outputs": [],
   "source": [
    "## controls\n",
    "controls = {}\n",
    "\n",
    "# diehl et al\n",
    "controls['diehl_data'] = [\"wildtype-I-CeuI\", \"wildtype-I-SceI\", \"nan\"] + [\"Non-Human-Target-114\", \"Non-Human-Target-122\",\n",
    "                                    \"Non-Human-Target-144\", \"Non-Human-Target-148\", \"Non-Human-Target-161\", \"Non-Human-Target-178\", \"Non-Human-Target-185\", \"Non-Human-Target-222\", \"Non-Human-Target-223\",\n",
    "                                    \"Non-Human-Target-224\", \"Non-Human-Target-23\", \"Non-Human-Target-243\", \"Non-Human-Target-245\", \"Non-Human-Target-249\", \"Non-Human-Target-292\", \"Non-Human-Target-298\",\n",
    "                                    \"Non-Human-Target-31\", \"Non-Human-Target-311\", \"Non-Human-Target-313\", \"Non-Human-Target-327\", \"Non-Human-Target-333\", \"Non-Human-Target-335\", \"Non-Human-Target-339\",\n",
    "                                    \"Non-Human-Target-341\", \"Non-Human-Target-343\", \"Non-Human-Target-389\", \"Non-Human-Target-39\", \"Non-Human-Target-397\", \"Non-Human-Target-398\", \"Non-Human-Target-40\", \n",
    "                                    \"Non-Human-Target-402\", \"Non-Human-Target-42\", \"Non-Human-Target-432\", \"Non-Human-Target-444\", \"Non-Human-Target-466\", \"Non-Human-Target-479\", \"Non-Human-Target-495\",\n",
    "                                    \"Non-Human-Target-512\", \"Non-Human-Target-515\", \"Non-Human-Target-526\", \"Non-Human-Target-532\", \"Non-Human-Target-542\", \"Non-Human-Target-547\", \"Non-Human-Target-55\", \n",
    "                                    \"Non-Human-Target-551\", \"Non-Human-Target-560\", \"Non-Human-Target-565\", \"Non-Human-Target-584\", \"Non-Human-Target-588\", \"Non-Human-Target-595\", \"Non-Human-Target-602\",\n",
    "                                    \"Non-Human-Target-622\", \"Non-Human-Target-636\", \"Non-Human-Target-637\", \"Non-Human-Target-654\", \"Non-Human-Target-659\", \"Non-Human-Target-668\", \"Non-Human-Target-676\",\n",
    "                                    \"Non-Human-Target-678\", \"Non-Human-Target-681\", \"Non-Human-Target-692\", \"Non-Human-Target-719\", \"Non-Human-Target-732\", \"Non-Human-Target-736\", \"Non-Human-Target-748\",\n",
    "                                    \"Non-Human-Target-752\", \"Non-Human-Target-766\", \"Non-Human-Target-782\", \"Non-Human-Target-789\", \"Non-Human-Target-798\", \"Non-Human-Target-799\", \"Non-Human-Target-801\",\n",
    "                                    \"Non-Human-Target-805\", \"Non-Human-Target-807\", \"Non-Human-Target-808\", \"Non-Human-Target-814\", \"Non-Human-Target-816\", \"Non-Human-Target-822\", \"Non-Human-Target-824\",\n",
    "                                    \"Non-Human-Target-827\", \"Non-Human-Target-828\", \"Non-Human-Target-835\", \"Non-Human-Target-85\", \"Non-Human-Target-857\", \"Non-Human-Target-863\", \"Non-Human-Target-864\",\n",
    "                                    \"Non-Human-Target-877\", \"Non-Human-Target-88\", \"Non-Human-Target-884\", \"Non-Human-Target-900\", \"Non-Human-Target-902\", \"Non-Human-Target-905\", \"Non-Human-Target-941\",\n",
    "                                    \"Non-Human-Target-942\", \"Non-Human-Target-953\", \"Non-Human-Target-954\", \"Non-Human-Target-958\", \"Non-Human-Target-959\", \"Non-Human-Target-965\", \"Non-Human-Target-969\",\n",
    "                                    \"Non-Human-Target-970\", \"Non-Human-Target-99\", \"Non-Human-Target-995\", \"Non-Human-Target-997\"]\n",
    "\n",
    "# horlbeck et al\n",
    "controls['horlbeck_data'] = [\"negative\"]\n",
    "\n",
    "# parrish et al\n",
    "controls['parrish_data'] = [\"nt\" + str(i+1) for i in range(975)] + [\"FAKE_GENE_\" + str(i+1) for i in range(50)]\n",
    "controls['parrish_data_original'] = [\"nt\" + str(i+1) for i in range(975)] + [\"FAKE_GENE_\" + str(i+1) for i in range(50)]\n",
    "\n",
    "# wong et al\n",
    "controls['wong_data'] = [\"DUMMYGUIDE\"]#, 'GFP', 'RFP'\n",
    "\n",
    "# zhao et al\n",
    "controls['zhao_data'] = [\"0\", 'CONTROL']\n",
    "\n",
    "# tang et al\n",
    "controls['shantang_data'] = ['0SAFE',\n",
    "                             '0SAFE-SAFE-GE',\n",
    "                             '0SAFE-SAFE-SP',\n",
    "                             '0SAFE-SAFE-MP',\n",
    "                             '0SAFE-SAFE-U2',\n",
    "                             '0SAFE-SAFE-DTKP',\n",
    "                             '0SAFE-SAFE-ACOC',\n",
    "                             '0SAFE-SAFE-TMM',\n",
    "                             '0SAFE-SAFE-U1',\n",
    "                             '0SAFE-SAFE-U3']\n",
    "\n",
    "controls['ito_data'] = ['AAVS1']\n",
    "\n",
    "controls['najm_data'] = [\"HPRT INTRON\",\n",
    "                         \"6T\",\n",
    "                         \"EEF2\",\n",
    "                         \"CD81\"]\n",
    "\n",
    "controls['shen_data'] = ['control']\n",
    "\n",
    "controls['shantang_2CL_data'] = ['0SAFE',\n",
    "                                 '0SAFE-SAFE-GE',\n",
    "                                 '0SAFE-SAFE-SP',\n",
    "                                 '0SAFE-SAFE-MP',\n",
    "                                 '0SAFE-SAFE-U2',\n",
    "                                 '0SAFE-SAFE-DTKP',\n",
    "                                 '0SAFE-SAFE-ACOC',\n",
    "                                 '0SAFE-SAFE-TMM',\n",
    "                                 '0SAFE-SAFE-U1',\n",
    "                                 '0SAFE-SAFE-U3']\n",
    "\n",
    "\n",
    "controls['shuai_data'] = ['0SAFE',\n",
    "                                 '0SAFE-SAFE-GE',\n",
    "                                 '0SAFE-SAFE-SP',\n",
    "                                 '0SAFE-SAFE-MP',\n",
    "                                 '0SAFE-SAFE-U2',\n",
    "                                 '0SAFE-SAFE-DTKP',\n",
    "                                 '0SAFE-SAFE-ACOC',\n",
    "                                 '0SAFE-SAFE-TMM',\n",
    "                                 '0SAFE-SAFE-U1',\n",
    "                                 '0SAFE-SAFE-U3']\n",
    "\n",
    "for study in controls:\n",
    "    controls[study] = [i.upper() for i in controls[study]]# + [i for i in controls[study]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "18c7ba8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "## conditions\n",
    "study_conditions = {}\n",
    "\n",
    "# diehl et al\n",
    "study_conditions['diehl_data'] = [['ctrl_1',\n",
    "                                    'ctrl_2'],\n",
    "                                   ['rep_1',\n",
    "                                    'rep_2',\n",
    "                                    'rep_3']]\n",
    "\n",
    "# ctrl_1;ctrl_2;rep_1;rep_2;rep_3\n",
    "# [\"wildtype-I-CeuI\", \"wildtype-I-SceI\", \"nan\"]\n",
    "\n",
    "# horlbeck et al\n",
    "study_conditions['horlbeck_data'] = {}\n",
    "# study_conditions['horlbeck_data']['JURKAT'] = [['JURKAT_barcode,T0,rep1',\n",
    "#                                                 'JURKAT_barcode,T0,rep2'],\n",
    "#                                                ['JURKAT_barcode,cyc,rep1',\n",
    "#                                                 'JURKAT_barcode,cyc,rep2',]]\n",
    "study_conditions['horlbeck_data']['JURKAT'] = [['JURKAT_tripleseq,T0,rep1',\n",
    "                                                'JURKAT_tripleseq,T0,rep2'],\n",
    "                                               ['JURKAT_tripleseq,cyc,rep1',\n",
    "                                                'JURKAT_tripleseq,cyc,rep2',]]\n",
    "\n",
    "# study_conditions['horlbeck_data']['K562'] = [['K562_barcode,T0,rep1',\n",
    "#                                         'K562_barcode,T0,rep2'],\n",
    "#                                        ['K562_barcode,cyc,rep1',\n",
    "#                                         'K562_barcode,cyc,rep2',]]\n",
    "study_conditions['horlbeck_data']['K562'] = [['K562_barcode,T0,rep1',\n",
    "                                        'K562_barcode,T0,rep2'],\n",
    "                                       ['K562_tripleseq,cyc,rep1',\n",
    "                                        'K562_tripleseq,cyc,rep2',]]\n",
    "\n",
    "# parrish et al\n",
    "study_conditions['parrish_data'] = [[\"plasmid_1\", \n",
    "                                     \"plasmid_2\",\n",
    "                                     \"plasmid_3\"],\n",
    "                                    [\"LTP_1\",\n",
    "                                     \"LTP_2\",\n",
    "                                     \"LTP_3\"]]\n",
    "study_conditions['parrish_data_original'] = [[\"plasmid_1\", \n",
    "                                     \"plasmid_2\",\n",
    "                                     \"plasmid_3\"],\n",
    "                                    [\"LTP_1\",\n",
    "                                     \"LTP_2\",\n",
    "                                     \"LTP_3\"]]\n",
    "\n",
    "# wong et al\n",
    "study_conditions['wong_data'] = [[\"day5 (Replicate 1)\", \n",
    "                                  \"day5 (Replicate 2)\"],\n",
    "                                 [\"day20 (Replicate 1)\",\n",
    "                                  \"day20 (Replicate 2)\"]]\n",
    "\n",
    "# zhao et al\n",
    "study_conditions['zhao_data'] = {}\n",
    "study_conditions['zhao_data'][\"HELA\"] = [[\"Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts\",\n",
    "                                          \"Hela_MV4_d3_2_S2_trimmed53_len_filtered_counts\"],\n",
    "                                         [\"Hela_MV4_d28_1_S5_trimmed53_len_filtered_counts\",\n",
    "                                          \"Hela_MV4_d28_2_S6_trimmed53_len_filtered_counts\"]]\n",
    "\n",
    "study_conditions['zhao_data'][\"A549\"] = [[\"A549_MV4_d3_1_S1_trimmed53_len_filtered_counts\",\n",
    "                                          \"A549_MV4_d3_2_S2_trimmed53_len_filtered_counts\"],\n",
    "                                         [\"A549_MV4_d28_1_S7_trimmed53_len_filtered_counts\",\n",
    "                                          \"A549_MV4_d28_2_S8_trimmed53_len_filtered_counts\"]]\n",
    "\n",
    "study_conditions['shantang_data'] = [[\"T0_1\", \n",
    "                                     \"T0_2\"],\n",
    "                                    [\"T12_1\",\n",
    "                                     \"T12_2\"]]\n",
    "\n",
    "study_conditions['shantang_data'] = [[\"T0_1\", \n",
    "                                     \"T0_2\"],\n",
    "                                    [\"T12_1\",\n",
    "                                     \"T12_2\"]]\n",
    "\n",
    "study_conditions['shuai_data'] = [['T1',\n",
    "                                   'T2',\n",
    "                                   'T3'],\n",
    "                                  ['F1',\n",
    "                                   'F2',\n",
    "                                   'F3']]\n",
    "\n",
    "\n",
    "study_conditions['shantang_2CL_data'] = {}\n",
    "study_conditions['shantang_2CL_data'][\"SAOS2\"] = [[\"S0A\", \n",
    "                                                   \"S0B\",\n",
    "                                                   \"S0C\"],\n",
    "                                                  [\"SEA\", \n",
    "                                                   \"SEB\",\n",
    "                                                   \"SEC\"]]\n",
    "study_conditions['shantang_2CL_data'][\"TT2\"] =   [[\"T0A\", \n",
    "                                                   \"T0B\",\n",
    "                                                   \"T0C\"],\n",
    "                                                  [\"TEA\", \n",
    "                                                   \"TEB\",\n",
    "                                                   \"TEC\"]]\n",
    "\n",
    "study_conditions['najm_data'] = [['pDNA_Reads'],\n",
    "                                 ['Rep_A_Reads',\n",
    "                                  'Rep_B_Reads',\n",
    "                                  'Rep_C_Reads']]\n",
    "\n",
    "# study_conditions['ito_data'] = [['pDNA_Reads'],\n",
    "#                                  ['Rep_A_Reads',\n",
    "#                                   'Rep_B_Reads',\n",
    "#                                   'Rep_C_Reads']]\n",
    "\n",
    "study_conditions['shen_data'] = {}\n",
    "study_conditions['shen_data']['HELA'] = [['Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_counts',\n",
    "       'Hela-CV4R4R_d3_2_S8_trimmed53_len_filtered_counts'],\n",
    "                                         ['Hela-CV4-d28-1_S5_trimmed53_len_filtered_counts',\n",
    "       'Hela-CV4-d28-2_S6_trimmed53_len_filtered_counts']]\n",
    "\n",
    "study_conditions['shen_data']['293T'] = [['293_CV4_d3_3_S1_trimmed53_len_filtered_counts',\n",
    "       '293_CV4_d3_4_S2_trimmed53_len_filtered_counts'],\n",
    "                                         ['293_CV4_d28_3_S7_trimmed53_len_filtered_counts',\n",
    "       '293_CV4_d28_4_S8_trimmed53_len_filtered_counts']]\n",
    "\n",
    "study_conditions['shen_data']['A549'] = [['A549_CV4_d3_1_S1_trimmed53_len_filtered_counts',\n",
    "                                            'A549_CV4_d3_2_S2_trimmed53_len_filtered_counts'],\n",
    "                                         ['A549_CV4_d28_1_S5_trimmed53_len_filtered_counts',\n",
    "                                          'A549_CV4_d28_2_S6_trimmed53_len_filtered_counts']]\n",
    "\n",
    "study_conditions['ito_data'] = {'MEWO': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'IPC298': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'HS944T': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'HS936T': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'MEL202': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'A549': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'MELJUSO': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'PATU8988S': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads']],\n",
    "                                 'HSC5': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'PK1': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
    "                                 'GI1': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']]}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "79c34487",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_placeholder_scores(curr_counts, sequence_ref):\n",
    "    # we should add genes to the KB that can later be modified following scoring\n",
    "    \n",
    "    # first, set the controls so they can be removed\n",
    "    curr_counts.loc[curr_counts['Guide 1'].isin(sequence_ref.loc[sequence_ref['sgRNA_target_name'] == 'control', 'sgRNA_guide_name'].values), 'Gene 1'] = 'CONTROL'\n",
    "    curr_counts.loc[curr_counts['Guide 2'].isin(sequence_ref.loc[sequence_ref['sgRNA_target_name'] == 'control', 'sgRNA_guide_name'].values), 'Gene 2'] = 'CONTROL'\n",
    "\n",
    "    idx = (curr_counts['Gene 1'] == 'CONTROL') | (curr_counts['Gene 2'] == 'CONTROL')\n",
    "    # remove them\n",
    "    curr_counts = curr_counts[~idx]\n",
    "\n",
    "    # add sorted genes so they can be removed\n",
    "    curr_counts['sorted_genes'] = ['|'.join(sorted([curr_counts['Gene 1'].iloc[i], curr_counts['Gene 2'].iloc[i]])) for i in range(curr_counts.shape[0])]\n",
    "    curr_counts.drop_duplicates(subset = ['sorted_genes', 'Cell Line'], keep = 'first', inplace = True)\n",
    "\n",
    "    # drop the same genes as well\n",
    "    curr_counts = curr_counts.loc[curr_counts['Gene 1'] != curr_counts['Gene 2']]\n",
    "    curr_counts.reset_index(drop = True, inplace = True)\n",
    "\n",
    "    # proceed to create the GI and return\n",
    "    curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "    curr_GI[\"Gene_A\"] = curr_counts['Gene 1'].values\n",
    "    curr_GI[\"Gene_B\"] = curr_counts['Gene 2'].values\n",
    "    curr_GI[\"Study_Source\"] = [curr_counts['Study'].iloc[0]] * curr_GI.shape[0]\n",
    "    curr_GI[\"Cell_Line\"] = curr_counts['Cell Line'].values\n",
    "    curr_GI = curr_GI.fillna(0)\n",
    "    \n",
    "    return(curr_GI)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "99255e9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_study_for_export(sequence_ref, counts_ref, score_ref, study_controls = None, study_conditions = None, can_control_be_substring = True, remove_unrelated_counts = False):\n",
    "    \n",
    "    ## make sure the columns are within each table, if not return error\n",
    "    sequence_ref_needed_columns = {'sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name'}\n",
    "    \n",
    "    if sequence_ref is not None:\n",
    "        if len(sequence_ref_needed_columns.difference(sequence_ref.columns)) > 0:\n",
    "            print('Error')\n",
    "            print('sequence_ref')\n",
    "            return\n",
    "        # reset index by default\n",
    "        sequence_ref.sort_values('sgRNA_target_name', ignore_index = True, inplace = True)\n",
    "        sequence_ref.reset_index(drop = True, inplace = True)\n",
    "    \n",
    "    counts_ref_needed_columns = {'Guide 1', 'Guide 2', 'Gene 1', 'Gene 2', 'Count Replicates', 'Cell Line', 'Study', 'Condition'}\n",
    "    if counts_ref is not None:\n",
    "        if len(counts_ref_needed_columns.difference(counts_ref.columns)) > 0:\n",
    "            print('Error')\n",
    "            print('counts_ref')\n",
    "            return\n",
    "        # reset index by default\n",
    "        counts_ref.reset_index(drop = True, inplace = True)\n",
    "    \n",
    "    score_ref_needed_columns = {'Gene_A', 'Gene_B', 'Study_Source', 'Cell_Line', 'GI_Score', 'GI_Cutoff', 'Stat_Score', 'Stat_Cutoff'}\n",
    "    if (score_ref is None) and (counts_ref is not None):\n",
    "        print('There are no scores, but there are counts...Generating Placeholder...')\n",
    "        score_ref = create_placeholder_scores(counts_ref.copy(), sequence_ref.copy())\n",
    "    if len(score_ref_needed_columns.difference(score_ref.columns)) > 0:\n",
    "        print('Error')\n",
    "        print('score_ref')\n",
    "        return\n",
    "    # reset index by default\n",
    "    score_ref.reset_index(drop = True, inplace = True)\n",
    "    \n",
    "    \n",
    "    ## prepare each table to be inserted to their respective tables\n",
    "    \n",
    "    print(\"Starting processing...\")\n",
    "    \n",
    "    ################################# first, handle the scores ref\n",
    "    print('Score reference...')\n",
    "    \n",
    "    # fill NA\n",
    "    score_ref = score_ref.fillna(0)\n",
    "    \n",
    "    for col in ['Gene_A', 'Gene_B', 'Cell_Line']:\n",
    "        score_ref[col] = [i.upper() for i in score_ref[col]]\n",
    "    \n",
    "    genes_A = score_ref['Gene_A'].values\n",
    "    genes_B = score_ref['Gene_B'].values\n",
    "    sorted_genes = []\n",
    "    for i in range(score_ref.shape[0]):\n",
    "        sorted_genes.append('_'.join(sorted([genes_A[i], genes_B[i]])))\n",
    "    score_ref[\"Gene Pair\"] = sorted_genes\n",
    "\n",
    "    # remove same ones\n",
    "    score_ref = score_ref.loc[~(score_ref[\"Gene_A\"].values == score_ref[\"Gene_B\"].values)]\n",
    "\n",
    "    # remove controls from SL scores\n",
    "    if study_controls is not None:\n",
    "        control_idx = np.array([False] * score_ref.shape[0])\n",
    "\n",
    "        for curr_control in study_controls:\n",
    "            \n",
    "            if can_control_be_substring:\n",
    "                control_idx = control_idx | np.array([True if curr_control in i else False for i in score_ref[\"Gene_A\"]]) | np.array([True if curr_control in i else False for i in score_ref[\"Gene_B\"]])\n",
    "                \n",
    "            control_idx = control_idx | np.array([True if i in curr_control else False for i in score_ref[\"Gene_A\"]]) | np.array([True if i in curr_control else False for i in score_ref[\"Gene_B\"]])\n",
    "\n",
    "        print('Controls within SL score that are removed: ')\n",
    "        print(control_idx.sum())\n",
    "        print('---')\n",
    "\n",
    "        score_ref = score_ref.loc[~control_idx]\n",
    "        \n",
    "    if (score_ref['Stat_Cutoff'].iloc[0] != 0) and (score_ref['GI_Cutoff'].iloc[0] != 0):\n",
    "        print('Both GI and Stat cutoffs are present...')\n",
    "        score_ref['SL_or_not'] = (score_ref['GI_Score'] <= (score_ref['GI_Cutoff'].iloc[0])) & (score_ref['Stat_Score'] <= (score_ref['Stat_Cutoff'].iloc[0]))\n",
    "    elif score_ref['GI_Cutoff'].iloc[0] != 0:\n",
    "        print('Only GI cutoff is present...')\n",
    "        score_ref['SL_or_not'] = score_ref['GI_Score'] <= score_ref['GI_Cutoff'].iloc[0]\n",
    "    elif score_ref['Stat_Score'].iloc[0] != 0:\n",
    "        print('Only Stat cutoff is present...')\n",
    "        score_ref['SL_or_not'] = score_ref['Stat_Score'] <= score_ref['Stat_Cutoff'].iloc[0]\n",
    "    else:\n",
    "        print('No scores/stats cutoffs are available, possibly generated. Setting all to be NOT SL')\n",
    "        score_ref['SL_or_not'] = [False] * score_ref.shape[0]\n",
    "        \n",
    "    score_ref.loc[score_ref['SL_or_not'], 'SL_or_not'] = 'SL'\n",
    "    score_ref.loc[score_ref['SL_or_not'] != 'SL', 'SL_or_not'] = 'Not SL'\n",
    "    \n",
    "    # rename the columns\n",
    "    score_ref = score_ref.rename(columns = {  'Gene_A': 'gene_1',\n",
    "                                    'Gene_B': 'gene_2',\n",
    "                                    'SL_or_not': 'SL_or_not',\n",
    "                                    'GI_Score': 'SL_score',\n",
    "                                    'Stat_Score': 'statistical_score',\n",
    "                                    'GI_Cutoff': 'SL_score_cutoff',\n",
    "                                    'Stat_Cutoff': 'statistical_score_cutoff',\n",
    "                                    'Cell_Line': 'cell_line_origin',\n",
    "                                    'Study_Source': 'study_origin',\n",
    "                                    'Gene Pair': 'gene_pair'})\n",
    "    \n",
    "    ################################# score ref - DONE\n",
    "    \n",
    "    print('Counts reference...')\n",
    "    \n",
    "    if counts_ref is not None:\n",
    "\n",
    "        for col in ['Guide 1', 'Guide 2', 'Gene 1', 'Gene 2', 'Cell Line']:\n",
    "            counts_ref[col] = [i.upper() for i in counts_ref[col]]\n",
    "\n",
    "\n",
    "        # label whether single, double, or control\n",
    "        sgRNA_true_pair_index = np.array([i for i in range(counts_ref.shape[0]) if (str(counts_ref[\"Gene 1\"].iloc[i]) not in study_controls) and (str(counts_ref[\"Gene 2\"].iloc[i]) not in study_controls) and (str(counts_ref[\"Gene 1\"].iloc[i]) != str(counts_ref[\"Gene 2\"].iloc[i]))])\n",
    "        print(' '.join([\"Number of double pairs:\", str(len(sgRNA_true_pair_index))]))\n",
    "\n",
    "        sgRNA_control_pair_index = np.array([i for i in range(counts_ref.shape[0]) if (str(counts_ref[\"Gene 1\"].iloc[i]) in study_controls) and (str(counts_ref[\"Gene 2\"].iloc[i]) in study_controls)])\n",
    "        print(' '.join([\"Number of controls:\", str(len(sgRNA_control_pair_index))]))\n",
    "\n",
    "        sgRNA_single_gene_index = np.array(sorted(list(set(range(counts_ref.shape[0])).difference(set(np.concatenate((sgRNA_true_pair_index, sgRNA_control_pair_index)))))))\n",
    "        print(' '.join([\"Number of singles:\", str(len(sgRNA_single_gene_index))]))\n",
    "\n",
    "        if (len(sgRNA_single_gene_index) + len(sgRNA_control_pair_index) + len(sgRNA_true_pair_index)) != counts_ref.shape[0]:\n",
    "            print('Missing annotation')\n",
    "            print(counts_ref.shape[0] - (len(sgRNA_single_gene_index) + len(sgRNA_control_pair_index) + len(sgRNA_true_pair_index)))\n",
    "\n",
    "        counts_ref['Target Type'] = 'N/A'\n",
    "        counts_ref['Target Type'].iloc[sgRNA_true_pair_index] = 'Dual'\n",
    "        counts_ref['Target Type'].iloc[sgRNA_control_pair_index] = 'Control'\n",
    "        counts_ref['Target Type'].iloc[sgRNA_single_gene_index] = 'Single'\n",
    "\n",
    "\n",
    "        if 'Type' in counts_ref.columns:\n",
    "            counts_ref = counts_ref.drop(columns = ['Type'])\n",
    "        if 'Sequencing' in counts_ref.columns:\n",
    "            counts_ref = counts_ref.drop(columns = ['Sequencing'])\n",
    "\n",
    "\n",
    "        ## seperate the replicate counts across T0 and TEnd\n",
    "        counts_ref['T0_counts'] = \"\"\n",
    "        counts_ref['T0_replicate_names'] = \"\"\n",
    "        counts_ref['TEnd_counts'] = \"\"\n",
    "        counts_ref['TEnd_replicate_names'] = \"\"\n",
    "\n",
    "        if isinstance(study_conditions, dict):\n",
    "            # for different cell lines within a study\n",
    "\n",
    "            for cell_line in study_conditions:\n",
    "                curr_conditions = study_conditions[cell_line]\n",
    "\n",
    "                # access the cell line counts\n",
    "                access_level = counts_ref.loc[counts_ref['Cell Line'] == cell_line].copy()\n",
    "\n",
    "                ## get all conditions\n",
    "                condition = access_level['Condition'].value_counts().index.tolist()\n",
    "                condition = condition[0].split(';')\n",
    "\n",
    "                # time point T_0\n",
    "                t_0_index = np.array([i for i in range(len(condition)) if condition[i] in curr_conditions[0]])\n",
    "                # time point T_end\n",
    "                t_end_index = np.array([i for i in range(len(condition)) if condition[i] in curr_conditions[1]])\n",
    "\n",
    "                # get counts\n",
    "                replicate_sep =  access_level[\"Count Replicates\"].apply(    \n",
    "                    lambda x: np.array(x.split(\";\"), dtype = np.float64)\n",
    "                )\n",
    "\n",
    "                # get time point \n",
    "                t_0_comb = replicate_sep.apply(\n",
    "                    lambda x: check_repeated_constructs(x, t_0_index)\n",
    "                ).apply(lambda x: ';'.join(x.astype(np.str_)))\n",
    "\n",
    "                t_end_comb = replicate_sep.apply(\n",
    "                    lambda x: check_repeated_constructs(x, t_end_index)#np.median(x[t_end_index])\n",
    "                ).apply(lambda x: ';'.join(x.astype(np.str_)))\n",
    "\n",
    "                access_level['T0_counts'] = t_0_comb\n",
    "                access_level['T0_replicate_names'] = ';'.join(curr_conditions[0])\n",
    "                access_level['TEnd_counts'] = t_end_comb\n",
    "                access_level['TEnd_replicate_names'] = ';'.join(curr_conditions[1])\n",
    "\n",
    "                counts_ref.loc[counts_ref['Cell Line'] == cell_line] = access_level\n",
    "        else:\n",
    "            # for only one cell line\n",
    "            curr_conditions = study_conditions\n",
    "\n",
    "            ## get all conditions\n",
    "            condition = counts_ref['Condition'].value_counts().index.tolist()\n",
    "            condition = condition[0].split(';')\n",
    "\n",
    "            # time point T_0\n",
    "            t_0_index = np.array([i for i in range(len(condition)) if condition[i] in curr_conditions[0]])\n",
    "            # time point T_end\n",
    "            t_end_index = np.array([i for i in range(len(condition)) if condition[i] in curr_conditions[1]])\n",
    "\n",
    "            # get counts\n",
    "            replicate_sep =  counts_ref[\"Count Replicates\"].apply(    \n",
    "                lambda x: np.array(x.split(\";\"), dtype = np.float64)\n",
    "            )\n",
    "\n",
    "            # get time point \n",
    "            t_0_comb = replicate_sep.apply(\n",
    "                lambda x: check_repeated_constructs(x, t_0_index)\n",
    "            ).apply(lambda x: ';'.join(x.astype(np.str_)))\n",
    "\n",
    "            t_end_comb = replicate_sep.apply(\n",
    "                lambda x: check_repeated_constructs(x, t_end_index)#np.median(x[t_end_index])\n",
    "            ).apply(lambda x: ';'.join(x.astype(np.str_)))\n",
    "\n",
    "            counts_ref['T0_counts'] = t_0_comb\n",
    "            counts_ref['T0_replicate_names'] = ';'.join(curr_conditions[0])\n",
    "            counts_ref['TEnd_counts'] = t_end_comb\n",
    "            counts_ref['TEnd_replicate_names'] = ';'.join(curr_conditions[1])\n",
    "\n",
    "\n",
    "        # proceed to add the orientation\n",
    "\n",
    "        unsorted_orientations = np.array(['|'.join([counts_ref[\"Gene 1\"].iloc[i], counts_ref[\"Gene 2\"].iloc[i]]) for i in range(counts_ref.shape[0])])\n",
    "        sorted_orientations = np.array(['|'.join(sorted([counts_ref[\"Gene 1\"].iloc[i], counts_ref[\"Gene 2\"].iloc[i]])) for i in range(counts_ref.shape[0])])\n",
    "\n",
    "        counts_ref['Gene Pair'] = sorted_orientations\n",
    "        counts_ref['Gene Pair Orientation'] = 'A_B'\n",
    "        counts_ref['Gene Pair Orientation'].loc[sorted_orientations != unsorted_orientations] = 'B_A'\n",
    "\n",
    "        # rename the columns\n",
    "        counts_ref = counts_ref.rename(columns = {'Gene Pair Orientation': 'gene_pair_orientation',\n",
    "                                        'Target Type': 'target_type',\n",
    "                                        'Gene Pair': 'gene_pair',\n",
    "                                        'Study': 'study_origin',\n",
    "                                        'Cell Line': 'cell_line_origin'})\n",
    "        \n",
    "        counts_ref_list = []\n",
    "        # applied for HORLBECK\n",
    "        if remove_unrelated_counts:\n",
    "            print('remove_unrelated_counts = TRUE')\n",
    "            counts_ref_list = []\n",
    "            for cl in sorted(list(set(counts_ref['cell_line_origin']))):\n",
    "                print('For cl = ' + cl)\n",
    "                temp_counts = counts_ref.loc[counts_ref['cell_line_origin'] == cl]\n",
    "                temp_scores = score_ref.loc[score_ref['cell_line_origin'] == cl]\n",
    "                \n",
    "                # get score genes + control\n",
    "                score_genes = list(set(temp_scores['gene_1'].tolist() + temp_scores['gene_2'].tolist())) + study_controls\n",
    "                if study_controls is not None:\n",
    "                    score_genes = score_genes + study_controls\n",
    "                # filter down\n",
    "                temp_counts_filt = temp_counts.loc[(temp_counts['Gene 1'].isin(score_genes) & temp_counts['Gene 2'].isin(score_genes)) | (temp_counts['target_type'] == 'Control')]\n",
    "                \n",
    "                removed = temp_counts.shape[0] - temp_counts_filt.shape[0]\n",
    "                if removed != 0:\n",
    "                    print('Removed a total of {rem} sgRNAs...'.format(rem = removed))\n",
    "                else:\n",
    "                    print('No unrelated counts found!')\n",
    "                    \n",
    "                counts_ref_list.append(temp_counts_filt)\n",
    "                \n",
    "            counts_ref = pd.concat(counts_ref_list, axis = 0)\n",
    "            counts_ref.reset_index(drop = True, inplace = True)\n",
    "\n",
    "    ################################# counts ref - DONE\n",
    "    \n",
    "    print('Sequence reference...')\n",
    "    if sequence_ref is not None:\n",
    "        for col in ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']:\n",
    "            sequence_ref[col] = [i.upper() for i in sequence_ref[col]]\n",
    "            \n",
    "        # set the target names to control\n",
    "        control_idx = np.array([True if str(sequence_ref['sgRNA_target_name'].iloc[i]) in study_controls else False for i in range(sequence_ref.shape[0])]) | np.array([True if str(sequence_ref['sgRNA_guide_name'].iloc[i]) in study_controls else False for i in range(sequence_ref.shape[0])])\n",
    "        sequence_ref.loc[control_idx, 'sgRNA_target_name'] = 'CONTROL'    \n",
    "        # add study origin as well\n",
    "        sequence_ref['study_origin'] = [counts_ref['study_origin'].iloc[0]] * sequence_ref.shape[0]\n",
    "        \n",
    "        # remove the sgRNAs that were not used\n",
    "#         all_used_sgRNAs = set(counts_ref['Guide 1'].tolist() + counts_ref['Guide 2'].tolist())\n",
    "#         all_used_sgRNAs = all_used_sgRNAs.intersection(set(sequence_ref['sgRNA_guide_name']))\n",
    "        \n",
    "#         print('Removed non-used sgRNAs:' + str((~(sequence_ref['sgRNA_guide_name'].isin(list(all_used_sgRNAs)))).sum()))\n",
    "        \n",
    "#         sequence_ref = sequence_ref.loc[sequence_ref['sgRNA_guide_name'].isin(list(all_used_sgRNAs))]\n",
    "        \n",
    "        sequence_ref.reset_index(drop=True, inplace = True)\n",
    "    \n",
    "    ################################# sequence ref - DONE\n",
    "    \n",
    "    print('Done! Returning...')\n",
    "    return({'sequence_ref': sequence_ref,\n",
    "            'counts_ref': counts_ref,\n",
    "            'score_ref': score_ref})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c1c6e1a",
   "metadata": {},
   "source": [
    "# temp = sequence_ref.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "646a564d",
   "metadata": {},
   "source": [
    "# study_controls = controls['najm_data']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a98e11b",
   "metadata": {},
   "source": [
    "db_inserts['sequence_ref']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c8ce22b",
   "metadata": {},
   "source": [
    "if sequence_ref is not None:\n",
    "    for col in ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']:\n",
    "        sequence_ref[col] = [i.upper() for i in sequence_ref[col]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9125f10",
   "metadata": {},
   "source": [
    "control_idx = np.array([True if str(sequence_ref['sgRNA_target_name'].iloc[i]) in study_controls else False for i in range(sequence_ref.shape[0])]) | np.array([True if str(sequence_ref['sgRNA_guide_name'].iloc[i]) in study_controls else False for i in range(sequence_ref.shape[0])])\n",
    "sequence_ref.loc[control_idx, 'sgRNA_target_name'] = 'CONTROL'  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ab5753a",
   "metadata": {},
   "source": [
    "study_controls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f09860d8",
   "metadata": {},
   "source": [
    "sequence_ref.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "eb8604a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def insert_study_to_db(engine_link, db_inserts):\n",
    "    \n",
    "    # first, get the metadata\n",
    "    db_metadata = sqlalchemy.MetaData(bind=engine_link)\n",
    "    db_metadata.reflect(engine_link)\n",
    "\n",
    "    # access the tables\n",
    "    sequence_table = db_metadata.tables['CDKO_EXPERIMENT_DESIGN']\n",
    "    counts_table = db_metadata.tables['CDKO_SGRNA_COUNTS']\n",
    "    scores_table = db_metadata.tables['CDKO_ORIGINAL_SL_RESULTS']\n",
    "\n",
    "    # then, start the session\n",
    "    engine_session = sessionmaker(bind=engine_link)\n",
    "    curr_session = engine_session()\n",
    "\n",
    "    # get number of records in each table\n",
    "    sequence_records_num = curr_session.query(sequence_table).count()\n",
    "    counts_records_num = curr_session.query(counts_table).count()\n",
    "    scores_records_num = curr_session.query(scores_table).count()\n",
    "\n",
    "    # get available gene pairs\n",
    "    available_gene_pairs = curr_session.query(sqlalchemy.func.max(counts_table.c.gene_pair_id)).first()[0]\n",
    "\n",
    "    # for the first instance\n",
    "    if available_gene_pairs is None:\n",
    "        available_gene_pairs = -1\n",
    "\n",
    "    # in the case of studies that doesn't have counts, but were added\n",
    "    temp = curr_session.query(sqlalchemy.func.max(scores_table.c.gene_pair_id)).first()[0]\n",
    "\n",
    "    if temp is not None:\n",
    "        available_gene_pairs = max(available_gene_pairs, temp)\n",
    "\n",
    "    # proceed to add the IDs to each table and reindex\n",
    "    if db_inserts['sequence_ref'] is not None:\n",
    "        sequence_insert = db_inserts['sequence_ref'].reset_index(drop=True)\n",
    "    else:\n",
    "        sequence_insert = None\n",
    "\n",
    "    if db_inserts['counts_ref'] is not None:\n",
    "        counts_insert = db_inserts['counts_ref'].reset_index(drop=True)\n",
    "    else:\n",
    "        counts_insert = None\n",
    "    score_insert = db_inserts['score_ref'].reset_index(drop=True)\n",
    "\n",
    "    # add the current number of records for proper insertion\n",
    "    if sequence_insert is not None:\n",
    "        sequence_insert.index += sequence_records_num\n",
    "    if counts_insert is not None:\n",
    "        counts_insert.index += counts_records_num\n",
    "    score_insert.index += scores_records_num\n",
    "\n",
    "    # set IDs\n",
    "    if sequence_insert is not None:\n",
    "        sequence_insert['sgRNA_id'] = sequence_insert.index\n",
    "    if counts_insert is not None:\n",
    "        counts_insert['sgRNA_pair_id'] = counts_insert.index\n",
    "    score_insert['id'] = score_insert.index\n",
    "    score_insert['gene_pair_id'] = score_insert['id'].copy()\n",
    "    \n",
    "    # update the gene pairs\n",
    "    print('Updating gene pairs with seperator |...')\n",
    "    if counts_insert is not None:\n",
    "        counts_insert['gene_pair'] = np.array(['|'.join(sorted([counts_insert[\"Gene 1\"].iloc[i], counts_insert[\"Gene 2\"].iloc[i]])) for i in range(counts_insert.shape[0])])\n",
    "    score_insert['gene_pair'] = np.array(['|'.join(sorted([score_insert[\"gene_1\"].iloc[i], score_insert[\"gene_2\"].iloc[i]])) for i in range(score_insert.shape[0])])\n",
    "\n",
    "    if (sequence_insert is not None) or (counts_insert is not None):\n",
    "        for_merging = sequence_insert.copy()\n",
    "        for_merging['ref_id'] = for_merging.index\n",
    "\n",
    "        # # add the foreign keys\n",
    "        counts_insert['FK_guide_1_id'] = counts_insert.merge(for_merging, how = 'left', left_on = 'Guide 1', right_on = 'sgRNA_guide_name')['ref_id'].values\n",
    "        counts_insert['FK_guide_2_id'] = counts_insert.merge(for_merging, how = 'left', left_on = 'Guide 2', right_on = 'sgRNA_guide_name')['ref_id'].values\n",
    "\n",
    "    if counts_insert is not None:\n",
    "        for_merging = score_insert.copy()\n",
    "\n",
    "        # in the case of multiple cell lines\n",
    "        for_merging['gene_pair+cell_line+study_origin'] = for_merging['gene_pair'] + '+' + for_merging['cell_line_origin'] + '+' + for_merging['study_origin']\n",
    "        counts_insert['gene_pair+cell_line+study_origin'] = counts_insert['gene_pair'] + '+' + counts_insert['cell_line_origin'] + '+' + counts_insert['study_origin']\n",
    "\n",
    "        # set the gene pair id\n",
    "        #counts_insert['gene_pair_id_all'] = np.NaN\n",
    "        # dual have the gene pair id\n",
    "        #counts_insert.loc[counts_insert['target_type'] == 'Dual', 'gene_pair_id_all'] = counts_insert.loc[counts_insert['target_type'] == 'Dual'].groupby(['gene_pair+cell_line+study_origin']).ngroup() + (available_gene_pairs + 1)\n",
    "        counts_insert['gene_pair_id_all'] = counts_insert.groupby(['gene_pair+cell_line+study_origin']).ngroup() + (available_gene_pairs + 1)\n",
    "\n",
    "        score_insert['gene_pair_id'] = for_merging.merge(counts_insert.drop_duplicates(subset = 'gene_pair+cell_line+study_origin'), how = 'left', left_on = 'gene_pair+cell_line+study_origin', right_on = 'gene_pair+cell_line+study_origin')['gene_pair_id_all'].values\n",
    "\n",
    "    ## check if there is any NA in the references\n",
    "    if (sequence_insert is not None) or (counts_insert is not None):\n",
    "        for col in ['FK_guide_1_id', 'FK_guide_2_id']:\n",
    "            if counts_insert[col].isna().sum() > 0:\n",
    "                print('NA in foreign keys: ' + col)\n",
    "    else:\n",
    "        print('No counts and sequences together')\n",
    "\n",
    "    print('Final QC...')\n",
    "    # Final quality control\n",
    "    if sequence_insert is not None:\n",
    "        sequence_insert = sequence_insert.applymap(lambda x: x.strip() if isinstance(x, str) else x, na_action='ignore')\n",
    "    if counts_insert is not None:\n",
    "        counts_insert = counts_insert.applymap(lambda x: x.strip() if isinstance(x, str) else x, na_action='ignore')\n",
    "    score_insert = score_insert.applymap(lambda x: x.strip() if isinstance(x, str) else x, na_action='ignore')\n",
    "\n",
    "    # proceed to insert to the database\n",
    "\n",
    "    # start the transaction, # insert only the columns we need\n",
    "    with engine_link.begin() as transaction:\n",
    "        # insert sequence\n",
    "        print('Beginning transaction...')\n",
    "\n",
    "        if sequence_insert is not None:\n",
    "            sequence_insert = sequence_insert.loc[:,['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name', 'study_origin', 'sgRNA_id']]\n",
    "            sequence_insert.to_sql(name = 'CDKO_EXPERIMENT_DESIGN', con = transaction, if_exists = 'append', index = False, index_label = 'sgRNA_id')\n",
    "\n",
    "            print('Done sequence')\n",
    "\n",
    "        # insert CDKO counts\n",
    "        if counts_insert is not None:\n",
    "            counts_insert = counts_insert.loc[:,['sgRNA_pair_id', 'FK_guide_1_id', 'FK_guide_2_id', 'gene_pair_id_all', 'gene_pair_orientation', 'T0_counts', 'T0_replicate_names', 'TEnd_counts', 'TEnd_replicate_names', 'target_type', 'study_origin', 'cell_line_origin']]\n",
    "\n",
    "            counts_insert = counts_insert.rename(columns = {'FK_guide_1_id': 'guide_1_id',\n",
    "                                                                   'FK_guide_2_id': 'guide_2_id',\n",
    "                                                                   'gene_pair_id_all': 'gene_pair_id'})\n",
    "\n",
    "            counts_insert.to_sql(name = 'CDKO_SGRNA_COUNTS', con = transaction, if_exists = 'append', index = False, index_label = 'sgRNA_pair_id')\n",
    "\n",
    "            print('Done counts')\n",
    "\n",
    "        # finally, insert scores\n",
    "        score_insert = score_insert.loc[:, ['gene_1', 'gene_2', 'study_origin', 'cell_line_origin', 'SL_score', 'SL_score_cutoff', 'statistical_score', 'statistical_score_cutoff', 'gene_pair', 'SL_or_not', 'gene_pair_id', 'id']]\n",
    "        score_insert.to_sql(name = 'CDKO_ORIGINAL_SL_RESULTS', con = transaction, if_exists = 'append', index = False, index_label = 'id')\n",
    "\n",
    "        print('Done score')\n",
    "\n",
    "        print('Successfully inserted!')\n",
    "\n",
    "        print('Added Record stats...')\n",
    "        if sequence_insert is not None:\n",
    "            print(' '.join(['Sequence insert:', str(sequence_insert.shape[0])]))\n",
    "        if counts_insert is not None:\n",
    "            print(' '.join(['Counts insert:', str(counts_insert.shape[0])]))\n",
    "        print(' '.join(['Score insert:', str(score_insert.shape[0])]))\n",
    "\n",
    "    print('Done!')\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac4d479b",
   "metadata": {},
   "source": [
    "guide_id, sequence_id\n",
    "gene1, gene2(control), guide1, guide2, replicates 1::N (sep = ;), \"single\", conditions, cell_line\n",
    "gene1, gene2, guide1, guide2, replicates 1::N (sep=;), \"dual\", conditions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5acccc33",
   "metadata": {},
   "source": [
    "## Diehl Data (HEK293T, RPE1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "f6806d4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "diehl_loc = os.path.join(learning_goals_loc_general, \"Diehl\")\n",
    "\n",
    "#GI is for RPE1\n",
    "atg_flux_gRNA = pd.read_excel(os.path.join(diehl_loc, \"Supplementary Table S5_ATG_Single_MPX_grna_readcounts.xlsx\"), skiprows = 1)\n",
    "atg_flux_gRNA_sequences = pd.ExcelFile(os.path.join(diehl_loc, \"Supplementary Table S1_DNA_oligonucleotides.xlsx\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e6f48e5",
   "metadata": {},
   "source": [
    "### Seq References"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "cbb04571",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = pd.DataFrame(columns = [\"Guide_id\", \"Sequence\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3d2ee64a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>h7SK-1N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGATNACTGAGGCTCAGTTAga...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>h7SK-1N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>h7SK-2N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGATNNCTGAGGCTCAGTTAga...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>h7SK-2N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>h7SK-3N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGANNNCTGAGGCTCAGTTAga...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>h7SK-3N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>h7SK-4N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGANNNNTGAGGCTCAGTTAga...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>h7SK-4N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hU6-1N</td>\n",
       "      <td>gctaTTTCtagctctaaaacCAGTGATGATNACTGAGGCTCgtttc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hU6-1N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hU6-2N</td>\n",
       "      <td>gctaTTTCtagctctaaaacCAGTGATGATNNCTGAGGCTCgtttc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hU6-2N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>hU6-3N</td>\n",
       "      <td>gctaTTTCtagctctaaaacCAGTGATGANNNCTGAGGCTCgtttc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hU6-3N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>hU6-4N</td>\n",
       "      <td>gctaTTTCtagctctaaaacCAGTGATGANNNNTGAGGCTCgtttc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hU6-4N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0                                                  1   2        3\n",
       "0  h7SK-1N  ccagcatagctcttaaacCAGTGATGATNACTGAGGCTCAGTTAga... NaN  h7SK-1N\n",
       "1  h7SK-2N  ccagcatagctcttaaacCAGTGATGATNNCTGAGGCTCAGTTAga... NaN  h7SK-2N\n",
       "2  h7SK-3N  ccagcatagctcttaaacCAGTGATGANNNCTGAGGCTCAGTTAga... NaN  h7SK-3N\n",
       "3  h7SK-4N  ccagcatagctcttaaacCAGTGATGANNNNTGAGGCTCAGTTAga... NaN  h7SK-4N\n",
       "4   hU6-1N  gctaTTTCtagctctaaaacCAGTGATGATNACTGAGGCTCgtttc... NaN   hU6-1N\n",
       "5   hU6-2N  gctaTTTCtagctctaaaacCAGTGATGATNNCTGAGGCTCgtttc... NaN   hU6-2N\n",
       "6   hU6-3N  gctaTTTCtagctctaaaacCAGTGATGANNNCTGAGGCTCgtttc... NaN   hU6-3N\n",
       "7   hU6-4N  gctaTTTCtagctctaaaacCAGTGATGANNNNTGAGGCTCgtttc... NaN   hU6-4N"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA_sequences.parse(\"Sheet1\", header = None).head(8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "e9d0c340",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_seq = atg_flux_gRNA_sequences.parse(\"Sheet1\", header = None).head(8)[[0, 1]]\n",
    "curr_seq.columns = [\"Guide_id\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, curr_seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "f46c4326",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>seuqence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GFP-mpx1#1</td>\n",
       "      <td>CCAGCATAGCTCTTAAACGTGCCCTGGCCCACCCTCGTCAGTTAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GFP-mpx1#2</td>\n",
       "      <td>CCAGCATAGCTCTTAAACGGAGCGCACCATCTTCTTCACAGTTAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GFP-mpx1#3</td>\n",
       "      <td>CCAGCATAGCTCTTAAACTGCCCTGGCCCACCCTCGTGCAGTTAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GFP-mpx1#4</td>\n",
       "      <td>CCAGCATAGCTCTTAAACCGCCCTCGCCGGACACGCTGCAGTTAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GFP-mpx1#5</td>\n",
       "      <td>CCAGCATAGCTCTTAAACGGCCCACCCTCGTGACCACCCAGTTAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>mCherry-mpx1#46</td>\n",
       "      <td>GCTATTTCTAGCTCTAAAACGAGGGCTTCAAGTGGGAGCGCGTTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>mCherry-mpx1#47</td>\n",
       "      <td>GCTATTTCTAGCTCTAAAACCCCAGCCCATGGTCTTCTTCCGTTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>mCherry-mpx1#48</td>\n",
       "      <td>GCTATTTCTAGCTCTAAAACCCGACATCCCCGACTACTTGCGTTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>mCherry-mpx1#49</td>\n",
       "      <td>GCTATTTCTAGCTCTAAAACACAACGTCAACATCAAGCTGCGTTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>mCherry-mpx1#50</td>\n",
       "      <td>GCTATTTCTAGCTCTAAAACTGCCGCCGGTGGAGTGGCGGCGTTTC...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               name                                           seuqence\n",
       "0        GFP-mpx1#1  CCAGCATAGCTCTTAAACGTGCCCTGGCCCACCCTCGTCAGTTAGA...\n",
       "1        GFP-mpx1#2  CCAGCATAGCTCTTAAACGGAGCGCACCATCTTCTTCACAGTTAGA...\n",
       "2        GFP-mpx1#3  CCAGCATAGCTCTTAAACTGCCCTGGCCCACCCTCGTGCAGTTAGA...\n",
       "3        GFP-mpx1#4  CCAGCATAGCTCTTAAACCGCCCTCGCCGGACACGCTGCAGTTAGA...\n",
       "4        GFP-mpx1#5  CCAGCATAGCTCTTAAACGGCCCACCCTCGTGACCACCCAGTTAGA...\n",
       "..              ...                                                ...\n",
       "95  mCherry-mpx1#46  GCTATTTCTAGCTCTAAAACGAGGGCTTCAAGTGGGAGCGCGTTTC...\n",
       "96  mCherry-mpx1#47  GCTATTTCTAGCTCTAAAACCCCAGCCCATGGTCTTCTTCCGTTTC...\n",
       "97  mCherry-mpx1#48  GCTATTTCTAGCTCTAAAACCCGACATCCCCGACTACTTGCGTTTC...\n",
       "98  mCherry-mpx1#49  GCTATTTCTAGCTCTAAAACACAACGTCAACATCAAGCTGCGTTTC...\n",
       "99  mCherry-mpx1#50  GCTATTTCTAGCTCTAAAACTGCCGCCGGTGGAGTGGCGGCGTTTC...\n",
       "\n",
       "[100 rows x 2 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA_sequences.parse(\"3Cs GFP-mCherry library\", header = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "ce7d83b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_seq = atg_flux_gRNA_sequences.parse(\"3Cs GFP-mCherry library\", header = 0)\n",
    "curr_seq.columns = [\"Guide_id\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, curr_seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "ee2a2938",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide</th>\n",
       "      <th>oligo sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCT4-KO-1</td>\n",
       "      <td>gctatttctagctctaaaacTAGTGGGACAATTGATGACTcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCT4-KO-2</td>\n",
       "      <td>gctatttctagctctaaaacCTATTTGATTATCCATCTGTcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCT4-KO-3</td>\n",
       "      <td>gctatttctagctctaaaacCATCAGTAGTCATCATTGCTcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCT4-KO-4</td>\n",
       "      <td>gctatttctagctctaaaacCAGGTCGAGACATGTCAGTCcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>IARS-KO-1</td>\n",
       "      <td>gctatttctagctctaaaacCTTTCCAACTTCGAGTCACAcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>ARNT-KO-4</td>\n",
       "      <td>ccagcatagctcttaaac TGGTCGCCGCTTAATAGCCC cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>BBX-KO-1</td>\n",
       "      <td>ccagcatagctcttaaac TCTTGCGAGGGGTATCTCCA cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>BBX-KO-2</td>\n",
       "      <td>ccagcatagctcttaaac CCATCGGATTCTTTTGCTTT cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>BBX-KO-3</td>\n",
       "      <td>ccagcatagctcttaaac CATGGAATTATGATCATTGA cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>BBX-KO-4</td>\n",
       "      <td>ccagcatagctcttaaac CGGTTATCAAGCCTGGGGTG  cagt...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>160 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          guide                                     oligo sequence\n",
       "1    CCT4-KO-1   gctatttctagctctaaaacTAGTGGGACAATTGATGACTcgtttc...\n",
       "2    CCT4-KO-2   gctatttctagctctaaaacCTATTTGATTATCCATCTGTcgtttc...\n",
       "3    CCT4-KO-3   gctatttctagctctaaaacCATCAGTAGTCATCATTGCTcgtttc...\n",
       "4    CCT4-KO-4   gctatttctagctctaaaacCAGGTCGAGACATGTCAGTCcgtttc...\n",
       "5    IARS-KO-1   gctatttctagctctaaaacCTTTCCAACTTCGAGTCACAcgtttc...\n",
       "..          ...                                                ...\n",
       "156  ARNT-KO-4    ccagcatagctcttaaac TGGTCGCCGCTTAATAGCCC cagtt...\n",
       "157   BBX-KO-1    ccagcatagctcttaaac TCTTGCGAGGGGTATCTCCA cagtt...\n",
       "158   BBX-KO-2    ccagcatagctcttaaac CCATCGGATTCTTTTGCTTT cagtt...\n",
       "159   BBX-KO-3    ccagcatagctcttaaac CATGGAATTATGATCATTGA cagtt...\n",
       "160   BBX-KO-4    ccagcatagctcttaaac CGGTTATCAAGCCTGGGGTG  cagt...\n",
       "\n",
       "[160 rows x 2 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA_sequences.parse(\"3Cs distorted library\", header = 0)[1:161]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "5d78146f",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_seq = atg_flux_gRNA_sequences.parse(\"3Cs distorted library\", header = 0)[1:161]\n",
    "curr_seq.columns = [\"Guide_id\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, curr_seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "9bfd2175",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide</th>\n",
       "      <th>oligo sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>Non-Human-Target-258-KO-1</td>\n",
       "      <td>gctatttctagctctaaaacCAACGGGTTCTCCCGGCTACcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>Non-Human-Target-754-KO-2</td>\n",
       "      <td>gctatttctagctctaaaacGTGCGATGTCGCTTCAACGTcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>Non-Human-Target-340-KO-3</td>\n",
       "      <td>gctatttctagctctaaaacCCCTGTGAAGGAGGCGTAAGcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>Non-Human-Target-196-KO-4</td>\n",
       "      <td>gctatttctagctctaaaacATATTTCGGCAGTTGCAGCAcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>Non-Human-Target-17-KO-5</td>\n",
       "      <td>gctatttctagctctaaaacAAATGCACAGATCGCTGATCcgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>Non-Human-Target-220-KO-152</td>\n",
       "      <td>ccagcatagctcttaaac ATGACATTGCGCGTCTACGG cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>Non-Human-Target-310-KO-153</td>\n",
       "      <td>ccagcatagctcttaaac CCAATGATAAGCCCGAACGG cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>Non-Human-Target-360-KO-154</td>\n",
       "      <td>ccagcatagctcttaaac CCTCGGGCGTAAATACTCAT cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>Non-Human-Target-727-KO-155</td>\n",
       "      <td>ccagcatagctcttaaac GTACACACTTATGCCATCAC cagtt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>Non-Human-Target-179-KO-156</td>\n",
       "      <td>ccagcatagctcttaaac ATAATCCGGAAACGCTCGAC cagtt...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>160 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            guide  \\\n",
       "163    Non-Human-Target-258-KO-1    \n",
       "164    Non-Human-Target-754-KO-2    \n",
       "165    Non-Human-Target-340-KO-3    \n",
       "166    Non-Human-Target-196-KO-4    \n",
       "167     Non-Human-Target-17-KO-5    \n",
       "..                            ...   \n",
       "318  Non-Human-Target-220-KO-152    \n",
       "319  Non-Human-Target-310-KO-153    \n",
       "320  Non-Human-Target-360-KO-154    \n",
       "321  Non-Human-Target-727-KO-155    \n",
       "322  Non-Human-Target-179-KO-156    \n",
       "\n",
       "                                        oligo sequence  \n",
       "163  gctatttctagctctaaaacCAACGGGTTCTCCCGGCTACcgtttc...  \n",
       "164  gctatttctagctctaaaacGTGCGATGTCGCTTCAACGTcgtttc...  \n",
       "165  gctatttctagctctaaaacCCCTGTGAAGGAGGCGTAAGcgtttc...  \n",
       "166  gctatttctagctctaaaacATATTTCGGCAGTTGCAGCAcgtttc...  \n",
       "167  gctatttctagctctaaaacAAATGCACAGATCGCTGATCcgtttc...  \n",
       "..                                                 ...  \n",
       "318   ccagcatagctcttaaac ATGACATTGCGCGTCTACGG cagtt...  \n",
       "319   ccagcatagctcttaaac CCAATGATAAGCCCGAACGG cagtt...  \n",
       "320   ccagcatagctcttaaac CCTCGGGCGTAAATACTCAT cagtt...  \n",
       "321   ccagcatagctcttaaac GTACACACTTATGCCATCAC cagtt...  \n",
       "322   ccagcatagctcttaaac ATAATCCGGAAACGCTCGAC cagtt...  \n",
       "\n",
       "[160 rows x 2 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA_sequences.parse(\"3Cs distorted library\", header = 0)[163:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c82e056b",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_seq = atg_flux_gRNA_sequences.parse(\"3Cs distorted library\", header = 0)[163:]\n",
    "curr_seq.columns = [\"Guide_id\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, curr_seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8bfeacb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>gRNA</th>\n",
       "      <th>homology</th>\n",
       "      <th>homology.1</th>\n",
       "      <th>sum(h7SK:v2)</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>acgtgaggctcccctcaaca</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaacacgtgaggctcccctcaacaCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-2-R</td>\n",
       "      <td>cgttggcgtactccatgaca</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaaccgttggcgtactccatgacaCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-3-R</td>\n",
       "      <td>ctgtcatcgaacgcaccttc</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaacctgtcatcgaacgcaccttcCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-4-R</td>\n",
       "      <td>cgttcttctccgagtgcagg</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaaccgttcttctccgagtgcaggCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3-KO-1-R</td>\n",
       "      <td>ttacctagtagtttcaaata</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaacttacctagtagtttcaaataCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>Non-Human-Target-941-KO-76-R</td>\n",
       "      <td>taacaccataaactttcaca</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaactaacaccataaactttcacaCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>Non-Human-Target-532-KO-77-R</td>\n",
       "      <td>ggtgacgccagcacgtcttc</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaacggtgacgccagcacgtcttcCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>Non-Human-Target-398-KO-78-R</td>\n",
       "      <td>ggtagccacataaacgggcg</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaacggtagccacataaacgggcgCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>Non-Human-Target-798-KO-79-R</td>\n",
       "      <td>cgcaaggacaatccagggta</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaaccgcaaggacaatccagggtaCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>Non-Human-Target-958-KO-80-R</td>\n",
       "      <td>gcttagtcacactcgctgaa</td>\n",
       "      <td>CAGTTAgaggtacccaagc</td>\n",
       "      <td>ccagcatagctcttaaac</td>\n",
       "      <td>ccagcatagctcttaaacgcttagtcacactcgctgaaCAGTTAga...</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>876 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             name                  gRNA             homology  \\\n",
       "0                     AKT1-KO-1-R  acgtgaggctcccctcaaca  CAGTTAgaggtacccaagc   \n",
       "1                     AKT1-KO-2-R  cgttggcgtactccatgaca  CAGTTAgaggtacccaagc   \n",
       "2                     AKT1-KO-3-R  ctgtcatcgaacgcaccttc  CAGTTAgaggtacccaagc   \n",
       "3                     AKT1-KO-4-R  cgttcttctccgagtgcagg  CAGTTAgaggtacccaagc   \n",
       "4                     AKT3-KO-1-R  ttacctagtagtttcaaata  CAGTTAgaggtacccaagc   \n",
       "..                            ...                   ...                  ...   \n",
       "871  Non-Human-Target-941-KO-76-R  taacaccataaactttcaca  CAGTTAgaggtacccaagc   \n",
       "872  Non-Human-Target-532-KO-77-R  ggtgacgccagcacgtcttc  CAGTTAgaggtacccaagc   \n",
       "873  Non-Human-Target-398-KO-78-R  ggtagccacataaacgggcg  CAGTTAgaggtacccaagc   \n",
       "874  Non-Human-Target-798-KO-79-R  cgcaaggacaatccagggta  CAGTTAgaggtacccaagc   \n",
       "875  Non-Human-Target-958-KO-80-R  gcttagtcacactcgctgaa  CAGTTAgaggtacccaagc   \n",
       "\n",
       "             homology.1                                       sum(h7SK:v2)  \\\n",
       "0    ccagcatagctcttaaac  ccagcatagctcttaaacacgtgaggctcccctcaacaCAGTTAga...   \n",
       "1    ccagcatagctcttaaac  ccagcatagctcttaaaccgttggcgtactccatgacaCAGTTAga...   \n",
       "2    ccagcatagctcttaaac  ccagcatagctcttaaacctgtcatcgaacgcaccttcCAGTTAga...   \n",
       "3    ccagcatagctcttaaac  ccagcatagctcttaaaccgttcttctccgagtgcaggCAGTTAga...   \n",
       "4    ccagcatagctcttaaac  ccagcatagctcttaaacttacctagtagtttcaaataCAGTTAga...   \n",
       "..                  ...                                                ...   \n",
       "871  ccagcatagctcttaaac  ccagcatagctcttaaactaacaccataaactttcacaCAGTTAga...   \n",
       "872  ccagcatagctcttaaac  ccagcatagctcttaaacggtgacgccagcacgtcttcCAGTTAga...   \n",
       "873  ccagcatagctcttaaac  ccagcatagctcttaaacggtagccacataaacgggcgCAGTTAga...   \n",
       "874  ccagcatagctcttaaac  ccagcatagctcttaaaccgcaaggacaatccagggtaCAGTTAga...   \n",
       "875  ccagcatagctcttaaac  ccagcatagctcttaaacgcttagtcacactcgctgaaCAGTTAga...   \n",
       "\n",
       "     length  \n",
       "0        57  \n",
       "1        57  \n",
       "2        57  \n",
       "3        57  \n",
       "4        57  \n",
       "..      ...  \n",
       "871      57  \n",
       "872      57  \n",
       "873      57  \n",
       "874      57  \n",
       "875      57  \n",
       "\n",
       "[876 rows x 6 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA_sequences.parse(\"3Cs ExtendedAutophagy library\", header = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cd33503a",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_seq = atg_flux_gRNA_sequences.parse(\"3Cs ExtendedAutophagy library\", header = 0)[[\"name\", \"gRNA\"]]\n",
    "curr_seq.columns = [\"Guide_id\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, curr_seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "5c8c2c46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>gRNAs</th>\n",
       "      <th>homology</th>\n",
       "      <th>homology.1</th>\n",
       "      <th>sum (U6:v1)</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AMBRA1-KO-1-R</td>\n",
       "      <td>ttctaggtatcaccgagaaa</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaacttctaggtatcaccgagaaaCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AMBRA1-KO-2-R</td>\n",
       "      <td>tgagagatactggatcatcc</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaactgagagatactggatcatccCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AMBRA1-KO-3-R</td>\n",
       "      <td>cttggcaggtccccagctcc</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaaccttggcaggtccccagctccCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMBRA1-KO-4-R</td>\n",
       "      <td>ccgtaatatagatattatgg</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaacccgtaatatagatattatggCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATG10-KO-1-R</td>\n",
       "      <td>acgttattgtgcagaattca</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaacacgttattgtgcagaattcaCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>Non-Human-Target-668-KO-22-R</td>\n",
       "      <td>ataaggtctttcggagtgcc</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaacataaggtctttcggagtgccCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>Non-Human-Target-953-KO-23-R</td>\n",
       "      <td>gtaatggcgtgctcacataa</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaacgtaatggcgtgctcacataaCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Non-Human-Target-551-KO-24-R</td>\n",
       "      <td>tcttttatattctagaagtc</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaactcttttatattctagaagtcCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>Non-Human-Target-148-KO-25-R</td>\n",
       "      <td>tacggtcaatgttggcgcct</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaactacggtcaatgttggcgcctCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>Non-Human-Target-995-KO-26-R</td>\n",
       "      <td>caagctatgcgttgaccaaa</td>\n",
       "      <td>gctaTTTCtagctctaaaac</td>\n",
       "      <td>Cgtttcgtcctttccacaa</td>\n",
       "      <td>gctaTTTCtagctctaaaaccaagctatgcgttgaccaaaCgtttc...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>282 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             name                 gRNAs              homology  \\\n",
       "0                   AMBRA1-KO-1-R  ttctaggtatcaccgagaaa  gctaTTTCtagctctaaaac   \n",
       "1                   AMBRA1-KO-2-R  tgagagatactggatcatcc  gctaTTTCtagctctaaaac   \n",
       "2                   AMBRA1-KO-3-R  cttggcaggtccccagctcc  gctaTTTCtagctctaaaac   \n",
       "3                   AMBRA1-KO-4-R  ccgtaatatagatattatgg  gctaTTTCtagctctaaaac   \n",
       "4                    ATG10-KO-1-R  acgttattgtgcagaattca  gctaTTTCtagctctaaaac   \n",
       "..                            ...                   ...                   ...   \n",
       "277  Non-Human-Target-668-KO-22-R  ataaggtctttcggagtgcc  gctaTTTCtagctctaaaac   \n",
       "278  Non-Human-Target-953-KO-23-R  gtaatggcgtgctcacataa  gctaTTTCtagctctaaaac   \n",
       "279  Non-Human-Target-551-KO-24-R  tcttttatattctagaagtc  gctaTTTCtagctctaaaac   \n",
       "280  Non-Human-Target-148-KO-25-R  tacggtcaatgttggcgcct  gctaTTTCtagctctaaaac   \n",
       "281  Non-Human-Target-995-KO-26-R  caagctatgcgttgaccaaa  gctaTTTCtagctctaaaac   \n",
       "\n",
       "              homology.1                                        sum (U6:v1)  \\\n",
       "0    Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaacttctaggtatcaccgagaaaCgtttc...   \n",
       "1    Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaactgagagatactggatcatccCgtttc...   \n",
       "2    Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaaccttggcaggtccccagctccCgtttc...   \n",
       "3    Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaacccgtaatatagatattatggCgtttc...   \n",
       "4    Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaacacgttattgtgcagaattcaCgtttc...   \n",
       "..                   ...                                                ...   \n",
       "277  Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaacataaggtctttcggagtgccCgtttc...   \n",
       "278  Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaacgtaatggcgtgctcacataaCgtttc...   \n",
       "279  Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaactcttttatattctagaagtcCgtttc...   \n",
       "280  Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaactacggtcaatgttggcgcctCgtttc...   \n",
       "281  Cgtttcgtcctttccacaa  gctaTTTCtagctctaaaaccaagctatgcgttgaccaaaCgtttc...   \n",
       "\n",
       "     length  \n",
       "0        59  \n",
       "1        59  \n",
       "2        59  \n",
       "3        59  \n",
       "4        59  \n",
       "..      ...  \n",
       "277      59  \n",
       "278      59  \n",
       "279      59  \n",
       "280      59  \n",
       "281      59  \n",
       "\n",
       "[282 rows x 6 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA_sequences.parse(\"3Cs CoreAutophagy library\", header = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9e8ae3b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_seq = atg_flux_gRNA_sequences.parse(\"3Cs CoreAutophagy library\", header = 0)[[\"name\", \"gRNAs\"]]\n",
    "curr_seq.columns = [\"Guide_id\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, curr_seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "34e43554",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide_id</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>h7SK-1N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGATNACTGAGGCTCAGTTAga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>h7SK-2N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGATNNCTGAGGCTCAGTTAga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>h7SK-3N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGANNNCTGAGGCTCAGTTAga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>h7SK-4N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGANNNNTGAGGCTCAGTTAga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hU6-1N</td>\n",
       "      <td>gctaTTTCtagctctaaaacCAGTGATGATNACTGAGGCTCgtttc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>Non-Human-Target-668-KO-22-R</td>\n",
       "      <td>ataaggtctttcggagtgcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>Non-Human-Target-953-KO-23-R</td>\n",
       "      <td>gtaatggcgtgctcacataa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Non-Human-Target-551-KO-24-R</td>\n",
       "      <td>tcttttatattctagaagtc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>Non-Human-Target-148-KO-25-R</td>\n",
       "      <td>tacggtcaatgttggcgcct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>Non-Human-Target-995-KO-26-R</td>\n",
       "      <td>caagctatgcgttgaccaaa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1586 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Guide_id  \\\n",
       "0                         h7SK-1N   \n",
       "1                         h7SK-2N   \n",
       "2                         h7SK-3N   \n",
       "3                         h7SK-4N   \n",
       "4                          hU6-1N   \n",
       "..                            ...   \n",
       "277  Non-Human-Target-668-KO-22-R   \n",
       "278  Non-Human-Target-953-KO-23-R   \n",
       "279  Non-Human-Target-551-KO-24-R   \n",
       "280  Non-Human-Target-148-KO-25-R   \n",
       "281  Non-Human-Target-995-KO-26-R   \n",
       "\n",
       "                                              Sequence  \n",
       "0    ccagcatagctcttaaacCAGTGATGATNACTGAGGCTCAGTTAga...  \n",
       "1    ccagcatagctcttaaacCAGTGATGATNNCTGAGGCTCAGTTAga...  \n",
       "2    ccagcatagctcttaaacCAGTGATGANNNCTGAGGCTCAGTTAga...  \n",
       "3    ccagcatagctcttaaacCAGTGATGANNNNTGAGGCTCAGTTAga...  \n",
       "4    gctaTTTCtagctctaaaacCAGTGATGATNACTGAGGCTCgtttc...  \n",
       "..                                                 ...  \n",
       "277                               ataaggtctttcggagtgcc  \n",
       "278                               gtaatggcgtgctcacataa  \n",
       "279                               tcttttatattctagaagtc  \n",
       "280                               tacggtcaatgttggcgcct  \n",
       "281                               caagctatgcgttgaccaaa  \n",
       "\n",
       "[1586 rows x 2 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "2df75d80",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_targets = np.array([i.split('-')[0] for i in sequence_ref['Guide_id']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "4826f677",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_targets = sgRNA_targets.astype('<U14')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "9520c5a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_targets[sgRNA_targets == 'Non'] = 'Control'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "574438b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['Guide_target'] = sgRNA_targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "574a2ad9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Control    266\n",
       "GFP         50\n",
       "mCherry     50\n",
       "KEAP1       12\n",
       "TP53        12\n",
       "          ... \n",
       "NEDD4        4\n",
       "NEDD4L       4\n",
       "hU6          4\n",
       "PEX13        4\n",
       "VMP1         4\n",
       "Name: Guide_target, Length: 223, dtype: int64"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref['Guide_target'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "64c29d56",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "77598086",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = pd.concat([sequence_ref, pd.DataFrame({'sgRNA_guide_name' : ['wildtype-I-CeuI', 'wildtype-I-SceI'],\n",
    "                     'sgRNA_guide_seq' : [\"ctcTAACTATAACGGTCCTAAGGTAGCGAaacTCGCTACCTTAGGACCGTTATAGTTA\", 'cacctagggataacagggtaatgttttagagctaGAAAtagcaagttaaaataaggctagtccgttatcaacttgaaaaagtggcaccgagtcggtgcTTTTTTggatccAAAAAAgcaccgactcggtgccactttttcaagttgataacggactagccttattttaacttgctaTTTCtagctctaaaacattaccctgttatcccta'],\n",
    "                     'sgRNA_target_name' : ['Control', 'Control']})])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "e8706fb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sequence_ref.drop_duplicates(subset = ['sgRNA_guide_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "f3f4dd79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>h7SK-1N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGATNACTGAGGCTCAGTTAga...</td>\n",
       "      <td>h7SK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>h7SK-2N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGATNNCTGAGGCTCAGTTAga...</td>\n",
       "      <td>h7SK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>h7SK-3N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGANNNCTGAGGCTCAGTTAga...</td>\n",
       "      <td>h7SK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>h7SK-4N</td>\n",
       "      <td>ccagcatagctcttaaacCAGTGATGANNNNTGAGGCTCAGTTAga...</td>\n",
       "      <td>h7SK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hU6-1N</td>\n",
       "      <td>gctaTTTCtagctctaaaacCAGTGATGATNACTGAGGCTCgtttc...</td>\n",
       "      <td>hU6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Non-Human-Target-551-KO-24-R</td>\n",
       "      <td>tcttttatattctagaagtc</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>Non-Human-Target-148-KO-25-R</td>\n",
       "      <td>tacggtcaatgttggcgcct</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>Non-Human-Target-995-KO-26-R</td>\n",
       "      <td>caagctatgcgttgaccaaa</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ctcTAACTATAACGGTCCTAAGGTAGCGAaacTCGCTACCTTAGGA...</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>cacctagggataacagggtaatgttttagagctaGAAAtagcaagt...</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1176 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sgRNA_guide_name  \\\n",
       "0                         h7SK-1N   \n",
       "1                         h7SK-2N   \n",
       "2                         h7SK-3N   \n",
       "3                         h7SK-4N   \n",
       "4                          hU6-1N   \n",
       "..                            ...   \n",
       "279  Non-Human-Target-551-KO-24-R   \n",
       "280  Non-Human-Target-148-KO-25-R   \n",
       "281  Non-Human-Target-995-KO-26-R   \n",
       "0                 wildtype-I-CeuI   \n",
       "1                 wildtype-I-SceI   \n",
       "\n",
       "                                       sgRNA_guide_seq sgRNA_target_name  \n",
       "0    ccagcatagctcttaaacCAGTGATGATNACTGAGGCTCAGTTAga...              h7SK  \n",
       "1    ccagcatagctcttaaacCAGTGATGATNNCTGAGGCTCAGTTAga...              h7SK  \n",
       "2    ccagcatagctcttaaacCAGTGATGANNNCTGAGGCTCAGTTAga...              h7SK  \n",
       "3    ccagcatagctcttaaacCAGTGATGANNNNTGAGGCTCAGTTAga...              h7SK  \n",
       "4    gctaTTTCtagctctaaaacCAGTGATGATNACTGAGGCTCgtttc...               hU6  \n",
       "..                                                 ...               ...  \n",
       "279                               tcttttatattctagaagtc           Control  \n",
       "280                               tacggtcaatgttggcgcct           Control  \n",
       "281                               caagctatgcgttgaccaaa           Control  \n",
       "0    ctcTAACTATAACGGTCCTAAGGTAGCGAaacTCGCTACCTTAGGA...           Control  \n",
       "1    cacctagggataacagggtaatgttttagagctaGAAAtagcaagt...           Control  \n",
       "\n",
       "[1176 rows x 3 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8a015d4",
   "metadata": {},
   "source": [
    "## Counts\n",
    "### Single sgRNAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "da12880f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide</th>\n",
       "      <th>Gene</th>\n",
       "      <th>library</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Guide1</th>\n",
       "      <th>Guide2</th>\n",
       "      <th>Gene_1</th>\n",
       "      <th>Gene_2</th>\n",
       "      <th>library.1</th>\n",
       "      <th>ProlifCTRL_1</th>\n",
       "      <th>ProlifCTRL_2</th>\n",
       "      <th>n1.1</th>\n",
       "      <th>n2.1</th>\n",
       "      <th>n3</th>\n",
       "      <th>Highgate_1</th>\n",
       "      <th>Highgate_2</th>\n",
       "      <th>Highgate_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>900.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>94</td>\n",
       "      <td>906</td>\n",
       "      <td>869</td>\n",
       "      <td>6</td>\n",
       "      <td>195</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>101</td>\n",
       "      <td>1140</td>\n",
       "      <td>1069</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>788.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>90</td>\n",
       "      <td>703</td>\n",
       "      <td>583</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>59</td>\n",
       "      <td>698</td>\n",
       "      <td>656</td>\n",
       "      <td>25</td>\n",
       "      <td>19</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3-KO-1-R</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>471.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>ATG10-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>ATG10</td>\n",
       "      <td>13</td>\n",
       "      <td>721</td>\n",
       "      <td>660</td>\n",
       "      <td>161</td>\n",
       "      <td>75</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248186</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-1-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>917</td>\n",
       "      <td>6290</td>\n",
       "      <td>6245</td>\n",
       "      <td>133</td>\n",
       "      <td>527</td>\n",
       "      <td>33</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248187</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-2-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>3271</td>\n",
       "      <td>4317</td>\n",
       "      <td>4129</td>\n",
       "      <td>268</td>\n",
       "      <td>237</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248188</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-3-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>4500</td>\n",
       "      <td>8497</td>\n",
       "      <td>8366</td>\n",
       "      <td>677</td>\n",
       "      <td>194</td>\n",
       "      <td>31</td>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248189</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-4-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>3800</td>\n",
       "      <td>3505</td>\n",
       "      <td>3541</td>\n",
       "      <td>62</td>\n",
       "      <td>124</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248190</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>1253</td>\n",
       "      <td>9485</td>\n",
       "      <td>14159</td>\n",
       "      <td>258</td>\n",
       "      <td>377</td>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>37</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>248191 rows  19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Guide  Gene  library    n1    n2  Unnamed: 5           Guide1  \\\n",
       "0       AKT1-KO-1-R  AKT1    900.0  99.0  34.0         NaN      AKT1-KO-1-R   \n",
       "1       AKT1-KO-2-R  AKT1   1223.0  54.0  44.0         NaN      AKT1-KO-1-R   \n",
       "2       AKT1-KO-3-R  AKT1    788.0  46.0  66.0         NaN      AKT1-KO-1-R   \n",
       "3       AKT1-KO-4-R  AKT1   1221.0  24.0  57.0         NaN      AKT1-KO-1-R   \n",
       "4       AKT3-KO-1-R  AKT3    471.0  18.0   5.0         NaN      AKT1-KO-1-R   \n",
       "...             ...   ...      ...   ...   ...         ...              ...   \n",
       "248186          NaN   NaN      NaN   NaN   NaN         NaN  wildtype-I-CeuI   \n",
       "248187          NaN   NaN      NaN   NaN   NaN         NaN  wildtype-I-CeuI   \n",
       "248188          NaN   NaN      NaN   NaN   NaN         NaN  wildtype-I-CeuI   \n",
       "248189          NaN   NaN      NaN   NaN   NaN         NaN  wildtype-I-CeuI   \n",
       "248190          NaN   NaN      NaN   NaN   NaN         NaN  wildtype-I-CeuI   \n",
       "\n",
       "                 Guide2           Gene_1           Gene_2  library.1  \\\n",
       "0         AMBRA1-KO-1-R             AKT1           AMBRA1         94   \n",
       "1         AMBRA1-KO-2-R             AKT1           AMBRA1        101   \n",
       "2         AMBRA1-KO-3-R             AKT1           AMBRA1         90   \n",
       "3         AMBRA1-KO-4-R             AKT1           AMBRA1         59   \n",
       "4          ATG10-KO-1-R             AKT1            ATG10         13   \n",
       "...                 ...              ...              ...        ...   \n",
       "248186   ZKSCAN3-KO-1-R  wildtype-I-CeuI          ZKSCAN3        917   \n",
       "248187   ZKSCAN3-KO-2-R  wildtype-I-CeuI          ZKSCAN3       3271   \n",
       "248188   ZKSCAN3-KO-3-R  wildtype-I-CeuI          ZKSCAN3       4500   \n",
       "248189   ZKSCAN3-KO-4-R  wildtype-I-CeuI          ZKSCAN3       3800   \n",
       "248190  wildtype-I-SceI  wildtype-I-CeuI  wildtype-I-SceI       1253   \n",
       "\n",
       "        ProlifCTRL_1  ProlifCTRL_2  n1.1  n2.1  n3  Highgate_1  Highgate_2  \\\n",
       "0                906           869     6   195   0           0           2   \n",
       "1               1140          1069     3     5   0           0           0   \n",
       "2                703           583    14    13  19           0           1   \n",
       "3                698           656    25    19  21           0           4   \n",
       "4                721           660   161    75  21           0           1   \n",
       "...              ...           ...   ...   ...  ..         ...         ...   \n",
       "248186          6290          6245   133   527  33           4           9   \n",
       "248187          4317          4129   268   237  15           1          16   \n",
       "248188          8497          8366   677   194  31          19          17   \n",
       "248189          3505          3541    62   124   9           3           2   \n",
       "248190          9485         14159   258   377  19           1          37   \n",
       "\n",
       "        Highgate_3  \n",
       "0                0  \n",
       "1                0  \n",
       "2                0  \n",
       "3                0  \n",
       "4                0  \n",
       "...            ...  \n",
       "248186           0  \n",
       "248187           0  \n",
       "248188           0  \n",
       "248189           0  \n",
       "248190         183  \n",
       "\n",
       "[248191 rows x 19 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "06cd4a10",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "5b317c51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide</th>\n",
       "      <th>Gene</th>\n",
       "      <th>library</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>900.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>788.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>66.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>57.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3-KO-1-R</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>471.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>ZKSCAN3-KO-1-R</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>725.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>ZKSCAN3-KO-2-R</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>1001.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>ZKSCAN3-KO-3-R</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>1808.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>ZKSCAN3-KO-4-R</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>340.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>4157.0</td>\n",
       "      <td>720.0</td>\n",
       "      <td>1754.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>877 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Guide             Gene  library     n1      n2\n",
       "0        AKT1-KO-1-R             AKT1    900.0   99.0    34.0\n",
       "1        AKT1-KO-2-R             AKT1   1223.0   54.0    44.0\n",
       "2        AKT1-KO-3-R             AKT1    788.0   46.0    66.0\n",
       "3        AKT1-KO-4-R             AKT1   1221.0   24.0    57.0\n",
       "4        AKT3-KO-1-R             AKT3    471.0   18.0     5.0\n",
       "..               ...              ...      ...    ...     ...\n",
       "872   ZKSCAN3-KO-1-R          ZKSCAN3    725.0    5.0     6.0\n",
       "873   ZKSCAN3-KO-2-R          ZKSCAN3   1001.0   33.0    46.0\n",
       "874   ZKSCAN3-KO-3-R          ZKSCAN3   1808.0   27.0    20.0\n",
       "875   ZKSCAN3-KO-4-R          ZKSCAN3    340.0    5.0     4.0\n",
       "876  wildtype-I-CeuI  wildtype-I-CeuI   4157.0  720.0  1754.0\n",
       "\n",
       "[877 rows x 5 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_gRNA.iloc[:, 0:5].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "5ef096e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_sgRNA = atg_flux_gRNA.iloc[:, 0:5].dropna()\n",
    "curr_sgRNA = curr_sgRNA.drop(\"library\", axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "1e29cd43",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_sgRNA['Count Replicates'] = curr_sgRNA[[\"n1\", \"n2\"]].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "curr_sgRNA = curr_sgRNA.drop([\"n1\", \"n2\"], axis = 1)\n",
    "curr_sgRNA.columns = [\"Guide 1\", \"Gene 1\", \"Count Replicates\"]\n",
    "curr_sgRNA[\"Type\"] = [\"Single\"] * curr_sgRNA.shape[0]\n",
    "curr_sgRNA[\"Sequencing\"] = [\"sgRNA\"] * curr_sgRNA.shape[0]\n",
    "curr_sgRNA[\"Condition\"] = [\"rep1;rep2\"] * curr_sgRNA.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1ea2f126",
   "metadata": {},
   "outputs": [],
   "source": [
    "#skip singles\n",
    "#sequence_ref = pd.concat([sequence_ref, curr_sgRNA])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e9f1c42",
   "metadata": {},
   "source": [
    "### Dual sgRNAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20c29aff",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_sgRNA = atg_flux_gRNA.iloc[:, 6:]\n",
    "curr_sgRNA = curr_sgRNA.drop([\"library.1\", \"Highgate_1\", \"Highgate_2\", \"Highgate_3\"], axis = 1)#\"ProlifCTRL_1\", \"ProlifCTRL_2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "81117123",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide1</th>\n",
       "      <th>Guide2</th>\n",
       "      <th>Gene_1</th>\n",
       "      <th>Gene_2</th>\n",
       "      <th>ProlifCTRL_1</th>\n",
       "      <th>ProlifCTRL_2</th>\n",
       "      <th>n1.1</th>\n",
       "      <th>n2.1</th>\n",
       "      <th>n3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>906</td>\n",
       "      <td>869</td>\n",
       "      <td>6</td>\n",
       "      <td>195</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>1140</td>\n",
       "      <td>1069</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>703</td>\n",
       "      <td>583</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>698</td>\n",
       "      <td>656</td>\n",
       "      <td>25</td>\n",
       "      <td>19</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>ATG10-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>ATG10</td>\n",
       "      <td>721</td>\n",
       "      <td>660</td>\n",
       "      <td>161</td>\n",
       "      <td>75</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248186</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-1-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>6290</td>\n",
       "      <td>6245</td>\n",
       "      <td>133</td>\n",
       "      <td>527</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248187</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-2-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>4317</td>\n",
       "      <td>4129</td>\n",
       "      <td>268</td>\n",
       "      <td>237</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248188</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-3-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>8497</td>\n",
       "      <td>8366</td>\n",
       "      <td>677</td>\n",
       "      <td>194</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248189</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-4-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>3505</td>\n",
       "      <td>3541</td>\n",
       "      <td>62</td>\n",
       "      <td>124</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248190</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>9485</td>\n",
       "      <td>14159</td>\n",
       "      <td>258</td>\n",
       "      <td>377</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>248191 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Guide1           Guide2           Gene_1           Gene_2  \\\n",
       "0           AKT1-KO-1-R    AMBRA1-KO-1-R             AKT1           AMBRA1   \n",
       "1           AKT1-KO-1-R    AMBRA1-KO-2-R             AKT1           AMBRA1   \n",
       "2           AKT1-KO-1-R    AMBRA1-KO-3-R             AKT1           AMBRA1   \n",
       "3           AKT1-KO-1-R    AMBRA1-KO-4-R             AKT1           AMBRA1   \n",
       "4           AKT1-KO-1-R     ATG10-KO-1-R             AKT1            ATG10   \n",
       "...                 ...              ...              ...              ...   \n",
       "248186  wildtype-I-CeuI   ZKSCAN3-KO-1-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248187  wildtype-I-CeuI   ZKSCAN3-KO-2-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248188  wildtype-I-CeuI   ZKSCAN3-KO-3-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248189  wildtype-I-CeuI   ZKSCAN3-KO-4-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248190  wildtype-I-CeuI  wildtype-I-SceI  wildtype-I-CeuI  wildtype-I-SceI   \n",
       "\n",
       "        ProlifCTRL_1  ProlifCTRL_2  n1.1  n2.1  n3  \n",
       "0                906           869     6   195   0  \n",
       "1               1140          1069     3     5   0  \n",
       "2                703           583    14    13  19  \n",
       "3                698           656    25    19  21  \n",
       "4                721           660   161    75  21  \n",
       "...              ...           ...   ...   ...  ..  \n",
       "248186          6290          6245   133   527  33  \n",
       "248187          4317          4129   268   237  15  \n",
       "248188          8497          8366   677   194  31  \n",
       "248189          3505          3541    62   124   9  \n",
       "248190          9485         14159   258   377  19  \n",
       "\n",
       "[248191 rows x 9 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_sgRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "3f9b80e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_sgRNA['Count Replicates'] = curr_sgRNA[[\"ProlifCTRL_1\", \"ProlifCTRL_2\", \"n1.1\", \"n2.1\", \"n3\"]].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "curr_sgRNA = curr_sgRNA.drop([\"ProlifCTRL_1\", \"ProlifCTRL_2\", \"n1.1\", \"n2.1\", \"n3\"], axis = 1)\n",
    "\n",
    "curr_sgRNA.columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\"]\n",
    "curr_sgRNA[\"Type\"] = [\"Dual\"] * curr_sgRNA.shape[0]\n",
    "curr_sgRNA[\"Sequencing\"] = [\"sgRNA\"] * curr_sgRNA.shape[0]\n",
    "curr_sgRNA[\"Condition\"] = [\"ctrl_1;ctrl_2;rep_1;rep_2;rep_3\"] * curr_sgRNA.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "f8bfab75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>906;869;6;195;0</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>1140;1069;3;5;0</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>703;583;14;13;19</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>698;656;25;19;21</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>ATG10-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>ATG10</td>\n",
       "      <td>721;660;161;75;21</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248186</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-1-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>6290;6245;133;527;33</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248187</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-2-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>4317;4129;268;237;15</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248188</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-3-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>8497;8366;677;194;31</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248189</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-4-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>3505;3541;62;124;9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248190</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>9485;14159;258;377;19</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>248191 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Guide 1          Guide 2           Gene 1           Gene 2  \\\n",
       "0           AKT1-KO-1-R    AMBRA1-KO-1-R             AKT1           AMBRA1   \n",
       "1           AKT1-KO-1-R    AMBRA1-KO-2-R             AKT1           AMBRA1   \n",
       "2           AKT1-KO-1-R    AMBRA1-KO-3-R             AKT1           AMBRA1   \n",
       "3           AKT1-KO-1-R    AMBRA1-KO-4-R             AKT1           AMBRA1   \n",
       "4           AKT1-KO-1-R     ATG10-KO-1-R             AKT1            ATG10   \n",
       "...                 ...              ...              ...              ...   \n",
       "248186  wildtype-I-CeuI   ZKSCAN3-KO-1-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248187  wildtype-I-CeuI   ZKSCAN3-KO-2-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248188  wildtype-I-CeuI   ZKSCAN3-KO-3-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248189  wildtype-I-CeuI   ZKSCAN3-KO-4-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248190  wildtype-I-CeuI  wildtype-I-SceI  wildtype-I-CeuI  wildtype-I-SceI   \n",
       "\n",
       "             Count Replicates  Type Sequencing  \\\n",
       "0             906;869;6;195;0  Dual      sgRNA   \n",
       "1             1140;1069;3;5;0  Dual      sgRNA   \n",
       "2            703;583;14;13;19  Dual      sgRNA   \n",
       "3            698;656;25;19;21  Dual      sgRNA   \n",
       "4           721;660;161;75;21  Dual      sgRNA   \n",
       "...                       ...   ...        ...   \n",
       "248186   6290;6245;133;527;33  Dual      sgRNA   \n",
       "248187   4317;4129;268;237;15  Dual      sgRNA   \n",
       "248188   8497;8366;677;194;31  Dual      sgRNA   \n",
       "248189     3505;3541;62;124;9  Dual      sgRNA   \n",
       "248190  9485;14159;258;377;19  Dual      sgRNA   \n",
       "\n",
       "                              Condition  \n",
       "0       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "1       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "2       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "3       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "4       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "...                                 ...  \n",
       "248186  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "248187  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "248188  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "248189  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "248190  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  \n",
       "\n",
       "[248191 rows x 8 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_sgRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "9d5b2f9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, curr_sgRNA])\n",
    "counts_ref[\"Cell Line\"] = [\"RPE1\"] * counts_ref.shape[0]\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['diehl_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "2efb15a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>906;869;6;195;0</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>1140;1069;3;5;0</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>703;583;14;13;19</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>698;656;25;19;21</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>ATG10-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>ATG10</td>\n",
       "      <td>721;660;161;75;21</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248186</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-1-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>6290;6245;133;527;33</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248187</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-2-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>4317;4129;268;237;15</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248188</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-3-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>8497;8366;677;194;31</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248189</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-4-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>3505;3541;62;124;9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248190</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>9485;14159;258;377;19</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>248191 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Guide 1          Guide 2           Gene 1           Gene 2  \\\n",
       "0           AKT1-KO-1-R    AMBRA1-KO-1-R             AKT1           AMBRA1   \n",
       "1           AKT1-KO-1-R    AMBRA1-KO-2-R             AKT1           AMBRA1   \n",
       "2           AKT1-KO-1-R    AMBRA1-KO-3-R             AKT1           AMBRA1   \n",
       "3           AKT1-KO-1-R    AMBRA1-KO-4-R             AKT1           AMBRA1   \n",
       "4           AKT1-KO-1-R     ATG10-KO-1-R             AKT1            ATG10   \n",
       "...                 ...              ...              ...              ...   \n",
       "248186  wildtype-I-CeuI   ZKSCAN3-KO-1-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248187  wildtype-I-CeuI   ZKSCAN3-KO-2-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248188  wildtype-I-CeuI   ZKSCAN3-KO-3-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248189  wildtype-I-CeuI   ZKSCAN3-KO-4-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248190  wildtype-I-CeuI  wildtype-I-SceI  wildtype-I-CeuI  wildtype-I-SceI   \n",
       "\n",
       "             Count Replicates  Type Sequencing Cell Line  \\\n",
       "0             906;869;6;195;0  Dual      sgRNA      RPE1   \n",
       "1             1140;1069;3;5;0  Dual      sgRNA      RPE1   \n",
       "2            703;583;14;13;19  Dual      sgRNA      RPE1   \n",
       "3            698;656;25;19;21  Dual      sgRNA      RPE1   \n",
       "4           721;660;161;75;21  Dual      sgRNA      RPE1   \n",
       "...                       ...   ...        ...       ...   \n",
       "248186   6290;6245;133;527;33  Dual      sgRNA      RPE1   \n",
       "248187   4317;4129;268;237;15  Dual      sgRNA      RPE1   \n",
       "248188   8497;8366;677;194;31  Dual      sgRNA      RPE1   \n",
       "248189     3505;3541;62;124;9  Dual      sgRNA      RPE1   \n",
       "248190  9485;14159;258;377;19  Dual      sgRNA      RPE1   \n",
       "\n",
       "                              Condition     Study  \n",
       "0       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "1       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "2       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "3       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "4       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "...                                 ...       ...  \n",
       "248186  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248187  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248188  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248189  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248190  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "\n",
       "[248191 rows x 10 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94667013",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "d421bfe3",
   "metadata": {},
   "outputs": [],
   "source": [
    "diehl_loc = os.path.join(learning_goals_loc_general, \"Diehl\")\n",
    "\n",
    "#GI is for RPE1\n",
    "atg_flux_GI = pd.read_excel(os.path.join(diehl_loc, \"Supplementary Table S6_ATG_GI_models_genes.xlsx\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "30ef3a84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene1</th>\n",
       "      <th>Gene_2</th>\n",
       "      <th>prolif_ctrl_avg_norm</th>\n",
       "      <th>post_facs_avg_norm</th>\n",
       "      <th>lfc</th>\n",
       "      <th>prolif_ctrl_avg_norm_h7SK</th>\n",
       "      <th>post_facs_avg_norm_h7SK</th>\n",
       "      <th>lfc_h7SK</th>\n",
       "      <th>prolif_ctrl_avg_norm_U6</th>\n",
       "      <th>post_facs_avg_norm_U6</th>\n",
       "      <th>...</th>\n",
       "      <th>max_model</th>\n",
       "      <th>max_dLFC</th>\n",
       "      <th>sum_model</th>\n",
       "      <th>sum_dLFC</th>\n",
       "      <th>min_model</th>\n",
       "      <th>min_dLFC</th>\n",
       "      <th>mult_model</th>\n",
       "      <th>mult_dLFC</th>\n",
       "      <th>log_model</th>\n",
       "      <th>log_dLFC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>49.509296</td>\n",
       "      <td>49.133852</td>\n",
       "      <td>-0.010982</td>\n",
       "      <td>66.726643</td>\n",
       "      <td>10.566977</td>\n",
       "      <td>-2.658700</td>\n",
       "      <td>40.667392</td>\n",
       "      <td>34.655258</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.195852</td>\n",
       "      <td>0.184870</td>\n",
       "      <td>-2.854552</td>\n",
       "      <td>2.843570</td>\n",
       "      <td>-2.658700</td>\n",
       "      <td>2.647718</td>\n",
       "      <td>0.520711</td>\n",
       "      <td>-0.531693</td>\n",
       "      <td>0.146446</td>\n",
       "      <td>-0.157428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>15.129326</td>\n",
       "      <td>67.584450</td>\n",
       "      <td>2.159344</td>\n",
       "      <td>20.007618</td>\n",
       "      <td>8.004394</td>\n",
       "      <td>-1.483954</td>\n",
       "      <td>40.667392</td>\n",
       "      <td>34.655258</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.195852</td>\n",
       "      <td>2.355195</td>\n",
       "      <td>-1.679806</td>\n",
       "      <td>3.839149</td>\n",
       "      <td>-1.483954</td>\n",
       "      <td>3.643298</td>\n",
       "      <td>0.290635</td>\n",
       "      <td>1.868709</td>\n",
       "      <td>0.113113</td>\n",
       "      <td>2.046230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>35.618412</td>\n",
       "      <td>84.113111</td>\n",
       "      <td>1.239708</td>\n",
       "      <td>36.815835</td>\n",
       "      <td>23.038215</td>\n",
       "      <td>-0.731591</td>\n",
       "      <td>40.667392</td>\n",
       "      <td>34.655258</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.195852</td>\n",
       "      <td>1.435559</td>\n",
       "      <td>-0.927443</td>\n",
       "      <td>2.167151</td>\n",
       "      <td>-0.731591</td>\n",
       "      <td>1.971299</td>\n",
       "      <td>0.143283</td>\n",
       "      <td>1.096424</td>\n",
       "      <td>0.071066</td>\n",
       "      <td>1.168641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARF6</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>31.232640</td>\n",
       "      <td>21.116277</td>\n",
       "      <td>-0.564699</td>\n",
       "      <td>44.586703</td>\n",
       "      <td>5.954327</td>\n",
       "      <td>-2.645284</td>\n",
       "      <td>40.667392</td>\n",
       "      <td>34.655258</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.195852</td>\n",
       "      <td>-0.368847</td>\n",
       "      <td>-2.841135</td>\n",
       "      <td>2.276436</td>\n",
       "      <td>-2.645284</td>\n",
       "      <td>2.080585</td>\n",
       "      <td>0.518083</td>\n",
       "      <td>-1.082782</td>\n",
       "      <td>0.146202</td>\n",
       "      <td>-0.710901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATF4</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>30.673669</td>\n",
       "      <td>61.306122</td>\n",
       "      <td>0.999030</td>\n",
       "      <td>43.737172</td>\n",
       "      <td>10.396138</td>\n",
       "      <td>-1.827212</td>\n",
       "      <td>40.667392</td>\n",
       "      <td>34.655258</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.195852</td>\n",
       "      <td>1.194882</td>\n",
       "      <td>-2.023064</td>\n",
       "      <td>3.022094</td>\n",
       "      <td>-1.827212</td>\n",
       "      <td>2.826242</td>\n",
       "      <td>0.357863</td>\n",
       "      <td>0.641168</td>\n",
       "      <td>0.125875</td>\n",
       "      <td>0.873156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25475</th>\n",
       "      <td>WIPI1</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>5.690665</td>\n",
       "      <td>2.879228</td>\n",
       "      <td>-0.982915</td>\n",
       "      <td>34.241181</td>\n",
       "      <td>6.979361</td>\n",
       "      <td>-2.134569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.134569</td>\n",
       "      <td>1.151654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.134569</td>\n",
       "      <td>1.151654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25476</th>\n",
       "      <td>WIPI2</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>11.798954</td>\n",
       "      <td>2.708389</td>\n",
       "      <td>-2.123152</td>\n",
       "      <td>68.029605</td>\n",
       "      <td>15.179626</td>\n",
       "      <td>-2.108286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.108286</td>\n",
       "      <td>-0.014867</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.108286</td>\n",
       "      <td>-0.014867</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25477</th>\n",
       "      <td>ZFYVE1</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>5.002701</td>\n",
       "      <td>1.683355</td>\n",
       "      <td>-1.571367</td>\n",
       "      <td>27.293784</td>\n",
       "      <td>5.270972</td>\n",
       "      <td>-2.183353</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.183353</td>\n",
       "      <td>0.611985</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.183353</td>\n",
       "      <td>0.611985</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25478</th>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>11.590480</td>\n",
       "      <td>3.562583</td>\n",
       "      <td>-1.701945</td>\n",
       "      <td>68.316257</td>\n",
       "      <td>9.883621</td>\n",
       "      <td>-2.791084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.791084</td>\n",
       "      <td>1.089139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.791084</td>\n",
       "      <td>1.089139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25479</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>493.915944</td>\n",
       "      <td>447.914490</td>\n",
       "      <td>-0.141042</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25480 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene1           Gene_2  prolif_ctrl_avg_norm  \\\n",
       "0                 AKT1           AMBRA1             49.509296   \n",
       "1                 AKT3           AMBRA1             15.129326   \n",
       "2               AMBRA1           AMBRA1             35.618412   \n",
       "3                 ARF6           AMBRA1             31.232640   \n",
       "4                 ATF4           AMBRA1             30.673669   \n",
       "...                ...              ...                   ...   \n",
       "25475            WIPI1  wildtype-I-SceI              5.690665   \n",
       "25476            WIPI2  wildtype-I-SceI             11.798954   \n",
       "25477           ZFYVE1  wildtype-I-SceI              5.002701   \n",
       "25478          ZKSCAN3  wildtype-I-SceI             11.590480   \n",
       "25479  wildtype-I-CeuI  wildtype-I-SceI            493.915944   \n",
       "\n",
       "       post_facs_avg_norm       lfc  prolif_ctrl_avg_norm_h7SK  \\\n",
       "0               49.133852 -0.010982                  66.726643   \n",
       "1               67.584450  2.159344                  20.007618   \n",
       "2               84.113111  1.239708                  36.815835   \n",
       "3               21.116277 -0.564699                  44.586703   \n",
       "4               61.306122  0.999030                  43.737172   \n",
       "...                   ...       ...                        ...   \n",
       "25475            2.879228 -0.982915                  34.241181   \n",
       "25476            2.708389 -2.123152                  68.029605   \n",
       "25477            1.683355 -1.571367                  27.293784   \n",
       "25478            3.562583 -1.701945                  68.316257   \n",
       "25479          447.914490 -0.141042                        NaN   \n",
       "\n",
       "       post_facs_avg_norm_h7SK  lfc_h7SK  prolif_ctrl_avg_norm_U6  \\\n",
       "0                    10.566977 -2.658700                40.667392   \n",
       "1                     8.004394 -1.483954                40.667392   \n",
       "2                    23.038215 -0.731591                40.667392   \n",
       "3                     5.954327 -2.645284                40.667392   \n",
       "4                    10.396138 -1.827212                40.667392   \n",
       "...                        ...       ...                      ...   \n",
       "25475                 6.979361 -2.134569                      NaN   \n",
       "25476                15.179626 -2.108286                      NaN   \n",
       "25477                 5.270972 -2.183353                      NaN   \n",
       "25478                 9.883621 -2.791084                      NaN   \n",
       "25479                      NaN       NaN                      NaN   \n",
       "\n",
       "       post_facs_avg_norm_U6  ...  max_model  max_dLFC  sum_model  sum_dLFC  \\\n",
       "0                  34.655258  ...  -0.195852  0.184870  -2.854552  2.843570   \n",
       "1                  34.655258  ...  -0.195852  2.355195  -1.679806  3.839149   \n",
       "2                  34.655258  ...  -0.195852  1.435559  -0.927443  2.167151   \n",
       "3                  34.655258  ...  -0.195852 -0.368847  -2.841135  2.276436   \n",
       "4                  34.655258  ...  -0.195852  1.194882  -2.023064  3.022094   \n",
       "...                      ...  ...        ...       ...        ...       ...   \n",
       "25475                    NaN  ...  -2.134569  1.151654        NaN       NaN   \n",
       "25476                    NaN  ...  -2.108286 -0.014867        NaN       NaN   \n",
       "25477                    NaN  ...  -2.183353  0.611985        NaN       NaN   \n",
       "25478                    NaN  ...  -2.791084  1.089139        NaN       NaN   \n",
       "25479                    NaN  ...        NaN       NaN        NaN       NaN   \n",
       "\n",
       "       min_model  min_dLFC  mult_model  mult_dLFC  log_model  log_dLFC  \n",
       "0      -2.658700  2.647718    0.520711  -0.531693   0.146446 -0.157428  \n",
       "1      -1.483954  3.643298    0.290635   1.868709   0.113113  2.046230  \n",
       "2      -0.731591  1.971299    0.143283   1.096424   0.071066  1.168641  \n",
       "3      -2.645284  2.080585    0.518083  -1.082782   0.146202 -0.710901  \n",
       "4      -1.827212  2.826242    0.357863   0.641168   0.125875  0.873156  \n",
       "...          ...       ...         ...        ...        ...       ...  \n",
       "25475  -2.134569  1.151654         NaN        NaN        NaN       NaN  \n",
       "25476  -2.108286 -0.014867         NaN        NaN        NaN       NaN  \n",
       "25477  -2.183353  0.611985         NaN        NaN        NaN       NaN  \n",
       "25478  -2.791084  1.089139         NaN        NaN        NaN       NaN  \n",
       "25479        NaN       NaN         NaN        NaN        NaN       NaN  \n",
       "\n",
       "[25480 rows x 21 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RPE1\n",
    "# negative LFC are depleted, -1 cutoff\n",
    "atg_flux_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "5a2c9703",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene1', 'Gene_2', 'prolif_ctrl_avg_norm', 'post_facs_avg_norm', 'lfc',\n",
       "       'prolif_ctrl_avg_norm_h7SK', 'post_facs_avg_norm_h7SK', 'lfc_h7SK',\n",
       "       'prolif_ctrl_avg_norm_U6', 'post_facs_avg_norm_U6', 'lfc_U6',\n",
       "       'max_model', 'max_dLFC', 'sum_model', 'sum_dLFC', 'min_model',\n",
       "       'min_dLFC', 'mult_model', 'mult_dLFC', 'log_model', 'log_dLFC'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_GI.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "1ce638c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "atg_flux_GI = atg_flux_GI.fillna(0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df37ab27",
   "metadata": {},
   "source": [
    "mult_dLFC is the best according to Defining Gene Expression. by R Mani"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "b8960e46",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = atg_flux_GI.loc[:, 'Gene1']\n",
    "curr_GI[\"Gene_B\"] = atg_flux_GI.loc[:, 'Gene_2']\n",
    "curr_GI[\"GI_Score\"] = atg_flux_GI.loc[:, 'lfc']\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['diehl_data']] * atg_flux_GI.shape[0]\n",
    "curr_GI[\"Cell_Line\"] = [\"RPE1\"] * atg_flux_GI.shape[0]\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"GI_Cutoff\"] = [-1] * atg_flux_GI.shape[0]\n",
    "curr_GI[\"Stat_Score\"] = [float(\"nan\")] * atg_flux_GI.shape[0]\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * atg_flux_GI.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "517e4541",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       -0.010982\n",
       "1        2.159344\n",
       "2        1.239708\n",
       "3       -0.564699\n",
       "4        0.999030\n",
       "           ...   \n",
       "25475   -0.982915\n",
       "25476   -2.123152\n",
       "25477   -1.571367\n",
       "25478   -1.701945\n",
       "25479   -0.141042\n",
       "Name: lfc, Length: 25480, dtype: float64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atg_flux_GI.loc[:, 'lfc']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "2f2511e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "sorted_genes = []\n",
    "for i in range(curr_GI.shape[0]):\n",
    "    gene_A = curr_GI[\"Gene_A\"].iloc[i]\n",
    "    gene_B = curr_GI[\"Gene_B\"].iloc[i]\n",
    "    sorted_genes.append('_'.join(sorted([gene_A, gene_B])))\n",
    "curr_GI[\"Sorted Genes\"] = sorted_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "889856dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "      <th>Sorted Genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.010982</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1_AMBRA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>2.159344</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT3_AMBRA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>1.239708</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMBRA1_AMBRA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARF6</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.564699</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMBRA1_ARF6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATF4</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>0.999030</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMBRA1_ATF4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25475</th>\n",
       "      <td>WIPI1</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.982915</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WIPI1_wildtype-I-SceI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25476</th>\n",
       "      <td>WIPI2</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-2.123152</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WIPI2_wildtype-I-SceI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25477</th>\n",
       "      <td>ZFYVE1</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-1.571367</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZFYVE1_wildtype-I-SceI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25478</th>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-1.701945</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZKSCAN3_wildtype-I-SceI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25479</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.141042</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI_wildtype-I-SceI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25480 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene_A           Gene_B Study_Source Cell_Line  GI_Score  \\\n",
       "0                 AKT1           AMBRA1     33956155      RPE1 -0.010982   \n",
       "1                 AKT3           AMBRA1     33956155      RPE1  2.159344   \n",
       "2               AMBRA1           AMBRA1     33956155      RPE1  1.239708   \n",
       "3                 ARF6           AMBRA1     33956155      RPE1 -0.564699   \n",
       "4                 ATF4           AMBRA1     33956155      RPE1  0.999030   \n",
       "...                ...              ...          ...       ...       ...   \n",
       "25475            WIPI1  wildtype-I-SceI     33956155      RPE1 -0.982915   \n",
       "25476            WIPI2  wildtype-I-SceI     33956155      RPE1 -2.123152   \n",
       "25477           ZFYVE1  wildtype-I-SceI     33956155      RPE1 -1.571367   \n",
       "25478          ZKSCAN3  wildtype-I-SceI     33956155      RPE1 -1.701945   \n",
       "25479  wildtype-I-CeuI  wildtype-I-SceI     33956155      RPE1 -0.141042   \n",
       "\n",
       "       GI_Cutoff  Stat_Score  Stat_Cutoff                     Sorted Genes  \n",
       "0             -1         NaN          NaN                      AKT1_AMBRA1  \n",
       "1             -1         NaN          NaN                      AKT3_AMBRA1  \n",
       "2             -1         NaN          NaN                    AMBRA1_AMBRA1  \n",
       "3             -1         NaN          NaN                      AMBRA1_ARF6  \n",
       "4             -1         NaN          NaN                      AMBRA1_ATF4  \n",
       "...          ...         ...          ...                              ...  \n",
       "25475         -1         NaN          NaN            WIPI1_wildtype-I-SceI  \n",
       "25476         -1         NaN          NaN            WIPI2_wildtype-I-SceI  \n",
       "25477         -1         NaN          NaN           ZFYVE1_wildtype-I-SceI  \n",
       "25478         -1         NaN          NaN          ZKSCAN3_wildtype-I-SceI  \n",
       "25479         -1         NaN          NaN  wildtype-I-CeuI_wildtype-I-SceI  \n",
       "\n",
       "[25480 rows x 9 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "8dff2287",
   "metadata": {},
   "outputs": [],
   "source": [
    "# average over duplicate SLs\n",
    "res = curr_GI.groupby('Sorted Genes', as_index = False)['GI_Score'].apply(lambda x: np.mean(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "cedff03a",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.merge(curr_GI, res, on=['Sorted Genes'], how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "cf334180",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = curr_GI.drop_duplicates(subset = ['Sorted Genes'])\n",
    "curr_GI['GI_Score'] = curr_GI['GI_Score_y']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "08bdd953",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>GI_Score_x</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "      <th>Sorted Genes</th>\n",
       "      <th>GI_Score_y</th>\n",
       "      <th>GI_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.010982</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT1_AMBRA1</td>\n",
       "      <td>-0.010982</td>\n",
       "      <td>-0.010982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>2.159344</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKT3_AMBRA1</td>\n",
       "      <td>2.159344</td>\n",
       "      <td>2.159344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>1.239708</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMBRA1_AMBRA1</td>\n",
       "      <td>1.239708</td>\n",
       "      <td>1.239708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARF6</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.564699</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMBRA1_ARF6</td>\n",
       "      <td>-0.564699</td>\n",
       "      <td>-0.564699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATF4</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>0.999030</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMBRA1_ATF4</td>\n",
       "      <td>0.999030</td>\n",
       "      <td>0.999030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25475</th>\n",
       "      <td>WIPI1</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.982915</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WIPI1_wildtype-I-SceI</td>\n",
       "      <td>-0.982915</td>\n",
       "      <td>-0.982915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25476</th>\n",
       "      <td>WIPI2</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-2.123152</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WIPI2_wildtype-I-SceI</td>\n",
       "      <td>-2.123152</td>\n",
       "      <td>-2.123152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25477</th>\n",
       "      <td>ZFYVE1</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-1.571367</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZFYVE1_wildtype-I-SceI</td>\n",
       "      <td>-1.571367</td>\n",
       "      <td>-1.571367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25478</th>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-1.701945</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZKSCAN3_wildtype-I-SceI</td>\n",
       "      <td>-1.701945</td>\n",
       "      <td>-1.701945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25479</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>33956155</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>-0.141042</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wildtype-I-CeuI_wildtype-I-SceI</td>\n",
       "      <td>-0.141042</td>\n",
       "      <td>-0.141042</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23400 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene_A           Gene_B Study_Source Cell_Line  GI_Score_x  \\\n",
       "0                 AKT1           AMBRA1     33956155      RPE1   -0.010982   \n",
       "1                 AKT3           AMBRA1     33956155      RPE1    2.159344   \n",
       "2               AMBRA1           AMBRA1     33956155      RPE1    1.239708   \n",
       "3                 ARF6           AMBRA1     33956155      RPE1   -0.564699   \n",
       "4                 ATF4           AMBRA1     33956155      RPE1    0.999030   \n",
       "...                ...              ...          ...       ...         ...   \n",
       "25475            WIPI1  wildtype-I-SceI     33956155      RPE1   -0.982915   \n",
       "25476            WIPI2  wildtype-I-SceI     33956155      RPE1   -2.123152   \n",
       "25477           ZFYVE1  wildtype-I-SceI     33956155      RPE1   -1.571367   \n",
       "25478          ZKSCAN3  wildtype-I-SceI     33956155      RPE1   -1.701945   \n",
       "25479  wildtype-I-CeuI  wildtype-I-SceI     33956155      RPE1   -0.141042   \n",
       "\n",
       "       GI_Cutoff  Stat_Score  Stat_Cutoff                     Sorted Genes  \\\n",
       "0             -1         NaN          NaN                      AKT1_AMBRA1   \n",
       "1             -1         NaN          NaN                      AKT3_AMBRA1   \n",
       "2             -1         NaN          NaN                    AMBRA1_AMBRA1   \n",
       "3             -1         NaN          NaN                      AMBRA1_ARF6   \n",
       "4             -1         NaN          NaN                      AMBRA1_ATF4   \n",
       "...          ...         ...          ...                              ...   \n",
       "25475         -1         NaN          NaN            WIPI1_wildtype-I-SceI   \n",
       "25476         -1         NaN          NaN            WIPI2_wildtype-I-SceI   \n",
       "25477         -1         NaN          NaN           ZFYVE1_wildtype-I-SceI   \n",
       "25478         -1         NaN          NaN          ZKSCAN3_wildtype-I-SceI   \n",
       "25479         -1         NaN          NaN  wildtype-I-CeuI_wildtype-I-SceI   \n",
       "\n",
       "       GI_Score_y  GI_Score  \n",
       "0       -0.010982 -0.010982  \n",
       "1        2.159344  2.159344  \n",
       "2        1.239708  1.239708  \n",
       "3       -0.564699 -0.564699  \n",
       "4        0.999030  0.999030  \n",
       "...           ...       ...  \n",
       "25475   -0.982915 -0.982915  \n",
       "25476   -2.123152 -2.123152  \n",
       "25477   -1.571367 -1.571367  \n",
       "25478   -1.701945 -1.701945  \n",
       "25479   -0.141042 -0.141042  \n",
       "\n",
       "[23400 rows x 11 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_GI\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "84c213a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>906;869;6;195;0</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-2-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>1140;1069;3;5;0</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-3-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>703;583;14;13;19</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>AMBRA1-KO-4-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>698;656;25;19;21</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1-KO-1-R</td>\n",
       "      <td>ATG10-KO-1-R</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>ATG10</td>\n",
       "      <td>721;660;161;75;21</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248186</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-1-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>6290;6245;133;527;33</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248187</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-2-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>4317;4129;268;237;15</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248188</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-3-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>8497;8366;677;194;31</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248189</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3-KO-4-R</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>3505;3541;62;124;9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248190</th>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>wildtype-I-CeuI</td>\n",
       "      <td>wildtype-I-SceI</td>\n",
       "      <td>9485;14159;258;377;19</td>\n",
       "      <td>Dual</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>ctrl_1;ctrl_2;rep_1;rep_2;rep_3</td>\n",
       "      <td>33956155</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>248191 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Guide 1          Guide 2           Gene 1           Gene 2  \\\n",
       "0           AKT1-KO-1-R    AMBRA1-KO-1-R             AKT1           AMBRA1   \n",
       "1           AKT1-KO-1-R    AMBRA1-KO-2-R             AKT1           AMBRA1   \n",
       "2           AKT1-KO-1-R    AMBRA1-KO-3-R             AKT1           AMBRA1   \n",
       "3           AKT1-KO-1-R    AMBRA1-KO-4-R             AKT1           AMBRA1   \n",
       "4           AKT1-KO-1-R     ATG10-KO-1-R             AKT1            ATG10   \n",
       "...                 ...              ...              ...              ...   \n",
       "248186  wildtype-I-CeuI   ZKSCAN3-KO-1-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248187  wildtype-I-CeuI   ZKSCAN3-KO-2-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248188  wildtype-I-CeuI   ZKSCAN3-KO-3-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248189  wildtype-I-CeuI   ZKSCAN3-KO-4-R  wildtype-I-CeuI          ZKSCAN3   \n",
       "248190  wildtype-I-CeuI  wildtype-I-SceI  wildtype-I-CeuI  wildtype-I-SceI   \n",
       "\n",
       "             Count Replicates  Type Sequencing Cell Line  \\\n",
       "0             906;869;6;195;0  Dual      sgRNA      RPE1   \n",
       "1             1140;1069;3;5;0  Dual      sgRNA      RPE1   \n",
       "2            703;583;14;13;19  Dual      sgRNA      RPE1   \n",
       "3            698;656;25;19;21  Dual      sgRNA      RPE1   \n",
       "4           721;660;161;75;21  Dual      sgRNA      RPE1   \n",
       "...                       ...   ...        ...       ...   \n",
       "248186   6290;6245;133;527;33  Dual      sgRNA      RPE1   \n",
       "248187   4317;4129;268;237;15  Dual      sgRNA      RPE1   \n",
       "248188   8497;8366;677;194;31  Dual      sgRNA      RPE1   \n",
       "248189     3505;3541;62;124;9  Dual      sgRNA      RPE1   \n",
       "248190  9485;14159;258;377;19  Dual      sgRNA      RPE1   \n",
       "\n",
       "                              Condition     Study  \n",
       "0       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "1       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "2       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "3       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "4       ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "...                                 ...       ...  \n",
       "248186  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248187  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248188  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248189  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "248190  ctrl_1;ctrl_2;rep_1;rep_2;rep_3  33956155  \n",
       "\n",
       "[248191 rows x 10 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "0412208e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "12615\n",
      "---\n",
      "Only GI cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 202768\n",
      "Number of controls: 2187\n",
      "Number of singles: 43236\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = curr_GI.copy(), study_controls = controls['diehl_data'], study_conditions = study_conditions['diehl_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "6e93e624",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF.loc[PREV_REF['Study_Source'] == 'Diehl',:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "c9eb700e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF.loc[(PREV_REF['Study_Source'] == 'Diehl') & (PREV_REF['Gene_A'] == PREV_REF['Gene_B']),:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "1be22da7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SL        6776\n",
       "Not SL    3944\n",
       "Name: SL_or_not, dtype: int64"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "2d988517",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF.loc[(PREV_REF['Study_Source'] == 'Diehl') & (PREV_REF['Gene Pair'] == 'ATG2B_MTOR'),:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "309b96c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ATG2B_MTOR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "3f3cfc81",
   "metadata": {},
   "outputs": [],
   "source": [
    "#set(PREV_REF.loc[PREV_REF['Study_Source'] == 'Diehl','Gene Pair']).difference(set(db_inserts['score_ref']['gene_pair']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "a8b7b56f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#db_inserts['score_ref']['gene_pair']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "c32f22ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "#(db_inserts['score_ref']['GI_Score_x'] < -1).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "e5299ede",
   "metadata": {},
   "outputs": [],
   "source": [
    "#(atg_flux_GI.loc[:, 'lfc'] < -1).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "4601be89",
   "metadata": {},
   "outputs": [],
   "source": [
    "#db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4679758e",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "de9b31c3",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6697c772",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "raw",
   "id": "5da97970",
   "metadata": {},
   "source": [
    "engine_link = SLKB_engine\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "335d20a9",
   "metadata": {},
   "source": [
    "# first, get the metadata\n",
    "db_metadata = sqlalchemy.MetaData(bind=engine_link)\n",
    "db_metadata.reflect(engine_link)\n",
    "\n",
    "# access the tables\n",
    "sequence_table = db_metadata.tables['CDKO_EXPERIMENT_DESIGN']\n",
    "counts_table = db_metadata.tables['CDKO_SGRNA_COUNTS']\n",
    "scores_table = db_metadata.tables['CDKO_ORIGINAL_SL_RESULTS']\n",
    "\n",
    "# then, start the session\n",
    "engine_session = sessionmaker(bind=engine_link)\n",
    "curr_session = engine_session()\n",
    "\n",
    "# get number of records in each table\n",
    "sequence_records_num = curr_session.query(sequence_table).count()\n",
    "counts_records_num = curr_session.query(counts_table).count()\n",
    "scores_records_num = curr_session.query(scores_table).count()\n",
    "\n",
    "# get available gene pairs\n",
    "available_gene_pairs = curr_session.query(sqlalchemy.func.max(counts_table.c.gene_pair_id)).first()[0]\n",
    "\n",
    "# for the first instance\n",
    "if available_gene_pairs is None:\n",
    "    available_gene_pairs = -1\n",
    "\n",
    "# in the case of studies that doesn't have counts, but were added\n",
    "temp = curr_session.query(sqlalchemy.func.max(scores_table.c.gene_pair_id)).first()[0]\n",
    "\n",
    "if temp is not None:\n",
    "    available_gene_pairs = max(available_gene_pairs, temp)\n",
    "\n",
    "# proceed to add the IDs to each table and reindex\n",
    "if db_inserts['sequence_ref'] is not None:\n",
    "    sequence_insert = db_inserts['sequence_ref'].reset_index(drop=True)\n",
    "else:\n",
    "    sequence_insert = None\n",
    "\n",
    "if db_inserts['counts_ref'] is not None:\n",
    "    counts_insert = db_inserts['counts_ref'].reset_index(drop=True)\n",
    "else:\n",
    "    counts_insert = None\n",
    "score_insert = db_inserts['score_ref'].reset_index(drop=True)\n",
    "\n",
    "# add the current number of records for proper insertion\n",
    "if sequence_insert is not None:\n",
    "    sequence_insert.index += sequence_records_num\n",
    "if counts_insert is not None:\n",
    "    counts_insert.index += counts_records_num\n",
    "score_insert.index += scores_records_num\n",
    "\n",
    "# set IDs\n",
    "if sequence_insert is not None:\n",
    "    sequence_insert['sgRNA_id'] = sequence_insert.index\n",
    "if counts_insert is not None:\n",
    "    counts_insert['sgRNA_pair_id'] = counts_insert.index\n",
    "score_insert['id'] = score_insert.index\n",
    "score_insert['gene_pair_id'] = score_insert['id'].copy()\n",
    "\n",
    "if (sequence_insert is not None) or (counts_insert is not None):\n",
    "    for_merging = sequence_insert.copy()\n",
    "    for_merging['ref_id'] = for_merging.index\n",
    "\n",
    "    # # add the foreign keys\n",
    "    counts_insert['FK_guide_1_id'] = counts_insert.merge(for_merging, how = 'left', left_on = 'Guide 1', right_on = 'sgRNA_guide_name')['ref_id'].values\n",
    "    counts_insert['FK_guide_2_id'] = counts_insert.merge(for_merging, how = 'left', left_on = 'Guide 2', right_on = 'sgRNA_guide_name')['ref_id'].values\n",
    "\n",
    "if counts_insert is not None:\n",
    "    for_merging = score_insert.copy()\n",
    "\n",
    "    # in the case of multiple cell lines\n",
    "    for_merging['gene_pair+cell_line+study_origin'] = for_merging['gene_pair'] + '+' + for_merging['cell_line_origin'] + '+' + for_merging['study_origin']\n",
    "    counts_insert['gene_pair+cell_line+study_origin'] = counts_insert['gene_pair'] + '+' + counts_insert['cell_line_origin'] + '+' + counts_insert['study_origin']\n",
    "\n",
    "    # set the gene pair id\n",
    "    #counts_insert['gene_pair_id_all'] = np.NaN\n",
    "    # dual have the gene pair id\n",
    "    #counts_insert.loc[counts_insert['target_type'] == 'Dual', 'gene_pair_id_all'] = counts_insert.loc[counts_insert['target_type'] == 'Dual'].groupby(['gene_pair+cell_line+study_origin']).ngroup() + (available_gene_pairs + 1)\n",
    "    counts_insert['gene_pair_id_all'] = counts_insert.groupby(['gene_pair+cell_line+study_origin']).ngroup() + (available_gene_pairs + 1)\n",
    "    \n",
    "    score_insert['gene_pair_id'] = for_merging.merge(counts_insert.drop_duplicates(subset = 'gene_pair+cell_line+study_origin'), how = 'left', left_on = 'gene_pair+cell_line+study_origin', right_on = 'gene_pair+cell_line+study_origin')['gene_pair_id_all'].values\n",
    "\n",
    "## check if there is any NA in the references\n",
    "if (sequence_insert is not None) or (counts_insert is not None):\n",
    "    for col in ['FK_guide_1_id', 'FK_guide_2_id']:\n",
    "        if counts_insert[col].isna().sum() > 0:\n",
    "            print('NA in foreign keys: ' + col)\n",
    "else:\n",
    "    print('No counts and sequences together')\n",
    "\n",
    "print('Final QC...')\n",
    "# Final quality control\n",
    "if sequence_insert is not None:\n",
    "    sequence_insert = sequence_insert.applymap(lambda x: x.strip() if isinstance(x, str) else x, na_action='ignore')\n",
    "if counts_insert is not None:\n",
    "    counts_insert = counts_insert.applymap(lambda x: x.strip() if isinstance(x, str) else x, na_action='ignore')\n",
    "score_insert = score_insert.applymap(lambda x: x.strip() if isinstance(x, str) else x, na_action='ignore')\n",
    "\n",
    "# proceed to insert to the database\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "1fee843c",
   "metadata": {},
   "source": [
    "score_insert['gene_pair_id']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "7208a75b",
   "metadata": {},
   "source": [
    "scores_records_num"
   ]
  },
  {
   "cell_type": "raw",
   "id": "3b6539b6",
   "metadata": {},
   "source": [
    "(counts_insert['gene_pair_id_all'] == 23400).sum()"
   ]
  },
  {
   "cell_type": "raw",
   "id": "6402e7b7",
   "metadata": {},
   "source": [
    "available_gene_pairs"
   ]
  },
  {
   "cell_type": "raw",
   "id": "7314cd11",
   "metadata": {},
   "source": [
    "temp"
   ]
  },
  {
   "cell_type": "raw",
   "id": "995ab4b2",
   "metadata": {},
   "source": [
    "counts_insert"
   ]
  },
  {
   "cell_type": "raw",
   "id": "f0628276",
   "metadata": {},
   "source": [
    "db_inserts['sequence_ref']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d0356fb5",
   "metadata": {},
   "source": [
    "score_insert['gene_pair_id'].isna().sum()"
   ]
  },
  {
   "cell_type": "raw",
   "id": "aa2a5bb8",
   "metadata": {},
   "source": [
    "score_insert['gene_pair_id']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "bbaf30bd",
   "metadata": {},
   "source": [
    "counts_insert['gene_pair_id']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "a5af9c6f",
   "metadata": {},
   "source": [
    "score_insert"
   ]
  },
  {
   "cell_type": "raw",
   "id": "18a4fde4",
   "metadata": {},
   "source": [
    "set(score_insert['gene_pair_id']).difference(set(counts_insert['gene_pair_id']))"
   ]
  },
  {
   "cell_type": "raw",
   "id": "0d1adf59",
   "metadata": {},
   "source": [
    "score_insert['gene_pair_id'] = score_insert['gene_pair_id'].astype(np.int64)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "27691127",
   "metadata": {},
   "source": [
    "counts_insert"
   ]
  },
  {
   "cell_type": "raw",
   "id": "cd782abf",
   "metadata": {},
   "source": [
    "set(score_insert['gene_pair_id']).difference(set(counts_insert['gene_pair_id']))\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "5ae1198f",
   "metadata": {},
   "source": [
    "counts_insert"
   ]
  },
  {
   "cell_type": "raw",
   "id": "61dba85a",
   "metadata": {},
   "source": [
    "score_insert['gene_pair_id']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "8eebaa5d",
   "metadata": {},
   "source": [
    "score_insert['gene_pair_id']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "fc8ecba4",
   "metadata": {},
   "source": [
    "counts_insert['gene_pair_id']"
   ]
  },
  {
   "cell_type": "raw",
   "id": "097dc731",
   "metadata": {},
   "source": [
    "\n",
    "# start the transaction, # insert only the columns we need\n",
    "with engine_link.begin() as transaction:\n",
    "    # insert sequence\n",
    "    print('Beginning transaction...')\n",
    "\n",
    "    if sequence_insert is not None:\n",
    "        sequence_insert = sequence_insert.loc[:,['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']]\n",
    "        sequence_insert.to_sql(name = 'CDKO_EXPERIMENT_DESIGN', con = transaction, if_exists = 'append', index = False, index_label = 'sgRNA_id')\n",
    "        \n",
    "        print('Done sequence')\n",
    "        \n",
    "    # insert CDKO counts\n",
    "    if counts_insert is not None:\n",
    "        counts_insert = counts_insert.loc[:,['sgRNA_pair_id', 'FK_guide_1_id', 'FK_guide_2_id', 'gene_pair_id_all', 'gene_pair_orientation', 'T0_counts', 'T0_replicate_names', 'TEnd_counts', 'TEnd_replicate_names', 'target_type', 'study_origin', 'cell_line_origin']]\n",
    "\n",
    "        counts_insert = counts_insert.rename(columns = {'FK_guide_1_id': 'guide_1_id',\n",
    "                                                               'FK_guide_2_id': 'guide_2_id',\n",
    "                                                               'gene_pair_id_all': 'gene_pair_id'})\n",
    "\n",
    "        counts_insert.to_sql(name = 'CDKO_SGRNA_COUNTS', con = transaction, if_exists = 'append', index = False, index_label = 'sgRNA_pair_id')\n",
    "        \n",
    "        print('Done counts')\n",
    "        \n",
    "    # finally, insert scores\n",
    "    score_insert = score_insert.loc[:, ['gene_1', 'gene_2', 'study_origin', 'cell_line_origin', 'SL_score', 'SL_score_cutoff', 'statistical_score', 'statistical_score_cutoff', 'gene_pair', 'SL_or_not', 'gene_pair_id', 'id']]\n",
    "    score_insert.to_sql(name = 'CDKO_ORIGINAL_SL_RESULTS', con = transaction, if_exists = 'append', index = False, index_label = 'id')\n",
    "    \n",
    "    print('Done score')\n",
    "\n",
    "    print('Successfully inserted!')\n",
    "\n",
    "    print('Added Record stats...')\n",
    "    if sequence_insert is not None:\n",
    "        print(' '.join(['Sequence insert:', str(sequence_insert.shape[0])]))\n",
    "    if counts_insert is not None:\n",
    "        print(' '.join(['Counts insert:', str(counts_insert.shape[0])]))\n",
    "    print(' '.join(['Score insert:', str(score_insert.shape[0])]))"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d17ae055",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "raw",
   "id": "51e954ce",
   "metadata": {},
   "source": [
    "score_insert"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "bfa2d704",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 1176\n",
      "Counts insert: 248191\n",
      "Score insert: 10720\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d5ed598",
   "metadata": {},
   "source": [
    "## Han Data (K562) No counts\n",
    "No counts are available"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "678734d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "han_loc = os.path.join(learning_goals_loc_general, \"Han\")\n",
    "\n",
    "ricin_CDKO = pd.read_excel(os.path.join(han_loc, \"7 - GI scores of Ricin-CDKO screen.xlsx\"))\n",
    "drug_CDKO = pd.read_excel(os.path.join(han_loc, \"4 - GI scores of DrugTarget-CDKO screen.xlsx\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "918c3fb4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_Pair</th>\n",
       "      <th>GeneA  Phenotype</th>\n",
       "      <th>GeneB  Phenotype</th>\n",
       "      <th>GeneAB Observed\\n Phenotype</th>\n",
       "      <th>GeneAB RawGI</th>\n",
       "      <th>GeneAB NormGI</th>\n",
       "      <th>GeneAB GIT score</th>\n",
       "      <th>GeneAB GIM score</th>\n",
       "      <th>Correlation of\\nGI profiles\\n(GIT score)</th>\n",
       "      <th>Correlation of\\nGI profiles\\n(GIM score)</th>\n",
       "      <th>q-values based on GIM score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B4GALT1__B4GALT3</td>\n",
       "      <td>4.574319</td>\n",
       "      <td>1.231391</td>\n",
       "      <td>9.000142</td>\n",
       "      <td>-3.328240</td>\n",
       "      <td>-4.150589</td>\n",
       "      <td>-23.494356</td>\n",
       "      <td>-24.621381</td>\n",
       "      <td>-0.389358</td>\n",
       "      <td>-0.383814</td>\n",
       "      <td>1.960799e-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SLC39A9__TRAPPC13</td>\n",
       "      <td>-5.004521</td>\n",
       "      <td>10.661590</td>\n",
       "      <td>8.043506</td>\n",
       "      <td>-2.838739</td>\n",
       "      <td>-3.546919</td>\n",
       "      <td>-17.692541</td>\n",
       "      <td>-24.275939</td>\n",
       "      <td>0.028368</td>\n",
       "      <td>0.072512</td>\n",
       "      <td>2.895902e-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC39A9__TRAPPC2L</td>\n",
       "      <td>-5.004521</td>\n",
       "      <td>8.633068</td>\n",
       "      <td>5.033394</td>\n",
       "      <td>-1.581442</td>\n",
       "      <td>-2.062448</td>\n",
       "      <td>-11.040980</td>\n",
       "      <td>-18.488956</td>\n",
       "      <td>-0.051350</td>\n",
       "      <td>0.035278</td>\n",
       "      <td>6.649633e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MAN1B1__MOGS</td>\n",
       "      <td>3.992183</td>\n",
       "      <td>2.377593</td>\n",
       "      <td>7.415340</td>\n",
       "      <td>-1.291890</td>\n",
       "      <td>-1.679146</td>\n",
       "      <td>-9.784799</td>\n",
       "      <td>-16.660538</td>\n",
       "      <td>-0.176978</td>\n",
       "      <td>-0.206039</td>\n",
       "      <td>2.107934e-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MAN2A1__MOGS</td>\n",
       "      <td>2.566985</td>\n",
       "      <td>2.377593</td>\n",
       "      <td>5.997311</td>\n",
       "      <td>-1.311117</td>\n",
       "      <td>-1.729301</td>\n",
       "      <td>-9.283981</td>\n",
       "      <td>-17.769419</td>\n",
       "      <td>-0.326060</td>\n",
       "      <td>-0.324798</td>\n",
       "      <td>2.535316e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3076</th>\n",
       "      <td>SAR1A__SAR1A</td>\n",
       "      <td>6.586146</td>\n",
       "      <td>6.586146</td>\n",
       "      <td>10.018327</td>\n",
       "      <td>2.097951</td>\n",
       "      <td>2.303398</td>\n",
       "      <td>10.144232</td>\n",
       "      <td>12.393864</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.576634e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3077</th>\n",
       "      <td>EED__TBL1XR1</td>\n",
       "      <td>5.256384</td>\n",
       "      <td>3.905370</td>\n",
       "      <td>7.379280</td>\n",
       "      <td>1.349383</td>\n",
       "      <td>1.735638</td>\n",
       "      <td>10.256917</td>\n",
       "      <td>18.996254</td>\n",
       "      <td>0.340102</td>\n",
       "      <td>0.349537</td>\n",
       "      <td>2.757013e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3078</th>\n",
       "      <td>SEC31A__SEC31A</td>\n",
       "      <td>4.740903</td>\n",
       "      <td>4.740903</td>\n",
       "      <td>7.211232</td>\n",
       "      <td>2.019320</td>\n",
       "      <td>2.852353</td>\n",
       "      <td>10.770399</td>\n",
       "      <td>12.522567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.230882e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3079</th>\n",
       "      <td>SEC24C__SLC39A9</td>\n",
       "      <td>6.633404</td>\n",
       "      <td>-5.004521</td>\n",
       "      <td>-0.143315</td>\n",
       "      <td>1.874443</td>\n",
       "      <td>1.980267</td>\n",
       "      <td>11.239310</td>\n",
       "      <td>16.727586</td>\n",
       "      <td>-0.061064</td>\n",
       "      <td>-0.076536</td>\n",
       "      <td>1.919277e-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3080</th>\n",
       "      <td>TRAPPC2L__TRAPPC2L</td>\n",
       "      <td>8.633068</td>\n",
       "      <td>8.633068</td>\n",
       "      <td>11.887001</td>\n",
       "      <td>2.725836</td>\n",
       "      <td>2.876552</td>\n",
       "      <td>12.641222</td>\n",
       "      <td>12.818615</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.730180e-12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3081 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene_Pair  GeneA  Phenotype  GeneB  Phenotype  \\\n",
       "0       B4GALT1__B4GALT3           4.574319           1.231391   \n",
       "1      SLC39A9__TRAPPC13          -5.004521          10.661590   \n",
       "2      SLC39A9__TRAPPC2L          -5.004521           8.633068   \n",
       "3           MAN1B1__MOGS           3.992183           2.377593   \n",
       "4           MAN2A1__MOGS           2.566985           2.377593   \n",
       "...                  ...                ...                ...   \n",
       "3076        SAR1A__SAR1A           6.586146           6.586146   \n",
       "3077        EED__TBL1XR1           5.256384           3.905370   \n",
       "3078      SEC31A__SEC31A           4.740903           4.740903   \n",
       "3079     SEC24C__SLC39A9           6.633404          -5.004521   \n",
       "3080  TRAPPC2L__TRAPPC2L           8.633068           8.633068   \n",
       "\n",
       "      GeneAB Observed\\n Phenotype  GeneAB RawGI  GeneAB NormGI  \\\n",
       "0                         9.000142     -3.328240      -4.150589   \n",
       "1                         8.043506     -2.838739      -3.546919   \n",
       "2                         5.033394     -1.581442      -2.062448   \n",
       "3                         7.415340     -1.291890      -1.679146   \n",
       "4                         5.997311     -1.311117      -1.729301   \n",
       "...                            ...           ...            ...   \n",
       "3076                     10.018327      2.097951       2.303398   \n",
       "3077                      7.379280      1.349383       1.735638   \n",
       "3078                      7.211232      2.019320       2.852353   \n",
       "3079                     -0.143315      1.874443       1.980267   \n",
       "3080                     11.887001      2.725836       2.876552   \n",
       "\n",
       "      GeneAB GIT score  GeneAB GIM score  \\\n",
       "0           -23.494356        -24.621381   \n",
       "1           -17.692541        -24.275939   \n",
       "2           -11.040980        -18.488956   \n",
       "3            -9.784799        -16.660538   \n",
       "4            -9.283981        -17.769419   \n",
       "...                ...               ...   \n",
       "3076         10.144232         12.393864   \n",
       "3077         10.256917         18.996254   \n",
       "3078         10.770399         12.522567   \n",
       "3079         11.239310         16.727586   \n",
       "3080         12.641222         12.818615   \n",
       "\n",
       "      Correlation of\\nGI profiles\\n(GIT score)  \\\n",
       "0                                    -0.389358   \n",
       "1                                     0.028368   \n",
       "2                                    -0.051350   \n",
       "3                                    -0.176978   \n",
       "4                                    -0.326060   \n",
       "...                                        ...   \n",
       "3076                                       NaN   \n",
       "3077                                  0.340102   \n",
       "3078                                       NaN   \n",
       "3079                                 -0.061064   \n",
       "3080                                       NaN   \n",
       "\n",
       "      Correlation of\\nGI profiles\\n(GIM score)  q-values based on GIM score  \n",
       "0                                    -0.383814                 1.960799e-22  \n",
       "1                                     0.072512                 2.895902e-22  \n",
       "2                                     0.035278                 6.649633e-17  \n",
       "3                                    -0.206039                 2.107934e-15  \n",
       "4                                    -0.324798                 2.535316e-16  \n",
       "...                                        ...                          ...  \n",
       "3076                                       NaN                 1.576634e-11  \n",
       "3077                                  0.349537                 2.757013e-17  \n",
       "3078                                       NaN                 1.230882e-11  \n",
       "3079                                 -0.076536                 1.919277e-15  \n",
       "3080                                       NaN                 6.730180e-12  \n",
       "\n",
       "[3081 rows x 11 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ricin_CDKO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "2a2920c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_Pair</th>\n",
       "      <th>GeneA  Phenotype</th>\n",
       "      <th>GeneB  Phenotype</th>\n",
       "      <th>GeneAB Observed\\n Phenotype</th>\n",
       "      <th>GeneAB RawGI</th>\n",
       "      <th>GeneAB NormGI</th>\n",
       "      <th>GeneAB GIT score</th>\n",
       "      <th>GeneAB GIM score</th>\n",
       "      <th>Correlation of\\nGI profiles\\n(GIT score)</th>\n",
       "      <th>Correlation of\\nGI profiles\\n(GIM score)</th>\n",
       "      <th>q-values based on GIM score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1__AKT2</td>\n",
       "      <td>-1.613746</td>\n",
       "      <td>-3.475764</td>\n",
       "      <td>-6.505799</td>\n",
       "      <td>-1.303365</td>\n",
       "      <td>-1.851698</td>\n",
       "      <td>-10.002585</td>\n",
       "      <td>-14.394872</td>\n",
       "      <td>-0.220295</td>\n",
       "      <td>-0.363181</td>\n",
       "      <td>3.230000e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIM1__PIM2</td>\n",
       "      <td>-3.018919</td>\n",
       "      <td>-2.275498</td>\n",
       "      <td>-6.978705</td>\n",
       "      <td>-1.643037</td>\n",
       "      <td>-2.228615</td>\n",
       "      <td>-9.855829</td>\n",
       "      <td>-14.423427</td>\n",
       "      <td>-0.165029</td>\n",
       "      <td>-0.315924</td>\n",
       "      <td>6.050000e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BCL2L1__MCL1</td>\n",
       "      <td>-5.370227</td>\n",
       "      <td>-2.488718</td>\n",
       "      <td>-10.066704</td>\n",
       "      <td>-1.970779</td>\n",
       "      <td>-2.274515</td>\n",
       "      <td>-8.359275</td>\n",
       "      <td>-10.374774</td>\n",
       "      <td>-0.010717</td>\n",
       "      <td>-0.052520</td>\n",
       "      <td>9.670000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TK1__TK1</td>\n",
       "      <td>-1.890373</td>\n",
       "      <td>-1.890373</td>\n",
       "      <td>-5.974678</td>\n",
       "      <td>-2.383823</td>\n",
       "      <td>-3.588639</td>\n",
       "      <td>-8.116117</td>\n",
       "      <td>-1.408927</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.622756e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALAD__GPI</td>\n",
       "      <td>-4.894231</td>\n",
       "      <td>-3.285434</td>\n",
       "      <td>-9.437143</td>\n",
       "      <td>-1.432968</td>\n",
       "      <td>-1.737535</td>\n",
       "      <td>-7.047904</td>\n",
       "      <td>-11.272882</td>\n",
       "      <td>-0.108534</td>\n",
       "      <td>-0.249004</td>\n",
       "      <td>2.140000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20985</th>\n",
       "      <td>PIK3CA__PIK3CA</td>\n",
       "      <td>-2.546795</td>\n",
       "      <td>-2.546795</td>\n",
       "      <td>-3.802928</td>\n",
       "      <td>1.609335</td>\n",
       "      <td>2.240952</td>\n",
       "      <td>7.348141</td>\n",
       "      <td>7.094248</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.590000e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20986</th>\n",
       "      <td>LCK__LCK</td>\n",
       "      <td>-2.887190</td>\n",
       "      <td>-2.887190</td>\n",
       "      <td>-3.557350</td>\n",
       "      <td>1.994927</td>\n",
       "      <td>2.784584</td>\n",
       "      <td>8.181569</td>\n",
       "      <td>6.979700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.850000e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20987</th>\n",
       "      <td>NPY4R__NPY4R</td>\n",
       "      <td>-2.263467</td>\n",
       "      <td>-2.263467</td>\n",
       "      <td>-2.580027</td>\n",
       "      <td>2.297937</td>\n",
       "      <td>3.294392</td>\n",
       "      <td>8.476707</td>\n",
       "      <td>5.956021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.220400e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20988</th>\n",
       "      <td>AKT2__AKT2</td>\n",
       "      <td>-3.475764</td>\n",
       "      <td>-3.475764</td>\n",
       "      <td>-5.327819</td>\n",
       "      <td>2.178670</td>\n",
       "      <td>2.464910</td>\n",
       "      <td>8.742763</td>\n",
       "      <td>8.521079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.390000e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20989</th>\n",
       "      <td>SKP2__SKP2</td>\n",
       "      <td>-3.289455</td>\n",
       "      <td>-3.289455</td>\n",
       "      <td>-4.626702</td>\n",
       "      <td>1.907078</td>\n",
       "      <td>2.340713</td>\n",
       "      <td>8.909761</td>\n",
       "      <td>11.119543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.440000e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20990 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gene_Pair  GeneA  Phenotype  GeneB  Phenotype  \\\n",
       "0          AKT1__AKT2          -1.613746          -3.475764   \n",
       "1          PIM1__PIM2          -3.018919          -2.275498   \n",
       "2        BCL2L1__MCL1          -5.370227          -2.488718   \n",
       "3            TK1__TK1          -1.890373          -1.890373   \n",
       "4           ALAD__GPI          -4.894231          -3.285434   \n",
       "...               ...                ...                ...   \n",
       "20985  PIK3CA__PIK3CA          -2.546795          -2.546795   \n",
       "20986        LCK__LCK          -2.887190          -2.887190   \n",
       "20987    NPY4R__NPY4R          -2.263467          -2.263467   \n",
       "20988      AKT2__AKT2          -3.475764          -3.475764   \n",
       "20989      SKP2__SKP2          -3.289455          -3.289455   \n",
       "\n",
       "       GeneAB Observed\\n Phenotype  GeneAB RawGI  GeneAB NormGI  \\\n",
       "0                         -6.505799     -1.303365      -1.851698   \n",
       "1                         -6.978705     -1.643037      -2.228615   \n",
       "2                        -10.066704     -1.970779      -2.274515   \n",
       "3                         -5.974678     -2.383823      -3.588639   \n",
       "4                         -9.437143     -1.432968      -1.737535   \n",
       "...                             ...           ...            ...   \n",
       "20985                     -3.802928      1.609335       2.240952   \n",
       "20986                     -3.557350      1.994927       2.784584   \n",
       "20987                     -2.580027      2.297937       3.294392   \n",
       "20988                     -5.327819      2.178670       2.464910   \n",
       "20989                     -4.626702      1.907078       2.340713   \n",
       "\n",
       "       GeneAB GIT score  GeneAB GIM score  \\\n",
       "0            -10.002585        -14.394872   \n",
       "1             -9.855829        -14.423427   \n",
       "2             -8.359275        -10.374774   \n",
       "3             -8.116117         -1.408927   \n",
       "4             -7.047904        -11.272882   \n",
       "...                 ...               ...   \n",
       "20985          7.348141          7.094248   \n",
       "20986          8.181569          6.979700   \n",
       "20987          8.476707          5.956021   \n",
       "20988          8.742763          8.521079   \n",
       "20989          8.909761         11.119543   \n",
       "\n",
       "       Correlation of\\nGI profiles\\n(GIT score)  \\\n",
       "0                                     -0.220295   \n",
       "1                                     -0.165029   \n",
       "2                                     -0.010717   \n",
       "3                                           NaN   \n",
       "4                                     -0.108534   \n",
       "...                                         ...   \n",
       "20985                                       NaN   \n",
       "20986                                       NaN   \n",
       "20987                                       NaN   \n",
       "20988                                       NaN   \n",
       "20989                                       NaN   \n",
       "\n",
       "       Correlation of\\nGI profiles\\n(GIM score)  q-values based on GIM score  \n",
       "0                                     -0.363181                 3.230000e-11  \n",
       "1                                     -0.315924                 6.050000e-11  \n",
       "2                                     -0.052520                 9.670000e-08  \n",
       "3                                           NaN                 1.622756e-01  \n",
       "4                                     -0.249004                 2.140000e-08  \n",
       "...                                         ...                          ...  \n",
       "20985                                       NaN                 7.590000e-05  \n",
       "20986                                       NaN                 8.850000e-05  \n",
       "20987                                       NaN                 5.220400e-04  \n",
       "20988                                       NaN                 4.390000e-06  \n",
       "20989                                       NaN                 2.440000e-08  \n",
       "\n",
       "[20990 rows x 11 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_CDKO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "ce82f50f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0        3.230000e-11\n",
       "1        6.050000e-11\n",
       "2        9.670000e-08\n",
       "3        1.622756e-01\n",
       "4        2.140000e-08\n",
       "             ...     \n",
       "20985    7.590000e-05\n",
       "20986    8.850000e-05\n",
       "20987    5.220400e-04\n",
       "20988    4.390000e-06\n",
       "20989    2.440000e-08\n",
       "Name: q-values based on GIM score, Length: 20990, dtype: float64"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_CDKO.loc[:, 'q-values based on GIM score']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "27066109",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0        -1.613745903|-3.475764088|-6.505798868\n",
       "1        -3.018919429|-2.275498445|-6.978705237\n",
       "2        -5.370226669|-2.488718326|-10.06670369\n",
       "3          -1.89037342|-1.89037342|-5.974678125\n",
       "4        -4.894230657|-3.285433863|-9.437142687\n",
       "                          ...                  \n",
       "20985    -2.546794911|-2.546794911|-3.802928425\n",
       "20986      -2.88719015|-2.88719015|-3.557350168\n",
       "20987    -2.263466552|-2.263466552|-2.580027035\n",
       "20988    -3.475764088|-3.475764088|-5.327819195\n",
       "20989    -3.289454989|-3.289454989|-4.626701805\n",
       "Length: 20990, dtype: object"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_CDKO.loc[:, ['GeneA  Phenotype', 'GeneB  Phenotype', 'GeneAB Observed\\n Phenotype']].astype('str').agg('|'.join, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "6ec33d19",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_A_list, gene_B_list = [], []\n",
    "for i in drug_CDKO['Gene_Pair']:\n",
    "    split_ab = i.split('__')\n",
    "    gene_A_list.append(split_ab[0])\n",
    "    gene_B_list.append(split_ab[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "30efb898",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"Phenotype\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = gene_A_list\n",
    "curr_GI[\"Gene_B\"] = gene_B_list\n",
    "curr_GI[\"GI_Score\"] = drug_CDKO.loc[:, ['GeneAB GIT score', 'GeneAB GIM score']].mean(axis = 1)\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['han_data']] * drug_CDKO.shape[0]\n",
    "curr_GI[\"Cell_Line\"] = [\"K562\"] * drug_CDKO.shape[0]\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"Phenotype\"] = drug_CDKO.loc[:, ['GeneA  Phenotype', 'GeneB  Phenotype', 'GeneAB Observed\\n Phenotype']].astype('str').agg('|'.join, axis=1)\n",
    "curr_GI[\"GI_Cutoff\"] = [-4] * drug_CDKO.shape[0]\n",
    "curr_GI[\"Stat_Score\"] = drug_CDKO.loc[:, 'q-values based on GIM score']\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * drug_CDKO.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "3843477d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       -12.198728\n",
       "1       -12.139628\n",
       "2        -9.367025\n",
       "3        -4.762522\n",
       "4        -9.160393\n",
       "           ...    \n",
       "20985     7.221194\n",
       "20986     7.580635\n",
       "20987     7.216364\n",
       "20988     8.631921\n",
       "20989    10.014652\n",
       "Length: 20990, dtype: float64"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_CDKO.loc[:, ['GeneAB GIT score', 'GeneAB GIM score']].mean(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "ffd31f94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_Pair</th>\n",
       "      <th>GeneA  Phenotype</th>\n",
       "      <th>GeneB  Phenotype</th>\n",
       "      <th>GeneAB Observed\\n Phenotype</th>\n",
       "      <th>GeneAB RawGI</th>\n",
       "      <th>GeneAB NormGI</th>\n",
       "      <th>GeneAB GIT score</th>\n",
       "      <th>GeneAB GIM score</th>\n",
       "      <th>Correlation of\\nGI profiles\\n(GIT score)</th>\n",
       "      <th>Correlation of\\nGI profiles\\n(GIM score)</th>\n",
       "      <th>q-values based on GIM score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1__AKT2</td>\n",
       "      <td>-1.613746</td>\n",
       "      <td>-3.475764</td>\n",
       "      <td>-6.505799</td>\n",
       "      <td>-1.303365</td>\n",
       "      <td>-1.851698</td>\n",
       "      <td>-10.002585</td>\n",
       "      <td>-14.394872</td>\n",
       "      <td>-0.220295</td>\n",
       "      <td>-0.363181</td>\n",
       "      <td>3.230000e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIM1__PIM2</td>\n",
       "      <td>-3.018919</td>\n",
       "      <td>-2.275498</td>\n",
       "      <td>-6.978705</td>\n",
       "      <td>-1.643037</td>\n",
       "      <td>-2.228615</td>\n",
       "      <td>-9.855829</td>\n",
       "      <td>-14.423427</td>\n",
       "      <td>-0.165029</td>\n",
       "      <td>-0.315924</td>\n",
       "      <td>6.050000e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BCL2L1__MCL1</td>\n",
       "      <td>-5.370227</td>\n",
       "      <td>-2.488718</td>\n",
       "      <td>-10.066704</td>\n",
       "      <td>-1.970779</td>\n",
       "      <td>-2.274515</td>\n",
       "      <td>-8.359275</td>\n",
       "      <td>-10.374774</td>\n",
       "      <td>-0.010717</td>\n",
       "      <td>-0.052520</td>\n",
       "      <td>9.670000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TK1__TK1</td>\n",
       "      <td>-1.890373</td>\n",
       "      <td>-1.890373</td>\n",
       "      <td>-5.974678</td>\n",
       "      <td>-2.383823</td>\n",
       "      <td>-3.588639</td>\n",
       "      <td>-8.116117</td>\n",
       "      <td>-1.408927</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.622756e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALAD__GPI</td>\n",
       "      <td>-4.894231</td>\n",
       "      <td>-3.285434</td>\n",
       "      <td>-9.437143</td>\n",
       "      <td>-1.432968</td>\n",
       "      <td>-1.737535</td>\n",
       "      <td>-7.047904</td>\n",
       "      <td>-11.272882</td>\n",
       "      <td>-0.108534</td>\n",
       "      <td>-0.249004</td>\n",
       "      <td>2.140000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20985</th>\n",
       "      <td>PIK3CA__PIK3CA</td>\n",
       "      <td>-2.546795</td>\n",
       "      <td>-2.546795</td>\n",
       "      <td>-3.802928</td>\n",
       "      <td>1.609335</td>\n",
       "      <td>2.240952</td>\n",
       "      <td>7.348141</td>\n",
       "      <td>7.094248</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.590000e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20986</th>\n",
       "      <td>LCK__LCK</td>\n",
       "      <td>-2.887190</td>\n",
       "      <td>-2.887190</td>\n",
       "      <td>-3.557350</td>\n",
       "      <td>1.994927</td>\n",
       "      <td>2.784584</td>\n",
       "      <td>8.181569</td>\n",
       "      <td>6.979700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.850000e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20987</th>\n",
       "      <td>NPY4R__NPY4R</td>\n",
       "      <td>-2.263467</td>\n",
       "      <td>-2.263467</td>\n",
       "      <td>-2.580027</td>\n",
       "      <td>2.297937</td>\n",
       "      <td>3.294392</td>\n",
       "      <td>8.476707</td>\n",
       "      <td>5.956021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.220400e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20988</th>\n",
       "      <td>AKT2__AKT2</td>\n",
       "      <td>-3.475764</td>\n",
       "      <td>-3.475764</td>\n",
       "      <td>-5.327819</td>\n",
       "      <td>2.178670</td>\n",
       "      <td>2.464910</td>\n",
       "      <td>8.742763</td>\n",
       "      <td>8.521079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.390000e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20989</th>\n",
       "      <td>SKP2__SKP2</td>\n",
       "      <td>-3.289455</td>\n",
       "      <td>-3.289455</td>\n",
       "      <td>-4.626702</td>\n",
       "      <td>1.907078</td>\n",
       "      <td>2.340713</td>\n",
       "      <td>8.909761</td>\n",
       "      <td>11.119543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.440000e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20990 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gene_Pair  GeneA  Phenotype  GeneB  Phenotype  \\\n",
       "0          AKT1__AKT2          -1.613746          -3.475764   \n",
       "1          PIM1__PIM2          -3.018919          -2.275498   \n",
       "2        BCL2L1__MCL1          -5.370227          -2.488718   \n",
       "3            TK1__TK1          -1.890373          -1.890373   \n",
       "4           ALAD__GPI          -4.894231          -3.285434   \n",
       "...               ...                ...                ...   \n",
       "20985  PIK3CA__PIK3CA          -2.546795          -2.546795   \n",
       "20986        LCK__LCK          -2.887190          -2.887190   \n",
       "20987    NPY4R__NPY4R          -2.263467          -2.263467   \n",
       "20988      AKT2__AKT2          -3.475764          -3.475764   \n",
       "20989      SKP2__SKP2          -3.289455          -3.289455   \n",
       "\n",
       "       GeneAB Observed\\n Phenotype  GeneAB RawGI  GeneAB NormGI  \\\n",
       "0                         -6.505799     -1.303365      -1.851698   \n",
       "1                         -6.978705     -1.643037      -2.228615   \n",
       "2                        -10.066704     -1.970779      -2.274515   \n",
       "3                         -5.974678     -2.383823      -3.588639   \n",
       "4                         -9.437143     -1.432968      -1.737535   \n",
       "...                             ...           ...            ...   \n",
       "20985                     -3.802928      1.609335       2.240952   \n",
       "20986                     -3.557350      1.994927       2.784584   \n",
       "20987                     -2.580027      2.297937       3.294392   \n",
       "20988                     -5.327819      2.178670       2.464910   \n",
       "20989                     -4.626702      1.907078       2.340713   \n",
       "\n",
       "       GeneAB GIT score  GeneAB GIM score  \\\n",
       "0            -10.002585        -14.394872   \n",
       "1             -9.855829        -14.423427   \n",
       "2             -8.359275        -10.374774   \n",
       "3             -8.116117         -1.408927   \n",
       "4             -7.047904        -11.272882   \n",
       "...                 ...               ...   \n",
       "20985          7.348141          7.094248   \n",
       "20986          8.181569          6.979700   \n",
       "20987          8.476707          5.956021   \n",
       "20988          8.742763          8.521079   \n",
       "20989          8.909761         11.119543   \n",
       "\n",
       "       Correlation of\\nGI profiles\\n(GIT score)  \\\n",
       "0                                     -0.220295   \n",
       "1                                     -0.165029   \n",
       "2                                     -0.010717   \n",
       "3                                           NaN   \n",
       "4                                     -0.108534   \n",
       "...                                         ...   \n",
       "20985                                       NaN   \n",
       "20986                                       NaN   \n",
       "20987                                       NaN   \n",
       "20988                                       NaN   \n",
       "20989                                       NaN   \n",
       "\n",
       "       Correlation of\\nGI profiles\\n(GIM score)  q-values based on GIM score  \n",
       "0                                     -0.363181                 3.230000e-11  \n",
       "1                                     -0.315924                 6.050000e-11  \n",
       "2                                     -0.052520                 9.670000e-08  \n",
       "3                                           NaN                 1.622756e-01  \n",
       "4                                     -0.249004                 2.140000e-08  \n",
       "...                                         ...                          ...  \n",
       "20985                                       NaN                 7.590000e-05  \n",
       "20986                                       NaN                 8.850000e-05  \n",
       "20987                                       NaN                 5.220400e-04  \n",
       "20988                                       NaN                 4.390000e-06  \n",
       "20989                                       NaN                 2.440000e-08  \n",
       "\n",
       "[20990 rows x 11 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_CDKO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "a6655bd6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Only GI cutoff is present...\n",
      "Counts reference...\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = None, counts_ref = None, score_ref = curr_GI.copy(), study_controls = None, study_conditions = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "b2983599",
   "metadata": {},
   "outputs": [],
   "source": [
    "# #HAN specific, only top 30 instead of the higher number (40)\n",
    "# db_inserts['score_ref'].sort_values('SL_score', inplace = True)\n",
    "# db_inserts['score_ref']['SL_or_not'] = [False] * db_inserts['score_ref'].shape[0]\n",
    "# #db_inserts['score_ref']['SL_or_not'][:31] = [True] * 31\n",
    "# db_inserts['score_ref'].sort_values('gene_pair', inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "787cfeb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "633cbbc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "No counts and sequences together\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Score insert: 20791\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f2195dd",
   "metadata": {},
   "source": [
    "## Horlbeck Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "a55d08e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16403553/ipykernel_128410/2129957103.py:6: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  sgRNA_counts = pd.read_csv(os.path.join(horlbeck_loc, \"3 - sgRNA Pair Read Counts Based on Barcode, sgRNA, and Triple Sequencing.txt\"), sep = \"\\t\")\n",
      "/tmp/slurmtmp.16403553/ipykernel_128410/2129957103.py:7: DtypeWarning: Columns (1,2,3,4,5,6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  sgRNA_phenotypes = pd.read_csv(os.path.join(horlbeck_loc, \"4 - sgRNA Pair Phenotypes.txt\"), sep = \"\\t\")\n"
     ]
    }
   ],
   "source": [
    "horlbeck_loc = os.path.join(learning_goals_loc_general, \"Horlbeck\")\n",
    "\n",
    "\n",
    "sgRNA_ref = pd.read_excel(os.path.join(horlbeck_loc, \"2 - Genes Included in the GI Map with Localization, Function, Neighboring Gene Annotation, and sgRNA Information.xlsx\"), sheet_name = \"Table S2b sgRNA sequences\")\n",
    "sgRNA_level_GI = pd.read_excel(os.path.join(horlbeck_loc, \"5 - sgRNA-Level and Gene-Level GI Scores and Correlations; GO Terms Used and Clusters Identified in K562, Jurkat, and Combined GI Maps.xlsx\"))\n",
    "sgRNA_counts = pd.read_csv(os.path.join(horlbeck_loc, \"3 - sgRNA Pair Read Counts Based on Barcode, sgRNA, and Triple Sequencing.txt\"), sep = \"\\t\")\n",
    "sgRNA_phenotypes = pd.read_csv(os.path.join(horlbeck_loc, \"4 - sgRNA Pair Phenotypes.txt\"), sep = \"\\t\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0cc7b38",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "2d252f5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = pd.DataFrame(columns = [\"Guide_ID\", \"Sequence\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f47b244d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref['Sequence'] = sgRNA_ref['sgRNA sequence']\n",
    "\n",
    "# sgRNA_ref[[\"Upstream Barcode\", \"sgRNA sequence\", \"Downstream Barcode\"]].apply(    \n",
    "#     lambda x: ''.join(x.astype(str)),\n",
    "#     axis=1\n",
    "# )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "8ce013e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>K562</th>\n",
       "      <th>K562.1</th>\n",
       "      <th>K562.2</th>\n",
       "      <th>K562.3</th>\n",
       "      <th>K562.4</th>\n",
       "      <th>K562.5</th>\n",
       "      <th>K562.6</th>\n",
       "      <th>K562.7</th>\n",
       "      <th>Jurkat</th>\n",
       "      <th>...</th>\n",
       "      <th>Jurkat.2</th>\n",
       "      <th>Jurkat.3</th>\n",
       "      <th>Jurkat.4</th>\n",
       "      <th>Jurkat.5</th>\n",
       "      <th>Jurkat.6</th>\n",
       "      <th>Jurkat.7</th>\n",
       "      <th>Jurkat.8</th>\n",
       "      <th>Jurkat.9</th>\n",
       "      <th>Jurkat.10</th>\n",
       "      <th>Jurkat.11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>...</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>...</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>...</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>combined name</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++AARS2_+_44281027.23-...</td>\n",
       "      <td>153.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>248.0</td>\n",
       "      <td>285.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>288</td>\n",
       "      <td>442</td>\n",
       "      <td>455.0</td>\n",
       "      <td>...</td>\n",
       "      <td>510.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>827.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>637.0</td>\n",
       "      <td>576.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>221.0</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044483</th>\n",
       "      <td>negative_control-10927++negative_control-10115</td>\n",
       "      <td>3128.0</td>\n",
       "      <td>4936.0</td>\n",
       "      <td>5474.0</td>\n",
       "      <td>5653.0</td>\n",
       "      <td>2395.0</td>\n",
       "      <td>3702.0</td>\n",
       "      <td>266</td>\n",
       "      <td>496</td>\n",
       "      <td>586.0</td>\n",
       "      <td>...</td>\n",
       "      <td>984.0</td>\n",
       "      <td>825.0</td>\n",
       "      <td>638.0</td>\n",
       "      <td>423.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>1014.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>485.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044484</th>\n",
       "      <td>negative_control-10927++negative_control-10130</td>\n",
       "      <td>2228.0</td>\n",
       "      <td>2131.0</td>\n",
       "      <td>2135.0</td>\n",
       "      <td>2909.0</td>\n",
       "      <td>1409.0</td>\n",
       "      <td>1481.0</td>\n",
       "      <td>294</td>\n",
       "      <td>298</td>\n",
       "      <td>371.0</td>\n",
       "      <td>...</td>\n",
       "      <td>651.0</td>\n",
       "      <td>743.0</td>\n",
       "      <td>402.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>737.0</td>\n",
       "      <td>840.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>335.0</td>\n",
       "      <td>411.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044485</th>\n",
       "      <td>negative_control-10927++negative_control-10899</td>\n",
       "      <td>4896.0</td>\n",
       "      <td>4908.0</td>\n",
       "      <td>5784.0</td>\n",
       "      <td>5686.0</td>\n",
       "      <td>3594.0</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>612</td>\n",
       "      <td>582</td>\n",
       "      <td>882.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1905.0</td>\n",
       "      <td>1571.0</td>\n",
       "      <td>849.0</td>\n",
       "      <td>707.0</td>\n",
       "      <td>1767.0</td>\n",
       "      <td>1771.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>313.0</td>\n",
       "      <td>1007.0</td>\n",
       "      <td>1002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044486</th>\n",
       "      <td>negative_control-10927++negative_control-10909</td>\n",
       "      <td>388.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>16.0</td>\n",
       "      <td>...</td>\n",
       "      <td>26.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044487</th>\n",
       "      <td>negative_control-10927++negative_control-10927</td>\n",
       "      <td>913.0</td>\n",
       "      <td>938.0</td>\n",
       "      <td>1557.0</td>\n",
       "      <td>1783.0</td>\n",
       "      <td>524.0</td>\n",
       "      <td>650.0</td>\n",
       "      <td>267</td>\n",
       "      <td>252</td>\n",
       "      <td>328.0</td>\n",
       "      <td>...</td>\n",
       "      <td>597.0</td>\n",
       "      <td>310.0</td>\n",
       "      <td>532.0</td>\n",
       "      <td>491.0</td>\n",
       "      <td>852.0</td>\n",
       "      <td>959.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>145.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1044488 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Unnamed: 0     K562   K562.1  \\\n",
       "0                                                      NaN  barcode  barcode   \n",
       "1                                                      NaN      cyc      cyc   \n",
       "2                                                      NaN     rep1     rep2   \n",
       "3                                            combined name      NaN      NaN   \n",
       "4        AARS2_+_44281027.23-P1P2++AARS2_+_44281027.23-...    153.0    180.0   \n",
       "...                                                    ...      ...      ...   \n",
       "1044483     negative_control-10927++negative_control-10115   3128.0   4936.0   \n",
       "1044484     negative_control-10927++negative_control-10130   2228.0   2131.0   \n",
       "1044485     negative_control-10927++negative_control-10899   4896.0   4908.0   \n",
       "1044486     negative_control-10927++negative_control-10909    388.0    139.0   \n",
       "1044487     negative_control-10927++negative_control-10927    913.0    938.0   \n",
       "\n",
       "         K562.2  K562.3     K562.4     K562.5   K562.6   K562.7   Jurkat  ...  \\\n",
       "0         sgRNA   sgRNA  tripleseq  tripleseq  barcode  barcode  barcode  ...   \n",
       "1           cyc     cyc        cyc        cyc       T0       T0       T0  ...   \n",
       "2          rep1    rep2       rep1       rep2     rep1     rep2     rep1  ...   \n",
       "3           NaN     NaN        NaN        NaN      NaN      NaN      NaN  ...   \n",
       "4         248.0   285.0      113.0       99.0      288      442    455.0  ...   \n",
       "...         ...     ...        ...        ...      ...      ...      ...  ...   \n",
       "1044483  5474.0  5653.0     2395.0     3702.0      266      496    586.0  ...   \n",
       "1044484  2135.0  2909.0     1409.0     1481.0      294      298    371.0  ...   \n",
       "1044485  5784.0  5686.0     3594.0     3673.0      612      582    882.0  ...   \n",
       "1044486   300.0   141.0      225.0      111.0       30       21     16.0  ...   \n",
       "1044487  1557.0  1783.0      524.0      650.0      267      252    328.0  ...   \n",
       "\n",
       "        Jurkat.2 Jurkat.3 Jurkat.4 Jurkat.5 Jurkat.6 Jurkat.7   Jurkat.8  \\\n",
       "0        barcode  barcode    sgRNA    sgRNA    sgRNA    sgRNA  tripleseq   \n",
       "1            cyc      cyc       T0       T0      cyc      cyc         T0   \n",
       "2           rep1     rep2     rep1     rep2     rep1     rep2       rep1   \n",
       "3            NaN      NaN      NaN      NaN      NaN      NaN        NaN   \n",
       "4          510.0    364.0    827.0    705.0    637.0    576.0      197.0   \n",
       "...          ...      ...      ...      ...      ...      ...        ...   \n",
       "1044483    984.0    825.0    638.0    423.0   1132.0   1014.0      301.0   \n",
       "1044484    651.0    743.0    402.0    338.0    737.0    840.0      191.0   \n",
       "1044485   1905.0   1571.0    849.0    707.0   1767.0   1771.0      427.0   \n",
       "1044486     26.0     11.0      9.0     11.0     23.0      7.0        6.0   \n",
       "1044487    597.0    310.0    532.0    491.0    852.0    959.0      125.0   \n",
       "\n",
       "          Jurkat.9  Jurkat.10  Jurkat.11  \n",
       "0        tripleseq  tripleseq  tripleseq  \n",
       "1               T0        cyc        cyc  \n",
       "2             rep2       rep1       rep2  \n",
       "3              NaN        NaN        NaN  \n",
       "4            165.0      221.0      160.0  \n",
       "...            ...        ...        ...  \n",
       "1044483      178.0      539.0      485.0  \n",
       "1044484      165.0      335.0      411.0  \n",
       "1044485      313.0     1007.0     1002.0  \n",
       "1044486        8.0       19.0        7.0  \n",
       "1044487      103.0      205.0      145.0  \n",
       "\n",
       "[1044488 rows x 21 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "e3e9018a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref = sgRNA_ref.drop([\"gene name\", \"sgRNA short name\", \"Upstream Barcode\", \"sgRNA sequence\", \"Downstream Barcode\"], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "add47a07",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref.columns = [\"Guide_ID\", \"Sequence\"]\n",
    "sequence_ref = pd.concat([sequence_ref, sgRNA_ref])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "d513dbf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide_ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>GAGTGGCAGCTGCAGCCCGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AARS2_+_44281044.23-P1P2</td>\n",
       "      <td>GGCTACGATGGCAGCGTCAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AATF_-_35306286.23-P1P2</td>\n",
       "      <td>GAGTGGCCGGTCCAGAGCTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AATF_-_35306346.23-P1P2</td>\n",
       "      <td>GGGATCAAGGCGAGAGGATC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB7_+_74375984.23-P1P2</td>\n",
       "      <td>GTAGCGGCTCAGGTCCGCAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>negative_control-10115</td>\n",
       "      <td>GCCGTACTACCGGCGGCCCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1019</th>\n",
       "      <td>negative_control-10130</td>\n",
       "      <td>GATGTAAATTAAGACGAAGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020</th>\n",
       "      <td>negative_control-10899</td>\n",
       "      <td>GAGTCCATTCAGGCGCGCCG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>negative_control-10909</td>\n",
       "      <td>GGAGCGGTCCGACCGAATGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>GTGCGATGCGTCGACACCGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1023 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Guide_ID              Sequence\n",
       "0     AARS2_+_44281027.23-P1P2  GAGTGGCAGCTGCAGCCCGG\n",
       "1     AARS2_+_44281044.23-P1P2  GGCTACGATGGCAGCGTCAG\n",
       "2      AATF_-_35306286.23-P1P2  GAGTGGCCGGTCCAGAGCTG\n",
       "3      AATF_-_35306346.23-P1P2  GGGATCAAGGCGAGAGGATC\n",
       "4     ABCB7_+_74375984.23-P1P2  GTAGCGGCTCAGGTCCGCAG\n",
       "...                        ...                   ...\n",
       "1018    negative_control-10115  GCCGTACTACCGGCGGCCCT\n",
       "1019    negative_control-10130  GATGTAAATTAAGACGAAGC\n",
       "1020    negative_control-10899  GAGTCCATTCAGGCGCGCCG\n",
       "1021    negative_control-10909  GGAGCGGTCCGACCGAATGG\n",
       "1022    negative_control-10927  GTGCGATGCGTCGACACCGG\n",
       "\n",
       "[1023 rows x 2 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "4e4601d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "target_names = np.array([i.split('_')[0] for i in sequence_ref['Guide_ID']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "efa31ce6",
   "metadata": {},
   "outputs": [],
   "source": [
    "target_names[target_names == 'negative'] = 'control'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "86a8dcac",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['targets'] = target_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "ffd7f134",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "ed49955f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>GAGTGGCAGCTGCAGCCCGG</td>\n",
       "      <td>AARS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AARS2_+_44281044.23-P1P2</td>\n",
       "      <td>GGCTACGATGGCAGCGTCAG</td>\n",
       "      <td>AARS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AATF_-_35306286.23-P1P2</td>\n",
       "      <td>GAGTGGCCGGTCCAGAGCTG</td>\n",
       "      <td>AATF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AATF_-_35306346.23-P1P2</td>\n",
       "      <td>GGGATCAAGGCGAGAGGATC</td>\n",
       "      <td>AATF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB7_+_74375984.23-P1P2</td>\n",
       "      <td>GTAGCGGCTCAGGTCCGCAG</td>\n",
       "      <td>ABCB7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>negative_control-10115</td>\n",
       "      <td>GCCGTACTACCGGCGGCCCT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1019</th>\n",
       "      <td>negative_control-10130</td>\n",
       "      <td>GATGTAAATTAAGACGAAGC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020</th>\n",
       "      <td>negative_control-10899</td>\n",
       "      <td>GAGTCCATTCAGGCGCGCCG</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>negative_control-10909</td>\n",
       "      <td>GGAGCGGTCCGACCGAATGG</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>GTGCGATGCGTCGACACCGG</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1023 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0     AARS2_+_44281027.23-P1P2  GAGTGGCAGCTGCAGCCCGG             AARS2\n",
       "1     AARS2_+_44281044.23-P1P2  GGCTACGATGGCAGCGTCAG             AARS2\n",
       "2      AATF_-_35306286.23-P1P2  GAGTGGCCGGTCCAGAGCTG              AATF\n",
       "3      AATF_-_35306346.23-P1P2  GGGATCAAGGCGAGAGGATC              AATF\n",
       "4     ABCB7_+_74375984.23-P1P2  GTAGCGGCTCAGGTCCGCAG             ABCB7\n",
       "...                        ...                   ...               ...\n",
       "1018    negative_control-10115  GCCGTACTACCGGCGGCCCT           control\n",
       "1019    negative_control-10130  GATGTAAATTAAGACGAAGC           control\n",
       "1020    negative_control-10899  GAGTCCATTCAGGCGCGCCG           control\n",
       "1021    negative_control-10909  GGAGCGGTCCGACCGAATGG           control\n",
       "1022    negative_control-10927  GTGCGATGCGTCGACACCGG           control\n",
       "\n",
       "[1023 rows x 3 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7bc158b",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "ed12d621",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "2118fd3b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>K562</th>\n",
       "      <th>K562.1</th>\n",
       "      <th>K562.2</th>\n",
       "      <th>K562.3</th>\n",
       "      <th>K562.4</th>\n",
       "      <th>K562.5</th>\n",
       "      <th>K562.6</th>\n",
       "      <th>K562.7</th>\n",
       "      <th>Jurkat</th>\n",
       "      <th>...</th>\n",
       "      <th>Jurkat.2</th>\n",
       "      <th>Jurkat.3</th>\n",
       "      <th>Jurkat.4</th>\n",
       "      <th>Jurkat.5</th>\n",
       "      <th>Jurkat.6</th>\n",
       "      <th>Jurkat.7</th>\n",
       "      <th>Jurkat.8</th>\n",
       "      <th>Jurkat.9</th>\n",
       "      <th>Jurkat.10</th>\n",
       "      <th>Jurkat.11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>...</td>\n",
       "      <td>barcode</td>\n",
       "      <td>barcode</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>sgRNA</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "      <td>tripleseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>...</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "      <td>T0</td>\n",
       "      <td>T0</td>\n",
       "      <td>cyc</td>\n",
       "      <td>cyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>...</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "      <td>rep1</td>\n",
       "      <td>rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>combined name</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++AARS2_+_44281027.23-...</td>\n",
       "      <td>153.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>248.0</td>\n",
       "      <td>285.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>288</td>\n",
       "      <td>442</td>\n",
       "      <td>455.0</td>\n",
       "      <td>...</td>\n",
       "      <td>510.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>827.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>637.0</td>\n",
       "      <td>576.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>221.0</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044483</th>\n",
       "      <td>negative_control-10927++negative_control-10115</td>\n",
       "      <td>3128.0</td>\n",
       "      <td>4936.0</td>\n",
       "      <td>5474.0</td>\n",
       "      <td>5653.0</td>\n",
       "      <td>2395.0</td>\n",
       "      <td>3702.0</td>\n",
       "      <td>266</td>\n",
       "      <td>496</td>\n",
       "      <td>586.0</td>\n",
       "      <td>...</td>\n",
       "      <td>984.0</td>\n",
       "      <td>825.0</td>\n",
       "      <td>638.0</td>\n",
       "      <td>423.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>1014.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>485.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044484</th>\n",
       "      <td>negative_control-10927++negative_control-10130</td>\n",
       "      <td>2228.0</td>\n",
       "      <td>2131.0</td>\n",
       "      <td>2135.0</td>\n",
       "      <td>2909.0</td>\n",
       "      <td>1409.0</td>\n",
       "      <td>1481.0</td>\n",
       "      <td>294</td>\n",
       "      <td>298</td>\n",
       "      <td>371.0</td>\n",
       "      <td>...</td>\n",
       "      <td>651.0</td>\n",
       "      <td>743.0</td>\n",
       "      <td>402.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>737.0</td>\n",
       "      <td>840.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>335.0</td>\n",
       "      <td>411.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044485</th>\n",
       "      <td>negative_control-10927++negative_control-10899</td>\n",
       "      <td>4896.0</td>\n",
       "      <td>4908.0</td>\n",
       "      <td>5784.0</td>\n",
       "      <td>5686.0</td>\n",
       "      <td>3594.0</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>612</td>\n",
       "      <td>582</td>\n",
       "      <td>882.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1905.0</td>\n",
       "      <td>1571.0</td>\n",
       "      <td>849.0</td>\n",
       "      <td>707.0</td>\n",
       "      <td>1767.0</td>\n",
       "      <td>1771.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>313.0</td>\n",
       "      <td>1007.0</td>\n",
       "      <td>1002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044486</th>\n",
       "      <td>negative_control-10927++negative_control-10909</td>\n",
       "      <td>388.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>16.0</td>\n",
       "      <td>...</td>\n",
       "      <td>26.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044487</th>\n",
       "      <td>negative_control-10927++negative_control-10927</td>\n",
       "      <td>913.0</td>\n",
       "      <td>938.0</td>\n",
       "      <td>1557.0</td>\n",
       "      <td>1783.0</td>\n",
       "      <td>524.0</td>\n",
       "      <td>650.0</td>\n",
       "      <td>267</td>\n",
       "      <td>252</td>\n",
       "      <td>328.0</td>\n",
       "      <td>...</td>\n",
       "      <td>597.0</td>\n",
       "      <td>310.0</td>\n",
       "      <td>532.0</td>\n",
       "      <td>491.0</td>\n",
       "      <td>852.0</td>\n",
       "      <td>959.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>145.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1044488 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Unnamed: 0     K562   K562.1  \\\n",
       "0                                                      NaN  barcode  barcode   \n",
       "1                                                      NaN      cyc      cyc   \n",
       "2                                                      NaN     rep1     rep2   \n",
       "3                                            combined name      NaN      NaN   \n",
       "4        AARS2_+_44281027.23-P1P2++AARS2_+_44281027.23-...    153.0    180.0   \n",
       "...                                                    ...      ...      ...   \n",
       "1044483     negative_control-10927++negative_control-10115   3128.0   4936.0   \n",
       "1044484     negative_control-10927++negative_control-10130   2228.0   2131.0   \n",
       "1044485     negative_control-10927++negative_control-10899   4896.0   4908.0   \n",
       "1044486     negative_control-10927++negative_control-10909    388.0    139.0   \n",
       "1044487     negative_control-10927++negative_control-10927    913.0    938.0   \n",
       "\n",
       "         K562.2  K562.3     K562.4     K562.5   K562.6   K562.7   Jurkat  ...  \\\n",
       "0         sgRNA   sgRNA  tripleseq  tripleseq  barcode  barcode  barcode  ...   \n",
       "1           cyc     cyc        cyc        cyc       T0       T0       T0  ...   \n",
       "2          rep1    rep2       rep1       rep2     rep1     rep2     rep1  ...   \n",
       "3           NaN     NaN        NaN        NaN      NaN      NaN      NaN  ...   \n",
       "4         248.0   285.0      113.0       99.0      288      442    455.0  ...   \n",
       "...         ...     ...        ...        ...      ...      ...      ...  ...   \n",
       "1044483  5474.0  5653.0     2395.0     3702.0      266      496    586.0  ...   \n",
       "1044484  2135.0  2909.0     1409.0     1481.0      294      298    371.0  ...   \n",
       "1044485  5784.0  5686.0     3594.0     3673.0      612      582    882.0  ...   \n",
       "1044486   300.0   141.0      225.0      111.0       30       21     16.0  ...   \n",
       "1044487  1557.0  1783.0      524.0      650.0      267      252    328.0  ...   \n",
       "\n",
       "        Jurkat.2 Jurkat.3 Jurkat.4 Jurkat.5 Jurkat.6 Jurkat.7   Jurkat.8  \\\n",
       "0        barcode  barcode    sgRNA    sgRNA    sgRNA    sgRNA  tripleseq   \n",
       "1            cyc      cyc       T0       T0      cyc      cyc         T0   \n",
       "2           rep1     rep2     rep1     rep2     rep1     rep2       rep1   \n",
       "3            NaN      NaN      NaN      NaN      NaN      NaN        NaN   \n",
       "4          510.0    364.0    827.0    705.0    637.0    576.0      197.0   \n",
       "...          ...      ...      ...      ...      ...      ...        ...   \n",
       "1044483    984.0    825.0    638.0    423.0   1132.0   1014.0      301.0   \n",
       "1044484    651.0    743.0    402.0    338.0    737.0    840.0      191.0   \n",
       "1044485   1905.0   1571.0    849.0    707.0   1767.0   1771.0      427.0   \n",
       "1044486     26.0     11.0      9.0     11.0     23.0      7.0        6.0   \n",
       "1044487    597.0    310.0    532.0    491.0    852.0    959.0      125.0   \n",
       "\n",
       "          Jurkat.9  Jurkat.10  Jurkat.11  \n",
       "0        tripleseq  tripleseq  tripleseq  \n",
       "1               T0        cyc        cyc  \n",
       "2             rep2       rep1       rep2  \n",
       "3              NaN        NaN        NaN  \n",
       "4            165.0      221.0      160.0  \n",
       "...            ...        ...        ...  \n",
       "1044483      178.0      539.0      485.0  \n",
       "1044484      165.0      335.0      411.0  \n",
       "1044485      313.0     1007.0     1002.0  \n",
       "1044486        8.0       19.0        7.0  \n",
       "1044487      103.0      205.0      145.0  \n",
       "\n",
       "[1044488 rows x 21 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "7ee74254",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['K562_barcode,T0,rep1',\n",
       " 'K562_barcode,T0,rep2',\n",
       " 'K562_tripleseq,cyc,rep1',\n",
       " 'K562_tripleseq,cyc,rep2']"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_conditions['horlbeck_data']['K562'][0] + study_conditions['horlbeck_data']['K562'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "71455172",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['K562.4', 'K562.5', 'K562.6', 'K562.7']"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "['K562.4', 'K562.5', 'K562.6', 'K562.7']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "7533e4fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "K562_counts = sgRNA_counts.iloc[:, 0:9].copy()\n",
    "K562_counts = K562_counts.fillna(0)\n",
    "K562_counts = K562_counts.iloc[4:, :]\n",
    "#K562_counts = K562_counts.dropna(subset = ['K562.4', 'K562.5', 'K562.6', 'K562.7'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "2aa70efc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>K562</th>\n",
       "      <th>K562.1</th>\n",
       "      <th>K562.2</th>\n",
       "      <th>K562.3</th>\n",
       "      <th>K562.4</th>\n",
       "      <th>K562.5</th>\n",
       "      <th>K562.6</th>\n",
       "      <th>K562.7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++ABCB7_+_74375984.23-...</td>\n",
       "      <td>210.0</td>\n",
       "      <td>207.0</td>\n",
       "      <td>215.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>559</td>\n",
       "      <td>458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++ABCB7_-_74376019.23-...</td>\n",
       "      <td>288.0</td>\n",
       "      <td>266.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>265.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>639</td>\n",
       "      <td>523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++ACTL6A_+_179280849.2...</td>\n",
       "      <td>890.0</td>\n",
       "      <td>644.0</td>\n",
       "      <td>639.0</td>\n",
       "      <td>851.0</td>\n",
       "      <td>422.0</td>\n",
       "      <td>502.0</td>\n",
       "      <td>1176</td>\n",
       "      <td>1609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++ACTR10_+_58666892.23...</td>\n",
       "      <td>186.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>240.0</td>\n",
       "      <td>445.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>624</td>\n",
       "      <td>828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++ACTR10_+_58666933.23...</td>\n",
       "      <td>191.0</td>\n",
       "      <td>198.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>510</td>\n",
       "      <td>585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044483</th>\n",
       "      <td>negative_control-10927++negative_control-10115</td>\n",
       "      <td>3128.0</td>\n",
       "      <td>4936.0</td>\n",
       "      <td>5474.0</td>\n",
       "      <td>5653.0</td>\n",
       "      <td>2395.0</td>\n",
       "      <td>3702.0</td>\n",
       "      <td>266</td>\n",
       "      <td>496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044484</th>\n",
       "      <td>negative_control-10927++negative_control-10130</td>\n",
       "      <td>2228.0</td>\n",
       "      <td>2131.0</td>\n",
       "      <td>2135.0</td>\n",
       "      <td>2909.0</td>\n",
       "      <td>1409.0</td>\n",
       "      <td>1481.0</td>\n",
       "      <td>294</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044485</th>\n",
       "      <td>negative_control-10927++negative_control-10899</td>\n",
       "      <td>4896.0</td>\n",
       "      <td>4908.0</td>\n",
       "      <td>5784.0</td>\n",
       "      <td>5686.0</td>\n",
       "      <td>3594.0</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>612</td>\n",
       "      <td>582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044486</th>\n",
       "      <td>negative_control-10927++negative_control-10909</td>\n",
       "      <td>388.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044487</th>\n",
       "      <td>negative_control-10927++negative_control-10927</td>\n",
       "      <td>913.0</td>\n",
       "      <td>938.0</td>\n",
       "      <td>1557.0</td>\n",
       "      <td>1783.0</td>\n",
       "      <td>524.0</td>\n",
       "      <td>650.0</td>\n",
       "      <td>267</td>\n",
       "      <td>252</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1044480 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Unnamed: 0    K562  K562.1  \\\n",
       "8        AARS2_+_44281027.23-P1P2++ABCB7_+_74375984.23-...   210.0   207.0   \n",
       "9        AARS2_+_44281027.23-P1P2++ABCB7_-_74376019.23-...   288.0   266.0   \n",
       "10       AARS2_+_44281027.23-P1P2++ACTL6A_+_179280849.2...   890.0   644.0   \n",
       "11       AARS2_+_44281027.23-P1P2++ACTR10_+_58666892.23...   186.0   404.0   \n",
       "12       AARS2_+_44281027.23-P1P2++ACTR10_+_58666933.23...   191.0   198.0   \n",
       "...                                                    ...     ...     ...   \n",
       "1044483     negative_control-10927++negative_control-10115  3128.0  4936.0   \n",
       "1044484     negative_control-10927++negative_control-10130  2228.0  2131.0   \n",
       "1044485     negative_control-10927++negative_control-10899  4896.0  4908.0   \n",
       "1044486     negative_control-10927++negative_control-10909   388.0   139.0   \n",
       "1044487     negative_control-10927++negative_control-10927   913.0   938.0   \n",
       "\n",
       "         K562.2  K562.3  K562.4  K562.5 K562.6 K562.7  \n",
       "8         215.0   208.0   100.0   110.0    559    458  \n",
       "9         258.0   265.0   133.0   167.0    639    523  \n",
       "10        639.0   851.0   422.0   502.0   1176   1609  \n",
       "11        240.0   445.0    42.0   280.0    624    828  \n",
       "12        123.0   213.0    84.0   102.0    510    585  \n",
       "...         ...     ...     ...     ...    ...    ...  \n",
       "1044483  5474.0  5653.0  2395.0  3702.0    266    496  \n",
       "1044484  2135.0  2909.0  1409.0  1481.0    294    298  \n",
       "1044485  5784.0  5686.0  3594.0  3673.0    612    582  \n",
       "1044486   300.0   141.0   225.0   111.0     30     21  \n",
       "1044487  1557.0  1783.0   524.0   650.0    267    252  \n",
       "\n",
       "[1044480 rows x 9 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "K562_counts.iloc[4:, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "6273a6ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "K562_count_replicates = K562_counts[K562_counts.columns[1:]].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "70c5ebda",
   "metadata": {},
   "outputs": [],
   "source": [
    "K562_conditions = ';'.join(sgRNA_counts.loc[0:2, K562_counts.columns[1:]].apply(    \n",
    "    lambda x: ','.join(x.astype(str)),\n",
    "    axis=0\n",
    ").tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "c1df212e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2;barcode,T0,rep1;barcode,T0,rep2'"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "K562_conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "bd8165ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['JURKAT_tripleseq,T0,rep1',\n",
       " 'JURKAT_tripleseq,T0,rep2',\n",
       " 'JURKAT_tripleseq,cyc,rep1',\n",
       " 'JURKAT_tripleseq,cyc,rep2']"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_conditions['horlbeck_data']['JURKAT'][0] + study_conditions['horlbeck_data']['JURKAT'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "d7b4171d",
   "metadata": {},
   "outputs": [],
   "source": [
    "JURKAT_counts = sgRNA_counts.iloc[:,np.r_[0, 9:21]].copy()\n",
    "JURKAT_counts = JURKAT_counts.fillna(0)\n",
    "JURKAT_counts = JURKAT_counts.iloc[4:, :]\n",
    "#JURKAT_counts = JURKAT_counts.dropna(subset = ['Jurkat.8', 'Jurkat.9', 'Jurkat.10', 'Jurkat.11'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "62cb3516",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Jurkat</th>\n",
       "      <th>Jurkat.1</th>\n",
       "      <th>Jurkat.2</th>\n",
       "      <th>Jurkat.3</th>\n",
       "      <th>Jurkat.4</th>\n",
       "      <th>Jurkat.5</th>\n",
       "      <th>Jurkat.6</th>\n",
       "      <th>Jurkat.7</th>\n",
       "      <th>Jurkat.8</th>\n",
       "      <th>Jurkat.9</th>\n",
       "      <th>Jurkat.10</th>\n",
       "      <th>Jurkat.11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++AARS2_+_44281027.23-...</td>\n",
       "      <td>455.0</td>\n",
       "      <td>438.0</td>\n",
       "      <td>510.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>827.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>637.0</td>\n",
       "      <td>576.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>221.0</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++AARS2_+_44281044.23-...</td>\n",
       "      <td>1124.0</td>\n",
       "      <td>868.0</td>\n",
       "      <td>872.0</td>\n",
       "      <td>843.0</td>\n",
       "      <td>1051.0</td>\n",
       "      <td>825.0</td>\n",
       "      <td>805.0</td>\n",
       "      <td>824.0</td>\n",
       "      <td>598.0</td>\n",
       "      <td>438.0</td>\n",
       "      <td>465.0</td>\n",
       "      <td>486.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++AATF_-_35306286.23-P1P2</td>\n",
       "      <td>482.0</td>\n",
       "      <td>386.0</td>\n",
       "      <td>377.0</td>\n",
       "      <td>377.0</td>\n",
       "      <td>545.0</td>\n",
       "      <td>424.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>382.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>202.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++AATF_-_35306346.23-P1P2</td>\n",
       "      <td>525.0</td>\n",
       "      <td>448.0</td>\n",
       "      <td>384.0</td>\n",
       "      <td>340.0</td>\n",
       "      <td>713.0</td>\n",
       "      <td>503.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>302.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2++ABCB7_+_74375984.23-...</td>\n",
       "      <td>668.0</td>\n",
       "      <td>563.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>206.0</td>\n",
       "      <td>547.0</td>\n",
       "      <td>416.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>216.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>81.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044483</th>\n",
       "      <td>negative_control-10927++negative_control-10115</td>\n",
       "      <td>586.0</td>\n",
       "      <td>440.0</td>\n",
       "      <td>984.0</td>\n",
       "      <td>825.0</td>\n",
       "      <td>638.0</td>\n",
       "      <td>423.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>1014.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>485.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044484</th>\n",
       "      <td>negative_control-10927++negative_control-10130</td>\n",
       "      <td>371.0</td>\n",
       "      <td>429.0</td>\n",
       "      <td>651.0</td>\n",
       "      <td>743.0</td>\n",
       "      <td>402.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>737.0</td>\n",
       "      <td>840.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>335.0</td>\n",
       "      <td>411.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044485</th>\n",
       "      <td>negative_control-10927++negative_control-10899</td>\n",
       "      <td>882.0</td>\n",
       "      <td>700.0</td>\n",
       "      <td>1905.0</td>\n",
       "      <td>1571.0</td>\n",
       "      <td>849.0</td>\n",
       "      <td>707.0</td>\n",
       "      <td>1767.0</td>\n",
       "      <td>1771.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>313.0</td>\n",
       "      <td>1007.0</td>\n",
       "      <td>1002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044486</th>\n",
       "      <td>negative_control-10927++negative_control-10909</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044487</th>\n",
       "      <td>negative_control-10927++negative_control-10927</td>\n",
       "      <td>328.0</td>\n",
       "      <td>270.0</td>\n",
       "      <td>597.0</td>\n",
       "      <td>310.0</td>\n",
       "      <td>532.0</td>\n",
       "      <td>491.0</td>\n",
       "      <td>852.0</td>\n",
       "      <td>959.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>145.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1044484 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Unnamed: 0  Jurkat Jurkat.1  \\\n",
       "4        AARS2_+_44281027.23-P1P2++AARS2_+_44281027.23-...   455.0    438.0   \n",
       "5        AARS2_+_44281027.23-P1P2++AARS2_+_44281044.23-...  1124.0    868.0   \n",
       "6        AARS2_+_44281027.23-P1P2++AATF_-_35306286.23-P1P2   482.0    386.0   \n",
       "7        AARS2_+_44281027.23-P1P2++AATF_-_35306346.23-P1P2   525.0    448.0   \n",
       "8        AARS2_+_44281027.23-P1P2++ABCB7_+_74375984.23-...   668.0    563.0   \n",
       "...                                                    ...     ...      ...   \n",
       "1044483     negative_control-10927++negative_control-10115   586.0    440.0   \n",
       "1044484     negative_control-10927++negative_control-10130   371.0    429.0   \n",
       "1044485     negative_control-10927++negative_control-10899   882.0    700.0   \n",
       "1044486     negative_control-10927++negative_control-10909    16.0     19.0   \n",
       "1044487     negative_control-10927++negative_control-10927   328.0    270.0   \n",
       "\n",
       "        Jurkat.2 Jurkat.3 Jurkat.4 Jurkat.5 Jurkat.6 Jurkat.7 Jurkat.8  \\\n",
       "4          510.0    364.0    827.0    705.0    637.0    576.0    197.0   \n",
       "5          872.0    843.0   1051.0    825.0    805.0    824.0    598.0   \n",
       "6          377.0    377.0    545.0    424.0    391.0    382.0    229.0   \n",
       "7          384.0    340.0    713.0    503.0    450.0    336.0    302.0   \n",
       "8          171.0    206.0    547.0    416.0    119.0    135.0    255.0   \n",
       "...          ...      ...      ...      ...      ...      ...      ...   \n",
       "1044483    984.0    825.0    638.0    423.0   1132.0   1014.0    301.0   \n",
       "1044484    651.0    743.0    402.0    338.0    737.0    840.0    191.0   \n",
       "1044485   1905.0   1571.0    849.0    707.0   1767.0   1771.0    427.0   \n",
       "1044486     26.0     11.0      9.0     11.0     23.0      7.0      6.0   \n",
       "1044487    597.0    310.0    532.0    491.0    852.0    959.0    125.0   \n",
       "\n",
       "        Jurkat.9 Jurkat.10 Jurkat.11  \n",
       "4          165.0     221.0     160.0  \n",
       "5          438.0     465.0     486.0  \n",
       "6          167.0     184.0     202.0  \n",
       "7          205.0     210.0     160.0  \n",
       "8          216.0      48.0      81.0  \n",
       "...          ...       ...       ...  \n",
       "1044483    178.0     539.0     485.0  \n",
       "1044484    165.0     335.0     411.0  \n",
       "1044485    313.0    1007.0    1002.0  \n",
       "1044486      8.0      19.0       7.0  \n",
       "1044487    103.0     205.0     145.0  \n",
       "\n",
       "[1044484 rows x 13 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "JURKAT_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "848f1931",
   "metadata": {},
   "outputs": [],
   "source": [
    "JURKAT_count_replicates = JURKAT_counts[JURKAT_counts.columns[1:]].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "cfd49577",
   "metadata": {},
   "outputs": [],
   "source": [
    "JURKAT_conditions = ';'.join(sgRNA_counts.loc[0:2, JURKAT_counts.columns[1:]].apply(    \n",
    "    lambda x: ','.join(x.astype(str)),\n",
    "    axis=0\n",
    ").tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "b61e4c24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,T0,rep1;sgRNA,T0,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,T0,rep1;tripleseq,T0,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2'"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "JURKAT_conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "a6b4b2da",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_1_list_K562 = []\n",
    "guide_2_list_K562 = []\n",
    "gene_1_list_K562 = []\n",
    "gene_2_list_K562 = []\n",
    "\n",
    "for i in K562_counts['Unnamed: 0']:\n",
    "    guide_1, guide_2 = str(i).split('++')\n",
    "    \n",
    "    guide_1_list_K562.append(guide_1)\n",
    "    guide_2_list_K562.append(guide_2)\n",
    "    \n",
    "    gene_1, gene_2 = guide_1.split('_')[0].upper(), guide_2.split('_')[0].upper()\n",
    "    \n",
    "    gene_1_list_K562.append(gene_1)\n",
    "    gene_2_list_K562.append(gene_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "6fdb25bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_1_list_JURKAT = []\n",
    "guide_2_list_JURKAT = []\n",
    "gene_1_list_JURKAT = []\n",
    "gene_2_list_JURKAT = []\n",
    "\n",
    "for i in JURKAT_counts['Unnamed: 0']:\n",
    "    guide_1, guide_2 = str(i).split('++')\n",
    "    \n",
    "    guide_1_list_JURKAT.append(guide_1)\n",
    "    guide_2_list_JURKAT.append(guide_2)\n",
    "    \n",
    "    gene_1, gene_2 = guide_1.split('_')[0].upper(), guide_2.split('_')[0].upper()\n",
    "    \n",
    "    gene_1_list_JURKAT.append(gene_1)\n",
    "    gene_2_list_JURKAT.append(gene_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "26104844",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref[\"Guide 1\"] = guide_1_list_K562 + guide_1_list_JURKAT\n",
    "counts_ref[\"Guide 2\"] = guide_2_list_K562 + guide_2_list_JURKAT\n",
    "counts_ref[\"Gene 1\"] = gene_1_list_K562 + gene_1_list_JURKAT\n",
    "counts_ref[\"Gene 2\"] = gene_2_list_K562 + gene_2_list_JURKAT\n",
    "counts_ref[\"Count Replicates\"] = K562_count_replicates.tolist() + JURKAT_count_replicates.tolist()\n",
    "counts_ref[\"Cell Line\"] = [\"K562\"] * len(guide_1_list_K562) + [\"JURKAT\"] * len(guide_1_list_JURKAT)\n",
    "counts_ref[\"Condition\"] = [K562_conditions] * len(guide_1_list_K562) + [JURKAT_conditions] * len(guide_1_list_JURKAT)\n",
    "counts_ref[\"Sequencing\"] = [\"Barcode, sgRNA, Tripleseq\"] * len(counts_ref[\"Guide 1\"])\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['horlbeck_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "a63c065a",
   "metadata": {},
   "outputs": [],
   "source": [
    "type_list = [\"Dual\"] * len(counts_ref[\"Guide 1\"])\n",
    "# for i in range(len(type_list)):\n",
    "#     if \"NEGATIVE\" == sequence_ref[\"Gene 1\"][i] == \"NEGATIVE\" in sequence_ref[\"Gene 2\"][i]:\n",
    "#         type_list[i] = \"Single\"\n",
    "counts_ref[\"Type\"] = type_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "7102f0c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>153.0;180.0;248.0;285.0;113.0;99.0;288;442</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AARS2_+_44281044.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>570.0;568.0;748.0;513.0;422.0;344.0;641;712</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AATF_-_35306286.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AATF</td>\n",
       "      <td>324.0;180.0;330.0;218.0;213.0;117.0;354;363</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AATF_-_35306346.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AATF</td>\n",
       "      <td>170.0;254.0;183.0;306.0;130.0;171.0;264;427</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>ABCB7_+_74375984.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>ABCB7</td>\n",
       "      <td>210.0;207.0;215.0;208.0;100.0;110.0;559;458</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088963</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10115</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>586.0;440.0;984.0;825.0;638.0;423.0;1132.0;101...</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088964</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10130</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>371.0;429.0;651.0;743.0;402.0;338.0;737.0;840....</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088965</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10899</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>882.0;700.0;1905.0;1571.0;849.0;707.0;1767.0;1...</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088966</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10909</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>16.0;19.0;26.0;11.0;9.0;11.0;23.0;7.0;6.0;8.0;...</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088967</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>328.0;270.0;597.0;310.0;532.0;491.0;852.0;959....</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2088968 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Guide 1                   Guide 2    Gene 1  \\\n",
       "0        AARS2_+_44281027.23-P1P2  AARS2_+_44281027.23-P1P2     AARS2   \n",
       "1        AARS2_+_44281027.23-P1P2  AARS2_+_44281044.23-P1P2     AARS2   \n",
       "2        AARS2_+_44281027.23-P1P2   AATF_-_35306286.23-P1P2     AARS2   \n",
       "3        AARS2_+_44281027.23-P1P2   AATF_-_35306346.23-P1P2     AARS2   \n",
       "4        AARS2_+_44281027.23-P1P2  ABCB7_+_74375984.23-P1P2     AARS2   \n",
       "...                           ...                       ...       ...   \n",
       "2088963    negative_control-10927    negative_control-10115  NEGATIVE   \n",
       "2088964    negative_control-10927    negative_control-10130  NEGATIVE   \n",
       "2088965    negative_control-10927    negative_control-10899  NEGATIVE   \n",
       "2088966    negative_control-10927    negative_control-10909  NEGATIVE   \n",
       "2088967    negative_control-10927    negative_control-10927  NEGATIVE   \n",
       "\n",
       "           Gene 2                                   Count Replicates  Type  \\\n",
       "0           AARS2         153.0;180.0;248.0;285.0;113.0;99.0;288;442  Dual   \n",
       "1           AARS2        570.0;568.0;748.0;513.0;422.0;344.0;641;712  Dual   \n",
       "2            AATF        324.0;180.0;330.0;218.0;213.0;117.0;354;363  Dual   \n",
       "3            AATF        170.0;254.0;183.0;306.0;130.0;171.0;264;427  Dual   \n",
       "4           ABCB7        210.0;207.0;215.0;208.0;100.0;110.0;559;458  Dual   \n",
       "...           ...                                                ...   ...   \n",
       "2088963  NEGATIVE  586.0;440.0;984.0;825.0;638.0;423.0;1132.0;101...  Dual   \n",
       "2088964  NEGATIVE  371.0;429.0;651.0;743.0;402.0;338.0;737.0;840....  Dual   \n",
       "2088965  NEGATIVE  882.0;700.0;1905.0;1571.0;849.0;707.0;1767.0;1...  Dual   \n",
       "2088966  NEGATIVE  16.0;19.0;26.0;11.0;9.0;11.0;23.0;7.0;6.0;8.0;...  Dual   \n",
       "2088967  NEGATIVE  328.0;270.0;597.0;310.0;532.0;491.0;852.0;959....  Dual   \n",
       "\n",
       "                        Sequencing Cell Line  \\\n",
       "0        Barcode, sgRNA, Tripleseq      K562   \n",
       "1        Barcode, sgRNA, Tripleseq      K562   \n",
       "2        Barcode, sgRNA, Tripleseq      K562   \n",
       "3        Barcode, sgRNA, Tripleseq      K562   \n",
       "4        Barcode, sgRNA, Tripleseq      K562   \n",
       "...                            ...       ...   \n",
       "2088963  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088964  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088965  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088966  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088967  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "\n",
       "                                                 Condition     Study  \n",
       "0        barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...  30033366  \n",
       "1        barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...  30033366  \n",
       "2        barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...  30033366  \n",
       "3        barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...  30033366  \n",
       "4        barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,re...  30033366  \n",
       "...                                                    ...       ...  \n",
       "2088963  barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...  30033366  \n",
       "2088964  barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...  30033366  \n",
       "2088965  barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...  30033366  \n",
       "2088966  barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...  30033366  \n",
       "2088967  barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,re...  30033366  \n",
       "\n",
       "[2088968 rows x 10 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "2d0685c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "K562_conditions = ';'.join([\"K562_\" + i for i in 'barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2;barcode,T0,rep1;barcode,T0,rep2'.split(';')])\n",
    "JURKAT_conditions = ';'.join([\"JURKAT_\" + i for i in 'barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,T0,rep1;sgRNA,T0,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,T0,rep1;tripleseq,T0,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2'.split(';')])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "6d94a6f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1044484"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(counts_ref['Condition'] == 'barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2;barcode,T0,rep1;barcode,T0,rep2').sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "8e363809",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1044484"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref['Cell Line'].value_counts()['K562']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "567e2f18",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Condition'][counts_ref['Condition'] == 'barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2;barcode,T0,rep1;barcode,T0,rep2'] = [K562_conditions] * counts_ref['Cell Line'].value_counts()['K562']\n",
    "counts_ref['Condition'][counts_ref['Condition'] == 'barcode,T0,rep1;barcode,T0,rep2;barcode,cyc,rep1;barcode,cyc,rep2;sgRNA,T0,rep1;sgRNA,T0,rep2;sgRNA,cyc,rep1;sgRNA,cyc,rep2;tripleseq,T0,rep1;tripleseq,T0,rep2;tripleseq,cyc,rep1;tripleseq,cyc,rep2'] = [JURKAT_conditions] *counts_ref['Cell Line'].value_counts()['JURKAT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "5f7df2b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>153.0;180.0;248.0;285.0;113.0;99.0;288;442</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AARS2_+_44281044.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>570.0;568.0;748.0;513.0;422.0;344.0;641;712</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AATF_-_35306286.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AATF</td>\n",
       "      <td>324.0;180.0;330.0;218.0;213.0;117.0;354;363</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>AATF_-_35306346.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>AATF</td>\n",
       "      <td>170.0;254.0;183.0;306.0;130.0;171.0;264;427</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2_+_44281027.23-P1P2</td>\n",
       "      <td>ABCB7_+_74375984.23-P1P2</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>ABCB7</td>\n",
       "      <td>210.0;207.0;215.0;208.0;100.0;110.0;559;458</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>K562</td>\n",
       "      <td>K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088963</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10115</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>586.0;440.0;984.0;825.0;638.0;423.0;1132.0;101...</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088964</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10130</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>371.0;429.0;651.0;743.0;402.0;338.0;737.0;840....</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088965</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10899</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>882.0;700.0;1905.0;1571.0;849.0;707.0;1767.0;1...</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088966</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10909</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>16.0;19.0;26.0;11.0;9.0;11.0;23.0;7.0;6.0;8.0;...</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2088967</th>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>negative_control-10927</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>NEGATIVE</td>\n",
       "      <td>328.0;270.0;597.0;310.0;532.0;491.0;852.0;959....</td>\n",
       "      <td>Dual</td>\n",
       "      <td>Barcode, sgRNA, Tripleseq</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...</td>\n",
       "      <td>30033366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2088968 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Guide 1                   Guide 2    Gene 1  \\\n",
       "0        AARS2_+_44281027.23-P1P2  AARS2_+_44281027.23-P1P2     AARS2   \n",
       "1        AARS2_+_44281027.23-P1P2  AARS2_+_44281044.23-P1P2     AARS2   \n",
       "2        AARS2_+_44281027.23-P1P2   AATF_-_35306286.23-P1P2     AARS2   \n",
       "3        AARS2_+_44281027.23-P1P2   AATF_-_35306346.23-P1P2     AARS2   \n",
       "4        AARS2_+_44281027.23-P1P2  ABCB7_+_74375984.23-P1P2     AARS2   \n",
       "...                           ...                       ...       ...   \n",
       "2088963    negative_control-10927    negative_control-10115  NEGATIVE   \n",
       "2088964    negative_control-10927    negative_control-10130  NEGATIVE   \n",
       "2088965    negative_control-10927    negative_control-10899  NEGATIVE   \n",
       "2088966    negative_control-10927    negative_control-10909  NEGATIVE   \n",
       "2088967    negative_control-10927    negative_control-10927  NEGATIVE   \n",
       "\n",
       "           Gene 2                                   Count Replicates  Type  \\\n",
       "0           AARS2         153.0;180.0;248.0;285.0;113.0;99.0;288;442  Dual   \n",
       "1           AARS2        570.0;568.0;748.0;513.0;422.0;344.0;641;712  Dual   \n",
       "2            AATF        324.0;180.0;330.0;218.0;213.0;117.0;354;363  Dual   \n",
       "3            AATF        170.0;254.0;183.0;306.0;130.0;171.0;264;427  Dual   \n",
       "4           ABCB7        210.0;207.0;215.0;208.0;100.0;110.0;559;458  Dual   \n",
       "...           ...                                                ...   ...   \n",
       "2088963  NEGATIVE  586.0;440.0;984.0;825.0;638.0;423.0;1132.0;101...  Dual   \n",
       "2088964  NEGATIVE  371.0;429.0;651.0;743.0;402.0;338.0;737.0;840....  Dual   \n",
       "2088965  NEGATIVE  882.0;700.0;1905.0;1571.0;849.0;707.0;1767.0;1...  Dual   \n",
       "2088966  NEGATIVE  16.0;19.0;26.0;11.0;9.0;11.0;23.0;7.0;6.0;8.0;...  Dual   \n",
       "2088967  NEGATIVE  328.0;270.0;597.0;310.0;532.0;491.0;852.0;959....  Dual   \n",
       "\n",
       "                        Sequencing Cell Line  \\\n",
       "0        Barcode, sgRNA, Tripleseq      K562   \n",
       "1        Barcode, sgRNA, Tripleseq      K562   \n",
       "2        Barcode, sgRNA, Tripleseq      K562   \n",
       "3        Barcode, sgRNA, Tripleseq      K562   \n",
       "4        Barcode, sgRNA, Tripleseq      K562   \n",
       "...                            ...       ...   \n",
       "2088963  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088964  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088965  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088966  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "2088967  Barcode, sgRNA, Tripleseq    JURKAT   \n",
       "\n",
       "                                                 Condition     Study  \n",
       "0        K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...  30033366  \n",
       "1        K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...  30033366  \n",
       "2        K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...  30033366  \n",
       "3        K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...  30033366  \n",
       "4        K562_barcode,cyc,rep1;K562_barcode,cyc,rep2;K5...  30033366  \n",
       "...                                                    ...       ...  \n",
       "2088963  JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...  30033366  \n",
       "2088964  JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...  30033366  \n",
       "2088965  JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...  30033366  \n",
       "2088966  JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...  30033366  \n",
       "2088967  JURKAT_barcode,T0,rep1;JURKAT_barcode,T0,rep2;...  30033366  \n",
       "\n",
       "[2088968 rows x 10 columns]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a0cc3b9",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 484,
   "id": "6b1149af",
   "metadata": {},
   "outputs": [],
   "source": [
    "Horlbeck_GI_JURKAT = pd.read_csv(os.path.join(learning_goals_loc_general, \"Horlbeck\", \"GI_Jurkat_w_score.txt\"), delim_whitespace=True, index_col = 0)\n",
    "Horlbeck_GI_K562 = pd.read_csv(os.path.join(learning_goals_loc_general, \"Horlbeck\", \"GI_K562_w_score.txt\"), delim_whitespace=True, index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 485,
   "id": "183022a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_learning_goal = Horlbeck_GI_JURKAT.copy()\n",
    "row_idx, col_idx = np.triu_indices(curr_learning_goal.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 486,
   "id": "a8e87e8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'FARSA'"
      ]
     },
     "execution_count": 486,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_learning_goal.columns[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 487,
   "id": "7d8cad38",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_learning_goal = Horlbeck_GI_JURKAT.copy()\n",
    "row_idx, col_idx = np.triu_indices(curr_learning_goal.shape[0])\n",
    "\n",
    "gene_A_list = []\n",
    "gene_B_list = []\n",
    "SL_scores = []\n",
    "\n",
    "\n",
    "for idx in range(len(row_idx)):\n",
    "    i = row_idx[idx]\n",
    "    j = col_idx[idx]\n",
    "    gene_A_list.append(curr_learning_goal.index[i])\n",
    "    gene_B_list.append(curr_learning_goal.columns[j])\n",
    "    SL_scores.append(curr_learning_goal.iloc[i, j])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "id": "9a2ec70b",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"Phenotype\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = gene_A_list\n",
    "curr_GI[\"Gene_B\"] = gene_B_list\n",
    "curr_GI[\"GI_Score\"] = SL_scores\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['horlbeck_data']] * len(gene_A_list)\n",
    "curr_GI[\"Cell_Line\"] = [\"JURKAT\"] * len(gene_A_list)\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"Phenotype\"] = [float(\"nan\")] * len(gene_A_list)\n",
    "curr_GI[\"GI_Cutoff\"] = [-3] * len(gene_A_list)\n",
    "curr_GI[\"Stat_Score\"] = [float(\"nan\")] * len(gene_A_list)\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * len(gene_A_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 489,
   "id": "29ea0c95",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI_JURKAT = curr_GI.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 490,
   "id": "7f16f055",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_learning_goal = Horlbeck_GI_K562.copy()\n",
    "row_idx, col_idx = np.triu_indices(curr_learning_goal.shape[0])\n",
    "\n",
    "gene_A_list = []\n",
    "gene_B_list = []\n",
    "SL_scores = []\n",
    "\n",
    "\n",
    "for idx in range(len(row_idx)):\n",
    "    i = row_idx[idx]\n",
    "    j = col_idx[idx]\n",
    "    gene_A_list.append(curr_learning_goal.index[i])\n",
    "    gene_B_list.append(curr_learning_goal.columns[j])\n",
    "    SL_scores.append(curr_learning_goal.iloc[i, j])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 491,
   "id": "e0b8a888",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"Phenotype\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = gene_A_list\n",
    "curr_GI[\"Gene_B\"] = gene_B_list\n",
    "curr_GI[\"GI_Score\"] = SL_scores\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['horlbeck_data']] * len(gene_A_list)\n",
    "curr_GI[\"Cell_Line\"] = [\"K562\"] * len(gene_A_list)\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"Phenotype\"] = [float(\"nan\")] * len(gene_A_list)\n",
    "curr_GI[\"GI_Cutoff\"] = [-3] * len(gene_A_list)\n",
    "curr_GI[\"Stat_Score\"] = [float(\"nan\")] * len(gene_A_list)\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * len(gene_A_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 492,
   "id": "f9439d33",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI_K562 = curr_GI.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 493,
   "id": "19b33333",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.concat([curr_GI_JURKAT, curr_GI_K562])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 494,
   "id": "a7704710",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Horlbeck specific, change the gane name\n",
    "curr_GI.loc[curr_GI['Gene_A'] == 'MINOS1.NBL1', 'Gene_A'] = 'MINOS1-NBL1'\n",
    "curr_GI.loc[curr_GI['Gene_B'] == 'MINOS1.NBL1', 'Gene_B'] = 'MINOS1-NBL1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 495,
   "id": "8e7d2e5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FARSA</td>\n",
       "      <td>FARSA</td>\n",
       "      <td>30033366</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.968186</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FARSA</td>\n",
       "      <td>LIAS</td>\n",
       "      <td>30033366</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.872620</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FARSA</td>\n",
       "      <td>TBC1D31</td>\n",
       "      <td>30033366</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.595562</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FARSA</td>\n",
       "      <td>COX7C</td>\n",
       "      <td>30033366</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.642902</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FARSA</td>\n",
       "      <td>ATP5J2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.626772</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100571</th>\n",
       "      <td>NEMF</td>\n",
       "      <td>RAE1</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.728041</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100572</th>\n",
       "      <td>NEMF</td>\n",
       "      <td>TGIF2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.564289</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100573</th>\n",
       "      <td>RAE1</td>\n",
       "      <td>RAE1</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.444090</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100574</th>\n",
       "      <td>RAE1</td>\n",
       "      <td>TGIF2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100575</th>\n",
       "      <td>TGIF2</td>\n",
       "      <td>TGIF2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.866674</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>176042 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene_A   Gene_B Study_Source Cell_Line  Phenotype  GI_Score  GI_Cutoff  \\\n",
       "0       FARSA    FARSA     30033366    JURKAT        NaN  0.968186         -3   \n",
       "1       FARSA     LIAS     30033366    JURKAT        NaN -2.872620         -3   \n",
       "2       FARSA  TBC1D31     30033366    JURKAT        NaN -0.595562         -3   \n",
       "3       FARSA    COX7C     30033366    JURKAT        NaN -2.642902         -3   \n",
       "4       FARSA   ATP5J2     30033366    JURKAT        NaN  0.626772         -3   \n",
       "...       ...      ...          ...       ...        ...       ...        ...   \n",
       "100571   NEMF     RAE1     30033366      K562        NaN -3.728041         -3   \n",
       "100572   NEMF    TGIF2     30033366      K562        NaN -0.564289         -3   \n",
       "100573   RAE1     RAE1     30033366      K562        NaN -5.444090         -3   \n",
       "100574   RAE1    TGIF2     30033366      K562        NaN  0.998261         -3   \n",
       "100575  TGIF2    TGIF2     30033366      K562        NaN -3.866674         -3   \n",
       "\n",
       "        Stat_Score  Stat_Cutoff  \n",
       "0              NaN          NaN  \n",
       "1              NaN          NaN  \n",
       "2              NaN          NaN  \n",
       "3              NaN          NaN  \n",
       "4              NaN          NaN  \n",
       "...            ...          ...  \n",
       "100571         NaN          NaN  \n",
       "100572         NaN          NaN  \n",
       "100573         NaN          NaN  \n",
       "100574         NaN          NaN  \n",
       "100575         NaN          NaN  \n",
       "\n",
       "[176042 rows x 9 columns]"
      ]
     },
     "execution_count": 495,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "3b66c617",
   "metadata": {},
   "outputs": [],
   "source": [
    "#curr_GI['Stat_Cutoff']\n",
    "#score_ref['Stat_Cutoff'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "54954cca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "Only GI cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 2011444\n",
      "Number of controls: 648\n",
      "Number of singles: 76876\n",
      "remove_unrelated_counts = TRUE\n",
      "For cl = JURKAT\n",
      "Removed a total of 303163 sgRNAs...\n",
      "For cl = K562\n",
      "Removed a total of 91908 sgRNAs...\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = curr_GI.copy(), study_controls = controls['horlbeck_data'], study_conditions = study_conditions['horlbeck_data'],  remove_unrelated_counts = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a08a9f20",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "35a73521",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['sequence_ref', 'counts_ref', 'score_ref'])"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "7e9e1cf5",
   "metadata": {},
   "outputs": [],
   "source": [
    "JURKAT_temp = db_inserts['score_ref'].loc[db_inserts['score_ref']['cell_line_origin'] == 'JURKAT']\n",
    "K562_temp = db_inserts['score_ref'].loc[db_inserts['score_ref']['cell_line_origin'] == 'K562']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "ead1080f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_1</th>\n",
       "      <th>gene_2</th>\n",
       "      <th>study_origin</th>\n",
       "      <th>cell_line_origin</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>SL_score</th>\n",
       "      <th>SL_score_cutoff</th>\n",
       "      <th>statistical_score</th>\n",
       "      <th>statistical_score_cutoff</th>\n",
       "      <th>gene_pair</th>\n",
       "      <th>SL_or_not</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75467</th>\n",
       "      <td>EIF3M</td>\n",
       "      <td>COX7C</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.908462</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>COX7C_EIF3M</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75468</th>\n",
       "      <td>EIF3M</td>\n",
       "      <td>COQ2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.660420</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>COQ2_EIF3M</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75469</th>\n",
       "      <td>EIF3M</td>\n",
       "      <td>PDSS2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.820470</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>EIF3M_PDSS2</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75470</th>\n",
       "      <td>EIF3M</td>\n",
       "      <td>FBXO28</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.190100</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>EIF3M_FBXO28</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75471</th>\n",
       "      <td>EIF3M</td>\n",
       "      <td>APOOL</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.267989</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>APOOL_EIF3M</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176034</th>\n",
       "      <td>ZNF598</td>\n",
       "      <td>RAE1</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.186801</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RAE1_ZNF598</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176035</th>\n",
       "      <td>ZNF598</td>\n",
       "      <td>TGIF2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.804539</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TGIF2_ZNF598</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176037</th>\n",
       "      <td>NEMF</td>\n",
       "      <td>RAE1</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-3.728041</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NEMF_RAE1</td>\n",
       "      <td>SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176038</th>\n",
       "      <td>NEMF</td>\n",
       "      <td>TGIF2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.564289</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NEMF_TGIF2</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176040</th>\n",
       "      <td>RAE1</td>\n",
       "      <td>TGIF2</td>\n",
       "      <td>30033366</td>\n",
       "      <td>K562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RAE1_TGIF2</td>\n",
       "      <td>Not SL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100128 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_1  gene_2 study_origin cell_line_origin  Phenotype  SL_score  \\\n",
       "75467    EIF3M   COX7C     30033366             K562        0.0 -1.908462   \n",
       "75468    EIF3M    COQ2     30033366             K562        0.0 -0.660420   \n",
       "75469    EIF3M   PDSS2     30033366             K562        0.0 -0.820470   \n",
       "75470    EIF3M  FBXO28     30033366             K562        0.0  1.190100   \n",
       "75471    EIF3M   APOOL     30033366             K562        0.0 -0.267989   \n",
       "...        ...     ...          ...              ...        ...       ...   \n",
       "176034  ZNF598    RAE1     30033366             K562        0.0 -0.186801   \n",
       "176035  ZNF598   TGIF2     30033366             K562        0.0  0.804539   \n",
       "176037    NEMF    RAE1     30033366             K562        0.0 -3.728041   \n",
       "176038    NEMF   TGIF2     30033366             K562        0.0 -0.564289   \n",
       "176040    RAE1   TGIF2     30033366             K562        0.0  0.998261   \n",
       "\n",
       "        SL_score_cutoff  statistical_score  statistical_score_cutoff  \\\n",
       "75467                -3                0.0                       0.0   \n",
       "75468                -3                0.0                       0.0   \n",
       "75469                -3                0.0                       0.0   \n",
       "75470                -3                0.0                       0.0   \n",
       "75471                -3                0.0                       0.0   \n",
       "...                 ...                ...                       ...   \n",
       "176034               -3                0.0                       0.0   \n",
       "176035               -3                0.0                       0.0   \n",
       "176037               -3                0.0                       0.0   \n",
       "176038               -3                0.0                       0.0   \n",
       "176040               -3                0.0                       0.0   \n",
       "\n",
       "           gene_pair SL_or_not  \n",
       "75467    COX7C_EIF3M    Not SL  \n",
       "75468     COQ2_EIF3M    Not SL  \n",
       "75469    EIF3M_PDSS2    Not SL  \n",
       "75470   EIF3M_FBXO28    Not SL  \n",
       "75471    APOOL_EIF3M    Not SL  \n",
       "...              ...       ...  \n",
       "176034   RAE1_ZNF598    Not SL  \n",
       "176035  TGIF2_ZNF598    Not SL  \n",
       "176037     NEMF_RAE1        SL  \n",
       "176038    NEMF_TGIF2    Not SL  \n",
       "176040    RAE1_TGIF2    Not SL  \n",
       "\n",
       "[100128 rows x 11 columns]"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "K562_temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "82237d69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "448\n"
     ]
    }
   ],
   "source": [
    "K562_genes = set(K562_temp['gene_1'].tolist() + K562_temp['gene_2'].tolist())\n",
    "print(len(K562_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "7951a8b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "388\n"
     ]
    }
   ],
   "source": [
    "JURKAT_genes = set(JURKAT_temp['gene_1'].tolist() + JURKAT_temp['gene_2'].tolist())\n",
    "print(len(JURKAT_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "id": "58f86733",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ADPRM',\n",
       " 'ANAPC13',\n",
       " 'APOOL',\n",
       " 'ATP6AP1',\n",
       " 'ATXN10',\n",
       " 'BUB1B',\n",
       " 'C14ORF178',\n",
       " 'CDC73',\n",
       " 'CDCA8',\n",
       " 'CEP63',\n",
       " 'CIT',\n",
       " 'CKAP5',\n",
       " 'CLSPN',\n",
       " 'COG3',\n",
       " 'COPS2',\n",
       " 'COPS5',\n",
       " 'COX6C',\n",
       " 'CYB5B',\n",
       " 'DCTN2',\n",
       " 'DNA2',\n",
       " 'DONSON',\n",
       " 'EIF3D',\n",
       " 'EMC6',\n",
       " 'ERH',\n",
       " 'EXOSC9',\n",
       " 'FARSB',\n",
       " 'GFI1B',\n",
       " 'GINS4',\n",
       " 'GOLT1B',\n",
       " 'GTF2E1',\n",
       " 'GTF2H1',\n",
       " 'HBS1L',\n",
       " 'KIF11',\n",
       " 'KIF14',\n",
       " 'LUC7L3',\n",
       " 'MAT2A',\n",
       " 'MRPL50',\n",
       " 'MRPS28',\n",
       " 'NAA50',\n",
       " 'NOL8',\n",
       " 'NSMCE4A',\n",
       " 'NUTF2',\n",
       " 'PAFAH1B1',\n",
       " 'PELO',\n",
       " 'PHF5A',\n",
       " 'PMPCB',\n",
       " 'PMVK',\n",
       " 'PRPF18',\n",
       " 'PSMC6',\n",
       " 'PSMD1',\n",
       " 'RBM22',\n",
       " 'RPL13',\n",
       " 'RPL8',\n",
       " 'SCYL1',\n",
       " 'SEC22B',\n",
       " 'SGOL1',\n",
       " 'SLC35A1',\n",
       " 'SLC39A9',\n",
       " 'SPATA5',\n",
       " 'SRP19',\n",
       " 'TAZ',\n",
       " 'TIMELESS',\n",
       " 'TIMM23B',\n",
       " 'TTI2',\n",
       " 'TUBGCP4',\n",
       " 'USMG5',\n",
       " 'VPS4A',\n",
       " 'VPS72',\n",
       " 'WDR70',\n",
       " 'ZNF574'}"
      ]
     },
     "execution_count": 496,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "K562_genes.difference(JURKAT_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 497,
   "id": "d1fc188f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'COPA',\n",
       " 'CWC22',\n",
       " 'DUT',\n",
       " 'EIF2B2',\n",
       " 'EIF4G2',\n",
       " 'MED10',\n",
       " 'MRPL11',\n",
       " 'MRPL35',\n",
       " 'SNRNP200',\n",
       " 'SPDL1'}"
      ]
     },
     "execution_count": 497,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "JURKAT_genes.difference(K562_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "c3ebf653",
   "metadata": {},
   "outputs": [],
   "source": [
    "JURKAT_temp = db_inserts['counts_ref'].loc[db_inserts['counts_ref']['cell_line_origin'] == 'JURKAT']\n",
    "K562_temp = db_inserts['counts_ref'].loc[db_inserts['counts_ref']['cell_line_origin'] == 'K562']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "9c19d9f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "449"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(K562_temp['Gene 1'].tolist() + K562_temp['Gene 2'].tolist()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "93748b6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'NEGATIVE'}"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(K562_temp['Gene 1'].tolist() + K562_temp['Gene 2'].tolist()).difference(K562_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "957cc89d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "389"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(JURKAT_temp['Gene 1'].tolist() + JURKAT_temp['Gene 2'].tolist()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "fc43cffa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'NEGATIVE'}"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(JURKAT_temp['Gene 1'].tolist() + JURKAT_temp['Gene 2'].tolist()).difference(JURKAT_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "6f1bf17c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 1023\n",
      "Counts insert: 1693897\n",
      "Score insert: 175206\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2f287de",
   "metadata": {},
   "source": [
    "## Ito Data (HSC5, MeWo, MELJUSO, IPC298, HS936T, HS944T, PATU8988S, PK1, A549 GI1, MEL202)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27586724",
   "metadata": {},
   "source": [
    "## Seq Ref (Create from Counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "19e8073d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ito_loc = os.path.join(learning_goals_loc_general, \"Ito\")\n",
    "\n",
    "shared_data = pd.read_csv(os.path.join(ito_loc, \"Target_oligo_sequences_genes_ParalogV1_Counts.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "c15fec83",
   "metadata": {},
   "outputs": [],
   "source": [
    "## ITO Specific, fix the March genes\n",
    "change = {'11-Mar': 'MARCH11',\n",
    "          '1-Mar': 'MARCH1',\n",
    "         '2-Mar': 'MARCH2',\n",
    "         '3-Mar': 'MARCH3',\n",
    "         '4-Mar': 'MARCH4',\n",
    "         '8-Mar': 'MARCH8',\n",
    "         '9-Mar': 'MARCH9',\n",
    "         }\n",
    "\n",
    "genes_1_temp = shared_data['Aureus_gene'].tolist().copy()\n",
    "genes_2_temp = shared_data['Pyogenes_gene'].tolist().copy()\n",
    "\n",
    "for gene in change:\n",
    "    genes_1_temp = [i.replace(gene, change[gene]) for i in genes_1_temp]\n",
    "    genes_2_temp = [i.replace(gene, change[gene]) for i in genes_2_temp]\n",
    "\n",
    "shared_data['Aureus_gene'] = genes_1_temp\n",
    "shared_data['Pyogenes_gene'] = genes_2_temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "71f19701",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo_sequence</th>\n",
       "      <th>Left_guide</th>\n",
       "      <th>Right_guide</th>\n",
       "      <th>Aureus_gene</th>\n",
       "      <th>Pyogenes_gene</th>\n",
       "      <th>pDNA</th>\n",
       "      <th>Meljuso_RepA</th>\n",
       "      <th>Meljuso_RepB</th>\n",
       "      <th>Meljuso_RepC</th>\n",
       "      <th>TIGI1_004_RepA</th>\n",
       "      <th>...</th>\n",
       "      <th>A549_REPB</th>\n",
       "      <th>A549_REPC</th>\n",
       "      <th>HSC5_REPA</th>\n",
       "      <th>HSC5_REPB</th>\n",
       "      <th>HSC5_REPC</th>\n",
       "      <th>HS936T_REPA</th>\n",
       "      <th>HS936T_REPB</th>\n",
       "      <th>HS936T_REPC</th>\n",
       "      <th>PATU8988S_REPA</th>\n",
       "      <th>PATU8988S_REPB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54756</th>\n",
       "      <td>TATGGATACTGTACCTTCCGG_CTGCAAGTATGAGTTCATCA</td>\n",
       "      <td>TATGGATACTGTACCTTCCGG</td>\n",
       "      <td>CTGCAAGTATGAGTTCATCA</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>2577</td>\n",
       "      <td>803</td>\n",
       "      <td>532</td>\n",
       "      <td>823</td>\n",
       "      <td>593</td>\n",
       "      <td>...</td>\n",
       "      <td>743</td>\n",
       "      <td>550</td>\n",
       "      <td>1121</td>\n",
       "      <td>1052</td>\n",
       "      <td>1208</td>\n",
       "      <td>187</td>\n",
       "      <td>1927</td>\n",
       "      <td>653</td>\n",
       "      <td>863</td>\n",
       "      <td>3090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54757</th>\n",
       "      <td>TATGGATACTGTACCTTCCGG_GCTCATTGACCGTACTGCTG</td>\n",
       "      <td>TATGGATACTGTACCTTCCGG</td>\n",
       "      <td>GCTCATTGACCGTACTGCTG</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>2680</td>\n",
       "      <td>794</td>\n",
       "      <td>502</td>\n",
       "      <td>989</td>\n",
       "      <td>526</td>\n",
       "      <td>...</td>\n",
       "      <td>705</td>\n",
       "      <td>589</td>\n",
       "      <td>1051</td>\n",
       "      <td>997</td>\n",
       "      <td>1092</td>\n",
       "      <td>316</td>\n",
       "      <td>1315</td>\n",
       "      <td>382</td>\n",
       "      <td>748</td>\n",
       "      <td>2094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54758</th>\n",
       "      <td>TATGGATACTGTACCTTCCGG_TTCTATCACGCCATCCAGCC</td>\n",
       "      <td>TATGGATACTGTACCTTCCGG</td>\n",
       "      <td>TTCTATCACGCCATCCAGCC</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>1730</td>\n",
       "      <td>617</td>\n",
       "      <td>351</td>\n",
       "      <td>1003</td>\n",
       "      <td>810</td>\n",
       "      <td>...</td>\n",
       "      <td>622</td>\n",
       "      <td>602</td>\n",
       "      <td>979</td>\n",
       "      <td>849</td>\n",
       "      <td>1213</td>\n",
       "      <td>456</td>\n",
       "      <td>1887</td>\n",
       "      <td>337</td>\n",
       "      <td>573</td>\n",
       "      <td>1947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54759</th>\n",
       "      <td>GATCGACTTGCAGATCGCCCT_TTCTATCACGCCATCCAGCC</td>\n",
       "      <td>GATCGACTTGCAGATCGCCCT</td>\n",
       "      <td>TTCTATCACGCCATCCAGCC</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>1214</td>\n",
       "      <td>654</td>\n",
       "      <td>482</td>\n",
       "      <td>777</td>\n",
       "      <td>427</td>\n",
       "      <td>...</td>\n",
       "      <td>774</td>\n",
       "      <td>765</td>\n",
       "      <td>1022</td>\n",
       "      <td>842</td>\n",
       "      <td>1197</td>\n",
       "      <td>557</td>\n",
       "      <td>1487</td>\n",
       "      <td>214</td>\n",
       "      <td>386</td>\n",
       "      <td>1398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54760</th>\n",
       "      <td>GATCGACTTGCAGATCGCCCT_GCTCATTGACCGTACTGCTG</td>\n",
       "      <td>GATCGACTTGCAGATCGCCCT</td>\n",
       "      <td>GCTCATTGACCGTACTGCTG</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>1775</td>\n",
       "      <td>637</td>\n",
       "      <td>566</td>\n",
       "      <td>802</td>\n",
       "      <td>609</td>\n",
       "      <td>...</td>\n",
       "      <td>902</td>\n",
       "      <td>600</td>\n",
       "      <td>1081</td>\n",
       "      <td>855</td>\n",
       "      <td>1103</td>\n",
       "      <td>307</td>\n",
       "      <td>542</td>\n",
       "      <td>304</td>\n",
       "      <td>574</td>\n",
       "      <td>1879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54761</th>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA_TTCTATCACGCCATCCAGCC</td>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA</td>\n",
       "      <td>TTCTATCACGCCATCCAGCC</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>2473</td>\n",
       "      <td>1088</td>\n",
       "      <td>875</td>\n",
       "      <td>1560</td>\n",
       "      <td>569</td>\n",
       "      <td>...</td>\n",
       "      <td>1562</td>\n",
       "      <td>1371</td>\n",
       "      <td>1819</td>\n",
       "      <td>1555</td>\n",
       "      <td>1835</td>\n",
       "      <td>494</td>\n",
       "      <td>1739</td>\n",
       "      <td>876</td>\n",
       "      <td>800</td>\n",
       "      <td>2368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54762</th>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA_GCTCATTGACCGTACTGCTG</td>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA</td>\n",
       "      <td>GCTCATTGACCGTACTGCTG</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>3258</td>\n",
       "      <td>1152</td>\n",
       "      <td>1001</td>\n",
       "      <td>1506</td>\n",
       "      <td>977</td>\n",
       "      <td>...</td>\n",
       "      <td>2057</td>\n",
       "      <td>1299</td>\n",
       "      <td>1899</td>\n",
       "      <td>1606</td>\n",
       "      <td>1733</td>\n",
       "      <td>901</td>\n",
       "      <td>2017</td>\n",
       "      <td>571</td>\n",
       "      <td>1171</td>\n",
       "      <td>3908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54763</th>\n",
       "      <td>GATCGACTTGCAGATCGCCCT_CTGCAAGTATGAGTTCATCA</td>\n",
       "      <td>GATCGACTTGCAGATCGCCCT</td>\n",
       "      <td>CTGCAAGTATGAGTTCATCA</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>435</td>\n",
       "      <td>375</td>\n",
       "      <td>206</td>\n",
       "      <td>307</td>\n",
       "      <td>187</td>\n",
       "      <td>...</td>\n",
       "      <td>208</td>\n",
       "      <td>198</td>\n",
       "      <td>444</td>\n",
       "      <td>277</td>\n",
       "      <td>505</td>\n",
       "      <td>194</td>\n",
       "      <td>889</td>\n",
       "      <td>240</td>\n",
       "      <td>106</td>\n",
       "      <td>544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54764</th>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA_CTGCAAGTATGAGTTCATCA</td>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA</td>\n",
       "      <td>CTGCAAGTATGAGTTCATCA</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>MARCH8</td>\n",
       "      <td>2188</td>\n",
       "      <td>899</td>\n",
       "      <td>828</td>\n",
       "      <td>1006</td>\n",
       "      <td>481</td>\n",
       "      <td>...</td>\n",
       "      <td>1216</td>\n",
       "      <td>679</td>\n",
       "      <td>1286</td>\n",
       "      <td>1254</td>\n",
       "      <td>1269</td>\n",
       "      <td>306</td>\n",
       "      <td>2114</td>\n",
       "      <td>741</td>\n",
       "      <td>798</td>\n",
       "      <td>2483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101232</th>\n",
       "      <td>TATGGATACTGTACCTTCCGG_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>TATGGATACTGTACCTTCCGG</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>1169</td>\n",
       "      <td>358</td>\n",
       "      <td>263</td>\n",
       "      <td>401</td>\n",
       "      <td>393</td>\n",
       "      <td>...</td>\n",
       "      <td>123</td>\n",
       "      <td>187</td>\n",
       "      <td>537</td>\n",
       "      <td>297</td>\n",
       "      <td>489</td>\n",
       "      <td>373</td>\n",
       "      <td>780</td>\n",
       "      <td>475</td>\n",
       "      <td>376</td>\n",
       "      <td>1009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101233</th>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CGCAGTGTCCCAGTGCAGCGA</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>3615</td>\n",
       "      <td>1313</td>\n",
       "      <td>1244</td>\n",
       "      <td>1678</td>\n",
       "      <td>1230</td>\n",
       "      <td>...</td>\n",
       "      <td>771</td>\n",
       "      <td>747</td>\n",
       "      <td>1711</td>\n",
       "      <td>1296</td>\n",
       "      <td>1356</td>\n",
       "      <td>690</td>\n",
       "      <td>3074</td>\n",
       "      <td>653</td>\n",
       "      <td>990</td>\n",
       "      <td>3595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101234</th>\n",
       "      <td>GATCGACTTGCAGATCGCCCT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>GATCGACTTGCAGATCGCCCT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>MARCH1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>1567</td>\n",
       "      <td>619</td>\n",
       "      <td>499</td>\n",
       "      <td>866</td>\n",
       "      <td>469</td>\n",
       "      <td>...</td>\n",
       "      <td>651</td>\n",
       "      <td>371</td>\n",
       "      <td>1005</td>\n",
       "      <td>789</td>\n",
       "      <td>799</td>\n",
       "      <td>457</td>\n",
       "      <td>1807</td>\n",
       "      <td>114</td>\n",
       "      <td>332</td>\n",
       "      <td>1459</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows  38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Oligo_sequence             Left_guide  \\\n",
       "54756   TATGGATACTGTACCTTCCGG_CTGCAAGTATGAGTTCATCA  TATGGATACTGTACCTTCCGG   \n",
       "54757   TATGGATACTGTACCTTCCGG_GCTCATTGACCGTACTGCTG  TATGGATACTGTACCTTCCGG   \n",
       "54758   TATGGATACTGTACCTTCCGG_TTCTATCACGCCATCCAGCC  TATGGATACTGTACCTTCCGG   \n",
       "54759   GATCGACTTGCAGATCGCCCT_TTCTATCACGCCATCCAGCC  GATCGACTTGCAGATCGCCCT   \n",
       "54760   GATCGACTTGCAGATCGCCCT_GCTCATTGACCGTACTGCTG  GATCGACTTGCAGATCGCCCT   \n",
       "54761   CGCAGTGTCCCAGTGCAGCGA_TTCTATCACGCCATCCAGCC  CGCAGTGTCCCAGTGCAGCGA   \n",
       "54762   CGCAGTGTCCCAGTGCAGCGA_GCTCATTGACCGTACTGCTG  CGCAGTGTCCCAGTGCAGCGA   \n",
       "54763   GATCGACTTGCAGATCGCCCT_CTGCAAGTATGAGTTCATCA  GATCGACTTGCAGATCGCCCT   \n",
       "54764   CGCAGTGTCCCAGTGCAGCGA_CTGCAAGTATGAGTTCATCA  CGCAGTGTCCCAGTGCAGCGA   \n",
       "101232  TATGGATACTGTACCTTCCGG_AGGGAGACATCCGTCGGAGA  TATGGATACTGTACCTTCCGG   \n",
       "101233  CGCAGTGTCCCAGTGCAGCGA_AGGGAGACATCCGTCGGAGA  CGCAGTGTCCCAGTGCAGCGA   \n",
       "101234  GATCGACTTGCAGATCGCCCT_AGGGAGACATCCGTCGGAGA  GATCGACTTGCAGATCGCCCT   \n",
       "\n",
       "                 Right_guide Aureus_gene Pyogenes_gene  pDNA  Meljuso_RepA  \\\n",
       "54756   CTGCAAGTATGAGTTCATCA      MARCH1        MARCH8  2577           803   \n",
       "54757   GCTCATTGACCGTACTGCTG      MARCH1        MARCH8  2680           794   \n",
       "54758   TTCTATCACGCCATCCAGCC      MARCH1        MARCH8  1730           617   \n",
       "54759   TTCTATCACGCCATCCAGCC      MARCH1        MARCH8  1214           654   \n",
       "54760   GCTCATTGACCGTACTGCTG      MARCH1        MARCH8  1775           637   \n",
       "54761   TTCTATCACGCCATCCAGCC      MARCH1        MARCH8  2473          1088   \n",
       "54762   GCTCATTGACCGTACTGCTG      MARCH1        MARCH8  3258          1152   \n",
       "54763   CTGCAAGTATGAGTTCATCA      MARCH1        MARCH8   435           375   \n",
       "54764   CTGCAAGTATGAGTTCATCA      MARCH1        MARCH8  2188           899   \n",
       "101232  AGGGAGACATCCGTCGGAGA      MARCH1         AAVS1  1169           358   \n",
       "101233  AGGGAGACATCCGTCGGAGA      MARCH1         AAVS1  3615          1313   \n",
       "101234  AGGGAGACATCCGTCGGAGA      MARCH1         AAVS1  1567           619   \n",
       "\n",
       "        Meljuso_RepB  Meljuso_RepC  TIGI1_004_RepA  ...  A549_REPB  A549_REPC  \\\n",
       "54756            532           823             593  ...        743        550   \n",
       "54757            502           989             526  ...        705        589   \n",
       "54758            351          1003             810  ...        622        602   \n",
       "54759            482           777             427  ...        774        765   \n",
       "54760            566           802             609  ...        902        600   \n",
       "54761            875          1560             569  ...       1562       1371   \n",
       "54762           1001          1506             977  ...       2057       1299   \n",
       "54763            206           307             187  ...        208        198   \n",
       "54764            828          1006             481  ...       1216        679   \n",
       "101232           263           401             393  ...        123        187   \n",
       "101233          1244          1678            1230  ...        771        747   \n",
       "101234           499           866             469  ...        651        371   \n",
       "\n",
       "        HSC5_REPA  HSC5_REPB  HSC5_REPC  HS936T_REPA  HS936T_REPB  \\\n",
       "54756        1121       1052       1208          187         1927   \n",
       "54757        1051        997       1092          316         1315   \n",
       "54758         979        849       1213          456         1887   \n",
       "54759        1022        842       1197          557         1487   \n",
       "54760        1081        855       1103          307          542   \n",
       "54761        1819       1555       1835          494         1739   \n",
       "54762        1899       1606       1733          901         2017   \n",
       "54763         444        277        505          194          889   \n",
       "54764        1286       1254       1269          306         2114   \n",
       "101232        537        297        489          373          780   \n",
       "101233       1711       1296       1356          690         3074   \n",
       "101234       1005        789        799          457         1807   \n",
       "\n",
       "        HS936T_REPC  PATU8988S_REPA  PATU8988S_REPB  \n",
       "54756           653             863            3090  \n",
       "54757           382             748            2094  \n",
       "54758           337             573            1947  \n",
       "54759           214             386            1398  \n",
       "54760           304             574            1879  \n",
       "54761           876             800            2368  \n",
       "54762           571            1171            3908  \n",
       "54763           240             106             544  \n",
       "54764           741             798            2483  \n",
       "101232          475             376            1009  \n",
       "101233          653             990            3595  \n",
       "101234          114             332            1459  \n",
       "\n",
       "[12 rows x 38 columns]"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shared_data.loc[shared_data['Aureus_gene'] == 'MARCH1',]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "30500ffc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo_sequence</th>\n",
       "      <th>Left_guide</th>\n",
       "      <th>Right_guide</th>\n",
       "      <th>Aureus_gene</th>\n",
       "      <th>Pyogenes_gene</th>\n",
       "      <th>pDNA</th>\n",
       "      <th>Meljuso_RepA</th>\n",
       "      <th>Meljuso_RepB</th>\n",
       "      <th>Meljuso_RepC</th>\n",
       "      <th>TIGI1_004_RepA</th>\n",
       "      <th>...</th>\n",
       "      <th>A549_REPB</th>\n",
       "      <th>A549_REPC</th>\n",
       "      <th>HSC5_REPA</th>\n",
       "      <th>HSC5_REPB</th>\n",
       "      <th>HSC5_REPC</th>\n",
       "      <th>HS936T_REPA</th>\n",
       "      <th>HS936T_REPB</th>\n",
       "      <th>HS936T_REPC</th>\n",
       "      <th>PATU8988S_REPA</th>\n",
       "      <th>PATU8988S_REPB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG_AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>1286</td>\n",
       "      <td>367</td>\n",
       "      <td>370</td>\n",
       "      <td>567</td>\n",
       "      <td>398</td>\n",
       "      <td>...</td>\n",
       "      <td>635</td>\n",
       "      <td>528</td>\n",
       "      <td>830</td>\n",
       "      <td>542</td>\n",
       "      <td>642</td>\n",
       "      <td>142</td>\n",
       "      <td>828</td>\n",
       "      <td>153</td>\n",
       "      <td>304</td>\n",
       "      <td>1139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA_GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA</td>\n",
       "      <td>GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>3430</td>\n",
       "      <td>1434</td>\n",
       "      <td>1234</td>\n",
       "      <td>1926</td>\n",
       "      <td>1115</td>\n",
       "      <td>...</td>\n",
       "      <td>1794</td>\n",
       "      <td>1326</td>\n",
       "      <td>1935</td>\n",
       "      <td>1471</td>\n",
       "      <td>1983</td>\n",
       "      <td>850</td>\n",
       "      <td>2335</td>\n",
       "      <td>1019</td>\n",
       "      <td>1118</td>\n",
       "      <td>3760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA_AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2002</td>\n",
       "      <td>770</td>\n",
       "      <td>596</td>\n",
       "      <td>959</td>\n",
       "      <td>603</td>\n",
       "      <td>...</td>\n",
       "      <td>1125</td>\n",
       "      <td>1007</td>\n",
       "      <td>1088</td>\n",
       "      <td>1078</td>\n",
       "      <td>1482</td>\n",
       "      <td>216</td>\n",
       "      <td>1792</td>\n",
       "      <td>811</td>\n",
       "      <td>589</td>\n",
       "      <td>1779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAGGGTGCCGAACAGCGGAGT_ACACGTAGTACATCACGCTC</td>\n",
       "      <td>CAGGGTGCCGAACAGCGGAGT</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>4822</td>\n",
       "      <td>1976</td>\n",
       "      <td>1329</td>\n",
       "      <td>2260</td>\n",
       "      <td>871</td>\n",
       "      <td>...</td>\n",
       "      <td>2116</td>\n",
       "      <td>1642</td>\n",
       "      <td>1651</td>\n",
       "      <td>1144</td>\n",
       "      <td>1810</td>\n",
       "      <td>514</td>\n",
       "      <td>2593</td>\n",
       "      <td>1099</td>\n",
       "      <td>1414</td>\n",
       "      <td>3356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG_ACACGTAGTACATCACGCTC</td>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2103</td>\n",
       "      <td>805</td>\n",
       "      <td>581</td>\n",
       "      <td>883</td>\n",
       "      <td>246</td>\n",
       "      <td>...</td>\n",
       "      <td>634</td>\n",
       "      <td>573</td>\n",
       "      <td>671</td>\n",
       "      <td>563</td>\n",
       "      <td>919</td>\n",
       "      <td>513</td>\n",
       "      <td>876</td>\n",
       "      <td>440</td>\n",
       "      <td>425</td>\n",
       "      <td>1677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110869</th>\n",
       "      <td>CACAGAAGTGACGTCAATAGT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CACAGAAGTGACGTCAATAGT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>2175</td>\n",
       "      <td>1237</td>\n",
       "      <td>1027</td>\n",
       "      <td>997</td>\n",
       "      <td>350</td>\n",
       "      <td>...</td>\n",
       "      <td>898</td>\n",
       "      <td>828</td>\n",
       "      <td>1068</td>\n",
       "      <td>832</td>\n",
       "      <td>1008</td>\n",
       "      <td>276</td>\n",
       "      <td>1672</td>\n",
       "      <td>619</td>\n",
       "      <td>728</td>\n",
       "      <td>2029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110870</th>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>1813</td>\n",
       "      <td>910</td>\n",
       "      <td>858</td>\n",
       "      <td>1193</td>\n",
       "      <td>731</td>\n",
       "      <td>...</td>\n",
       "      <td>447</td>\n",
       "      <td>507</td>\n",
       "      <td>1158</td>\n",
       "      <td>693</td>\n",
       "      <td>1030</td>\n",
       "      <td>507</td>\n",
       "      <td>2091</td>\n",
       "      <td>426</td>\n",
       "      <td>433</td>\n",
       "      <td>1581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110871</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>2343</td>\n",
       "      <td>812</td>\n",
       "      <td>609</td>\n",
       "      <td>1039</td>\n",
       "      <td>637</td>\n",
       "      <td>...</td>\n",
       "      <td>1039</td>\n",
       "      <td>749</td>\n",
       "      <td>992</td>\n",
       "      <td>711</td>\n",
       "      <td>978</td>\n",
       "      <td>229</td>\n",
       "      <td>2151</td>\n",
       "      <td>711</td>\n",
       "      <td>875</td>\n",
       "      <td>2359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110872</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_GAGCTTGTAGGACGTTGTGA</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>GAGCTTGTAGGACGTTGTGA</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>1688</td>\n",
       "      <td>371</td>\n",
       "      <td>355</td>\n",
       "      <td>534</td>\n",
       "      <td>453</td>\n",
       "      <td>...</td>\n",
       "      <td>760</td>\n",
       "      <td>445</td>\n",
       "      <td>492</td>\n",
       "      <td>350</td>\n",
       "      <td>442</td>\n",
       "      <td>241</td>\n",
       "      <td>675</td>\n",
       "      <td>243</td>\n",
       "      <td>564</td>\n",
       "      <td>1499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110873</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_AATGTACAGACTGCTACGAG</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>AATGTACAGACTGCTACGAG</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>970</td>\n",
       "      <td>271</td>\n",
       "      <td>227</td>\n",
       "      <td>299</td>\n",
       "      <td>115</td>\n",
       "      <td>...</td>\n",
       "      <td>436</td>\n",
       "      <td>391</td>\n",
       "      <td>406</td>\n",
       "      <td>275</td>\n",
       "      <td>416</td>\n",
       "      <td>278</td>\n",
       "      <td>1050</td>\n",
       "      <td>287</td>\n",
       "      <td>256</td>\n",
       "      <td>693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110874 rows  38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Oligo_sequence             Left_guide  \\\n",
       "0       GCCAAAGGTGCTGACACCGTG_AAATAATGGGAGCCCCCCAG  GCCAAAGGTGCTGACACCGTG   \n",
       "1       AGGACGTGTCCATGCGCCGCA_GCGCTGGCAAGACGTGTCGA  AGGACGTGTCCATGCGCCGCA   \n",
       "2       AGGACGTGTCCATGCGCCGCA_AAATAATGGGAGCCCCCCAG  AGGACGTGTCCATGCGCCGCA   \n",
       "3       CAGGGTGCCGAACAGCGGAGT_ACACGTAGTACATCACGCTC  CAGGGTGCCGAACAGCGGAGT   \n",
       "4       GCCAAAGGTGCTGACACCGTG_ACACGTAGTACATCACGCTC  GCCAAAGGTGCTGACACCGTG   \n",
       "...                                            ...                    ...   \n",
       "110869  CACAGAAGTGACGTCAATAGT_AGGGAGACATCCGTCGGAGA  CACAGAAGTGACGTCAATAGT   \n",
       "110870  CCCACCTAGGCACAAAGCCGT_AGGGAGACATCCGTCGGAGA  CCCACCTAGGCACAAAGCCGT   \n",
       "110871  CTGAGCGGATCCTCCCCGTGT_ACCTGCAGAACCACTAGCAT  CTGAGCGGATCCTCCCCGTGT   \n",
       "110872  CTGAGCGGATCCTCCCCGTGT_GAGCTTGTAGGACGTTGTGA  CTGAGCGGATCCTCCCCGTGT   \n",
       "110873  CTGAGCGGATCCTCCCCGTGT_AATGTACAGACTGCTACGAG  CTGAGCGGATCCTCCCCGTGT   \n",
       "\n",
       "                 Right_guide Aureus_gene Pyogenes_gene  pDNA  Meljuso_RepA  \\\n",
       "0       AAATAATGGGAGCCCCCCAG         ABO       A3GALT2  1286           367   \n",
       "1       GCGCTGGCAAGACGTGTCGA         ABO       A3GALT2  3430          1434   \n",
       "2       AAATAATGGGAGCCCCCCAG         ABO       A3GALT2  2002           770   \n",
       "3       ACACGTAGTACATCACGCTC         ABO       A3GALT2  4822          1976   \n",
       "4       ACACGTAGTACATCACGCTC         ABO       A3GALT2  2103           805   \n",
       "...                      ...         ...           ...   ...           ...   \n",
       "110869  AGGGAGACATCCGTCGGAGA        ZXDC         AAVS1  2175          1237   \n",
       "110870  AGGGAGACATCCGTCGGAGA        ZXDC         AAVS1  1813           910   \n",
       "110871  ACCTGCAGAACCACTAGCAT       AAVS1          ZXDC  2343           812   \n",
       "110872  GAGCTTGTAGGACGTTGTGA       AAVS1          ZXDC  1688           371   \n",
       "110873  AATGTACAGACTGCTACGAG       AAVS1          ZXDC   970           271   \n",
       "\n",
       "        Meljuso_RepB  Meljuso_RepC  TIGI1_004_RepA  ...  A549_REPB  A549_REPC  \\\n",
       "0                370           567             398  ...        635        528   \n",
       "1               1234          1926            1115  ...       1794       1326   \n",
       "2                596           959             603  ...       1125       1007   \n",
       "3               1329          2260             871  ...       2116       1642   \n",
       "4                581           883             246  ...        634        573   \n",
       "...              ...           ...             ...  ...        ...        ...   \n",
       "110869          1027           997             350  ...        898        828   \n",
       "110870           858          1193             731  ...        447        507   \n",
       "110871           609          1039             637  ...       1039        749   \n",
       "110872           355           534             453  ...        760        445   \n",
       "110873           227           299             115  ...        436        391   \n",
       "\n",
       "        HSC5_REPA  HSC5_REPB  HSC5_REPC  HS936T_REPA  HS936T_REPB  \\\n",
       "0             830        542        642          142          828   \n",
       "1            1935       1471       1983          850         2335   \n",
       "2            1088       1078       1482          216         1792   \n",
       "3            1651       1144       1810          514         2593   \n",
       "4             671        563        919          513          876   \n",
       "...           ...        ...        ...          ...          ...   \n",
       "110869       1068        832       1008          276         1672   \n",
       "110870       1158        693       1030          507         2091   \n",
       "110871        992        711        978          229         2151   \n",
       "110872        492        350        442          241          675   \n",
       "110873        406        275        416          278         1050   \n",
       "\n",
       "        HS936T_REPC  PATU8988S_REPA  PATU8988S_REPB  \n",
       "0               153             304            1139  \n",
       "1              1019            1118            3760  \n",
       "2               811             589            1779  \n",
       "3              1099            1414            3356  \n",
       "4               440             425            1677  \n",
       "...             ...             ...             ...  \n",
       "110869          619             728            2029  \n",
       "110870          426             433            1581  \n",
       "110871          711             875            2359  \n",
       "110872          243             564            1499  \n",
       "110873          287             256             693  \n",
       "\n",
       "[110874 rows x 38 columns]"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shared_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "c436faa5",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_to_guide = {}\n",
    "target_counts = {}\n",
    "for i in range(shared_data.shape[0]):\n",
    "    \n",
    "    guide_1_seq = shared_data.loc[i, 'Left_guide']\n",
    "    guide_2_seq = shared_data.loc[i, 'Right_guide']\n",
    "    \n",
    "    target_1 = shared_data.loc[i, 'Aureus_gene']\n",
    "    target_2 = shared_data.loc[i, 'Pyogenes_gene']\n",
    "    \n",
    "    if (guide_1_seq not in seq_to_guide):\n",
    "        if (target_1 not in target_counts):\n",
    "            target_counts[target_1] = 1\n",
    "        else:\n",
    "            target_counts[target_1] += 1\n",
    "        seq_to_guide[guide_1_seq] = '_'.join([target_1, str(target_counts[target_1])])\n",
    "    if (guide_2_seq not in seq_to_guide):\n",
    "        if (target_2 not in target_counts):\n",
    "            target_counts[target_2] = 1\n",
    "        else:\n",
    "            target_counts[target_2] += 1\n",
    "        seq_to_guide[guide_2_seq] = '_'.join([target_2, str(target_counts[target_2])])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "73face7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MARCH1_3'"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_to_guide['CGCAGTGTCCCAGTGCAGCGA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "f841323e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19706"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(seq_to_guide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "a5403bd5",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_list = []\n",
    "guide_list = []\n",
    "\n",
    "for seq in seq_to_guide:\n",
    "    seq_list.append(seq)\n",
    "    guide_list.append(seq_to_guide[seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "d98bc5d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'MARCH1_1' in guide_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "7be5e032",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = pd.DataFrame({'sgRNA_guide_name': guide_list,\n",
    "                             'sgRNA_guide_seq': seq_list})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "365bdf78",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['sgRNA_target_name'] = [i.split('_')[0] for i in sequence_ref['sgRNA_guide_name']]\n",
    "sequence_ref.loc[sequence_ref['sgRNA_target_name'] == 'AAVS1', 'sgRNA_target_name'] = 'control'\n",
    "sequence_ref.sort_values('sgRNA_target_name', inplace = True, ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "77674138",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2_1</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2_2</td>\n",
       "      <td>GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2_3</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2_4</td>\n",
       "      <td>GCAAAGATAGTCAGCCCAATG</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2_5</td>\n",
       "      <td>GGCCGAAGGGCAGACGCCCAT</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19701</th>\n",
       "      <td>ZXDC_4</td>\n",
       "      <td>ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>ZXDC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19702</th>\n",
       "      <td>ZXDC_3</td>\n",
       "      <td>CTCTTACCTCAGCTAGAAGCT</td>\n",
       "      <td>ZXDC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19703</th>\n",
       "      <td>ZXDC_2</td>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT</td>\n",
       "      <td>ZXDC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19704</th>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19705</th>\n",
       "      <td>AAVS1_2</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19706 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sgRNA_guide_name        sgRNA_guide_seq sgRNA_target_name\n",
       "0            A3GALT2_1   AAATAATGGGAGCCCCCCAG           A3GALT2\n",
       "1            A3GALT2_2   GCGCTGGCAAGACGTGTCGA           A3GALT2\n",
       "2            A3GALT2_3   ACACGTAGTACATCACGCTC           A3GALT2\n",
       "3            A3GALT2_4  GCAAAGATAGTCAGCCCAATG           A3GALT2\n",
       "4            A3GALT2_5  GGCCGAAGGGCAGACGCCCAT           A3GALT2\n",
       "...                ...                    ...               ...\n",
       "19701           ZXDC_4   ACCTGCAGAACCACTAGCAT              ZXDC\n",
       "19702           ZXDC_3  CTCTTACCTCAGCTAGAAGCT              ZXDC\n",
       "19703           ZXDC_2  CCCACCTAGGCACAAAGCCGT              ZXDC\n",
       "19704          AAVS1_1  CTGAGCGGATCCTCCCCGTGT           control\n",
       "19705          AAVS1_2   AGGGAGACATCCGTCGGAGA           control\n",
       "\n",
       "[19706 rows x 3 columns]"
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f35d13fd",
   "metadata": {},
   "source": [
    "## Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "3e5a3f02",
   "metadata": {},
   "outputs": [],
   "source": [
    "#counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])\n",
    "#counts_ref = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "deabd860",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo_sequence</th>\n",
       "      <th>Left_guide</th>\n",
       "      <th>Right_guide</th>\n",
       "      <th>Aureus_gene</th>\n",
       "      <th>Pyogenes_gene</th>\n",
       "      <th>pDNA</th>\n",
       "      <th>Meljuso_RepA</th>\n",
       "      <th>Meljuso_RepB</th>\n",
       "      <th>Meljuso_RepC</th>\n",
       "      <th>TIGI1_004_RepA</th>\n",
       "      <th>...</th>\n",
       "      <th>A549_REPB</th>\n",
       "      <th>A549_REPC</th>\n",
       "      <th>HSC5_REPA</th>\n",
       "      <th>HSC5_REPB</th>\n",
       "      <th>HSC5_REPC</th>\n",
       "      <th>HS936T_REPA</th>\n",
       "      <th>HS936T_REPB</th>\n",
       "      <th>HS936T_REPC</th>\n",
       "      <th>PATU8988S_REPA</th>\n",
       "      <th>PATU8988S_REPB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG_AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>1286</td>\n",
       "      <td>367</td>\n",
       "      <td>370</td>\n",
       "      <td>567</td>\n",
       "      <td>398</td>\n",
       "      <td>...</td>\n",
       "      <td>635</td>\n",
       "      <td>528</td>\n",
       "      <td>830</td>\n",
       "      <td>542</td>\n",
       "      <td>642</td>\n",
       "      <td>142</td>\n",
       "      <td>828</td>\n",
       "      <td>153</td>\n",
       "      <td>304</td>\n",
       "      <td>1139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA_GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA</td>\n",
       "      <td>GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>3430</td>\n",
       "      <td>1434</td>\n",
       "      <td>1234</td>\n",
       "      <td>1926</td>\n",
       "      <td>1115</td>\n",
       "      <td>...</td>\n",
       "      <td>1794</td>\n",
       "      <td>1326</td>\n",
       "      <td>1935</td>\n",
       "      <td>1471</td>\n",
       "      <td>1983</td>\n",
       "      <td>850</td>\n",
       "      <td>2335</td>\n",
       "      <td>1019</td>\n",
       "      <td>1118</td>\n",
       "      <td>3760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA_AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2002</td>\n",
       "      <td>770</td>\n",
       "      <td>596</td>\n",
       "      <td>959</td>\n",
       "      <td>603</td>\n",
       "      <td>...</td>\n",
       "      <td>1125</td>\n",
       "      <td>1007</td>\n",
       "      <td>1088</td>\n",
       "      <td>1078</td>\n",
       "      <td>1482</td>\n",
       "      <td>216</td>\n",
       "      <td>1792</td>\n",
       "      <td>811</td>\n",
       "      <td>589</td>\n",
       "      <td>1779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAGGGTGCCGAACAGCGGAGT_ACACGTAGTACATCACGCTC</td>\n",
       "      <td>CAGGGTGCCGAACAGCGGAGT</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>4822</td>\n",
       "      <td>1976</td>\n",
       "      <td>1329</td>\n",
       "      <td>2260</td>\n",
       "      <td>871</td>\n",
       "      <td>...</td>\n",
       "      <td>2116</td>\n",
       "      <td>1642</td>\n",
       "      <td>1651</td>\n",
       "      <td>1144</td>\n",
       "      <td>1810</td>\n",
       "      <td>514</td>\n",
       "      <td>2593</td>\n",
       "      <td>1099</td>\n",
       "      <td>1414</td>\n",
       "      <td>3356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG_ACACGTAGTACATCACGCTC</td>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2103</td>\n",
       "      <td>805</td>\n",
       "      <td>581</td>\n",
       "      <td>883</td>\n",
       "      <td>246</td>\n",
       "      <td>...</td>\n",
       "      <td>634</td>\n",
       "      <td>573</td>\n",
       "      <td>671</td>\n",
       "      <td>563</td>\n",
       "      <td>919</td>\n",
       "      <td>513</td>\n",
       "      <td>876</td>\n",
       "      <td>440</td>\n",
       "      <td>425</td>\n",
       "      <td>1677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110869</th>\n",
       "      <td>CACAGAAGTGACGTCAATAGT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CACAGAAGTGACGTCAATAGT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>2175</td>\n",
       "      <td>1237</td>\n",
       "      <td>1027</td>\n",
       "      <td>997</td>\n",
       "      <td>350</td>\n",
       "      <td>...</td>\n",
       "      <td>898</td>\n",
       "      <td>828</td>\n",
       "      <td>1068</td>\n",
       "      <td>832</td>\n",
       "      <td>1008</td>\n",
       "      <td>276</td>\n",
       "      <td>1672</td>\n",
       "      <td>619</td>\n",
       "      <td>728</td>\n",
       "      <td>2029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110870</th>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>1813</td>\n",
       "      <td>910</td>\n",
       "      <td>858</td>\n",
       "      <td>1193</td>\n",
       "      <td>731</td>\n",
       "      <td>...</td>\n",
       "      <td>447</td>\n",
       "      <td>507</td>\n",
       "      <td>1158</td>\n",
       "      <td>693</td>\n",
       "      <td>1030</td>\n",
       "      <td>507</td>\n",
       "      <td>2091</td>\n",
       "      <td>426</td>\n",
       "      <td>433</td>\n",
       "      <td>1581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110871</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>2343</td>\n",
       "      <td>812</td>\n",
       "      <td>609</td>\n",
       "      <td>1039</td>\n",
       "      <td>637</td>\n",
       "      <td>...</td>\n",
       "      <td>1039</td>\n",
       "      <td>749</td>\n",
       "      <td>992</td>\n",
       "      <td>711</td>\n",
       "      <td>978</td>\n",
       "      <td>229</td>\n",
       "      <td>2151</td>\n",
       "      <td>711</td>\n",
       "      <td>875</td>\n",
       "      <td>2359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110872</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_GAGCTTGTAGGACGTTGTGA</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>GAGCTTGTAGGACGTTGTGA</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>1688</td>\n",
       "      <td>371</td>\n",
       "      <td>355</td>\n",
       "      <td>534</td>\n",
       "      <td>453</td>\n",
       "      <td>...</td>\n",
       "      <td>760</td>\n",
       "      <td>445</td>\n",
       "      <td>492</td>\n",
       "      <td>350</td>\n",
       "      <td>442</td>\n",
       "      <td>241</td>\n",
       "      <td>675</td>\n",
       "      <td>243</td>\n",
       "      <td>564</td>\n",
       "      <td>1499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110873</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_AATGTACAGACTGCTACGAG</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>AATGTACAGACTGCTACGAG</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>970</td>\n",
       "      <td>271</td>\n",
       "      <td>227</td>\n",
       "      <td>299</td>\n",
       "      <td>115</td>\n",
       "      <td>...</td>\n",
       "      <td>436</td>\n",
       "      <td>391</td>\n",
       "      <td>406</td>\n",
       "      <td>275</td>\n",
       "      <td>416</td>\n",
       "      <td>278</td>\n",
       "      <td>1050</td>\n",
       "      <td>287</td>\n",
       "      <td>256</td>\n",
       "      <td>693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110874 rows  38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Oligo_sequence             Left_guide  \\\n",
       "0       GCCAAAGGTGCTGACACCGTG_AAATAATGGGAGCCCCCCAG  GCCAAAGGTGCTGACACCGTG   \n",
       "1       AGGACGTGTCCATGCGCCGCA_GCGCTGGCAAGACGTGTCGA  AGGACGTGTCCATGCGCCGCA   \n",
       "2       AGGACGTGTCCATGCGCCGCA_AAATAATGGGAGCCCCCCAG  AGGACGTGTCCATGCGCCGCA   \n",
       "3       CAGGGTGCCGAACAGCGGAGT_ACACGTAGTACATCACGCTC  CAGGGTGCCGAACAGCGGAGT   \n",
       "4       GCCAAAGGTGCTGACACCGTG_ACACGTAGTACATCACGCTC  GCCAAAGGTGCTGACACCGTG   \n",
       "...                                            ...                    ...   \n",
       "110869  CACAGAAGTGACGTCAATAGT_AGGGAGACATCCGTCGGAGA  CACAGAAGTGACGTCAATAGT   \n",
       "110870  CCCACCTAGGCACAAAGCCGT_AGGGAGACATCCGTCGGAGA  CCCACCTAGGCACAAAGCCGT   \n",
       "110871  CTGAGCGGATCCTCCCCGTGT_ACCTGCAGAACCACTAGCAT  CTGAGCGGATCCTCCCCGTGT   \n",
       "110872  CTGAGCGGATCCTCCCCGTGT_GAGCTTGTAGGACGTTGTGA  CTGAGCGGATCCTCCCCGTGT   \n",
       "110873  CTGAGCGGATCCTCCCCGTGT_AATGTACAGACTGCTACGAG  CTGAGCGGATCCTCCCCGTGT   \n",
       "\n",
       "                 Right_guide Aureus_gene Pyogenes_gene  pDNA  Meljuso_RepA  \\\n",
       "0       AAATAATGGGAGCCCCCCAG         ABO       A3GALT2  1286           367   \n",
       "1       GCGCTGGCAAGACGTGTCGA         ABO       A3GALT2  3430          1434   \n",
       "2       AAATAATGGGAGCCCCCCAG         ABO       A3GALT2  2002           770   \n",
       "3       ACACGTAGTACATCACGCTC         ABO       A3GALT2  4822          1976   \n",
       "4       ACACGTAGTACATCACGCTC         ABO       A3GALT2  2103           805   \n",
       "...                      ...         ...           ...   ...           ...   \n",
       "110869  AGGGAGACATCCGTCGGAGA        ZXDC         AAVS1  2175          1237   \n",
       "110870  AGGGAGACATCCGTCGGAGA        ZXDC         AAVS1  1813           910   \n",
       "110871  ACCTGCAGAACCACTAGCAT       AAVS1          ZXDC  2343           812   \n",
       "110872  GAGCTTGTAGGACGTTGTGA       AAVS1          ZXDC  1688           371   \n",
       "110873  AATGTACAGACTGCTACGAG       AAVS1          ZXDC   970           271   \n",
       "\n",
       "        Meljuso_RepB  Meljuso_RepC  TIGI1_004_RepA  ...  A549_REPB  A549_REPC  \\\n",
       "0                370           567             398  ...        635        528   \n",
       "1               1234          1926            1115  ...       1794       1326   \n",
       "2                596           959             603  ...       1125       1007   \n",
       "3               1329          2260             871  ...       2116       1642   \n",
       "4                581           883             246  ...        634        573   \n",
       "...              ...           ...             ...  ...        ...        ...   \n",
       "110869          1027           997             350  ...        898        828   \n",
       "110870           858          1193             731  ...        447        507   \n",
       "110871           609          1039             637  ...       1039        749   \n",
       "110872           355           534             453  ...        760        445   \n",
       "110873           227           299             115  ...        436        391   \n",
       "\n",
       "        HSC5_REPA  HSC5_REPB  HSC5_REPC  HS936T_REPA  HS936T_REPB  \\\n",
       "0             830        542        642          142          828   \n",
       "1            1935       1471       1983          850         2335   \n",
       "2            1088       1078       1482          216         1792   \n",
       "3            1651       1144       1810          514         2593   \n",
       "4             671        563        919          513          876   \n",
       "...           ...        ...        ...          ...          ...   \n",
       "110869       1068        832       1008          276         1672   \n",
       "110870       1158        693       1030          507         2091   \n",
       "110871        992        711        978          229         2151   \n",
       "110872        492        350        442          241          675   \n",
       "110873        406        275        416          278         1050   \n",
       "\n",
       "        HS936T_REPC  PATU8988S_REPA  PATU8988S_REPB  \n",
       "0               153             304            1139  \n",
       "1              1019            1118            3760  \n",
       "2               811             589            1779  \n",
       "3              1099            1414            3356  \n",
       "4               440             425            1677  \n",
       "...             ...             ...             ...  \n",
       "110869          619             728            2029  \n",
       "110870          426             433            1581  \n",
       "110871          711             875            2359  \n",
       "110872          243             564            1499  \n",
       "110873          287             256             693  \n",
       "\n",
       "[110874 rows x 38 columns]"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shared_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "197f8045",
   "metadata": {},
   "outputs": [],
   "source": [
    "# capitalize CL data and make sure the cell line matches the reported data\n",
    "temp = shared_data.columns.tolist()\n",
    "temp[6:] = [i.upper().replace('HS944', 'HS944T').replace('TI', '').replace('_004', '').replace('_003', '') for i in temp[6:]]\n",
    "shared_data.columns = temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "a535c4da",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get all cell lines\n",
    "cell_lines = set(item.split('_')[0] for item in shared_data.columns[6:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "27c40eca",
   "metadata": {},
   "outputs": [],
   "source": [
    "shared_data['Guide 1'] = [seq_to_guide[seq] for seq in shared_data['Left_guide']]\n",
    "shared_data['Guide 2'] = [seq_to_guide[seq] for seq in shared_data['Right_guide']]\n",
    "shared_data['Gene 1'] = shared_data['Aureus_gene']\n",
    "shared_data['Gene 2'] = shared_data['Pyogenes_gene']\n",
    "shared_data['Count Replicates'] = None\n",
    "shared_data['Cell Line'] = None\n",
    "shared_data['Condition'] = None\n",
    "shared_data['Type'] = None\n",
    "shared_data['Sequencing'] = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "404fdf34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo_sequence</th>\n",
       "      <th>Left_guide</th>\n",
       "      <th>Right_guide</th>\n",
       "      <th>Aureus_gene</th>\n",
       "      <th>Pyogenes_gene</th>\n",
       "      <th>pDNA</th>\n",
       "      <th>MELJUSO_REPA</th>\n",
       "      <th>MELJUSO_REPB</th>\n",
       "      <th>MELJUSO_REPC</th>\n",
       "      <th>GI1_REPA</th>\n",
       "      <th>...</th>\n",
       "      <th>PATU8988S_REPB</th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG_AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>1286</td>\n",
       "      <td>367</td>\n",
       "      <td>370</td>\n",
       "      <td>567</td>\n",
       "      <td>398</td>\n",
       "      <td>...</td>\n",
       "      <td>1139</td>\n",
       "      <td>ABO_1</td>\n",
       "      <td>A3GALT2_1</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA_GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA</td>\n",
       "      <td>GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>3430</td>\n",
       "      <td>1434</td>\n",
       "      <td>1234</td>\n",
       "      <td>1926</td>\n",
       "      <td>1115</td>\n",
       "      <td>...</td>\n",
       "      <td>3760</td>\n",
       "      <td>ABO_2</td>\n",
       "      <td>A3GALT2_2</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA_AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>AGGACGTGTCCATGCGCCGCA</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2002</td>\n",
       "      <td>770</td>\n",
       "      <td>596</td>\n",
       "      <td>959</td>\n",
       "      <td>603</td>\n",
       "      <td>...</td>\n",
       "      <td>1779</td>\n",
       "      <td>ABO_2</td>\n",
       "      <td>A3GALT2_1</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAGGGTGCCGAACAGCGGAGT_ACACGTAGTACATCACGCTC</td>\n",
       "      <td>CAGGGTGCCGAACAGCGGAGT</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>4822</td>\n",
       "      <td>1976</td>\n",
       "      <td>1329</td>\n",
       "      <td>2260</td>\n",
       "      <td>871</td>\n",
       "      <td>...</td>\n",
       "      <td>3356</td>\n",
       "      <td>ABO_3</td>\n",
       "      <td>A3GALT2_3</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG_ACACGTAGTACATCACGCTC</td>\n",
       "      <td>GCCAAAGGTGCTGACACCGTG</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2103</td>\n",
       "      <td>805</td>\n",
       "      <td>581</td>\n",
       "      <td>883</td>\n",
       "      <td>246</td>\n",
       "      <td>...</td>\n",
       "      <td>1677</td>\n",
       "      <td>ABO_1</td>\n",
       "      <td>A3GALT2_3</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110869</th>\n",
       "      <td>CACAGAAGTGACGTCAATAGT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CACAGAAGTGACGTCAATAGT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>2175</td>\n",
       "      <td>1237</td>\n",
       "      <td>1027</td>\n",
       "      <td>997</td>\n",
       "      <td>350</td>\n",
       "      <td>...</td>\n",
       "      <td>2029</td>\n",
       "      <td>ZXDC_1</td>\n",
       "      <td>AAVS1_2</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110870</th>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT_AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>1813</td>\n",
       "      <td>910</td>\n",
       "      <td>858</td>\n",
       "      <td>1193</td>\n",
       "      <td>731</td>\n",
       "      <td>...</td>\n",
       "      <td>1581</td>\n",
       "      <td>ZXDC_2</td>\n",
       "      <td>AAVS1_2</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110871</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>2343</td>\n",
       "      <td>812</td>\n",
       "      <td>609</td>\n",
       "      <td>1039</td>\n",
       "      <td>637</td>\n",
       "      <td>...</td>\n",
       "      <td>2359</td>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>ZXDC_4</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110872</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_GAGCTTGTAGGACGTTGTGA</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>GAGCTTGTAGGACGTTGTGA</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>1688</td>\n",
       "      <td>371</td>\n",
       "      <td>355</td>\n",
       "      <td>534</td>\n",
       "      <td>453</td>\n",
       "      <td>...</td>\n",
       "      <td>1499</td>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>ZXDC_6</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110873</th>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT_AATGTACAGACTGCTACGAG</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>AATGTACAGACTGCTACGAG</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>970</td>\n",
       "      <td>271</td>\n",
       "      <td>227</td>\n",
       "      <td>299</td>\n",
       "      <td>115</td>\n",
       "      <td>...</td>\n",
       "      <td>693</td>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>ZXDC_5</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110874 rows  47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Oligo_sequence             Left_guide  \\\n",
       "0       GCCAAAGGTGCTGACACCGTG_AAATAATGGGAGCCCCCCAG  GCCAAAGGTGCTGACACCGTG   \n",
       "1       AGGACGTGTCCATGCGCCGCA_GCGCTGGCAAGACGTGTCGA  AGGACGTGTCCATGCGCCGCA   \n",
       "2       AGGACGTGTCCATGCGCCGCA_AAATAATGGGAGCCCCCCAG  AGGACGTGTCCATGCGCCGCA   \n",
       "3       CAGGGTGCCGAACAGCGGAGT_ACACGTAGTACATCACGCTC  CAGGGTGCCGAACAGCGGAGT   \n",
       "4       GCCAAAGGTGCTGACACCGTG_ACACGTAGTACATCACGCTC  GCCAAAGGTGCTGACACCGTG   \n",
       "...                                            ...                    ...   \n",
       "110869  CACAGAAGTGACGTCAATAGT_AGGGAGACATCCGTCGGAGA  CACAGAAGTGACGTCAATAGT   \n",
       "110870  CCCACCTAGGCACAAAGCCGT_AGGGAGACATCCGTCGGAGA  CCCACCTAGGCACAAAGCCGT   \n",
       "110871  CTGAGCGGATCCTCCCCGTGT_ACCTGCAGAACCACTAGCAT  CTGAGCGGATCCTCCCCGTGT   \n",
       "110872  CTGAGCGGATCCTCCCCGTGT_GAGCTTGTAGGACGTTGTGA  CTGAGCGGATCCTCCCCGTGT   \n",
       "110873  CTGAGCGGATCCTCCCCGTGT_AATGTACAGACTGCTACGAG  CTGAGCGGATCCTCCCCGTGT   \n",
       "\n",
       "                 Right_guide Aureus_gene Pyogenes_gene  pDNA  MELJUSO_REPA  \\\n",
       "0       AAATAATGGGAGCCCCCCAG         ABO       A3GALT2  1286           367   \n",
       "1       GCGCTGGCAAGACGTGTCGA         ABO       A3GALT2  3430          1434   \n",
       "2       AAATAATGGGAGCCCCCCAG         ABO       A3GALT2  2002           770   \n",
       "3       ACACGTAGTACATCACGCTC         ABO       A3GALT2  4822          1976   \n",
       "4       ACACGTAGTACATCACGCTC         ABO       A3GALT2  2103           805   \n",
       "...                      ...         ...           ...   ...           ...   \n",
       "110869  AGGGAGACATCCGTCGGAGA        ZXDC         AAVS1  2175          1237   \n",
       "110870  AGGGAGACATCCGTCGGAGA        ZXDC         AAVS1  1813           910   \n",
       "110871  ACCTGCAGAACCACTAGCAT       AAVS1          ZXDC  2343           812   \n",
       "110872  GAGCTTGTAGGACGTTGTGA       AAVS1          ZXDC  1688           371   \n",
       "110873  AATGTACAGACTGCTACGAG       AAVS1          ZXDC   970           271   \n",
       "\n",
       "        MELJUSO_REPB  MELJUSO_REPC  GI1_REPA  ...  PATU8988S_REPB  Guide 1  \\\n",
       "0                370           567       398  ...            1139    ABO_1   \n",
       "1               1234          1926      1115  ...            3760    ABO_2   \n",
       "2                596           959       603  ...            1779    ABO_2   \n",
       "3               1329          2260       871  ...            3356    ABO_3   \n",
       "4                581           883       246  ...            1677    ABO_1   \n",
       "...              ...           ...       ...  ...             ...      ...   \n",
       "110869          1027           997       350  ...            2029   ZXDC_1   \n",
       "110870           858          1193       731  ...            1581   ZXDC_2   \n",
       "110871           609          1039       637  ...            2359  AAVS1_1   \n",
       "110872           355           534       453  ...            1499  AAVS1_1   \n",
       "110873           227           299       115  ...             693  AAVS1_1   \n",
       "\n",
       "          Guide 2  Gene 1   Gene 2  Count Replicates  Cell Line  Condition  \\\n",
       "0       A3GALT2_1     ABO  A3GALT2              None       None       None   \n",
       "1       A3GALT2_2     ABO  A3GALT2              None       None       None   \n",
       "2       A3GALT2_1     ABO  A3GALT2              None       None       None   \n",
       "3       A3GALT2_3     ABO  A3GALT2              None       None       None   \n",
       "4       A3GALT2_3     ABO  A3GALT2              None       None       None   \n",
       "...           ...     ...      ...               ...        ...        ...   \n",
       "110869    AAVS1_2    ZXDC    AAVS1              None       None       None   \n",
       "110870    AAVS1_2    ZXDC    AAVS1              None       None       None   \n",
       "110871     ZXDC_4   AAVS1     ZXDC              None       None       None   \n",
       "110872     ZXDC_6   AAVS1     ZXDC              None       None       None   \n",
       "110873     ZXDC_5   AAVS1     ZXDC              None       None       None   \n",
       "\n",
       "        Type  Sequencing  \n",
       "0       None        None  \n",
       "1       None        None  \n",
       "2       None        None  \n",
       "3       None        None  \n",
       "4       None        None  \n",
       "...      ...         ...  \n",
       "110869  None        None  \n",
       "110870  None        None  \n",
       "110871  None        None  \n",
       "110872  None        None  \n",
       "110873  None        None  \n",
       "\n",
       "[110874 rows x 47 columns]"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shared_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "d557943b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'A549',\n",
       " 'GI1',\n",
       " 'HS936T',\n",
       " 'HS944T',\n",
       " 'HSC5',\n",
       " 'IPC298',\n",
       " 'MEL202',\n",
       " 'MELJUSO',\n",
       " 'MEWO',\n",
       " 'PATU8988S',\n",
       " 'PK1'}"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "f17cb216",
   "metadata": {},
   "outputs": [],
   "source": [
    "study_conditions['ito_data'] = {}\n",
    "counts_ref = []\n",
    "\n",
    "#plasmid DNA is shared in every cell line, and every cell line has 3 replicates\n",
    "for cl in cell_lines:\n",
    "    curr_subset = shared_data[[\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Cell Line\", \"Condition\"]].copy()\n",
    "    # prepare data\n",
    "    if cl == 'PATU8988S':\n",
    "        curr_subset['Count Replicates'] = shared_data[['pDNA', cl + '_REPA', cl + '_REPB']].apply(    \n",
    "        lambda x: ';'.join(x.astype(str)),\n",
    "        axis=1\n",
    "        )\n",
    "        curr_subset[\"Condition\"] = ';'.join(['pDNA_Reads', 'Rep_A_Reads', 'Rep_B_Reads'])\n",
    "        study_conditions['ito_data'][cl] = [['pDNA_Reads'],\n",
    "                                           ['Rep_A_Reads',\n",
    "                                            'Rep_B_Reads']]\n",
    "    else:\n",
    "        curr_subset['Count Replicates'] = shared_data[['pDNA', cl + '_REPA', cl + '_REPB', cl + '_REPC']].apply(    \n",
    "        lambda x: ';'.join(x.astype(str)),\n",
    "        axis=1\n",
    "        )\n",
    "        curr_subset[\"Condition\"] = ';'.join(['pDNA_Reads', 'Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads'])\n",
    "        study_conditions['ito_data'][cl] = [['pDNA_Reads'],\n",
    "                                           ['Rep_A_Reads',\n",
    "                                            'Rep_B_Reads',\n",
    "                                            'Rep_C_Reads']]\n",
    "    curr_subset['Cell Line'] = cl\n",
    "    \n",
    "    \n",
    "    counts_ref.append(curr_subset)\n",
    "    \n",
    "counts_ref = pd.concat(counts_ref, axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "20529d97",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Study'] = [study_name_to_pubmed_id['ito_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e77efb79",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "cc808dc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "ito_loc = os.path.join(learning_goals_loc_general, \"Ito\")\n",
    "\n",
    "inferred_LFC = pd.read_excel(os.path.join(ito_loc, \"Supplementary Tables.xlsx\"), sheet_name = \"Supplementary table 5\", skiprows = 3)\n",
    "gemini_SL_score = pd.read_excel(os.path.join(ito_loc, \"Supplementary Tables.xlsx\"), sheet_name = \"Supplementary table 8\", skiprows = 3)\n",
    "gemini_FDR_score = pd.read_excel(os.path.join(ito_loc, \"Supplementary Tables.xlsx\"), sheet_name = \"Supplementary table 10\", skiprows = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "58c25784",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2351"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(inferred_LFC['A549_LUNG'] < 0).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "ff9f57d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene 1</th>\n",
       "      <th>gene 2</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>0.227807</td>\n",
       "      <td>0.326647</td>\n",
       "      <td>0.274665</td>\n",
       "      <td>0.152530</td>\n",
       "      <td>0.505637</td>\n",
       "      <td>0.070358</td>\n",
       "      <td>0.373297</td>\n",
       "      <td>0.189235</td>\n",
       "      <td>0.086962</td>\n",
       "      <td>0.222683</td>\n",
       "      <td>0.374522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>ABO</td>\n",
       "      <td>0.469130</td>\n",
       "      <td>0.026932</td>\n",
       "      <td>0.033748</td>\n",
       "      <td>0.127937</td>\n",
       "      <td>0.243435</td>\n",
       "      <td>-0.013569</td>\n",
       "      <td>0.350034</td>\n",
       "      <td>0.326564</td>\n",
       "      <td>0.160496</td>\n",
       "      <td>0.103914</td>\n",
       "      <td>0.242022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GBGT1</td>\n",
       "      <td>0.282571</td>\n",
       "      <td>-0.149230</td>\n",
       "      <td>-0.098934</td>\n",
       "      <td>-0.085788</td>\n",
       "      <td>-0.218262</td>\n",
       "      <td>-0.144974</td>\n",
       "      <td>0.301605</td>\n",
       "      <td>0.221792</td>\n",
       "      <td>0.092004</td>\n",
       "      <td>0.024993</td>\n",
       "      <td>-0.139925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GLT6D1</td>\n",
       "      <td>0.164828</td>\n",
       "      <td>-0.077907</td>\n",
       "      <td>0.100564</td>\n",
       "      <td>0.072888</td>\n",
       "      <td>-0.203077</td>\n",
       "      <td>-0.131575</td>\n",
       "      <td>0.377808</td>\n",
       "      <td>0.253479</td>\n",
       "      <td>0.068689</td>\n",
       "      <td>0.046162</td>\n",
       "      <td>-0.017748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC</td>\n",
       "      <td>AADACL2</td>\n",
       "      <td>0.019660</td>\n",
       "      <td>0.307797</td>\n",
       "      <td>-0.193339</td>\n",
       "      <td>0.011265</td>\n",
       "      <td>0.051313</td>\n",
       "      <td>0.258473</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>-0.115806</td>\n",
       "      <td>-0.158089</td>\n",
       "      <td>0.016727</td>\n",
       "      <td>0.071597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8344</th>\n",
       "      <td>ZNF362</td>\n",
       "      <td>ZNF384</td>\n",
       "      <td>0.492029</td>\n",
       "      <td>0.158074</td>\n",
       "      <td>-0.063436</td>\n",
       "      <td>0.186603</td>\n",
       "      <td>0.357797</td>\n",
       "      <td>0.676371</td>\n",
       "      <td>0.272882</td>\n",
       "      <td>0.132875</td>\n",
       "      <td>0.053505</td>\n",
       "      <td>0.009301</td>\n",
       "      <td>0.041217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8345</th>\n",
       "      <td>ZNF423</td>\n",
       "      <td>ZNF521</td>\n",
       "      <td>0.437752</td>\n",
       "      <td>0.189360</td>\n",
       "      <td>0.449655</td>\n",
       "      <td>0.217239</td>\n",
       "      <td>0.368404</td>\n",
       "      <td>0.227551</td>\n",
       "      <td>0.382624</td>\n",
       "      <td>0.242068</td>\n",
       "      <td>0.182690</td>\n",
       "      <td>0.266521</td>\n",
       "      <td>0.598274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8346</th>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>ZNRF2</td>\n",
       "      <td>0.264205</td>\n",
       "      <td>0.588086</td>\n",
       "      <td>0.340166</td>\n",
       "      <td>0.123244</td>\n",
       "      <td>0.301472</td>\n",
       "      <td>0.150680</td>\n",
       "      <td>0.070817</td>\n",
       "      <td>-0.161118</td>\n",
       "      <td>0.115489</td>\n",
       "      <td>0.044563</td>\n",
       "      <td>0.344359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8347</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.634686</td>\n",
       "      <td>0.138514</td>\n",
       "      <td>0.681951</td>\n",
       "      <td>0.408652</td>\n",
       "      <td>0.391544</td>\n",
       "      <td>0.089159</td>\n",
       "      <td>0.294511</td>\n",
       "      <td>0.302321</td>\n",
       "      <td>0.194356</td>\n",
       "      <td>0.313704</td>\n",
       "      <td>0.635204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8348</th>\n",
       "      <td>ZXDB</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.575260</td>\n",
       "      <td>0.390036</td>\n",
       "      <td>0.570246</td>\n",
       "      <td>0.347302</td>\n",
       "      <td>0.293765</td>\n",
       "      <td>0.306559</td>\n",
       "      <td>0.260811</td>\n",
       "      <td>0.423686</td>\n",
       "      <td>0.119967</td>\n",
       "      <td>0.266421</td>\n",
       "      <td>0.482381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8349 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene 1   gene 2  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0     A3GALT2    AAVS1   0.227807                    0.326647     0.274665   \n",
       "1     A3GALT2      ABO   0.469130                    0.026932     0.033748   \n",
       "2     A3GALT2    GBGT1   0.282571                   -0.149230    -0.098934   \n",
       "3     A3GALT2   GLT6D1   0.164828                   -0.077907     0.100564   \n",
       "4       AADAC  AADACL2   0.019660                    0.307797    -0.193339   \n",
       "...       ...      ...        ...                         ...          ...   \n",
       "8344   ZNF362   ZNF384   0.492029                    0.158074    -0.063436   \n",
       "8345   ZNF423   ZNF521   0.437752                    0.189360     0.449655   \n",
       "8346    ZNRF1    ZNRF2   0.264205                    0.588086     0.340166   \n",
       "8347     ZXDA     ZXDC   0.634686                    0.138514     0.681951   \n",
       "8348     ZXDB     ZXDC   0.575260                    0.390036     0.570246   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0        0.152530   0.505637     0.070358     0.373297      0.189235   \n",
       "1        0.127937   0.243435    -0.013569     0.350034      0.326564   \n",
       "2       -0.085788  -0.218262    -0.144974     0.301605      0.221792   \n",
       "3        0.072888  -0.203077    -0.131575     0.377808      0.253479   \n",
       "4        0.011265   0.051313     0.258473     0.166144     -0.115806   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "8344     0.186603   0.357797     0.676371     0.272882      0.132875   \n",
       "8345     0.217239   0.368404     0.227551     0.382624      0.242068   \n",
       "8346     0.123244   0.301472     0.150680     0.070817     -0.161118   \n",
       "8347     0.408652   0.391544     0.089159     0.294511      0.302321   \n",
       "8348     0.347302   0.293765     0.306559     0.260811      0.423686   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS  \n",
       "0      0.086962            0.222683      0.374522  \n",
       "1      0.160496            0.103914      0.242022  \n",
       "2      0.092004            0.024993     -0.139925  \n",
       "3      0.068689            0.046162     -0.017748  \n",
       "4     -0.158089            0.016727      0.071597  \n",
       "...         ...                 ...           ...  \n",
       "8344   0.053505            0.009301      0.041217  \n",
       "8345   0.182690            0.266521      0.598274  \n",
       "8346   0.115489            0.044563      0.344359  \n",
       "8347   0.194356            0.313704      0.635204  \n",
       "8348   0.119967            0.266421      0.482381  \n",
       "\n",
       "[8349 rows x 13 columns]"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inferred_LFC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "0ecbf89a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2;ABO</td>\n",
       "      <td>-0.241323</td>\n",
       "      <td>0.299715</td>\n",
       "      <td>0.240917</td>\n",
       "      <td>0.024593</td>\n",
       "      <td>0.218901</td>\n",
       "      <td>0.083927</td>\n",
       "      <td>-0.077458</td>\n",
       "      <td>-0.137329</td>\n",
       "      <td>-0.073534</td>\n",
       "      <td>0.118768</td>\n",
       "      <td>0.132501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2;GBGT1</td>\n",
       "      <td>-0.054765</td>\n",
       "      <td>0.348611</td>\n",
       "      <td>0.303613</td>\n",
       "      <td>0.234600</td>\n",
       "      <td>0.363287</td>\n",
       "      <td>0.215332</td>\n",
       "      <td>-0.150396</td>\n",
       "      <td>-0.032557</td>\n",
       "      <td>-0.005041</td>\n",
       "      <td>0.197135</td>\n",
       "      <td>0.360829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2;GLT6D1</td>\n",
       "      <td>0.062979</td>\n",
       "      <td>0.200752</td>\n",
       "      <td>0.138522</td>\n",
       "      <td>0.079642</td>\n",
       "      <td>0.540686</td>\n",
       "      <td>0.201932</td>\n",
       "      <td>-0.140968</td>\n",
       "      <td>-0.105989</td>\n",
       "      <td>0.018274</td>\n",
       "      <td>0.130981</td>\n",
       "      <td>0.366193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AADAC;AADACL2</td>\n",
       "      <td>0.352531</td>\n",
       "      <td>0.002588</td>\n",
       "      <td>0.373891</td>\n",
       "      <td>0.168104</td>\n",
       "      <td>0.270224</td>\n",
       "      <td>-0.037628</td>\n",
       "      <td>0.067509</td>\n",
       "      <td>0.314345</td>\n",
       "      <td>0.300868</td>\n",
       "      <td>0.090486</td>\n",
       "      <td>0.227840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC;AADACL3</td>\n",
       "      <td>0.257603</td>\n",
       "      <td>0.406478</td>\n",
       "      <td>0.277750</td>\n",
       "      <td>0.426975</td>\n",
       "      <td>0.142934</td>\n",
       "      <td>0.160611</td>\n",
       "      <td>0.100796</td>\n",
       "      <td>-0.026351</td>\n",
       "      <td>0.374977</td>\n",
       "      <td>0.129949</td>\n",
       "      <td>0.133039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4854</th>\n",
       "      <td>ZNF362;ZNF384</td>\n",
       "      <td>-0.447865</td>\n",
       "      <td>-0.024214</td>\n",
       "      <td>0.078828</td>\n",
       "      <td>-0.046643</td>\n",
       "      <td>-0.216521</td>\n",
       "      <td>-0.325250</td>\n",
       "      <td>-0.308825</td>\n",
       "      <td>-0.187287</td>\n",
       "      <td>0.089056</td>\n",
       "      <td>-0.170584</td>\n",
       "      <td>0.056888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4855</th>\n",
       "      <td>ZNF423;ZNF521</td>\n",
       "      <td>-0.185968</td>\n",
       "      <td>0.119744</td>\n",
       "      <td>-0.105866</td>\n",
       "      <td>0.026296</td>\n",
       "      <td>0.093787</td>\n",
       "      <td>0.161879</td>\n",
       "      <td>-0.119130</td>\n",
       "      <td>-0.070896</td>\n",
       "      <td>-0.025065</td>\n",
       "      <td>-0.127838</td>\n",
       "      <td>-0.071252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4856</th>\n",
       "      <td>ZNRF1;ZNRF2</td>\n",
       "      <td>0.067613</td>\n",
       "      <td>-0.193392</td>\n",
       "      <td>0.081893</td>\n",
       "      <td>0.009510</td>\n",
       "      <td>0.015224</td>\n",
       "      <td>0.057746</td>\n",
       "      <td>0.187882</td>\n",
       "      <td>0.201913</td>\n",
       "      <td>0.048178</td>\n",
       "      <td>0.108719</td>\n",
       "      <td>-0.181989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4857</th>\n",
       "      <td>ZXDA;ZXDC</td>\n",
       "      <td>-0.400328</td>\n",
       "      <td>-0.001426</td>\n",
       "      <td>-0.385589</td>\n",
       "      <td>-0.156762</td>\n",
       "      <td>-0.163813</td>\n",
       "      <td>0.006693</td>\n",
       "      <td>-0.236212</td>\n",
       "      <td>-0.102695</td>\n",
       "      <td>-0.113680</td>\n",
       "      <td>-0.142767</td>\n",
       "      <td>-0.331000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4858</th>\n",
       "      <td>ZXDB;ZXDC</td>\n",
       "      <td>-0.340902</td>\n",
       "      <td>-0.252948</td>\n",
       "      <td>-0.208500</td>\n",
       "      <td>-0.095413</td>\n",
       "      <td>-0.066034</td>\n",
       "      <td>-0.210707</td>\n",
       "      <td>-0.202512</td>\n",
       "      <td>-0.292406</td>\n",
       "      <td>-0.039290</td>\n",
       "      <td>-0.095484</td>\n",
       "      <td>-0.245977</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4859 rows  12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Unnamed: 0  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0        A3GALT2;ABO  -0.241323                    0.299715     0.240917   \n",
       "1      A3GALT2;GBGT1  -0.054765                    0.348611     0.303613   \n",
       "2     A3GALT2;GLT6D1   0.062979                    0.200752     0.138522   \n",
       "3      AADAC;AADACL2   0.352531                    0.002588     0.373891   \n",
       "4      AADAC;AADACL3   0.257603                    0.406478     0.277750   \n",
       "...              ...        ...                         ...          ...   \n",
       "4854   ZNF362;ZNF384  -0.447865                   -0.024214     0.078828   \n",
       "4855   ZNF423;ZNF521  -0.185968                    0.119744    -0.105866   \n",
       "4856     ZNRF1;ZNRF2   0.067613                   -0.193392     0.081893   \n",
       "4857       ZXDA;ZXDC  -0.400328                   -0.001426    -0.385589   \n",
       "4858       ZXDB;ZXDC  -0.340902                   -0.252948    -0.208500   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0        0.024593   0.218901     0.083927    -0.077458     -0.137329   \n",
       "1        0.234600   0.363287     0.215332    -0.150396     -0.032557   \n",
       "2        0.079642   0.540686     0.201932    -0.140968     -0.105989   \n",
       "3        0.168104   0.270224    -0.037628     0.067509      0.314345   \n",
       "4        0.426975   0.142934     0.160611     0.100796     -0.026351   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "4854    -0.046643  -0.216521    -0.325250    -0.308825     -0.187287   \n",
       "4855     0.026296   0.093787     0.161879    -0.119130     -0.070896   \n",
       "4856     0.009510   0.015224     0.057746     0.187882      0.201913   \n",
       "4857    -0.156762  -0.163813     0.006693    -0.236212     -0.102695   \n",
       "4858    -0.095413  -0.066034    -0.210707    -0.202512     -0.292406   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS  \n",
       "0     -0.073534            0.118768      0.132501  \n",
       "1     -0.005041            0.197135      0.360829  \n",
       "2      0.018274            0.130981      0.366193  \n",
       "3      0.300868            0.090486      0.227840  \n",
       "4      0.374977            0.129949      0.133039  \n",
       "...         ...                 ...           ...  \n",
       "4854   0.089056           -0.170584      0.056888  \n",
       "4855  -0.025065           -0.127838     -0.071252  \n",
       "4856   0.048178            0.108719     -0.181989  \n",
       "4857  -0.113680           -0.142767     -0.331000  \n",
       "4858  -0.039290           -0.095484     -0.245977  \n",
       "\n",
       "[4859 rows x 12 columns]"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gemini_SL_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "dc81c513",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2;ABO</td>\n",
       "      <td>0.921199</td>\n",
       "      <td>0.332505</td>\n",
       "      <td>0.424278</td>\n",
       "      <td>0.837115</td>\n",
       "      <td>0.336008</td>\n",
       "      <td>0.764955</td>\n",
       "      <td>0.935477</td>\n",
       "      <td>0.921540</td>\n",
       "      <td>0.955418</td>\n",
       "      <td>0.584562</td>\n",
       "      <td>0.563288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2;GBGT1</td>\n",
       "      <td>0.751001</td>\n",
       "      <td>0.263071</td>\n",
       "      <td>0.307590</td>\n",
       "      <td>0.381261</td>\n",
       "      <td>0.135696</td>\n",
       "      <td>0.547510</td>\n",
       "      <td>0.977996</td>\n",
       "      <td>0.755984</td>\n",
       "      <td>0.896801</td>\n",
       "      <td>0.380175</td>\n",
       "      <td>0.172056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2;GLT6D1</td>\n",
       "      <td>0.570461</td>\n",
       "      <td>0.487910</td>\n",
       "      <td>0.622668</td>\n",
       "      <td>0.741919</td>\n",
       "      <td>0.020182</td>\n",
       "      <td>0.571000</td>\n",
       "      <td>0.974356</td>\n",
       "      <td>0.884787</td>\n",
       "      <td>0.866267</td>\n",
       "      <td>0.553824</td>\n",
       "      <td>0.165021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AADAC;AADACL2</td>\n",
       "      <td>0.139850</td>\n",
       "      <td>0.776708</td>\n",
       "      <td>0.196587</td>\n",
       "      <td>0.542847</td>\n",
       "      <td>0.248709</td>\n",
       "      <td>0.904911</td>\n",
       "      <td>0.726176</td>\n",
       "      <td>0.058205</td>\n",
       "      <td>0.204460</td>\n",
       "      <td>0.652104</td>\n",
       "      <td>0.389172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC;AADACL3</td>\n",
       "      <td>0.252618</td>\n",
       "      <td>0.191253</td>\n",
       "      <td>0.354247</td>\n",
       "      <td>0.067707</td>\n",
       "      <td>0.491352</td>\n",
       "      <td>0.642586</td>\n",
       "      <td>0.659100</td>\n",
       "      <td>0.742404</td>\n",
       "      <td>0.090457</td>\n",
       "      <td>0.556450</td>\n",
       "      <td>0.562338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4854</th>\n",
       "      <td>ZNF362;ZNF384</td>\n",
       "      <td>0.981917</td>\n",
       "      <td>0.806990</td>\n",
       "      <td>0.725792</td>\n",
       "      <td>0.920152</td>\n",
       "      <td>0.978128</td>\n",
       "      <td>0.997034</td>\n",
       "      <td>0.999124</td>\n",
       "      <td>0.960840</td>\n",
       "      <td>0.735925</td>\n",
       "      <td>0.969437</td>\n",
       "      <td>0.688228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4855</th>\n",
       "      <td>ZNF423;ZNF521</td>\n",
       "      <td>0.886454</td>\n",
       "      <td>0.616188</td>\n",
       "      <td>0.921000</td>\n",
       "      <td>0.834580</td>\n",
       "      <td>0.597923</td>\n",
       "      <td>0.640425</td>\n",
       "      <td>0.964026</td>\n",
       "      <td>0.830805</td>\n",
       "      <td>0.918407</td>\n",
       "      <td>0.948940</td>\n",
       "      <td>0.849562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4856</th>\n",
       "      <td>ZNRF1;ZNRF2</td>\n",
       "      <td>0.562628</td>\n",
       "      <td>0.936957</td>\n",
       "      <td>0.720903</td>\n",
       "      <td>0.858420</td>\n",
       "      <td>0.754345</td>\n",
       "      <td>0.801502</td>\n",
       "      <td>0.475185</td>\n",
       "      <td>0.186117</td>\n",
       "      <td>0.818102</td>\n",
       "      <td>0.609197</td>\n",
       "      <td>0.934004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4857</th>\n",
       "      <td>ZXDA;ZXDC</td>\n",
       "      <td>0.970700</td>\n",
       "      <td>0.781413</td>\n",
       "      <td>0.992768</td>\n",
       "      <td>0.978433</td>\n",
       "      <td>0.956582</td>\n",
       "      <td>0.862882</td>\n",
       "      <td>0.995520</td>\n",
       "      <td>0.880310</td>\n",
       "      <td>0.973317</td>\n",
       "      <td>0.957046</td>\n",
       "      <td>0.984665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4858</th>\n",
       "      <td>ZXDB;ZXDC</td>\n",
       "      <td>0.958072</td>\n",
       "      <td>0.960969</td>\n",
       "      <td>0.959880</td>\n",
       "      <td>0.954196</td>\n",
       "      <td>0.875061</td>\n",
       "      <td>0.985088</td>\n",
       "      <td>0.991263</td>\n",
       "      <td>0.993564</td>\n",
       "      <td>0.931364</td>\n",
       "      <td>0.927340</td>\n",
       "      <td>0.962825</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4859 rows  12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Unnamed: 0  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0        A3GALT2;ABO   0.921199                    0.332505     0.424278   \n",
       "1      A3GALT2;GBGT1   0.751001                    0.263071     0.307590   \n",
       "2     A3GALT2;GLT6D1   0.570461                    0.487910     0.622668   \n",
       "3      AADAC;AADACL2   0.139850                    0.776708     0.196587   \n",
       "4      AADAC;AADACL3   0.252618                    0.191253     0.354247   \n",
       "...              ...        ...                         ...          ...   \n",
       "4854   ZNF362;ZNF384   0.981917                    0.806990     0.725792   \n",
       "4855   ZNF423;ZNF521   0.886454                    0.616188     0.921000   \n",
       "4856     ZNRF1;ZNRF2   0.562628                    0.936957     0.720903   \n",
       "4857       ZXDA;ZXDC   0.970700                    0.781413     0.992768   \n",
       "4858       ZXDB;ZXDC   0.958072                    0.960969     0.959880   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0        0.837115   0.336008     0.764955     0.935477      0.921540   \n",
       "1        0.381261   0.135696     0.547510     0.977996      0.755984   \n",
       "2        0.741919   0.020182     0.571000     0.974356      0.884787   \n",
       "3        0.542847   0.248709     0.904911     0.726176      0.058205   \n",
       "4        0.067707   0.491352     0.642586     0.659100      0.742404   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "4854     0.920152   0.978128     0.997034     0.999124      0.960840   \n",
       "4855     0.834580   0.597923     0.640425     0.964026      0.830805   \n",
       "4856     0.858420   0.754345     0.801502     0.475185      0.186117   \n",
       "4857     0.978433   0.956582     0.862882     0.995520      0.880310   \n",
       "4858     0.954196   0.875061     0.985088     0.991263      0.993564   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS  \n",
       "0      0.955418            0.584562      0.563288  \n",
       "1      0.896801            0.380175      0.172056  \n",
       "2      0.866267            0.553824      0.165021  \n",
       "3      0.204460            0.652104      0.389172  \n",
       "4      0.090457            0.556450      0.562338  \n",
       "...         ...                 ...           ...  \n",
       "4854   0.735925            0.969437      0.688228  \n",
       "4855   0.918407            0.948940      0.849562  \n",
       "4856   0.818102            0.609197      0.934004  \n",
       "4857   0.973317            0.957046      0.984665  \n",
       "4858   0.931364            0.927340      0.962825  \n",
       "\n",
       "[4859 rows x 12 columns]"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gemini_FDR_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "d5da24fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# updated_gemini_SL_score = inferred_LFC.copy()\n",
    "# merged = updated_gemini_SL_score[[\"gene 1\", \"gene 2\"]].agg(';'.join, axis=1) \n",
    "# for i in range(len(merged)):\n",
    "#     pair = merged[i]\n",
    "#     ref_list = gemini_SL_score[\"Unnamed: 0\"].tolist()\n",
    "#     if pair not in ref_list:\n",
    "#         updated_gemini_SL_score.iloc[i, 2:] = float(\"nan\")\n",
    "#     else:\n",
    "#         updated_gemini_SL_score.iloc[i, 2:] = gemini_SL_score.iloc[ref_list.index(pair), 1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "7e234bf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "updated_gemini_SL_score = inferred_LFC.copy()\n",
    "merged = updated_gemini_SL_score[[\"gene 1\", \"gene 2\"]].agg(';'.join, axis=1) \n",
    "for i in range(len(merged)):\n",
    "    pair = merged[i]\n",
    "    ref_list = gemini_SL_score[\"Unnamed: 0\"].tolist()\n",
    "    if pair not in ref_list:\n",
    "        updated_gemini_SL_score.iloc[i, 2:] = float(\"nan\")\n",
    "    else:\n",
    "        updated_gemini_SL_score.iloc[i, 2:] = gemini_SL_score.iloc[ref_list.index(pair), 1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "c6201400",
   "metadata": {},
   "outputs": [],
   "source": [
    "updated_gemini_FDR_score = inferred_LFC.copy()\n",
    "merged = updated_gemini_FDR_score[[\"gene 1\", \"gene 2\"]].agg(';'.join, axis=1) \n",
    "for i in range(len(merged)):\n",
    "    pair = merged[i]\n",
    "    ref_list = gemini_FDR_score[\"Unnamed: 0\"].tolist()\n",
    "    if pair not in ref_list:\n",
    "        updated_gemini_FDR_score.iloc[i, 2:] = float(\"nan\")\n",
    "    else:\n",
    "        updated_gemini_FDR_score.iloc[i, 2:] = gemini_FDR_score.iloc[ref_list.index(pair), 1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "97a644a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene 1</th>\n",
       "      <th>gene 2</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>ABO</td>\n",
       "      <td>0.921199</td>\n",
       "      <td>0.332505</td>\n",
       "      <td>0.424278</td>\n",
       "      <td>0.837115</td>\n",
       "      <td>0.336008</td>\n",
       "      <td>0.764955</td>\n",
       "      <td>0.935477</td>\n",
       "      <td>0.921540</td>\n",
       "      <td>0.955418</td>\n",
       "      <td>0.584562</td>\n",
       "      <td>0.563288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GBGT1</td>\n",
       "      <td>0.751001</td>\n",
       "      <td>0.263071</td>\n",
       "      <td>0.307590</td>\n",
       "      <td>0.381261</td>\n",
       "      <td>0.135696</td>\n",
       "      <td>0.547510</td>\n",
       "      <td>0.977996</td>\n",
       "      <td>0.755984</td>\n",
       "      <td>0.896801</td>\n",
       "      <td>0.380175</td>\n",
       "      <td>0.172056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GLT6D1</td>\n",
       "      <td>0.570461</td>\n",
       "      <td>0.487910</td>\n",
       "      <td>0.622668</td>\n",
       "      <td>0.741919</td>\n",
       "      <td>0.020182</td>\n",
       "      <td>0.571000</td>\n",
       "      <td>0.974356</td>\n",
       "      <td>0.884787</td>\n",
       "      <td>0.866267</td>\n",
       "      <td>0.553824</td>\n",
       "      <td>0.165021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC</td>\n",
       "      <td>AADACL2</td>\n",
       "      <td>0.139850</td>\n",
       "      <td>0.776708</td>\n",
       "      <td>0.196587</td>\n",
       "      <td>0.542847</td>\n",
       "      <td>0.248709</td>\n",
       "      <td>0.904911</td>\n",
       "      <td>0.726176</td>\n",
       "      <td>0.058205</td>\n",
       "      <td>0.204460</td>\n",
       "      <td>0.652104</td>\n",
       "      <td>0.389172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8344</th>\n",
       "      <td>ZNF362</td>\n",
       "      <td>ZNF384</td>\n",
       "      <td>0.981917</td>\n",
       "      <td>0.806990</td>\n",
       "      <td>0.725792</td>\n",
       "      <td>0.920152</td>\n",
       "      <td>0.978128</td>\n",
       "      <td>0.997034</td>\n",
       "      <td>0.999124</td>\n",
       "      <td>0.960840</td>\n",
       "      <td>0.735925</td>\n",
       "      <td>0.969437</td>\n",
       "      <td>0.688228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8345</th>\n",
       "      <td>ZNF423</td>\n",
       "      <td>ZNF521</td>\n",
       "      <td>0.886454</td>\n",
       "      <td>0.616188</td>\n",
       "      <td>0.921000</td>\n",
       "      <td>0.834580</td>\n",
       "      <td>0.597923</td>\n",
       "      <td>0.640425</td>\n",
       "      <td>0.964026</td>\n",
       "      <td>0.830805</td>\n",
       "      <td>0.918407</td>\n",
       "      <td>0.948940</td>\n",
       "      <td>0.849562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8346</th>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>ZNRF2</td>\n",
       "      <td>0.562628</td>\n",
       "      <td>0.936957</td>\n",
       "      <td>0.720903</td>\n",
       "      <td>0.858420</td>\n",
       "      <td>0.754345</td>\n",
       "      <td>0.801502</td>\n",
       "      <td>0.475185</td>\n",
       "      <td>0.186117</td>\n",
       "      <td>0.818102</td>\n",
       "      <td>0.609197</td>\n",
       "      <td>0.934004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8347</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.970700</td>\n",
       "      <td>0.781413</td>\n",
       "      <td>0.992768</td>\n",
       "      <td>0.978433</td>\n",
       "      <td>0.956582</td>\n",
       "      <td>0.862882</td>\n",
       "      <td>0.995520</td>\n",
       "      <td>0.880310</td>\n",
       "      <td>0.973317</td>\n",
       "      <td>0.957046</td>\n",
       "      <td>0.984665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8348</th>\n",
       "      <td>ZXDB</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.958072</td>\n",
       "      <td>0.960969</td>\n",
       "      <td>0.959880</td>\n",
       "      <td>0.954196</td>\n",
       "      <td>0.875061</td>\n",
       "      <td>0.985088</td>\n",
       "      <td>0.991263</td>\n",
       "      <td>0.993564</td>\n",
       "      <td>0.931364</td>\n",
       "      <td>0.927340</td>\n",
       "      <td>0.962825</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8349 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene 1   gene 2  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0     A3GALT2    AAVS1        NaN                         NaN          NaN   \n",
       "1     A3GALT2      ABO   0.921199                    0.332505     0.424278   \n",
       "2     A3GALT2    GBGT1   0.751001                    0.263071     0.307590   \n",
       "3     A3GALT2   GLT6D1   0.570461                    0.487910     0.622668   \n",
       "4       AADAC  AADACL2   0.139850                    0.776708     0.196587   \n",
       "...       ...      ...        ...                         ...          ...   \n",
       "8344   ZNF362   ZNF384   0.981917                    0.806990     0.725792   \n",
       "8345   ZNF423   ZNF521   0.886454                    0.616188     0.921000   \n",
       "8346    ZNRF1    ZNRF2   0.562628                    0.936957     0.720903   \n",
       "8347     ZXDA     ZXDC   0.970700                    0.781413     0.992768   \n",
       "8348     ZXDB     ZXDC   0.958072                    0.960969     0.959880   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0             NaN        NaN          NaN          NaN           NaN   \n",
       "1        0.837115   0.336008     0.764955     0.935477      0.921540   \n",
       "2        0.381261   0.135696     0.547510     0.977996      0.755984   \n",
       "3        0.741919   0.020182     0.571000     0.974356      0.884787   \n",
       "4        0.542847   0.248709     0.904911     0.726176      0.058205   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "8344     0.920152   0.978128     0.997034     0.999124      0.960840   \n",
       "8345     0.834580   0.597923     0.640425     0.964026      0.830805   \n",
       "8346     0.858420   0.754345     0.801502     0.475185      0.186117   \n",
       "8347     0.978433   0.956582     0.862882     0.995520      0.880310   \n",
       "8348     0.954196   0.875061     0.985088     0.991263      0.993564   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS  \n",
       "0           NaN                 NaN           NaN  \n",
       "1      0.955418            0.584562      0.563288  \n",
       "2      0.896801            0.380175      0.172056  \n",
       "3      0.866267            0.553824      0.165021  \n",
       "4      0.204460            0.652104      0.389172  \n",
       "...         ...                 ...           ...  \n",
       "8344   0.735925            0.969437      0.688228  \n",
       "8345   0.918407            0.948940      0.849562  \n",
       "8346   0.818102            0.609197      0.934004  \n",
       "8347   0.973317            0.957046      0.984665  \n",
       "8348   0.931364            0.927340      0.962825  \n",
       "\n",
       "[8349 rows x 13 columns]"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "updated_gemini_FDR_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "5a8b12b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "gemini_SL_score['gene 1'] = [i.split(';')[0] for i in gemini_SL_score['Unnamed: 0']]\n",
    "gemini_SL_score['gene 2'] = [i.split(';')[1] for i in gemini_SL_score['Unnamed: 0']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "6f7b21e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4859"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(gemini_SL_score['Unnamed: 0'] == gemini_FDR_score['Unnamed: 0']).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "27d7b734",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2;ABO</td>\n",
       "      <td>0.921199</td>\n",
       "      <td>0.332505</td>\n",
       "      <td>0.424278</td>\n",
       "      <td>0.837115</td>\n",
       "      <td>0.336008</td>\n",
       "      <td>0.764955</td>\n",
       "      <td>0.935477</td>\n",
       "      <td>0.921540</td>\n",
       "      <td>0.955418</td>\n",
       "      <td>0.584562</td>\n",
       "      <td>0.563288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2;GBGT1</td>\n",
       "      <td>0.751001</td>\n",
       "      <td>0.263071</td>\n",
       "      <td>0.307590</td>\n",
       "      <td>0.381261</td>\n",
       "      <td>0.135696</td>\n",
       "      <td>0.547510</td>\n",
       "      <td>0.977996</td>\n",
       "      <td>0.755984</td>\n",
       "      <td>0.896801</td>\n",
       "      <td>0.380175</td>\n",
       "      <td>0.172056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2;GLT6D1</td>\n",
       "      <td>0.570461</td>\n",
       "      <td>0.487910</td>\n",
       "      <td>0.622668</td>\n",
       "      <td>0.741919</td>\n",
       "      <td>0.020182</td>\n",
       "      <td>0.571000</td>\n",
       "      <td>0.974356</td>\n",
       "      <td>0.884787</td>\n",
       "      <td>0.866267</td>\n",
       "      <td>0.553824</td>\n",
       "      <td>0.165021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AADAC;AADACL2</td>\n",
       "      <td>0.139850</td>\n",
       "      <td>0.776708</td>\n",
       "      <td>0.196587</td>\n",
       "      <td>0.542847</td>\n",
       "      <td>0.248709</td>\n",
       "      <td>0.904911</td>\n",
       "      <td>0.726176</td>\n",
       "      <td>0.058205</td>\n",
       "      <td>0.204460</td>\n",
       "      <td>0.652104</td>\n",
       "      <td>0.389172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC;AADACL3</td>\n",
       "      <td>0.252618</td>\n",
       "      <td>0.191253</td>\n",
       "      <td>0.354247</td>\n",
       "      <td>0.067707</td>\n",
       "      <td>0.491352</td>\n",
       "      <td>0.642586</td>\n",
       "      <td>0.659100</td>\n",
       "      <td>0.742404</td>\n",
       "      <td>0.090457</td>\n",
       "      <td>0.556450</td>\n",
       "      <td>0.562338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4854</th>\n",
       "      <td>ZNF362;ZNF384</td>\n",
       "      <td>0.981917</td>\n",
       "      <td>0.806990</td>\n",
       "      <td>0.725792</td>\n",
       "      <td>0.920152</td>\n",
       "      <td>0.978128</td>\n",
       "      <td>0.997034</td>\n",
       "      <td>0.999124</td>\n",
       "      <td>0.960840</td>\n",
       "      <td>0.735925</td>\n",
       "      <td>0.969437</td>\n",
       "      <td>0.688228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4855</th>\n",
       "      <td>ZNF423;ZNF521</td>\n",
       "      <td>0.886454</td>\n",
       "      <td>0.616188</td>\n",
       "      <td>0.921000</td>\n",
       "      <td>0.834580</td>\n",
       "      <td>0.597923</td>\n",
       "      <td>0.640425</td>\n",
       "      <td>0.964026</td>\n",
       "      <td>0.830805</td>\n",
       "      <td>0.918407</td>\n",
       "      <td>0.948940</td>\n",
       "      <td>0.849562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4856</th>\n",
       "      <td>ZNRF1;ZNRF2</td>\n",
       "      <td>0.562628</td>\n",
       "      <td>0.936957</td>\n",
       "      <td>0.720903</td>\n",
       "      <td>0.858420</td>\n",
       "      <td>0.754345</td>\n",
       "      <td>0.801502</td>\n",
       "      <td>0.475185</td>\n",
       "      <td>0.186117</td>\n",
       "      <td>0.818102</td>\n",
       "      <td>0.609197</td>\n",
       "      <td>0.934004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4857</th>\n",
       "      <td>ZXDA;ZXDC</td>\n",
       "      <td>0.970700</td>\n",
       "      <td>0.781413</td>\n",
       "      <td>0.992768</td>\n",
       "      <td>0.978433</td>\n",
       "      <td>0.956582</td>\n",
       "      <td>0.862882</td>\n",
       "      <td>0.995520</td>\n",
       "      <td>0.880310</td>\n",
       "      <td>0.973317</td>\n",
       "      <td>0.957046</td>\n",
       "      <td>0.984665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4858</th>\n",
       "      <td>ZXDB;ZXDC</td>\n",
       "      <td>0.958072</td>\n",
       "      <td>0.960969</td>\n",
       "      <td>0.959880</td>\n",
       "      <td>0.954196</td>\n",
       "      <td>0.875061</td>\n",
       "      <td>0.985088</td>\n",
       "      <td>0.991263</td>\n",
       "      <td>0.993564</td>\n",
       "      <td>0.931364</td>\n",
       "      <td>0.927340</td>\n",
       "      <td>0.962825</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4859 rows  12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Unnamed: 0  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0        A3GALT2;ABO   0.921199                    0.332505     0.424278   \n",
       "1      A3GALT2;GBGT1   0.751001                    0.263071     0.307590   \n",
       "2     A3GALT2;GLT6D1   0.570461                    0.487910     0.622668   \n",
       "3      AADAC;AADACL2   0.139850                    0.776708     0.196587   \n",
       "4      AADAC;AADACL3   0.252618                    0.191253     0.354247   \n",
       "...              ...        ...                         ...          ...   \n",
       "4854   ZNF362;ZNF384   0.981917                    0.806990     0.725792   \n",
       "4855   ZNF423;ZNF521   0.886454                    0.616188     0.921000   \n",
       "4856     ZNRF1;ZNRF2   0.562628                    0.936957     0.720903   \n",
       "4857       ZXDA;ZXDC   0.970700                    0.781413     0.992768   \n",
       "4858       ZXDB;ZXDC   0.958072                    0.960969     0.959880   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0        0.837115   0.336008     0.764955     0.935477      0.921540   \n",
       "1        0.381261   0.135696     0.547510     0.977996      0.755984   \n",
       "2        0.741919   0.020182     0.571000     0.974356      0.884787   \n",
       "3        0.542847   0.248709     0.904911     0.726176      0.058205   \n",
       "4        0.067707   0.491352     0.642586     0.659100      0.742404   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "4854     0.920152   0.978128     0.997034     0.999124      0.960840   \n",
       "4855     0.834580   0.597923     0.640425     0.964026      0.830805   \n",
       "4856     0.858420   0.754345     0.801502     0.475185      0.186117   \n",
       "4857     0.978433   0.956582     0.862882     0.995520      0.880310   \n",
       "4858     0.954196   0.875061     0.985088     0.991263      0.993564   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS  \n",
       "0      0.955418            0.584562      0.563288  \n",
       "1      0.896801            0.380175      0.172056  \n",
       "2      0.866267            0.553824      0.165021  \n",
       "3      0.204460            0.652104      0.389172  \n",
       "4      0.090457            0.556450      0.562338  \n",
       "...         ...                 ...           ...  \n",
       "4854   0.735925            0.969437      0.688228  \n",
       "4855   0.918407            0.948940      0.849562  \n",
       "4856   0.818102            0.609197      0.934004  \n",
       "4857   0.973317            0.957046      0.984665  \n",
       "4858   0.931364            0.927340      0.962825  \n",
       "\n",
       "[4859 rows x 12 columns]"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gemini_FDR_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "9973f20d",
   "metadata": {},
   "outputs": [],
   "source": [
    "gemini_FDR_score['gene 1'] = [i.split(';')[0] for i in gemini_FDR_score['Unnamed: 0']]\n",
    "gemini_FDR_score['gene 2'] = [i.split(';')[1] for i in gemini_FDR_score['Unnamed: 0']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "5ec27996",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "34b26670",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A549\n",
      "###############################\n",
      "GI1\n",
      "###############################\n",
      "HS936T\n",
      "###############################\n",
      "HS944T\n",
      "###############################\n",
      "HSC5\n",
      "###############################\n",
      "IPC298\n",
      "###############################\n",
      "MEL202\n",
      "###############################\n",
      "MELJUSO\n",
      "###############################\n",
      "MEWO\n",
      "###############################\n",
      "PATU8988S\n",
      "###############################\n",
      "PK1\n",
      "###############################\n"
     ]
    }
   ],
   "source": [
    "for cell_line in inferred_LFC.columns[2:]:\n",
    "    orig_cell_line = cell_line\n",
    "    curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"Phenotype\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "    curr_GI[\"Gene_A\"] = gemini_SL_score[\"gene 1\"]\n",
    "    curr_GI[\"Gene_B\"] = gemini_SL_score[\"gene 2\"]\n",
    "    curr_GI[\"GI_Score\"] = gemini_SL_score.loc[:, cell_line]\n",
    "    \n",
    "    curr_GI = curr_GI.fillna(0)\n",
    "    curr_GI[\"GI_Cutoff\"] = [float(\"nan\")] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Stat_Score\"] = gemini_FDR_score.loc[:, cell_line]\n",
    "    curr_GI[\"Stat_Cutoff\"] = [0.05] * len(curr_GI[\"Gene_A\"])\n",
    "    \n",
    "    # Total number of GIs (negatives)\n",
    "    cell_line = cell_line.split('_')[0]\n",
    "    curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['ito_data']] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Cell_Line\"] = [cell_line] * len(curr_GI[\"Gene_A\"])\n",
    "    \n",
    "    all_GI = pd.concat([all_GI, curr_GI])\n",
    "    \n",
    "#     all_data[\"ito_GI_\" + cell_line] = curr_GI\n",
    "#     all_data[\"Notes\"][\"ito_GI_\" + cell_line] = {}\n",
    "#     all_data[\"Notes\"][\"ito_GI_\" + cell_line][\"Control\"] = \"AAVS1\"\n",
    "#     all_data[\"Notes\"][\"ito_GI_\" + cell_line][\"Phenotype\"] = \"log fold change\"\n",
    "\n",
    "    print(cell_line)\n",
    "#     print(\"Total Neg Phenotype: \" + str(sum(curr_GI[\"GI_Score\"] < 0)))\n",
    "#     print(\"Total Neg Phenotype: \" + str(sum(curr_GI[\"Phenotype\"] < 0)))\n",
    "#     print(\"Total Sig Neg Phenotype: \" + str(sum((gemini_FDR_score[orig_cell_line] < 0.05))))#(gemini_SL_score[orig_cell_line] < 0) & \n",
    "#     print(\"Percentages\")\n",
    "#     print(\"Total Neg Phenotype: \" + str(100 * sum(curr_GI[\"GI_Score\"] < 0)/len(curr_GI[\"GI_Score\"])))\n",
    "#     print(\"Total Neg Phenotype: \" + str(100 * sum(curr_GI[\"Phenotype\"] < 0)/len(curr_GI[\"Phenotype\"])))\n",
    "#     print(\"Total Sig Neg Phenotype: \" + str(100 * sum((gemini_SL_score[orig_cell_line] > 0) & (gemini_FDR_score[orig_cell_line] < 0.05))/len(gemini_FDR_score[orig_cell_line])))\n",
    "# #     curr_GI_matrix = create_matrix_from_df(curr_GI)\n",
    "#     all_str += \"{}% {} ({} GI), \".format(round(100 * sum(curr_GI[\"GI_Score\"] < 0)/len(curr_GI[\"GI_Score\"]), 2),\n",
    "#                                   cell_line,\n",
    "#                                   sum(curr_GI[\"GI_Score\"] > 0)\n",
    "          \n",
    "#     )\n",
    "#     sig_str += \"{}% {} ({} SL), \".format(round(100 * sum((gemini_SL_score[orig_cell_line] > 0) & (gemini_FDR_score[orig_cell_line] < 0.05))/len(gemini_FDR_score[orig_cell_line]), 2),\n",
    "#                                   cell_line,\n",
    "#                                   sum((gemini_SL_score[orig_cell_line] < 0) & (gemini_FDR_score[orig_cell_line] < 0.05))\n",
    "#          \n",
    "#    )\n",
    "#     curr_GI_matrix_cell_lines = make_cell_line_matrix(curr_GI_matrix, cell_line = cell_line)\n",
    "#     curr_GI_matrix.to_csv(os.path.join(ito_loc, \"GI_matrix.csv\"))\n",
    "#     curr_GI_matrix_cell_lines.to_csv(os.path.join(ito_loc, \"CellLine_matrix.csv\"))\n",
    "    print(\"###############################\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "3d671557",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "      <th>gene 1</th>\n",
       "      <th>gene 2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2;ABO</td>\n",
       "      <td>0.921199</td>\n",
       "      <td>0.332505</td>\n",
       "      <td>0.424278</td>\n",
       "      <td>0.837115</td>\n",
       "      <td>0.336008</td>\n",
       "      <td>0.764955</td>\n",
       "      <td>0.935477</td>\n",
       "      <td>0.921540</td>\n",
       "      <td>0.955418</td>\n",
       "      <td>0.584562</td>\n",
       "      <td>0.563288</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>ABO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2;GBGT1</td>\n",
       "      <td>0.751001</td>\n",
       "      <td>0.263071</td>\n",
       "      <td>0.307590</td>\n",
       "      <td>0.381261</td>\n",
       "      <td>0.135696</td>\n",
       "      <td>0.547510</td>\n",
       "      <td>0.977996</td>\n",
       "      <td>0.755984</td>\n",
       "      <td>0.896801</td>\n",
       "      <td>0.380175</td>\n",
       "      <td>0.172056</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GBGT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2;GLT6D1</td>\n",
       "      <td>0.570461</td>\n",
       "      <td>0.487910</td>\n",
       "      <td>0.622668</td>\n",
       "      <td>0.741919</td>\n",
       "      <td>0.020182</td>\n",
       "      <td>0.571000</td>\n",
       "      <td>0.974356</td>\n",
       "      <td>0.884787</td>\n",
       "      <td>0.866267</td>\n",
       "      <td>0.553824</td>\n",
       "      <td>0.165021</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GLT6D1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AADAC;AADACL2</td>\n",
       "      <td>0.139850</td>\n",
       "      <td>0.776708</td>\n",
       "      <td>0.196587</td>\n",
       "      <td>0.542847</td>\n",
       "      <td>0.248709</td>\n",
       "      <td>0.904911</td>\n",
       "      <td>0.726176</td>\n",
       "      <td>0.058205</td>\n",
       "      <td>0.204460</td>\n",
       "      <td>0.652104</td>\n",
       "      <td>0.389172</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>AADACL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC;AADACL3</td>\n",
       "      <td>0.252618</td>\n",
       "      <td>0.191253</td>\n",
       "      <td>0.354247</td>\n",
       "      <td>0.067707</td>\n",
       "      <td>0.491352</td>\n",
       "      <td>0.642586</td>\n",
       "      <td>0.659100</td>\n",
       "      <td>0.742404</td>\n",
       "      <td>0.090457</td>\n",
       "      <td>0.556450</td>\n",
       "      <td>0.562338</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>AADACL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4854</th>\n",
       "      <td>ZNF362;ZNF384</td>\n",
       "      <td>0.981917</td>\n",
       "      <td>0.806990</td>\n",
       "      <td>0.725792</td>\n",
       "      <td>0.920152</td>\n",
       "      <td>0.978128</td>\n",
       "      <td>0.997034</td>\n",
       "      <td>0.999124</td>\n",
       "      <td>0.960840</td>\n",
       "      <td>0.735925</td>\n",
       "      <td>0.969437</td>\n",
       "      <td>0.688228</td>\n",
       "      <td>ZNF362</td>\n",
       "      <td>ZNF384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4855</th>\n",
       "      <td>ZNF423;ZNF521</td>\n",
       "      <td>0.886454</td>\n",
       "      <td>0.616188</td>\n",
       "      <td>0.921000</td>\n",
       "      <td>0.834580</td>\n",
       "      <td>0.597923</td>\n",
       "      <td>0.640425</td>\n",
       "      <td>0.964026</td>\n",
       "      <td>0.830805</td>\n",
       "      <td>0.918407</td>\n",
       "      <td>0.948940</td>\n",
       "      <td>0.849562</td>\n",
       "      <td>ZNF423</td>\n",
       "      <td>ZNF521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4856</th>\n",
       "      <td>ZNRF1;ZNRF2</td>\n",
       "      <td>0.562628</td>\n",
       "      <td>0.936957</td>\n",
       "      <td>0.720903</td>\n",
       "      <td>0.858420</td>\n",
       "      <td>0.754345</td>\n",
       "      <td>0.801502</td>\n",
       "      <td>0.475185</td>\n",
       "      <td>0.186117</td>\n",
       "      <td>0.818102</td>\n",
       "      <td>0.609197</td>\n",
       "      <td>0.934004</td>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>ZNRF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4857</th>\n",
       "      <td>ZXDA;ZXDC</td>\n",
       "      <td>0.970700</td>\n",
       "      <td>0.781413</td>\n",
       "      <td>0.992768</td>\n",
       "      <td>0.978433</td>\n",
       "      <td>0.956582</td>\n",
       "      <td>0.862882</td>\n",
       "      <td>0.995520</td>\n",
       "      <td>0.880310</td>\n",
       "      <td>0.973317</td>\n",
       "      <td>0.957046</td>\n",
       "      <td>0.984665</td>\n",
       "      <td>ZXDA</td>\n",
       "      <td>ZXDC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4858</th>\n",
       "      <td>ZXDB;ZXDC</td>\n",
       "      <td>0.958072</td>\n",
       "      <td>0.960969</td>\n",
       "      <td>0.959880</td>\n",
       "      <td>0.954196</td>\n",
       "      <td>0.875061</td>\n",
       "      <td>0.985088</td>\n",
       "      <td>0.991263</td>\n",
       "      <td>0.993564</td>\n",
       "      <td>0.931364</td>\n",
       "      <td>0.927340</td>\n",
       "      <td>0.962825</td>\n",
       "      <td>ZXDB</td>\n",
       "      <td>ZXDC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4859 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Unnamed: 0  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0        A3GALT2;ABO   0.921199                    0.332505     0.424278   \n",
       "1      A3GALT2;GBGT1   0.751001                    0.263071     0.307590   \n",
       "2     A3GALT2;GLT6D1   0.570461                    0.487910     0.622668   \n",
       "3      AADAC;AADACL2   0.139850                    0.776708     0.196587   \n",
       "4      AADAC;AADACL3   0.252618                    0.191253     0.354247   \n",
       "...              ...        ...                         ...          ...   \n",
       "4854   ZNF362;ZNF384   0.981917                    0.806990     0.725792   \n",
       "4855   ZNF423;ZNF521   0.886454                    0.616188     0.921000   \n",
       "4856     ZNRF1;ZNRF2   0.562628                    0.936957     0.720903   \n",
       "4857       ZXDA;ZXDC   0.970700                    0.781413     0.992768   \n",
       "4858       ZXDB;ZXDC   0.958072                    0.960969     0.959880   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0        0.837115   0.336008     0.764955     0.935477      0.921540   \n",
       "1        0.381261   0.135696     0.547510     0.977996      0.755984   \n",
       "2        0.741919   0.020182     0.571000     0.974356      0.884787   \n",
       "3        0.542847   0.248709     0.904911     0.726176      0.058205   \n",
       "4        0.067707   0.491352     0.642586     0.659100      0.742404   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "4854     0.920152   0.978128     0.997034     0.999124      0.960840   \n",
       "4855     0.834580   0.597923     0.640425     0.964026      0.830805   \n",
       "4856     0.858420   0.754345     0.801502     0.475185      0.186117   \n",
       "4857     0.978433   0.956582     0.862882     0.995520      0.880310   \n",
       "4858     0.954196   0.875061     0.985088     0.991263      0.993564   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS   gene 1   gene 2  \n",
       "0      0.955418            0.584562      0.563288  A3GALT2      ABO  \n",
       "1      0.896801            0.380175      0.172056  A3GALT2    GBGT1  \n",
       "2      0.866267            0.553824      0.165021  A3GALT2   GLT6D1  \n",
       "3      0.204460            0.652104      0.389172    AADAC  AADACL2  \n",
       "4      0.090457            0.556450      0.562338    AADAC  AADACL3  \n",
       "...         ...                 ...           ...      ...      ...  \n",
       "4854   0.735925            0.969437      0.688228   ZNF362   ZNF384  \n",
       "4855   0.918407            0.948940      0.849562   ZNF423   ZNF521  \n",
       "4856   0.818102            0.609197      0.934004    ZNRF1    ZNRF2  \n",
       "4857   0.973317            0.957046      0.984665     ZXDA     ZXDC  \n",
       "4858   0.931364            0.927340      0.962825     ZXDB     ZXDC  \n",
       "\n",
       "[4859 rows x 14 columns]"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gemini_FDR_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "3f2002a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene 1</th>\n",
       "      <th>gene 2</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>GI1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HS936T_SKIN</th>\n",
       "      <th>HS944T_SKIN</th>\n",
       "      <th>HSC5_SKIN</th>\n",
       "      <th>IPC298_SKIN</th>\n",
       "      <th>MEL202_UVEA</th>\n",
       "      <th>MELJUSO_SKIN</th>\n",
       "      <th>MEWO_SKIN</th>\n",
       "      <th>PATU8988S_PANCREAS</th>\n",
       "      <th>PK1_PANCREAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>ABO</td>\n",
       "      <td>0.921199</td>\n",
       "      <td>0.332505</td>\n",
       "      <td>0.424278</td>\n",
       "      <td>0.837115</td>\n",
       "      <td>0.336008</td>\n",
       "      <td>0.764955</td>\n",
       "      <td>0.935477</td>\n",
       "      <td>0.921540</td>\n",
       "      <td>0.955418</td>\n",
       "      <td>0.584562</td>\n",
       "      <td>0.563288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GBGT1</td>\n",
       "      <td>0.751001</td>\n",
       "      <td>0.263071</td>\n",
       "      <td>0.307590</td>\n",
       "      <td>0.381261</td>\n",
       "      <td>0.135696</td>\n",
       "      <td>0.547510</td>\n",
       "      <td>0.977996</td>\n",
       "      <td>0.755984</td>\n",
       "      <td>0.896801</td>\n",
       "      <td>0.380175</td>\n",
       "      <td>0.172056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GLT6D1</td>\n",
       "      <td>0.570461</td>\n",
       "      <td>0.487910</td>\n",
       "      <td>0.622668</td>\n",
       "      <td>0.741919</td>\n",
       "      <td>0.020182</td>\n",
       "      <td>0.571000</td>\n",
       "      <td>0.974356</td>\n",
       "      <td>0.884787</td>\n",
       "      <td>0.866267</td>\n",
       "      <td>0.553824</td>\n",
       "      <td>0.165021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC</td>\n",
       "      <td>AADACL2</td>\n",
       "      <td>0.139850</td>\n",
       "      <td>0.776708</td>\n",
       "      <td>0.196587</td>\n",
       "      <td>0.542847</td>\n",
       "      <td>0.248709</td>\n",
       "      <td>0.904911</td>\n",
       "      <td>0.726176</td>\n",
       "      <td>0.058205</td>\n",
       "      <td>0.204460</td>\n",
       "      <td>0.652104</td>\n",
       "      <td>0.389172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8344</th>\n",
       "      <td>ZNF362</td>\n",
       "      <td>ZNF384</td>\n",
       "      <td>0.981917</td>\n",
       "      <td>0.806990</td>\n",
       "      <td>0.725792</td>\n",
       "      <td>0.920152</td>\n",
       "      <td>0.978128</td>\n",
       "      <td>0.997034</td>\n",
       "      <td>0.999124</td>\n",
       "      <td>0.960840</td>\n",
       "      <td>0.735925</td>\n",
       "      <td>0.969437</td>\n",
       "      <td>0.688228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8345</th>\n",
       "      <td>ZNF423</td>\n",
       "      <td>ZNF521</td>\n",
       "      <td>0.886454</td>\n",
       "      <td>0.616188</td>\n",
       "      <td>0.921000</td>\n",
       "      <td>0.834580</td>\n",
       "      <td>0.597923</td>\n",
       "      <td>0.640425</td>\n",
       "      <td>0.964026</td>\n",
       "      <td>0.830805</td>\n",
       "      <td>0.918407</td>\n",
       "      <td>0.948940</td>\n",
       "      <td>0.849562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8346</th>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>ZNRF2</td>\n",
       "      <td>0.562628</td>\n",
       "      <td>0.936957</td>\n",
       "      <td>0.720903</td>\n",
       "      <td>0.858420</td>\n",
       "      <td>0.754345</td>\n",
       "      <td>0.801502</td>\n",
       "      <td>0.475185</td>\n",
       "      <td>0.186117</td>\n",
       "      <td>0.818102</td>\n",
       "      <td>0.609197</td>\n",
       "      <td>0.934004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8347</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.970700</td>\n",
       "      <td>0.781413</td>\n",
       "      <td>0.992768</td>\n",
       "      <td>0.978433</td>\n",
       "      <td>0.956582</td>\n",
       "      <td>0.862882</td>\n",
       "      <td>0.995520</td>\n",
       "      <td>0.880310</td>\n",
       "      <td>0.973317</td>\n",
       "      <td>0.957046</td>\n",
       "      <td>0.984665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8348</th>\n",
       "      <td>ZXDB</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.958072</td>\n",
       "      <td>0.960969</td>\n",
       "      <td>0.959880</td>\n",
       "      <td>0.954196</td>\n",
       "      <td>0.875061</td>\n",
       "      <td>0.985088</td>\n",
       "      <td>0.991263</td>\n",
       "      <td>0.993564</td>\n",
       "      <td>0.931364</td>\n",
       "      <td>0.927340</td>\n",
       "      <td>0.962825</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8349 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene 1   gene 2  A549_LUNG  GI1_CENTRAL_NERVOUS_SYSTEM  HS936T_SKIN  \\\n",
       "0     A3GALT2    AAVS1        NaN                         NaN          NaN   \n",
       "1     A3GALT2      ABO   0.921199                    0.332505     0.424278   \n",
       "2     A3GALT2    GBGT1   0.751001                    0.263071     0.307590   \n",
       "3     A3GALT2   GLT6D1   0.570461                    0.487910     0.622668   \n",
       "4       AADAC  AADACL2   0.139850                    0.776708     0.196587   \n",
       "...       ...      ...        ...                         ...          ...   \n",
       "8344   ZNF362   ZNF384   0.981917                    0.806990     0.725792   \n",
       "8345   ZNF423   ZNF521   0.886454                    0.616188     0.921000   \n",
       "8346    ZNRF1    ZNRF2   0.562628                    0.936957     0.720903   \n",
       "8347     ZXDA     ZXDC   0.970700                    0.781413     0.992768   \n",
       "8348     ZXDB     ZXDC   0.958072                    0.960969     0.959880   \n",
       "\n",
       "      HS944T_SKIN  HSC5_SKIN  IPC298_SKIN  MEL202_UVEA  MELJUSO_SKIN  \\\n",
       "0             NaN        NaN          NaN          NaN           NaN   \n",
       "1        0.837115   0.336008     0.764955     0.935477      0.921540   \n",
       "2        0.381261   0.135696     0.547510     0.977996      0.755984   \n",
       "3        0.741919   0.020182     0.571000     0.974356      0.884787   \n",
       "4        0.542847   0.248709     0.904911     0.726176      0.058205   \n",
       "...           ...        ...          ...          ...           ...   \n",
       "8344     0.920152   0.978128     0.997034     0.999124      0.960840   \n",
       "8345     0.834580   0.597923     0.640425     0.964026      0.830805   \n",
       "8346     0.858420   0.754345     0.801502     0.475185      0.186117   \n",
       "8347     0.978433   0.956582     0.862882     0.995520      0.880310   \n",
       "8348     0.954196   0.875061     0.985088     0.991263      0.993564   \n",
       "\n",
       "      MEWO_SKIN  PATU8988S_PANCREAS  PK1_PANCREAS  \n",
       "0           NaN                 NaN           NaN  \n",
       "1      0.955418            0.584562      0.563288  \n",
       "2      0.896801            0.380175      0.172056  \n",
       "3      0.866267            0.553824      0.165021  \n",
       "4      0.204460            0.652104      0.389172  \n",
       "...         ...                 ...           ...  \n",
       "8344   0.735925            0.969437      0.688228  \n",
       "8345   0.918407            0.948940      0.849562  \n",
       "8346   0.818102            0.609197      0.934004  \n",
       "8347   0.973317            0.957046      0.984665  \n",
       "8348   0.931364            0.927340      0.962825  \n",
       "\n",
       "[8349 rows x 13 columns]"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "updated_gemini_FDR_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "cebb7e02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'MEWO': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'HSC5': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'PATU8988S': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads']],\n",
       " 'PK1': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'MELJUSO': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'HS944T': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'HS936T': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'MEL202': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'A549': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'GI1': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']],\n",
       " 'IPC298': [['pDNA_Reads'], ['Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads']]}"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_conditions['ito_data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "0674710b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "Only Stat cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 1002870\n",
      "Number of controls: 0\n",
      "Number of singles: 216744\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = all_GI.copy(), study_controls = controls['ito_data'], study_conditions = study_conditions['ito_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "5b9de586",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "      <th>study_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2_1</td>\n",
       "      <td>AAATAATGGGAGCCCCCCAG</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A3GALT2_2</td>\n",
       "      <td>GCGCTGGCAAGACGTGTCGA</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3GALT2_3</td>\n",
       "      <td>ACACGTAGTACATCACGCTC</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2_4</td>\n",
       "      <td>GCAAAGATAGTCAGCCCAATG</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2_5</td>\n",
       "      <td>GGCCGAAGGGCAGACGCCCAT</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19701</th>\n",
       "      <td>ZXDC_4</td>\n",
       "      <td>ACCTGCAGAACCACTAGCAT</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19702</th>\n",
       "      <td>ZXDC_3</td>\n",
       "      <td>CTCTTACCTCAGCTAGAAGCT</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19703</th>\n",
       "      <td>ZXDC_2</td>\n",
       "      <td>CCCACCTAGGCACAAAGCCGT</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19704</th>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>CTGAGCGGATCCTCCCCGTGT</td>\n",
       "      <td>CONTROL</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19705</th>\n",
       "      <td>AAVS1_2</td>\n",
       "      <td>AGGGAGACATCCGTCGGAGA</td>\n",
       "      <td>CONTROL</td>\n",
       "      <td>34857952</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19706 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sgRNA_guide_name        sgRNA_guide_seq sgRNA_target_name study_origin\n",
       "0            A3GALT2_1   AAATAATGGGAGCCCCCCAG           A3GALT2     34857952\n",
       "1            A3GALT2_2   GCGCTGGCAAGACGTGTCGA           A3GALT2     34857952\n",
       "2            A3GALT2_3   ACACGTAGTACATCACGCTC           A3GALT2     34857952\n",
       "3            A3GALT2_4  GCAAAGATAGTCAGCCCAATG           A3GALT2     34857952\n",
       "4            A3GALT2_5  GGCCGAAGGGCAGACGCCCAT           A3GALT2     34857952\n",
       "...                ...                    ...               ...          ...\n",
       "19701           ZXDC_4   ACCTGCAGAACCACTAGCAT              ZXDC     34857952\n",
       "19702           ZXDC_3  CTCTTACCTCAGCTAGAAGCT              ZXDC     34857952\n",
       "19703           ZXDC_2  CCCACCTAGGCACAAAGCCGT              ZXDC     34857952\n",
       "19704          AAVS1_1  CTGAGCGGATCCTCCCCGTGT           CONTROL     34857952\n",
       "19705          AAVS1_2   AGGGAGACATCCGTCGGAGA           CONTROL     34857952\n",
       "\n",
       "[19706 rows x 4 columns]"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['sequence_ref']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "0de7b46d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>cell_line_origin</th>\n",
       "      <th>Condition</th>\n",
       "      <th>study_origin</th>\n",
       "      <th>target_type</th>\n",
       "      <th>T0_counts</th>\n",
       "      <th>T0_replicate_names</th>\n",
       "      <th>TEnd_counts</th>\n",
       "      <th>TEnd_replicate_names</th>\n",
       "      <th>gene_pair</th>\n",
       "      <th>gene_pair_orientation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABO_1</td>\n",
       "      <td>A3GALT2_1</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>1286;1707;358;184</td>\n",
       "      <td>MEWO</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Dual</td>\n",
       "      <td>1286.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>1707.0;358.0;184.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>A3GALT2|ABO</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABO_2</td>\n",
       "      <td>A3GALT2_2</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>3430;2962;703;328</td>\n",
       "      <td>MEWO</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Dual</td>\n",
       "      <td>3430.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>2962.0;703.0;328.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>A3GALT2|ABO</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABO_2</td>\n",
       "      <td>A3GALT2_1</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2002;1954;505;210</td>\n",
       "      <td>MEWO</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Dual</td>\n",
       "      <td>2002.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>1954.0;505.0;210.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>A3GALT2|ABO</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABO_3</td>\n",
       "      <td>A3GALT2_3</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>4822;4475;1140;555</td>\n",
       "      <td>MEWO</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Dual</td>\n",
       "      <td>4822.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>4475.0;1140.0;555.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>A3GALT2|ABO</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABO_1</td>\n",
       "      <td>A3GALT2_3</td>\n",
       "      <td>ABO</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>2103;1801;463;202</td>\n",
       "      <td>MEWO</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Dual</td>\n",
       "      <td>2103.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>1801.0;463.0;202.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>A3GALT2|ABO</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219609</th>\n",
       "      <td>ZXDC_1</td>\n",
       "      <td>AAVS1_2</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>2175;991;1650;422</td>\n",
       "      <td>IPC298</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Single</td>\n",
       "      <td>2175.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>991.0;1650.0;422.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>AAVS1|ZXDC</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219610</th>\n",
       "      <td>ZXDC_2</td>\n",
       "      <td>AAVS1_2</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>1813;704;1148;340</td>\n",
       "      <td>IPC298</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Single</td>\n",
       "      <td>1813.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>704.0;1148.0;340.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>AAVS1|ZXDC</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219611</th>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>ZXDC_4</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>2343;388;808;246</td>\n",
       "      <td>IPC298</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Single</td>\n",
       "      <td>2343.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>388.0;808.0;246.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>AAVS1|ZXDC</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219612</th>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>ZXDC_6</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>1688;436;826;198</td>\n",
       "      <td>IPC298</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Single</td>\n",
       "      <td>1688.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>436.0;826.0;198.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>AAVS1|ZXDC</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219613</th>\n",
       "      <td>AAVS1_1</td>\n",
       "      <td>ZXDC_5</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>970;217;460;136</td>\n",
       "      <td>IPC298</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>34857952</td>\n",
       "      <td>Single</td>\n",
       "      <td>970.0</td>\n",
       "      <td>pDNA_Reads</td>\n",
       "      <td>217.0;460.0;136.0</td>\n",
       "      <td>Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>AAVS1|ZXDC</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1219614 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Guide 1    Guide 2 Gene 1   Gene 2    Count Replicates  \\\n",
       "0          ABO_1  A3GALT2_1    ABO  A3GALT2   1286;1707;358;184   \n",
       "1          ABO_2  A3GALT2_2    ABO  A3GALT2   3430;2962;703;328   \n",
       "2          ABO_2  A3GALT2_1    ABO  A3GALT2   2002;1954;505;210   \n",
       "3          ABO_3  A3GALT2_3    ABO  A3GALT2  4822;4475;1140;555   \n",
       "4          ABO_1  A3GALT2_3    ABO  A3GALT2   2103;1801;463;202   \n",
       "...          ...        ...    ...      ...                 ...   \n",
       "1219609   ZXDC_1    AAVS1_2   ZXDC    AAVS1   2175;991;1650;422   \n",
       "1219610   ZXDC_2    AAVS1_2   ZXDC    AAVS1   1813;704;1148;340   \n",
       "1219611  AAVS1_1     ZXDC_4  AAVS1     ZXDC    2343;388;808;246   \n",
       "1219612  AAVS1_1     ZXDC_6  AAVS1     ZXDC    1688;436;826;198   \n",
       "1219613  AAVS1_1     ZXDC_5  AAVS1     ZXDC     970;217;460;136   \n",
       "\n",
       "        cell_line_origin                                       Condition  \\\n",
       "0                   MEWO  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1                   MEWO  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "2                   MEWO  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "3                   MEWO  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "4                   MEWO  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "...                  ...                                             ...   \n",
       "1219609           IPC298  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219610           IPC298  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219611           IPC298  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219612           IPC298  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219613           IPC298  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "\n",
       "        study_origin target_type T0_counts T0_replicate_names  \\\n",
       "0           34857952        Dual    1286.0         pDNA_Reads   \n",
       "1           34857952        Dual    3430.0         pDNA_Reads   \n",
       "2           34857952        Dual    2002.0         pDNA_Reads   \n",
       "3           34857952        Dual    4822.0         pDNA_Reads   \n",
       "4           34857952        Dual    2103.0         pDNA_Reads   \n",
       "...              ...         ...       ...                ...   \n",
       "1219609     34857952      Single    2175.0         pDNA_Reads   \n",
       "1219610     34857952      Single    1813.0         pDNA_Reads   \n",
       "1219611     34857952      Single    2343.0         pDNA_Reads   \n",
       "1219612     34857952      Single    1688.0         pDNA_Reads   \n",
       "1219613     34857952      Single     970.0         pDNA_Reads   \n",
       "\n",
       "                 TEnd_counts                 TEnd_replicate_names  \\\n",
       "0         1707.0;358.0;184.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1         2962.0;703.0;328.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "2         1954.0;505.0;210.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "3        4475.0;1140.0;555.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "4         1801.0;463.0;202.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "...                      ...                                  ...   \n",
       "1219609   991.0;1650.0;422.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219610   704.0;1148.0;340.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219611    388.0;808.0;246.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219612    436.0;826.0;198.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "1219613    217.0;460.0;136.0  Rep_A_Reads;Rep_B_Reads;Rep_C_Reads   \n",
       "\n",
       "           gene_pair gene_pair_orientation  \n",
       "0        A3GALT2|ABO                   B_A  \n",
       "1        A3GALT2|ABO                   B_A  \n",
       "2        A3GALT2|ABO                   B_A  \n",
       "3        A3GALT2|ABO                   B_A  \n",
       "4        A3GALT2|ABO                   B_A  \n",
       "...              ...                   ...  \n",
       "1219609   AAVS1|ZXDC                   B_A  \n",
       "1219610   AAVS1|ZXDC                   B_A  \n",
       "1219611   AAVS1|ZXDC                   A_B  \n",
       "1219612   AAVS1|ZXDC                   A_B  \n",
       "1219613   AAVS1|ZXDC                   A_B  \n",
       "\n",
       "[1219614 rows x 15 columns]"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['counts_ref']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3d68179",
   "metadata": {},
   "outputs": [],
   "source": [
    "## ITO Specific, fix the March genes\n",
    "change = {'1-MAR': 'MARCH1',\n",
    "         '2-MAR': 'MARCH2',\n",
    "         '3-MAR': 'MARCH3',\n",
    "         '4-MAR': 'MARCH4',\n",
    "         '8-MAR': 'MARCH8',\n",
    "         '9-MAR': 'MARCH9',\n",
    "         '11-MAR': 'MARCH11'}\n",
    "\n",
    "for gene in change:\n",
    "    # update sequence ref\n",
    "    db_inserts['sequence_ref'].loc[db_inserts['sequence_ref']['sgRNA_target'] == gene, 'sgRNA_target'] = change[gene]\n",
    "    \n",
    "    # update counts ref\n",
    "    db_inserts['counts_ref'].loc[db_inserts['counts_ref']['sgRNA_target_1'] == gene, 'sgRNA_target_1'] = change[gene]\n",
    "    db_inserts['counts_ref'].loc[db_inserts['counts_ref']['sgRNA_target_1'] == gene, 'sgRNA_target_1'] = change[gene]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "aa74e8d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF['Study_Source'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "db71c8f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF.loc[PREV_REF['Study_Source'] == 'Ito', :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "3e01b80d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(set(PREV_REF.loc[PREV_REF['Study_Source'] == 'Ito', 'Gene Pair']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "c8e230bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#set(PREV_REF.loc[PREV_REF['Study_Source'] == 'Ito', 'Gene Pair']).difference(set(db_inserts['score_ref']['gene_pair']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "be7a72d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(set(db_inserts['score_ref']['gene_pair']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "9ec3341e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 19706\n",
      "Counts insert: 1219614\n",
      "Score insert: 53449\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "858fb79d",
   "metadata": {},
   "source": [
    "## Parrish Data (PC9, HeLa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "212539d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "parrish_loc = os.path.join(learning_goals_loc_general, \"Parrish\")\n",
    "\n",
    "sgRNA_ref = pd.read_excel(os.path.join(parrish_loc, \"1 - pgPEN library composition.xlsx\"), skiprows = 1)\n",
    "PC9_sgRNA = pd.read_excel(os.path.join(parrish_loc, \"3 - PC9 screen pgRNA CRISPR scores and normalized counts across three biological replicates.xlsx\"), skiprows = 1)\n",
    "HELA_sgRNA = pd.read_excel(os.path.join(parrish_loc, \"6 - HeLa screen pgRNA CRISPR scores and normalized counts across three biological replicates.xlsx\"), skiprows = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b89058f",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "bad0468b",
   "metadata": {},
   "outputs": [],
   "source": [
    "old_to_new_map = {}\n",
    "old_IDs = sgRNA_ref.loc[:, 'pgRNA_ID'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "ee325c2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pgRNA_ID</th>\n",
       "      <th>target1_sgRNA_seq</th>\n",
       "      <th>target2_sgRNA_seq</th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>target_type</th>\n",
       "      <th>target1</th>\n",
       "      <th>target2</th>\n",
       "      <th>target1_ensembl_id</th>\n",
       "      <th>target2_ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M_nt1</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "      <td>ATACGGCCGAAGCCCCTTCA</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M_nt2</td>\n",
       "      <td>ATGTCTCATGAACTACCCTG</td>\n",
       "      <td>CTCTGTGAGATGTCCCGAAA</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M_nt3</td>\n",
       "      <td>TGAAATGAAACTTCACACTG</td>\n",
       "      <td>AGGGCACCCGGTTCATACGC</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M_nt4</td>\n",
       "      <td>TTACTCATATAGGATCCCAA</td>\n",
       "      <td>CCAATGATAAGCCCGAACGG</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M_nt5</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "      <td>TTCGAGGTCCGGACAGGTCG</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33165</th>\n",
       "      <td>ZYG11B_ZYG11A_pg5</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>AATGCACTATCTGAAATGCC</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33166</th>\n",
       "      <td>ZYG11B_ZYG11A_pg6</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>CATCACCCTGAGAAATCGCT</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33167</th>\n",
       "      <td>ZYG11B_ZYG11A_pg7</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>GCTCTCAACCTAACACGCCA</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33168</th>\n",
       "      <td>ZYG11B_ZYG11A_pg8</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>TAGAACTGTTTATACGACTG</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33169</th>\n",
       "      <td>ZYG11B_ZYG11A_pg9</td>\n",
       "      <td>GCAGTGGCTTTGCAACCATG</td>\n",
       "      <td>AATGCACTATCTGAAATGCC</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33170 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                pgRNA_ID     target1_sgRNA_seq     target2_sgRNA_seq  \\\n",
       "0                A2M_nt1  ACTGCATCTGTGCAAACGGG  ATACGGCCGAAGCCCCTTCA   \n",
       "1                A2M_nt2  ATGTCTCATGAACTACCCTG  CTCTGTGAGATGTCCCGAAA   \n",
       "2                A2M_nt3  TGAAATGAAACTTCACACTG  AGGGCACCCGGTTCATACGC   \n",
       "3                A2M_nt4  TTACTCATATAGGATCCCAA  CCAATGATAAGCCCGAACGG   \n",
       "4                A2M_nt5  ACTGCATCTGTGCAAACGGG  TTCGAGGTCCGGACAGGTCG   \n",
       "...                  ...                   ...                   ...   \n",
       "33165  ZYG11B_ZYG11A_pg5  CAAGTTCCACTAACTTGTGG  AATGCACTATCTGAAATGCC   \n",
       "33166  ZYG11B_ZYG11A_pg6  CAAGTTCCACTAACTTGTGG  CATCACCCTGAGAAATCGCT   \n",
       "33167  ZYG11B_ZYG11A_pg7  CAAGTTCCACTAACTTGTGG  GCTCTCAACCTAACACGCCA   \n",
       "33168  ZYG11B_ZYG11A_pg8  CAAGTTCCACTAACTTGTGG  TAGAACTGTTTATACGACTG   \n",
       "33169  ZYG11B_ZYG11A_pg9  GCAGTGGCTTTGCAACCATG  AATGCACTATCTGAAATGCC   \n",
       "\n",
       "        paralog_pair target_type target1 target2 target1_ensembl_id  \\\n",
       "0            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "1            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "2            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "3            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "4            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "...              ...         ...     ...     ...                ...   \n",
       "33165  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33166  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33167  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33168  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33169  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "\n",
       "      target2_ensembl_id  \n",
       "0                    NaN  \n",
       "1                    NaN  \n",
       "2                    NaN  \n",
       "3                    NaN  \n",
       "4                    NaN  \n",
       "...                  ...  \n",
       "33165    ENSG00000203995  \n",
       "33166    ENSG00000203995  \n",
       "33167    ENSG00000203995  \n",
       "33168    ENSG00000203995  \n",
       "33169    ENSG00000203995  \n",
       "\n",
       "[33170 rows x 9 columns]"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "60b45ce7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene_gene    16396\n",
       "ntc_gene      8149\n",
       "gene_ntc      8125\n",
       "ntc_ntc        500\n",
       "Name: target_type, dtype: int64"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref['target_type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "0e518844",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_control_single_sgRNA_sequences = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "6141a2e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_control_single_sgRNA_sequences += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'ntc_gene']['target1_sgRNA_seq'].values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "e0fb1022",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_control_single_sgRNA_sequences += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_ntc']['target2_sgRNA_seq'].values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "6dc84648",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_control_single_sgRNA_sequences = set(all_control_single_sgRNA_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "070c642b",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_control_dual_sgRNA_sequences = set(sgRNA_ref.loc[sgRNA_ref['target_type'] == 'ntc_ntc']['target1_sgRNA_seq'].values.tolist() + sgRNA_ref.loc[sgRNA_ref['target_type'] == 'ntc_ntc']['target2_sgRNA_seq'].values.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "56055e17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "975"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_control_single_sgRNA_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "2c7dc828",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "967"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_control_dual_sgRNA_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "7e0baf36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_control_dual_sgRNA_sequences.difference(all_control_single_sgRNA_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "baa60ef8",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_to_control_map = {}\n",
    "i = 1\n",
    "for seq in all_control_single_sgRNA_sequences:\n",
    "    seq_to_control_map[seq] = \"nt\" + str(i)\n",
    "    i += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "346598c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_gene_sgRNA_sequences = []\n",
    "all_gene_sgRNA_sequences += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'ntc_gene']['target2_sgRNA_seq'].values.tolist()\n",
    "all_gene_sgRNA_sequences += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_ntc']['target1_sgRNA_seq'].values.tolist()\n",
    "all_gene_sgRNA_sequences += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_gene']['target1_sgRNA_seq'].values.tolist()\n",
    "all_gene_sgRNA_sequences += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_gene']['target2_sgRNA_seq'].values.tolist()\n",
    "\n",
    "all_genes = []\n",
    "all_genes += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'ntc_gene']['target2'].values.tolist()\n",
    "all_genes += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_ntc']['target1'].values.tolist()\n",
    "all_genes += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_gene']['target1'].values.tolist()\n",
    "all_genes += sgRNA_ref.loc[sgRNA_ref['target_type'] == 'gene_gene']['target2'].values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "e69e2002",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_to_gene_map = {}\n",
    "for i in range(len(all_genes)):\n",
    "    if all_genes[i] not in seq_to_gene_map:\n",
    "        seq_to_gene_map[all_genes[i]] = []\n",
    "    if all_gene_sgRNA_sequences[i] not in seq_to_gene_map[all_genes[i]]:\n",
    "        seq_to_gene_map[all_genes[i]].append(all_gene_sgRNA_sequences[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "6354ecb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_to_guide_map = {}\n",
    "for i in range(len(sgRNA_ref['pgRNA_ID'])):\n",
    "    curr_ID = sgRNA_ref['pgRNA_ID'][i]\n",
    "    target1_seq = sgRNA_ref['target1_sgRNA_seq'][i]\n",
    "    target2_seq = sgRNA_ref['target2_sgRNA_seq'][i]\n",
    "    target1_name = sgRNA_ref['target1'][i]\n",
    "    target2_name = sgRNA_ref['target2'][i]\n",
    "    cond = sgRNA_ref['target_type'][i]\n",
    "    \n",
    "    new_target_1 = target1_name\n",
    "    new_target_2 = target2_name\n",
    "    if cond == 'gene_gene':\n",
    "        new_target_1 = target1_name + '+' + str(seq_to_gene_map[target1_name].index(target1_seq) + 1)\n",
    "        new_target_2 = target2_name + '+' + str(seq_to_gene_map[target2_name].index(target2_seq) + 1)\n",
    "    elif cond == 'gene_ntc':\n",
    "        new_target_1 = target1_name + '+' + str(seq_to_gene_map[target1_name].index(target1_seq) + 1)\n",
    "        new_target_2 = curr_ID.split('_')[1]\n",
    "    elif cond == 'ntc_gene':\n",
    "        new_target_1 = curr_ID.split('_')[0]\n",
    "        new_target_2 = target2_name + '+' + str(seq_to_gene_map[target2_name].index(target2_seq) + 1)\n",
    "    else:\n",
    "        continue\n",
    "    new_guide = '_'.join([new_target_1, new_target_2])\n",
    "    \n",
    "    guide_to_guide_map[curr_ID] = new_guide\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "ea749d41",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref['pgRNA_ID'] =  [guide_to_guide_map[i] if i in guide_to_guide_map else i for i in sgRNA_ref['pgRNA_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "e73dbed5",
   "metadata": {},
   "outputs": [],
   "source": [
    "## replace the original values\n",
    "curr_idx = sgRNA_ref['target_type'] == 'gene_ntc'\n",
    "\n",
    "gene_1 = [i.split('_')[0] for i in sgRNA_ref.loc[curr_idx]['pgRNA_ID']]\n",
    "gene_2 = [seq_to_control_map[i] for i in sgRNA_ref.loc[curr_idx]['target2_sgRNA_seq']]\n",
    "\n",
    "sgRNA_ref.loc[curr_idx, 'pgRNA_ID'] = [gene_1[i] + \"_\" + gene_2[i] for i in range(sum(curr_idx))]#[gene_1[i] + \"+\" + str(i+1) + \"_\" + gene_2[i] for i in range(sum(curr_idx))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "c2e51c11",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_idx = sgRNA_ref['target_type'] == 'ntc_gene'\n",
    "\n",
    "gene_1 = [seq_to_control_map[i] for i in sgRNA_ref.loc[curr_idx]['target1_sgRNA_seq']]\n",
    "gene_2 = [i.split('_')[1] for i in sgRNA_ref.loc[curr_idx]['pgRNA_ID']]\n",
    "\n",
    "sgRNA_ref.loc[curr_idx, 'pgRNA_ID'] = [gene_1[i] + \"_\" + gene_2[i] for i in range(sum(curr_idx))]#[gene_1[i] + \"_\" + gene_2[i] + \"+\" + str(i+1) for i in range(sum(curr_idx))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "c7d34398",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_idx = sgRNA_ref['target_type'] == 'ntc_ntc'\n",
    "\n",
    "gene_1 = [seq_to_control_map[i] for i in sgRNA_ref.loc[curr_idx]['target1_sgRNA_seq']]\n",
    "gene_2 = [seq_to_control_map[i] for i in sgRNA_ref.loc[curr_idx]['target2_sgRNA_seq']]\n",
    "\n",
    "sgRNA_ref.loc[curr_idx, 'pgRNA_ID'] = [gene_1[i] + \"_\" + gene_2[i] for i in range(sum(curr_idx))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "7a1f751d",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_IDs = sgRNA_ref['pgRNA_ID'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "a5f63c56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(old_IDs) == len(new_IDs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "7a4ecfe1",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(len(old_IDs)):\n",
    "    old_to_new_map[old_IDs[i]] = new_IDs[i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "1365c366",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pgRNA_ID</th>\n",
       "      <th>target1_sgRNA_seq</th>\n",
       "      <th>target2_sgRNA_seq</th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>target_type</th>\n",
       "      <th>target1</th>\n",
       "      <th>target2</th>\n",
       "      <th>target1_ensembl_id</th>\n",
       "      <th>target2_ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1_nt404</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "      <td>ATACGGCCGAAGCCCCTTCA</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2_nt598</td>\n",
       "      <td>ATGTCTCATGAACTACCCTG</td>\n",
       "      <td>CTCTGTGAGATGTCCCGAAA</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3_nt847</td>\n",
       "      <td>TGAAATGAAACTTCACACTG</td>\n",
       "      <td>AGGGCACCCGGTTCATACGC</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4_nt731</td>\n",
       "      <td>TTACTCATATAGGATCCCAA</td>\n",
       "      <td>CCAATGATAAGCCCGAACGG</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1_nt54</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "      <td>TTCGAGGTCCGGACAGGTCG</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>gene_ntc</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33165</th>\n",
       "      <td>ZYG11B+2_ZYG11A+1</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>AATGCACTATCTGAAATGCC</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33166</th>\n",
       "      <td>ZYG11B+2_ZYG11A+2</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>CATCACCCTGAGAAATCGCT</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33167</th>\n",
       "      <td>ZYG11B+2_ZYG11A+3</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>GCTCTCAACCTAACACGCCA</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33168</th>\n",
       "      <td>ZYG11B+2_ZYG11A+4</td>\n",
       "      <td>CAAGTTCCACTAACTTGTGG</td>\n",
       "      <td>TAGAACTGTTTATACGACTG</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33169</th>\n",
       "      <td>ZYG11B+3_ZYG11A+1</td>\n",
       "      <td>GCAGTGGCTTTGCAACCATG</td>\n",
       "      <td>AATGCACTATCTGAAATGCC</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>gene_gene</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33170 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                pgRNA_ID     target1_sgRNA_seq     target2_sgRNA_seq  \\\n",
       "0            A2M+1_nt404  ACTGCATCTGTGCAAACGGG  ATACGGCCGAAGCCCCTTCA   \n",
       "1            A2M+2_nt598  ATGTCTCATGAACTACCCTG  CTCTGTGAGATGTCCCGAAA   \n",
       "2            A2M+3_nt847  TGAAATGAAACTTCACACTG  AGGGCACCCGGTTCATACGC   \n",
       "3            A2M+4_nt731  TTACTCATATAGGATCCCAA  CCAATGATAAGCCCGAACGG   \n",
       "4             A2M+1_nt54  ACTGCATCTGTGCAAACGGG  TTCGAGGTCCGGACAGGTCG   \n",
       "...                  ...                   ...                   ...   \n",
       "33165  ZYG11B+2_ZYG11A+1  CAAGTTCCACTAACTTGTGG  AATGCACTATCTGAAATGCC   \n",
       "33166  ZYG11B+2_ZYG11A+2  CAAGTTCCACTAACTTGTGG  CATCACCCTGAGAAATCGCT   \n",
       "33167  ZYG11B+2_ZYG11A+3  CAAGTTCCACTAACTTGTGG  GCTCTCAACCTAACACGCCA   \n",
       "33168  ZYG11B+2_ZYG11A+4  CAAGTTCCACTAACTTGTGG  TAGAACTGTTTATACGACTG   \n",
       "33169  ZYG11B+3_ZYG11A+1  GCAGTGGCTTTGCAACCATG  AATGCACTATCTGAAATGCC   \n",
       "\n",
       "        paralog_pair target_type target1 target2 target1_ensembl_id  \\\n",
       "0            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "1            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "2            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "3            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "4            A2M|PZP    gene_ntc     A2M     NaN    ENSG00000175899   \n",
       "...              ...         ...     ...     ...                ...   \n",
       "33165  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33166  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33167  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33168  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "33169  ZYG11B|ZYG11A   gene_gene  ZYG11B  ZYG11A    ENSG00000162378   \n",
       "\n",
       "      target2_ensembl_id  \n",
       "0                    NaN  \n",
       "1                    NaN  \n",
       "2                    NaN  \n",
       "3                    NaN  \n",
       "4                    NaN  \n",
       "...                  ...  \n",
       "33165    ENSG00000203995  \n",
       "33166    ENSG00000203995  \n",
       "33167    ENSG00000203995  \n",
       "33168    ENSG00000203995  \n",
       "33169    ENSG00000203995  \n",
       "\n",
       "[33170 rows x 9 columns]"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "0f3071dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "guides = []\n",
    "guides += [i.split('_')[0] for i in sgRNA_ref['pgRNA_ID'].tolist()]\n",
    "guides += [i.split('_')[1] for i in sgRNA_ref['pgRNA_ID'].tolist()]\n",
    "sequences = []\n",
    "sequences += sgRNA_ref['target1_sgRNA_seq'].tolist()\n",
    "sequences += sgRNA_ref['target2_sgRNA_seq'].tolist()\n",
    "\n",
    "sgRNA_ref2 = pd.DataFrame({\"Guide_ID\": guides, \n",
    "                           \"Sequence\" : sequences})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "ed04c732",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide_ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>ATGTCTCATGAACTACCCTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>TGAAATGAAACTTCACACTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>TTACTCATATAGGATCCCAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66335</th>\n",
       "      <td>ZYG11A+1</td>\n",
       "      <td>AATGCACTATCTGAAATGCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66336</th>\n",
       "      <td>ZYG11A+2</td>\n",
       "      <td>CATCACCCTGAGAAATCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66337</th>\n",
       "      <td>ZYG11A+3</td>\n",
       "      <td>GCTCTCAACCTAACACGCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66338</th>\n",
       "      <td>ZYG11A+4</td>\n",
       "      <td>TAGAACTGTTTATACGACTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66339</th>\n",
       "      <td>ZYG11A+1</td>\n",
       "      <td>AATGCACTATCTGAAATGCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66340 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Guide_ID              Sequence\n",
       "0         A2M+1  ACTGCATCTGTGCAAACGGG\n",
       "1         A2M+2  ATGTCTCATGAACTACCCTG\n",
       "2         A2M+3  TGAAATGAAACTTCACACTG\n",
       "3         A2M+4  TTACTCATATAGGATCCCAA\n",
       "4         A2M+1  ACTGCATCTGTGCAAACGGG\n",
       "...         ...                   ...\n",
       "66335  ZYG11A+1  AATGCACTATCTGAAATGCC\n",
       "66336  ZYG11A+2  CATCACCCTGAGAAATCGCT\n",
       "66337  ZYG11A+3  GCTCTCAACCTAACACGCCA\n",
       "66338  ZYG11A+4  TAGAACTGTTTATACGACTG\n",
       "66339  ZYG11A+1  AATGCACTATCTGAAATGCC\n",
       "\n",
       "[66340 rows x 2 columns]"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "83ae2bd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sgRNA_ref2.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "6db7dad6",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['Target'] = [i.split('+')[0] for i in sequence_ref['Guide_ID']]\n",
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "eca12ac5",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sequence_ref.drop_duplicates(subset=['sgRNA_guide_name'])\n",
    "sequence_ref = sequence_ref.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "1daacfa3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>ATGTCTCATGAACTACCCTG</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>TGAAATGAAACTTCACACTG</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>TTACTCATATAGGATCCCAA</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC+1</td>\n",
       "      <td>AAGTCTGAAGCACTAAGAAG</td>\n",
       "      <td>AADAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9189</th>\n",
       "      <td>PLAG1+4</td>\n",
       "      <td>TCGCCGGTTCTACACCCGAA</td>\n",
       "      <td>PLAG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9190</th>\n",
       "      <td>PRPF40A+4</td>\n",
       "      <td>TTGAAGTACTACCTGTATCG</td>\n",
       "      <td>PRPF40A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9191</th>\n",
       "      <td>PXDN+4</td>\n",
       "      <td>TGAAAACCTACGCGGAGTCG</td>\n",
       "      <td>PXDN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9192</th>\n",
       "      <td>RHEB+4</td>\n",
       "      <td>GCAAATTGTTGGATATGGTG</td>\n",
       "      <td>RHEB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9193</th>\n",
       "      <td>VASH1+4</td>\n",
       "      <td>GGAAAGGATGTGGAAACACG</td>\n",
       "      <td>VASH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9194 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0               A2M+1  ACTGCATCTGTGCAAACGGG               A2M\n",
       "1               A2M+2  ATGTCTCATGAACTACCCTG               A2M\n",
       "2               A2M+3  TGAAATGAAACTTCACACTG               A2M\n",
       "3               A2M+4  TTACTCATATAGGATCCCAA               A2M\n",
       "4             AADAC+1  AAGTCTGAAGCACTAAGAAG             AADAC\n",
       "...               ...                   ...               ...\n",
       "9189          PLAG1+4  TCGCCGGTTCTACACCCGAA             PLAG1\n",
       "9190        PRPF40A+4  TTGAAGTACTACCTGTATCG           PRPF40A\n",
       "9191           PXDN+4  TGAAAACCTACGCGGAGTCG              PXDN\n",
       "9192           RHEB+4  GCAAATTGTTGGATATGGTG              RHEB\n",
       "9193          VASH1+4  GGAAAGGATGTGGAAACACG             VASH1\n",
       "\n",
       "[9194 rows x 3 columns]"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e826bcbe",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "3711bfe1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace the guide ids\n",
    "PC9_sgRNA['pgRNA_ID'] = [old_to_new_map[i] if i in old_to_new_map else i for i in PC9_sgRNA['pgRNA_ID']]\n",
    "HELA_sgRNA['pgRNA'] = [old_to_new_map[i] if i in old_to_new_map else i for i in HELA_sgRNA['pgRNA']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "bd96aac9",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "9e0f81ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pgRNA_ID</th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>pgRNA_target</th>\n",
       "      <th>broad_target_type</th>\n",
       "      <th>mean_CRISPR_score</th>\n",
       "      <th>RepA_CRISPR_score</th>\n",
       "      <th>RepB_CRISPR_score</th>\n",
       "      <th>RepC_CRISPR_score</th>\n",
       "      <th>RepA_plasmid_count</th>\n",
       "      <th>RepB_plasmid_count</th>\n",
       "      <th>RepC_plasmid_count</th>\n",
       "      <th>RepA_LTP_count</th>\n",
       "      <th>RepB_LTP_count</th>\n",
       "      <th>RepC_LTP_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1_nt404</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.368602</td>\n",
       "      <td>0.495471</td>\n",
       "      <td>0.196626</td>\n",
       "      <td>0.413707</td>\n",
       "      <td>2435.90</td>\n",
       "      <td>2367.70</td>\n",
       "      <td>2182.40</td>\n",
       "      <td>3364.10</td>\n",
       "      <td>2893.90</td>\n",
       "      <td>3027.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2_nt598</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.483738</td>\n",
       "      <td>0.303549</td>\n",
       "      <td>0.531184</td>\n",
       "      <td>0.616480</td>\n",
       "      <td>868.75</td>\n",
       "      <td>844.39</td>\n",
       "      <td>778.32</td>\n",
       "      <td>1102.40</td>\n",
       "      <td>1224.90</td>\n",
       "      <td>1207.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3_nt847</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.368685</td>\n",
       "      <td>-0.456572</td>\n",
       "      <td>0.736245</td>\n",
       "      <td>0.826380</td>\n",
       "      <td>911.10</td>\n",
       "      <td>885.56</td>\n",
       "      <td>816.27</td>\n",
       "      <td>826.22</td>\n",
       "      <td>1426.70</td>\n",
       "      <td>1421.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4_nt731</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.041019</td>\n",
       "      <td>-1.270615</td>\n",
       "      <td>0.010252</td>\n",
       "      <td>1.383419</td>\n",
       "      <td>416.65</td>\n",
       "      <td>404.97</td>\n",
       "      <td>373.28</td>\n",
       "      <td>263.42</td>\n",
       "      <td>450.09</td>\n",
       "      <td>883.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1_nt54</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>-0.196169</td>\n",
       "      <td>-0.170177</td>\n",
       "      <td>-0.495369</td>\n",
       "      <td>0.077040</td>\n",
       "      <td>1852.50</td>\n",
       "      <td>1800.50</td>\n",
       "      <td>1659.60</td>\n",
       "      <td>1906.60</td>\n",
       "      <td>1544.90</td>\n",
       "      <td>1913.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32735</th>\n",
       "      <td>ZYG11B+2_ZYG11A+1</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.325519</td>\n",
       "      <td>-0.609131</td>\n",
       "      <td>-0.399958</td>\n",
       "      <td>0.032532</td>\n",
       "      <td>1926.80</td>\n",
       "      <td>1872.80</td>\n",
       "      <td>1726.20</td>\n",
       "      <td>1495.80</td>\n",
       "      <td>1525.80</td>\n",
       "      <td>1759.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32736</th>\n",
       "      <td>ZYG11B+2_ZYG11A+2</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.606071</td>\n",
       "      <td>-0.907578</td>\n",
       "      <td>-0.333128</td>\n",
       "      <td>-0.577508</td>\n",
       "      <td>2270.00</td>\n",
       "      <td>2206.30</td>\n",
       "      <td>2033.70</td>\n",
       "      <td>1544.50</td>\n",
       "      <td>1860.00</td>\n",
       "      <td>1482.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32737</th>\n",
       "      <td>ZYG11B+2_ZYG11A+3</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.763798</td>\n",
       "      <td>-0.759516</td>\n",
       "      <td>-0.782239</td>\n",
       "      <td>-0.749639</td>\n",
       "      <td>1378.80</td>\n",
       "      <td>1340.10</td>\n",
       "      <td>1235.20</td>\n",
       "      <td>1001.40</td>\n",
       "      <td>897.78</td>\n",
       "      <td>818.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32738</th>\n",
       "      <td>ZYG11B+2_ZYG11A+4</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.418372</td>\n",
       "      <td>0.221395</td>\n",
       "      <td>-1.528546</td>\n",
       "      <td>0.052036</td>\n",
       "      <td>272.29</td>\n",
       "      <td>264.66</td>\n",
       "      <td>243.95</td>\n",
       "      <td>305.19</td>\n",
       "      <td>120.58</td>\n",
       "      <td>251.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32739</th>\n",
       "      <td>ZYG11B+3_ZYG11A+1</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.149848</td>\n",
       "      <td>0.733271</td>\n",
       "      <td>-1.331134</td>\n",
       "      <td>0.148318</td>\n",
       "      <td>367.38</td>\n",
       "      <td>357.08</td>\n",
       "      <td>329.14</td>\n",
       "      <td>516.39</td>\n",
       "      <td>180.27</td>\n",
       "      <td>357.56</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32740 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                pgRNA_ID   paralog_pair   pgRNA_target broad_target_type  \\\n",
       "0            A2M+1_nt404        A2M|PZP        A2M|ntc  single_targeting   \n",
       "1            A2M+2_nt598        A2M|PZP        A2M|ntc  single_targeting   \n",
       "2            A2M+3_nt847        A2M|PZP        A2M|ntc  single_targeting   \n",
       "3            A2M+4_nt731        A2M|PZP        A2M|ntc  single_targeting   \n",
       "4             A2M+1_nt54        A2M|PZP        A2M|ntc  single_targeting   \n",
       "...                  ...            ...            ...               ...   \n",
       "32735  ZYG11B+2_ZYG11A+1  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32736  ZYG11B+2_ZYG11A+2  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32737  ZYG11B+2_ZYG11A+3  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32738  ZYG11B+2_ZYG11A+4  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32739  ZYG11B+3_ZYG11A+1  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "\n",
       "       mean_CRISPR_score  RepA_CRISPR_score  RepB_CRISPR_score  \\\n",
       "0               0.368602           0.495471           0.196626   \n",
       "1               0.483738           0.303549           0.531184   \n",
       "2               0.368685          -0.456572           0.736245   \n",
       "3               0.041019          -1.270615           0.010252   \n",
       "4              -0.196169          -0.170177          -0.495369   \n",
       "...                  ...                ...                ...   \n",
       "32735          -0.325519          -0.609131          -0.399958   \n",
       "32736          -0.606071          -0.907578          -0.333128   \n",
       "32737          -0.763798          -0.759516          -0.782239   \n",
       "32738          -0.418372           0.221395          -1.528546   \n",
       "32739          -0.149848           0.733271          -1.331134   \n",
       "\n",
       "       RepC_CRISPR_score  RepA_plasmid_count  RepB_plasmid_count  \\\n",
       "0               0.413707             2435.90             2367.70   \n",
       "1               0.616480              868.75              844.39   \n",
       "2               0.826380              911.10              885.56   \n",
       "3               1.383419              416.65              404.97   \n",
       "4               0.077040             1852.50             1800.50   \n",
       "...                  ...                 ...                 ...   \n",
       "32735           0.032532             1926.80             1872.80   \n",
       "32736          -0.577508             2270.00             2206.30   \n",
       "32737          -0.749639             1378.80             1340.10   \n",
       "32738           0.052036              272.29              264.66   \n",
       "32739           0.148318              367.38              357.08   \n",
       "\n",
       "       RepC_plasmid_count  RepA_LTP_count  RepB_LTP_count  RepC_LTP_count  \n",
       "0                 2182.40         3364.10         2893.90         3027.60  \n",
       "1                  778.32         1102.40         1224.90         1207.40  \n",
       "2                  816.27          826.22         1426.70         1421.20  \n",
       "3                  373.28          263.42          450.09          883.55  \n",
       "4                 1659.60         1906.60         1544.90         1913.50  \n",
       "...                   ...             ...             ...             ...  \n",
       "32735             1726.20         1495.80         1525.80         1759.30  \n",
       "32736             2033.70         1544.50         1860.00         1482.10  \n",
       "32737             1235.20         1001.40          897.78          818.77  \n",
       "32738              243.95          305.19          120.58          251.33  \n",
       "32739              329.14          516.39          180.27          357.56  \n",
       "\n",
       "[32740 rows x 14 columns]"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PC9_sgRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "817747c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pgRNA</th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>pgRNA_target</th>\n",
       "      <th>broad_target_type</th>\n",
       "      <th>mean_CRISPR_score</th>\n",
       "      <th>RepA_CRISPR_score</th>\n",
       "      <th>RepB_CRISPR_score</th>\n",
       "      <th>RepC_CRISPR_score</th>\n",
       "      <th>RepA_plasmid_count</th>\n",
       "      <th>RepB_plasmid_count</th>\n",
       "      <th>RepC_plasmid_count</th>\n",
       "      <th>RepA_LTP_count</th>\n",
       "      <th>RepB_LTP_count</th>\n",
       "      <th>RepC_LTP_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNPO2+1_nt687</td>\n",
       "      <td>TNPO2|TNPO1</td>\n",
       "      <td>TNPO2|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>1.404089</td>\n",
       "      <td>1.839640</td>\n",
       "      <td>1.220548</td>\n",
       "      <td>1.152079</td>\n",
       "      <td>238.76</td>\n",
       "      <td>266.44</td>\n",
       "      <td>267.51</td>\n",
       "      <td>1555.70</td>\n",
       "      <td>1106.90</td>\n",
       "      <td>1069.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>nt625_SEC23A+2</td>\n",
       "      <td>SEC23B|SEC23A</td>\n",
       "      <td>ntc|SEC23A</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>1.376141</td>\n",
       "      <td>1.719647</td>\n",
       "      <td>1.432025</td>\n",
       "      <td>0.976752</td>\n",
       "      <td>448.63</td>\n",
       "      <td>500.64</td>\n",
       "      <td>502.65</td>\n",
       "      <td>2625.00</td>\n",
       "      <td>2544.70</td>\n",
       "      <td>1691.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nt294_IL1RAPL1+4</td>\n",
       "      <td>IL1RAPL2|IL1RAPL1</td>\n",
       "      <td>ntc|IL1RAPL1</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.401452</td>\n",
       "      <td>1.764124</td>\n",
       "      <td>-0.273009</td>\n",
       "      <td>-0.286759</td>\n",
       "      <td>238.76</td>\n",
       "      <td>266.44</td>\n",
       "      <td>267.51</td>\n",
       "      <td>1455.30</td>\n",
       "      <td>263.25</td>\n",
       "      <td>261.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nt133_TGFB2+3</td>\n",
       "      <td>TGFB3|TGFB2</td>\n",
       "      <td>ntc|TGFB2</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.399684</td>\n",
       "      <td>1.297354</td>\n",
       "      <td>-0.345809</td>\n",
       "      <td>0.247506</td>\n",
       "      <td>2651.80</td>\n",
       "      <td>2959.30</td>\n",
       "      <td>2971.20</td>\n",
       "      <td>10675.00</td>\n",
       "      <td>2726.60</td>\n",
       "      <td>4892.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>nt625_PAFAH1B2+2</td>\n",
       "      <td>PAFAH1B3|PAFAH1B2</td>\n",
       "      <td>ntc|PAFAH1B2</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>1.474571</td>\n",
       "      <td>1.432591</td>\n",
       "      <td>1.727355</td>\n",
       "      <td>1.263767</td>\n",
       "      <td>1123.30</td>\n",
       "      <td>1253.50</td>\n",
       "      <td>1258.50</td>\n",
       "      <td>5096.60</td>\n",
       "      <td>8451.50</td>\n",
       "      <td>5602.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31828</th>\n",
       "      <td>nt787_nt679</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FAKE_GENE_25</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.025413</td>\n",
       "      <td>-0.600200</td>\n",
       "      <td>0.054393</td>\n",
       "      <td>0.469568</td>\n",
       "      <td>364.51</td>\n",
       "      <td>406.77</td>\n",
       "      <td>408.40</td>\n",
       "      <td>358.82</td>\n",
       "      <td>750.63</td>\n",
       "      <td>1125.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31829</th>\n",
       "      <td>nt401_nt577</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FAKE_GENE_14</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-1.004934</td>\n",
       "      <td>-0.574149</td>\n",
       "      <td>-1.615088</td>\n",
       "      <td>-0.825566</td>\n",
       "      <td>1119.00</td>\n",
       "      <td>1248.80</td>\n",
       "      <td>1253.80</td>\n",
       "      <td>1127.20</td>\n",
       "      <td>463.07</td>\n",
       "      <td>971.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31830</th>\n",
       "      <td>nt412_nt190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FAKE_GENE_25</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.114991</td>\n",
       "      <td>-0.576438</td>\n",
       "      <td>0.239666</td>\n",
       "      <td>-0.008203</td>\n",
       "      <td>814.83</td>\n",
       "      <td>909.30</td>\n",
       "      <td>912.96</td>\n",
       "      <td>819.14</td>\n",
       "      <td>2003.40</td>\n",
       "      <td>1575.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31831</th>\n",
       "      <td>nt159_nt616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FAKE_GENE_25</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.127443</td>\n",
       "      <td>-0.579808</td>\n",
       "      <td>0.163396</td>\n",
       "      <td>0.034084</td>\n",
       "      <td>1275.40</td>\n",
       "      <td>1423.20</td>\n",
       "      <td>1428.90</td>\n",
       "      <td>1278.30</td>\n",
       "      <td>2913.70</td>\n",
       "      <td>2569.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31832</th>\n",
       "      <td>nt796_nt822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FAKE_GENE_30</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.350594</td>\n",
       "      <td>-0.584801</td>\n",
       "      <td>-0.037319</td>\n",
       "      <td>-0.429662</td>\n",
       "      <td>1644.10</td>\n",
       "      <td>1834.70</td>\n",
       "      <td>1842.10</td>\n",
       "      <td>1640.60</td>\n",
       "      <td>3097.70</td>\n",
       "      <td>2103.90</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31833 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  pgRNA       paralog_pair  pgRNA_target    broad_target_type  \\\n",
       "0         TNPO2+1_nt687        TNPO2|TNPO1     TNPO2|ntc     single_targeting   \n",
       "1        nt625_SEC23A+2      SEC23B|SEC23A    ntc|SEC23A     single_targeting   \n",
       "2      nt294_IL1RAPL1+4  IL1RAPL2|IL1RAPL1  ntc|IL1RAPL1     single_targeting   \n",
       "3         nt133_TGFB2+3        TGFB3|TGFB2     ntc|TGFB2     single_targeting   \n",
       "4      nt625_PAFAH1B2+2  PAFAH1B3|PAFAH1B2  ntc|PAFAH1B2     single_targeting   \n",
       "...                 ...                ...           ...                  ...   \n",
       "31828       nt787_nt679                NaN  FAKE_GENE_25  double_nontargeting   \n",
       "31829       nt401_nt577                NaN  FAKE_GENE_14  double_nontargeting   \n",
       "31830       nt412_nt190                NaN  FAKE_GENE_25  double_nontargeting   \n",
       "31831       nt159_nt616                NaN  FAKE_GENE_25  double_nontargeting   \n",
       "31832       nt796_nt822                NaN  FAKE_GENE_30  double_nontargeting   \n",
       "\n",
       "       mean_CRISPR_score  RepA_CRISPR_score  RepB_CRISPR_score  \\\n",
       "0               1.404089           1.839640           1.220548   \n",
       "1               1.376141           1.719647           1.432025   \n",
       "2               0.401452           1.764124          -0.273009   \n",
       "3               0.399684           1.297354          -0.345809   \n",
       "4               1.474571           1.432591           1.727355   \n",
       "...                  ...                ...                ...   \n",
       "31828          -0.025413          -0.600200           0.054393   \n",
       "31829          -1.004934          -0.574149          -1.615088   \n",
       "31830          -0.114991          -0.576438           0.239666   \n",
       "31831          -0.127443          -0.579808           0.163396   \n",
       "31832          -0.350594          -0.584801          -0.037319   \n",
       "\n",
       "       RepC_CRISPR_score  RepA_plasmid_count  RepB_plasmid_count  \\\n",
       "0               1.152079              238.76              266.44   \n",
       "1               0.976752              448.63              500.64   \n",
       "2              -0.286759              238.76              266.44   \n",
       "3               0.247506             2651.80             2959.30   \n",
       "4               1.263767             1123.30             1253.50   \n",
       "...                  ...                 ...                 ...   \n",
       "31828           0.469568              364.51              406.77   \n",
       "31829          -0.825566             1119.00             1248.80   \n",
       "31830          -0.008203              814.83              909.30   \n",
       "31831           0.034084             1275.40             1423.20   \n",
       "31832          -0.429662             1644.10             1834.70   \n",
       "\n",
       "       RepC_plasmid_count  RepA_LTP_count  RepB_LTP_count  RepC_LTP_count  \n",
       "0                  267.51         1555.70         1106.90         1069.80  \n",
       "1                  502.65         2625.00         2544.70         1691.10  \n",
       "2                  267.51         1455.30          263.25          261.14  \n",
       "3                 2971.20        10675.00         2726.60         4892.20  \n",
       "4                 1258.50         5096.60         8451.50         5602.00  \n",
       "...                   ...             ...             ...             ...  \n",
       "31828              408.40          358.82          750.63         1125.70  \n",
       "31829             1253.80         1127.20          463.07          971.92  \n",
       "31830              912.96          819.14         2003.40         1575.30  \n",
       "31831             1428.90         1278.30         2913.70         2569.30  \n",
       "31832             1842.10         1640.60         3097.70         2103.90  \n",
       "\n",
       "[31833 rows x 14 columns]"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HELA_sgRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "aa1b5031",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts = PC9_sgRNA.copy()\n",
    "curr_counts = curr_counts.fillna(0)\n",
    "curr_counts.index = range(curr_counts.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "3001d7cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['pgRNA_ID', 'paralog_pair', 'pgRNA_target', 'broad_target_type',\n",
       "       'mean_CRISPR_score', 'RepA_CRISPR_score', 'RepB_CRISPR_score',\n",
       "       'RepC_CRISPR_score', 'RepA_plasmid_count', 'RepB_plasmid_count',\n",
       "       'RepC_plasmid_count', 'RepA_LTP_count', 'RepB_LTP_count',\n",
       "       'RepC_LTP_count'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "fda1edc5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pgRNA_ID</th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>pgRNA_target</th>\n",
       "      <th>broad_target_type</th>\n",
       "      <th>mean_CRISPR_score</th>\n",
       "      <th>RepA_CRISPR_score</th>\n",
       "      <th>RepB_CRISPR_score</th>\n",
       "      <th>RepC_CRISPR_score</th>\n",
       "      <th>RepA_plasmid_count</th>\n",
       "      <th>RepB_plasmid_count</th>\n",
       "      <th>RepC_plasmid_count</th>\n",
       "      <th>RepA_LTP_count</th>\n",
       "      <th>RepB_LTP_count</th>\n",
       "      <th>RepC_LTP_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1_nt404</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.368602</td>\n",
       "      <td>0.495471</td>\n",
       "      <td>0.196626</td>\n",
       "      <td>0.413707</td>\n",
       "      <td>2435.90</td>\n",
       "      <td>2367.70</td>\n",
       "      <td>2182.40</td>\n",
       "      <td>3364.10</td>\n",
       "      <td>2893.90</td>\n",
       "      <td>3027.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2_nt598</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.483738</td>\n",
       "      <td>0.303549</td>\n",
       "      <td>0.531184</td>\n",
       "      <td>0.616480</td>\n",
       "      <td>868.75</td>\n",
       "      <td>844.39</td>\n",
       "      <td>778.32</td>\n",
       "      <td>1102.40</td>\n",
       "      <td>1224.90</td>\n",
       "      <td>1207.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3_nt847</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.368685</td>\n",
       "      <td>-0.456572</td>\n",
       "      <td>0.736245</td>\n",
       "      <td>0.826380</td>\n",
       "      <td>911.10</td>\n",
       "      <td>885.56</td>\n",
       "      <td>816.27</td>\n",
       "      <td>826.22</td>\n",
       "      <td>1426.70</td>\n",
       "      <td>1421.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4_nt731</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.041019</td>\n",
       "      <td>-1.270615</td>\n",
       "      <td>0.010252</td>\n",
       "      <td>1.383419</td>\n",
       "      <td>416.65</td>\n",
       "      <td>404.97</td>\n",
       "      <td>373.28</td>\n",
       "      <td>263.42</td>\n",
       "      <td>450.09</td>\n",
       "      <td>883.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1_nt54</td>\n",
       "      <td>A2M|PZP</td>\n",
       "      <td>A2M|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>-0.196169</td>\n",
       "      <td>-0.170177</td>\n",
       "      <td>-0.495369</td>\n",
       "      <td>0.077040</td>\n",
       "      <td>1852.50</td>\n",
       "      <td>1800.50</td>\n",
       "      <td>1659.60</td>\n",
       "      <td>1906.60</td>\n",
       "      <td>1544.90</td>\n",
       "      <td>1913.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32735</th>\n",
       "      <td>ZYG11B+2_ZYG11A+1</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.325519</td>\n",
       "      <td>-0.609131</td>\n",
       "      <td>-0.399958</td>\n",
       "      <td>0.032532</td>\n",
       "      <td>1926.80</td>\n",
       "      <td>1872.80</td>\n",
       "      <td>1726.20</td>\n",
       "      <td>1495.80</td>\n",
       "      <td>1525.80</td>\n",
       "      <td>1759.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32736</th>\n",
       "      <td>ZYG11B+2_ZYG11A+2</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.606071</td>\n",
       "      <td>-0.907578</td>\n",
       "      <td>-0.333128</td>\n",
       "      <td>-0.577508</td>\n",
       "      <td>2270.00</td>\n",
       "      <td>2206.30</td>\n",
       "      <td>2033.70</td>\n",
       "      <td>1544.50</td>\n",
       "      <td>1860.00</td>\n",
       "      <td>1482.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32737</th>\n",
       "      <td>ZYG11B+2_ZYG11A+3</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.763798</td>\n",
       "      <td>-0.759516</td>\n",
       "      <td>-0.782239</td>\n",
       "      <td>-0.749639</td>\n",
       "      <td>1378.80</td>\n",
       "      <td>1340.10</td>\n",
       "      <td>1235.20</td>\n",
       "      <td>1001.40</td>\n",
       "      <td>897.78</td>\n",
       "      <td>818.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32738</th>\n",
       "      <td>ZYG11B+2_ZYG11A+4</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.418372</td>\n",
       "      <td>0.221395</td>\n",
       "      <td>-1.528546</td>\n",
       "      <td>0.052036</td>\n",
       "      <td>272.29</td>\n",
       "      <td>264.66</td>\n",
       "      <td>243.95</td>\n",
       "      <td>305.19</td>\n",
       "      <td>120.58</td>\n",
       "      <td>251.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32739</th>\n",
       "      <td>ZYG11B+3_ZYG11A+1</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>ZYG11B|ZYG11A</td>\n",
       "      <td>double_targeting</td>\n",
       "      <td>-0.149848</td>\n",
       "      <td>0.733271</td>\n",
       "      <td>-1.331134</td>\n",
       "      <td>0.148318</td>\n",
       "      <td>367.38</td>\n",
       "      <td>357.08</td>\n",
       "      <td>329.14</td>\n",
       "      <td>516.39</td>\n",
       "      <td>180.27</td>\n",
       "      <td>357.56</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32740 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                pgRNA_ID   paralog_pair   pgRNA_target broad_target_type  \\\n",
       "0            A2M+1_nt404        A2M|PZP        A2M|ntc  single_targeting   \n",
       "1            A2M+2_nt598        A2M|PZP        A2M|ntc  single_targeting   \n",
       "2            A2M+3_nt847        A2M|PZP        A2M|ntc  single_targeting   \n",
       "3            A2M+4_nt731        A2M|PZP        A2M|ntc  single_targeting   \n",
       "4             A2M+1_nt54        A2M|PZP        A2M|ntc  single_targeting   \n",
       "...                  ...            ...            ...               ...   \n",
       "32735  ZYG11B+2_ZYG11A+1  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32736  ZYG11B+2_ZYG11A+2  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32737  ZYG11B+2_ZYG11A+3  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32738  ZYG11B+2_ZYG11A+4  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "32739  ZYG11B+3_ZYG11A+1  ZYG11B|ZYG11A  ZYG11B|ZYG11A  double_targeting   \n",
       "\n",
       "       mean_CRISPR_score  RepA_CRISPR_score  RepB_CRISPR_score  \\\n",
       "0               0.368602           0.495471           0.196626   \n",
       "1               0.483738           0.303549           0.531184   \n",
       "2               0.368685          -0.456572           0.736245   \n",
       "3               0.041019          -1.270615           0.010252   \n",
       "4              -0.196169          -0.170177          -0.495369   \n",
       "...                  ...                ...                ...   \n",
       "32735          -0.325519          -0.609131          -0.399958   \n",
       "32736          -0.606071          -0.907578          -0.333128   \n",
       "32737          -0.763798          -0.759516          -0.782239   \n",
       "32738          -0.418372           0.221395          -1.528546   \n",
       "32739          -0.149848           0.733271          -1.331134   \n",
       "\n",
       "       RepC_CRISPR_score  RepA_plasmid_count  RepB_plasmid_count  \\\n",
       "0               0.413707             2435.90             2367.70   \n",
       "1               0.616480              868.75              844.39   \n",
       "2               0.826380              911.10              885.56   \n",
       "3               1.383419              416.65              404.97   \n",
       "4               0.077040             1852.50             1800.50   \n",
       "...                  ...                 ...                 ...   \n",
       "32735           0.032532             1926.80             1872.80   \n",
       "32736          -0.577508             2270.00             2206.30   \n",
       "32737          -0.749639             1378.80             1340.10   \n",
       "32738           0.052036              272.29              264.66   \n",
       "32739           0.148318              367.38              357.08   \n",
       "\n",
       "       RepC_plasmid_count  RepA_LTP_count  RepB_LTP_count  RepC_LTP_count  \n",
       "0                 2182.40         3364.10         2893.90         3027.60  \n",
       "1                  778.32         1102.40         1224.90         1207.40  \n",
       "2                  816.27          826.22         1426.70         1421.20  \n",
       "3                  373.28          263.42          450.09          883.55  \n",
       "4                 1659.60         1906.60         1544.90         1913.50  \n",
       "...                   ...             ...             ...             ...  \n",
       "32735             1726.20         1495.80         1525.80         1759.30  \n",
       "32736             2033.70         1544.50         1860.00         1482.10  \n",
       "32737             1235.20         1001.40          897.78          818.77  \n",
       "32738              243.95          305.19          120.58          251.33  \n",
       "32739              329.14          516.39          180.27          357.56  \n",
       "\n",
       "[32740 rows x 14 columns]"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "52b0747a",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_counts = curr_counts[['RepA_plasmid_count', 'RepB_plasmid_count', 'RepC_plasmid_count', 'RepA_LTP_count', 'RepB_LTP_count', 'RepC_LTP_count']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "dbb06781",
   "metadata": {},
   "outputs": [],
   "source": [
    "type_list = [\"Single\"] * curr_counts.shape[0]\n",
    "for i in range(len(type_list)):\n",
    "    if \"single\" not in curr_counts.loc[i, 'broad_target_type']:\n",
    "        type_list[i] = \"Dual\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "77d1f489",
   "metadata": {},
   "outputs": [],
   "source": [
    "targeted_genes = curr_counts['pgRNA_target'].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "fc67dd18",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts = curr_counts.drop(['pgRNA_target', 'paralog_pair', 'mean_CRISPR_score', \n",
    "                                'RepA_CRISPR_score', 'RepB_CRISPR_score', 'RepC_CRISPR_score',\n",
    "                               'RepA_plasmid_count', 'RepB_plasmid_count', 'RepC_plasmid_count', \n",
    "                                'RepA_LTP_count', 'RepB_LTP_count', 'RepC_LTP_count', 'broad_target_type'], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "75b3b2d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts.columns = [\"Guide 1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "af27789d",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_1_list = np.array([i.split('_')[0].split('+')[0] for i in curr_counts['Guide 1']], dtype = object)\n",
    "gene_2_list = np.array([i.split('_')[1].split('+')[0] for i in curr_counts['Guide 1']], dtype = object)\n",
    "guide_2_list = [i.split('_')[1] for i in curr_counts['Guide 1'].tolist()]\n",
    "curr_counts['Guide 1'] = [i.split('_')[0] for i in curr_counts['Guide 1'].tolist()]\n",
    "\n",
    "gene_1_list = gene_1_list.tolist()\n",
    "gene_2_list = gene_2_list.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "aa0e2efa",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts['Gene 1'] = gene_1_list\n",
    "curr_counts['Guide 2'] = guide_2_list\n",
    "curr_counts['Gene 2'] = gene_2_list\n",
    "curr_counts['Count Replicates'] = all_counts\n",
    "curr_counts['Type'] = type_list\n",
    "curr_counts['Cell Line'] = [\"PC9\"] * curr_counts.shape[0]\n",
    "curr_counts['Condition'] = [\"plasmid;plasmid;plasmid;LTP;LTP;LTP\"] * curr_counts.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "1f0189da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt404</td>\n",
       "      <td>nt404</td>\n",
       "      <td>2435.9;2367.7;2182.4;3364.1;2893.9;3027.6</td>\n",
       "      <td>Single</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt598</td>\n",
       "      <td>nt598</td>\n",
       "      <td>868.75;844.39;778.32;1102.4;1224.9;1207.4</td>\n",
       "      <td>Single</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt847</td>\n",
       "      <td>nt847</td>\n",
       "      <td>911.1;885.56;816.27;826.22;1426.7;1421.2</td>\n",
       "      <td>Single</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt731</td>\n",
       "      <td>nt731</td>\n",
       "      <td>416.65;404.97;373.28;263.42;450.09;883.55</td>\n",
       "      <td>Single</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt54</td>\n",
       "      <td>nt54</td>\n",
       "      <td>1852.5;1800.5;1659.6;1906.6;1544.9;1913.5</td>\n",
       "      <td>Single</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32735</th>\n",
       "      <td>ZYG11B+2</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A+1</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>1926.8;1872.8;1726.2;1495.8;1525.8;1759.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32736</th>\n",
       "      <td>ZYG11B+2</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A+2</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>2270.0;2206.3;2033.7;1544.5;1860.0;1482.1</td>\n",
       "      <td>Dual</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32737</th>\n",
       "      <td>ZYG11B+2</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A+3</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>1378.8;1340.1;1235.2;1001.4;897.78;818.77</td>\n",
       "      <td>Dual</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32738</th>\n",
       "      <td>ZYG11B+2</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A+4</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>272.29;264.66;243.95;305.19;120.58;251.33</td>\n",
       "      <td>Dual</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32739</th>\n",
       "      <td>ZYG11B+3</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ZYG11A+1</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>367.38;357.08;329.14;516.39;180.27;357.56</td>\n",
       "      <td>Dual</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32740 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Guide 1  Gene 1   Guide 2  Gene 2  \\\n",
       "0         A2M+1     A2M     nt404   nt404   \n",
       "1         A2M+2     A2M     nt598   nt598   \n",
       "2         A2M+3     A2M     nt847   nt847   \n",
       "3         A2M+4     A2M     nt731   nt731   \n",
       "4         A2M+1     A2M      nt54    nt54   \n",
       "...         ...     ...       ...     ...   \n",
       "32735  ZYG11B+2  ZYG11B  ZYG11A+1  ZYG11A   \n",
       "32736  ZYG11B+2  ZYG11B  ZYG11A+2  ZYG11A   \n",
       "32737  ZYG11B+2  ZYG11B  ZYG11A+3  ZYG11A   \n",
       "32738  ZYG11B+2  ZYG11B  ZYG11A+4  ZYG11A   \n",
       "32739  ZYG11B+3  ZYG11B  ZYG11A+1  ZYG11A   \n",
       "\n",
       "                                Count Replicates    Type Cell Line  \\\n",
       "0      2435.9;2367.7;2182.4;3364.1;2893.9;3027.6  Single       PC9   \n",
       "1      868.75;844.39;778.32;1102.4;1224.9;1207.4  Single       PC9   \n",
       "2       911.1;885.56;816.27;826.22;1426.7;1421.2  Single       PC9   \n",
       "3      416.65;404.97;373.28;263.42;450.09;883.55  Single       PC9   \n",
       "4      1852.5;1800.5;1659.6;1906.6;1544.9;1913.5  Single       PC9   \n",
       "...                                          ...     ...       ...   \n",
       "32735  1926.8;1872.8;1726.2;1495.8;1525.8;1759.3    Dual       PC9   \n",
       "32736  2270.0;2206.3;2033.7;1544.5;1860.0;1482.1    Dual       PC9   \n",
       "32737  1378.8;1340.1;1235.2;1001.4;897.78;818.77    Dual       PC9   \n",
       "32738  272.29;264.66;243.95;305.19;120.58;251.33    Dual       PC9   \n",
       "32739  367.38;357.08;329.14;516.39;180.27;357.56    Dual       PC9   \n",
       "\n",
       "                                 Condition  \n",
       "0      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "1      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "2      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "3      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "4      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "...                                    ...  \n",
       "32735  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "32736  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "32737  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "32738  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "32739  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "\n",
       "[32740 rows x 8 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "c4b1ae7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, curr_counts])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "c36c16dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts = HELA_sgRNA.copy()\n",
    "#curr_counts = curr_counts.dropna()\n",
    "curr_counts = curr_counts.fillna(0)\n",
    "curr_counts.index = range(curr_counts.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "7ae6f8bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['pgRNA', 'paralog_pair', 'pgRNA_target', 'broad_target_type',\n",
       "       'mean_CRISPR_score', 'RepA_CRISPR_score', 'RepB_CRISPR_score',\n",
       "       'RepC_CRISPR_score', 'RepA_plasmid_count', 'RepB_plasmid_count',\n",
       "       'RepC_plasmid_count', 'RepA_LTP_count', 'RepB_LTP_count',\n",
       "       'RepC_LTP_count'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "3163f89f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pgRNA</th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>pgRNA_target</th>\n",
       "      <th>broad_target_type</th>\n",
       "      <th>mean_CRISPR_score</th>\n",
       "      <th>RepA_CRISPR_score</th>\n",
       "      <th>RepB_CRISPR_score</th>\n",
       "      <th>RepC_CRISPR_score</th>\n",
       "      <th>RepA_plasmid_count</th>\n",
       "      <th>RepB_plasmid_count</th>\n",
       "      <th>RepC_plasmid_count</th>\n",
       "      <th>RepA_LTP_count</th>\n",
       "      <th>RepB_LTP_count</th>\n",
       "      <th>RepC_LTP_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNPO2+1_nt687</td>\n",
       "      <td>TNPO2|TNPO1</td>\n",
       "      <td>TNPO2|ntc</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>1.404089</td>\n",
       "      <td>1.839640</td>\n",
       "      <td>1.220548</td>\n",
       "      <td>1.152079</td>\n",
       "      <td>238.76</td>\n",
       "      <td>266.44</td>\n",
       "      <td>267.51</td>\n",
       "      <td>1555.70</td>\n",
       "      <td>1106.90</td>\n",
       "      <td>1069.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>nt625_SEC23A+2</td>\n",
       "      <td>SEC23B|SEC23A</td>\n",
       "      <td>ntc|SEC23A</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>1.376141</td>\n",
       "      <td>1.719647</td>\n",
       "      <td>1.432025</td>\n",
       "      <td>0.976752</td>\n",
       "      <td>448.63</td>\n",
       "      <td>500.64</td>\n",
       "      <td>502.65</td>\n",
       "      <td>2625.00</td>\n",
       "      <td>2544.70</td>\n",
       "      <td>1691.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nt294_IL1RAPL1+4</td>\n",
       "      <td>IL1RAPL2|IL1RAPL1</td>\n",
       "      <td>ntc|IL1RAPL1</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.401452</td>\n",
       "      <td>1.764124</td>\n",
       "      <td>-0.273009</td>\n",
       "      <td>-0.286759</td>\n",
       "      <td>238.76</td>\n",
       "      <td>266.44</td>\n",
       "      <td>267.51</td>\n",
       "      <td>1455.30</td>\n",
       "      <td>263.25</td>\n",
       "      <td>261.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nt133_TGFB2+3</td>\n",
       "      <td>TGFB3|TGFB2</td>\n",
       "      <td>ntc|TGFB2</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>0.399684</td>\n",
       "      <td>1.297354</td>\n",
       "      <td>-0.345809</td>\n",
       "      <td>0.247506</td>\n",
       "      <td>2651.80</td>\n",
       "      <td>2959.30</td>\n",
       "      <td>2971.20</td>\n",
       "      <td>10675.00</td>\n",
       "      <td>2726.60</td>\n",
       "      <td>4892.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>nt625_PAFAH1B2+2</td>\n",
       "      <td>PAFAH1B3|PAFAH1B2</td>\n",
       "      <td>ntc|PAFAH1B2</td>\n",
       "      <td>single_targeting</td>\n",
       "      <td>1.474571</td>\n",
       "      <td>1.432591</td>\n",
       "      <td>1.727355</td>\n",
       "      <td>1.263767</td>\n",
       "      <td>1123.30</td>\n",
       "      <td>1253.50</td>\n",
       "      <td>1258.50</td>\n",
       "      <td>5096.60</td>\n",
       "      <td>8451.50</td>\n",
       "      <td>5602.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31828</th>\n",
       "      <td>nt787_nt679</td>\n",
       "      <td>0</td>\n",
       "      <td>FAKE_GENE_25</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.025413</td>\n",
       "      <td>-0.600200</td>\n",
       "      <td>0.054393</td>\n",
       "      <td>0.469568</td>\n",
       "      <td>364.51</td>\n",
       "      <td>406.77</td>\n",
       "      <td>408.40</td>\n",
       "      <td>358.82</td>\n",
       "      <td>750.63</td>\n",
       "      <td>1125.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31829</th>\n",
       "      <td>nt401_nt577</td>\n",
       "      <td>0</td>\n",
       "      <td>FAKE_GENE_14</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-1.004934</td>\n",
       "      <td>-0.574149</td>\n",
       "      <td>-1.615088</td>\n",
       "      <td>-0.825566</td>\n",
       "      <td>1119.00</td>\n",
       "      <td>1248.80</td>\n",
       "      <td>1253.80</td>\n",
       "      <td>1127.20</td>\n",
       "      <td>463.07</td>\n",
       "      <td>971.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31830</th>\n",
       "      <td>nt412_nt190</td>\n",
       "      <td>0</td>\n",
       "      <td>FAKE_GENE_25</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.114991</td>\n",
       "      <td>-0.576438</td>\n",
       "      <td>0.239666</td>\n",
       "      <td>-0.008203</td>\n",
       "      <td>814.83</td>\n",
       "      <td>909.30</td>\n",
       "      <td>912.96</td>\n",
       "      <td>819.14</td>\n",
       "      <td>2003.40</td>\n",
       "      <td>1575.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31831</th>\n",
       "      <td>nt159_nt616</td>\n",
       "      <td>0</td>\n",
       "      <td>FAKE_GENE_25</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.127443</td>\n",
       "      <td>-0.579808</td>\n",
       "      <td>0.163396</td>\n",
       "      <td>0.034084</td>\n",
       "      <td>1275.40</td>\n",
       "      <td>1423.20</td>\n",
       "      <td>1428.90</td>\n",
       "      <td>1278.30</td>\n",
       "      <td>2913.70</td>\n",
       "      <td>2569.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31832</th>\n",
       "      <td>nt796_nt822</td>\n",
       "      <td>0</td>\n",
       "      <td>FAKE_GENE_30</td>\n",
       "      <td>double_nontargeting</td>\n",
       "      <td>-0.350594</td>\n",
       "      <td>-0.584801</td>\n",
       "      <td>-0.037319</td>\n",
       "      <td>-0.429662</td>\n",
       "      <td>1644.10</td>\n",
       "      <td>1834.70</td>\n",
       "      <td>1842.10</td>\n",
       "      <td>1640.60</td>\n",
       "      <td>3097.70</td>\n",
       "      <td>2103.90</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31833 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  pgRNA       paralog_pair  pgRNA_target    broad_target_type  \\\n",
       "0         TNPO2+1_nt687        TNPO2|TNPO1     TNPO2|ntc     single_targeting   \n",
       "1        nt625_SEC23A+2      SEC23B|SEC23A    ntc|SEC23A     single_targeting   \n",
       "2      nt294_IL1RAPL1+4  IL1RAPL2|IL1RAPL1  ntc|IL1RAPL1     single_targeting   \n",
       "3         nt133_TGFB2+3        TGFB3|TGFB2     ntc|TGFB2     single_targeting   \n",
       "4      nt625_PAFAH1B2+2  PAFAH1B3|PAFAH1B2  ntc|PAFAH1B2     single_targeting   \n",
       "...                 ...                ...           ...                  ...   \n",
       "31828       nt787_nt679                  0  FAKE_GENE_25  double_nontargeting   \n",
       "31829       nt401_nt577                  0  FAKE_GENE_14  double_nontargeting   \n",
       "31830       nt412_nt190                  0  FAKE_GENE_25  double_nontargeting   \n",
       "31831       nt159_nt616                  0  FAKE_GENE_25  double_nontargeting   \n",
       "31832       nt796_nt822                  0  FAKE_GENE_30  double_nontargeting   \n",
       "\n",
       "       mean_CRISPR_score  RepA_CRISPR_score  RepB_CRISPR_score  \\\n",
       "0               1.404089           1.839640           1.220548   \n",
       "1               1.376141           1.719647           1.432025   \n",
       "2               0.401452           1.764124          -0.273009   \n",
       "3               0.399684           1.297354          -0.345809   \n",
       "4               1.474571           1.432591           1.727355   \n",
       "...                  ...                ...                ...   \n",
       "31828          -0.025413          -0.600200           0.054393   \n",
       "31829          -1.004934          -0.574149          -1.615088   \n",
       "31830          -0.114991          -0.576438           0.239666   \n",
       "31831          -0.127443          -0.579808           0.163396   \n",
       "31832          -0.350594          -0.584801          -0.037319   \n",
       "\n",
       "       RepC_CRISPR_score  RepA_plasmid_count  RepB_plasmid_count  \\\n",
       "0               1.152079              238.76              266.44   \n",
       "1               0.976752              448.63              500.64   \n",
       "2              -0.286759              238.76              266.44   \n",
       "3               0.247506             2651.80             2959.30   \n",
       "4               1.263767             1123.30             1253.50   \n",
       "...                  ...                 ...                 ...   \n",
       "31828           0.469568              364.51              406.77   \n",
       "31829          -0.825566             1119.00             1248.80   \n",
       "31830          -0.008203              814.83              909.30   \n",
       "31831           0.034084             1275.40             1423.20   \n",
       "31832          -0.429662             1644.10             1834.70   \n",
       "\n",
       "       RepC_plasmid_count  RepA_LTP_count  RepB_LTP_count  RepC_LTP_count  \n",
       "0                  267.51         1555.70         1106.90         1069.80  \n",
       "1                  502.65         2625.00         2544.70         1691.10  \n",
       "2                  267.51         1455.30          263.25          261.14  \n",
       "3                 2971.20        10675.00         2726.60         4892.20  \n",
       "4                 1258.50         5096.60         8451.50         5602.00  \n",
       "...                   ...             ...             ...             ...  \n",
       "31828              408.40          358.82          750.63         1125.70  \n",
       "31829             1253.80         1127.20          463.07          971.92  \n",
       "31830              912.96          819.14         2003.40         1575.30  \n",
       "31831             1428.90         1278.30         2913.70         2569.30  \n",
       "31832             1842.10         1640.60         3097.70         2103.90  \n",
       "\n",
       "[31833 rows x 14 columns]"
      ]
     },
     "execution_count": 244,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "b7da074f",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_counts = curr_counts[['RepA_plasmid_count', 'RepB_plasmid_count', 'RepC_plasmid_count', 'RepA_LTP_count', 'RepB_LTP_count', 'RepC_LTP_count']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "6dd8c727",
   "metadata": {},
   "outputs": [],
   "source": [
    "type_list = [\"Single\"] * curr_counts.shape[0]\n",
    "for i in range(len(type_list)):\n",
    "    if \"single\" not in curr_counts.loc[i, 'broad_target_type']:\n",
    "        type_list[i] = \"Dual\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "cdd7e52c",
   "metadata": {},
   "outputs": [],
   "source": [
    "targeted_genes = curr_counts['pgRNA_target'].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "a542a0cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts = curr_counts.drop(['pgRNA_target', 'paralog_pair', 'mean_CRISPR_score', \n",
    "                                'RepA_CRISPR_score', 'RepB_CRISPR_score', 'RepC_CRISPR_score',\n",
    "                               'RepA_plasmid_count', 'RepB_plasmid_count', 'RepC_plasmid_count', \n",
    "                                'RepA_LTP_count', 'RepB_LTP_count', 'RepC_LTP_count', 'broad_target_type'], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "40d8290d",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts.columns = [\"Guide 1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "e103d856",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0           TNPO2+1_nt687\n",
       "1          nt625_SEC23A+2\n",
       "2        nt294_IL1RAPL1+4\n",
       "3           nt133_TGFB2+3\n",
       "4        nt625_PAFAH1B2+2\n",
       "               ...       \n",
       "31828         nt787_nt679\n",
       "31829         nt401_nt577\n",
       "31830         nt412_nt190\n",
       "31831         nt159_nt616\n",
       "31832         nt796_nt822\n",
       "Name: Guide 1, Length: 31833, dtype: object"
      ]
     },
     "execution_count": 250,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts['Guide 1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "id": "6ab9d6c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_1_list = np.array([i.split('_')[0].split('+')[0] for i in curr_counts['Guide 1']], dtype = object)\n",
    "gene_2_list = np.array([i.split('_')[1].split('+')[0] for i in curr_counts['Guide 1']], dtype = object)\n",
    "guide_2_list = [i.split('_')[1] for i in curr_counts['Guide 1'].tolist()]\n",
    "curr_counts['Guide 1'] = [i.split('_')[0] for i in curr_counts['Guide 1'].tolist()]\n",
    "\n",
    "gene_1_list = gene_1_list.tolist()\n",
    "gene_2_list = gene_2_list.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "88fc8827",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts['Gene 1'] = gene_1_list\n",
    "curr_counts['Guide 2'] = guide_2_list\n",
    "curr_counts['Gene 2'] = gene_2_list\n",
    "curr_counts['Count Replicates'] = all_counts\n",
    "curr_counts['Type'] = type_list\n",
    "curr_counts['Cell Line'] = [\"HELA\"] * curr_counts.shape[0]\n",
    "curr_counts['Condition'] = [\"plasmid;plasmid;plasmid;LTP;LTP;LTP\"] * curr_counts.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "id": "b46bff42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNPO2+1</td>\n",
       "      <td>TNPO2</td>\n",
       "      <td>nt687</td>\n",
       "      <td>nt687</td>\n",
       "      <td>238.76;266.44;267.51;1555.7;1106.9;1069.8</td>\n",
       "      <td>Single</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>nt625</td>\n",
       "      <td>nt625</td>\n",
       "      <td>SEC23A+2</td>\n",
       "      <td>SEC23A</td>\n",
       "      <td>448.63;500.64;502.65;2625.0;2544.7;1691.1</td>\n",
       "      <td>Single</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nt294</td>\n",
       "      <td>nt294</td>\n",
       "      <td>IL1RAPL1+4</td>\n",
       "      <td>IL1RAPL1</td>\n",
       "      <td>238.76;266.44;267.51;1455.3;263.25;261.14</td>\n",
       "      <td>Single</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nt133</td>\n",
       "      <td>nt133</td>\n",
       "      <td>TGFB2+3</td>\n",
       "      <td>TGFB2</td>\n",
       "      <td>2651.8;2959.3;2971.2;10675.0;2726.6;4892.2</td>\n",
       "      <td>Single</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>nt625</td>\n",
       "      <td>nt625</td>\n",
       "      <td>PAFAH1B2+2</td>\n",
       "      <td>PAFAH1B2</td>\n",
       "      <td>1123.3;1253.5;1258.5;5096.6;8451.5;5602.0</td>\n",
       "      <td>Single</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31828</th>\n",
       "      <td>nt787</td>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>nt679</td>\n",
       "      <td>364.51;406.77;408.4;358.82;750.63;1125.7</td>\n",
       "      <td>Dual</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31829</th>\n",
       "      <td>nt401</td>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>nt577</td>\n",
       "      <td>1119.0;1248.8;1253.8;1127.2;463.07;971.92</td>\n",
       "      <td>Dual</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31830</th>\n",
       "      <td>nt412</td>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>nt190</td>\n",
       "      <td>814.83;909.3;912.96;819.14;2003.4;1575.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31831</th>\n",
       "      <td>nt159</td>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>nt616</td>\n",
       "      <td>1275.4;1423.2;1428.9;1278.3;2913.7;2569.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31832</th>\n",
       "      <td>nt796</td>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>nt822</td>\n",
       "      <td>1644.1;1834.7;1842.1;1640.6;3097.7;2103.9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31833 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Guide 1 Gene 1     Guide 2    Gene 2  \\\n",
       "0      TNPO2+1  TNPO2       nt687     nt687   \n",
       "1        nt625  nt625    SEC23A+2    SEC23A   \n",
       "2        nt294  nt294  IL1RAPL1+4  IL1RAPL1   \n",
       "3        nt133  nt133     TGFB2+3     TGFB2   \n",
       "4        nt625  nt625  PAFAH1B2+2  PAFAH1B2   \n",
       "...        ...    ...         ...       ...   \n",
       "31828    nt787  nt787       nt679     nt679   \n",
       "31829    nt401  nt401       nt577     nt577   \n",
       "31830    nt412  nt412       nt190     nt190   \n",
       "31831    nt159  nt159       nt616     nt616   \n",
       "31832    nt796  nt796       nt822     nt822   \n",
       "\n",
       "                                 Count Replicates    Type Cell Line  \\\n",
       "0       238.76;266.44;267.51;1555.7;1106.9;1069.8  Single      HELA   \n",
       "1       448.63;500.64;502.65;2625.0;2544.7;1691.1  Single      HELA   \n",
       "2       238.76;266.44;267.51;1455.3;263.25;261.14  Single      HELA   \n",
       "3      2651.8;2959.3;2971.2;10675.0;2726.6;4892.2  Single      HELA   \n",
       "4       1123.3;1253.5;1258.5;5096.6;8451.5;5602.0  Single      HELA   \n",
       "...                                           ...     ...       ...   \n",
       "31828    364.51;406.77;408.4;358.82;750.63;1125.7    Dual      HELA   \n",
       "31829   1119.0;1248.8;1253.8;1127.2;463.07;971.92    Dual      HELA   \n",
       "31830    814.83;909.3;912.96;819.14;2003.4;1575.3    Dual      HELA   \n",
       "31831   1275.4;1423.2;1428.9;1278.3;2913.7;2569.3    Dual      HELA   \n",
       "31832   1644.1;1834.7;1842.1;1640.6;3097.7;2103.9    Dual      HELA   \n",
       "\n",
       "                                 Condition  \n",
       "0      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "1      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "2      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "3      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "4      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "...                                    ...  \n",
       "31828  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31829  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31830  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31831  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31832  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "\n",
       "[31833 rows x 8 columns]"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "829a19aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, curr_counts])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "7b503fec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>nt404</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt404</td>\n",
       "      <td>2435.9;2367.7;2182.4;3364.1;2893.9;3027.6</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>nt598</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt598</td>\n",
       "      <td>868.75;844.39;778.32;1102.4;1224.9;1207.4</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>nt847</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt847</td>\n",
       "      <td>911.1;885.56;816.27;826.22;1426.7;1421.2</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>nt731</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt731</td>\n",
       "      <td>416.65;404.97;373.28;263.42;450.09;883.55</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>nt54</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt54</td>\n",
       "      <td>1852.5;1800.5;1659.6;1906.6;1544.9;1913.5</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31828</th>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>364.51;406.77;408.4;358.82;750.63;1125.7</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31829</th>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>1119.0;1248.8;1253.8;1127.2;463.07;971.92</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31830</th>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>814.83;909.3;912.96;819.14;2003.4;1575.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31831</th>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>1275.4;1423.2;1428.9;1278.3;2913.7;2569.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31832</th>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>1644.1;1834.7;1842.1;1640.6;3097.7;2103.9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid;plasmid;plasmid;LTP;LTP;LTP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64573 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Guide 1 Guide 2 Gene 1 Gene 2  \\\n",
       "0       A2M+1   nt404    A2M  nt404   \n",
       "1       A2M+2   nt598    A2M  nt598   \n",
       "2       A2M+3   nt847    A2M  nt847   \n",
       "3       A2M+4   nt731    A2M  nt731   \n",
       "4       A2M+1    nt54    A2M   nt54   \n",
       "...       ...     ...    ...    ...   \n",
       "31828   nt787   nt679  nt787  nt679   \n",
       "31829   nt401   nt577  nt401  nt577   \n",
       "31830   nt412   nt190  nt412  nt190   \n",
       "31831   nt159   nt616  nt159  nt616   \n",
       "31832   nt796   nt822  nt796  nt822   \n",
       "\n",
       "                                Count Replicates    Type Sequencing Cell Line  \\\n",
       "0      2435.9;2367.7;2182.4;3364.1;2893.9;3027.6  Single        NaN       PC9   \n",
       "1      868.75;844.39;778.32;1102.4;1224.9;1207.4  Single        NaN       PC9   \n",
       "2       911.1;885.56;816.27;826.22;1426.7;1421.2  Single        NaN       PC9   \n",
       "3      416.65;404.97;373.28;263.42;450.09;883.55  Single        NaN       PC9   \n",
       "4      1852.5;1800.5;1659.6;1906.6;1544.9;1913.5  Single        NaN       PC9   \n",
       "...                                          ...     ...        ...       ...   \n",
       "31828   364.51;406.77;408.4;358.82;750.63;1125.7    Dual        NaN      HELA   \n",
       "31829  1119.0;1248.8;1253.8;1127.2;463.07;971.92    Dual        NaN      HELA   \n",
       "31830   814.83;909.3;912.96;819.14;2003.4;1575.3    Dual        NaN      HELA   \n",
       "31831  1275.4;1423.2;1428.9;1278.3;2913.7;2569.3    Dual        NaN      HELA   \n",
       "31832  1644.1;1834.7;1842.1;1640.6;3097.7;2103.9    Dual        NaN      HELA   \n",
       "\n",
       "                                 Condition  \n",
       "0      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "1      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "2      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "3      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "4      plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "...                                    ...  \n",
       "31828  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31829  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31830  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31831  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "31832  plasmid;plasmid;plasmid;LTP;LTP;LTP  \n",
       "\n",
       "[64573 rows x 9 columns]"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "9c22cc15",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Condition'] = 'plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3'\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['parrish_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "1e6b8091",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>nt404</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt404</td>\n",
       "      <td>2435.9;2367.7;2182.4;3364.1;2893.9;3027.6</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>nt598</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt598</td>\n",
       "      <td>868.75;844.39;778.32;1102.4;1224.9;1207.4</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>nt847</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt847</td>\n",
       "      <td>911.1;885.56;816.27;826.22;1426.7;1421.2</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>nt731</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt731</td>\n",
       "      <td>416.65;404.97;373.28;263.42;450.09;883.55</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>nt54</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt54</td>\n",
       "      <td>1852.5;1800.5;1659.6;1906.6;1544.9;1913.5</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31828</th>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>364.51;406.77;408.4;358.82;750.63;1125.7</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31829</th>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>1119.0;1248.8;1253.8;1127.2;463.07;971.92</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31830</th>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>814.83;909.3;912.96;819.14;2003.4;1575.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31831</th>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>1275.4;1423.2;1428.9;1278.3;2913.7;2569.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31832</th>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>1644.1;1834.7;1842.1;1640.6;3097.7;2103.9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64573 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Guide 1 Guide 2 Gene 1 Gene 2  \\\n",
       "0       A2M+1   nt404    A2M  nt404   \n",
       "1       A2M+2   nt598    A2M  nt598   \n",
       "2       A2M+3   nt847    A2M  nt847   \n",
       "3       A2M+4   nt731    A2M  nt731   \n",
       "4       A2M+1    nt54    A2M   nt54   \n",
       "...       ...     ...    ...    ...   \n",
       "31828   nt787   nt679  nt787  nt679   \n",
       "31829   nt401   nt577  nt401  nt577   \n",
       "31830   nt412   nt190  nt412  nt190   \n",
       "31831   nt159   nt616  nt159  nt616   \n",
       "31832   nt796   nt822  nt796  nt822   \n",
       "\n",
       "                                Count Replicates    Type Sequencing Cell Line  \\\n",
       "0      2435.9;2367.7;2182.4;3364.1;2893.9;3027.6  Single        NaN       PC9   \n",
       "1      868.75;844.39;778.32;1102.4;1224.9;1207.4  Single        NaN       PC9   \n",
       "2       911.1;885.56;816.27;826.22;1426.7;1421.2  Single        NaN       PC9   \n",
       "3      416.65;404.97;373.28;263.42;450.09;883.55  Single        NaN       PC9   \n",
       "4      1852.5;1800.5;1659.6;1906.6;1544.9;1913.5  Single        NaN       PC9   \n",
       "...                                          ...     ...        ...       ...   \n",
       "31828   364.51;406.77;408.4;358.82;750.63;1125.7    Dual        NaN      HELA   \n",
       "31829  1119.0;1248.8;1253.8;1127.2;463.07;971.92    Dual        NaN      HELA   \n",
       "31830   814.83;909.3;912.96;819.14;2003.4;1575.3    Dual        NaN      HELA   \n",
       "31831  1275.4;1423.2;1428.9;1278.3;2913.7;2569.3    Dual        NaN      HELA   \n",
       "31832  1644.1;1834.7;1842.1;1640.6;3097.7;2103.9    Dual        NaN      HELA   \n",
       "\n",
       "                                             Condition     Study  \n",
       "0      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "1      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "2      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "3      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "4      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "...                                                ...       ...  \n",
       "31828  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31829  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31830  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31831  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31832  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "\n",
       "[64573 rows x 10 columns]"
      ]
     },
     "execution_count": 257,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11fa9491",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "91702e3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "parrish_loc = os.path.join(learning_goals_loc_general, \"Parrish\")\n",
    "\n",
    "PC9_HeLa_GIs = pd.read_excel(os.path.join(parrish_loc, \"4 - PC9 and HeLa screen paralog genetic interaction (GI) scores.xlsx\"), skiprows = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "ac9a25d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paralog_pair</th>\n",
       "      <th>target1</th>\n",
       "      <th>target2</th>\n",
       "      <th>GI_flag</th>\n",
       "      <th>PC9_GI_flag</th>\n",
       "      <th>PC9_GI_score_rank</th>\n",
       "      <th>PC9_GI_score</th>\n",
       "      <th>PC9_GI_fdr</th>\n",
       "      <th>HeLa_GI_flag</th>\n",
       "      <th>HeLa_GI_score_rank</th>\n",
       "      <th>HeLa_GI_score</th>\n",
       "      <th>HeLa_GI_fdr</th>\n",
       "      <th>PC9_DKO_expected_CS</th>\n",
       "      <th>PC9_DKO_observed_CS</th>\n",
       "      <th>HeLa_DKO_expected_CS</th>\n",
       "      <th>HeLa_DKO_observed_CS</th>\n",
       "      <th>same_chr</th>\n",
       "      <th>same_chr_dist</th>\n",
       "      <th>proximity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCNL2|CCNL1</td>\n",
       "      <td>CCNL2</td>\n",
       "      <td>CCNL1</td>\n",
       "      <td>synthetic_lethal</td>\n",
       "      <td>SL_in_PC9</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.313788</td>\n",
       "      <td>0.000165</td>\n",
       "      <td>SL_in_HeLa</td>\n",
       "      <td>10</td>\n",
       "      <td>-1.261084</td>\n",
       "      <td>0.026168</td>\n",
       "      <td>0.375079</td>\n",
       "      <td>-1.987383</td>\n",
       "      <td>-0.099315</td>\n",
       "      <td>-1.332104</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDK6|CDK4</td>\n",
       "      <td>CDK6</td>\n",
       "      <td>CDK4</td>\n",
       "      <td>synthetic_lethal</td>\n",
       "      <td>SL_in_PC9</td>\n",
       "      <td>2</td>\n",
       "      <td>-1.537568</td>\n",
       "      <td>0.000340</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>210</td>\n",
       "      <td>-0.251834</td>\n",
       "      <td>0.307389</td>\n",
       "      <td>-2.309117</td>\n",
       "      <td>-3.408344</td>\n",
       "      <td>-0.451694</td>\n",
       "      <td>-0.632285</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSK3B|GSK3A</td>\n",
       "      <td>GSK3B</td>\n",
       "      <td>GSK3A</td>\n",
       "      <td>synthetic_lethal</td>\n",
       "      <td>SL_in_PC9</td>\n",
       "      <td>3</td>\n",
       "      <td>-1.429767</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>SL_in_HeLa</td>\n",
       "      <td>4</td>\n",
       "      <td>-1.659580</td>\n",
       "      <td>0.012451</td>\n",
       "      <td>-0.343343</td>\n",
       "      <td>-1.691436</td>\n",
       "      <td>0.649198</td>\n",
       "      <td>-1.073316</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G3BP2|G3BP1</td>\n",
       "      <td>G3BP2</td>\n",
       "      <td>G3BP1</td>\n",
       "      <td>synthetic_lethal</td>\n",
       "      <td>SL_in_PC9</td>\n",
       "      <td>4</td>\n",
       "      <td>-1.429397</td>\n",
       "      <td>0.004693</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>663</td>\n",
       "      <td>0.021784</td>\n",
       "      <td>0.951159</td>\n",
       "      <td>-0.260061</td>\n",
       "      <td>-1.622894</td>\n",
       "      <td>-0.858532</td>\n",
       "      <td>-0.715920</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CNOT8|CNOT7</td>\n",
       "      <td>CNOT8</td>\n",
       "      <td>CNOT7</td>\n",
       "      <td>synthetic_lethal</td>\n",
       "      <td>SL_in_PC9</td>\n",
       "      <td>5</td>\n",
       "      <td>-1.208182</td>\n",
       "      <td>0.004693</td>\n",
       "      <td>SL_in_HeLa</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.063079</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.775066</td>\n",
       "      <td>-0.554362</td>\n",
       "      <td>0.166142</td>\n",
       "      <td>-1.900996</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1025</th>\n",
       "      <td>ACTL6A|ACTL6B</td>\n",
       "      <td>ACTL6A</td>\n",
       "      <td>ACTL6B</td>\n",
       "      <td>buffering</td>\n",
       "      <td>buffering_in_PC9</td>\n",
       "      <td>1026</td>\n",
       "      <td>0.728780</td>\n",
       "      <td>0.029700</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>121</td>\n",
       "      <td>-0.388877</td>\n",
       "      <td>0.033472</td>\n",
       "      <td>-1.909384</td>\n",
       "      <td>-0.814789</td>\n",
       "      <td>-0.574970</td>\n",
       "      <td>-0.877579</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>CLEC4C|CLEC6A</td>\n",
       "      <td>CLEC4C</td>\n",
       "      <td>CLEC6A</td>\n",
       "      <td>buffering</td>\n",
       "      <td>buffering_in_PC9</td>\n",
       "      <td>1027</td>\n",
       "      <td>0.748590</td>\n",
       "      <td>0.000489</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>751</td>\n",
       "      <td>0.069986</td>\n",
       "      <td>0.789673</td>\n",
       "      <td>-0.330330</td>\n",
       "      <td>0.497574</td>\n",
       "      <td>-0.308274</td>\n",
       "      <td>-0.184525</td>\n",
       "      <td>True</td>\n",
       "      <td>726547.0</td>\n",
       "      <td>less_than_1Mb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1027</th>\n",
       "      <td>ZFAND6|ZFAND5</td>\n",
       "      <td>ZFAND6</td>\n",
       "      <td>ZFAND5</td>\n",
       "      <td>buffering</td>\n",
       "      <td>buffering_in_PC9</td>\n",
       "      <td>1028</td>\n",
       "      <td>0.763221</td>\n",
       "      <td>0.039909</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>951</td>\n",
       "      <td>0.249774</td>\n",
       "      <td>0.107288</td>\n",
       "      <td>-1.434036</td>\n",
       "      <td>-0.391247</td>\n",
       "      <td>-0.486414</td>\n",
       "      <td>-0.161166</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>METTL7B|METTL7A</td>\n",
       "      <td>METTL7B</td>\n",
       "      <td>METTL7A</td>\n",
       "      <td>buffering</td>\n",
       "      <td>buffering_in_PC9</td>\n",
       "      <td>1029</td>\n",
       "      <td>0.789800</td>\n",
       "      <td>0.026260</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>338</td>\n",
       "      <td>-0.142798</td>\n",
       "      <td>0.424612</td>\n",
       "      <td>-1.460019</td>\n",
       "      <td>-0.385937</td>\n",
       "      <td>-1.060845</td>\n",
       "      <td>-1.058157</td>\n",
       "      <td>True</td>\n",
       "      <td>4758206.0</td>\n",
       "      <td>greater_than_1Mb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1029</th>\n",
       "      <td>PMPCB|UQCRC1</td>\n",
       "      <td>PMPCB</td>\n",
       "      <td>UQCRC1</td>\n",
       "      <td>buffering</td>\n",
       "      <td>buffering_in_PC9</td>\n",
       "      <td>1030</td>\n",
       "      <td>0.931335</td>\n",
       "      <td>0.001425</td>\n",
       "      <td>neither_in_HeLa</td>\n",
       "      <td>1028</td>\n",
       "      <td>0.591570</td>\n",
       "      <td>0.188484</td>\n",
       "      <td>-3.405244</td>\n",
       "      <td>-1.836689</td>\n",
       "      <td>-3.371047</td>\n",
       "      <td>-2.352422</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diff_chr</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1030 rows  19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         paralog_pair  target1  target2           GI_flag       PC9_GI_flag  \\\n",
       "0         CCNL2|CCNL1    CCNL2    CCNL1  synthetic_lethal         SL_in_PC9   \n",
       "1           CDK6|CDK4     CDK6     CDK4  synthetic_lethal         SL_in_PC9   \n",
       "2         GSK3B|GSK3A    GSK3B    GSK3A  synthetic_lethal         SL_in_PC9   \n",
       "3         G3BP2|G3BP1    G3BP2    G3BP1  synthetic_lethal         SL_in_PC9   \n",
       "4         CNOT8|CNOT7    CNOT8    CNOT7  synthetic_lethal         SL_in_PC9   \n",
       "...               ...      ...      ...               ...               ...   \n",
       "1025    ACTL6A|ACTL6B   ACTL6A   ACTL6B         buffering  buffering_in_PC9   \n",
       "1026    CLEC4C|CLEC6A   CLEC4C   CLEC6A         buffering  buffering_in_PC9   \n",
       "1027    ZFAND6|ZFAND5   ZFAND6   ZFAND5         buffering  buffering_in_PC9   \n",
       "1028  METTL7B|METTL7A  METTL7B  METTL7A         buffering  buffering_in_PC9   \n",
       "1029     PMPCB|UQCRC1    PMPCB   UQCRC1         buffering  buffering_in_PC9   \n",
       "\n",
       "      PC9_GI_score_rank  PC9_GI_score  PC9_GI_fdr     HeLa_GI_flag  \\\n",
       "0                     1     -2.313788    0.000165       SL_in_HeLa   \n",
       "1                     2     -1.537568    0.000340  neither_in_HeLa   \n",
       "2                     3     -1.429767    0.001618       SL_in_HeLa   \n",
       "3                     4     -1.429397    0.004693  neither_in_HeLa   \n",
       "4                     5     -1.208182    0.004693       SL_in_HeLa   \n",
       "...                 ...           ...         ...              ...   \n",
       "1025               1026      0.728780    0.029700  neither_in_HeLa   \n",
       "1026               1027      0.748590    0.000489  neither_in_HeLa   \n",
       "1027               1028      0.763221    0.039909  neither_in_HeLa   \n",
       "1028               1029      0.789800    0.026260  neither_in_HeLa   \n",
       "1029               1030      0.931335    0.001425  neither_in_HeLa   \n",
       "\n",
       "      HeLa_GI_score_rank  HeLa_GI_score  HeLa_GI_fdr  PC9_DKO_expected_CS  \\\n",
       "0                     10      -1.261084     0.026168             0.375079   \n",
       "1                    210      -0.251834     0.307389            -2.309117   \n",
       "2                      4      -1.659580     0.012451            -0.343343   \n",
       "3                    663       0.021784     0.951159            -0.260061   \n",
       "4                      1      -2.063079     0.000022             0.775066   \n",
       "...                  ...            ...          ...                  ...   \n",
       "1025                 121      -0.388877     0.033472            -1.909384   \n",
       "1026                 751       0.069986     0.789673            -0.330330   \n",
       "1027                 951       0.249774     0.107288            -1.434036   \n",
       "1028                 338      -0.142798     0.424612            -1.460019   \n",
       "1029                1028       0.591570     0.188484            -3.405244   \n",
       "\n",
       "      PC9_DKO_observed_CS  HeLa_DKO_expected_CS  HeLa_DKO_observed_CS  \\\n",
       "0               -1.987383             -0.099315             -1.332104   \n",
       "1               -3.408344             -0.451694             -0.632285   \n",
       "2               -1.691436              0.649198             -1.073316   \n",
       "3               -1.622894             -0.858532             -0.715920   \n",
       "4               -0.554362              0.166142             -1.900996   \n",
       "...                   ...                   ...                   ...   \n",
       "1025            -0.814789             -0.574970             -0.877579   \n",
       "1026             0.497574             -0.308274             -0.184525   \n",
       "1027            -0.391247             -0.486414             -0.161166   \n",
       "1028            -0.385937             -1.060845             -1.058157   \n",
       "1029            -1.836689             -3.371047             -2.352422   \n",
       "\n",
       "      same_chr  same_chr_dist         proximity  \n",
       "0        False            NaN          diff_chr  \n",
       "1        False            NaN          diff_chr  \n",
       "2        False            NaN          diff_chr  \n",
       "3        False            NaN          diff_chr  \n",
       "4        False            NaN          diff_chr  \n",
       "...        ...            ...               ...  \n",
       "1025     False            NaN          diff_chr  \n",
       "1026      True       726547.0     less_than_1Mb  \n",
       "1027     False            NaN          diff_chr  \n",
       "1028      True      4758206.0  greater_than_1Mb  \n",
       "1029     False            NaN          diff_chr  \n",
       "\n",
       "[1030 rows x 19 columns]"
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PC9_HeLa_GIs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "a0e61f6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['paralog_pair', 'target1', 'target2', 'GI_flag', 'PC9_GI_flag',\n",
       "       'PC9_GI_score_rank', 'PC9_GI_score', 'PC9_GI_fdr', 'HeLa_GI_flag',\n",
       "       'HeLa_GI_score_rank', 'HeLa_GI_score', 'HeLa_GI_fdr',\n",
       "       'PC9_DKO_expected_CS', 'PC9_DKO_observed_CS', 'HeLa_DKO_expected_CS',\n",
       "       'HeLa_DKO_observed_CS', 'same_chr', 'same_chr_dist', 'proximity'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PC9_HeLa_GIs.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "370695cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"Phenotype\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "7d350d97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PC9\n",
      "###############################\n",
      "HeLa\n",
      "###############################\n"
     ]
    }
   ],
   "source": [
    "for cell_line in ['PC9', 'HeLa']:\n",
    "    curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"Phenotype\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "    curr_GI[\"Gene_A\"] = PC9_HeLa_GIs[\"target1\"]\n",
    "    curr_GI[\"Gene_B\"] = PC9_HeLa_GIs[\"target2\"]\n",
    "    curr_GI[\"GI_Score\"] = PC9_HeLa_GIs.loc[:, cell_line + \"_GI_score\"]\n",
    "    \n",
    "    curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['parrish_data']] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Cell_Line\"] = [cell_line] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI = curr_GI.fillna(0)\n",
    "    curr_GI[\"GI_Cutoff\"] = [-0.5] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Stat_Score\"] = PC9_HeLa_GIs.loc[:, cell_line + \"_GI_fdr\"]\n",
    "    curr_GI[\"Stat_Cutoff\"] = [0.1] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Phenotype\"] = [float(\"nan\")] * len(curr_GI[\"Gene_A\"])\n",
    "    \n",
    "    all_GI = pd.concat([all_GI, curr_GI])\n",
    "    print(cell_line)\n",
    "    print(\"###############################\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "5df32b41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCNL2</td>\n",
       "      <td>CCNL1</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.313788</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.000165</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDK6</td>\n",
       "      <td>CDK4</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.537568</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.00034</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSK3B</td>\n",
       "      <td>GSK3A</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.429767</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G3BP2</td>\n",
       "      <td>G3BP1</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.429397</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.004693</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CNOT8</td>\n",
       "      <td>CNOT7</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.208182</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.004693</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1025</th>\n",
       "      <td>ACTL6A</td>\n",
       "      <td>ACTL6B</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.388877</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.033472</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>CLEC4C</td>\n",
       "      <td>CLEC6A</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.069986</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.789673</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1027</th>\n",
       "      <td>ZFAND6</td>\n",
       "      <td>ZFAND5</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.249774</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.107288</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>METTL7B</td>\n",
       "      <td>METTL7A</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.142798</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.424612</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1029</th>\n",
       "      <td>PMPCB</td>\n",
       "      <td>UQCRC1</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.59157</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.188484</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2060 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene_A   Gene_B Study_Source Cell_Line Phenotype  GI_Score GI_Cutoff  \\\n",
       "0       CCNL2    CCNL1     34469736       PC9       NaN -2.313788      -0.5   \n",
       "1        CDK6     CDK4     34469736       PC9       NaN -1.537568      -0.5   \n",
       "2       GSK3B    GSK3A     34469736       PC9       NaN -1.429767      -0.5   \n",
       "3       G3BP2    G3BP1     34469736       PC9       NaN -1.429397      -0.5   \n",
       "4       CNOT8    CNOT7     34469736       PC9       NaN -1.208182      -0.5   \n",
       "...       ...      ...          ...       ...       ...       ...       ...   \n",
       "1025   ACTL6A   ACTL6B     34469736      HeLa       NaN -0.388877      -0.5   \n",
       "1026   CLEC4C   CLEC6A     34469736      HeLa       NaN  0.069986      -0.5   \n",
       "1027   ZFAND6   ZFAND5     34469736      HeLa       NaN  0.249774      -0.5   \n",
       "1028  METTL7B  METTL7A     34469736      HeLa       NaN -0.142798      -0.5   \n",
       "1029    PMPCB   UQCRC1     34469736      HeLa       NaN   0.59157      -0.5   \n",
       "\n",
       "     Stat_Score Stat_Cutoff  \n",
       "0      0.000165         0.1  \n",
       "1       0.00034         0.1  \n",
       "2      0.001618         0.1  \n",
       "3      0.004693         0.1  \n",
       "4      0.004693         0.1  \n",
       "...         ...         ...  \n",
       "1025   0.033472         0.1  \n",
       "1026   0.789673         0.1  \n",
       "1027   0.107288         0.1  \n",
       "1028   0.424612         0.1  \n",
       "1029   0.188484         0.1  \n",
       "\n",
       "[2060 rows x 9 columns]"
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "84391f72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>ACTGCATCTGTGCAAACGGG</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>ATGTCTCATGAACTACCCTG</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>TGAAATGAAACTTCACACTG</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>TTACTCATATAGGATCCCAA</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC+1</td>\n",
       "      <td>AAGTCTGAAGCACTAAGAAG</td>\n",
       "      <td>AADAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9189</th>\n",
       "      <td>PLAG1+4</td>\n",
       "      <td>TCGCCGGTTCTACACCCGAA</td>\n",
       "      <td>PLAG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9190</th>\n",
       "      <td>PRPF40A+4</td>\n",
       "      <td>TTGAAGTACTACCTGTATCG</td>\n",
       "      <td>PRPF40A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9191</th>\n",
       "      <td>PXDN+4</td>\n",
       "      <td>TGAAAACCTACGCGGAGTCG</td>\n",
       "      <td>PXDN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9192</th>\n",
       "      <td>RHEB+4</td>\n",
       "      <td>GCAAATTGTTGGATATGGTG</td>\n",
       "      <td>RHEB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9193</th>\n",
       "      <td>VASH1+4</td>\n",
       "      <td>GGAAAGGATGTGGAAACACG</td>\n",
       "      <td>VASH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9194 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0               A2M+1  ACTGCATCTGTGCAAACGGG               A2M\n",
       "1               A2M+2  ATGTCTCATGAACTACCCTG               A2M\n",
       "2               A2M+3  TGAAATGAAACTTCACACTG               A2M\n",
       "3               A2M+4  TTACTCATATAGGATCCCAA               A2M\n",
       "4             AADAC+1  AAGTCTGAAGCACTAAGAAG             AADAC\n",
       "...               ...                   ...               ...\n",
       "9189          PLAG1+4  TCGCCGGTTCTACACCCGAA             PLAG1\n",
       "9190        PRPF40A+4  TTGAAGTACTACCTGTATCG           PRPF40A\n",
       "9191           PXDN+4  TGAAAACCTACGCGGAGTCG              PXDN\n",
       "9192           RHEB+4  GCAAATTGTTGGATATGGTG              RHEB\n",
       "9193          VASH1+4  GGAAAGGATGTGGAAACACG             VASH1\n",
       "\n",
       "[9194 rows x 3 columns]"
      ]
     },
     "execution_count": 264,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "b5d74d4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>nt404</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt404</td>\n",
       "      <td>2435.9;2367.7;2182.4;3364.1;2893.9;3027.6</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M+2</td>\n",
       "      <td>nt598</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt598</td>\n",
       "      <td>868.75;844.39;778.32;1102.4;1224.9;1207.4</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M+3</td>\n",
       "      <td>nt847</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt847</td>\n",
       "      <td>911.1;885.56;816.27;826.22;1426.7;1421.2</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M+4</td>\n",
       "      <td>nt731</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt731</td>\n",
       "      <td>416.65;404.97;373.28;263.42;450.09;883.55</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M+1</td>\n",
       "      <td>nt54</td>\n",
       "      <td>A2M</td>\n",
       "      <td>nt54</td>\n",
       "      <td>1852.5;1800.5;1659.6;1906.6;1544.9;1913.5</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PC9</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31828</th>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>nt787</td>\n",
       "      <td>nt679</td>\n",
       "      <td>364.51;406.77;408.4;358.82;750.63;1125.7</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31829</th>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>nt401</td>\n",
       "      <td>nt577</td>\n",
       "      <td>1119.0;1248.8;1253.8;1127.2;463.07;971.92</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31830</th>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>nt412</td>\n",
       "      <td>nt190</td>\n",
       "      <td>814.83;909.3;912.96;819.14;2003.4;1575.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31831</th>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>nt159</td>\n",
       "      <td>nt616</td>\n",
       "      <td>1275.4;1423.2;1428.9;1278.3;2913.7;2569.3</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31832</th>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>nt796</td>\n",
       "      <td>nt822</td>\n",
       "      <td>1644.1;1834.7;1842.1;1640.6;3097.7;2103.9</td>\n",
       "      <td>Dual</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HELA</td>\n",
       "      <td>plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3</td>\n",
       "      <td>34469736</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64573 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Guide 1 Guide 2 Gene 1 Gene 2  \\\n",
       "0       A2M+1   nt404    A2M  nt404   \n",
       "1       A2M+2   nt598    A2M  nt598   \n",
       "2       A2M+3   nt847    A2M  nt847   \n",
       "3       A2M+4   nt731    A2M  nt731   \n",
       "4       A2M+1    nt54    A2M   nt54   \n",
       "...       ...     ...    ...    ...   \n",
       "31828   nt787   nt679  nt787  nt679   \n",
       "31829   nt401   nt577  nt401  nt577   \n",
       "31830   nt412   nt190  nt412  nt190   \n",
       "31831   nt159   nt616  nt159  nt616   \n",
       "31832   nt796   nt822  nt796  nt822   \n",
       "\n",
       "                                Count Replicates    Type Sequencing Cell Line  \\\n",
       "0      2435.9;2367.7;2182.4;3364.1;2893.9;3027.6  Single        NaN       PC9   \n",
       "1      868.75;844.39;778.32;1102.4;1224.9;1207.4  Single        NaN       PC9   \n",
       "2       911.1;885.56;816.27;826.22;1426.7;1421.2  Single        NaN       PC9   \n",
       "3      416.65;404.97;373.28;263.42;450.09;883.55  Single        NaN       PC9   \n",
       "4      1852.5;1800.5;1659.6;1906.6;1544.9;1913.5  Single        NaN       PC9   \n",
       "...                                          ...     ...        ...       ...   \n",
       "31828   364.51;406.77;408.4;358.82;750.63;1125.7    Dual        NaN      HELA   \n",
       "31829  1119.0;1248.8;1253.8;1127.2;463.07;971.92    Dual        NaN      HELA   \n",
       "31830   814.83;909.3;912.96;819.14;2003.4;1575.3    Dual        NaN      HELA   \n",
       "31831  1275.4;1423.2;1428.9;1278.3;2913.7;2569.3    Dual        NaN      HELA   \n",
       "31832  1644.1;1834.7;1842.1;1640.6;3097.7;2103.9    Dual        NaN      HELA   \n",
       "\n",
       "                                             Condition     Study  \n",
       "0      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "1      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "2      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "3      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "4      plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "...                                                ...       ...  \n",
       "31828  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31829  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31830  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31831  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "31832  plasmid_1;plasmid_2;plasmid_3;LTP_1;LTP_2;LTP_3  34469736  \n",
       "\n",
       "[64573 rows x 10 columns]"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "1cb096c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['plasmid_1', 'plasmid_2', 'plasmid_3'], ['LTP_1', 'LTP_2', 'LTP_3']]"
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_conditions['parrish_data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "51261466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCNL2</td>\n",
       "      <td>CCNL1</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.313788</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.000165</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDK6</td>\n",
       "      <td>CDK4</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.537568</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.00034</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSK3B</td>\n",
       "      <td>GSK3A</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.429767</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G3BP2</td>\n",
       "      <td>G3BP1</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.429397</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.004693</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CNOT8</td>\n",
       "      <td>CNOT7</td>\n",
       "      <td>34469736</td>\n",
       "      <td>PC9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.208182</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.004693</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1025</th>\n",
       "      <td>ACTL6A</td>\n",
       "      <td>ACTL6B</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.388877</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.033472</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>CLEC4C</td>\n",
       "      <td>CLEC6A</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.069986</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.789673</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1027</th>\n",
       "      <td>ZFAND6</td>\n",
       "      <td>ZFAND5</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.249774</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.107288</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>METTL7B</td>\n",
       "      <td>METTL7A</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.142798</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.424612</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1029</th>\n",
       "      <td>PMPCB</td>\n",
       "      <td>UQCRC1</td>\n",
       "      <td>34469736</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.59157</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.188484</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2060 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene_A   Gene_B Study_Source Cell_Line Phenotype  GI_Score GI_Cutoff  \\\n",
       "0       CCNL2    CCNL1     34469736       PC9       NaN -2.313788      -0.5   \n",
       "1        CDK6     CDK4     34469736       PC9       NaN -1.537568      -0.5   \n",
       "2       GSK3B    GSK3A     34469736       PC9       NaN -1.429767      -0.5   \n",
       "3       G3BP2    G3BP1     34469736       PC9       NaN -1.429397      -0.5   \n",
       "4       CNOT8    CNOT7     34469736       PC9       NaN -1.208182      -0.5   \n",
       "...       ...      ...          ...       ...       ...       ...       ...   \n",
       "1025   ACTL6A   ACTL6B     34469736      HeLa       NaN -0.388877      -0.5   \n",
       "1026   CLEC4C   CLEC6A     34469736      HeLa       NaN  0.069986      -0.5   \n",
       "1027   ZFAND6   ZFAND5     34469736      HeLa       NaN  0.249774      -0.5   \n",
       "1028  METTL7B  METTL7A     34469736      HeLa       NaN -0.142798      -0.5   \n",
       "1029    PMPCB   UQCRC1     34469736      HeLa       NaN   0.59157      -0.5   \n",
       "\n",
       "     Stat_Score Stat_Cutoff  \n",
       "0      0.000165         0.1  \n",
       "1       0.00034         0.1  \n",
       "2      0.001618         0.1  \n",
       "3      0.004693         0.1  \n",
       "4      0.004693         0.1  \n",
       "...         ...         ...  \n",
       "1025   0.033472         0.1  \n",
       "1026   0.789673         0.1  \n",
       "1027   0.107288         0.1  \n",
       "1028   0.424612         0.1  \n",
       "1029   0.188484         0.1  \n",
       "\n",
       "[2060 rows x 9 columns]"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "fef2f674",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "Both GI and Stat cutoffs are present...\n",
      "Counts reference...\n",
      "Number of double pairs: 31630\n",
      "Number of controls: 983\n",
      "Number of singles: 31960\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = all_GI.copy(), study_controls = controls['parrish_data'], study_conditions = study_conditions['parrish_data'], can_control_be_substring = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "e9a7ca86",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF['Study_Source'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "2023bf75",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF.loc[PREV_REF['Study_Source'] == 'Parrish', :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "05be4b12",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Not SL    1903\n",
       "SL         157\n",
       "Name: SL_or_not, dtype: int64"
      ]
     },
     "execution_count": 271,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "745683db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 9194\n",
      "Counts insert: 64573\n",
      "Score insert: 2060\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "506d1038",
   "metadata": {},
   "source": [
    "## Shen Data (HeLa, 293T, A549)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e2491c3",
   "metadata": {},
   "source": [
    "## Sequence Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "0499a936",
   "metadata": {},
   "outputs": [],
   "source": [
    "shen_loc = os.path.join(learning_goals_loc_general, \"Shen\")\n",
    "\n",
    "lib_file = pd.read_excel(os.path.join(shen_loc, \"2 - Raw and processed data from CRISPR competitive growth assay.xlsx\"), sheet_name = \"Table_S2A_CRISPR-KO-gRNAs\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "93230070",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#</th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_A_chr</th>\n",
       "      <th>Gene_A_pos</th>\n",
       "      <th>Gene_A_seq</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Gene_B_chr</th>\n",
       "      <th>Gene_B_pos</th>\n",
       "      <th>Gene_B_seq</th>\n",
       "      <th>FinalSequence</th>\n",
       "      <th>SequenceID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>41276018.0</td>\n",
       "      <td>TCTTGTGCTGACTTACCAGA</td>\n",
       "      <td>NonTargeting0412</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCACGCTGTACAGACGACAA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGTCTTGTGCTGACTTAC...</td>\n",
       "      <td>BRCA1_chr17_41276018__NonTargeting0412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>NonTargeting0352</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCCATTCTAGTCCCGGCATA</td>\n",
       "      <td>SETD2</td>\n",
       "      <td>chr3</td>\n",
       "      <td>47142972.0</td>\n",
       "      <td>AACGTTAACTCTGAGCCTGA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGCCATTCTAGTCCCGG...</td>\n",
       "      <td>NonTargeting0352__SETD2_chr3_47142972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>41256141.0</td>\n",
       "      <td>TGAACCCGAAAATCCTTCCT</td>\n",
       "      <td>NonTargeting0362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAGTGATGCTTAGACTCCGT</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGTGAACCCGAAAATCCT...</td>\n",
       "      <td>BRCA1_chr17_41256141__NonTargeting0362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>NonTargeting0352</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCCATTCTAGTCCCGGCATA</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13</td>\n",
       "      <td>32912006.0</td>\n",
       "      <td>CTGTCTACCTGACCAATCGA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGCCATTCTAGTCCCGG...</td>\n",
       "      <td>NonTargeting0352__BRCA2_chr13_32912006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>RB1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>48934148.0</td>\n",
       "      <td>TTCAGGGGAAGTATTACAAA</td>\n",
       "      <td>NonTargeting0362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAGTGATGCTTAGACTCCGT</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGTTCAGGGGAAGTATTA...</td>\n",
       "      <td>RB1_chr13_48934148__NonTargeting0362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24904</th>\n",
       "      <td>24905</td>\n",
       "      <td>NonTargeting0362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAGTGATGCTTAGACTCCGT</td>\n",
       "      <td>NonTargeting0084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TGGTTCCGTAGGTCGGTATA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGAGTGATGCTTAGACT...</td>\n",
       "      <td>NonTargeting0362__NonTargeting0084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24905</th>\n",
       "      <td>24906</td>\n",
       "      <td>NonTargeting0002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGCTTCCGCGGCCCGTTCAA</td>\n",
       "      <td>NonTargeting0084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TGGTTCCGTAGGTCGGTATA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCGCTTCCGCGGCCCGT...</td>\n",
       "      <td>NonTargeting0002__NonTargeting0084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24906</th>\n",
       "      <td>24907</td>\n",
       "      <td>NonTargeting0002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGCTTCCGCGGCCCGTTCAA</td>\n",
       "      <td>NonTargeting0082</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACGTCCATACTGTCGGCTAC</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCGCTTCCGCGGCCCGT...</td>\n",
       "      <td>NonTargeting0002__NonTargeting0082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24907</th>\n",
       "      <td>24908</td>\n",
       "      <td>NonTargeting0276</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTCCACGGTAAAATCGGTCA</td>\n",
       "      <td>NonTargeting0008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGGCCCGCATAGGATATCGC</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGTTCCACGGTAAAATCG...</td>\n",
       "      <td>NonTargeting0276__NonTargeting0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24908</th>\n",
       "      <td>24909</td>\n",
       "      <td>NonTargeting0098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GTACGGGGCGATCATCCACA</td>\n",
       "      <td>NonTargeting0515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCCCTATGCAGACTACAATT</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGTACGGGGCGATCATC...</td>\n",
       "      <td>NonTargeting0098__NonTargeting0515</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24909 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           #            Gene_A Gene_A_chr  Gene_A_pos            Gene_A_seq  \\\n",
       "0          1             BRCA1      chr17  41276018.0  TCTTGTGCTGACTTACCAGA   \n",
       "1          2  NonTargeting0352        NaN         NaN  GCCATTCTAGTCCCGGCATA   \n",
       "2          3             BRCA1      chr17  41256141.0  TGAACCCGAAAATCCTTCCT   \n",
       "3          4  NonTargeting0352        NaN         NaN  GCCATTCTAGTCCCGGCATA   \n",
       "4          5               RB1      chr13  48934148.0  TTCAGGGGAAGTATTACAAA   \n",
       "...      ...               ...        ...         ...                   ...   \n",
       "24904  24905  NonTargeting0362        NaN         NaN  GAGTGATGCTTAGACTCCGT   \n",
       "24905  24906  NonTargeting0002        NaN         NaN  CGCTTCCGCGGCCCGTTCAA   \n",
       "24906  24907  NonTargeting0002        NaN         NaN  CGCTTCCGCGGCCCGTTCAA   \n",
       "24907  24908  NonTargeting0276        NaN         NaN  TTCCACGGTAAAATCGGTCA   \n",
       "24908  24909  NonTargeting0098        NaN         NaN  GTACGGGGCGATCATCCACA   \n",
       "\n",
       "                 Gene_B Gene_B_chr  Gene_B_pos            Gene_B_seq  \\\n",
       "0      NonTargeting0412        NaN         NaN  GCACGCTGTACAGACGACAA   \n",
       "1                 SETD2       chr3  47142972.0  AACGTTAACTCTGAGCCTGA   \n",
       "2      NonTargeting0362        NaN         NaN  GAGTGATGCTTAGACTCCGT   \n",
       "3                 BRCA2      chr13  32912006.0  CTGTCTACCTGACCAATCGA   \n",
       "4      NonTargeting0362        NaN         NaN  GAGTGATGCTTAGACTCCGT   \n",
       "...                 ...        ...         ...                   ...   \n",
       "24904  NonTargeting0084        NaN         NaN  TGGTTCCGTAGGTCGGTATA   \n",
       "24905  NonTargeting0084        NaN         NaN  TGGTTCCGTAGGTCGGTATA   \n",
       "24906  NonTargeting0082        NaN         NaN  ACGTCCATACTGTCGGCTAC   \n",
       "24907  NonTargeting0008        NaN         NaN  GGGCCCGCATAGGATATCGC   \n",
       "24908  NonTargeting0515        NaN         NaN  CCCCTATGCAGACTACAATT   \n",
       "\n",
       "                                           FinalSequence  \\\n",
       "0      tatatatcttgtggaaaggacgaaacACCGTCTTGTGCTGACTTAC...   \n",
       "1      tatatatcttgtggaaaggacgaaacACCGGCCATTCTAGTCCCGG...   \n",
       "2      tatatatcttgtggaaaggacgaaacACCGTGAACCCGAAAATCCT...   \n",
       "3      tatatatcttgtggaaaggacgaaacACCGGCCATTCTAGTCCCGG...   \n",
       "4      tatatatcttgtggaaaggacgaaacACCGTTCAGGGGAAGTATTA...   \n",
       "...                                                  ...   \n",
       "24904  tatatatcttgtggaaaggacgaaacACCGGAGTGATGCTTAGACT...   \n",
       "24905  tatatatcttgtggaaaggacgaaacACCGCGCTTCCGCGGCCCGT...   \n",
       "24906  tatatatcttgtggaaaggacgaaacACCGCGCTTCCGCGGCCCGT...   \n",
       "24907  tatatatcttgtggaaaggacgaaacACCGTTCCACGGTAAAATCG...   \n",
       "24908  tatatatcttgtggaaaggacgaaacACCGGTACGGGGCGATCATC...   \n",
       "\n",
       "                                   SequenceID  \n",
       "0      BRCA1_chr17_41276018__NonTargeting0412  \n",
       "1       NonTargeting0352__SETD2_chr3_47142972  \n",
       "2      BRCA1_chr17_41256141__NonTargeting0362  \n",
       "3      NonTargeting0352__BRCA2_chr13_32912006  \n",
       "4        RB1_chr13_48934148__NonTargeting0362  \n",
       "...                                       ...  \n",
       "24904      NonTargeting0362__NonTargeting0084  \n",
       "24905      NonTargeting0002__NonTargeting0084  \n",
       "24906      NonTargeting0002__NonTargeting0082  \n",
       "24907      NonTargeting0276__NonTargeting0008  \n",
       "24908      NonTargeting0098__NonTargeting0515  \n",
       "\n",
       "[24909 rows x 11 columns]"
      ]
     },
     "execution_count": 274,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "060ddccf",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = pd.DataFrame(columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "4e8b65c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['sgRNA_guide_name'] = [i.split('__')[0] for i in lib_file['SequenceID']] + [i.split('__')[1] for i in lib_file['SequenceID']]\n",
    "sequence_ref['sgRNA_guide_seq'] = lib_file['Gene_A_seq'].tolist() + lib_file['Gene_B_seq'].tolist()\n",
    "sequence_ref['sgRNA_target_name'] = lib_file['Gene_A'].tolist() + lib_file['Gene_B'].tolist()\n",
    "sequence_ref.drop_duplicates('sgRNA_guide_name', inplace = True, ignore_index = True)\n",
    "control_idx = np.array([True if 'NonTargeting' in i else False for i in sequence_ref['sgRNA_target_name']])\n",
    "sequence_ref.loc[control_idx, 'sgRNA_target_name'] = 'control'\n",
    "sequence_ref.reset_index(inplace = True, drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "id": "8f431a58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "control     47\n",
       "EGFR         3\n",
       "FNTA         3\n",
       "TOP1         3\n",
       "HSP90AA1     3\n",
       "            ..\n",
       "BRCA2        3\n",
       "ARID2        3\n",
       "BTK          3\n",
       "CHEK1        3\n",
       "CHEK2        3\n",
       "Name: sgRNA_target_name, Length: 74, dtype: int64"
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref['sgRNA_target_name'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "251f57ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA1_chr17_41276018</td>\n",
       "      <td>TCTTGTGCTGACTTACCAGA</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NonTargeting0352</td>\n",
       "      <td>GCCATTCTAGTCCCGGCATA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA1_chr17_41256141</td>\n",
       "      <td>TGAACCCGAAAATCCTTCCT</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RB1_chr13_48934148</td>\n",
       "      <td>TTCAGGGGAAGTATTACAAA</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NonTargeting0412</td>\n",
       "      <td>GCACGCTGTACAGACGACAA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>NonTargeting0028</td>\n",
       "      <td>CGCGACGACTCAACCTAGTC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>NonTargeting0152</td>\n",
       "      <td>TTCACCGTCCACGTGCGCAT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>NonTargeting0002</td>\n",
       "      <td>CGCTTCCGCGGCCCGTTCAA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>NonTargeting0157</td>\n",
       "      <td>GGGTGCCCACTAATAGCCGC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>NonTargeting0093</td>\n",
       "      <td>TGCGGCGTAATGCTTGAAAG</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>266 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0    BRCA1_chr17_41276018  TCTTGTGCTGACTTACCAGA             BRCA1\n",
       "1        NonTargeting0352  GCCATTCTAGTCCCGGCATA           control\n",
       "2    BRCA1_chr17_41256141  TGAACCCGAAAATCCTTCCT             BRCA1\n",
       "3      RB1_chr13_48934148  TTCAGGGGAAGTATTACAAA               RB1\n",
       "4        NonTargeting0412  GCACGCTGTACAGACGACAA           control\n",
       "..                    ...                   ...               ...\n",
       "261      NonTargeting0028  CGCGACGACTCAACCTAGTC           control\n",
       "262      NonTargeting0152  TTCACCGTCCACGTGCGCAT           control\n",
       "263      NonTargeting0002  CGCTTCCGCGGCCCGTTCAA           control\n",
       "264      NonTargeting0157  GGGTGCCCACTAATAGCCGC           control\n",
       "265      NonTargeting0093  TGCGGCGTAATGCTTGAAAG           control\n",
       "\n",
       "[266 rows x 3 columns]"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47f6cfba",
   "metadata": {},
   "source": [
    "## Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "f20c18b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "shen_loc = os.path.join(learning_goals_loc_general, \"Shen\")\n",
    "\n",
    "cl_skip_footers = {}\n",
    "cl_skip_footers['HeLa'] = 3\n",
    "cl_skip_footers['293T'] = 2\n",
    "cl_skip_footers['A549'] = 5\n",
    "\n",
    "counts_HELA = pd.read_excel(os.path.join(shen_loc, \"raw_counts_HeLa-A549-293T_NatMeth-2017.xlsx\"), sheet_name = \"HeLa\", skiprows = 1, skipfooter = 3)\n",
    "counts_293T = pd.read_excel(os.path.join(shen_loc, \"raw_counts_HeLa-A549-293T_NatMeth-2017.xlsx\"), sheet_name = \"293T\", skiprows = 1, skipfooter = 2)\n",
    "counts_A549 = pd.read_excel(os.path.join(shen_loc, \"raw_counts_HeLa-A549-293T_NatMeth-2017.xlsx\"), sheet_name = \"A549\", skiprows = 1, skipfooter = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "b86b7c4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['construct_id', 'Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d3-2_S2_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4R4R_d3_2_S8_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d14-1_S3_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d14-2_S4_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d21-1_S1_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d21-2_S2_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d28-1_S5_trimmed53_len_filtered_counts',\n",
       "       'Hela-CV4-d28-2_S6_trimmed53_len_filtered_counts', 'Unnamed: 11',\n",
       "       'Unnamed: 12'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 280,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_HELA.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "2cb37d50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['construct_id', '293_CV4_d3_3_S1_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d3_4_S2_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d14_3_S3_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d14_4_S4_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d21_4_S6_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d21_3_S5_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d28_3_S7_trimmed53_len_filtered_counts',\n",
       "       '293_CV4_d28_4_S8_trimmed53_len_filtered_counts', 'Unnamed: 9',\n",
       "       'Unnamed: 10'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 281,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_293T.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "378b3064",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['construct_id', 'A549_CV4_d3_1_S1_trimmed53_len_filtered_counts',\n",
       "       'A549_CV4_d3_2_S2_trimmed53_len_filtered_counts',\n",
       "       'A549_CV4_d14_1_S3_trimmed53_len_filtered_counts',\n",
       "       'A549_CV4_d14_2_S4_trimmed53_len_filtered_counts',\n",
       "       'A549-CV4-d21-1_S3_trimmed53_len_filtered_counts',\n",
       "       'A549-CV4-d21-2_S4_trimmed53_len_filtered_counts',\n",
       "       'A549-CV4-d28-1_S5_trimmed53_len_filtered_counts',\n",
       "       'A549_CV4_d28_1_S5_trimmed53_len_filtered_counts',\n",
       "       'A549-CV4-d28-2_S6_trimmed53_len_filtered_counts',\n",
       "       'A549_CV4_d28_2_S6_trimmed53_len_filtered_counts', 'Unnamed: 11',\n",
       "       'Unnamed: 12'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_A549.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "aea66b71",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp_seq_ref = sequence_ref.copy()\n",
    "temp_seq_ref.index = temp_seq_ref['sgRNA_guide_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "5c1246d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "study_conditions['shen_data'] = {}\n",
    "study_conditions['shen_data']['HELA'] = [['Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_counts',\n",
    "       'Hela-CV4R4R_d3_2_S8_trimmed53_len_filtered_counts'],\n",
    "                                         ['Hela-CV4-d28-1_S5_trimmed53_len_filtered_counts',\n",
    "       'Hela-CV4-d28-2_S6_trimmed53_len_filtered_counts']]\n",
    "\n",
    "study_conditions['shen_data']['293T'] = [['293_CV4_d3_3_S1_trimmed53_len_filtered_counts',\n",
    "       '293_CV4_d3_4_S2_trimmed53_len_filtered_counts'],\n",
    "                                         ['293_CV4_d28_3_S7_trimmed53_len_filtered_counts',\n",
    "       '293_CV4_d28_4_S8_trimmed53_len_filtered_counts']]\n",
    "\n",
    "study_conditions['shen_data']['A549'] = [['A549_CV4_d3_1_S1_trimmed53_len_filtered_counts',\n",
    "                                            'A549_CV4_d3_2_S2_trimmed53_len_filtered_counts'],\n",
    "                                         ['A549_CV4_d28_1_S5_trimmed53_len_filtered_counts',\n",
    "                                          'A549_CV4_d28_2_S6_trimmed53_len_filtered_counts']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "281f1ed1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CL: HELA\n",
      "CL: 293T\n",
      "CL: A549\n"
     ]
    }
   ],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\", \"Study\"])\n",
    "\n",
    "for cl in cl_skip_footers.keys():\n",
    "    curr_counts = pd.read_excel(os.path.join(shen_loc, \"raw_counts_HeLa-A549-293T_NatMeth-2017.xlsx\"), sheet_name = cl, skiprows = 1, skipfooter = cl_skip_footers[cl])\n",
    "    cl = cl.upper()\n",
    "    print('CL: ' + cl)\n",
    "\n",
    "    # drop last 2 columns\n",
    "    curr_counts.drop(curr_counts.columns[(len(curr_counts.columns)-2):len(curr_counts.columns)], axis = 1, inplace = True)\n",
    "    curr_counts['construct_id'] = [i.replace('ControlGuideForHuman', '') for i in curr_counts['construct_id']]\n",
    "    \n",
    "    # get the whole sample at columns\n",
    "    study_columns = curr_counts.columns[1:].tolist()\n",
    "\n",
    "    curr_counts['construct_id'] = [i.replace('ControlGuideForHuman', '') for i in curr_counts['construct_id']]\n",
    "    curr_counts['Guide 1'] = [i.split('__')[0] for i in curr_counts['construct_id']]\n",
    "    curr_counts['Guide 2'] = [i.split('__')[1] for i in curr_counts['construct_id']]\n",
    "    curr_counts['Gene 1'] = [temp_seq_ref.loc[i, 'sgRNA_target_name'] for i in curr_counts['Guide 1']]\n",
    "    curr_counts['Gene 2'] = [temp_seq_ref.loc[i, 'sgRNA_target_name'] for i in curr_counts['Guide 2']]\n",
    "    curr_counts['Type'] = None\n",
    "    curr_counts['Sequencing'] = None\n",
    "    curr_counts['Cell Line'] = cl\n",
    "    curr_counts['Condition'] = cl\n",
    "    curr_counts['Study'] = study_name_to_pubmed_id['shen_data']\n",
    "    \n",
    "    curr_counts[\"Count Replicates\"] = curr_counts[study_columns].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    "    )\n",
    "    curr_counts[\"Condition\"] = ';'.join(study_columns)\n",
    "    \n",
    "    \n",
    "    counts_ref = pd.concat([counts_ref, curr_counts[counts_ref.columns]])\n",
    "    \n",
    "    \n",
    "    #\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "22a238b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1_chr9_133589790</td>\n",
       "      <td>ADA_chr20_43257696</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ADA</td>\n",
       "      <td>0;3;198;127;146;243;89;238;61;211</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL1_chr9_133589790</td>\n",
       "      <td>ADA_chr20_43257769</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ADA</td>\n",
       "      <td>417;7;616;494;142;242;112;263;42;270</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1_chr9_133589790</td>\n",
       "      <td>ALK_chr2_29541172</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>2;0;659;351;75;94;279;108;342;101</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL1_chr9_133589790</td>\n",
       "      <td>ALK_chr2_30143522</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>169;10;238;563;266;248;204;330;163;191</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL1_chr9_133589790</td>\n",
       "      <td>ARID1A_chr1_27056288</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>2;949;176;342;258;372;472;497;515;254</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24903</th>\n",
       "      <td>WEE1_chr11_9595845</td>\n",
       "      <td>TSC1_chr9_135804256</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>157;331;63;416;22;270;77;26;191;66</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24904</th>\n",
       "      <td>WEE1_chr11_9595845</td>\n",
       "      <td>TYMS_chr18_657730</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>958;1209;643;843;721;1239;881;286;2032;790</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24905</th>\n",
       "      <td>WEE1_chr11_9595845</td>\n",
       "      <td>TYMS_chr18_659665</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>333;640;127;190;118;38;97;33;3;0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24906</th>\n",
       "      <td>WEE1_chr11_9595845</td>\n",
       "      <td>VEGFA_chr6_43742077</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>VEGFA</td>\n",
       "      <td>1010;1709;544;403;812;450;684;236;506;171</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24907</th>\n",
       "      <td>WEE1_chr11_9595845</td>\n",
       "      <td>VHL_chr3_10183773</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>VHL</td>\n",
       "      <td>620;568;365;374;212;362;173;47;349;96</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74724 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Guide 1               Guide 2 Gene 1  Gene 2  \\\n",
       "0      ABL1_chr9_133589790    ADA_chr20_43257696   ABL1     ADA   \n",
       "1      ABL1_chr9_133589790    ADA_chr20_43257769   ABL1     ADA   \n",
       "2      ABL1_chr9_133589790     ALK_chr2_29541172   ABL1     ALK   \n",
       "3      ABL1_chr9_133589790     ALK_chr2_30143522   ABL1     ALK   \n",
       "4      ABL1_chr9_133589790  ARID1A_chr1_27056288   ABL1  ARID1A   \n",
       "...                    ...                   ...    ...     ...   \n",
       "24903   WEE1_chr11_9595845   TSC1_chr9_135804256   WEE1    TSC1   \n",
       "24904   WEE1_chr11_9595845     TYMS_chr18_657730   WEE1    TYMS   \n",
       "24905   WEE1_chr11_9595845     TYMS_chr18_659665   WEE1    TYMS   \n",
       "24906   WEE1_chr11_9595845   VEGFA_chr6_43742077   WEE1   VEGFA   \n",
       "24907   WEE1_chr11_9595845     VHL_chr3_10183773   WEE1     VHL   \n",
       "\n",
       "                                 Count Replicates  Type Sequencing Cell Line  \\\n",
       "0               0;3;198;127;146;243;89;238;61;211  None       None      HELA   \n",
       "1            417;7;616;494;142;242;112;263;42;270  None       None      HELA   \n",
       "2               2;0;659;351;75;94;279;108;342;101  None       None      HELA   \n",
       "3          169;10;238;563;266;248;204;330;163;191  None       None      HELA   \n",
       "4           2;949;176;342;258;372;472;497;515;254  None       None      HELA   \n",
       "...                                           ...   ...        ...       ...   \n",
       "24903          157;331;63;416;22;270;77;26;191;66  None       None      A549   \n",
       "24904  958;1209;643;843;721;1239;881;286;2032;790  None       None      A549   \n",
       "24905            333;640;127;190;118;38;97;33;3;0  None       None      A549   \n",
       "24906   1010;1709;544;403;812;450;684;236;506;171  None       None      A549   \n",
       "24907       620;568;365;374;212;362;173;47;349;96  None       None      A549   \n",
       "\n",
       "                                               Condition     Study  \n",
       "0      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "1      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "2      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "3      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "4      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "...                                                  ...       ...  \n",
       "24903  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "24904  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "24905  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "24906  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "24907  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  28319113  \n",
       "\n",
       "[74724 rows x 10 columns]"
      ]
     },
     "execution_count": 286,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56ade458",
   "metadata": {},
   "source": [
    "## Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "3533d6ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "shen_loc = os.path.join(learning_goals_loc_general, \"Shen\")\n",
    "\n",
    "HELA_A549_293T = pd.read_excel(os.path.join(shen_loc, \"2 - Raw and processed data from CRISPR competitive growth assay.xlsx\"), sheet_name = \"Table_S2B_CRISPR-KO-assay\")\n",
    "SL_labels = pd.read_excel(os.path.join(shen_loc, \"3 - List of hit genetic interactions from CRISPR assay.xlsx\"), skiprows = 1, sheet_name = \"TableS3B_CRISPR-KO-hits\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "c2f16abb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_gene</th>\n",
       "      <th>geneA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>fA_HeLa</th>\n",
       "      <th>fB_HeLa</th>\n",
       "      <th>fA+fB_HeLa</th>\n",
       "      <th>pi_HeLa</th>\n",
       "      <th>sd_HeLa</th>\n",
       "      <th>PP_HeLa</th>\n",
       "      <th>abs_pi_HeLa</th>\n",
       "      <th>...</th>\n",
       "      <th>fA_293T</th>\n",
       "      <th>fB_293T</th>\n",
       "      <th>fA+fB_293T</th>\n",
       "      <th>pi_293T</th>\n",
       "      <th>sd_293T</th>\n",
       "      <th>PP_293T</th>\n",
       "      <th>abs_pi_293T</th>\n",
       "      <th>FDR_left_293T</th>\n",
       "      <th>FDR_right__293T</th>\n",
       "      <th>293T_Z</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1_ADA</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ADA</td>\n",
       "      <td>-0.010891</td>\n",
       "      <td>-0.023380</td>\n",
       "      <td>-0.034270</td>\n",
       "      <td>-0.038661</td>\n",
       "      <td>0.036023</td>\n",
       "      <td>0.704</td>\n",
       "      <td>0.038661</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039056</td>\n",
       "      <td>-0.026479</td>\n",
       "      <td>-0.065535</td>\n",
       "      <td>0.005756</td>\n",
       "      <td>0.031896</td>\n",
       "      <td>0.207</td>\n",
       "      <td>0.005756</td>\n",
       "      <td>0.800666</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.198972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL1_AKT1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>-0.010891</td>\n",
       "      <td>-0.014149</td>\n",
       "      <td>-0.025039</td>\n",
       "      <td>-0.009330</td>\n",
       "      <td>0.037599</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.009330</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039056</td>\n",
       "      <td>-0.024441</td>\n",
       "      <td>-0.063497</td>\n",
       "      <td>-0.015256</td>\n",
       "      <td>0.027199</td>\n",
       "      <td>0.423</td>\n",
       "      <td>0.015256</td>\n",
       "      <td>0.557710</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.527369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1_ALK</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>-0.010891</td>\n",
       "      <td>-0.013610</td>\n",
       "      <td>-0.024501</td>\n",
       "      <td>-0.015444</td>\n",
       "      <td>0.020846</td>\n",
       "      <td>0.793</td>\n",
       "      <td>0.015444</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039056</td>\n",
       "      <td>-0.024261</td>\n",
       "      <td>-0.063316</td>\n",
       "      <td>-0.034626</td>\n",
       "      <td>0.027275</td>\n",
       "      <td>0.783</td>\n",
       "      <td>0.034626</td>\n",
       "      <td>0.407198</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.196939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL1_APC</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>APC</td>\n",
       "      <td>-0.010891</td>\n",
       "      <td>-0.010225</td>\n",
       "      <td>-0.021116</td>\n",
       "      <td>-0.011350</td>\n",
       "      <td>0.019483</td>\n",
       "      <td>0.512</td>\n",
       "      <td>0.011350</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039056</td>\n",
       "      <td>-0.022621</td>\n",
       "      <td>-0.061677</td>\n",
       "      <td>-0.007403</td>\n",
       "      <td>0.008770</td>\n",
       "      <td>0.669</td>\n",
       "      <td>0.007403</td>\n",
       "      <td>0.639785</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.255905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL1_ARID1A</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>-0.010891</td>\n",
       "      <td>-0.016038</td>\n",
       "      <td>-0.026929</td>\n",
       "      <td>-0.029612</td>\n",
       "      <td>0.032232</td>\n",
       "      <td>0.600</td>\n",
       "      <td>0.029612</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039056</td>\n",
       "      <td>-0.018836</td>\n",
       "      <td>-0.057892</td>\n",
       "      <td>-0.039848</td>\n",
       "      <td>0.033482</td>\n",
       "      <td>0.755</td>\n",
       "      <td>0.039848</td>\n",
       "      <td>0.392145</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.377448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2625</th>\n",
       "      <td>VEGFA_VHL</td>\n",
       "      <td>VEGFA</td>\n",
       "      <td>VHL</td>\n",
       "      <td>-0.019953</td>\n",
       "      <td>-0.081441</td>\n",
       "      <td>-0.101394</td>\n",
       "      <td>-0.027529</td>\n",
       "      <td>0.019448</td>\n",
       "      <td>0.850</td>\n",
       "      <td>0.027529</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.042977</td>\n",
       "      <td>-0.035299</td>\n",
       "      <td>-0.078276</td>\n",
       "      <td>0.017510</td>\n",
       "      <td>0.018634</td>\n",
       "      <td>0.648</td>\n",
       "      <td>0.017510</td>\n",
       "      <td>0.865491</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.605266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2626</th>\n",
       "      <td>VEGFA_WEE1</td>\n",
       "      <td>VEGFA</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>-0.019953</td>\n",
       "      <td>-0.024776</td>\n",
       "      <td>-0.044729</td>\n",
       "      <td>-0.045977</td>\n",
       "      <td>0.025619</td>\n",
       "      <td>0.931</td>\n",
       "      <td>0.045977</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.042977</td>\n",
       "      <td>-0.047175</td>\n",
       "      <td>-0.090152</td>\n",
       "      <td>-0.020847</td>\n",
       "      <td>0.028925</td>\n",
       "      <td>0.526</td>\n",
       "      <td>0.020847</td>\n",
       "      <td>0.499459</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.720636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2627</th>\n",
       "      <td>VHL_WEE1</td>\n",
       "      <td>VHL</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>-0.081441</td>\n",
       "      <td>-0.024776</td>\n",
       "      <td>-0.106217</td>\n",
       "      <td>-0.045337</td>\n",
       "      <td>0.038023</td>\n",
       "      <td>0.783</td>\n",
       "      <td>0.045337</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.035299</td>\n",
       "      <td>-0.047175</td>\n",
       "      <td>-0.082474</td>\n",
       "      <td>0.001916</td>\n",
       "      <td>0.021769</td>\n",
       "      <td>0.106</td>\n",
       "      <td>0.001916</td>\n",
       "      <td>0.769568</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.066241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2628</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2629</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.028929</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2630 rows  33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_gene  geneA   geneB   fA_HeLa   fB_HeLa  fA+fB_HeLa   pi_HeLa  \\\n",
       "0        ABL1_ADA   ABL1     ADA -0.010891 -0.023380   -0.034270 -0.038661   \n",
       "1       ABL1_AKT1   ABL1    AKT1 -0.010891 -0.014149   -0.025039 -0.009330   \n",
       "2        ABL1_ALK   ABL1     ALK -0.010891 -0.013610   -0.024501 -0.015444   \n",
       "3        ABL1_APC   ABL1     APC -0.010891 -0.010225   -0.021116 -0.011350   \n",
       "4     ABL1_ARID1A   ABL1  ARID1A -0.010891 -0.016038   -0.026929 -0.029612   \n",
       "...           ...    ...     ...       ...       ...         ...       ...   \n",
       "2625    VEGFA_VHL  VEGFA     VHL -0.019953 -0.081441   -0.101394 -0.027529   \n",
       "2626   VEGFA_WEE1  VEGFA    WEE1 -0.019953 -0.024776   -0.044729 -0.045977   \n",
       "2627     VHL_WEE1    VHL    WEE1 -0.081441 -0.024776   -0.106217 -0.045337   \n",
       "2628          NaN    NaN     NaN       NaN       NaN         NaN       NaN   \n",
       "2629          NaN    NaN     NaN       NaN       NaN         NaN       NaN   \n",
       "\n",
       "       sd_HeLa  PP_HeLa  abs_pi_HeLa  ...   fA_293T   fB_293T  fA+fB_293T  \\\n",
       "0     0.036023    0.704     0.038661  ... -0.039056 -0.026479   -0.065535   \n",
       "1     0.037599    0.182     0.009330  ... -0.039056 -0.024441   -0.063497   \n",
       "2     0.020846    0.793     0.015444  ... -0.039056 -0.024261   -0.063316   \n",
       "3     0.019483    0.512     0.011350  ... -0.039056 -0.022621   -0.061677   \n",
       "4     0.032232    0.600     0.029612  ... -0.039056 -0.018836   -0.057892   \n",
       "...        ...      ...          ...  ...       ...       ...         ...   \n",
       "2625  0.019448    0.850     0.027529  ... -0.042977 -0.035299   -0.078276   \n",
       "2626  0.025619    0.931     0.045977  ... -0.042977 -0.047175   -0.090152   \n",
       "2627  0.038023    0.783     0.045337  ... -0.035299 -0.047175   -0.082474   \n",
       "2628       NaN      NaN          NaN  ...       NaN       NaN         NaN   \n",
       "2629       NaN      NaN          NaN  ...       NaN       NaN         NaN   \n",
       "\n",
       "       pi_293T   sd_293T  PP_293T  abs_pi_293T  FDR_left_293T  \\\n",
       "0     0.005756  0.031896    0.207     0.005756       0.800666   \n",
       "1    -0.015256  0.027199    0.423     0.015256       0.557710   \n",
       "2    -0.034626  0.027275    0.783     0.034626       0.407198   \n",
       "3    -0.007403  0.008770    0.669     0.007403       0.639785   \n",
       "4    -0.039848  0.033482    0.755     0.039848       0.392145   \n",
       "...        ...       ...      ...          ...            ...   \n",
       "2625  0.017510  0.018634    0.648     0.017510       0.865491   \n",
       "2626 -0.020847  0.028925    0.526     0.020847       0.499459   \n",
       "2627  0.001916  0.021769    0.106     0.001916       0.769568   \n",
       "2628       NaN       NaN      NaN          NaN            NaN   \n",
       "2629  0.028929       NaN      NaN          NaN            NaN   \n",
       "\n",
       "      FDR_right__293T    293T_Z  \n",
       "0                 1.0  0.198972  \n",
       "1                 1.0 -0.527369  \n",
       "2                 1.0 -1.196939  \n",
       "3                 1.0 -0.255905  \n",
       "4                 1.0 -1.377448  \n",
       "...               ...       ...  \n",
       "2625              1.0  0.605266  \n",
       "2626              1.0 -0.720636  \n",
       "2627              1.0  0.066241  \n",
       "2628              NaN       NaN  \n",
       "2629              NaN       NaN  \n",
       "\n",
       "[2630 rows x 33 columns]"
      ]
     },
     "execution_count": 288,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HELA_A549_293T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "5461db67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Interaction</th>\n",
       "      <th>geneA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>HeLa_Z</th>\n",
       "      <th>A549_Z</th>\n",
       "      <th>293T_Z</th>\n",
       "      <th>Hit_Cell_Line</th>\n",
       "      <th>Interaction_type</th>\n",
       "      <th>Conserved?</th>\n",
       "      <th>Validation?</th>\n",
       "      <th>PriorSL_report?</th>\n",
       "      <th>PMID / ref</th>\n",
       "      <th>geneA.1</th>\n",
       "      <th>geneB.1</th>\n",
       "      <th>geneA.2</th>\n",
       "      <th>geneB.2</th>\n",
       "      <th>geneA.3</th>\n",
       "      <th>geneB.3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA1_WEE1</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>-0.420565</td>\n",
       "      <td>-1.101509</td>\n",
       "      <td>-5.859982</td>\n",
       "      <td>293T</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>25964244</td>\n",
       "      <td>16.81875</td>\n",
       "      <td>19.7427</td>\n",
       "      <td>9.098395</td>\n",
       "      <td>5.24626</td>\n",
       "      <td>14.7990</td>\n",
       "      <td>23.8672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2_CDK9</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>CDK9</td>\n",
       "      <td>-0.581225</td>\n",
       "      <td>1.231318</td>\n",
       "      <td>-5.130425</td>\n",
       "      <td>293T</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.670472</td>\n",
       "      <td>11.60678</td>\n",
       "      <td>0.469309</td>\n",
       "      <td>6.88599</td>\n",
       "      <td>1.6771</td>\n",
       "      <td>21.0598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FNTA_TOP1</td>\n",
       "      <td>FNTA</td>\n",
       "      <td>TOP1</td>\n",
       "      <td>-1.700798</td>\n",
       "      <td>-2.046243</td>\n",
       "      <td>-4.888855</td>\n",
       "      <td>293T</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.97785</td>\n",
       "      <td>3.776725</td>\n",
       "      <td>10.53671</td>\n",
       "      <td>4.040535</td>\n",
       "      <td>35.2184</td>\n",
       "      <td>89.1235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CHEK1_IGF1R</td>\n",
       "      <td>CHEK1</td>\n",
       "      <td>IGF1R</td>\n",
       "      <td>-1.694748</td>\n",
       "      <td>-1.732183</td>\n",
       "      <td>-4.781605</td>\n",
       "      <td>293T</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.18234</td>\n",
       "      <td>3.231415</td>\n",
       "      <td>5.582225</td>\n",
       "      <td>11.213015</td>\n",
       "      <td>22.6362</td>\n",
       "      <td>7.9260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1_VEGFA</td>\n",
       "      <td>CHEK1</td>\n",
       "      <td>VEGFA</td>\n",
       "      <td>-2.323439</td>\n",
       "      <td>-2.475270</td>\n",
       "      <td>-4.599751</td>\n",
       "      <td>293T</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.18234</td>\n",
       "      <td>21.09175</td>\n",
       "      <td>5.582225</td>\n",
       "      <td>9.042085</td>\n",
       "      <td>22.6362</td>\n",
       "      <td>15.8428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>CHEK2_KMT2D</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>-3.060446</td>\n",
       "      <td>0.617330</td>\n",
       "      <td>1.211524</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.545135</td>\n",
       "      <td>0.635619</td>\n",
       "      <td>5.51226</td>\n",
       "      <td>1.142524</td>\n",
       "      <td>40.8714</td>\n",
       "      <td>3.9515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>MAP2K1_VHL</td>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>VHL</td>\n",
       "      <td>-3.586458</td>\n",
       "      <td>-3.138479</td>\n",
       "      <td>1.971959</td>\n",
       "      <td>Hela, A549</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>conserved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>18948595</td>\n",
       "      <td>11.77885</td>\n",
       "      <td>8.94539</td>\n",
       "      <td>5.56903</td>\n",
       "      <td>4.59031</td>\n",
       "      <td>51.4010</td>\n",
       "      <td>27.9101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>CHEK1_SRC</td>\n",
       "      <td>CHEK1</td>\n",
       "      <td>SRC</td>\n",
       "      <td>-3.586796</td>\n",
       "      <td>0.240960</td>\n",
       "      <td>2.935314</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>Synthetic Lethal</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>21148814</td>\n",
       "      <td>9.18234</td>\n",
       "      <td>0.004156</td>\n",
       "      <td>5.582225</td>\n",
       "      <td>32.8265</td>\n",
       "      <td>22.6362</td>\n",
       "      <td>11.3629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>KIT_TP53</td>\n",
       "      <td>KIT</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-0.705059</td>\n",
       "      <td>-0.652426</td>\n",
       "      <td>3.102322</td>\n",
       "      <td>293T</td>\n",
       "      <td>Epistasis</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1.605015</td>\n",
       "      <td>0.000841</td>\n",
       "      <td>12.2452</td>\n",
       "      <td>3.1454</td>\n",
       "      <td>42.3220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>KIT_SMO</td>\n",
       "      <td>KIT</td>\n",
       "      <td>SMO</td>\n",
       "      <td>0.305340</td>\n",
       "      <td>-2.233035</td>\n",
       "      <td>3.561980</td>\n",
       "      <td>293T</td>\n",
       "      <td>Epistasis</td>\n",
       "      <td>private</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000841</td>\n",
       "      <td>2.68528</td>\n",
       "      <td>3.1454</td>\n",
       "      <td>16.5413</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162 rows  18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Interaction   geneA  geneB    HeLa_Z    A549_Z    293T_Z Hit_Cell_Line  \\\n",
       "0     BRCA1_WEE1   BRCA1   WEE1 -0.420565 -1.101509 -5.859982          293T   \n",
       "1     BRCA2_CDK9   BRCA2   CDK9 -0.581225  1.231318 -5.130425          293T   \n",
       "2      FNTA_TOP1    FNTA   TOP1 -1.700798 -2.046243 -4.888855          293T   \n",
       "3    CHEK1_IGF1R   CHEK1  IGF1R -1.694748 -1.732183 -4.781605          293T   \n",
       "4    CHEK1_VEGFA   CHEK1  VEGFA -2.323439 -2.475270 -4.599751          293T   \n",
       "..           ...     ...    ...       ...       ...       ...           ...   \n",
       "157  CHEK2_KMT2D   CHEK2  KMT2D -3.060446  0.617330  1.211524          HeLa   \n",
       "158   MAP2K1_VHL  MAP2K1    VHL -3.586458 -3.138479  1.971959    Hela, A549   \n",
       "159    CHEK1_SRC   CHEK1    SRC -3.586796  0.240960  2.935314          HeLa   \n",
       "160     KIT_TP53     KIT   TP53 -0.705059 -0.652426  3.102322          293T   \n",
       "161      KIT_SMO     KIT    SMO  0.305340 -2.233035  3.561980          293T   \n",
       "\n",
       "     Interaction_type Conserved? Validation? PriorSL_report? PMID / ref  \\\n",
       "0    Synthetic Lethal    private         NaN             yes   25964244   \n",
       "1    Synthetic Lethal    private         NaN             NaN        NaN   \n",
       "2    Synthetic Lethal    private         NaN             NaN        NaN   \n",
       "3    Synthetic Lethal    private         NaN             NaN        NaN   \n",
       "4    Synthetic Lethal    private         NaN             NaN        NaN   \n",
       "..                ...        ...         ...             ...        ...   \n",
       "157  Synthetic Lethal    private         NaN             NaN        NaN   \n",
       "158  Synthetic Lethal  conserved         NaN             yes   18948595   \n",
       "159  Synthetic Lethal    private         NaN             yes   21148814   \n",
       "160         Epistasis    private         NaN             NaN        NaN   \n",
       "161         Epistasis    private         NaN             NaN        NaN   \n",
       "\n",
       "       geneA.1   geneB.1   geneA.2    geneB.2  geneA.3  geneB.3  \n",
       "0     16.81875   19.7427  9.098395    5.24626  14.7990  23.8672  \n",
       "1     0.670472  11.60678  0.469309    6.88599   1.6771  21.0598  \n",
       "2     12.97785  3.776725  10.53671   4.040535  35.2184  89.1235  \n",
       "3      9.18234  3.231415  5.582225  11.213015  22.6362   7.9260  \n",
       "4      9.18234  21.09175  5.582225   9.042085  22.6362  15.8428  \n",
       "..         ...       ...       ...        ...      ...      ...  \n",
       "157  10.545135  0.635619   5.51226   1.142524  40.8714   3.9515  \n",
       "158   11.77885   8.94539   5.56903    4.59031  51.4010  27.9101  \n",
       "159    9.18234  0.004156  5.582225    32.8265  22.6362  11.3629  \n",
       "160          0  1.605015  0.000841    12.2452   3.1454  42.3220  \n",
       "161          0         0  0.000841    2.68528   3.1454  16.5413  \n",
       "\n",
       "[162 rows x 18 columns]"
      ]
     },
     "execution_count": 289,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SL_labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "6044e2fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "42ceaa01",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "293T\n",
      "###############################\n",
      "A549\n",
      "###############################\n",
      "HELA\n",
      "###############################\n"
     ]
    }
   ],
   "source": [
    "for cell_line in ['293T', 'A549', 'HeLa']:\n",
    "    curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "    curr_GI[\"Gene_A\"] = HELA_A549_293T[\"geneA\"]\n",
    "    curr_GI[\"Gene_B\"] = HELA_A549_293T[\"geneB\"]\n",
    "    curr_GI[\"GI_Score\"] = HELA_A549_293T.loc[:, cell_line + \"_Z\"]\n",
    "\n",
    "    curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['shen_data']] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"GI_Cutoff\"] = [-3] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Stat_Score\"] = HELA_A549_293T.loc[:, \"FDR_left_\" + cell_line]\n",
    "    cell_line = cell_line.upper()\n",
    "    curr_GI[\"Cell_Line\"] = [cell_line] * len(curr_GI[\"Gene_A\"])\n",
    "    \n",
    "    curr_GI = curr_GI.fillna(0)\n",
    "    curr_GI = curr_GI[:-2]\n",
    "    \n",
    "    all_GI = pd.concat([all_GI, curr_GI])\n",
    "    \n",
    "    \n",
    "    print(cell_line)\n",
    "    print(\"###############################\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "c875727c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "Only GI cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 70956\n",
      "Number of controls: 1797\n",
      "Number of singles: 1971\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = all_GI.copy(), study_controls = controls['shen_data'], study_conditions = study_conditions['shen_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "7bbf3bc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>cell_line_origin</th>\n",
       "      <th>Condition</th>\n",
       "      <th>study_origin</th>\n",
       "      <th>target_type</th>\n",
       "      <th>T0_counts</th>\n",
       "      <th>T0_replicate_names</th>\n",
       "      <th>TEnd_counts</th>\n",
       "      <th>TEnd_replicate_names</th>\n",
       "      <th>gene_pair</th>\n",
       "      <th>gene_pair_orientation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1_CHR9_133589790</td>\n",
       "      <td>ADA_CHR20_43257696</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ADA</td>\n",
       "      <td>0;3;198;127;146;243;89;238;61;211</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>198.0;127.0</td>\n",
       "      <td>Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...</td>\n",
       "      <td>61.0;211.0</td>\n",
       "      <td>Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>ABL1|ADA</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL1_CHR9_133589790</td>\n",
       "      <td>ADA_CHR20_43257769</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ADA</td>\n",
       "      <td>417;7;616;494;142;242;112;263;42;270</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>616.0;494.0</td>\n",
       "      <td>Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...</td>\n",
       "      <td>42.0;270.0</td>\n",
       "      <td>Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>ABL1|ADA</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1_CHR9_133589790</td>\n",
       "      <td>ALK_CHR2_29541172</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>2;0;659;351;75;94;279;108;342;101</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>659.0;351.0</td>\n",
       "      <td>Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...</td>\n",
       "      <td>342.0;101.0</td>\n",
       "      <td>Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>ABL1|ALK</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL1_CHR9_133589790</td>\n",
       "      <td>ALK_CHR2_30143522</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>169;10;238;563;266;248;204;330;163;191</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>238.0;563.0</td>\n",
       "      <td>Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...</td>\n",
       "      <td>163.0;191.0</td>\n",
       "      <td>Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>ABL1|ALK</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL1_CHR9_133589790</td>\n",
       "      <td>ARID1A_CHR1_27056288</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>2;949;176;342;258;372;472;497;515;254</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>176.0;342.0</td>\n",
       "      <td>Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...</td>\n",
       "      <td>515.0;254.0</td>\n",
       "      <td>Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>ABL1|ARID1A</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74719</th>\n",
       "      <td>WEE1_CHR11_9595845</td>\n",
       "      <td>TSC1_CHR9_135804256</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>157;331;63;416;22;270;77;26;191;66</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>157.0;331.0</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>26.0;66.0</td>\n",
       "      <td>A549_CV4_d28_1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>TSC1|WEE1</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74720</th>\n",
       "      <td>WEE1_CHR11_9595845</td>\n",
       "      <td>TYMS_CHR18_657730</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>958;1209;643;843;721;1239;881;286;2032;790</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>958.0;1209.0</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>286.0;790.0</td>\n",
       "      <td>A549_CV4_d28_1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>TYMS|WEE1</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74721</th>\n",
       "      <td>WEE1_CHR11_9595845</td>\n",
       "      <td>TYMS_CHR18_659665</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>333;640;127;190;118;38;97;33;3;0</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>333.0;640.0</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>33.0;0.0</td>\n",
       "      <td>A549_CV4_d28_1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>TYMS|WEE1</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74722</th>\n",
       "      <td>WEE1_CHR11_9595845</td>\n",
       "      <td>VEGFA_CHR6_43742077</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>VEGFA</td>\n",
       "      <td>1010;1709;544;403;812;450;684;236;506;171</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>1010.0;1709.0</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>236.0;171.0</td>\n",
       "      <td>A549_CV4_d28_1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>VEGFA|WEE1</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74723</th>\n",
       "      <td>WEE1_CHR11_9595845</td>\n",
       "      <td>VHL_CHR3_10183773</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>VHL</td>\n",
       "      <td>620;568;365;374;212;362;173;47;349;96</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>28319113</td>\n",
       "      <td>Dual</td>\n",
       "      <td>620.0;568.0</td>\n",
       "      <td>A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>47.0;96.0</td>\n",
       "      <td>A549_CV4_d28_1_S5_trimmed53_len_filtered_count...</td>\n",
       "      <td>VHL|WEE1</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74724 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Guide 1               Guide 2 Gene 1  Gene 2  \\\n",
       "0      ABL1_CHR9_133589790    ADA_CHR20_43257696   ABL1     ADA   \n",
       "1      ABL1_CHR9_133589790    ADA_CHR20_43257769   ABL1     ADA   \n",
       "2      ABL1_CHR9_133589790     ALK_CHR2_29541172   ABL1     ALK   \n",
       "3      ABL1_CHR9_133589790     ALK_CHR2_30143522   ABL1     ALK   \n",
       "4      ABL1_CHR9_133589790  ARID1A_CHR1_27056288   ABL1  ARID1A   \n",
       "...                    ...                   ...    ...     ...   \n",
       "74719   WEE1_CHR11_9595845   TSC1_CHR9_135804256   WEE1    TSC1   \n",
       "74720   WEE1_CHR11_9595845     TYMS_CHR18_657730   WEE1    TYMS   \n",
       "74721   WEE1_CHR11_9595845     TYMS_CHR18_659665   WEE1    TYMS   \n",
       "74722   WEE1_CHR11_9595845   VEGFA_CHR6_43742077   WEE1   VEGFA   \n",
       "74723   WEE1_CHR11_9595845     VHL_CHR3_10183773   WEE1     VHL   \n",
       "\n",
       "                                 Count Replicates cell_line_origin  \\\n",
       "0               0;3;198;127;146;243;89;238;61;211             HELA   \n",
       "1            417;7;616;494;142;242;112;263;42;270             HELA   \n",
       "2               2;0;659;351;75;94;279;108;342;101             HELA   \n",
       "3          169;10;238;563;266;248;204;330;163;191             HELA   \n",
       "4           2;949;176;342;258;372;472;497;515;254             HELA   \n",
       "...                                           ...              ...   \n",
       "74719          157;331;63;416;22;270;77;26;191;66             A549   \n",
       "74720  958;1209;643;843;721;1239;881;286;2032;790             A549   \n",
       "74721            333;640;127;190;118;38;97;33;3;0             A549   \n",
       "74722   1010;1709;544;403;812;450;684;236;506;171             A549   \n",
       "74723       620;568;365;374;212;362;173;47;349;96             A549   \n",
       "\n",
       "                                               Condition study_origin  \\\n",
       "0      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "1      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "2      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "3      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "4      Hela-CV4-d3-1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "...                                                  ...          ...   \n",
       "74719  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "74720  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "74721  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "74722  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "74723  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...     28319113   \n",
       "\n",
       "      target_type      T0_counts  \\\n",
       "0            Dual    198.0;127.0   \n",
       "1            Dual    616.0;494.0   \n",
       "2            Dual    659.0;351.0   \n",
       "3            Dual    238.0;563.0   \n",
       "4            Dual    176.0;342.0   \n",
       "...           ...            ...   \n",
       "74719        Dual    157.0;331.0   \n",
       "74720        Dual   958.0;1209.0   \n",
       "74721        Dual    333.0;640.0   \n",
       "74722        Dual  1010.0;1709.0   \n",
       "74723        Dual    620.0;568.0   \n",
       "\n",
       "                                      T0_replicate_names  TEnd_counts  \\\n",
       "0      Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...   61.0;211.0   \n",
       "1      Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...   42.0;270.0   \n",
       "2      Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...  342.0;101.0   \n",
       "3      Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...  163.0;191.0   \n",
       "4      Hela-CV4R4R_d3_1_S7_trimmed53_len_filtered_cou...  515.0;254.0   \n",
       "...                                                  ...          ...   \n",
       "74719  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...    26.0;66.0   \n",
       "74720  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  286.0;790.0   \n",
       "74721  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...     33.0;0.0   \n",
       "74722  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...  236.0;171.0   \n",
       "74723  A549_CV4_d3_1_S1_trimmed53_len_filtered_counts...    47.0;96.0   \n",
       "\n",
       "                                    TEnd_replicate_names    gene_pair  \\\n",
       "0      Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...     ABL1|ADA   \n",
       "1      Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...     ABL1|ADA   \n",
       "2      Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...     ABL1|ALK   \n",
       "3      Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...     ABL1|ALK   \n",
       "4      Hela-CV4-d28-1_S5_trimmed53_len_filtered_count...  ABL1|ARID1A   \n",
       "...                                                  ...          ...   \n",
       "74719  A549_CV4_d28_1_S5_trimmed53_len_filtered_count...    TSC1|WEE1   \n",
       "74720  A549_CV4_d28_1_S5_trimmed53_len_filtered_count...    TYMS|WEE1   \n",
       "74721  A549_CV4_d28_1_S5_trimmed53_len_filtered_count...    TYMS|WEE1   \n",
       "74722  A549_CV4_d28_1_S5_trimmed53_len_filtered_count...   VEGFA|WEE1   \n",
       "74723  A549_CV4_d28_1_S5_trimmed53_len_filtered_count...     VHL|WEE1   \n",
       "\n",
       "      gene_pair_orientation  \n",
       "0                       A_B  \n",
       "1                       A_B  \n",
       "2                       A_B  \n",
       "3                       A_B  \n",
       "4                       A_B  \n",
       "...                     ...  \n",
       "74719                   B_A  \n",
       "74720                   B_A  \n",
       "74721                   B_A  \n",
       "74722                   B_A  \n",
       "74723                   B_A  \n",
       "\n",
       "[74724 rows x 15 columns]"
      ]
     },
     "execution_count": 293,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['counts_ref']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "3dcb0eeb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Not SL    7716\n",
       "SL         168\n",
       "Name: SL_or_not, dtype: int64"
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "1116e745",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 266\n",
      "Counts insert: 74724\n",
      "Score insert: 7884\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb7a8b19",
   "metadata": {},
   "source": [
    "## Thompson (A375, MeWo, RPE1) No Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "265b7f4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "thompson_loc = os.path.join(learning_goals_loc_general, \"Thompson\")\n",
    "\n",
    "all_pairs = pd.read_excel(os.path.join(thompson_loc, \"1 - Gene pairs used in the screen and analysis results.xlsx\"), skiprows = 2)\n",
    "statistically_sig = pd.read_excel(os.path.join(thompson_loc, \"5 - Statistically significant gene pairs after filtering.xlsx\"), skiprows = 3)\n",
    "\n",
    "#sgRNAs = pd.read_excel(os.path.join(thompson_loc, \"4 - gRNAs used for library construction.xlsx\"), skiprows = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "122a9122",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pair</th>\n",
       "      <th>GENE PAIR</th>\n",
       "      <th>Gene pair class</th>\n",
       "      <th>A375_D28_rra_fdr_low</th>\n",
       "      <th>A375_D28_t_fdr_low</th>\n",
       "      <th>A375_D14_rra_fdr_low</th>\n",
       "      <th>A375_D14_t_fdr_low</th>\n",
       "      <th>Passes A375 filter?</th>\n",
       "      <th>Mewo_D28_rra_fdr_low</th>\n",
       "      <th>Mewo_D28_t_fdr_low</th>\n",
       "      <th>...</th>\n",
       "      <th>MEWO_D14_Bagel_Gene1</th>\n",
       "      <th>MEWO_D14_Bagel_Gene2</th>\n",
       "      <th>RPE_D14_Bagel_Gene1</th>\n",
       "      <th>RPE_D14_Bagel_Gene2</th>\n",
       "      <th>A375_D28_Bagel_Gene1</th>\n",
       "      <th>A375_D28_Bagel_Gene2</th>\n",
       "      <th>MEWO_D28_Bagel_Gene1</th>\n",
       "      <th>MEWO_D28_Bagel_Gene2</th>\n",
       "      <th>RPE_D28_Bagel_Gene1</th>\n",
       "      <th>RPE_D28_Bagel_Gene2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>AARS2_AARS</td>\n",
       "      <td>Paralogous_gene_pair</td>\n",
       "      <td>0.228786</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>No</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ABCB7_ABCB6</td>\n",
       "      <td>Paralogous_gene_pair</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>No</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ABCF1_ABCF3</td>\n",
       "      <td>Paralogous_gene_pair</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>No</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ABHD12B_ABHD12</td>\n",
       "      <td>Paralogous_gene_pair</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ABL1_EGFR</td>\n",
       "      <td>SynLethDB_gene_pair</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1187</th>\n",
       "      <td>1188</td>\n",
       "      <td>ZSWIM3_CENPF</td>\n",
       "      <td>Bioinformatically_derived_pair</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1188</th>\n",
       "      <td>1189</td>\n",
       "      <td>ZSWIM3_WDHD1</td>\n",
       "      <td>Bioinformatically_derived_pair</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>8.774696e-03</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.019452</td>\n",
       "      <td>No</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>1.180000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1189</th>\n",
       "      <td>1190</td>\n",
       "      <td>ZYX_EXO1</td>\n",
       "      <td>Bioinformatically_derived_pair</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>5.970000e-09</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>0.041627</td>\n",
       "      <td>No</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1190</th>\n",
       "      <td>1191</td>\n",
       "      <td>ZYX_MKI67</td>\n",
       "      <td>Bioinformatically_derived_pair</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1191</th>\n",
       "      <td>1192</td>\n",
       "      <td>ZYX_TYMS</td>\n",
       "      <td>Bioinformatically_derived_pair</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>1.868735e-01</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>7.933596e-01</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1192 rows  42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Pair       GENE PAIR                 Gene pair class  \\\n",
       "0        1      AARS2_AARS            Paralogous_gene_pair   \n",
       "1        2     ABCB7_ABCB6            Paralogous_gene_pair   \n",
       "2        3     ABCF1_ABCF3            Paralogous_gene_pair   \n",
       "3        4  ABHD12B_ABHD12            Paralogous_gene_pair   \n",
       "4        5       ABL1_EGFR             SynLethDB_gene_pair   \n",
       "...    ...             ...                             ...   \n",
       "1187  1188    ZSWIM3_CENPF  Bioinformatically_derived_pair   \n",
       "1188  1189    ZSWIM3_WDHD1  Bioinformatically_derived_pair   \n",
       "1189  1190        ZYX_EXO1  Bioinformatically_derived_pair   \n",
       "1190  1191       ZYX_MKI67  Bioinformatically_derived_pair   \n",
       "1191  1192        ZYX_TYMS  Bioinformatically_derived_pair   \n",
       "\n",
       "      A375_D28_rra_fdr_low  A375_D28_t_fdr_low  A375_D14_rra_fdr_low  \\\n",
       "0                 0.228786        1.000000e+00              0.999996   \n",
       "1                 0.999996        1.000000e+00              0.999996   \n",
       "2                 0.999996        1.000000e+00              0.999996   \n",
       "3                 0.999996        1.000000e+00              0.999996   \n",
       "4                 0.999996        1.000000e+00              0.999996   \n",
       "...                    ...                 ...                   ...   \n",
       "1187              0.999996        1.000000e+00              0.999996   \n",
       "1188              0.000028        8.774696e-03              0.000030   \n",
       "1189              0.000028        5.970000e-09              0.999996   \n",
       "1190              0.999996        1.000000e+00              0.999996   \n",
       "1191              0.000028        1.868735e-01              0.999996   \n",
       "\n",
       "      A375_D14_t_fdr_low Passes A375 filter?  Mewo_D28_rra_fdr_low  \\\n",
       "0               1.000000                  No              0.999996   \n",
       "1               1.000000                  No              0.999996   \n",
       "2               1.000000                  No              0.999996   \n",
       "3               1.000000                 Yes              0.999996   \n",
       "4               1.000000                 Yes              0.999996   \n",
       "...                  ...                 ...                   ...   \n",
       "1187            1.000000                 Yes              0.999996   \n",
       "1188            0.019452                  No              0.000030   \n",
       "1189            0.041627                  No              0.999996   \n",
       "1190            1.000000                 Yes              0.999996   \n",
       "1191            1.000000                 Yes              0.999996   \n",
       "\n",
       "      Mewo_D28_t_fdr_low  ...  MEWO_D14_Bagel_Gene1  MEWO_D14_Bagel_Gene2  \\\n",
       "0           1.000000e+00  ...                   0.0                   1.0   \n",
       "1           1.000000e+00  ...                   0.0                   0.0   \n",
       "2           1.000000e+00  ...                   0.0                   0.0   \n",
       "3           1.000000e+00  ...                   0.0                   0.0   \n",
       "4           1.000000e+00  ...                   0.0                   0.0   \n",
       "...                  ...  ...                   ...                   ...   \n",
       "1187        1.000000e+00  ...                   0.0                   0.0   \n",
       "1188        1.180000e-07  ...                   0.0                   1.0   \n",
       "1189        1.000000e+00  ...                   0.0                   0.0   \n",
       "1190        1.000000e+00  ...                   0.0                   0.0   \n",
       "1191        7.933596e-01  ...                   0.0                   0.0   \n",
       "\n",
       "     RPE_D14_Bagel_Gene1  RPE_D14_Bagel_Gene2  A375_D28_Bagel_Gene1  \\\n",
       "0                    0.0                  1.0                   1.0   \n",
       "1                    0.0                  0.0                   0.0   \n",
       "2                    0.0                  0.0                   1.0   \n",
       "3                    0.0                  0.0                   0.0   \n",
       "4                    0.0                  0.0                   0.0   \n",
       "...                  ...                  ...                   ...   \n",
       "1187                 0.0                  0.0                   0.0   \n",
       "1188                 0.0                  1.0                   0.0   \n",
       "1189                 0.0                  0.0                   1.0   \n",
       "1190                 0.0                  0.0                   1.0   \n",
       "1191                 0.0                  0.0                   1.0   \n",
       "\n",
       "      A375_D28_Bagel_Gene2  MEWO_D28_Bagel_Gene1 MEWO_D28_Bagel_Gene2  \\\n",
       "0                      1.0                   1.0                  1.0   \n",
       "1                      0.0                   0.0                  0.0   \n",
       "2                      0.0                   1.0                  0.0   \n",
       "3                      0.0                   0.0                  0.0   \n",
       "4                      0.0                   0.0                  0.0   \n",
       "...                    ...                   ...                  ...   \n",
       "1187                   0.0                   0.0                  0.0   \n",
       "1188                   1.0                   0.0                  1.0   \n",
       "1189                   0.0                   0.0                  0.0   \n",
       "1190                   0.0                   0.0                  0.0   \n",
       "1191                   0.0                   0.0                  0.0   \n",
       "\n",
       "      RPE_D28_Bagel_Gene1  RPE_D28_Bagel_Gene2  \n",
       "0                     1.0                  1.0  \n",
       "1                     0.0                  0.0  \n",
       "2                     1.0                  0.0  \n",
       "3                     0.0                  0.0  \n",
       "4                     0.0                  0.0  \n",
       "...                   ...                  ...  \n",
       "1187                  0.0                  0.0  \n",
       "1188                  0.0                  1.0  \n",
       "1189                  0.0                  0.0  \n",
       "1190                  0.0                  0.0  \n",
       "1191                  0.0                  0.0  \n",
       "\n",
       "[1192 rows x 42 columns]"
      ]
     },
     "execution_count": 297,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_pairs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "fced23a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A375</th>\n",
       "      <th>Mewo</th>\n",
       "      <th>RPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AP2A2_AP2A1</td>\n",
       "      <td>ARID1A_ARID1B</td>\n",
       "      <td>ALAS1_ALAS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARID4B_ARID4A</td>\n",
       "      <td>EAF1_EAF2</td>\n",
       "      <td>ALKBH4_ATR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASF1A_ASF1B</td>\n",
       "      <td>ARPC5_ARPC5L</td>\n",
       "      <td>ARFGEF1_ARFGEF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCNL2_CCNL1</td>\n",
       "      <td>SRRM3_SRRM2</td>\n",
       "      <td>ARPC1A_ARPC1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CDS1_CDS2</td>\n",
       "      <td>CCNT2_CCNT1</td>\n",
       "      <td>ASF1A_ASF1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CHMP1B_CHMP1A</td>\n",
       "      <td>CHML_CHM</td>\n",
       "      <td>ATP6V1C2_ATP6V1C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CLASP1_CLASP2</td>\n",
       "      <td>UAP1_UAP1L1</td>\n",
       "      <td>CCNI2_CCNI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CNOT8_CNOT7</td>\n",
       "      <td>CPPED1_ASPM</td>\n",
       "      <td>CCNT2_CCNT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CSTF2T_CSTF2</td>\n",
       "      <td>CRK_CRKL</td>\n",
       "      <td>CDK13_CDK12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>DNAJC6_GAK</td>\n",
       "      <td>FAM50B_FAM50A</td>\n",
       "      <td>CDS1_CDS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>DYNC1LI1_DYNC1LI2</td>\n",
       "      <td>DDX39B_DDX39A</td>\n",
       "      <td>CHML_CHM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DYNLT1_DYNLT3</td>\n",
       "      <td>DSN1_HTT</td>\n",
       "      <td>CLASP1_CLASP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EAF1_EAF2</td>\n",
       "      <td>DSN1_SRRM2</td>\n",
       "      <td>CNOT8_CNOT7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>EXOC3L1_EXOC3</td>\n",
       "      <td>SAR1B_SAR1A</td>\n",
       "      <td>CPPED1_ASPM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FAM50B_FAM50A</td>\n",
       "      <td>EPC2_EPC1</td>\n",
       "      <td>CSTF2T_CSTF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>FKBP5_FKBP4</td>\n",
       "      <td>FAH_EXO1</td>\n",
       "      <td>DAGLA_DAGLB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>FNIP2_FNIP1</td>\n",
       "      <td>G3BP2_G3BP1</td>\n",
       "      <td>DCP1A_DCP1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>G3BP2_G3BP1</td>\n",
       "      <td>TTC7A_TTC7B</td>\n",
       "      <td>EGFR_PRKCE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GNB5_ASPM</td>\n",
       "      <td>IDH3B_IDH3G</td>\n",
       "      <td>ELL2_ELL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HOMER3_HNRNPA2B1</td>\n",
       "      <td>GNPDA2_DSN1</td>\n",
       "      <td>FAM50B_FAM50A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>IDE_NRDC</td>\n",
       "      <td>HOMER3_DSN1</td>\n",
       "      <td>FBXW7_FEN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>IDH3B_IDH3G</td>\n",
       "      <td>LARP1B_LARP1</td>\n",
       "      <td>G3BP2_G3BP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>KAT2B_KAT2A</td>\n",
       "      <td>INPP5K_ASPM</td>\n",
       "      <td>GCFC2_PAXBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>LARP1B_LARP1</td>\n",
       "      <td>KMT2A_KMT2B</td>\n",
       "      <td>GRB10_CCNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PARP2_RAD21</td>\n",
       "      <td>MLKL_EXO1</td>\n",
       "      <td>HOMER3_HNRNPA2B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PTPRS_ASPM</td>\n",
       "      <td>ASF1A_ASF1B</td>\n",
       "      <td>IDE_NRDC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>RANBP17_XPO7</td>\n",
       "      <td>RAD54L_NDUFA9</td>\n",
       "      <td>LARP1B_LARP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>RANBP6_IPO5</td>\n",
       "      <td>RAD54L_PFDN5</td>\n",
       "      <td>LIMK2_LIMK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SAR1B_SAR1A</td>\n",
       "      <td>RIN2_EXO1</td>\n",
       "      <td>MET_VHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SCO1_SCO2</td>\n",
       "      <td>RNF135_ASPM</td>\n",
       "      <td>NAV1_CCNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SEC23A_SEC23B</td>\n",
       "      <td>RPP25L_RPP25</td>\n",
       "      <td>NCK2_NCK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SEC24B_SEC24A</td>\n",
       "      <td>RRAGC_RRAGD</td>\n",
       "      <td>PARP1_RAD21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>SEPHS1_SEPHS2</td>\n",
       "      <td>RRAS_ASPM</td>\n",
       "      <td>PARP1_STAG3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>SLC25A37_SLC25A28</td>\n",
       "      <td>CLASP1_CLASP2</td>\n",
       "      <td>PARP2_RAD21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SMARCA2_SMARCA4</td>\n",
       "      <td>SLC25A37_SLC25A28</td>\n",
       "      <td>PARP2_STAG3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SNRPB2_SNRPA</td>\n",
       "      <td>CNOT8_CNOT7</td>\n",
       "      <td>PEX5L_PEX5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>SRRM3_SRRM2</td>\n",
       "      <td>TMEM101_DSN1</td>\n",
       "      <td>PLEKHO1_CCNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TASP1_ASRGL1</td>\n",
       "      <td>UPF3A_UPF3B</td>\n",
       "      <td>PPA2_PPA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>TNFRSF1A_HNRNPA2B1</td>\n",
       "      <td>CSTF2T_CSTF2</td>\n",
       "      <td>PSENEN_AC002398.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>TTC7A_TTC7B</td>\n",
       "      <td>CHMP1B_CHMP1A</td>\n",
       "      <td>RANBP6_IPO5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>UAP1_UAP1L1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RB1_SKP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>XRCC1_PARG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RHPN1_RHPN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>YIF1A_YIF1B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAR1B_SAR1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ZC3H6_ZC3H4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SMARCA2_SMARCA4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SMUG1_SMG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STXBP6_EXOC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAF5L_TAF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TLK1_TLK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TMEM185B_TMEM185A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNFRSF1A_HNRNPA2B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTC7A_TTC7B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>USP38_USP35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>USP42_USP36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>XYLT1_XYLT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YIF1A_YIF1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZC3H6_ZC3H4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZNF175_EXO1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  A375               Mewo                 RPE\n",
       "0          AP2A2_AP2A1      ARID1A_ARID1B         ALAS1_ALAS2\n",
       "1        ARID4B_ARID4A          EAF1_EAF2          ALKBH4_ATR\n",
       "2          ASF1A_ASF1B       ARPC5_ARPC5L     ARFGEF1_ARFGEF2\n",
       "3          CCNL2_CCNL1        SRRM3_SRRM2       ARPC1A_ARPC1B\n",
       "4            CDS1_CDS2        CCNT2_CCNT1         ASF1A_ASF1B\n",
       "5        CHMP1B_CHMP1A           CHML_CHM   ATP6V1C2_ATP6V1C1\n",
       "6        CLASP1_CLASP2        UAP1_UAP1L1          CCNI2_CCNI\n",
       "7          CNOT8_CNOT7        CPPED1_ASPM         CCNT2_CCNT1\n",
       "8         CSTF2T_CSTF2           CRK_CRKL         CDK13_CDK12\n",
       "9           DNAJC6_GAK      FAM50B_FAM50A           CDS1_CDS2\n",
       "10   DYNC1LI1_DYNC1LI2      DDX39B_DDX39A            CHML_CHM\n",
       "11       DYNLT1_DYNLT3           DSN1_HTT       CLASP1_CLASP2\n",
       "12           EAF1_EAF2         DSN1_SRRM2         CNOT8_CNOT7\n",
       "13       EXOC3L1_EXOC3        SAR1B_SAR1A         CPPED1_ASPM\n",
       "14       FAM50B_FAM50A          EPC2_EPC1        CSTF2T_CSTF2\n",
       "15         FKBP5_FKBP4           FAH_EXO1         DAGLA_DAGLB\n",
       "16         FNIP2_FNIP1        G3BP2_G3BP1         DCP1A_DCP1B\n",
       "17         G3BP2_G3BP1        TTC7A_TTC7B          EGFR_PRKCE\n",
       "18           GNB5_ASPM        IDH3B_IDH3G            ELL2_ELL\n",
       "19    HOMER3_HNRNPA2B1        GNPDA2_DSN1       FAM50B_FAM50A\n",
       "20            IDE_NRDC        HOMER3_DSN1          FBXW7_FEN1\n",
       "21         IDH3B_IDH3G       LARP1B_LARP1         G3BP2_G3BP1\n",
       "22         KAT2B_KAT2A        INPP5K_ASPM        GCFC2_PAXBP1\n",
       "23        LARP1B_LARP1        KMT2A_KMT2B         GRB10_CCNB1\n",
       "24         PARP2_RAD21          MLKL_EXO1    HOMER3_HNRNPA2B1\n",
       "25          PTPRS_ASPM        ASF1A_ASF1B            IDE_NRDC\n",
       "26        RANBP17_XPO7      RAD54L_NDUFA9        LARP1B_LARP1\n",
       "27         RANBP6_IPO5       RAD54L_PFDN5         LIMK2_LIMK1\n",
       "28         SAR1B_SAR1A          RIN2_EXO1             MET_VHL\n",
       "29           SCO1_SCO2        RNF135_ASPM          NAV1_CCNB1\n",
       "30       SEC23A_SEC23B       RPP25L_RPP25           NCK2_NCK1\n",
       "31       SEC24B_SEC24A        RRAGC_RRAGD         PARP1_RAD21\n",
       "32       SEPHS1_SEPHS2          RRAS_ASPM         PARP1_STAG3\n",
       "33   SLC25A37_SLC25A28      CLASP1_CLASP2         PARP2_RAD21\n",
       "34     SMARCA2_SMARCA4  SLC25A37_SLC25A28         PARP2_STAG3\n",
       "35        SNRPB2_SNRPA        CNOT8_CNOT7          PEX5L_PEX5\n",
       "36         SRRM3_SRRM2       TMEM101_DSN1       PLEKHO1_CCNB1\n",
       "37        TASP1_ASRGL1        UPF3A_UPF3B           PPA2_PPA1\n",
       "38  TNFRSF1A_HNRNPA2B1       CSTF2T_CSTF2   PSENEN_AC002398.9\n",
       "39         TTC7A_TTC7B      CHMP1B_CHMP1A         RANBP6_IPO5\n",
       "40         UAP1_UAP1L1                NaN            RB1_SKP2\n",
       "41          XRCC1_PARG                NaN         RHPN1_RHPN2\n",
       "42         YIF1A_YIF1B                NaN         SAR1B_SAR1A\n",
       "43         ZC3H6_ZC3H4                NaN     SMARCA2_SMARCA4\n",
       "44                 NaN                NaN          SMUG1_SMG1\n",
       "45                 NaN                NaN        STXBP6_EXOC1\n",
       "46                 NaN                NaN          TAF5L_TAF5\n",
       "47                 NaN                NaN           TLK1_TLK2\n",
       "48                 NaN                NaN   TMEM185B_TMEM185A\n",
       "49                 NaN                NaN  TNFRSF1A_HNRNPA2B1\n",
       "50                 NaN                NaN         TTC7A_TTC7B\n",
       "51                 NaN                NaN         USP38_USP35\n",
       "52                 NaN                NaN         USP42_USP36\n",
       "53                 NaN                NaN         XYLT1_XYLT2\n",
       "54                 NaN                NaN         YIF1A_YIF1B\n",
       "55                 NaN                NaN         ZC3H6_ZC3H4\n",
       "56                 NaN                NaN         ZNF175_EXO1"
      ]
     },
     "execution_count": 298,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "statistically_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "id": "d69c17bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_A_list, gene_B_list = [], []\n",
    "for i in all_pairs['GENE PAIR']:\n",
    "    split_ab = i.split('_')\n",
    "    gene_A_list.append(split_ab[0])\n",
    "    gene_B_list.append(split_ab[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "55af543a",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "for cell_line in [\"A375\", \"Mewo\", \"RPE1\"]:\n",
    "    curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"GI_Score\"])\n",
    "    curr_GI[\"Gene_A\"] = gene_A_list\n",
    "    curr_GI[\"Gene_B\"] = gene_B_list\n",
    "    #curr_GI[\"GI_Score\"] = None#all_pairs[cell_line + \"_D14_t_fdr_low\"]\n",
    "    \n",
    "    curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['thompson_data']] * len(curr_GI[\"Gene_A\"])\n",
    "    curr_GI[\"Cell_Line\"] = [cell_line] * len(curr_GI[\"Gene_A\"])\n",
    "    \n",
    "    all_GI = pd.concat([all_GI, curr_GI])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "525e741c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>AARS</td>\n",
       "      <td>33637726</td>\n",
       "      <td>A375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCB7</td>\n",
       "      <td>ABCB6</td>\n",
       "      <td>33637726</td>\n",
       "      <td>A375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCF1</td>\n",
       "      <td>ABCF3</td>\n",
       "      <td>33637726</td>\n",
       "      <td>A375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABHD12B</td>\n",
       "      <td>ABHD12</td>\n",
       "      <td>33637726</td>\n",
       "      <td>A375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>33637726</td>\n",
       "      <td>A375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1187</th>\n",
       "      <td>ZSWIM3</td>\n",
       "      <td>CENPF</td>\n",
       "      <td>33637726</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1188</th>\n",
       "      <td>ZSWIM3</td>\n",
       "      <td>WDHD1</td>\n",
       "      <td>33637726</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1189</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>EXO1</td>\n",
       "      <td>33637726</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1190</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>MKI67</td>\n",
       "      <td>33637726</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1191</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>33637726</td>\n",
       "      <td>RPE1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3576 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene_A  Gene_B Study_Source Cell_Line GI_Score GI_Cutoff Stat_Score  \\\n",
       "0       AARS2    AARS     33637726      A375      NaN       NaN        NaN   \n",
       "1       ABCB7   ABCB6     33637726      A375      NaN       NaN        NaN   \n",
       "2       ABCF1   ABCF3     33637726      A375      NaN       NaN        NaN   \n",
       "3     ABHD12B  ABHD12     33637726      A375      NaN       NaN        NaN   \n",
       "4        ABL1    EGFR     33637726      A375      NaN       NaN        NaN   \n",
       "...       ...     ...          ...       ...      ...       ...        ...   \n",
       "1187   ZSWIM3   CENPF     33637726      RPE1      NaN       NaN        NaN   \n",
       "1188   ZSWIM3   WDHD1     33637726      RPE1      NaN       NaN        NaN   \n",
       "1189      ZYX    EXO1     33637726      RPE1      NaN       NaN        NaN   \n",
       "1190      ZYX   MKI67     33637726      RPE1      NaN       NaN        NaN   \n",
       "1191      ZYX    TYMS     33637726      RPE1      NaN       NaN        NaN   \n",
       "\n",
       "     Stat_Cutoff  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4            NaN  \n",
       "...          ...  \n",
       "1187         NaN  \n",
       "1188         NaN  \n",
       "1189         NaN  \n",
       "1190         NaN  \n",
       "1191         NaN  \n",
       "\n",
       "[3576 rows x 8 columns]"
      ]
     },
     "execution_count": 301,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "5968fdd9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "No scores/stats cutoffs are available, possibly generated. Setting all to be NOT SL\n",
      "Counts reference...\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = None, counts_ref = None, score_ref = all_GI.copy(), study_controls = None, study_conditions = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "119ad9d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = db_inserts['score_ref'].copy()\n",
    "all_GI['SL_or_not'] = 'Not SL'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "e2c9f4e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp = statistically_sig['A375'].dropna()\n",
    "temp = ['_'.join(sorted(i.split('_'))) for i in temp]\n",
    "all_GI.loc[all_GI['gene_pair'].isin(temp) & (all_GI['cell_line_origin'] == 'A375'), 'SL_or_not'] = 'SL'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "06dac8ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp = statistically_sig['Mewo'].dropna()\n",
    "temp = ['_'.join(sorted(i.split('_'))) for i in temp]\n",
    "all_GI.loc[all_GI['gene_pair'].isin(temp) & (all_GI['cell_line_origin'] == 'MEWO'), 'SL_or_not'] = 'SL'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "id": "5e5e7a1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp = statistically_sig['RPE'].dropna()\n",
    "temp = ['_'.join(sorted(i.split('_'))) for i in temp]\n",
    "all_GI.loc[all_GI['gene_pair'].isin(temp) & (all_GI['cell_line_origin'] == 'RPE1'), 'SL_or_not'] = 'SL'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "c9bee1f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "db_inserts['score_ref'] = all_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "0988beb7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Not SL    3435\n",
       "SL         141\n",
       "Name: SL_or_not, dtype: int64"
      ]
     },
     "execution_count": 308,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "f6d117ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "No counts and sequences together\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Score insert: 3576\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24261942",
   "metadata": {},
   "source": [
    "## Wong Data (OVCAR8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "a47f80bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/apps/python/3.9-2022.05/lib/python3.9/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "wong_loc = os.path.join(learning_goals_loc_general, \"Wong\")\n",
    "\n",
    "all_sgRNA = pd.read_excel(os.path.join(wong_loc, \"barcode_counts.xlsx\"), skiprows = 2)\n",
    "sgRNA_ref = pd.read_excel(os.path.join(wong_loc, \"sequences_ref.xlsx\"))\n",
    "#sgRNA_ref = pd.read_excel(os.path.join(wong_loc, \"pnas.1517883113.sd03.xlsx\"), skiprows = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "158904f9",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "39d649f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide_ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GFP_1</td>\n",
       "      <td>GGGCGAGGAGCTGTTCACCG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RFP_1</td>\n",
       "      <td>CACCCAGACCATGAAGATCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RFP_2</td>\n",
       "      <td>CCACTTCAAGTGCACATCCG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NF1_1</td>\n",
       "      <td>GTTGTGCTCAGTACTGACTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NF1_4</td>\n",
       "      <td>TTTCAGCTTCCAATAAAAAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>ING4_2</td>\n",
       "      <td>CCTAGAAGGCCGGACTCAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>ING4_3</td>\n",
       "      <td>GGCACTACTCATATACTCAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>ING5_1</td>\n",
       "      <td>GATCTGCTTCAAAGCGCGCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>ING5_2</td>\n",
       "      <td>CTTCCAGCTGATGCGAGAGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>GAAGTTCCTCTGAAGTTCGC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>160 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Guide_ID              Sequence\n",
       "0      GFP_1  GGGCGAGGAGCTGTTCACCG\n",
       "1      RFP_1  CACCCAGACCATGAAGATCA\n",
       "2      RFP_2  CCACTTCAAGTGCACATCCG\n",
       "3      NF1_1  GTTGTGCTCAGTACTGACTT\n",
       "4      NF1_4  TTTCAGCTTCCAATAAAAAC\n",
       "..       ...                   ...\n",
       "155   ING4_2  CCTAGAAGGCCGGACTCAAA\n",
       "156   ING4_3  GGCACTACTCATATACTCAG\n",
       "157   ING5_1  GATCTGCTTCAAAGCGCGCC\n",
       "158   ING5_2  CTTCCAGCTGATGCGAGAGC\n",
       "159   ING5_3  GAAGTTCCTCTGAAGTTCGC\n",
       "\n",
       "[160 rows x 2 columns]"
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "id": "bdf03af5",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sgRNA_ref.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "id": "95294a4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['Target'] = [i.split('_')[0] for i in sequence_ref['Guide_ID']]\n",
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "ec87153b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GFP_1</td>\n",
       "      <td>GGGCGAGGAGCTGTTCACCG</td>\n",
       "      <td>GFP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RFP_1</td>\n",
       "      <td>CACCCAGACCATGAAGATCA</td>\n",
       "      <td>RFP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RFP_2</td>\n",
       "      <td>CCACTTCAAGTGCACATCCG</td>\n",
       "      <td>RFP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NF1_1</td>\n",
       "      <td>GTTGTGCTCAGTACTGACTT</td>\n",
       "      <td>NF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NF1_4</td>\n",
       "      <td>TTTCAGCTTCCAATAAAAAC</td>\n",
       "      <td>NF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>ING4_2</td>\n",
       "      <td>CCTAGAAGGCCGGACTCAAA</td>\n",
       "      <td>ING4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>ING4_3</td>\n",
       "      <td>GGCACTACTCATATACTCAG</td>\n",
       "      <td>ING4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>ING5_1</td>\n",
       "      <td>GATCTGCTTCAAAGCGCGCC</td>\n",
       "      <td>ING5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>ING5_2</td>\n",
       "      <td>CTTCCAGCTGATGCGAGAGC</td>\n",
       "      <td>ING5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>GAAGTTCCTCTGAAGTTCGC</td>\n",
       "      <td>ING5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>160 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0              GFP_1  GGGCGAGGAGCTGTTCACCG               GFP\n",
       "1              RFP_1  CACCCAGACCATGAAGATCA               RFP\n",
       "2              RFP_2  CCACTTCAAGTGCACATCCG               RFP\n",
       "3              NF1_1  GTTGTGCTCAGTACTGACTT               NF1\n",
       "4              NF1_4  TTTCAGCTTCCAATAAAAAC               NF1\n",
       "..               ...                   ...               ...\n",
       "155           ING4_2  CCTAGAAGGCCGGACTCAAA              ING4\n",
       "156           ING4_3  GGCACTACTCATATACTCAG              ING4\n",
       "157           ING5_1  GATCTGCTTCAAAGCGCGCC              ING5\n",
       "158           ING5_2  CTTCCAGCTGATGCGAGAGC              ING5\n",
       "159           ING5_3  GAAGTTCCTCTGAAGTTCGC              ING5\n",
       "\n",
       "[160 rows x 3 columns]"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3905ae5e",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "465faabe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Key</th>\n",
       "      <th>2-wise gRNA combination</th>\n",
       "      <th>sgRNA-A</th>\n",
       "      <th>sgRNA-B</th>\n",
       "      <th>day5 (Replicate 1)</th>\n",
       "      <th>day15 (Replicate 1)</th>\n",
       "      <th>day20 (Replicate 1)</th>\n",
       "      <th>day5 (Replicate 2)</th>\n",
       "      <th>day15 (Replicate 2)</th>\n",
       "      <th>day20 (Replicate 2)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1,1</td>\n",
       "      <td>dummyguide_1 + dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>1987.0</td>\n",
       "      <td>1618.0</td>\n",
       "      <td>1919.0</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>987.0</td>\n",
       "      <td>1485.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1,2</td>\n",
       "      <td>dummyguide_1 + dummyguide_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_2</td>\n",
       "      <td>757.0</td>\n",
       "      <td>868.0</td>\n",
       "      <td>679.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>465.0</td>\n",
       "      <td>531.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1,3</td>\n",
       "      <td>dummyguide_1 + dummyguide_3</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_3</td>\n",
       "      <td>1657.0</td>\n",
       "      <td>1901.0</td>\n",
       "      <td>1558.0</td>\n",
       "      <td>1166.0</td>\n",
       "      <td>1091.0</td>\n",
       "      <td>1243.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1,4</td>\n",
       "      <td>dummyguide_1 + DNMT1_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>DNMT1_1</td>\n",
       "      <td>1178.0</td>\n",
       "      <td>1555.0</td>\n",
       "      <td>900.0</td>\n",
       "      <td>1093.0</td>\n",
       "      <td>972.0</td>\n",
       "      <td>638.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1,5</td>\n",
       "      <td>dummyguide_1 + DNMT1_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>DNMT1_2</td>\n",
       "      <td>2103.0</td>\n",
       "      <td>2150.0</td>\n",
       "      <td>1291.0</td>\n",
       "      <td>1775.0</td>\n",
       "      <td>1387.0</td>\n",
       "      <td>2080.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23404</th>\n",
       "      <td>153,149</td>\n",
       "      <td>ING5_3 + ING4_2</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING4_2</td>\n",
       "      <td>1056.0</td>\n",
       "      <td>1430.0</td>\n",
       "      <td>1560.0</td>\n",
       "      <td>1264.0</td>\n",
       "      <td>826.0</td>\n",
       "      <td>1195.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23405</th>\n",
       "      <td>153,150</td>\n",
       "      <td>ING5_3 + ING4_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING4_3</td>\n",
       "      <td>1344.0</td>\n",
       "      <td>1051.0</td>\n",
       "      <td>2309.0</td>\n",
       "      <td>1552.0</td>\n",
       "      <td>1063.0</td>\n",
       "      <td>987.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23406</th>\n",
       "      <td>153,151</td>\n",
       "      <td>ING5_3 + ING5_1</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1129.0</td>\n",
       "      <td>1222.0</td>\n",
       "      <td>942.0</td>\n",
       "      <td>745.0</td>\n",
       "      <td>896.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23407</th>\n",
       "      <td>153,152</td>\n",
       "      <td>ING5_3 + ING5_2</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>922.0</td>\n",
       "      <td>715.0</td>\n",
       "      <td>1047.0</td>\n",
       "      <td>783.0</td>\n",
       "      <td>452.0</td>\n",
       "      <td>886.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23408</th>\n",
       "      <td>153,153</td>\n",
       "      <td>ING5_3 + ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>387.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>414.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>395.0</td>\n",
       "      <td>573.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23409 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Key      2-wise gRNA combination       sgRNA-A       sgRNA-B  \\\n",
       "0          1,1  dummyguide_1 + dummyguide_1  dummyguide_1  dummyguide_1   \n",
       "1          1,2  dummyguide_1 + dummyguide_2  dummyguide_1  dummyguide_2   \n",
       "2          1,3  dummyguide_1 + dummyguide_3  dummyguide_1  dummyguide_3   \n",
       "3          1,4       dummyguide_1 + DNMT1_1  dummyguide_1       DNMT1_1   \n",
       "4          1,5       dummyguide_1 + DNMT1_2  dummyguide_1       DNMT1_2   \n",
       "...        ...                          ...           ...           ...   \n",
       "23404  153,149              ING5_3 + ING4_2        ING5_3        ING4_2   \n",
       "23405  153,150              ING5_3 + ING4_3        ING5_3        ING4_3   \n",
       "23406  153,151              ING5_3 + ING5_1        ING5_3        ING5_1   \n",
       "23407  153,152              ING5_3 + ING5_2        ING5_3        ING5_2   \n",
       "23408  153,153              ING5_3 + ING5_3        ING5_3        ING5_3   \n",
       "\n",
       "       day5 (Replicate 1)  day15 (Replicate 1)  day20 (Replicate 1)  \\\n",
       "0                  1987.0               1618.0               1919.0   \n",
       "1                   757.0                868.0                679.0   \n",
       "2                  1657.0               1901.0               1558.0   \n",
       "3                  1178.0               1555.0                900.0   \n",
       "4                  2103.0               2150.0               1291.0   \n",
       "...                   ...                  ...                  ...   \n",
       "23404              1056.0               1430.0               1560.0   \n",
       "23405              1344.0               1051.0               2309.0   \n",
       "23406              1120.0               1129.0               1222.0   \n",
       "23407               922.0                715.0               1047.0   \n",
       "23408               387.0                539.0                414.0   \n",
       "\n",
       "       day5 (Replicate 2)  day15 (Replicate 2)  day20 (Replicate 2)  \n",
       "0                  1299.0                987.0               1485.0  \n",
       "1                   511.0                465.0                531.0  \n",
       "2                  1166.0               1091.0               1243.0  \n",
       "3                  1093.0                972.0                638.0  \n",
       "4                  1775.0               1387.0               2080.0  \n",
       "...                   ...                  ...                  ...  \n",
       "23404              1264.0                826.0               1195.0  \n",
       "23405              1552.0               1063.0                987.0  \n",
       "23406               942.0                745.0                896.0  \n",
       "23407               783.0                452.0                886.0  \n",
       "23408               511.0                395.0                573.0  \n",
       "\n",
       "[23409 rows x 10 columns]"
      ]
     },
     "execution_count": 315,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_sgRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "2b97ee50",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "id": "71156d92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['day5 (Replicate 1)', 'day15 (Replicate 1)', 'day20 (Replicate 1)',\n",
       "       'day5 (Replicate 2)', 'day15 (Replicate 2)', 'day20 (Replicate 2)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 317,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_sgRNA.columns[4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "a1869a57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'day5 (Replicate 1);day15 (Replicate 1);day20 (Replicate 1);day5 (Replicate 2);day15 (Replicate 2);day20 (Replicate 2)'"
      ]
     },
     "execution_count": 318,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "';'.join(all_sgRNA.columns[4:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "3e975796",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_sgRNA['Condition'] = ';'.join(all_sgRNA.columns[4:])\n",
    "\n",
    "all_sgRNA['Count Replicates'] = all_sgRNA[['day5 (Replicate 1)', 'day15 (Replicate 1)', 'day20 (Replicate 1)',\n",
    "       'day5 (Replicate 2)', 'day15 (Replicate 2)', 'day20 (Replicate 2)']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "id": "9f913b4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "type_list = [\"Dual\"] * all_sgRNA.shape[0]\n",
    "for i in range(len(all_sgRNA['2-wise gRNA combination'])):\n",
    "    if \"dummyguide\" in all_sgRNA.loc[i, '2-wise gRNA combination']:\n",
    "        type_list[i] = \"Single\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "6455ce00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Key</th>\n",
       "      <th>2-wise gRNA combination</th>\n",
       "      <th>sgRNA-A</th>\n",
       "      <th>sgRNA-B</th>\n",
       "      <th>day5 (Replicate 1)</th>\n",
       "      <th>day15 (Replicate 1)</th>\n",
       "      <th>day20 (Replicate 1)</th>\n",
       "      <th>day5 (Replicate 2)</th>\n",
       "      <th>day15 (Replicate 2)</th>\n",
       "      <th>day20 (Replicate 2)</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Count Replicates</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1,1</td>\n",
       "      <td>dummyguide_1 + dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>1987.0</td>\n",
       "      <td>1618.0</td>\n",
       "      <td>1919.0</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>987.0</td>\n",
       "      <td>1485.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>1987.0;1618.0;1919.0;1299.0;987.0;1485.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1,2</td>\n",
       "      <td>dummyguide_1 + dummyguide_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_2</td>\n",
       "      <td>757.0</td>\n",
       "      <td>868.0</td>\n",
       "      <td>679.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>465.0</td>\n",
       "      <td>531.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>757.0;868.0;679.0;510.99999999999994;465.0;531.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1,3</td>\n",
       "      <td>dummyguide_1 + dummyguide_3</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_3</td>\n",
       "      <td>1657.0</td>\n",
       "      <td>1901.0</td>\n",
       "      <td>1558.0</td>\n",
       "      <td>1166.0</td>\n",
       "      <td>1091.0</td>\n",
       "      <td>1243.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>1657.0;1901.0;1558.0;1165.9999999999998;1091.0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1,4</td>\n",
       "      <td>dummyguide_1 + DNMT1_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>DNMT1_1</td>\n",
       "      <td>1178.0</td>\n",
       "      <td>1555.0</td>\n",
       "      <td>900.0</td>\n",
       "      <td>1093.0</td>\n",
       "      <td>972.0</td>\n",
       "      <td>638.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>1178.0;1555.0;900.0;1093.0;972.0000000000001;6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1,5</td>\n",
       "      <td>dummyguide_1 + DNMT1_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>DNMT1_2</td>\n",
       "      <td>2103.0</td>\n",
       "      <td>2150.0</td>\n",
       "      <td>1291.0</td>\n",
       "      <td>1775.0</td>\n",
       "      <td>1387.0</td>\n",
       "      <td>2080.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>2103.0;2150.0;1291.0;1775.0000000000002;1387.0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23404</th>\n",
       "      <td>153,149</td>\n",
       "      <td>ING5_3 + ING4_2</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING4_2</td>\n",
       "      <td>1056.0</td>\n",
       "      <td>1430.0</td>\n",
       "      <td>1560.0</td>\n",
       "      <td>1264.0</td>\n",
       "      <td>826.0</td>\n",
       "      <td>1195.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>1056.0;1430.0;1560.0;1264.0;826.0;1195.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23405</th>\n",
       "      <td>153,150</td>\n",
       "      <td>ING5_3 + ING4_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING4_3</td>\n",
       "      <td>1344.0</td>\n",
       "      <td>1051.0</td>\n",
       "      <td>2309.0</td>\n",
       "      <td>1552.0</td>\n",
       "      <td>1063.0</td>\n",
       "      <td>987.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>1344.0;1051.0;2309.0;1552.0;1063.0;987.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23406</th>\n",
       "      <td>153,151</td>\n",
       "      <td>ING5_3 + ING5_1</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1129.0</td>\n",
       "      <td>1222.0</td>\n",
       "      <td>942.0</td>\n",
       "      <td>745.0</td>\n",
       "      <td>896.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>1120.0;1129.0;1222.0;942.0;745.0;896.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23407</th>\n",
       "      <td>153,152</td>\n",
       "      <td>ING5_3 + ING5_2</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>922.0</td>\n",
       "      <td>715.0</td>\n",
       "      <td>1047.0</td>\n",
       "      <td>783.0</td>\n",
       "      <td>452.0</td>\n",
       "      <td>886.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>921.9999999999999;715.0;1047.0;783.0;452.0;886.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23408</th>\n",
       "      <td>153,153</td>\n",
       "      <td>ING5_3 + ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>387.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>414.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>395.0</td>\n",
       "      <td>573.0</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>387.0;539.0;414.0;510.99999999999994;395.0;573.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23409 rows  12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Key      2-wise gRNA combination       sgRNA-A       sgRNA-B  \\\n",
       "0          1,1  dummyguide_1 + dummyguide_1  dummyguide_1  dummyguide_1   \n",
       "1          1,2  dummyguide_1 + dummyguide_2  dummyguide_1  dummyguide_2   \n",
       "2          1,3  dummyguide_1 + dummyguide_3  dummyguide_1  dummyguide_3   \n",
       "3          1,4       dummyguide_1 + DNMT1_1  dummyguide_1       DNMT1_1   \n",
       "4          1,5       dummyguide_1 + DNMT1_2  dummyguide_1       DNMT1_2   \n",
       "...        ...                          ...           ...           ...   \n",
       "23404  153,149              ING5_3 + ING4_2        ING5_3        ING4_2   \n",
       "23405  153,150              ING5_3 + ING4_3        ING5_3        ING4_3   \n",
       "23406  153,151              ING5_3 + ING5_1        ING5_3        ING5_1   \n",
       "23407  153,152              ING5_3 + ING5_2        ING5_3        ING5_2   \n",
       "23408  153,153              ING5_3 + ING5_3        ING5_3        ING5_3   \n",
       "\n",
       "       day5 (Replicate 1)  day15 (Replicate 1)  day20 (Replicate 1)  \\\n",
       "0                  1987.0               1618.0               1919.0   \n",
       "1                   757.0                868.0                679.0   \n",
       "2                  1657.0               1901.0               1558.0   \n",
       "3                  1178.0               1555.0                900.0   \n",
       "4                  2103.0               2150.0               1291.0   \n",
       "...                   ...                  ...                  ...   \n",
       "23404              1056.0               1430.0               1560.0   \n",
       "23405              1344.0               1051.0               2309.0   \n",
       "23406              1120.0               1129.0               1222.0   \n",
       "23407               922.0                715.0               1047.0   \n",
       "23408               387.0                539.0                414.0   \n",
       "\n",
       "       day5 (Replicate 2)  day15 (Replicate 2)  day20 (Replicate 2)  \\\n",
       "0                  1299.0                987.0               1485.0   \n",
       "1                   511.0                465.0                531.0   \n",
       "2                  1166.0               1091.0               1243.0   \n",
       "3                  1093.0                972.0                638.0   \n",
       "4                  1775.0               1387.0               2080.0   \n",
       "...                   ...                  ...                  ...   \n",
       "23404              1264.0                826.0               1195.0   \n",
       "23405              1552.0               1063.0                987.0   \n",
       "23406               942.0                745.0                896.0   \n",
       "23407               783.0                452.0                886.0   \n",
       "23408               511.0                395.0                573.0   \n",
       "\n",
       "                                               Condition  \\\n",
       "0      day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "1      day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "2      day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "3      day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "4      day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "...                                                  ...   \n",
       "23404  day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "23405  day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "23406  day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "23407  day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "23408  day5 (Replicate 1);day15 (Replicate 1);day20 (...   \n",
       "\n",
       "                                        Count Replicates  \n",
       "0               1987.0;1618.0;1919.0;1299.0;987.0;1485.0  \n",
       "1       757.0;868.0;679.0;510.99999999999994;465.0;531.0  \n",
       "2      1657.0;1901.0;1558.0;1165.9999999999998;1091.0...  \n",
       "3      1178.0;1555.0;900.0;1093.0;972.0000000000001;6...  \n",
       "4      2103.0;2150.0;1291.0;1775.0000000000002;1387.0...  \n",
       "...                                                  ...  \n",
       "23404           1056.0;1430.0;1560.0;1264.0;826.0;1195.0  \n",
       "23405           1344.0;1051.0;2309.0;1552.0;1063.0;987.0  \n",
       "23406             1120.0;1129.0;1222.0;942.0;745.0;896.0  \n",
       "23407   921.9999999999999;715.0;1047.0;783.0;452.0;886.0  \n",
       "23408   387.0;539.0;414.0;510.99999999999994;395.0;573.0  \n",
       "\n",
       "[23409 rows x 12 columns]"
      ]
     },
     "execution_count": 321,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_sgRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "id": "277065e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_1_list = []\n",
    "guide_2_list = []\n",
    "gene_1_list = []\n",
    "gene_2_list = []\n",
    "\n",
    "for i in all_sgRNA['2-wise gRNA combination']:\n",
    "    guide_1, guide_2 = i.split(' + ')\n",
    "    \n",
    "    guide_1_list.append(guide_1)\n",
    "    guide_2_list.append(guide_2)\n",
    "    \n",
    "    gene_1, gene_2 = guide_1.split('_')[0].upper(), guide_2.split('_')[0].upper()\n",
    "    \n",
    "    gene_1_list.append(gene_1)\n",
    "    gene_2_list.append(gene_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "id": "5ba397a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_sgRNA = all_sgRNA.drop(['2-wise gRNA combination', 'Key', 'sgRNA-A', 'sgRNA-B', 'day5 (Replicate 1)', 'day15 (Replicate 1)', 'day20 (Replicate 1)',\n",
    "                            'day5 (Replicate 2)', 'day15 (Replicate 2)', 'day20 (Replicate 2)'], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "id": "aae55b0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_sgRNA['Guide 1'] = guide_1_list\n",
    "all_sgRNA['Guide 2'] = guide_2_list\n",
    "all_sgRNA['Gene 1'] = gene_1_list\n",
    "all_sgRNA['Gene 2'] = gene_2_list\n",
    "all_sgRNA['Type'] = type_list\n",
    "all_sgRNA['Sequencing'] = ['CombiGEM-CRISPR'] * all_sgRNA.shape[0]\n",
    "all_sgRNA['Cell Line'] = ['OVCAR8'] * all_sgRNA.shape[0]\n",
    "all_sgRNA[\"Study\"] = [study_name_to_pubmed_id['wong_data']] * all_sgRNA.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "2ce7e0f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, all_sgRNA])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "id": "af2f11d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2baf334",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "id": "61efae81",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/apps/python/3.9-2022.05/lib/python3.9/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "wong_loc = os.path.join(learning_goals_loc_general, \"Wong\")\n",
    "\n",
    "statistically_sig = pd.read_excel(os.path.join(wong_loc, \"OVCAR8-ADR Cell Proliferation.xlsx\"))\n",
    "all_SL = pd.read_excel(os.path.join(wong_loc, \"log2_ratios.xlsx\"), skiprows = 2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "id": "c24ebc4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2-wise gRNA combination</th>\n",
       "      <th>sgRNA-A</th>\n",
       "      <th>sgRNA-B</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 1)</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 2)</th>\n",
       "      <th>Log2 ratio  Day 15/Day5 (Replicate 1)</th>\n",
       "      <th>Log2 ratio  Day 15/Day5 (Replicate 2)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dummyguide_1 + dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>0.205770</td>\n",
       "      <td>0.534492</td>\n",
       "      <td>-0.441889</td>\n",
       "      <td>-0.084731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>dummyguide_2 + dummyguide_1</td>\n",
       "      <td>dummyguide_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>-0.175483</td>\n",
       "      <td>-0.491619</td>\n",
       "      <td>-0.157136</td>\n",
       "      <td>0.320183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>dummyguide_3 + dummyguide_1</td>\n",
       "      <td>dummyguide_3</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>-0.182784</td>\n",
       "      <td>-0.010254</td>\n",
       "      <td>-0.087128</td>\n",
       "      <td>0.372476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT1_1 + dummyguide_1</td>\n",
       "      <td>DNMT1_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>0.152098</td>\n",
       "      <td>-0.430941</td>\n",
       "      <td>-0.148725</td>\n",
       "      <td>0.111234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DNMT1_2 + dummyguide_1</td>\n",
       "      <td>DNMT1_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>-0.005401</td>\n",
       "      <td>0.504039</td>\n",
       "      <td>-0.261538</td>\n",
       "      <td>-0.125159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11202</th>\n",
       "      <td>ING5_3 + ING5_1</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_1</td>\n",
       "      <td>-0.007535</td>\n",
       "      <td>0.324034</td>\n",
       "      <td>-0.043012</td>\n",
       "      <td>-0.441795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11203</th>\n",
       "      <td>ING5_2 + ING5_2</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>0.091024</td>\n",
       "      <td>1.334409</td>\n",
       "      <td>0.256652</td>\n",
       "      <td>-0.542246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11204</th>\n",
       "      <td>ING5_3 + ING5_2</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>0.447290</td>\n",
       "      <td>0.598651</td>\n",
       "      <td>-0.425111</td>\n",
       "      <td>-0.362895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11205</th>\n",
       "      <td>BRD4_3 + ING5_3</td>\n",
       "      <td>BRD4_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>0.073002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.020841</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11206</th>\n",
       "      <td>ING5_3 + ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>-0.421028</td>\n",
       "      <td>0.481833</td>\n",
       "      <td>0.332443</td>\n",
       "      <td>-0.059922</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11207 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           2-wise gRNA combination       sgRNA-A       sgRNA-B  \\\n",
       "0      dummyguide_1 + dummyguide_1  dummyguide_1  dummyguide_1   \n",
       "1      dummyguide_2 + dummyguide_1  dummyguide_2  dummyguide_1   \n",
       "2      dummyguide_3 + dummyguide_1  dummyguide_3  dummyguide_1   \n",
       "3           DNMT1_1 + dummyguide_1       DNMT1_1  dummyguide_1   \n",
       "4           DNMT1_2 + dummyguide_1       DNMT1_2  dummyguide_1   \n",
       "...                            ...           ...           ...   \n",
       "11202              ING5_3 + ING5_1        ING5_3        ING5_1   \n",
       "11203              ING5_2 + ING5_2        ING5_2        ING5_2   \n",
       "11204              ING5_3 + ING5_2        ING5_3        ING5_2   \n",
       "11205              BRD4_3 + ING5_3        BRD4_3        ING5_3   \n",
       "11206              ING5_3 + ING5_3        ING5_3        ING5_3   \n",
       "\n",
       "       Log2 ratio  Day 20/Day15 (Replicate 1)  \\\n",
       "0                                     0.205770   \n",
       "1                                    -0.175483   \n",
       "2                                    -0.182784   \n",
       "3                                     0.152098   \n",
       "4                                    -0.005401   \n",
       "...                                        ...   \n",
       "11202                                -0.007535   \n",
       "11203                                 0.091024   \n",
       "11204                                 0.447290   \n",
       "11205                                 0.073002   \n",
       "11206                                -0.421028   \n",
       "\n",
       "       Log2 ratio  Day 20/Day15 (Replicate 2)  \\\n",
       "0                                     0.534492   \n",
       "1                                    -0.491619   \n",
       "2                                    -0.010254   \n",
       "3                                    -0.430941   \n",
       "4                                     0.504039   \n",
       "...                                        ...   \n",
       "11202                                 0.324034   \n",
       "11203                                 1.334409   \n",
       "11204                                 0.598651   \n",
       "11205                                      NaN   \n",
       "11206                                 0.481833   \n",
       "\n",
       "       Log2 ratio  Day 15/Day5 (Replicate 1)  \\\n",
       "0                                   -0.441889   \n",
       "1                                   -0.157136   \n",
       "2                                   -0.087128   \n",
       "3                                   -0.148725   \n",
       "4                                   -0.261538   \n",
       "...                                       ...   \n",
       "11202                               -0.043012   \n",
       "11203                                0.256652   \n",
       "11204                               -0.425111   \n",
       "11205                               -0.020841   \n",
       "11206                                0.332443   \n",
       "\n",
       "       Log2 ratio  Day 15/Day5 (Replicate 2)  \n",
       "0                                   -0.084731  \n",
       "1                                    0.320183  \n",
       "2                                    0.372476  \n",
       "3                                    0.111234  \n",
       "4                                   -0.125159  \n",
       "...                                       ...  \n",
       "11202                               -0.441795  \n",
       "11203                               -0.542246  \n",
       "11204                               -0.362895  \n",
       "11205                                     NaN  \n",
       "11206                               -0.059922  \n",
       "\n",
       "[11207 rows x 7 columns]"
      ]
     },
     "execution_count": 328,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_SL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "eca1b445",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA-A</th>\n",
       "      <th>sgRNA-B</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 1)</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 2)</th>\n",
       "      <th>Q-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4_sg3</td>\n",
       "      <td>MLL_sg3</td>\n",
       "      <td>-2.61</td>\n",
       "      <td>-3.32</td>\n",
       "      <td>7.010000e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMI1_sg2</td>\n",
       "      <td>HDAC2_sg3</td>\n",
       "      <td>-3.91</td>\n",
       "      <td>-1.93</td>\n",
       "      <td>3.410000e-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMI1_sg2</td>\n",
       "      <td>KDM1B_sg3</td>\n",
       "      <td>-1.34</td>\n",
       "      <td>-3.71</td>\n",
       "      <td>1.200000e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ING3_sg3</td>\n",
       "      <td>BMI1_sg1</td>\n",
       "      <td>-3.68</td>\n",
       "      <td>-1.11</td>\n",
       "      <td>4.570000e-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRD4_sg3</td>\n",
       "      <td>KDM6A_sg2</td>\n",
       "      <td>-2.30</td>\n",
       "      <td>-2.20</td>\n",
       "      <td>1.520000e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>PRMT5_sg2</td>\n",
       "      <td>MBD1_sg1</td>\n",
       "      <td>-0.94</td>\n",
       "      <td>-1.04</td>\n",
       "      <td>2.470000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>EP300_sg3</td>\n",
       "      <td>MBD1_sg3</td>\n",
       "      <td>-0.92</td>\n",
       "      <td>-1.03</td>\n",
       "      <td>2.930000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>ING3_sg1</td>\n",
       "      <td>BRD4_sg3</td>\n",
       "      <td>-0.99</td>\n",
       "      <td>-0.96</td>\n",
       "      <td>2.930000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>KDM1A_sg1</td>\n",
       "      <td>HDAC2_sg3</td>\n",
       "      <td>-0.94</td>\n",
       "      <td>-0.95</td>\n",
       "      <td>4.710000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>PRMT5_sg2</td>\n",
       "      <td>CARM1_sg1</td>\n",
       "      <td>-0.95</td>\n",
       "      <td>-0.91</td>\n",
       "      <td>5.780000e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sgRNA-A    sgRNA-B  Log2 ratio  Day 20/Day15 (Replicate 1)  \\\n",
       "0    BRD4_sg3    MLL_sg3                                    -2.61   \n",
       "1    BMI1_sg2  HDAC2_sg3                                    -3.91   \n",
       "2    BMI1_sg2  KDM1B_sg3                                    -1.34   \n",
       "3    ING3_sg3   BMI1_sg1                                    -3.68   \n",
       "4    BRD4_sg3  KDM6A_sg2                                    -2.30   \n",
       "..        ...        ...                                      ...   \n",
       "56  PRMT5_sg2   MBD1_sg1                                    -0.94   \n",
       "57  EP300_sg3   MBD1_sg3                                    -0.92   \n",
       "58   ING3_sg1   BRD4_sg3                                    -0.99   \n",
       "59  KDM1A_sg1  HDAC2_sg3                                    -0.94   \n",
       "60  PRMT5_sg2  CARM1_sg1                                    -0.95   \n",
       "\n",
       "    Log2 ratio  Day 20/Day15 (Replicate 2)       Q-value  \n",
       "0                                     -3.32  7.010000e-33  \n",
       "1                                     -1.93  3.410000e-32  \n",
       "2                                     -3.71  1.200000e-23  \n",
       "3                                     -1.11  4.570000e-21  \n",
       "4                                     -2.20  1.520000e-18  \n",
       "..                                      ...           ...  \n",
       "56                                    -1.04  2.470000e-03  \n",
       "57                                    -1.03  2.930000e-03  \n",
       "58                                    -0.96  2.930000e-03  \n",
       "59                                    -0.95  4.710000e-03  \n",
       "60                                    -0.91  5.780000e-03  \n",
       "\n",
       "[61 rows x 5 columns]"
      ]
     },
     "execution_count": 329,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "statistically_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "id": "976e8614",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['2-wise gRNA combination', 'sgRNA-A', 'sgRNA-B',\n",
       "       'Log2 ratio  Day 20/Day15 (Replicate 1)',\n",
       "       'Log2 ratio  Day 20/Day15 (Replicate 2)',\n",
       "       'Log2 ratio  Day 15/Day5 (Replicate 1)',\n",
       "       'Log2 ratio  Day 15/Day5 (Replicate 2)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 330,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_SL.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "id": "949a6637",
   "metadata": {},
   "outputs": [],
   "source": [
    "Wong_map_sig = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "id": "150a47d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_A_list = all_SL['sgRNA-A']\n",
    "gene_B_list = all_SL['sgRNA-B']\n",
    "\n",
    "for row in range(statistically_sig.shape[0]):\n",
    "    gene_A = statistically_sig.loc[row, 'sgRNA-A'].split('_')[0]\n",
    "    gene_B = statistically_sig.loc[row, 'sgRNA-B'].split('_')[0]\n",
    "    \n",
    "    # sorted for uniqueness\n",
    "    gene_A, gene_B = sorted([gene_A, gene_B])\n",
    "    \n",
    "    if gene_A == 'dummyguide' or gene_B == 'dummyguide':\n",
    "        continue\n",
    "    \n",
    "    key = gene_A + \"_\" + gene_B\n",
    "    if key not in Wong_map_sig:\n",
    "        Wong_map_sig[key] = {}\n",
    "        Wong_map_sig[key]['Log2 ratio  Day 20/Day15'] = statistically_sig.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 1)'] + statistically_sig.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 2)']\n",
    "    else:\n",
    "        Wong_map_sig[key]['Log2 ratio  Day 20/Day15'] += statistically_sig.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 1)'] + statistically_sig.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 2)']\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "ad1c9036",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_SL = all_SL.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "c126b89a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Wong_map = {}\n",
    "Wong_map_repeats = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "52a4bc9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_combined = statistically_sig[['sgRNA-A', 'sgRNA-B']].agg(' + '.join, axis=1).tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "id": "762c4932",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_A_list = all_SL['sgRNA-A']\n",
    "gene_B_list = all_SL['sgRNA-B']\n",
    "\n",
    "for row in range(all_SL.shape[0]):\n",
    "    gene_A = all_SL.loc[row, 'sgRNA-A'].split('_')[0]\n",
    "    gene_B = all_SL.loc[row, 'sgRNA-B'].split('_')[0]\n",
    "\n",
    "    # sorted for uniqueness\n",
    "    gene_A, gene_B = sorted([gene_A, gene_B])\n",
    "    \n",
    "#     if gene_A == 'dummyguide' or gene_B == 'dummyguide':\n",
    "#         continue\n",
    "    \n",
    "    key = gene_A + \"_\" + gene_B\n",
    "        \n",
    "    if key not in Wong_map:\n",
    "        Wong_map[key] = {}\n",
    "        Wong_map[key]['Log2 ratio  Day 20/Day15'] = all_SL.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 1)'] + all_SL.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 2)']\n",
    "        Wong_map[key]['Log2 ratio  Day 15/Day5'] = all_SL.loc[row, 'Log2 ratio  Day 15/Day5 (Replicate 1)'] + all_SL.loc[row, 'Log2 ratio  Day 15/Day5 (Replicate 2)']\n",
    "        Wong_map[key][\"Q_Val\"] = 0\n",
    "        Wong_map_repeats[key] = 1\n",
    "    else:\n",
    "        Wong_map[key]['Log2 ratio  Day 20/Day15'] += all_SL.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 1)'] + all_SL.loc[row, 'Log2 ratio  Day 20/Day15 (Replicate 2)']\n",
    "        Wong_map[key]['Log2 ratio  Day 15/Day5'] += all_SL.loc[row, 'Log2 ratio  Day 15/Day5 (Replicate 1)'] + all_SL.loc[row, 'Log2 ratio  Day 15/Day5 (Replicate 2)']\n",
    "        Wong_map_repeats[key] += 1\n",
    "      \n",
    "    curr_combined_1 = gene_A + \"_sg\" + all_SL.loc[row, 'sgRNA-A'].split('_')[1]  + \" + \" + gene_B + \"_sg\" + all_SL.loc[row, 'sgRNA-B'].split('_')[1]\n",
    "    curr_combined_2 = gene_B + \"_sg\" + all_SL.loc[row, 'sgRNA-B'].split('_')[1]  + \" + \" + gene_A + \"_sg\" + all_SL.loc[row, 'sgRNA-A'].split('_')[1]\n",
    "    \n",
    "    if curr_combined_1 in sig_combined:\n",
    "        Wong_map[key][\"Q_Val\"] += statistically_sig.loc[sig_combined.index(curr_combined_1), \"Q-value\"]\n",
    "    elif curr_combined_2 in sig_combined:\n",
    "        Wong_map[key][\"Q_Val\"] += statistically_sig.loc[sig_combined.index(curr_combined_2), \"Q-value\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "id": "b2f2122b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA-A</th>\n",
       "      <th>sgRNA-B</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 1)</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 2)</th>\n",
       "      <th>Q-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4_sg3</td>\n",
       "      <td>MLL_sg3</td>\n",
       "      <td>-2.61</td>\n",
       "      <td>-3.32</td>\n",
       "      <td>7.010000e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMI1_sg2</td>\n",
       "      <td>HDAC2_sg3</td>\n",
       "      <td>-3.91</td>\n",
       "      <td>-1.93</td>\n",
       "      <td>3.410000e-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMI1_sg2</td>\n",
       "      <td>KDM1B_sg3</td>\n",
       "      <td>-1.34</td>\n",
       "      <td>-3.71</td>\n",
       "      <td>1.200000e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ING3_sg3</td>\n",
       "      <td>BMI1_sg1</td>\n",
       "      <td>-3.68</td>\n",
       "      <td>-1.11</td>\n",
       "      <td>4.570000e-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRD4_sg3</td>\n",
       "      <td>KDM6A_sg2</td>\n",
       "      <td>-2.30</td>\n",
       "      <td>-2.20</td>\n",
       "      <td>1.520000e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>PRMT5_sg2</td>\n",
       "      <td>MBD1_sg1</td>\n",
       "      <td>-0.94</td>\n",
       "      <td>-1.04</td>\n",
       "      <td>2.470000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>EP300_sg3</td>\n",
       "      <td>MBD1_sg3</td>\n",
       "      <td>-0.92</td>\n",
       "      <td>-1.03</td>\n",
       "      <td>2.930000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>ING3_sg1</td>\n",
       "      <td>BRD4_sg3</td>\n",
       "      <td>-0.99</td>\n",
       "      <td>-0.96</td>\n",
       "      <td>2.930000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>KDM1A_sg1</td>\n",
       "      <td>HDAC2_sg3</td>\n",
       "      <td>-0.94</td>\n",
       "      <td>-0.95</td>\n",
       "      <td>4.710000e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>PRMT5_sg2</td>\n",
       "      <td>CARM1_sg1</td>\n",
       "      <td>-0.95</td>\n",
       "      <td>-0.91</td>\n",
       "      <td>5.780000e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sgRNA-A    sgRNA-B  Log2 ratio  Day 20/Day15 (Replicate 1)  \\\n",
       "0    BRD4_sg3    MLL_sg3                                    -2.61   \n",
       "1    BMI1_sg2  HDAC2_sg3                                    -3.91   \n",
       "2    BMI1_sg2  KDM1B_sg3                                    -1.34   \n",
       "3    ING3_sg3   BMI1_sg1                                    -3.68   \n",
       "4    BRD4_sg3  KDM6A_sg2                                    -2.30   \n",
       "..        ...        ...                                      ...   \n",
       "56  PRMT5_sg2   MBD1_sg1                                    -0.94   \n",
       "57  EP300_sg3   MBD1_sg3                                    -0.92   \n",
       "58   ING3_sg1   BRD4_sg3                                    -0.99   \n",
       "59  KDM1A_sg1  HDAC2_sg3                                    -0.94   \n",
       "60  PRMT5_sg2  CARM1_sg1                                    -0.95   \n",
       "\n",
       "    Log2 ratio  Day 20/Day15 (Replicate 2)       Q-value  \n",
       "0                                     -3.32  7.010000e-33  \n",
       "1                                     -1.93  3.410000e-32  \n",
       "2                                     -3.71  1.200000e-23  \n",
       "3                                     -1.11  4.570000e-21  \n",
       "4                                     -2.20  1.520000e-18  \n",
       "..                                      ...           ...  \n",
       "56                                    -1.04  2.470000e-03  \n",
       "57                                    -1.03  2.930000e-03  \n",
       "58                                    -0.96  2.930000e-03  \n",
       "59                                    -0.95  4.710000e-03  \n",
       "60                                    -0.91  5.780000e-03  \n",
       "\n",
       "[61 rows x 5 columns]"
      ]
     },
     "execution_count": 337,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "statistically_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "id": "4431a7a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "for key in Wong_map.keys():\n",
    "    Wong_map[key]['Log2 ratio  Day 20/Day15'] = Wong_map[key]['Log2 ratio  Day 20/Day15']/Wong_map_repeats[key]\n",
    "    Wong_map[key]['Log2 ratio  Day 15/Day5'] = Wong_map[key]['Log2 ratio  Day 15/Day5']/Wong_map_repeats[key]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "id": "63c17b26",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2-wise gRNA combination</th>\n",
       "      <th>sgRNA-A</th>\n",
       "      <th>sgRNA-B</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 1)</th>\n",
       "      <th>Log2 ratio  Day 20/Day15 (Replicate 2)</th>\n",
       "      <th>Log2 ratio  Day 15/Day5 (Replicate 1)</th>\n",
       "      <th>Log2 ratio  Day 15/Day5 (Replicate 2)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dummyguide_1 + dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>0.205770</td>\n",
       "      <td>0.534492</td>\n",
       "      <td>-0.441889</td>\n",
       "      <td>-0.084731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>dummyguide_2 + dummyguide_1</td>\n",
       "      <td>dummyguide_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>-0.175483</td>\n",
       "      <td>-0.491619</td>\n",
       "      <td>-0.157136</td>\n",
       "      <td>0.320183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>dummyguide_3 + dummyguide_1</td>\n",
       "      <td>dummyguide_3</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>-0.182784</td>\n",
       "      <td>-0.010254</td>\n",
       "      <td>-0.087128</td>\n",
       "      <td>0.372476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT1_1 + dummyguide_1</td>\n",
       "      <td>DNMT1_1</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>0.152098</td>\n",
       "      <td>-0.430941</td>\n",
       "      <td>-0.148725</td>\n",
       "      <td>0.111234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DNMT1_2 + dummyguide_1</td>\n",
       "      <td>DNMT1_2</td>\n",
       "      <td>dummyguide_1</td>\n",
       "      <td>-0.005401</td>\n",
       "      <td>0.504039</td>\n",
       "      <td>-0.261538</td>\n",
       "      <td>-0.125159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11202</th>\n",
       "      <td>ING5_3 + ING5_1</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_1</td>\n",
       "      <td>-0.007535</td>\n",
       "      <td>0.324034</td>\n",
       "      <td>-0.043012</td>\n",
       "      <td>-0.441795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11203</th>\n",
       "      <td>ING5_2 + ING5_2</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>0.091024</td>\n",
       "      <td>1.334409</td>\n",
       "      <td>0.256652</td>\n",
       "      <td>-0.542246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11204</th>\n",
       "      <td>ING5_3 + ING5_2</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>0.447290</td>\n",
       "      <td>0.598651</td>\n",
       "      <td>-0.425111</td>\n",
       "      <td>-0.362895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11205</th>\n",
       "      <td>BRD4_3 + ING5_3</td>\n",
       "      <td>BRD4_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>0.073002</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.020841</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11206</th>\n",
       "      <td>ING5_3 + ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>-0.421028</td>\n",
       "      <td>0.481833</td>\n",
       "      <td>0.332443</td>\n",
       "      <td>-0.059922</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11207 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           2-wise gRNA combination       sgRNA-A       sgRNA-B  \\\n",
       "0      dummyguide_1 + dummyguide_1  dummyguide_1  dummyguide_1   \n",
       "1      dummyguide_2 + dummyguide_1  dummyguide_2  dummyguide_1   \n",
       "2      dummyguide_3 + dummyguide_1  dummyguide_3  dummyguide_1   \n",
       "3           DNMT1_1 + dummyguide_1       DNMT1_1  dummyguide_1   \n",
       "4           DNMT1_2 + dummyguide_1       DNMT1_2  dummyguide_1   \n",
       "...                            ...           ...           ...   \n",
       "11202              ING5_3 + ING5_1        ING5_3        ING5_1   \n",
       "11203              ING5_2 + ING5_2        ING5_2        ING5_2   \n",
       "11204              ING5_3 + ING5_2        ING5_3        ING5_2   \n",
       "11205              BRD4_3 + ING5_3        BRD4_3        ING5_3   \n",
       "11206              ING5_3 + ING5_3        ING5_3        ING5_3   \n",
       "\n",
       "       Log2 ratio  Day 20/Day15 (Replicate 1)  \\\n",
       "0                                     0.205770   \n",
       "1                                    -0.175483   \n",
       "2                                    -0.182784   \n",
       "3                                     0.152098   \n",
       "4                                    -0.005401   \n",
       "...                                        ...   \n",
       "11202                                -0.007535   \n",
       "11203                                 0.091024   \n",
       "11204                                 0.447290   \n",
       "11205                                 0.073002   \n",
       "11206                                -0.421028   \n",
       "\n",
       "       Log2 ratio  Day 20/Day15 (Replicate 2)  \\\n",
       "0                                     0.534492   \n",
       "1                                    -0.491619   \n",
       "2                                    -0.010254   \n",
       "3                                    -0.430941   \n",
       "4                                     0.504039   \n",
       "...                                        ...   \n",
       "11202                                 0.324034   \n",
       "11203                                 1.334409   \n",
       "11204                                 0.598651   \n",
       "11205                                 0.000000   \n",
       "11206                                 0.481833   \n",
       "\n",
       "       Log2 ratio  Day 15/Day5 (Replicate 1)  \\\n",
       "0                                   -0.441889   \n",
       "1                                   -0.157136   \n",
       "2                                   -0.087128   \n",
       "3                                   -0.148725   \n",
       "4                                   -0.261538   \n",
       "...                                       ...   \n",
       "11202                               -0.043012   \n",
       "11203                                0.256652   \n",
       "11204                               -0.425111   \n",
       "11205                               -0.020841   \n",
       "11206                                0.332443   \n",
       "\n",
       "       Log2 ratio  Day 15/Day5 (Replicate 2)  \n",
       "0                                   -0.084731  \n",
       "1                                    0.320183  \n",
       "2                                    0.372476  \n",
       "3                                    0.111234  \n",
       "4                                   -0.125159  \n",
       "...                                       ...  \n",
       "11202                               -0.441795  \n",
       "11203                               -0.542246  \n",
       "11204                               -0.362895  \n",
       "11205                                0.000000  \n",
       "11206                               -0.059922  \n",
       "\n",
       "[11207 rows x 7 columns]"
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_SL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "34a55488",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_A_list = []\n",
    "gene_B_list = []\n",
    "SL_scores = []\n",
    "phenotype_scores = []\n",
    "stat_scores = []\n",
    "\n",
    "for row in range(all_SL.shape[0]):\n",
    "    gene_A = all_SL.loc[row, 'sgRNA-A'].split('_')[0]\n",
    "    gene_B = all_SL.loc[row, 'sgRNA-B'].split('_')[0]\n",
    "    \n",
    "    # sorted for uniqueness\n",
    "    gene_A, gene_B = sorted([gene_A, gene_B])\n",
    "    \n",
    "    key = gene_A + \"_\" + gene_B\n",
    "    \n",
    "    gene_A_list.append(gene_A)\n",
    "    gene_B_list.append(gene_B)\n",
    "    SL_scores.append(Wong_map[key]['Log2 ratio  Day 20/Day15'])\n",
    "    phenotype_scores.append(Wong_map[key]['Log2 ratio  Day 15/Day5'])\n",
    "    stat_scores.append(Wong_map[key]['Q_Val'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "id": "fc6294b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = gene_A_list\n",
    "curr_GI[\"Gene_B\"] = gene_B_list\n",
    "curr_GI[\"GI_Score\"] = SL_scores\n",
    "curr_GI[\"Stat_Score\"] = stat_scores\n",
    "curr_GI[\"GI_Cutoff\"] = [-0.90] * len(gene_A_list)\n",
    "curr_GI[\"Stat_Cutoff\"] = [0.01] * len(gene_A_list)\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['wong_data']] * len(gene_A_list)\n",
    "curr_GI[\"Cell_Line\"] = [\"OVCAR8\"] * len(gene_A_list)\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI['Sorted_Genes'] = ['_'.join(sorted([curr_GI[\"Gene_A\"][i], curr_GI[\"Gene_B\"][i]])) for i in range(curr_GI.shape[0])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "id": "dd1344b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp = curr_GI.groupby('Sorted_Genes')['GI_Score'].apply(lambda x: np.mean(x))\n",
    "curr_GI = curr_GI.merge(temp, how = 'left', left_on = 'Sorted_Genes', right_on = temp.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "id": "cb0914bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = curr_GI.drop(columns = 'GI_Score_x')\n",
    "curr_GI = curr_GI.rename(columns = {'GI_Score_y': 'GI_Score'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "id": "657527d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = curr_GI.groupby([\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\", \"Sorted_Genes\"], as_index=False).count()\n",
    "curr_GI = curr_GI.drop(columns = 'Sorted_Genes')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "12b70f05",
   "metadata": {},
   "outputs": [],
   "source": [
    "#curr_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "b915afa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "id": "1971e7c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#curr_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "id": "c7d48136",
   "metadata": {},
   "outputs": [],
   "source": [
    "#controls['wong_data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "e7162588",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "50\n",
      "---\n",
      "Both GI and Stat cutoffs are present...\n",
      "Counts reference...\n",
      "Number of double pairs: 22050\n",
      "Number of controls: 9\n",
      "Number of singles: 1350\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = curr_GI.copy(), study_controls = controls['wong_data'], study_conditions = study_conditions['wong_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "id": "d26c550e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16403553/ipykernel_128410/4045206025.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db_inserts['score_ref']['SL_or_not'].loc[db_inserts['score_ref']['gene_pair'].isin(sig_gene_pairs)] = 'SL'\n"
     ]
    }
   ],
   "source": [
    "# Wong Specific\n",
    "sig_gene_pairs = list(set(['_'.join(sorted([statistically_sig['sgRNA-A'].iloc[i].split('_')[0], statistically_sig['sgRNA-B'].iloc[i].split('_')[0]])) for i in range(statistically_sig.shape[0])]))\n",
    "db_inserts['score_ref']['SL_or_not'] = ['Not SL'] * db_inserts['score_ref'].shape[0]\n",
    "db_inserts['score_ref']['SL_or_not'].loc[db_inserts['score_ref']['gene_pair'].isin(sig_gene_pairs)] = 'SL'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "id": "ed63d6eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>cell_line_origin</th>\n",
       "      <th>Condition</th>\n",
       "      <th>study_origin</th>\n",
       "      <th>target_type</th>\n",
       "      <th>T0_counts</th>\n",
       "      <th>T0_replicate_names</th>\n",
       "      <th>TEnd_counts</th>\n",
       "      <th>TEnd_replicate_names</th>\n",
       "      <th>gene_pair</th>\n",
       "      <th>gene_pair_orientation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DUMMYGUIDE_1</td>\n",
       "      <td>DUMMYGUIDE_1</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>1987.0;1618.0;1919.0;1299.0;987.0;1485.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Control</td>\n",
       "      <td>1987.0;1299.0</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>1919.0;1485.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>DUMMYGUIDE|DUMMYGUIDE</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DUMMYGUIDE_1</td>\n",
       "      <td>DUMMYGUIDE_2</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>757.0;868.0;679.0;510.99999999999994;465.0;531.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Control</td>\n",
       "      <td>757.0;510.99999999999994</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>679.0;531.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>DUMMYGUIDE|DUMMYGUIDE</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DUMMYGUIDE_1</td>\n",
       "      <td>DUMMYGUIDE_3</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>1657.0;1901.0;1558.0;1165.9999999999998;1091.0...</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Control</td>\n",
       "      <td>1657.0;1165.9999999999998</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>1558.0;1243.0000000000002</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>DUMMYGUIDE|DUMMYGUIDE</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DUMMYGUIDE_1</td>\n",
       "      <td>DNMT1_1</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>DNMT1</td>\n",
       "      <td>1178.0;1555.0;900.0;1093.0;972.0000000000001;6...</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Single</td>\n",
       "      <td>1178.0;1093.0</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>900.0;638.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>DNMT1|DUMMYGUIDE</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DUMMYGUIDE_1</td>\n",
       "      <td>DNMT1_2</td>\n",
       "      <td>DUMMYGUIDE</td>\n",
       "      <td>DNMT1</td>\n",
       "      <td>2103.0;2150.0;1291.0;1775.0000000000002;1387.0...</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Single</td>\n",
       "      <td>2103.0;1775.0000000000002</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>1291.0;2080.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>DNMT1|DUMMYGUIDE</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23404</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING4_2</td>\n",
       "      <td>ING5</td>\n",
       "      <td>ING4</td>\n",
       "      <td>1056.0;1430.0;1560.0;1264.0;826.0;1195.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Dual</td>\n",
       "      <td>1056.0;1264.0</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>1560.0;1195.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>ING4|ING5</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23405</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING4_3</td>\n",
       "      <td>ING5</td>\n",
       "      <td>ING4</td>\n",
       "      <td>1344.0;1051.0;2309.0;1552.0;1063.0;987.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Dual</td>\n",
       "      <td>1344.0;1552.0</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>2309.0;987.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>ING4|ING5</td>\n",
       "      <td>B_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23406</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_1</td>\n",
       "      <td>ING5</td>\n",
       "      <td>ING5</td>\n",
       "      <td>1120.0;1129.0;1222.0;942.0;745.0;896.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Single</td>\n",
       "      <td>1120.0;942.0</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>1222.0;896.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>ING5|ING5</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23407</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_2</td>\n",
       "      <td>ING5</td>\n",
       "      <td>ING5</td>\n",
       "      <td>921.9999999999999;715.0;1047.0;783.0;452.0;886.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Single</td>\n",
       "      <td>921.9999999999999;783.0</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>1047.0;886.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>ING5|ING5</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23408</th>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5_3</td>\n",
       "      <td>ING5</td>\n",
       "      <td>ING5</td>\n",
       "      <td>387.0;539.0;414.0;510.99999999999994;395.0;573.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>day5 (Replicate 1);day15 (Replicate 1);day20 (...</td>\n",
       "      <td>26864203</td>\n",
       "      <td>Single</td>\n",
       "      <td>387.0;510.99999999999994</td>\n",
       "      <td>day5 (Replicate 1);day5 (Replicate 2)</td>\n",
       "      <td>414.0;573.0</td>\n",
       "      <td>day20 (Replicate 1);day20 (Replicate 2)</td>\n",
       "      <td>ING5|ING5</td>\n",
       "      <td>A_B</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23409 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Guide 1       Guide 2      Gene 1      Gene 2  \\\n",
       "0      DUMMYGUIDE_1  DUMMYGUIDE_1  DUMMYGUIDE  DUMMYGUIDE   \n",
       "1      DUMMYGUIDE_1  DUMMYGUIDE_2  DUMMYGUIDE  DUMMYGUIDE   \n",
       "2      DUMMYGUIDE_1  DUMMYGUIDE_3  DUMMYGUIDE  DUMMYGUIDE   \n",
       "3      DUMMYGUIDE_1       DNMT1_1  DUMMYGUIDE       DNMT1   \n",
       "4      DUMMYGUIDE_1       DNMT1_2  DUMMYGUIDE       DNMT1   \n",
       "...             ...           ...         ...         ...   \n",
       "23404        ING5_3        ING4_2        ING5        ING4   \n",
       "23405        ING5_3        ING4_3        ING5        ING4   \n",
       "23406        ING5_3        ING5_1        ING5        ING5   \n",
       "23407        ING5_3        ING5_2        ING5        ING5   \n",
       "23408        ING5_3        ING5_3        ING5        ING5   \n",
       "\n",
       "                                        Count Replicates cell_line_origin  \\\n",
       "0               1987.0;1618.0;1919.0;1299.0;987.0;1485.0           OVCAR8   \n",
       "1       757.0;868.0;679.0;510.99999999999994;465.0;531.0           OVCAR8   \n",
       "2      1657.0;1901.0;1558.0;1165.9999999999998;1091.0...           OVCAR8   \n",
       "3      1178.0;1555.0;900.0;1093.0;972.0000000000001;6...           OVCAR8   \n",
       "4      2103.0;2150.0;1291.0;1775.0000000000002;1387.0...           OVCAR8   \n",
       "...                                                  ...              ...   \n",
       "23404           1056.0;1430.0;1560.0;1264.0;826.0;1195.0           OVCAR8   \n",
       "23405           1344.0;1051.0;2309.0;1552.0;1063.0;987.0           OVCAR8   \n",
       "23406             1120.0;1129.0;1222.0;942.0;745.0;896.0           OVCAR8   \n",
       "23407   921.9999999999999;715.0;1047.0;783.0;452.0;886.0           OVCAR8   \n",
       "23408   387.0;539.0;414.0;510.99999999999994;395.0;573.0           OVCAR8   \n",
       "\n",
       "                                               Condition study_origin  \\\n",
       "0      day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "1      day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "2      day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "3      day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "4      day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "...                                                  ...          ...   \n",
       "23404  day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "23405  day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "23406  day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "23407  day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "23408  day5 (Replicate 1);day15 (Replicate 1);day20 (...     26864203   \n",
       "\n",
       "      target_type                  T0_counts  \\\n",
       "0         Control              1987.0;1299.0   \n",
       "1         Control   757.0;510.99999999999994   \n",
       "2         Control  1657.0;1165.9999999999998   \n",
       "3          Single              1178.0;1093.0   \n",
       "4          Single  2103.0;1775.0000000000002   \n",
       "...           ...                        ...   \n",
       "23404        Dual              1056.0;1264.0   \n",
       "23405        Dual              1344.0;1552.0   \n",
       "23406      Single               1120.0;942.0   \n",
       "23407      Single    921.9999999999999;783.0   \n",
       "23408      Single   387.0;510.99999999999994   \n",
       "\n",
       "                          T0_replicate_names                TEnd_counts  \\\n",
       "0      day5 (Replicate 1);day5 (Replicate 2)              1919.0;1485.0   \n",
       "1      day5 (Replicate 1);day5 (Replicate 2)                679.0;531.0   \n",
       "2      day5 (Replicate 1);day5 (Replicate 2)  1558.0;1243.0000000000002   \n",
       "3      day5 (Replicate 1);day5 (Replicate 2)                900.0;638.0   \n",
       "4      day5 (Replicate 1);day5 (Replicate 2)              1291.0;2080.0   \n",
       "...                                      ...                        ...   \n",
       "23404  day5 (Replicate 1);day5 (Replicate 2)              1560.0;1195.0   \n",
       "23405  day5 (Replicate 1);day5 (Replicate 2)               2309.0;987.0   \n",
       "23406  day5 (Replicate 1);day5 (Replicate 2)               1222.0;896.0   \n",
       "23407  day5 (Replicate 1);day5 (Replicate 2)               1047.0;886.0   \n",
       "23408  day5 (Replicate 1);day5 (Replicate 2)                414.0;573.0   \n",
       "\n",
       "                          TEnd_replicate_names              gene_pair  \\\n",
       "0      day20 (Replicate 1);day20 (Replicate 2)  DUMMYGUIDE|DUMMYGUIDE   \n",
       "1      day20 (Replicate 1);day20 (Replicate 2)  DUMMYGUIDE|DUMMYGUIDE   \n",
       "2      day20 (Replicate 1);day20 (Replicate 2)  DUMMYGUIDE|DUMMYGUIDE   \n",
       "3      day20 (Replicate 1);day20 (Replicate 2)       DNMT1|DUMMYGUIDE   \n",
       "4      day20 (Replicate 1);day20 (Replicate 2)       DNMT1|DUMMYGUIDE   \n",
       "...                                        ...                    ...   \n",
       "23404  day20 (Replicate 1);day20 (Replicate 2)              ING4|ING5   \n",
       "23405  day20 (Replicate 1);day20 (Replicate 2)              ING4|ING5   \n",
       "23406  day20 (Replicate 1);day20 (Replicate 2)              ING5|ING5   \n",
       "23407  day20 (Replicate 1);day20 (Replicate 2)              ING5|ING5   \n",
       "23408  day20 (Replicate 1);day20 (Replicate 2)              ING5|ING5   \n",
       "\n",
       "      gene_pair_orientation  \n",
       "0                       A_B  \n",
       "1                       A_B  \n",
       "2                       A_B  \n",
       "3                       B_A  \n",
       "4                       B_A  \n",
       "...                     ...  \n",
       "23404                   B_A  \n",
       "23405                   B_A  \n",
       "23406                   A_B  \n",
       "23407                   A_B  \n",
       "23408                   A_B  \n",
       "\n",
       "[23409 rows x 15 columns]"
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['counts_ref']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "id": "7a4a49b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 160\n",
      "Counts insert: 23409\n",
      "Score insert: 1225\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c6bf8e4",
   "metadata": {},
   "source": [
    "## Zhao Data (A549, HELA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "id": "fc2a31b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "zhao_loc = os.path.join(learning_goals_loc_general, \"Zhao\")\n",
    "\n",
    "Zhao_excel = pd.ExcelFile(os.path.join(zhao_loc, \"1 - sgRNA Counts.xlsx\"))\n",
    "\n",
    "sgRNA_ref = Zhao_excel.parse(\"Metabolism_dual_spacers\")\n",
    "Zhao_sgRNA_A549 =  Zhao_excel.parse(\"Raw counts from A549 screen\")\n",
    "Zhao_sgRNA_HELA =  Zhao_excel.parse(\"Raw counts from HeLa screen\")\n",
    "Zhao_sgRNA_plasmid = Zhao_excel.parse(\"Raw counts from plasmid library\")\n",
    "\n",
    "# Zhao_sgRNA_A549 =  pd.read_excel(os.path.join(zhao_loc, \"1 - sgRNA Counts.xlsx\"), sheet_name = \"Raw counts from A549 screen\")\n",
    "# Zhao_sgRNA_HELA =  pd.read_excel(os.path.join(zhao_loc, \"1 - sgRNA Counts.xlsx\"), sheet_name = \"Raw counts from HeLa screen\")\n",
    "# Zhao_sgRNA_plasmid =  pd.read_excel(os.path.join(zhao_loc, \"1 - sgRNA Counts.xlsx\"), sheet_name = \"Raw counts from plasmid library\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ac3e637",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "19fab44e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SequenceID (GeneA_chr_position__GeneB_chr_position)</th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_A_spacer sequence</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Gene_B_spacer sequence</th>\n",
       "      <th>Sequence for oligo synthesis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>CTGGCTTGCGCCTTGCCATG</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>GGTACCTATTGTTGAACCAG</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>CTGGCTTGCGCCTTGCCATG</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>CTCCTCTGGGGTATACTTGA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>CTGGCTTGCGCCTTGCCATG</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>GAAAGGGCTGGGTACGAGTG</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>CTGGCTTGCGCCTTGCCATG</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>TGTGGCGCGTCTGTGCGCGA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>CTGGCTTGCGCCTTGCCATG</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>GATACAGATGATCGCTCCGA</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836__RPIA_chr2_88991246</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>GAGGGCTTACCGGTGTCGGC</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>GGGCCAAGACCCGCCCGTAG</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_747467</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>GAGGGCTTACCGGTGTCGGC</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>CTCGACATAGCAAGACCGAG</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755880</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>GAGGGCTTACCGGTGTCGGC</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>CATCGACGAGTACAAGCCCC</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755937</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>GAGGGCTTACCGGTGTCGGC</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>AGCACAGATGCCCGCTTACC</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836__TKT_chr3_53263054</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>GAGGGCTTACCGGTGTCGGC</td>\n",
       "      <td>TKT</td>\n",
       "      <td>TCTGTAGCAACGCCATCACT</td>\n",
       "      <td>tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11934 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      SequenceID (GeneA_chr_position__GeneB_chr_position) Gene_A  \\\n",
       "0             ALDOA_chr16_30078215__ALDOB_chr9_104188893   ALDOA   \n",
       "1             ALDOA_chr16_30078215__ALDOC_chr17_26901140   ALDOA   \n",
       "2             ALDOA_chr16_30078215__ALDOC_chr17_26902525   ALDOA   \n",
       "3             ALDOA_chr16_30078215__DLAT_chr11_111896198   ALDOA   \n",
       "4             ALDOA_chr16_30078215__DLAT_chr11_111899333   ALDOA   \n",
       "...                                                  ...     ...   \n",
       "11929             TPI1_chr12_6976836__RPIA_chr2_88991246    TPI1   \n",
       "11930            TPI1_chr12_6976836__TALDO1_chr11_747467    TPI1   \n",
       "11931            TPI1_chr12_6976836__TALDO1_chr11_755880    TPI1   \n",
       "11932            TPI1_chr12_6976836__TALDO1_chr11_755937    TPI1   \n",
       "11933              TPI1_chr12_6976836__TKT_chr3_53263054    TPI1   \n",
       "\n",
       "      Gene_A_spacer sequence  Gene_B Gene_B_spacer sequence  \\\n",
       "0       CTGGCTTGCGCCTTGCCATG   ALDOB   GGTACCTATTGTTGAACCAG   \n",
       "1       CTGGCTTGCGCCTTGCCATG   ALDOC   CTCCTCTGGGGTATACTTGA   \n",
       "2       CTGGCTTGCGCCTTGCCATG   ALDOC   GAAAGGGCTGGGTACGAGTG   \n",
       "3       CTGGCTTGCGCCTTGCCATG    DLAT   TGTGGCGCGTCTGTGCGCGA   \n",
       "4       CTGGCTTGCGCCTTGCCATG    DLAT   GATACAGATGATCGCTCCGA   \n",
       "...                      ...     ...                    ...   \n",
       "11929   GAGGGCTTACCGGTGTCGGC    RPIA   GGGCCAAGACCCGCCCGTAG   \n",
       "11930   GAGGGCTTACCGGTGTCGGC  TALDO1   CTCGACATAGCAAGACCGAG   \n",
       "11931   GAGGGCTTACCGGTGTCGGC  TALDO1   CATCGACGAGTACAAGCCCC   \n",
       "11932   GAGGGCTTACCGGTGTCGGC  TALDO1   AGCACAGATGCCCGCTTACC   \n",
       "11933   GAGGGCTTACCGGTGTCGGC     TKT   TCTGTAGCAACGCCATCACT   \n",
       "\n",
       "                            Sequence for oligo synthesis  \n",
       "0      tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...  \n",
       "1      tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...  \n",
       "2      tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...  \n",
       "3      tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...  \n",
       "4      tatatatcttgtggaaaggacgaaacACCGCTGGCTTGCGCCTTGC...  \n",
       "...                                                  ...  \n",
       "11929  tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...  \n",
       "11930  tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...  \n",
       "11931  tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...  \n",
       "11932  tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...  \n",
       "11933  tatatatcttgtggaaaggacgaaacACCGGAGGGCTTACCGGTGT...  \n",
       "\n",
       "[11934 rows x 6 columns]"
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "id": "294e5572",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_list = []\n",
    "seq_list = []\n",
    "for i in range(len(sgRNA_ref['SequenceID (GeneA_chr_position__GeneB_chr_position)'])):\n",
    "    probes = sgRNA_ref['SequenceID (GeneA_chr_position__GeneB_chr_position)'][i]\n",
    "    curr_A, curr_B = probes.split('__')\n",
    "    \n",
    "    if curr_A not in guide_list:\n",
    "        guide_list.append(curr_A)\n",
    "        seq_list.append(sgRNA_ref['Gene_A_spacer sequence'][i])\n",
    "        \n",
    "    if curr_B not in guide_list:\n",
    "        guide_list.append(curr_B)\n",
    "        seq_list.append(sgRNA_ref['Gene_B_spacer sequence'][i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "id": "0a5f755d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref = pd.DataFrame({\"Guide_ID\": guide_list,\n",
    "                         \"Sequence\": seq_list})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "id": "37939c49",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sgRNA_ref.copy()\n",
    "sequence_ref['Target'] = [i.split('_')[0] for i in sequence_ref['Guide_ID']]\n",
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "id": "3f4f81d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "idx = np.array([i for i in range(len(sequence_ref['sgRNA_guide_name'])) if 'HUMAN' in sequence_ref['sgRNA_guide_name'][i].upper()])\n",
    "sequence_ref.loc[idx, 'sgRNA_target_name'] = 'control'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "cad36ea1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>CTGGCTTGCGCCTTGCCATG</td>\n",
       "      <td>ALDOA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOB_chr9_104188893</td>\n",
       "      <td>GGTACCTATTGTTGAACCAG</td>\n",
       "      <td>ALDOB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOC_chr17_26901140</td>\n",
       "      <td>CTCCTCTGGGGTATACTTGA</td>\n",
       "      <td>ALDOC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOC_chr17_26902525</td>\n",
       "      <td>GAAAGGGCTGGGTACGAGTG</td>\n",
       "      <td>ALDOC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DLAT_chr11_111896198</td>\n",
       "      <td>TGTGGCGCGTCTGTGCGCGA</td>\n",
       "      <td>DLAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>LDHB_chr12_21796959</td>\n",
       "      <td>AGGGGAGAGTCGGCTCAATC</td>\n",
       "      <td>LDHB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>G6PD_chrX_153775081</td>\n",
       "      <td>CCCGCCCCCGCCGATTAAAT</td>\n",
       "      <td>G6PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>PGLS_chr19_17622472</td>\n",
       "      <td>CGCCCTCGCCATGGCCGCGC</td>\n",
       "      <td>PGLS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>PGAM1_chr10_99186191</td>\n",
       "      <td>CAGGCGCTACGAGGTGCGGA</td>\n",
       "      <td>PGAM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>PDHA2_chr4_96761286</td>\n",
       "      <td>GCACTCCGTGAAGAATATGC</td>\n",
       "      <td>PDHA2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>156 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0    ALDOA_chr16_30078215  CTGGCTTGCGCCTTGCCATG             ALDOA\n",
       "1    ALDOB_chr9_104188893  GGTACCTATTGTTGAACCAG             ALDOB\n",
       "2    ALDOC_chr17_26901140  CTCCTCTGGGGTATACTTGA             ALDOC\n",
       "3    ALDOC_chr17_26902525  GAAAGGGCTGGGTACGAGTG             ALDOC\n",
       "4    DLAT_chr11_111896198  TGTGGCGCGTCTGTGCGCGA              DLAT\n",
       "..                    ...                   ...               ...\n",
       "151   LDHB_chr12_21796959  AGGGGAGAGTCGGCTCAATC              LDHB\n",
       "152   G6PD_chrX_153775081  CCCGCCCCCGCCGATTAAAT              G6PD\n",
       "153   PGLS_chr19_17622472  CGCCCTCGCCATGGCCGCGC              PGLS\n",
       "154  PGAM1_chr10_99186191  CAGGCGCTACGAGGTGCGGA             PGAM1\n",
       "155   PDHA2_chr4_96761286  GCACTCCGTGAAGAATATGC             PDHA2\n",
       "\n",
       "[156 rows x 3 columns]"
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "13415acf",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "id": "d6daeace",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "id": "6ddcb357",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SequenceID (GeneA_chr_position__GeneB_chr_position)</th>\n",
       "      <th>MV4_plasmid_S7_trimmed53_len_filtered_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOB_chr9_104188893</td>\n",
       "      <td>442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26901140</td>\n",
       "      <td>1199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26902525</td>\n",
       "      <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111896198</td>\n",
       "      <td>1012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111899333</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836__RPIA_chr2_88991246</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_747467</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755880</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755937</td>\n",
       "      <td>600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836__TKT_chr3_53263054</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11934 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      SequenceID (GeneA_chr_position__GeneB_chr_position)  \\\n",
       "0             ALDOA_chr16_30078215__ALDOB_chr9_104188893    \n",
       "1             ALDOA_chr16_30078215__ALDOC_chr17_26901140    \n",
       "2             ALDOA_chr16_30078215__ALDOC_chr17_26902525    \n",
       "3             ALDOA_chr16_30078215__DLAT_chr11_111896198    \n",
       "4             ALDOA_chr16_30078215__DLAT_chr11_111899333    \n",
       "...                                                  ...    \n",
       "11929             TPI1_chr12_6976836__RPIA_chr2_88991246    \n",
       "11930            TPI1_chr12_6976836__TALDO1_chr11_747467    \n",
       "11931            TPI1_chr12_6976836__TALDO1_chr11_755880    \n",
       "11932            TPI1_chr12_6976836__TALDO1_chr11_755937    \n",
       "11933              TPI1_chr12_6976836__TKT_chr3_53263054    \n",
       "\n",
       "       MV4_plasmid_S7_trimmed53_len_filtered_counts  \n",
       "0                                               442  \n",
       "1                                              1199  \n",
       "2                                               284  \n",
       "3                                              1012  \n",
       "4                                               589  \n",
       "...                                             ...  \n",
       "11929                                           194  \n",
       "11930                                           167  \n",
       "11931                                           144  \n",
       "11932                                           600  \n",
       "11933                                           296  \n",
       "\n",
       "[11934 rows x 2 columns]"
      ]
     },
     "execution_count": 361,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Zhao_sgRNA_plasmid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "id": "d5cc22a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "Zhao_sgRNA_A549['MV4_plasmid_S7_trimmed53_len_filtered_counts'] = Zhao_sgRNA_plasmid['MV4_plasmid_S7_trimmed53_len_filtered_counts']\n",
    "Zhao_sgRNA_HELA['MV4_plasmid_S7_trimmed53_len_filtered_counts'] = Zhao_sgRNA_plasmid['MV4_plasmid_S7_trimmed53_len_filtered_counts']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "id": "93e44d66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>probeA</th>\n",
       "      <th>probeB</th>\n",
       "      <th>geneA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d3_2_S2_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d14_1_S3_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d14_2_S4_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d20_1_S5_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d20_2_S6_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d28_1_S7_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d28_2_S8_trimmed53_len_filtered_counts</th>\n",
       "      <th>MV4_plasmid_S7_trimmed53_len_filtered_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>697</td>\n",
       "      <td>526</td>\n",
       "      <td>761</td>\n",
       "      <td>1144</td>\n",
       "      <td>728</td>\n",
       "      <td>1107</td>\n",
       "      <td>852</td>\n",
       "      <td>794</td>\n",
       "      <td>442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>1370</td>\n",
       "      <td>1127</td>\n",
       "      <td>1452</td>\n",
       "      <td>1594</td>\n",
       "      <td>1434</td>\n",
       "      <td>1870</td>\n",
       "      <td>1771</td>\n",
       "      <td>917</td>\n",
       "      <td>1199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>492</td>\n",
       "      <td>465</td>\n",
       "      <td>380</td>\n",
       "      <td>604</td>\n",
       "      <td>266</td>\n",
       "      <td>708</td>\n",
       "      <td>267</td>\n",
       "      <td>383</td>\n",
       "      <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>1507</td>\n",
       "      <td>1170</td>\n",
       "      <td>1248</td>\n",
       "      <td>1444</td>\n",
       "      <td>968</td>\n",
       "      <td>1294</td>\n",
       "      <td>1214</td>\n",
       "      <td>824</td>\n",
       "      <td>1012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>678</td>\n",
       "      <td>635</td>\n",
       "      <td>643</td>\n",
       "      <td>771</td>\n",
       "      <td>537</td>\n",
       "      <td>887</td>\n",
       "      <td>378</td>\n",
       "      <td>517</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836__RPIA_chr2_88991246</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>RPIA_chr2_88991246</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>267</td>\n",
       "      <td>355</td>\n",
       "      <td>147</td>\n",
       "      <td>139</td>\n",
       "      <td>178</td>\n",
       "      <td>70</td>\n",
       "      <td>254</td>\n",
       "      <td>97</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_747467</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_747467</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>194</td>\n",
       "      <td>221</td>\n",
       "      <td>109</td>\n",
       "      <td>139</td>\n",
       "      <td>220</td>\n",
       "      <td>39</td>\n",
       "      <td>294</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755880</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755880</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>227</td>\n",
       "      <td>365</td>\n",
       "      <td>224</td>\n",
       "      <td>292</td>\n",
       "      <td>161</td>\n",
       "      <td>169</td>\n",
       "      <td>188</td>\n",
       "      <td>64</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755937</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755937</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>418</td>\n",
       "      <td>681</td>\n",
       "      <td>526</td>\n",
       "      <td>772</td>\n",
       "      <td>531</td>\n",
       "      <td>1127</td>\n",
       "      <td>640</td>\n",
       "      <td>954</td>\n",
       "      <td>600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836__TKT_chr3_53263054</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TKT_chr3_53263054</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>587</td>\n",
       "      <td>607</td>\n",
       "      <td>509</td>\n",
       "      <td>240</td>\n",
       "      <td>620</td>\n",
       "      <td>567</td>\n",
       "      <td>632</td>\n",
       "      <td>286</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11934 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ID                probeA  \\\n",
       "0      ALDOA_chr16_30078215__ALDOB_chr9_104188893  ALDOA_chr16_30078215   \n",
       "1      ALDOA_chr16_30078215__ALDOC_chr17_26901140  ALDOA_chr16_30078215   \n",
       "2      ALDOA_chr16_30078215__ALDOC_chr17_26902525  ALDOA_chr16_30078215   \n",
       "3      ALDOA_chr16_30078215__DLAT_chr11_111896198  ALDOA_chr16_30078215   \n",
       "4      ALDOA_chr16_30078215__DLAT_chr11_111899333  ALDOA_chr16_30078215   \n",
       "...                                           ...                   ...   \n",
       "11929      TPI1_chr12_6976836__RPIA_chr2_88991246    TPI1_chr12_6976836   \n",
       "11930     TPI1_chr12_6976836__TALDO1_chr11_747467    TPI1_chr12_6976836   \n",
       "11931     TPI1_chr12_6976836__TALDO1_chr11_755880    TPI1_chr12_6976836   \n",
       "11932     TPI1_chr12_6976836__TALDO1_chr11_755937    TPI1_chr12_6976836   \n",
       "11933       TPI1_chr12_6976836__TKT_chr3_53263054    TPI1_chr12_6976836   \n",
       "\n",
       "                     probeB  geneA   geneB  \\\n",
       "0      ALDOB_chr9_104188893  ALDOA   ALDOB   \n",
       "1      ALDOC_chr17_26901140  ALDOA   ALDOC   \n",
       "2      ALDOC_chr17_26902525  ALDOA   ALDOC   \n",
       "3      DLAT_chr11_111896198  ALDOA    DLAT   \n",
       "4      DLAT_chr11_111899333  ALDOA    DLAT   \n",
       "...                     ...    ...     ...   \n",
       "11929    RPIA_chr2_88991246   TPI1    RPIA   \n",
       "11930   TALDO1_chr11_747467   TPI1  TALDO1   \n",
       "11931   TALDO1_chr11_755880   TPI1  TALDO1   \n",
       "11932   TALDO1_chr11_755937   TPI1  TALDO1   \n",
       "11933     TKT_chr3_53263054   TPI1     TKT   \n",
       "\n",
       "       A549_MV4_d3_1_S1_trimmed53_len_filtered_counts  \\\n",
       "0                                                 697   \n",
       "1                                                1370   \n",
       "2                                                 492   \n",
       "3                                                1507   \n",
       "4                                                 678   \n",
       "...                                               ...   \n",
       "11929                                             267   \n",
       "11930                                             194   \n",
       "11931                                             227   \n",
       "11932                                             418   \n",
       "11933                                             587   \n",
       "\n",
       "       A549_MV4_d3_2_S2_trimmed53_len_filtered_counts  \\\n",
       "0                                                 526   \n",
       "1                                                1127   \n",
       "2                                                 465   \n",
       "3                                                1170   \n",
       "4                                                 635   \n",
       "...                                               ...   \n",
       "11929                                             355   \n",
       "11930                                             221   \n",
       "11931                                             365   \n",
       "11932                                             681   \n",
       "11933                                             607   \n",
       "\n",
       "       A549_MV4_d14_1_S3_trimmed53_len_filtered_counts  \\\n",
       "0                                                  761   \n",
       "1                                                 1452   \n",
       "2                                                  380   \n",
       "3                                                 1248   \n",
       "4                                                  643   \n",
       "...                                                ...   \n",
       "11929                                              147   \n",
       "11930                                              109   \n",
       "11931                                              224   \n",
       "11932                                              526   \n",
       "11933                                              509   \n",
       "\n",
       "       A549_MV4_d14_2_S4_trimmed53_len_filtered_counts  \\\n",
       "0                                                 1144   \n",
       "1                                                 1594   \n",
       "2                                                  604   \n",
       "3                                                 1444   \n",
       "4                                                  771   \n",
       "...                                                ...   \n",
       "11929                                              139   \n",
       "11930                                              139   \n",
       "11931                                              292   \n",
       "11932                                              772   \n",
       "11933                                              240   \n",
       "\n",
       "       A549_MV4_d20_1_S5_trimmed53_len_filtered_counts  \\\n",
       "0                                                  728   \n",
       "1                                                 1434   \n",
       "2                                                  266   \n",
       "3                                                  968   \n",
       "4                                                  537   \n",
       "...                                                ...   \n",
       "11929                                              178   \n",
       "11930                                              220   \n",
       "11931                                              161   \n",
       "11932                                              531   \n",
       "11933                                              620   \n",
       "\n",
       "       A549_MV4_d20_2_S6_trimmed53_len_filtered_counts  \\\n",
       "0                                                 1107   \n",
       "1                                                 1870   \n",
       "2                                                  708   \n",
       "3                                                 1294   \n",
       "4                                                  887   \n",
       "...                                                ...   \n",
       "11929                                               70   \n",
       "11930                                               39   \n",
       "11931                                              169   \n",
       "11932                                             1127   \n",
       "11933                                              567   \n",
       "\n",
       "       A549_MV4_d28_1_S7_trimmed53_len_filtered_counts  \\\n",
       "0                                                  852   \n",
       "1                                                 1771   \n",
       "2                                                  267   \n",
       "3                                                 1214   \n",
       "4                                                  378   \n",
       "...                                                ...   \n",
       "11929                                              254   \n",
       "11930                                              294   \n",
       "11931                                              188   \n",
       "11932                                              640   \n",
       "11933                                              632   \n",
       "\n",
       "       A549_MV4_d28_2_S8_trimmed53_len_filtered_counts  \\\n",
       "0                                                  794   \n",
       "1                                                  917   \n",
       "2                                                  383   \n",
       "3                                                  824   \n",
       "4                                                  517   \n",
       "...                                                ...   \n",
       "11929                                               97   \n",
       "11930                                                1   \n",
       "11931                                               64   \n",
       "11932                                              954   \n",
       "11933                                              286   \n",
       "\n",
       "       MV4_plasmid_S7_trimmed53_len_filtered_counts  \n",
       "0                                               442  \n",
       "1                                              1199  \n",
       "2                                               284  \n",
       "3                                              1012  \n",
       "4                                               589  \n",
       "...                                             ...  \n",
       "11929                                           194  \n",
       "11930                                           167  \n",
       "11931                                           144  \n",
       "11932                                           600  \n",
       "11933                                           296  \n",
       "\n",
       "[11934 rows x 14 columns]"
      ]
     },
     "execution_count": 363,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Zhao_sgRNA_A549"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "id": "e00f10b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts = Zhao_sgRNA_A549.copy()\n",
    "curr_counts = curr_counts.drop([\"ID\"], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "id": "0d7dfb51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['probeA', 'probeB', 'geneA', 'geneB'], dtype='object')"
      ]
     },
     "execution_count": 365,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts.columns[:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "368994a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['A549_MV4_d3_1_S1_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d3_2_S2_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d14_1_S3_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d14_2_S4_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d20_1_S5_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d20_2_S6_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d28_1_S7_trimmed53_len_filtered_counts',\n",
       "       'A549_MV4_d28_2_S8_trimmed53_len_filtered_counts',\n",
       "       'MV4_plasmid_S7_trimmed53_len_filtered_counts'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 366,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts.columns[4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "id": "2e81aec0",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts.columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\"] + curr_counts.columns[4:].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "id": "893075b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "replicates = curr_counts.columns[4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "id": "1858b115",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts['Count Replicates'] = curr_counts[replicates].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "curr_counts = curr_counts.drop(replicates, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "id": "702bff5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts[\"Condition\"] = [';'.join(replicates)] * curr_counts.shape[0]\n",
    "curr_counts[\"Cell Line\"] = [\"A549\"] * curr_counts.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "id": "0c14582c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Cell Line</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>697;526;761;1144;728;1107;852;794;442</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>1370;1127;1452;1594;1434;1870;1771;917;1199</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>492;465;380;604;266;708;267;383;284</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>1507;1170;1248;1444;968;1294;1214;824;1012</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>678;635;643;771;537;887;378;517;589</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>RPIA_chr2_88991246</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>267;355;147;139;178;70;254;97;194</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_747467</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>194;221;109;139;220;39;294;1;167</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755880</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>227;365;224;292;161;169;188;64;144</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755937</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>418;681;526;772;531;1127;640;954;600</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TKT_chr3_53263054</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>587;607;509;240;620;567;632;286;296</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11934 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Guide 1               Guide 2 Gene 1  Gene 2  \\\n",
       "0      ALDOA_chr16_30078215  ALDOB_chr9_104188893  ALDOA   ALDOB   \n",
       "1      ALDOA_chr16_30078215  ALDOC_chr17_26901140  ALDOA   ALDOC   \n",
       "2      ALDOA_chr16_30078215  ALDOC_chr17_26902525  ALDOA   ALDOC   \n",
       "3      ALDOA_chr16_30078215  DLAT_chr11_111896198  ALDOA    DLAT   \n",
       "4      ALDOA_chr16_30078215  DLAT_chr11_111899333  ALDOA    DLAT   \n",
       "...                     ...                   ...    ...     ...   \n",
       "11929    TPI1_chr12_6976836    RPIA_chr2_88991246   TPI1    RPIA   \n",
       "11930    TPI1_chr12_6976836   TALDO1_chr11_747467   TPI1  TALDO1   \n",
       "11931    TPI1_chr12_6976836   TALDO1_chr11_755880   TPI1  TALDO1   \n",
       "11932    TPI1_chr12_6976836   TALDO1_chr11_755937   TPI1  TALDO1   \n",
       "11933    TPI1_chr12_6976836     TKT_chr3_53263054   TPI1     TKT   \n",
       "\n",
       "                                  Count Replicates  \\\n",
       "0            697;526;761;1144;728;1107;852;794;442   \n",
       "1      1370;1127;1452;1594;1434;1870;1771;917;1199   \n",
       "2              492;465;380;604;266;708;267;383;284   \n",
       "3       1507;1170;1248;1444;968;1294;1214;824;1012   \n",
       "4              678;635;643;771;537;887;378;517;589   \n",
       "...                                            ...   \n",
       "11929            267;355;147;139;178;70;254;97;194   \n",
       "11930             194;221;109;139;220;39;294;1;167   \n",
       "11931           227;365;224;292;161;169;188;64;144   \n",
       "11932         418;681;526;772;531;1127;640;954;600   \n",
       "11933          587;607;509;240;620;567;632;286;296   \n",
       "\n",
       "                                               Condition Cell Line  \n",
       "0      A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "1      A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "2      A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "3      A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "4      A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "...                                                  ...       ...  \n",
       "11929  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "11930  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "11931  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "11932  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "11933  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...      A549  \n",
       "\n",
       "[11934 rows x 7 columns]"
      ]
     },
     "execution_count": 371,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "id": "b5efbb93",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, curr_counts])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "id": "2348984f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>probeA</th>\n",
       "      <th>probeB</th>\n",
       "      <th>geneA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d3_2_S2_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d14_1_S3_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d14_2_S4_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d20_1_S5_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d20_2_S6_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d28_1_S7_trimmed53_len_filtered_counts</th>\n",
       "      <th>A549_MV4_d28_2_S8_trimmed53_len_filtered_counts</th>\n",
       "      <th>MV4_plasmid_S7_trimmed53_len_filtered_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>697</td>\n",
       "      <td>526</td>\n",
       "      <td>761</td>\n",
       "      <td>1144</td>\n",
       "      <td>728</td>\n",
       "      <td>1107</td>\n",
       "      <td>852</td>\n",
       "      <td>794</td>\n",
       "      <td>442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>1370</td>\n",
       "      <td>1127</td>\n",
       "      <td>1452</td>\n",
       "      <td>1594</td>\n",
       "      <td>1434</td>\n",
       "      <td>1870</td>\n",
       "      <td>1771</td>\n",
       "      <td>917</td>\n",
       "      <td>1199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>492</td>\n",
       "      <td>465</td>\n",
       "      <td>380</td>\n",
       "      <td>604</td>\n",
       "      <td>266</td>\n",
       "      <td>708</td>\n",
       "      <td>267</td>\n",
       "      <td>383</td>\n",
       "      <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>1507</td>\n",
       "      <td>1170</td>\n",
       "      <td>1248</td>\n",
       "      <td>1444</td>\n",
       "      <td>968</td>\n",
       "      <td>1294</td>\n",
       "      <td>1214</td>\n",
       "      <td>824</td>\n",
       "      <td>1012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>678</td>\n",
       "      <td>635</td>\n",
       "      <td>643</td>\n",
       "      <td>771</td>\n",
       "      <td>537</td>\n",
       "      <td>887</td>\n",
       "      <td>378</td>\n",
       "      <td>517</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836__RPIA_chr2_88991246</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>RPIA_chr2_88991246</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>267</td>\n",
       "      <td>355</td>\n",
       "      <td>147</td>\n",
       "      <td>139</td>\n",
       "      <td>178</td>\n",
       "      <td>70</td>\n",
       "      <td>254</td>\n",
       "      <td>97</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_747467</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_747467</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>194</td>\n",
       "      <td>221</td>\n",
       "      <td>109</td>\n",
       "      <td>139</td>\n",
       "      <td>220</td>\n",
       "      <td>39</td>\n",
       "      <td>294</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755880</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755880</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>227</td>\n",
       "      <td>365</td>\n",
       "      <td>224</td>\n",
       "      <td>292</td>\n",
       "      <td>161</td>\n",
       "      <td>169</td>\n",
       "      <td>188</td>\n",
       "      <td>64</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755937</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755937</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>418</td>\n",
       "      <td>681</td>\n",
       "      <td>526</td>\n",
       "      <td>772</td>\n",
       "      <td>531</td>\n",
       "      <td>1127</td>\n",
       "      <td>640</td>\n",
       "      <td>954</td>\n",
       "      <td>600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836__TKT_chr3_53263054</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TKT_chr3_53263054</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>587</td>\n",
       "      <td>607</td>\n",
       "      <td>509</td>\n",
       "      <td>240</td>\n",
       "      <td>620</td>\n",
       "      <td>567</td>\n",
       "      <td>632</td>\n",
       "      <td>286</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11934 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ID                probeA  \\\n",
       "0      ALDOA_chr16_30078215__ALDOB_chr9_104188893  ALDOA_chr16_30078215   \n",
       "1      ALDOA_chr16_30078215__ALDOC_chr17_26901140  ALDOA_chr16_30078215   \n",
       "2      ALDOA_chr16_30078215__ALDOC_chr17_26902525  ALDOA_chr16_30078215   \n",
       "3      ALDOA_chr16_30078215__DLAT_chr11_111896198  ALDOA_chr16_30078215   \n",
       "4      ALDOA_chr16_30078215__DLAT_chr11_111899333  ALDOA_chr16_30078215   \n",
       "...                                           ...                   ...   \n",
       "11929      TPI1_chr12_6976836__RPIA_chr2_88991246    TPI1_chr12_6976836   \n",
       "11930     TPI1_chr12_6976836__TALDO1_chr11_747467    TPI1_chr12_6976836   \n",
       "11931     TPI1_chr12_6976836__TALDO1_chr11_755880    TPI1_chr12_6976836   \n",
       "11932     TPI1_chr12_6976836__TALDO1_chr11_755937    TPI1_chr12_6976836   \n",
       "11933       TPI1_chr12_6976836__TKT_chr3_53263054    TPI1_chr12_6976836   \n",
       "\n",
       "                     probeB  geneA   geneB  \\\n",
       "0      ALDOB_chr9_104188893  ALDOA   ALDOB   \n",
       "1      ALDOC_chr17_26901140  ALDOA   ALDOC   \n",
       "2      ALDOC_chr17_26902525  ALDOA   ALDOC   \n",
       "3      DLAT_chr11_111896198  ALDOA    DLAT   \n",
       "4      DLAT_chr11_111899333  ALDOA    DLAT   \n",
       "...                     ...    ...     ...   \n",
       "11929    RPIA_chr2_88991246   TPI1    RPIA   \n",
       "11930   TALDO1_chr11_747467   TPI1  TALDO1   \n",
       "11931   TALDO1_chr11_755880   TPI1  TALDO1   \n",
       "11932   TALDO1_chr11_755937   TPI1  TALDO1   \n",
       "11933     TKT_chr3_53263054   TPI1     TKT   \n",
       "\n",
       "       A549_MV4_d3_1_S1_trimmed53_len_filtered_counts  \\\n",
       "0                                                 697   \n",
       "1                                                1370   \n",
       "2                                                 492   \n",
       "3                                                1507   \n",
       "4                                                 678   \n",
       "...                                               ...   \n",
       "11929                                             267   \n",
       "11930                                             194   \n",
       "11931                                             227   \n",
       "11932                                             418   \n",
       "11933                                             587   \n",
       "\n",
       "       A549_MV4_d3_2_S2_trimmed53_len_filtered_counts  \\\n",
       "0                                                 526   \n",
       "1                                                1127   \n",
       "2                                                 465   \n",
       "3                                                1170   \n",
       "4                                                 635   \n",
       "...                                               ...   \n",
       "11929                                             355   \n",
       "11930                                             221   \n",
       "11931                                             365   \n",
       "11932                                             681   \n",
       "11933                                             607   \n",
       "\n",
       "       A549_MV4_d14_1_S3_trimmed53_len_filtered_counts  \\\n",
       "0                                                  761   \n",
       "1                                                 1452   \n",
       "2                                                  380   \n",
       "3                                                 1248   \n",
       "4                                                  643   \n",
       "...                                                ...   \n",
       "11929                                              147   \n",
       "11930                                              109   \n",
       "11931                                              224   \n",
       "11932                                              526   \n",
       "11933                                              509   \n",
       "\n",
       "       A549_MV4_d14_2_S4_trimmed53_len_filtered_counts  \\\n",
       "0                                                 1144   \n",
       "1                                                 1594   \n",
       "2                                                  604   \n",
       "3                                                 1444   \n",
       "4                                                  771   \n",
       "...                                                ...   \n",
       "11929                                              139   \n",
       "11930                                              139   \n",
       "11931                                              292   \n",
       "11932                                              772   \n",
       "11933                                              240   \n",
       "\n",
       "       A549_MV4_d20_1_S5_trimmed53_len_filtered_counts  \\\n",
       "0                                                  728   \n",
       "1                                                 1434   \n",
       "2                                                  266   \n",
       "3                                                  968   \n",
       "4                                                  537   \n",
       "...                                                ...   \n",
       "11929                                              178   \n",
       "11930                                              220   \n",
       "11931                                              161   \n",
       "11932                                              531   \n",
       "11933                                              620   \n",
       "\n",
       "       A549_MV4_d20_2_S6_trimmed53_len_filtered_counts  \\\n",
       "0                                                 1107   \n",
       "1                                                 1870   \n",
       "2                                                  708   \n",
       "3                                                 1294   \n",
       "4                                                  887   \n",
       "...                                                ...   \n",
       "11929                                               70   \n",
       "11930                                               39   \n",
       "11931                                              169   \n",
       "11932                                             1127   \n",
       "11933                                              567   \n",
       "\n",
       "       A549_MV4_d28_1_S7_trimmed53_len_filtered_counts  \\\n",
       "0                                                  852   \n",
       "1                                                 1771   \n",
       "2                                                  267   \n",
       "3                                                 1214   \n",
       "4                                                  378   \n",
       "...                                                ...   \n",
       "11929                                              254   \n",
       "11930                                              294   \n",
       "11931                                              188   \n",
       "11932                                              640   \n",
       "11933                                              632   \n",
       "\n",
       "       A549_MV4_d28_2_S8_trimmed53_len_filtered_counts  \\\n",
       "0                                                  794   \n",
       "1                                                  917   \n",
       "2                                                  383   \n",
       "3                                                  824   \n",
       "4                                                  517   \n",
       "...                                                ...   \n",
       "11929                                               97   \n",
       "11930                                                1   \n",
       "11931                                               64   \n",
       "11932                                              954   \n",
       "11933                                              286   \n",
       "\n",
       "       MV4_plasmid_S7_trimmed53_len_filtered_counts  \n",
       "0                                               442  \n",
       "1                                              1199  \n",
       "2                                               284  \n",
       "3                                              1012  \n",
       "4                                               589  \n",
       "...                                             ...  \n",
       "11929                                           194  \n",
       "11930                                           167  \n",
       "11931                                           144  \n",
       "11932                                           600  \n",
       "11933                                           296  \n",
       "\n",
       "[11934 rows x 14 columns]"
      ]
     },
     "execution_count": 373,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Zhao_sgRNA_A549"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "id": "971173b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>probeA</th>\n",
       "      <th>probeB</th>\n",
       "      <th>geneA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d3_2_S2_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d14_1_S3_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d14_2_S4_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d20_1_S9_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d20_2_S10_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d28_1_S5_trimmed53_len_filtered_counts</th>\n",
       "      <th>Hela_MV4_d28_2_S6_trimmed53_len_filtered_counts</th>\n",
       "      <th>MV4_plasmid_S7_trimmed53_len_filtered_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>640</td>\n",
       "      <td>422</td>\n",
       "      <td>621</td>\n",
       "      <td>563</td>\n",
       "      <td>818</td>\n",
       "      <td>450</td>\n",
       "      <td>782</td>\n",
       "      <td>367</td>\n",
       "      <td>442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>1026</td>\n",
       "      <td>736</td>\n",
       "      <td>1455</td>\n",
       "      <td>1263</td>\n",
       "      <td>1504</td>\n",
       "      <td>2381</td>\n",
       "      <td>1358</td>\n",
       "      <td>1371</td>\n",
       "      <td>1199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215__ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>305</td>\n",
       "      <td>365</td>\n",
       "      <td>375</td>\n",
       "      <td>397</td>\n",
       "      <td>434</td>\n",
       "      <td>806</td>\n",
       "      <td>319</td>\n",
       "      <td>560</td>\n",
       "      <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>1008</td>\n",
       "      <td>947</td>\n",
       "      <td>1201</td>\n",
       "      <td>1283</td>\n",
       "      <td>1859</td>\n",
       "      <td>2112</td>\n",
       "      <td>1562</td>\n",
       "      <td>1604</td>\n",
       "      <td>1012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215__DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>206</td>\n",
       "      <td>620</td>\n",
       "      <td>885</td>\n",
       "      <td>855</td>\n",
       "      <td>958</td>\n",
       "      <td>1159</td>\n",
       "      <td>693</td>\n",
       "      <td>648</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836__RPIA_chr2_88991246</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>RPIA_chr2_88991246</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>206</td>\n",
       "      <td>133</td>\n",
       "      <td>95</td>\n",
       "      <td>34</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_747467</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_747467</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>177</td>\n",
       "      <td>242</td>\n",
       "      <td>102</td>\n",
       "      <td>90</td>\n",
       "      <td>58</td>\n",
       "      <td>81</td>\n",
       "      <td>11</td>\n",
       "      <td>38</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755880</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755880</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>115</td>\n",
       "      <td>161</td>\n",
       "      <td>49</td>\n",
       "      <td>95</td>\n",
       "      <td>11</td>\n",
       "      <td>43</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836__TALDO1_chr11_755937</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755937</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>504</td>\n",
       "      <td>487</td>\n",
       "      <td>399</td>\n",
       "      <td>401</td>\n",
       "      <td>280</td>\n",
       "      <td>288</td>\n",
       "      <td>230</td>\n",
       "      <td>217</td>\n",
       "      <td>600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836__TKT_chr3_53263054</td>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TKT_chr3_53263054</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>414</td>\n",
       "      <td>337</td>\n",
       "      <td>93</td>\n",
       "      <td>126</td>\n",
       "      <td>78</td>\n",
       "      <td>113</td>\n",
       "      <td>31</td>\n",
       "      <td>93</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11934 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ID                probeA  \\\n",
       "0      ALDOA_chr16_30078215__ALDOB_chr9_104188893  ALDOA_chr16_30078215   \n",
       "1      ALDOA_chr16_30078215__ALDOC_chr17_26901140  ALDOA_chr16_30078215   \n",
       "2      ALDOA_chr16_30078215__ALDOC_chr17_26902525  ALDOA_chr16_30078215   \n",
       "3      ALDOA_chr16_30078215__DLAT_chr11_111896198  ALDOA_chr16_30078215   \n",
       "4      ALDOA_chr16_30078215__DLAT_chr11_111899333  ALDOA_chr16_30078215   \n",
       "...                                           ...                   ...   \n",
       "11929      TPI1_chr12_6976836__RPIA_chr2_88991246    TPI1_chr12_6976836   \n",
       "11930     TPI1_chr12_6976836__TALDO1_chr11_747467    TPI1_chr12_6976836   \n",
       "11931     TPI1_chr12_6976836__TALDO1_chr11_755880    TPI1_chr12_6976836   \n",
       "11932     TPI1_chr12_6976836__TALDO1_chr11_755937    TPI1_chr12_6976836   \n",
       "11933       TPI1_chr12_6976836__TKT_chr3_53263054    TPI1_chr12_6976836   \n",
       "\n",
       "                     probeB  geneA   geneB  \\\n",
       "0      ALDOB_chr9_104188893  ALDOA   ALDOB   \n",
       "1      ALDOC_chr17_26901140  ALDOA   ALDOC   \n",
       "2      ALDOC_chr17_26902525  ALDOA   ALDOC   \n",
       "3      DLAT_chr11_111896198  ALDOA    DLAT   \n",
       "4      DLAT_chr11_111899333  ALDOA    DLAT   \n",
       "...                     ...    ...     ...   \n",
       "11929    RPIA_chr2_88991246   TPI1    RPIA   \n",
       "11930   TALDO1_chr11_747467   TPI1  TALDO1   \n",
       "11931   TALDO1_chr11_755880   TPI1  TALDO1   \n",
       "11932   TALDO1_chr11_755937   TPI1  TALDO1   \n",
       "11933     TKT_chr3_53263054   TPI1     TKT   \n",
       "\n",
       "       Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts  \\\n",
       "0                                                 640   \n",
       "1                                                1026   \n",
       "2                                                 305   \n",
       "3                                                1008   \n",
       "4                                                 206   \n",
       "...                                               ...   \n",
       "11929                                             206   \n",
       "11930                                             177   \n",
       "11931                                             115   \n",
       "11932                                             504   \n",
       "11933                                             414   \n",
       "\n",
       "       Hela_MV4_d3_2_S2_trimmed53_len_filtered_counts  \\\n",
       "0                                                 422   \n",
       "1                                                 736   \n",
       "2                                                 365   \n",
       "3                                                 947   \n",
       "4                                                 620   \n",
       "...                                               ...   \n",
       "11929                                             133   \n",
       "11930                                             242   \n",
       "11931                                             161   \n",
       "11932                                             487   \n",
       "11933                                             337   \n",
       "\n",
       "       Hela_MV4_d14_1_S3_trimmed53_len_filtered_counts  \\\n",
       "0                                                  621   \n",
       "1                                                 1455   \n",
       "2                                                  375   \n",
       "3                                                 1201   \n",
       "4                                                  885   \n",
       "...                                                ...   \n",
       "11929                                               95   \n",
       "11930                                              102   \n",
       "11931                                               49   \n",
       "11932                                              399   \n",
       "11933                                               93   \n",
       "\n",
       "       Hela_MV4_d14_2_S4_trimmed53_len_filtered_counts  \\\n",
       "0                                                  563   \n",
       "1                                                 1263   \n",
       "2                                                  397   \n",
       "3                                                 1283   \n",
       "4                                                  855   \n",
       "...                                                ...   \n",
       "11929                                               34   \n",
       "11930                                               90   \n",
       "11931                                               95   \n",
       "11932                                              401   \n",
       "11933                                              126   \n",
       "\n",
       "       Hela_MV4_d20_1_S9_trimmed53_len_filtered_counts  \\\n",
       "0                                                  818   \n",
       "1                                                 1504   \n",
       "2                                                  434   \n",
       "3                                                 1859   \n",
       "4                                                  958   \n",
       "...                                                ...   \n",
       "11929                                               56   \n",
       "11930                                               58   \n",
       "11931                                               11   \n",
       "11932                                              280   \n",
       "11933                                               78   \n",
       "\n",
       "       Hela_MV4_d20_2_S10_trimmed53_len_filtered_counts  \\\n",
       "0                                                   450   \n",
       "1                                                  2381   \n",
       "2                                                   806   \n",
       "3                                                  2112   \n",
       "4                                                  1159   \n",
       "...                                                 ...   \n",
       "11929                                                47   \n",
       "11930                                                81   \n",
       "11931                                                43   \n",
       "11932                                               288   \n",
       "11933                                               113   \n",
       "\n",
       "       Hela_MV4_d28_1_S5_trimmed53_len_filtered_counts  \\\n",
       "0                                                  782   \n",
       "1                                                 1358   \n",
       "2                                                  319   \n",
       "3                                                 1562   \n",
       "4                                                  693   \n",
       "...                                                ...   \n",
       "11929                                                2   \n",
       "11930                                               11   \n",
       "11931                                               16   \n",
       "11932                                              230   \n",
       "11933                                               31   \n",
       "\n",
       "       Hela_MV4_d28_2_S6_trimmed53_len_filtered_counts  \\\n",
       "0                                                  367   \n",
       "1                                                 1371   \n",
       "2                                                  560   \n",
       "3                                                 1604   \n",
       "4                                                  648   \n",
       "...                                                ...   \n",
       "11929                                               19   \n",
       "11930                                               38   \n",
       "11931                                                5   \n",
       "11932                                              217   \n",
       "11933                                               93   \n",
       "\n",
       "       MV4_plasmid_S7_trimmed53_len_filtered_counts  \n",
       "0                                               442  \n",
       "1                                              1199  \n",
       "2                                               284  \n",
       "3                                              1012  \n",
       "4                                               589  \n",
       "...                                             ...  \n",
       "11929                                           194  \n",
       "11930                                           167  \n",
       "11931                                           144  \n",
       "11932                                           600  \n",
       "11933                                           296  \n",
       "\n",
       "[11934 rows x 14 columns]"
      ]
     },
     "execution_count": 374,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Zhao_sgRNA_HELA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "id": "e2a62ee8",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts = Zhao_sgRNA_HELA.copy()\n",
    "curr_counts = curr_counts.drop([\"ID\"], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "id": "4e736283",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['probeA', 'probeB', 'geneA', 'geneB'], dtype='object')"
      ]
     },
     "execution_count": 376,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts.columns[:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "id": "311d4420",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d3_2_S2_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d14_1_S3_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d14_2_S4_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d20_1_S9_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d20_2_S10_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d28_1_S5_trimmed53_len_filtered_counts',\n",
       "       'Hela_MV4_d28_2_S6_trimmed53_len_filtered_counts',\n",
       "       'MV4_plasmid_S7_trimmed53_len_filtered_counts'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 377,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_counts.columns[4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "id": "6c32a985",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts.columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\"] + curr_counts.columns[4:].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "id": "f78af8d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "replicates = curr_counts.columns[4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "id": "57dde0d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts['Count Replicates'] = curr_counts[replicates].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "curr_counts = curr_counts.drop(replicates, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "e1608201",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_counts[\"Condition\"] = [';'.join(replicates)] * curr_counts.shape[0]\n",
    "curr_counts[\"Cell Line\"] = [\"HELA\"] * curr_counts.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "id": "d71bc74a",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, curr_counts])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "id": "9deeb956",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref[\"Sequencing\"] = [\"Combinatorial CRISPR\"] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "id": "a3a5e243",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref.loc[counts_ref['Gene 1'] == 0, 'Gene 1'] = 'control'\n",
    "counts_ref.loc[counts_ref['Gene 2'] == 0, 'Gene 2'] = 'control'\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['zhao_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "id": "b3959fb3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOB_chr9_104188893</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>697;526;761;1144;728;1107;852;794;442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26901140</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>1370;1127;1452;1594;1434;1870;1771;917;1199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>ALDOC_chr17_26902525</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>492;465;380;604;266;708;267;383;284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111896198</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>1507;1170;1248;1444;968;1294;1214;824;1012</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_chr16_30078215</td>\n",
       "      <td>DLAT_chr11_111899333</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>678;635;643;771;537;887;378;517;589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>A549</td>\n",
       "      <td>A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11929</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>RPIA_chr2_88991246</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>206;133;95;34;56;47;2;19;194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_747467</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>177;242;102;90;58;81;11;38;167</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11931</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755880</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>115;161;49;95;11;43;16;5;144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TALDO1_chr11_755937</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>504;487;399;401;280;288;230;217;600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11933</th>\n",
       "      <td>TPI1_chr12_6976836</td>\n",
       "      <td>TKT_chr3_53263054</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>414;337;93;126;78;113;31;93;296</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>HELA</td>\n",
       "      <td>Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...</td>\n",
       "      <td>29452643</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23868 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Guide 1               Guide 2 Gene 1  Gene 2  \\\n",
       "0      ALDOA_chr16_30078215  ALDOB_chr9_104188893  ALDOA   ALDOB   \n",
       "1      ALDOA_chr16_30078215  ALDOC_chr17_26901140  ALDOA   ALDOC   \n",
       "2      ALDOA_chr16_30078215  ALDOC_chr17_26902525  ALDOA   ALDOC   \n",
       "3      ALDOA_chr16_30078215  DLAT_chr11_111896198  ALDOA    DLAT   \n",
       "4      ALDOA_chr16_30078215  DLAT_chr11_111899333  ALDOA    DLAT   \n",
       "...                     ...                   ...    ...     ...   \n",
       "11929    TPI1_chr12_6976836    RPIA_chr2_88991246   TPI1    RPIA   \n",
       "11930    TPI1_chr12_6976836   TALDO1_chr11_747467   TPI1  TALDO1   \n",
       "11931    TPI1_chr12_6976836   TALDO1_chr11_755880   TPI1  TALDO1   \n",
       "11932    TPI1_chr12_6976836   TALDO1_chr11_755937   TPI1  TALDO1   \n",
       "11933    TPI1_chr12_6976836     TKT_chr3_53263054   TPI1     TKT   \n",
       "\n",
       "                                  Count Replicates Type            Sequencing  \\\n",
       "0            697;526;761;1144;728;1107;852;794;442  NaN  Combinatorial CRISPR   \n",
       "1      1370;1127;1452;1594;1434;1870;1771;917;1199  NaN  Combinatorial CRISPR   \n",
       "2              492;465;380;604;266;708;267;383;284  NaN  Combinatorial CRISPR   \n",
       "3       1507;1170;1248;1444;968;1294;1214;824;1012  NaN  Combinatorial CRISPR   \n",
       "4              678;635;643;771;537;887;378;517;589  NaN  Combinatorial CRISPR   \n",
       "...                                            ...  ...                   ...   \n",
       "11929                 206;133;95;34;56;47;2;19;194  NaN  Combinatorial CRISPR   \n",
       "11930               177;242;102;90;58;81;11;38;167  NaN  Combinatorial CRISPR   \n",
       "11931                 115;161;49;95;11;43;16;5;144  NaN  Combinatorial CRISPR   \n",
       "11932          504;487;399;401;280;288;230;217;600  NaN  Combinatorial CRISPR   \n",
       "11933              414;337;93;126;78;113;31;93;296  NaN  Combinatorial CRISPR   \n",
       "\n",
       "      Cell Line                                          Condition     Study  \n",
       "0          A549  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "1          A549  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "2          A549  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "3          A549  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "4          A549  A549_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "...         ...                                                ...       ...  \n",
       "11929      HELA  Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "11930      HELA  Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "11931      HELA  Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "11932      HELA  Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "11933      HELA  Hela_MV4_d3_1_S1_trimmed53_len_filtered_counts...  29452643  \n",
       "\n",
       "[23868 rows x 10 columns]"
      ]
     },
     "execution_count": 385,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e69e4dd2",
   "metadata": {},
   "source": [
    "## SL Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "id": "65757609",
   "metadata": {},
   "outputs": [],
   "source": [
    "zhao_loc = os.path.join(learning_goals_loc_general, \"Zhao\", 'created')\n",
    "\n",
    "Zhao_A549 =  pd.read_excel(os.path.join(zhao_loc, \"Zhao_GI.xlsx\"), sheet_name = \"A549 pi scores\")\n",
    "Zhao_HELA =  pd.read_excel(os.path.join(zhao_loc, \"Zhao_GI.xlsx\"), sheet_name = \"HeLa pi socres\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 387,
   "id": "3b3c3a15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_gene</th>\n",
       "      <th>geneA</th>\n",
       "      <th>fA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>fB</th>\n",
       "      <th>fA+fB</th>\n",
       "      <th>pi</th>\n",
       "      <th>sd</th>\n",
       "      <th>PP</th>\n",
       "      <th>abs pi</th>\n",
       "      <th>FDR left</th>\n",
       "      <th>FDR right</th>\n",
       "      <th>z</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_ALDOB</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.047146</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>-0.005978</td>\n",
       "      <td>-0.053124</td>\n",
       "      <td>0.004084</td>\n",
       "      <td>0.018323</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.004084</td>\n",
       "      <td>0.761518</td>\n",
       "      <td>1</td>\n",
       "      <td>0.169437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_ALDOC</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.047146</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>-0.018737</td>\n",
       "      <td>-0.065883</td>\n",
       "      <td>-0.058298</td>\n",
       "      <td>0.034592</td>\n",
       "      <td>0.912</td>\n",
       "      <td>0.058298</td>\n",
       "      <td>0.205590</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.418505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_DLAT</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.047146</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>-0.012969</td>\n",
       "      <td>-0.060115</td>\n",
       "      <td>-0.071055</td>\n",
       "      <td>0.037351</td>\n",
       "      <td>0.949</td>\n",
       "      <td>0.071055</td>\n",
       "      <td>0.178348</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.947731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_DLD</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.047146</td>\n",
       "      <td>DLD</td>\n",
       "      <td>-0.027896</td>\n",
       "      <td>-0.075042</td>\n",
       "      <td>-0.065549</td>\n",
       "      <td>0.016643</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.065549</td>\n",
       "      <td>0.184315</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.719313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_ENO1</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.047146</td>\n",
       "      <td>ENO1</td>\n",
       "      <td>-0.044006</td>\n",
       "      <td>-0.091152</td>\n",
       "      <td>-0.048887</td>\n",
       "      <td>0.024589</td>\n",
       "      <td>0.958</td>\n",
       "      <td>0.048887</td>\n",
       "      <td>0.251583</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.028083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>RPIA_TKT</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>-0.030796</td>\n",
       "      <td>TKT</td>\n",
       "      <td>-0.036411</td>\n",
       "      <td>-0.067207</td>\n",
       "      <td>-0.014492</td>\n",
       "      <td>0.020125</td>\n",
       "      <td>0.525</td>\n",
       "      <td>0.014492</td>\n",
       "      <td>0.462568</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.601182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1271</th>\n",
       "      <td>RPIA_TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>-0.030796</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>-0.030328</td>\n",
       "      <td>-0.061124</td>\n",
       "      <td>-0.026897</td>\n",
       "      <td>0.027609</td>\n",
       "      <td>0.675</td>\n",
       "      <td>0.026897</td>\n",
       "      <td>0.372878</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.115811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1272</th>\n",
       "      <td>TALDO1_TKT</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>-0.031732</td>\n",
       "      <td>TKT</td>\n",
       "      <td>-0.036411</td>\n",
       "      <td>-0.068143</td>\n",
       "      <td>-0.080038</td>\n",
       "      <td>0.031667</td>\n",
       "      <td>0.990</td>\n",
       "      <td>0.080038</td>\n",
       "      <td>0.130720</td>\n",
       "      <td>1</td>\n",
       "      <td>-3.320382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1273</th>\n",
       "      <td>TALDO1_TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>-0.031732</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>-0.030328</td>\n",
       "      <td>-0.062060</td>\n",
       "      <td>-0.056565</td>\n",
       "      <td>0.037941</td>\n",
       "      <td>0.860</td>\n",
       "      <td>0.056565</td>\n",
       "      <td>0.208709</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.346589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>TKT_TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>-0.036411</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>-0.030328</td>\n",
       "      <td>-0.066739</td>\n",
       "      <td>-0.069036</td>\n",
       "      <td>0.021031</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.069036</td>\n",
       "      <td>0.171769</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.863983</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1275 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_gene   geneA        fA  geneB        fB     fA+fB        pi  \\\n",
       "0     ALDOA_ALDOB   ALDOA -0.047146  ALDOB -0.005978 -0.053124  0.004084   \n",
       "1     ALDOA_ALDOC   ALDOA -0.047146  ALDOC -0.018737 -0.065883 -0.058298   \n",
       "2      ALDOA_DLAT   ALDOA -0.047146   DLAT -0.012969 -0.060115 -0.071055   \n",
       "3       ALDOA_DLD   ALDOA -0.047146    DLD -0.027896 -0.075042 -0.065549   \n",
       "4      ALDOA_ENO1   ALDOA -0.047146   ENO1 -0.044006 -0.091152 -0.048887   \n",
       "...           ...     ...       ...    ...       ...       ...       ...   \n",
       "1270     RPIA_TKT    RPIA -0.030796    TKT -0.036411 -0.067207 -0.014492   \n",
       "1271    RPIA_TPI1    RPIA -0.030796   TPI1 -0.030328 -0.061124 -0.026897   \n",
       "1272   TALDO1_TKT  TALDO1 -0.031732    TKT -0.036411 -0.068143 -0.080038   \n",
       "1273  TALDO1_TPI1  TALDO1 -0.031732   TPI1 -0.030328 -0.062060 -0.056565   \n",
       "1274     TKT_TPI1     TKT -0.036411   TPI1 -0.030328 -0.066739 -0.069036   \n",
       "\n",
       "            sd     PP    abs pi  FDR left  FDR right         z  \n",
       "0     0.018323  0.163  0.004084  0.761518          1  0.169437  \n",
       "1     0.034592  0.912  0.058298  0.205590          1 -2.418505  \n",
       "2     0.037351  0.949  0.071055  0.178348          1 -2.947731  \n",
       "3     0.016643  1.000  0.065549  0.184315          1 -2.719313  \n",
       "4     0.024589  0.958  0.048887  0.251583          1 -2.028083  \n",
       "...        ...    ...       ...       ...        ...       ...  \n",
       "1270  0.020125  0.525  0.014492  0.462568          1 -0.601182  \n",
       "1271  0.027609  0.675  0.026897  0.372878          1 -1.115811  \n",
       "1272  0.031667  0.990  0.080038  0.130720          1 -3.320382  \n",
       "1273  0.037941  0.860  0.056565  0.208709          1 -2.346589  \n",
       "1274  0.021031  0.998  0.069036  0.171769          1 -2.863983  \n",
       "\n",
       "[1275 rows x 13 columns]"
      ]
     },
     "execution_count": 387,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Zhao_A549"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "id": "29344494",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_gene</th>\n",
       "      <th>geneA</th>\n",
       "      <th>fA</th>\n",
       "      <th>geneB</th>\n",
       "      <th>fB</th>\n",
       "      <th>fA+fB</th>\n",
       "      <th>pi</th>\n",
       "      <th>sd</th>\n",
       "      <th>PP</th>\n",
       "      <th>abs pi</th>\n",
       "      <th>FDR left</th>\n",
       "      <th>FDR right</th>\n",
       "      <th>z</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDOA_ALDOB</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.090395</td>\n",
       "      <td>ALDOB</td>\n",
       "      <td>-0.023448</td>\n",
       "      <td>-0.113843</td>\n",
       "      <td>0.004951</td>\n",
       "      <td>0.022351</td>\n",
       "      <td>0.165</td>\n",
       "      <td>0.004951</td>\n",
       "      <td>0.790159</td>\n",
       "      <td>1</td>\n",
       "      <td>0.188401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALDOA_ALDOC</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.090395</td>\n",
       "      <td>ALDOC</td>\n",
       "      <td>-0.027559</td>\n",
       "      <td>-0.117955</td>\n",
       "      <td>-0.049581</td>\n",
       "      <td>0.049662</td>\n",
       "      <td>0.684</td>\n",
       "      <td>0.049581</td>\n",
       "      <td>0.369074</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.886880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALDOA_DLAT</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.090395</td>\n",
       "      <td>DLAT</td>\n",
       "      <td>0.006216</td>\n",
       "      <td>-0.084180</td>\n",
       "      <td>-0.055467</td>\n",
       "      <td>0.052478</td>\n",
       "      <td>0.697</td>\n",
       "      <td>0.055467</td>\n",
       "      <td>0.355821</td>\n",
       "      <td>1</td>\n",
       "      <td>-2.110882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALDOA_DLD</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.090395</td>\n",
       "      <td>DLD</td>\n",
       "      <td>-0.019896</td>\n",
       "      <td>-0.110292</td>\n",
       "      <td>-0.019820</td>\n",
       "      <td>0.026161</td>\n",
       "      <td>0.555</td>\n",
       "      <td>0.019820</td>\n",
       "      <td>0.517702</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.754265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALDOA_ENO1</td>\n",
       "      <td>ALDOA</td>\n",
       "      <td>-0.090395</td>\n",
       "      <td>ENO1</td>\n",
       "      <td>-0.090564</td>\n",
       "      <td>-0.180959</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>0.065412</td>\n",
       "      <td>0.046</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>0.787505</td>\n",
       "      <td>1</td>\n",
       "      <td>0.148370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>RPIA_TKT</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>-0.037973</td>\n",
       "      <td>TKT</td>\n",
       "      <td>-0.065134</td>\n",
       "      <td>-0.103107</td>\n",
       "      <td>-0.034815</td>\n",
       "      <td>0.029087</td>\n",
       "      <td>0.745</td>\n",
       "      <td>0.034815</td>\n",
       "      <td>0.402030</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.324924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1271</th>\n",
       "      <td>RPIA_TPI1</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>-0.037973</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>-0.089012</td>\n",
       "      <td>-0.126985</td>\n",
       "      <td>-0.024122</td>\n",
       "      <td>0.034493</td>\n",
       "      <td>0.519</td>\n",
       "      <td>0.024122</td>\n",
       "      <td>0.479355</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.918003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1272</th>\n",
       "      <td>TALDO1_TKT</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>-0.026272</td>\n",
       "      <td>TKT</td>\n",
       "      <td>-0.065134</td>\n",
       "      <td>-0.091406</td>\n",
       "      <td>-0.004685</td>\n",
       "      <td>0.021982</td>\n",
       "      <td>0.158</td>\n",
       "      <td>0.004685</td>\n",
       "      <td>0.689172</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.178284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1273</th>\n",
       "      <td>TALDO1_TPI1</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>-0.026272</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>-0.089012</td>\n",
       "      <td>-0.115284</td>\n",
       "      <td>-0.036088</td>\n",
       "      <td>0.034647</td>\n",
       "      <td>0.694</td>\n",
       "      <td>0.036088</td>\n",
       "      <td>0.401973</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.373376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>TKT_TPI1</td>\n",
       "      <td>TKT</td>\n",
       "      <td>-0.065134</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>-0.089012</td>\n",
       "      <td>-0.154145</td>\n",
       "      <td>0.006468</td>\n",
       "      <td>0.027732</td>\n",
       "      <td>0.204</td>\n",
       "      <td>0.006468</td>\n",
       "      <td>0.800446</td>\n",
       "      <td>1</td>\n",
       "      <td>0.246129</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1275 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_gene   geneA        fA  geneB        fB     fA+fB        pi  \\\n",
       "0     ALDOA_ALDOB   ALDOA -0.090395  ALDOB -0.023448 -0.113843  0.004951   \n",
       "1     ALDOA_ALDOC   ALDOA -0.090395  ALDOC -0.027559 -0.117955 -0.049581   \n",
       "2      ALDOA_DLAT   ALDOA -0.090395   DLAT  0.006216 -0.084180 -0.055467   \n",
       "3       ALDOA_DLD   ALDOA -0.090395    DLD -0.019896 -0.110292 -0.019820   \n",
       "4      ALDOA_ENO1   ALDOA -0.090395   ENO1 -0.090564 -0.180959  0.003899   \n",
       "...           ...     ...       ...    ...       ...       ...       ...   \n",
       "1270     RPIA_TKT    RPIA -0.037973    TKT -0.065134 -0.103107 -0.034815   \n",
       "1271    RPIA_TPI1    RPIA -0.037973   TPI1 -0.089012 -0.126985 -0.024122   \n",
       "1272   TALDO1_TKT  TALDO1 -0.026272    TKT -0.065134 -0.091406 -0.004685   \n",
       "1273  TALDO1_TPI1  TALDO1 -0.026272   TPI1 -0.089012 -0.115284 -0.036088   \n",
       "1274     TKT_TPI1     TKT -0.065134   TPI1 -0.089012 -0.154145  0.006468   \n",
       "\n",
       "            sd     PP    abs pi  FDR left  FDR right         z  \n",
       "0     0.022351  0.165  0.004951  0.790159          1  0.188401  \n",
       "1     0.049662  0.684  0.049581  0.369074          1 -1.886880  \n",
       "2     0.052478  0.697  0.055467  0.355821          1 -2.110882  \n",
       "3     0.026161  0.555  0.019820  0.517702          1 -0.754265  \n",
       "4     0.065412  0.046  0.003899  0.787505          1  0.148370  \n",
       "...        ...    ...       ...       ...        ...       ...  \n",
       "1270  0.029087  0.745  0.034815  0.402030          1 -1.324924  \n",
       "1271  0.034493  0.519  0.024122  0.479355          1 -0.918003  \n",
       "1272  0.021982  0.158  0.004685  0.689172          1 -0.178284  \n",
       "1273  0.034647  0.694  0.036088  0.401973          1 -1.373376  \n",
       "1274  0.027732  0.204  0.006468  0.800446          1  0.246129  \n",
       "\n",
       "[1275 rows x 13 columns]"
      ]
     },
     "execution_count": 388,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Zhao_HELA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "id": "4b53d6cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "id": "fa86b3c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = Zhao_A549[\"geneA\"]\n",
    "curr_GI[\"Gene_B\"] = Zhao_A549[\"geneB\"]\n",
    "curr_GI[\"GI_Score\"] = Zhao_A549[\"z\"]\n",
    "curr_GI[\"GI_Cutoff\"] = [-3] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['zhao_data']] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI[\"Cell_Line\"] = [\"A549\"] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"Stat_Score\"] = Zhao_A549.loc[:, \"FDR left\"]\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * len(curr_GI[\"Gene_A\"])\n",
    "\n",
    "all_GI = pd.concat([all_GI, curr_GI])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "id": "9e5e75c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = Zhao_HELA[\"geneA\"]\n",
    "curr_GI[\"Gene_B\"] = Zhao_HELA[\"geneB\"]\n",
    "curr_GI[\"GI_Score\"] = Zhao_HELA[\"z\"]\n",
    "curr_GI[\"GI_Cutoff\"] = [-3] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['zhao_data']] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI[\"Cell_Line\"] = [\"HELA\"] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"Stat_Score\"] = Zhao_HELA.loc[:, \"FDR left\"]\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * len(curr_GI[\"Gene_A\"])\n",
    "\n",
    "all_GI = pd.concat([all_GI, curr_GI])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "id": "097d2426",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "Only GI cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 22950\n",
      "Number of controls: 0\n",
      "Number of singles: 918\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = all_GI.copy(), study_controls = controls['zhao_data'], study_conditions = study_conditions['zhao_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 393,
   "id": "6a1233eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF['Study_Source'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 394,
   "id": "8869daf1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PREV_REF.loc[PREV_REF['Study_Source'] == 'Zhao', :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "id": "09f17140",
   "metadata": {},
   "outputs": [],
   "source": [
    "#db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "id": "bc47de37",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 156\n",
      "Counts insert: 23868\n",
      "Score insert: 2550\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3daa8be2",
   "metadata": {},
   "source": [
    "## ShanTang Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "id": "c6642b06",
   "metadata": {},
   "outputs": [],
   "source": [
    "shantang_loc = os.path.join(learning_goals_loc_general, \"ShanTang\")\n",
    "shan_data = pd.read_csv(os.path.join(shantang_loc, \"DKO_22RV1_RawCounts_1.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "c9f67582",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "      <th>T0_2</th>\n",
       "      <th>T0_3</th>\n",
       "      <th>T12_2</th>\n",
       "      <th>T12_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>22</td>\n",
       "      <td>17</td>\n",
       "      <td>36</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4525</td>\n",
       "      <td>GCATGGCCTCCACTTGCAA</td>\n",
       "      <td>57</td>\n",
       "      <td>73</td>\n",
       "      <td>107</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4590</td>\n",
       "      <td>GCACCAGTCTATGCCACCAC</td>\n",
       "      <td>78</td>\n",
       "      <td>55</td>\n",
       "      <td>124</td>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe_safe_DTKP</td>\n",
       "      <td>0Safe_safe_DTKP_204550.5261</td>\n",
       "      <td>GTGAATTTAAGGCACAACCC</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe_safe_ACOC</td>\n",
       "      <td>0Safe_safe_ACOC_204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe_safe_DTKP</td>\n",
       "      <td>0Safe_safe_DTKP_204550.5308</td>\n",
       "      <td>GCTGGACTATGCCAGGACCT</td>\n",
       "      <td>17</td>\n",
       "      <td>27</td>\n",
       "      <td>38</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60625</th>\n",
       "      <td>WNT5A_4__UPF3A_4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>UPF3A_4</td>\n",
       "      <td>GACCCAGTCTATCAAAAGACC</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "      <td>227</td>\n",
       "      <td>235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60626</th>\n",
       "      <td>WNT5A_4__WNT5A_1</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_1</td>\n",
       "      <td>GAGGATGCGAGCACTCTCGT</td>\n",
       "      <td>339</td>\n",
       "      <td>282</td>\n",
       "      <td>455</td>\n",
       "      <td>564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60627</th>\n",
       "      <td>WNT5A_4__WNT5A_2</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_2</td>\n",
       "      <td>GAGTATCAATTCCGACATCGA</td>\n",
       "      <td>288</td>\n",
       "      <td>308</td>\n",
       "      <td>448</td>\n",
       "      <td>597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60628</th>\n",
       "      <td>WNT5A_4__WNT5A_3</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_3</td>\n",
       "      <td>GAGTTAACTTCCAACAGGGGG</td>\n",
       "      <td>201</td>\n",
       "      <td>198</td>\n",
       "      <td>333</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60629</th>\n",
       "      <td>WNT5A_4__WNT5A_4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A_4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>198</td>\n",
       "      <td>204</td>\n",
       "      <td>328</td>\n",
       "      <td>492</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>60630 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id      target_a_id  \\\n",
       "0      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe_safe_ACOC   \n",
       "1      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe_safe_ACOC   \n",
       "2      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe_safe_ACOC   \n",
       "3      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...  0Safe_safe_ACOC   \n",
       "4      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...  0Safe_safe_ACOC   \n",
       "...                                                  ...              ...   \n",
       "60625                                   WNT5A_4__UPF3A_4            WNT5A   \n",
       "60626                                   WNT5A_4__WNT5A_1            WNT5A   \n",
       "60627                                   WNT5A_4__WNT5A_2            WNT5A   \n",
       "60628                                   WNT5A_4__WNT5A_3            WNT5A   \n",
       "60629                                   WNT5A_4__WNT5A_4            WNT5A   \n",
       "\n",
       "                        probe_a_id            probe_a_seq      target_b_id  \\\n",
       "0      0Safe_safe_ACOC_204550.4522    GTGTATTTGGCTTCCAAAA  0Safe_safe_ACOC   \n",
       "1      0Safe_safe_ACOC_204550.4522    GTGTATTTGGCTTCCAAAA  0Safe_safe_ACOC   \n",
       "2      0Safe_safe_ACOC_204550.4522    GTGTATTTGGCTTCCAAAA  0Safe_safe_ACOC   \n",
       "3      0Safe_safe_ACOC_204550.4522    GTGTATTTGGCTTCCAAAA  0Safe_safe_DTKP   \n",
       "4      0Safe_safe_ACOC_204550.4522    GTGTATTTGGCTTCCAAAA  0Safe_safe_DTKP   \n",
       "...                            ...                    ...              ...   \n",
       "60625                      WNT5A_4  GTAACTTCCAACAGGGGGTGG            UPF3A   \n",
       "60626                      WNT5A_4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60627                      WNT5A_4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60628                      WNT5A_4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60629                      WNT5A_4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "\n",
       "                        probe_b_id            probe_b_seq  T0_2  T0_3  T12_2  \\\n",
       "0      0Safe_safe_ACOC_204550.4522    GTGTATTTGGCTTCCAAAA    22    17     36   \n",
       "1      0Safe_safe_ACOC_204550.4525    GCATGGCCTCCACTTGCAA    57    73    107   \n",
       "2      0Safe_safe_ACOC_204550.4590   GCACCAGTCTATGCCACCAC    78    55    124   \n",
       "3      0Safe_safe_DTKP_204550.5261   GTGAATTTAAGGCACAACCC    42    41     62   \n",
       "4      0Safe_safe_DTKP_204550.5308   GCTGGACTATGCCAGGACCT    17    27     38   \n",
       "...                            ...                    ...   ...   ...    ...   \n",
       "60625                      UPF3A_4  GACCCAGTCTATCAAAAGACC   105    97    227   \n",
       "60626                      WNT5A_1   GAGGATGCGAGCACTCTCGT   339   282    455   \n",
       "60627                      WNT5A_2  GAGTATCAATTCCGACATCGA   288   308    448   \n",
       "60628                      WNT5A_3  GAGTTAACTTCCAACAGGGGG   201   198    333   \n",
       "60629                      WNT5A_4  GTAACTTCCAACAGGGGGTGG   198   204    328   \n",
       "\n",
       "       T12_3  \n",
       "0         43  \n",
       "1         98  \n",
       "2        166  \n",
       "3         62  \n",
       "4         50  \n",
       "...      ...  \n",
       "60625    235  \n",
       "60626    564  \n",
       "60627    597  \n",
       "60628    416  \n",
       "60629    492  \n",
       "\n",
       "[60630 rows x 11 columns]"
      ]
     },
     "execution_count": 398,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shan_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "819f669a",
   "metadata": {},
   "outputs": [],
   "source": [
    "for column in ['target_a_id', 'probe_a_id', 'target_b_id', 'probe_b_id']:\n",
    "    shan_data[column] = ['-'.join(i.split('_')) for i in shan_data[column]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "cca56f23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "      <th>T0_2</th>\n",
       "      <th>T0_3</th>\n",
       "      <th>T12_2</th>\n",
       "      <th>T12_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>22</td>\n",
       "      <td>17</td>\n",
       "      <td>36</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4525</td>\n",
       "      <td>GCATGGCCTCCACTTGCAA</td>\n",
       "      <td>57</td>\n",
       "      <td>73</td>\n",
       "      <td>107</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4590</td>\n",
       "      <td>GCACCAGTCTATGCCACCAC</td>\n",
       "      <td>78</td>\n",
       "      <td>55</td>\n",
       "      <td>124</td>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5261</td>\n",
       "      <td>GTGAATTTAAGGCACAACCC</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5308</td>\n",
       "      <td>GCTGGACTATGCCAGGACCT</td>\n",
       "      <td>17</td>\n",
       "      <td>27</td>\n",
       "      <td>38</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60625</th>\n",
       "      <td>WNT5A_4__UPF3A_4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>UPF3A-4</td>\n",
       "      <td>GACCCAGTCTATCAAAAGACC</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "      <td>227</td>\n",
       "      <td>235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60626</th>\n",
       "      <td>WNT5A_4__WNT5A_1</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-1</td>\n",
       "      <td>GAGGATGCGAGCACTCTCGT</td>\n",
       "      <td>339</td>\n",
       "      <td>282</td>\n",
       "      <td>455</td>\n",
       "      <td>564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60627</th>\n",
       "      <td>WNT5A_4__WNT5A_2</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-2</td>\n",
       "      <td>GAGTATCAATTCCGACATCGA</td>\n",
       "      <td>288</td>\n",
       "      <td>308</td>\n",
       "      <td>448</td>\n",
       "      <td>597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60628</th>\n",
       "      <td>WNT5A_4__WNT5A_3</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-3</td>\n",
       "      <td>GAGTTAACTTCCAACAGGGGG</td>\n",
       "      <td>201</td>\n",
       "      <td>198</td>\n",
       "      <td>333</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60629</th>\n",
       "      <td>WNT5A_4__WNT5A_4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>198</td>\n",
       "      <td>204</td>\n",
       "      <td>328</td>\n",
       "      <td>492</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>60630 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id      target_a_id  \\\n",
       "0      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe-safe-ACOC   \n",
       "1      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe-safe-ACOC   \n",
       "2      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe-safe-ACOC   \n",
       "3      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...  0Safe-safe-ACOC   \n",
       "4      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...  0Safe-safe-ACOC   \n",
       "...                                                  ...              ...   \n",
       "60625                                   WNT5A_4__UPF3A_4            WNT5A   \n",
       "60626                                   WNT5A_4__WNT5A_1            WNT5A   \n",
       "60627                                   WNT5A_4__WNT5A_2            WNT5A   \n",
       "60628                                   WNT5A_4__WNT5A_3            WNT5A   \n",
       "60629                                   WNT5A_4__WNT5A_4            WNT5A   \n",
       "\n",
       "                        probe_a_id            probe_a_seq      target_b_id  \\\n",
       "0      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-ACOC   \n",
       "1      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-ACOC   \n",
       "2      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-ACOC   \n",
       "3      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-DTKP   \n",
       "4      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-DTKP   \n",
       "...                            ...                    ...              ...   \n",
       "60625                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            UPF3A   \n",
       "60626                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60627                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60628                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60629                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "\n",
       "                        probe_b_id            probe_b_seq  T0_2  T0_3  T12_2  \\\n",
       "0      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA    22    17     36   \n",
       "1      0Safe-safe-ACOC-204550.4525    GCATGGCCTCCACTTGCAA    57    73    107   \n",
       "2      0Safe-safe-ACOC-204550.4590   GCACCAGTCTATGCCACCAC    78    55    124   \n",
       "3      0Safe-safe-DTKP-204550.5261   GTGAATTTAAGGCACAACCC    42    41     62   \n",
       "4      0Safe-safe-DTKP-204550.5308   GCTGGACTATGCCAGGACCT    17    27     38   \n",
       "...                            ...                    ...   ...   ...    ...   \n",
       "60625                      UPF3A-4  GACCCAGTCTATCAAAAGACC   105    97    227   \n",
       "60626                      WNT5A-1   GAGGATGCGAGCACTCTCGT   339   282    455   \n",
       "60627                      WNT5A-2  GAGTATCAATTCCGACATCGA   288   308    448   \n",
       "60628                      WNT5A-3  GAGTTAACTTCCAACAGGGGG   201   198    333   \n",
       "60629                      WNT5A-4  GTAACTTCCAACAGGGGGTGG   198   204    328   \n",
       "\n",
       "       T12_3  \n",
       "0         43  \n",
       "1         98  \n",
       "2        166  \n",
       "3         62  \n",
       "4         50  \n",
       "...      ...  \n",
       "60625    235  \n",
       "60626    564  \n",
       "60627    597  \n",
       "60628    416  \n",
       "60629    492  \n",
       "\n",
       "[60630 rows x 11 columns]"
      ]
     },
     "execution_count": 400,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shan_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1cd47648",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "6c32fde6",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref = pd.DataFrame(columns = [\"Guide_ID\", \"Sequence\"])\n",
    "sgRNA_ref['Guide_ID'] = pd.concat([shan_data['probe_a_id'], shan_data['probe_b_id']])\n",
    "sgRNA_ref['Sequence'] = pd.concat([shan_data['probe_a_seq'], shan_data['probe_b_seq']])\n",
    "sgRNA_ref.index = sgRNA_ref['Guide_ID']\n",
    "sgRNA_ref = sgRNA_ref.drop_duplicates()\n",
    "sgRNA_ref = sgRNA_ref.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "id": "81be6ea3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide_ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4525</td>\n",
       "      <td>GCATGGCCTCCACTTGCAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4590</td>\n",
       "      <td>GCACCAGTCTATGCCACCAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe-safe-DTKP-204550.5261</td>\n",
       "      <td>GTGAATTTAAGGCACAACCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe-safe-DTKP-204550.5308</td>\n",
       "      <td>GCTGGACTATGCCAGGACCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>UPF3A-4</td>\n",
       "      <td>GACCCAGTCTATCAAAAGACC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>WNT5A-1</td>\n",
       "      <td>GAGGATGCGAGCACTCTCGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>WNT5A-2</td>\n",
       "      <td>GAGTATCAATTCCGACATCGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>WNT5A-3</td>\n",
       "      <td>GAGTTAACTTCCAACAGGGGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>247 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Guide_ID               Sequence\n",
       "0    0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA\n",
       "1    0Safe-safe-ACOC-204550.4525    GCATGGCCTCCACTTGCAA\n",
       "2    0Safe-safe-ACOC-204550.4590   GCACCAGTCTATGCCACCAC\n",
       "3    0Safe-safe-DTKP-204550.5261   GTGAATTTAAGGCACAACCC\n",
       "4    0Safe-safe-DTKP-204550.5308   GCTGGACTATGCCAGGACCT\n",
       "..                           ...                    ...\n",
       "242                      UPF3A-4  GACCCAGTCTATCAAAAGACC\n",
       "243                      WNT5A-1   GAGGATGCGAGCACTCTCGT\n",
       "244                      WNT5A-2  GAGTATCAATTCCGACATCGA\n",
       "245                      WNT5A-3  GAGTTAACTTCCAACAGGGGG\n",
       "246                      WNT5A-4  GTAACTTCCAACAGGGGGTGG\n",
       "\n",
       "[247 rows x 2 columns]"
      ]
     },
     "execution_count": 402,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "id": "7a1bc87a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sgRNA_ref.copy()\n",
    "sequence_ref['Target'] = [i.split('-')[0] for i in sequence_ref['Guide_ID']]\n",
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']\n",
    "sequence_ref.loc[sequence_ref['sgRNA_target_name'] == '0Safe', 'sgRNA_target_name'] = 'control'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "id": "9ffa3f18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4525</td>\n",
       "      <td>GCATGGCCTCCACTTGCAA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4590</td>\n",
       "      <td>GCACCAGTCTATGCCACCAC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe-safe-DTKP-204550.5261</td>\n",
       "      <td>GTGAATTTAAGGCACAACCC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe-safe-DTKP-204550.5308</td>\n",
       "      <td>GCTGGACTATGCCAGGACCT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>UPF3A-4</td>\n",
       "      <td>GACCCAGTCTATCAAAAGACC</td>\n",
       "      <td>UPF3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>WNT5A-1</td>\n",
       "      <td>GAGGATGCGAGCACTCTCGT</td>\n",
       "      <td>WNT5A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>WNT5A-2</td>\n",
       "      <td>GAGTATCAATTCCGACATCGA</td>\n",
       "      <td>WNT5A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>WNT5A-3</td>\n",
       "      <td>GAGTTAACTTCCAACAGGGGG</td>\n",
       "      <td>WNT5A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>247 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                sgRNA_guide_name        sgRNA_guide_seq sgRNA_target_name\n",
       "0    0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA           control\n",
       "1    0Safe-safe-ACOC-204550.4525    GCATGGCCTCCACTTGCAA           control\n",
       "2    0Safe-safe-ACOC-204550.4590   GCACCAGTCTATGCCACCAC           control\n",
       "3    0Safe-safe-DTKP-204550.5261   GTGAATTTAAGGCACAACCC           control\n",
       "4    0Safe-safe-DTKP-204550.5308   GCTGGACTATGCCAGGACCT           control\n",
       "..                           ...                    ...               ...\n",
       "242                      UPF3A-4  GACCCAGTCTATCAAAAGACC             UPF3A\n",
       "243                      WNT5A-1   GAGGATGCGAGCACTCTCGT             WNT5A\n",
       "244                      WNT5A-2  GAGTATCAATTCCGACATCGA             WNT5A\n",
       "245                      WNT5A-3  GAGTTAACTTCCAACAGGGGG             WNT5A\n",
       "246                      WNT5A-4  GTAACTTCCAACAGGGGGTGG             WNT5A\n",
       "\n",
       "[247 rows x 3 columns]"
      ]
     },
     "execution_count": 404,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "183acfa6",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "id": "4b8059af",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "9165a275",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Guide 1'] = shan_data['probe_a_id']\n",
    "counts_ref['Guide 2'] = shan_data['probe_b_id']\n",
    "counts_ref['Gene 1'] = shan_data['target_a_id']\n",
    "counts_ref['Gene 2'] = shan_data['target_b_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "id": "2933a007",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "      <th>T0_2</th>\n",
       "      <th>T0_3</th>\n",
       "      <th>T12_2</th>\n",
       "      <th>T12_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>22</td>\n",
       "      <td>17</td>\n",
       "      <td>36</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4525</td>\n",
       "      <td>GCATGGCCTCCACTTGCAA</td>\n",
       "      <td>57</td>\n",
       "      <td>73</td>\n",
       "      <td>107</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4590</td>\n",
       "      <td>GCACCAGTCTATGCCACCAC</td>\n",
       "      <td>78</td>\n",
       "      <td>55</td>\n",
       "      <td>124</td>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5261</td>\n",
       "      <td>GTGAATTTAAGGCACAACCC</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>GTGTATTTGGCTTCCAAAA</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5308</td>\n",
       "      <td>GCTGGACTATGCCAGGACCT</td>\n",
       "      <td>17</td>\n",
       "      <td>27</td>\n",
       "      <td>38</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60625</th>\n",
       "      <td>WNT5A_4__UPF3A_4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>UPF3A-4</td>\n",
       "      <td>GACCCAGTCTATCAAAAGACC</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "      <td>227</td>\n",
       "      <td>235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60626</th>\n",
       "      <td>WNT5A_4__WNT5A_1</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-1</td>\n",
       "      <td>GAGGATGCGAGCACTCTCGT</td>\n",
       "      <td>339</td>\n",
       "      <td>282</td>\n",
       "      <td>455</td>\n",
       "      <td>564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60627</th>\n",
       "      <td>WNT5A_4__WNT5A_2</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-2</td>\n",
       "      <td>GAGTATCAATTCCGACATCGA</td>\n",
       "      <td>288</td>\n",
       "      <td>308</td>\n",
       "      <td>448</td>\n",
       "      <td>597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60628</th>\n",
       "      <td>WNT5A_4__WNT5A_3</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-3</td>\n",
       "      <td>GAGTTAACTTCCAACAGGGGG</td>\n",
       "      <td>201</td>\n",
       "      <td>198</td>\n",
       "      <td>333</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60629</th>\n",
       "      <td>WNT5A_4__WNT5A_4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>GTAACTTCCAACAGGGGGTGG</td>\n",
       "      <td>198</td>\n",
       "      <td>204</td>\n",
       "      <td>328</td>\n",
       "      <td>492</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>60630 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id      target_a_id  \\\n",
       "0      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe-safe-ACOC   \n",
       "1      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe-safe-ACOC   \n",
       "2      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...  0Safe-safe-ACOC   \n",
       "3      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...  0Safe-safe-ACOC   \n",
       "4      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...  0Safe-safe-ACOC   \n",
       "...                                                  ...              ...   \n",
       "60625                                   WNT5A_4__UPF3A_4            WNT5A   \n",
       "60626                                   WNT5A_4__WNT5A_1            WNT5A   \n",
       "60627                                   WNT5A_4__WNT5A_2            WNT5A   \n",
       "60628                                   WNT5A_4__WNT5A_3            WNT5A   \n",
       "60629                                   WNT5A_4__WNT5A_4            WNT5A   \n",
       "\n",
       "                        probe_a_id            probe_a_seq      target_b_id  \\\n",
       "0      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-ACOC   \n",
       "1      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-ACOC   \n",
       "2      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-ACOC   \n",
       "3      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-DTKP   \n",
       "4      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA  0Safe-safe-DTKP   \n",
       "...                            ...                    ...              ...   \n",
       "60625                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            UPF3A   \n",
       "60626                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60627                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60628                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "60629                      WNT5A-4  GTAACTTCCAACAGGGGGTGG            WNT5A   \n",
       "\n",
       "                        probe_b_id            probe_b_seq  T0_2  T0_3  T12_2  \\\n",
       "0      0Safe-safe-ACOC-204550.4522    GTGTATTTGGCTTCCAAAA    22    17     36   \n",
       "1      0Safe-safe-ACOC-204550.4525    GCATGGCCTCCACTTGCAA    57    73    107   \n",
       "2      0Safe-safe-ACOC-204550.4590   GCACCAGTCTATGCCACCAC    78    55    124   \n",
       "3      0Safe-safe-DTKP-204550.5261   GTGAATTTAAGGCACAACCC    42    41     62   \n",
       "4      0Safe-safe-DTKP-204550.5308   GCTGGACTATGCCAGGACCT    17    27     38   \n",
       "...                            ...                    ...   ...   ...    ...   \n",
       "60625                      UPF3A-4  GACCCAGTCTATCAAAAGACC   105    97    227   \n",
       "60626                      WNT5A-1   GAGGATGCGAGCACTCTCGT   339   282    455   \n",
       "60627                      WNT5A-2  GAGTATCAATTCCGACATCGA   288   308    448   \n",
       "60628                      WNT5A-3  GAGTTAACTTCCAACAGGGGG   201   198    333   \n",
       "60629                      WNT5A-4  GTAACTTCCAACAGGGGGTGG   198   204    328   \n",
       "\n",
       "       T12_3  \n",
       "0         43  \n",
       "1         98  \n",
       "2        166  \n",
       "3         62  \n",
       "4         50  \n",
       "...      ...  \n",
       "60625    235  \n",
       "60626    564  \n",
       "60627    597  \n",
       "60628    416  \n",
       "60629    492  \n",
       "\n",
       "[60630 rows x 11 columns]"
      ]
     },
     "execution_count": 407,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shan_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "id": "66388c7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Count Replicates'] = shan_data[['T0_2', 'T0_3', 'T12_2', 'T12_3']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "id": "8b71e0e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Sequencing'] = ['Combinatorial CRISPR'] * counts_ref.shape[0]\n",
    "counts_ref['Cell Line'] = ['22Rv1'] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "id": "8929b59b",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref[\"Condition\"] = [';'.join(['T0_1', 'T0_2', 'T12_1', 'T12_2'])] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "id": "39a8db99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>22;17;36;43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4525</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>57;73;107;98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4590</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>78;55;124;166</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5261</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>42;41;62;62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5308</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>17;27;38;50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60625</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>UPF3A-4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>105;97;227;235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60626</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-1</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>339;282;455;564</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60627</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-2</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>288;308;448;597</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60628</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-3</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>201;198;333;416</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60629</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>198;204;328;492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>60630 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Guide 1                      Guide 2  \\\n",
       "0      0Safe-safe-ACOC-204550.4522  0Safe-safe-ACOC-204550.4522   \n",
       "1      0Safe-safe-ACOC-204550.4522  0Safe-safe-ACOC-204550.4525   \n",
       "2      0Safe-safe-ACOC-204550.4522  0Safe-safe-ACOC-204550.4590   \n",
       "3      0Safe-safe-ACOC-204550.4522  0Safe-safe-DTKP-204550.5261   \n",
       "4      0Safe-safe-ACOC-204550.4522  0Safe-safe-DTKP-204550.5308   \n",
       "...                            ...                          ...   \n",
       "60625                      WNT5A-4                      UPF3A-4   \n",
       "60626                      WNT5A-4                      WNT5A-1   \n",
       "60627                      WNT5A-4                      WNT5A-2   \n",
       "60628                      WNT5A-4                      WNT5A-3   \n",
       "60629                      WNT5A-4                      WNT5A-4   \n",
       "\n",
       "                Gene 1           Gene 2 Count Replicates Type  \\\n",
       "0      0Safe-safe-ACOC  0Safe-safe-ACOC      22;17;36;43  NaN   \n",
       "1      0Safe-safe-ACOC  0Safe-safe-ACOC     57;73;107;98  NaN   \n",
       "2      0Safe-safe-ACOC  0Safe-safe-ACOC    78;55;124;166  NaN   \n",
       "3      0Safe-safe-ACOC  0Safe-safe-DTKP      42;41;62;62  NaN   \n",
       "4      0Safe-safe-ACOC  0Safe-safe-DTKP      17;27;38;50  NaN   \n",
       "...                ...              ...              ...  ...   \n",
       "60625            WNT5A            UPF3A   105;97;227;235  NaN   \n",
       "60626            WNT5A            WNT5A  339;282;455;564  NaN   \n",
       "60627            WNT5A            WNT5A  288;308;448;597  NaN   \n",
       "60628            WNT5A            WNT5A  201;198;333;416  NaN   \n",
       "60629            WNT5A            WNT5A  198;204;328;492  NaN   \n",
       "\n",
       "                 Sequencing Cell Line              Condition  \n",
       "0      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "1      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "2      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "3      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "4      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "...                     ...       ...                    ...  \n",
       "60625  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "60626  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "60627  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "60628  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "60629  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  \n",
       "\n",
       "[60630 rows x 9 columns]"
      ]
     },
     "execution_count": 411,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "id": "55ad06c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "idx = (counts_ref['Gene 1'].isin(['luciferase', 'EGFP', 'LacZ'])) | (counts_ref['Gene 2'].isin(['luciferase', 'EGFP', 'LacZ']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "id": "77e72c14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11699"
      ]
     },
     "execution_count": 413,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "id": "10bb1ec5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16403553/ipykernel_128410/3197312670.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  counts_ref[\"Study\"] = [study_name_to_pubmed_id['shantang_data']] * counts_ref.shape[0]\n"
     ]
    }
   ],
   "source": [
    "# remove the 3 of them\n",
    "counts_ref = counts_ref[~idx]\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['shantang_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "id": "55486f35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>22;17;36;43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4525</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>57;73;107;98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-ACOC-204550.4590</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>78;55;124;166</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5261</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>42;41;62;62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe-safe-ACOC-204550.4522</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5308</td>\n",
       "      <td>0Safe-safe-ACOC</td>\n",
       "      <td>0Safe-safe-DTKP</td>\n",
       "      <td>17;27;38;50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60625</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>UPF3A-4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>105;97;227;235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60626</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-1</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>339;282;455;564</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60627</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-2</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>288;308;448;597</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60628</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-3</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>201;198;333;416</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60629</th>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A-4</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>198;204;328;492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>T0_1;T0_2;T12_1;T12_2</td>\n",
       "      <td>36060092</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48931 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Guide 1                      Guide 2  \\\n",
       "0      0Safe-safe-ACOC-204550.4522  0Safe-safe-ACOC-204550.4522   \n",
       "1      0Safe-safe-ACOC-204550.4522  0Safe-safe-ACOC-204550.4525   \n",
       "2      0Safe-safe-ACOC-204550.4522  0Safe-safe-ACOC-204550.4590   \n",
       "3      0Safe-safe-ACOC-204550.4522  0Safe-safe-DTKP-204550.5261   \n",
       "4      0Safe-safe-ACOC-204550.4522  0Safe-safe-DTKP-204550.5308   \n",
       "...                            ...                          ...   \n",
       "60625                      WNT5A-4                      UPF3A-4   \n",
       "60626                      WNT5A-4                      WNT5A-1   \n",
       "60627                      WNT5A-4                      WNT5A-2   \n",
       "60628                      WNT5A-4                      WNT5A-3   \n",
       "60629                      WNT5A-4                      WNT5A-4   \n",
       "\n",
       "                Gene 1           Gene 2 Count Replicates Type  \\\n",
       "0      0Safe-safe-ACOC  0Safe-safe-ACOC      22;17;36;43  NaN   \n",
       "1      0Safe-safe-ACOC  0Safe-safe-ACOC     57;73;107;98  NaN   \n",
       "2      0Safe-safe-ACOC  0Safe-safe-ACOC    78;55;124;166  NaN   \n",
       "3      0Safe-safe-ACOC  0Safe-safe-DTKP      42;41;62;62  NaN   \n",
       "4      0Safe-safe-ACOC  0Safe-safe-DTKP      17;27;38;50  NaN   \n",
       "...                ...              ...              ...  ...   \n",
       "60625            WNT5A            UPF3A   105;97;227;235  NaN   \n",
       "60626            WNT5A            WNT5A  339;282;455;564  NaN   \n",
       "60627            WNT5A            WNT5A  288;308;448;597  NaN   \n",
       "60628            WNT5A            WNT5A  201;198;333;416  NaN   \n",
       "60629            WNT5A            WNT5A  198;204;328;492  NaN   \n",
       "\n",
       "                 Sequencing Cell Line              Condition     Study  \n",
       "0      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "1      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "2      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "3      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "4      Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "...                     ...       ...                    ...       ...  \n",
       "60625  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "60626  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "60627  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "60628  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "60629  Combinatorial CRISPR     22Rv1  T0_1;T0_2;T12_1;T12_2  36060092  \n",
       "\n",
       "[48931 rows x 10 columns]"
      ]
     },
     "execution_count": 415,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "893cd799",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "id": "8ee72522",
   "metadata": {},
   "outputs": [],
   "source": [
    "shantang_loc = os.path.join(learning_goals_loc_general, \"ShanTang\")\n",
    "shan_SL = pd.read_csv(os.path.join(shantang_loc, \"shan_sgRNA_level_calculated_121722.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "4814816d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Pair (Original)</th>\n",
       "      <th>GI_Replicate_Z_Score_0</th>\n",
       "      <th>GI_Replicate_Z_Score_1</th>\n",
       "      <th>GI_Replicate_Average</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT3_AR</td>\n",
       "      <td>-0.241486</td>\n",
       "      <td>-1.712796</td>\n",
       "      <td>-0.977141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3_AURKA</td>\n",
       "      <td>0.687381</td>\n",
       "      <td>-0.385961</td>\n",
       "      <td>0.150710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT3_BMP6</td>\n",
       "      <td>-0.938292</td>\n",
       "      <td>-0.793694</td>\n",
       "      <td>-0.865993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT3_CCNE2</td>\n",
       "      <td>0.262964</td>\n",
       "      <td>-0.366995</td>\n",
       "      <td>-0.052016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3_CDC6</td>\n",
       "      <td>-0.238720</td>\n",
       "      <td>0.188453</td>\n",
       "      <td>-0.025134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>ULK1_UPF3A</td>\n",
       "      <td>0.433624</td>\n",
       "      <td>-0.659875</td>\n",
       "      <td>-0.113125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>ULK1_WNT5A</td>\n",
       "      <td>0.184594</td>\n",
       "      <td>-0.416153</td>\n",
       "      <td>-0.115779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>UPF2_UPF3A</td>\n",
       "      <td>0.047645</td>\n",
       "      <td>1.053003</td>\n",
       "      <td>0.550324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>UPF2_WNT5A</td>\n",
       "      <td>0.068202</td>\n",
       "      <td>0.713417</td>\n",
       "      <td>0.390810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>UPF3A_WNT5A</td>\n",
       "      <td>0.020426</td>\n",
       "      <td>0.239067</td>\n",
       "      <td>0.129747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1225 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene Pair (Original)  GI_Replicate_Z_Score_0  GI_Replicate_Z_Score_1  \\\n",
       "0                 AKT3_AR               -0.241486               -1.712796   \n",
       "1              AKT3_AURKA                0.687381               -0.385961   \n",
       "2               AKT3_BMP6               -0.938292               -0.793694   \n",
       "3              AKT3_CCNE2                0.262964               -0.366995   \n",
       "4               AKT3_CDC6               -0.238720                0.188453   \n",
       "...                   ...                     ...                     ...   \n",
       "1220           ULK1_UPF3A                0.433624               -0.659875   \n",
       "1221           ULK1_WNT5A                0.184594               -0.416153   \n",
       "1222           UPF2_UPF3A                0.047645                1.053003   \n",
       "1223           UPF2_WNT5A                0.068202                0.713417   \n",
       "1224          UPF3A_WNT5A                0.020426                0.239067   \n",
       "\n",
       "      GI_Replicate_Average  \n",
       "0                -0.977141  \n",
       "1                 0.150710  \n",
       "2                -0.865993  \n",
       "3                -0.052016  \n",
       "4                -0.025134  \n",
       "...                    ...  \n",
       "1220             -0.113125  \n",
       "1221             -0.115779  \n",
       "1222              0.550324  \n",
       "1223              0.390810  \n",
       "1224              0.129747  \n",
       "\n",
       "[1225 rows x 4 columns]"
      ]
     },
     "execution_count": 417,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shan_SL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "id": "954dadd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "curr_GI[\"Gene_A\"] = [i.split('_')[0] for i in shan_SL['Gene Pair (Original)']]\n",
    "curr_GI[\"Gene_B\"] = [i.split('_')[1] for i in shan_SL['Gene Pair (Original)']]\n",
    "curr_GI[\"GI_Score\"] = shan_SL['GI_Replicate_Average']\n",
    "curr_GI[\"GI_Cutoff\"] = [-0.361] * len(curr_GI[\"Gene_A\"])#-0.2989\n",
    "curr_GI[\"Study_Source\"] = [study_name_to_pubmed_id['shantang_data']] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI[\"Cell_Line\"] = [\"22Rv1\"] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI = curr_GI.fillna(0)\n",
    "curr_GI[\"Stat_Score\"] = [float(\"nan\")] * len(curr_GI[\"Gene_A\"])\n",
    "curr_GI[\"Stat_Cutoff\"] = [float(\"nan\")] * len(curr_GI[\"Gene_A\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "id": "b4e21dc1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>AR</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>-0.977141</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>AURKA</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>0.150710</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>BMP6</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>-0.865993</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>CCNE2</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>-0.052016</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>CDC6</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>-0.025134</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>ULK1</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>-0.113125</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>ULK1</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>-0.115779</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>UPF2</td>\n",
       "      <td>UPF3A</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>0.550324</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>UPF2</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>0.390810</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>UPF3A</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>36060092</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>0.129747</td>\n",
       "      <td>-0.361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1225 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene_A Gene_B Study_Source Cell_Line  GI_Score  GI_Cutoff  Stat_Score  \\\n",
       "0      AKT3     AR     36060092     22Rv1 -0.977141     -0.361         NaN   \n",
       "1      AKT3  AURKA     36060092     22Rv1  0.150710     -0.361         NaN   \n",
       "2      AKT3   BMP6     36060092     22Rv1 -0.865993     -0.361         NaN   \n",
       "3      AKT3  CCNE2     36060092     22Rv1 -0.052016     -0.361         NaN   \n",
       "4      AKT3   CDC6     36060092     22Rv1 -0.025134     -0.361         NaN   \n",
       "...     ...    ...          ...       ...       ...        ...         ...   \n",
       "1220   ULK1  UPF3A     36060092     22Rv1 -0.113125     -0.361         NaN   \n",
       "1221   ULK1  WNT5A     36060092     22Rv1 -0.115779     -0.361         NaN   \n",
       "1222   UPF2  UPF3A     36060092     22Rv1  0.550324     -0.361         NaN   \n",
       "1223   UPF2  WNT5A     36060092     22Rv1  0.390810     -0.361         NaN   \n",
       "1224  UPF3A  WNT5A     36060092     22Rv1  0.129747     -0.361         NaN   \n",
       "\n",
       "      Stat_Cutoff  \n",
       "0             NaN  \n",
       "1             NaN  \n",
       "2             NaN  \n",
       "3             NaN  \n",
       "4             NaN  \n",
       "...           ...  \n",
       "1220          NaN  \n",
       "1221          NaN  \n",
       "1222          NaN  \n",
       "1223          NaN  \n",
       "1224          NaN  \n",
       "\n",
       "[1225 rows x 8 columns]"
      ]
     },
     "execution_count": 419,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "2a9a8f79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 420,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(curr_GI['GI_Score'] < -1).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "id": "a1536515",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['T0_1', 'T0_2'], ['T12_1', 'T12_2']]"
      ]
     },
     "execution_count": 421,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_conditions['shantang_data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "id": "af315c7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "Only GI cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 37767\n",
      "Number of controls: 614\n",
      "Number of singles: 10550\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = curr_GI.copy(), study_controls = controls['shantang_data'], study_conditions = study_conditions['shantang_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 423,
   "id": "336ae1e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Not SL    951\n",
       "SL        274\n",
       "Name: SL_or_not, dtype: int64"
      ]
     },
     "execution_count": 423,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 424,
   "id": "b289942d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 247\n",
      "Counts insert: 48931\n",
      "Score insert: 1225\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5a5fe8f",
   "metadata": {},
   "source": [
    "## Najm Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37810e81",
   "metadata": {},
   "source": [
    "## Sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9706b193",
   "metadata": {},
   "outputs": [],
   "source": [
    "# najm_loc = os.path.join(learning_goals_loc_general, \"Najm\", \"Najm_29251726_supplementary\")\n",
    "\n",
    "# #GI is for RPE1\n",
    "# synleth_lib = pd.read_csv(os.path.join(najm_loc, \"Supplementary Table 2 SynLet library.txt\"), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "022ae307",
   "metadata": {},
   "outputs": [],
   "source": [
    "# synleth_lib['Guide_Name'] = synleth_lib['Gene'] + '_' + synleth_lib['Well ID'] + '_' + synleth_lib['Cas9']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "07fedfe2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# synleth_lib = synleth_lib.loc[:,['Guide_Name', 'sgRNA sequence', 'Gene']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "3a4c871a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sequence_ref = synleth_lib.copy()\n",
    "# sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "531b91e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sequence_ref.loc[sequence_ref['sgRNA_guide_seq'] == 'AAAAAAAGAGTCGAATGTTTT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "378bf983",
   "metadata": {},
   "outputs": [],
   "source": [
    "# seq_to_guide = {}\n",
    "# for i in range(sequence_ref.shape[0]):\n",
    "#     seq_to_guide[sequence_ref.loc[i, 'sgRNA_guide_seq']] = sequence_ref.loc[i, 'sgRNA_guide_name']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38fe4b6c",
   "metadata": {},
   "source": [
    "## Sequence (Create from counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "48821422",
   "metadata": {},
   "outputs": [],
   "source": [
    "najm_loc = os.path.join(learning_goals_loc_general, \"Najm\", \"Najm_29251726_supplementary\")\n",
    "\n",
    "#GI is for RPE1\n",
    "synleth_lib = pd.read_csv(os.path.join(najm_loc, \"Supplementary Table 3 SynLet screening data.txt\"), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e08dd343",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>U6 Sequence;H1 Sequence</th>\n",
       "      <th>U6 gene</th>\n",
       "      <th>U6 well</th>\n",
       "      <th>H1 gene</th>\n",
       "      <th>H1 well</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Time Point</th>\n",
       "      <th>Rep A Reads</th>\n",
       "      <th>Rep B Reads</th>\n",
       "      <th>Rep C Reads</th>\n",
       "      <th>...</th>\n",
       "      <th>Rep B Log Norm</th>\n",
       "      <th>Rep C Log Norm</th>\n",
       "      <th>Rep A LFC</th>\n",
       "      <th>Rep B LFC</th>\n",
       "      <th>Rep C LFC</th>\n",
       "      <th>Avg LFC</th>\n",
       "      <th>U6 expected</th>\n",
       "      <th>H1 expected</th>\n",
       "      <th>Sum Expected</th>\n",
       "      <th>Measured - Expected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAAAAAAGAGTCGAATGTTTT</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>6T</td>\n",
       "      <td>H12</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2438</td>\n",
       "      <td>1997</td>\n",
       "      <td>2175.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8.035362</td>\n",
       "      <td>8.057403</td>\n",
       "      <td>0.193558</td>\n",
       "      <td>0.523903</td>\n",
       "      <td>0.545944</td>\n",
       "      <td>0.421135</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.579674</td>\n",
       "      <td>1.021109</td>\n",
       "      <td>-0.599974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAAGAGTCCACTCTGCACTTG</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>UBC</td>\n",
       "      <td>F10</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2176</td>\n",
       "      <td>1385</td>\n",
       "      <td>1485.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.509841</td>\n",
       "      <td>7.509365</td>\n",
       "      <td>0.062268</td>\n",
       "      <td>0.030280</td>\n",
       "      <td>0.029804</td>\n",
       "      <td>0.040784</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.272248</td>\n",
       "      <td>0.713683</td>\n",
       "      <td>-0.672899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AACAGCTCCGTGTACTGAGGC</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>CD81</td>\n",
       "      <td>G08</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2294</td>\n",
       "      <td>1579</td>\n",
       "      <td>1844.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.697994</td>\n",
       "      <td>7.820199</td>\n",
       "      <td>0.617168</td>\n",
       "      <td>0.697543</td>\n",
       "      <td>0.819748</td>\n",
       "      <td>0.711486</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.593754</td>\n",
       "      <td>1.035189</td>\n",
       "      <td>-0.323703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAGACGAAATTGAAGACGAAG</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>CD81</td>\n",
       "      <td>G12</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2169</td>\n",
       "      <td>1229</td>\n",
       "      <td>1705.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.338445</td>\n",
       "      <td>7.707652</td>\n",
       "      <td>0.461431</td>\n",
       "      <td>0.262669</td>\n",
       "      <td>0.631877</td>\n",
       "      <td>0.451992</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.696023</td>\n",
       "      <td>1.137459</td>\n",
       "      <td>-0.685466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAGCGTACTGCTCATCATCGT</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>D01</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>1771</td>\n",
       "      <td>984</td>\n",
       "      <td>1526.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.019908</td>\n",
       "      <td>7.548444</td>\n",
       "      <td>0.452550</td>\n",
       "      <td>0.225973</td>\n",
       "      <td>0.754510</td>\n",
       "      <td>0.477678</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.474748</td>\n",
       "      <td>0.916183</td>\n",
       "      <td>-0.438506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92155</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTGCTTTCATTTAATGCTACA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>PARP2</td>\n",
       "      <td>G03</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>134</td>\n",
       "      <td>118</td>\n",
       "      <td>123.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.379506</td>\n",
       "      <td>6.124881</td>\n",
       "      <td>0.526160</td>\n",
       "      <td>0.290210</td>\n",
       "      <td>0.254625</td>\n",
       "      <td>0.356998</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.225575</td>\n",
       "      <td>0.611683</td>\n",
       "      <td>-0.254684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92156</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTGGGACGAGTCCTGTGAGAA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>D04</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>117</td>\n",
       "      <td>119</td>\n",
       "      <td>102.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.112980</td>\n",
       "      <td>6.015239</td>\n",
       "      <td>0.442162</td>\n",
       "      <td>0.411885</td>\n",
       "      <td>0.097741</td>\n",
       "      <td>0.317263</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.115624</td>\n",
       "      <td>0.501732</td>\n",
       "      <td>-0.184469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92157</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTAGGAATTGCTGTTGGGAC</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>H05</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>149</td>\n",
       "      <td>126</td>\n",
       "      <td>136.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.522797</td>\n",
       "      <td>6.324821</td>\n",
       "      <td>0.477853</td>\n",
       "      <td>0.183833</td>\n",
       "      <td>0.197976</td>\n",
       "      <td>0.286554</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.105574</td>\n",
       "      <td>0.491681</td>\n",
       "      <td>-0.205127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92158</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTCCATCACTTGGTTGAATA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>B02</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>169</td>\n",
       "      <td>147</td>\n",
       "      <td>202.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.088410</td>\n",
       "      <td>6.466948</td>\n",
       "      <td>0.515907</td>\n",
       "      <td>0.261833</td>\n",
       "      <td>0.621462</td>\n",
       "      <td>0.466401</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.388544</td>\n",
       "      <td>0.774652</td>\n",
       "      <td>-0.308251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92159</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTGCCTGTGAGGTGGCCGCC</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>CD81</td>\n",
       "      <td>H01</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>135</td>\n",
       "      <td>116</td>\n",
       "      <td>127.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.425130</td>\n",
       "      <td>5.689091</td>\n",
       "      <td>0.972570</td>\n",
       "      <td>0.701629</td>\n",
       "      <td>0.736039</td>\n",
       "      <td>0.803413</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.683571</td>\n",
       "      <td>1.069679</td>\n",
       "      <td>-0.266266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92160 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          U6 Sequence;H1 Sequence      U6 gene U6 well  \\\n",
       "0      AAAGTGGAACTCAGGACATG;AAAAAAAGAGTCGAATGTTTT  HPRT intron     G08   \n",
       "1      AAAGTGGAACTCAGGACATG;AAAGAGTCCACTCTGCACTTG  HPRT intron     G08   \n",
       "2      AAAGTGGAACTCAGGACATG;AACAGCTCCGTGTACTGAGGC  HPRT intron     G08   \n",
       "3      AAAGTGGAACTCAGGACATG;AAGACGAAATTGAAGACGAAG  HPRT intron     G08   \n",
       "4      AAAGTGGAACTCAGGACATG;AAGCGTACTGCTCATCATCGT  HPRT intron     G08   \n",
       "...                                           ...          ...     ...   \n",
       "92155  TTCTGACTACAACATCCAGA;TTGCTTTCATTTAATGCTACA          UBB     F03   \n",
       "92156  TTCTGACTACAACATCCAGA;TTGGGACGAGTCCTGTGAGAA          UBB     F03   \n",
       "92157  TTCTGACTACAACATCCAGA;TTTAGGAATTGCTGTTGGGAC          UBB     F03   \n",
       "92158  TTCTGACTACAACATCCAGA;TTTCCATCACTTGGTTGAATA          UBB     F03   \n",
       "92159  TTCTGACTACAACATCCAGA;TTTGCCTGTGAGGTGGCCGCC          UBB     F03   \n",
       "\n",
       "           H1 gene H1 well Cell Line Time Point  Rep A Reads  Rep B Reads  \\\n",
       "0               6T     H12      786O     Day 21         2438         1997   \n",
       "1              UBC     F10      786O     Day 21         2176         1385   \n",
       "2             CD81     G08      786O     Day 21         2294         1579   \n",
       "3             CD81     G12      786O     Day 21         2169         1229   \n",
       "4         HSP90AA1     D01      786O     Day 21         1771          984   \n",
       "...            ...     ...       ...        ...          ...          ...   \n",
       "92155        PARP2     G03    OVCAR8     Day 11          134          118   \n",
       "92156       IMPDH1     D04    OVCAR8     Day 11          117          119   \n",
       "92157  HPRT intron     H05    OVCAR8     Day 11          149          126   \n",
       "92158       BCL2A1     B02    OVCAR8     Day 11          169          147   \n",
       "92159         CD81     H01    OVCAR8     Day 11          135          116   \n",
       "\n",
       "       Rep C Reads  ...  Rep B Log Norm  Rep C Log Norm  Rep A LFC  Rep B LFC  \\\n",
       "0           2175.0  ...        8.035362        8.057403   0.193558   0.523903   \n",
       "1           1485.0  ...        7.509841        7.509365   0.062268   0.030280   \n",
       "2           1844.0  ...        7.697994        7.820199   0.617168   0.697543   \n",
       "3           1705.0  ...        7.338445        7.707652   0.461431   0.262669   \n",
       "4           1526.0  ...        7.019908        7.548444   0.452550   0.225973   \n",
       "...            ...  ...             ...             ...        ...        ...   \n",
       "92155        123.0  ...        6.379506        6.124881   0.526160   0.290210   \n",
       "92156        102.0  ...        6.112980        6.015239   0.442162   0.411885   \n",
       "92157        136.0  ...        6.522797        6.324821   0.477853   0.183833   \n",
       "92158        202.0  ...        7.088410        6.466948   0.515907   0.261833   \n",
       "92159        127.0  ...        6.425130        5.689091   0.972570   0.701629   \n",
       "\n",
       "       Rep C LFC   Avg LFC  U6 expected  H1 expected  Sum Expected  \\\n",
       "0       0.545944  0.421135     0.441435     0.579674      1.021109   \n",
       "1       0.029804  0.040784     0.441435     0.272248      0.713683   \n",
       "2       0.819748  0.711486     0.441435     0.593754      1.035189   \n",
       "3       0.631877  0.451992     0.441435     0.696023      1.137459   \n",
       "4       0.754510  0.477678     0.441435     0.474748      0.916183   \n",
       "...          ...       ...          ...          ...           ...   \n",
       "92155   0.254625  0.356998     0.386108     0.225575      0.611683   \n",
       "92156   0.097741  0.317263     0.386108     0.115624      0.501732   \n",
       "92157   0.197976  0.286554     0.386108     0.105574      0.491681   \n",
       "92158   0.621462  0.466401     0.386108     0.388544      0.774652   \n",
       "92159   0.736039  0.803413     0.386108     0.683571      1.069679   \n",
       "\n",
       "       Measured - Expected  \n",
       "0                -0.599974  \n",
       "1                -0.672899  \n",
       "2                -0.323703  \n",
       "3                -0.685466  \n",
       "4                -0.438506  \n",
       "...                    ...  \n",
       "92155            -0.254684  \n",
       "92156            -0.184469  \n",
       "92157            -0.205127  \n",
       "92158            -0.308251  \n",
       "92159            -0.266266  \n",
       "\n",
       "[92160 rows x 23 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "59cbb899",
   "metadata": {},
   "outputs": [],
   "source": [
    "synleth_lib['CL+TIME'] = synleth_lib['Cell Line'] + '_' + synleth_lib['Time Point']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "90593327",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          786O_Day 21\n",
       "1          786O_Day 21\n",
       "2          786O_Day 21\n",
       "3          786O_Day 21\n",
       "4          786O_Day 21\n",
       "             ...      \n",
       "92155    OVCAR8_Day 11\n",
       "92156    OVCAR8_Day 11\n",
       "92157    OVCAR8_Day 11\n",
       "92158    OVCAR8_Day 11\n",
       "92159    OVCAR8_Day 11\n",
       "Name: CL+TIME, Length: 92160, dtype: object"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib['CL+TIME']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f2548f86",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'786O_Day 11',\n",
       " '786O_Day 21',\n",
       " 'A375_Day 21',\n",
       " 'A375_Day 9',\n",
       " 'A549_Day 21',\n",
       " 'HT29_Day 21',\n",
       " 'Meljuso_Day 21',\n",
       " 'Meljuso_Day 9',\n",
       " 'OVCAR8_Day 11',\n",
       " 'OVCAR8_Day 21'}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(synleth_lib['CL+TIME'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1871d53c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# only get the final time points for each cell line\n",
    "synleth_lib = synleth_lib.loc[synleth_lib['Time Point'] == 'Day 21']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "ebe13b34",
   "metadata": {},
   "outputs": [],
   "source": [
    "synleth_lib.reset_index(drop = True, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1645db00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'786O', 'A375', 'A549', 'HT29', 'Meljuso', 'OVCAR8'}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(synleth_lib['Cell Line'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d9c65dd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_to_guide = {}\n",
    "target_counts = {}\n",
    "for i in range(synleth_lib.shape[0]):\n",
    "    split = synleth_lib.loc[i, 'U6 Sequence;H1 Sequence'].split(';')\n",
    "    \n",
    "    guide_1_seq = split[0]\n",
    "    guide_2_seq = split[1]\n",
    "    \n",
    "    target_1 = synleth_lib.loc[i, 'U6 gene']\n",
    "    target_2 = synleth_lib.loc[i, 'H1 gene']\n",
    "    \n",
    "    if (guide_1_seq not in seq_to_guide):\n",
    "        if (target_1 not in target_counts):\n",
    "            target_counts[target_1] = 1\n",
    "        else:\n",
    "            target_counts[target_1] += 1\n",
    "        seq_to_guide[guide_1_seq] = '_'.join([target_1, str(target_counts[target_1])])\n",
    "    if (guide_2_seq not in seq_to_guide):\n",
    "        if (target_2 not in target_counts):\n",
    "            target_counts[target_2] = 1\n",
    "        else:\n",
    "            target_counts[target_2] += 1\n",
    "        seq_to_guide[guide_2_seq] = '_'.join([target_2, str(target_counts[target_2])])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "53d8b9c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "192"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(seq_to_guide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e93b9968",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_list = []\n",
    "guide_list = []\n",
    "\n",
    "for seq in seq_to_guide:\n",
    "    seq_list.append(seq)\n",
    "    guide_list.append(seq_to_guide[seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ae4d00e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = pd.DataFrame({'sgRNA_guide_name': guide_list,\n",
    "                             'sgRNA_guide_seq': seq_list})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "b1ea08b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref['sgRNA_target_name'] = [i.split('_')[0] for i in sequence_ref['sgRNA_guide_name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9dfcb03e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>AAAGTGGAACTCAGGACATG</td>\n",
       "      <td>HPRT intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6T_1</td>\n",
       "      <td>AAAAAAAGAGTCGAATGTTTT</td>\n",
       "      <td>6T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UBC_1</td>\n",
       "      <td>AAAGAGTCCACTCTGCACTTG</td>\n",
       "      <td>UBC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CD81_1</td>\n",
       "      <td>AACAGCTCCGTGTACTGAGGC</td>\n",
       "      <td>CD81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CD81_2</td>\n",
       "      <td>AAGACGAAATTGAAGACGAAG</td>\n",
       "      <td>CD81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>AKT1_6</td>\n",
       "      <td>TGTCATGGAGTACGCCAACG</td>\n",
       "      <td>AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>BCL2_6</td>\n",
       "      <td>TGTCGCAGAGGGGCTACGAG</td>\n",
       "      <td>BCL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>BCL2L10_6</td>\n",
       "      <td>TGTTGCTGGCCGACTACCTG</td>\n",
       "      <td>BCL2L10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>UBB_5</td>\n",
       "      <td>TGTTGTAGTCAGAAAGAGTA</td>\n",
       "      <td>UBB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>TTCTGACTACAACATCCAGA</td>\n",
       "      <td>UBB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sgRNA_guide_name        sgRNA_guide_seq sgRNA_target_name\n",
       "0      HPRT intron_1   AAAGTGGAACTCAGGACATG       HPRT intron\n",
       "1               6T_1  AAAAAAAGAGTCGAATGTTTT                6T\n",
       "2              UBC_1  AAAGAGTCCACTCTGCACTTG               UBC\n",
       "3             CD81_1  AACAGCTCCGTGTACTGAGGC              CD81\n",
       "4             CD81_2  AAGACGAAATTGAAGACGAAG              CD81\n",
       "..               ...                    ...               ...\n",
       "187           AKT1_6   TGTCATGGAGTACGCCAACG              AKT1\n",
       "188           BCL2_6   TGTCGCAGAGGGGCTACGAG              BCL2\n",
       "189        BCL2L10_6   TGTTGCTGGCCGACTACCTG           BCL2L10\n",
       "190            UBB_5   TGTTGTAGTCAGAAAGAGTA               UBB\n",
       "191            UBB_6   TTCTGACTACAACATCCAGA               UBB\n",
       "\n",
       "[192 rows x 3 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8732f6f",
   "metadata": {},
   "source": [
    "## Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "a40bd7aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "b0d1a5fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "najm_loc = os.path.join(learning_goals_loc_general, \"Najm\", \"Najm_29251726_supplementary\")\n",
    "\n",
    "#GI is for RPE1\n",
    "synleth_lib = pd.read_csv(os.path.join(najm_loc, \"Supplementary Table 3 SynLet screening data.txt\"), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "65b8cb81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>U6 Sequence;H1 Sequence</th>\n",
       "      <th>U6 gene</th>\n",
       "      <th>U6 well</th>\n",
       "      <th>H1 gene</th>\n",
       "      <th>H1 well</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Time Point</th>\n",
       "      <th>Rep A Reads</th>\n",
       "      <th>Rep B Reads</th>\n",
       "      <th>Rep C Reads</th>\n",
       "      <th>...</th>\n",
       "      <th>Rep B Log Norm</th>\n",
       "      <th>Rep C Log Norm</th>\n",
       "      <th>Rep A LFC</th>\n",
       "      <th>Rep B LFC</th>\n",
       "      <th>Rep C LFC</th>\n",
       "      <th>Avg LFC</th>\n",
       "      <th>U6 expected</th>\n",
       "      <th>H1 expected</th>\n",
       "      <th>Sum Expected</th>\n",
       "      <th>Measured - Expected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAAAAAAGAGTCGAATGTTTT</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>6T</td>\n",
       "      <td>H12</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2438</td>\n",
       "      <td>1997</td>\n",
       "      <td>2175.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8.035362</td>\n",
       "      <td>8.057403</td>\n",
       "      <td>0.193558</td>\n",
       "      <td>0.523903</td>\n",
       "      <td>0.545944</td>\n",
       "      <td>0.421135</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.579674</td>\n",
       "      <td>1.021109</td>\n",
       "      <td>-0.599974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAAGAGTCCACTCTGCACTTG</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>UBC</td>\n",
       "      <td>F10</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2176</td>\n",
       "      <td>1385</td>\n",
       "      <td>1485.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.509841</td>\n",
       "      <td>7.509365</td>\n",
       "      <td>0.062268</td>\n",
       "      <td>0.030280</td>\n",
       "      <td>0.029804</td>\n",
       "      <td>0.040784</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.272248</td>\n",
       "      <td>0.713683</td>\n",
       "      <td>-0.672899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AACAGCTCCGTGTACTGAGGC</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>CD81</td>\n",
       "      <td>G08</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2294</td>\n",
       "      <td>1579</td>\n",
       "      <td>1844.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.697994</td>\n",
       "      <td>7.820199</td>\n",
       "      <td>0.617168</td>\n",
       "      <td>0.697543</td>\n",
       "      <td>0.819748</td>\n",
       "      <td>0.711486</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.593754</td>\n",
       "      <td>1.035189</td>\n",
       "      <td>-0.323703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAGACGAAATTGAAGACGAAG</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>CD81</td>\n",
       "      <td>G12</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2169</td>\n",
       "      <td>1229</td>\n",
       "      <td>1705.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.338445</td>\n",
       "      <td>7.707652</td>\n",
       "      <td>0.461431</td>\n",
       "      <td>0.262669</td>\n",
       "      <td>0.631877</td>\n",
       "      <td>0.451992</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.696023</td>\n",
       "      <td>1.137459</td>\n",
       "      <td>-0.685466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAGCGTACTGCTCATCATCGT</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>D01</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>1771</td>\n",
       "      <td>984</td>\n",
       "      <td>1526.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.019908</td>\n",
       "      <td>7.548444</td>\n",
       "      <td>0.452550</td>\n",
       "      <td>0.225973</td>\n",
       "      <td>0.754510</td>\n",
       "      <td>0.477678</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.474748</td>\n",
       "      <td>0.916183</td>\n",
       "      <td>-0.438506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92155</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTGCTTTCATTTAATGCTACA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>PARP2</td>\n",
       "      <td>G03</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>134</td>\n",
       "      <td>118</td>\n",
       "      <td>123.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.379506</td>\n",
       "      <td>6.124881</td>\n",
       "      <td>0.526160</td>\n",
       "      <td>0.290210</td>\n",
       "      <td>0.254625</td>\n",
       "      <td>0.356998</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.225575</td>\n",
       "      <td>0.611683</td>\n",
       "      <td>-0.254684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92156</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTGGGACGAGTCCTGTGAGAA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>D04</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>117</td>\n",
       "      <td>119</td>\n",
       "      <td>102.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.112980</td>\n",
       "      <td>6.015239</td>\n",
       "      <td>0.442162</td>\n",
       "      <td>0.411885</td>\n",
       "      <td>0.097741</td>\n",
       "      <td>0.317263</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.115624</td>\n",
       "      <td>0.501732</td>\n",
       "      <td>-0.184469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92157</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTAGGAATTGCTGTTGGGAC</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>H05</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>149</td>\n",
       "      <td>126</td>\n",
       "      <td>136.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.522797</td>\n",
       "      <td>6.324821</td>\n",
       "      <td>0.477853</td>\n",
       "      <td>0.183833</td>\n",
       "      <td>0.197976</td>\n",
       "      <td>0.286554</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.105574</td>\n",
       "      <td>0.491681</td>\n",
       "      <td>-0.205127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92158</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTCCATCACTTGGTTGAATA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>B02</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>169</td>\n",
       "      <td>147</td>\n",
       "      <td>202.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.088410</td>\n",
       "      <td>6.466948</td>\n",
       "      <td>0.515907</td>\n",
       "      <td>0.261833</td>\n",
       "      <td>0.621462</td>\n",
       "      <td>0.466401</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.388544</td>\n",
       "      <td>0.774652</td>\n",
       "      <td>-0.308251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92159</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTGCCTGTGAGGTGGCCGCC</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>CD81</td>\n",
       "      <td>H01</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 11</td>\n",
       "      <td>135</td>\n",
       "      <td>116</td>\n",
       "      <td>127.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.425130</td>\n",
       "      <td>5.689091</td>\n",
       "      <td>0.972570</td>\n",
       "      <td>0.701629</td>\n",
       "      <td>0.736039</td>\n",
       "      <td>0.803413</td>\n",
       "      <td>0.386108</td>\n",
       "      <td>0.683571</td>\n",
       "      <td>1.069679</td>\n",
       "      <td>-0.266266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92160 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          U6 Sequence;H1 Sequence      U6 gene U6 well  \\\n",
       "0      AAAGTGGAACTCAGGACATG;AAAAAAAGAGTCGAATGTTTT  HPRT intron     G08   \n",
       "1      AAAGTGGAACTCAGGACATG;AAAGAGTCCACTCTGCACTTG  HPRT intron     G08   \n",
       "2      AAAGTGGAACTCAGGACATG;AACAGCTCCGTGTACTGAGGC  HPRT intron     G08   \n",
       "3      AAAGTGGAACTCAGGACATG;AAGACGAAATTGAAGACGAAG  HPRT intron     G08   \n",
       "4      AAAGTGGAACTCAGGACATG;AAGCGTACTGCTCATCATCGT  HPRT intron     G08   \n",
       "...                                           ...          ...     ...   \n",
       "92155  TTCTGACTACAACATCCAGA;TTGCTTTCATTTAATGCTACA          UBB     F03   \n",
       "92156  TTCTGACTACAACATCCAGA;TTGGGACGAGTCCTGTGAGAA          UBB     F03   \n",
       "92157  TTCTGACTACAACATCCAGA;TTTAGGAATTGCTGTTGGGAC          UBB     F03   \n",
       "92158  TTCTGACTACAACATCCAGA;TTTCCATCACTTGGTTGAATA          UBB     F03   \n",
       "92159  TTCTGACTACAACATCCAGA;TTTGCCTGTGAGGTGGCCGCC          UBB     F03   \n",
       "\n",
       "           H1 gene H1 well Cell Line Time Point  Rep A Reads  Rep B Reads  \\\n",
       "0               6T     H12      786O     Day 21         2438         1997   \n",
       "1              UBC     F10      786O     Day 21         2176         1385   \n",
       "2             CD81     G08      786O     Day 21         2294         1579   \n",
       "3             CD81     G12      786O     Day 21         2169         1229   \n",
       "4         HSP90AA1     D01      786O     Day 21         1771          984   \n",
       "...            ...     ...       ...        ...          ...          ...   \n",
       "92155        PARP2     G03    OVCAR8     Day 11          134          118   \n",
       "92156       IMPDH1     D04    OVCAR8     Day 11          117          119   \n",
       "92157  HPRT intron     H05    OVCAR8     Day 11          149          126   \n",
       "92158       BCL2A1     B02    OVCAR8     Day 11          169          147   \n",
       "92159         CD81     H01    OVCAR8     Day 11          135          116   \n",
       "\n",
       "       Rep C Reads  ...  Rep B Log Norm  Rep C Log Norm  Rep A LFC  Rep B LFC  \\\n",
       "0           2175.0  ...        8.035362        8.057403   0.193558   0.523903   \n",
       "1           1485.0  ...        7.509841        7.509365   0.062268   0.030280   \n",
       "2           1844.0  ...        7.697994        7.820199   0.617168   0.697543   \n",
       "3           1705.0  ...        7.338445        7.707652   0.461431   0.262669   \n",
       "4           1526.0  ...        7.019908        7.548444   0.452550   0.225973   \n",
       "...            ...  ...             ...             ...        ...        ...   \n",
       "92155        123.0  ...        6.379506        6.124881   0.526160   0.290210   \n",
       "92156        102.0  ...        6.112980        6.015239   0.442162   0.411885   \n",
       "92157        136.0  ...        6.522797        6.324821   0.477853   0.183833   \n",
       "92158        202.0  ...        7.088410        6.466948   0.515907   0.261833   \n",
       "92159        127.0  ...        6.425130        5.689091   0.972570   0.701629   \n",
       "\n",
       "       Rep C LFC   Avg LFC  U6 expected  H1 expected  Sum Expected  \\\n",
       "0       0.545944  0.421135     0.441435     0.579674      1.021109   \n",
       "1       0.029804  0.040784     0.441435     0.272248      0.713683   \n",
       "2       0.819748  0.711486     0.441435     0.593754      1.035189   \n",
       "3       0.631877  0.451992     0.441435     0.696023      1.137459   \n",
       "4       0.754510  0.477678     0.441435     0.474748      0.916183   \n",
       "...          ...       ...          ...          ...           ...   \n",
       "92155   0.254625  0.356998     0.386108     0.225575      0.611683   \n",
       "92156   0.097741  0.317263     0.386108     0.115624      0.501732   \n",
       "92157   0.197976  0.286554     0.386108     0.105574      0.491681   \n",
       "92158   0.621462  0.466401     0.386108     0.388544      0.774652   \n",
       "92159   0.736039  0.803413     0.386108     0.683571      1.069679   \n",
       "\n",
       "       Measured - Expected  \n",
       "0                -0.599974  \n",
       "1                -0.672899  \n",
       "2                -0.323703  \n",
       "3                -0.685466  \n",
       "4                -0.438506  \n",
       "...                    ...  \n",
       "92155            -0.254684  \n",
       "92156            -0.184469  \n",
       "92157            -0.205127  \n",
       "92158            -0.308251  \n",
       "92159            -0.266266  \n",
       "\n",
       "[92160 rows x 23 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cbd6a922",
   "metadata": {},
   "outputs": [],
   "source": [
    "synleth_lib['CL+TIME'] = synleth_lib['Cell Line'] + '_' + synleth_lib['Time Point']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a3539a4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          786O_Day 21\n",
       "1          786O_Day 21\n",
       "2          786O_Day 21\n",
       "3          786O_Day 21\n",
       "4          786O_Day 21\n",
       "             ...      \n",
       "92155    OVCAR8_Day 11\n",
       "92156    OVCAR8_Day 11\n",
       "92157    OVCAR8_Day 11\n",
       "92158    OVCAR8_Day 11\n",
       "92159    OVCAR8_Day 11\n",
       "Name: CL+TIME, Length: 92160, dtype: object"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib['CL+TIME']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "ad8a50f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'786O_Day 11',\n",
       " '786O_Day 21',\n",
       " 'A375_Day 21',\n",
       " 'A375_Day 9',\n",
       " 'A549_Day 21',\n",
       " 'HT29_Day 21',\n",
       " 'Meljuso_Day 21',\n",
       " 'Meljuso_Day 9',\n",
       " 'OVCAR8_Day 11',\n",
       " 'OVCAR8_Day 21'}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(synleth_lib['CL+TIME'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "dc9fb050",
   "metadata": {},
   "outputs": [],
   "source": [
    "# only get the final time points for each cell line\n",
    "synleth_lib = synleth_lib.loc[synleth_lib['Time Point'] == 'Day 21']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "798d7cf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "synleth_lib.reset_index(drop = True, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "682e0dc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'786O', 'A375', 'A549', 'HT29', 'Meljuso', 'OVCAR8'}"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(synleth_lib['Cell Line'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c7b583a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>U6 Sequence;H1 Sequence</th>\n",
       "      <th>U6 gene</th>\n",
       "      <th>U6 well</th>\n",
       "      <th>H1 gene</th>\n",
       "      <th>H1 well</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Time Point</th>\n",
       "      <th>Rep A Reads</th>\n",
       "      <th>Rep B Reads</th>\n",
       "      <th>Rep C Reads</th>\n",
       "      <th>...</th>\n",
       "      <th>Rep C Log Norm</th>\n",
       "      <th>Rep A LFC</th>\n",
       "      <th>Rep B LFC</th>\n",
       "      <th>Rep C LFC</th>\n",
       "      <th>Avg LFC</th>\n",
       "      <th>U6 expected</th>\n",
       "      <th>H1 expected</th>\n",
       "      <th>Sum Expected</th>\n",
       "      <th>Measured - Expected</th>\n",
       "      <th>CL+TIME</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAAAAAAGAGTCGAATGTTTT</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>6T</td>\n",
       "      <td>H12</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2438</td>\n",
       "      <td>1997</td>\n",
       "      <td>2175.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8.057403</td>\n",
       "      <td>0.193558</td>\n",
       "      <td>0.523903</td>\n",
       "      <td>0.545944</td>\n",
       "      <td>0.421135</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.579674</td>\n",
       "      <td>1.021109</td>\n",
       "      <td>-0.599974</td>\n",
       "      <td>786O_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAAGAGTCCACTCTGCACTTG</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>UBC</td>\n",
       "      <td>F10</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2176</td>\n",
       "      <td>1385</td>\n",
       "      <td>1485.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.509365</td>\n",
       "      <td>0.062268</td>\n",
       "      <td>0.030280</td>\n",
       "      <td>0.029804</td>\n",
       "      <td>0.040784</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.272248</td>\n",
       "      <td>0.713683</td>\n",
       "      <td>-0.672899</td>\n",
       "      <td>786O_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AACAGCTCCGTGTACTGAGGC</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>CD81</td>\n",
       "      <td>G08</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2294</td>\n",
       "      <td>1579</td>\n",
       "      <td>1844.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.820199</td>\n",
       "      <td>0.617168</td>\n",
       "      <td>0.697543</td>\n",
       "      <td>0.819748</td>\n",
       "      <td>0.711486</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.593754</td>\n",
       "      <td>1.035189</td>\n",
       "      <td>-0.323703</td>\n",
       "      <td>786O_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAGACGAAATTGAAGACGAAG</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>CD81</td>\n",
       "      <td>G12</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>2169</td>\n",
       "      <td>1229</td>\n",
       "      <td>1705.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.707652</td>\n",
       "      <td>0.461431</td>\n",
       "      <td>0.262669</td>\n",
       "      <td>0.631877</td>\n",
       "      <td>0.451992</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.696023</td>\n",
       "      <td>1.137459</td>\n",
       "      <td>-0.685466</td>\n",
       "      <td>786O_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAGTGGAACTCAGGACATG;AAGCGTACTGCTCATCATCGT</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>G08</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>D01</td>\n",
       "      <td>786O</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>1771</td>\n",
       "      <td>984</td>\n",
       "      <td>1526.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.548444</td>\n",
       "      <td>0.452550</td>\n",
       "      <td>0.225973</td>\n",
       "      <td>0.754510</td>\n",
       "      <td>0.477678</td>\n",
       "      <td>0.441435</td>\n",
       "      <td>0.474748</td>\n",
       "      <td>0.916183</td>\n",
       "      <td>-0.438506</td>\n",
       "      <td>786O_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55291</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTGCTTTCATTTAATGCTACA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>PARP2</td>\n",
       "      <td>G03</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>140</td>\n",
       "      <td>93</td>\n",
       "      <td>291.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.124881</td>\n",
       "      <td>-0.486649</td>\n",
       "      <td>-1.041673</td>\n",
       "      <td>-0.625293</td>\n",
       "      <td>-0.717872</td>\n",
       "      <td>-0.161739</td>\n",
       "      <td>0.242457</td>\n",
       "      <td>0.080718</td>\n",
       "      <td>-0.798590</td>\n",
       "      <td>OVCAR8_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55292</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTGGGACGAGTCCTGTGAGAA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>D04</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>185</td>\n",
       "      <td>270</td>\n",
       "      <td>418.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.015239</td>\n",
       "      <td>0.018028</td>\n",
       "      <td>0.577453</td>\n",
       "      <td>-0.002889</td>\n",
       "      <td>0.197531</td>\n",
       "      <td>-0.161739</td>\n",
       "      <td>0.743171</td>\n",
       "      <td>0.581432</td>\n",
       "      <td>-0.383901</td>\n",
       "      <td>OVCAR8_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55293</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTAGGAATTGCTGTTGGGAC</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>H05</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>215</td>\n",
       "      <td>208</td>\n",
       "      <td>337.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.324821</td>\n",
       "      <td>-0.077820</td>\n",
       "      <td>-0.104056</td>\n",
       "      <td>-0.617864</td>\n",
       "      <td>-0.266580</td>\n",
       "      <td>-0.161739</td>\n",
       "      <td>0.350282</td>\n",
       "      <td>0.188542</td>\n",
       "      <td>-0.455122</td>\n",
       "      <td>OVCAR8_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55294</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTCCATCACTTGGTTGAATA</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>B02</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>178</td>\n",
       "      <td>186</td>\n",
       "      <td>486.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.466948</td>\n",
       "      <td>-0.488464</td>\n",
       "      <td>-0.405178</td>\n",
       "      <td>-0.240275</td>\n",
       "      <td>-0.377972</td>\n",
       "      <td>-0.161739</td>\n",
       "      <td>0.323509</td>\n",
       "      <td>0.161769</td>\n",
       "      <td>-0.539742</td>\n",
       "      <td>OVCAR8_Day 21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55295</th>\n",
       "      <td>TTCTGACTACAACATCCAGA;TTTGCCTGTGAGGTGGCCGCC</td>\n",
       "      <td>UBB</td>\n",
       "      <td>F03</td>\n",
       "      <td>CD81</td>\n",
       "      <td>H01</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>Day 21</td>\n",
       "      <td>145</td>\n",
       "      <td>96</td>\n",
       "      <td>244.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.689091</td>\n",
       "      <td>-0.001233</td>\n",
       "      <td>-0.561411</td>\n",
       "      <td>-0.437512</td>\n",
       "      <td>-0.333385</td>\n",
       "      <td>-0.161739</td>\n",
       "      <td>0.664896</td>\n",
       "      <td>0.503157</td>\n",
       "      <td>-0.836542</td>\n",
       "      <td>OVCAR8_Day 21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55296 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          U6 Sequence;H1 Sequence      U6 gene U6 well  \\\n",
       "0      AAAGTGGAACTCAGGACATG;AAAAAAAGAGTCGAATGTTTT  HPRT intron     G08   \n",
       "1      AAAGTGGAACTCAGGACATG;AAAGAGTCCACTCTGCACTTG  HPRT intron     G08   \n",
       "2      AAAGTGGAACTCAGGACATG;AACAGCTCCGTGTACTGAGGC  HPRT intron     G08   \n",
       "3      AAAGTGGAACTCAGGACATG;AAGACGAAATTGAAGACGAAG  HPRT intron     G08   \n",
       "4      AAAGTGGAACTCAGGACATG;AAGCGTACTGCTCATCATCGT  HPRT intron     G08   \n",
       "...                                           ...          ...     ...   \n",
       "55291  TTCTGACTACAACATCCAGA;TTGCTTTCATTTAATGCTACA          UBB     F03   \n",
       "55292  TTCTGACTACAACATCCAGA;TTGGGACGAGTCCTGTGAGAA          UBB     F03   \n",
       "55293  TTCTGACTACAACATCCAGA;TTTAGGAATTGCTGTTGGGAC          UBB     F03   \n",
       "55294  TTCTGACTACAACATCCAGA;TTTCCATCACTTGGTTGAATA          UBB     F03   \n",
       "55295  TTCTGACTACAACATCCAGA;TTTGCCTGTGAGGTGGCCGCC          UBB     F03   \n",
       "\n",
       "           H1 gene H1 well Cell Line Time Point  Rep A Reads  Rep B Reads  \\\n",
       "0               6T     H12      786O     Day 21         2438         1997   \n",
       "1              UBC     F10      786O     Day 21         2176         1385   \n",
       "2             CD81     G08      786O     Day 21         2294         1579   \n",
       "3             CD81     G12      786O     Day 21         2169         1229   \n",
       "4         HSP90AA1     D01      786O     Day 21         1771          984   \n",
       "...            ...     ...       ...        ...          ...          ...   \n",
       "55291        PARP2     G03    OVCAR8     Day 21          140           93   \n",
       "55292       IMPDH1     D04    OVCAR8     Day 21          185          270   \n",
       "55293  HPRT intron     H05    OVCAR8     Day 21          215          208   \n",
       "55294       BCL2A1     B02    OVCAR8     Day 21          178          186   \n",
       "55295         CD81     H01    OVCAR8     Day 21          145           96   \n",
       "\n",
       "       Rep C Reads  ...  Rep C Log Norm  Rep A LFC  Rep B LFC  Rep C LFC  \\\n",
       "0           2175.0  ...        8.057403   0.193558   0.523903   0.545944   \n",
       "1           1485.0  ...        7.509365   0.062268   0.030280   0.029804   \n",
       "2           1844.0  ...        7.820199   0.617168   0.697543   0.819748   \n",
       "3           1705.0  ...        7.707652   0.461431   0.262669   0.631877   \n",
       "4           1526.0  ...        7.548444   0.452550   0.225973   0.754510   \n",
       "...            ...  ...             ...        ...        ...        ...   \n",
       "55291        291.0  ...        6.124881  -0.486649  -1.041673  -0.625293   \n",
       "55292        418.0  ...        6.015239   0.018028   0.577453  -0.002889   \n",
       "55293        337.0  ...        6.324821  -0.077820  -0.104056  -0.617864   \n",
       "55294        486.0  ...        6.466948  -0.488464  -0.405178  -0.240275   \n",
       "55295        244.0  ...        5.689091  -0.001233  -0.561411  -0.437512   \n",
       "\n",
       "        Avg LFC  U6 expected  H1 expected  Sum Expected  Measured - Expected  \\\n",
       "0      0.421135     0.441435     0.579674      1.021109            -0.599974   \n",
       "1      0.040784     0.441435     0.272248      0.713683            -0.672899   \n",
       "2      0.711486     0.441435     0.593754      1.035189            -0.323703   \n",
       "3      0.451992     0.441435     0.696023      1.137459            -0.685466   \n",
       "4      0.477678     0.441435     0.474748      0.916183            -0.438506   \n",
       "...         ...          ...          ...           ...                  ...   \n",
       "55291 -0.717872    -0.161739     0.242457      0.080718            -0.798590   \n",
       "55292  0.197531    -0.161739     0.743171      0.581432            -0.383901   \n",
       "55293 -0.266580    -0.161739     0.350282      0.188542            -0.455122   \n",
       "55294 -0.377972    -0.161739     0.323509      0.161769            -0.539742   \n",
       "55295 -0.333385    -0.161739     0.664896      0.503157            -0.836542   \n",
       "\n",
       "             CL+TIME  \n",
       "0        786O_Day 21  \n",
       "1        786O_Day 21  \n",
       "2        786O_Day 21  \n",
       "3        786O_Day 21  \n",
       "4        786O_Day 21  \n",
       "...              ...  \n",
       "55291  OVCAR8_Day 21  \n",
       "55292  OVCAR8_Day 21  \n",
       "55293  OVCAR8_Day 21  \n",
       "55294  OVCAR8_Day 21  \n",
       "55295  OVCAR8_Day 21  \n",
       "\n",
       "[55296 rows x 24 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "bfb8a807",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16407139/ipykernel_195360/2887337210.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  synleth_lib['Gene 1'] = synleth_lib['U6 gene']\n",
      "/tmp/slurmtmp.16407139/ipykernel_195360/2887337210.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  synleth_lib['Gene 2'] = synleth_lib['H1 gene']\n"
     ]
    }
   ],
   "source": [
    "synleth_lib['Gene 1'] = synleth_lib['U6 gene']\n",
    "synleth_lib['Gene 2'] = synleth_lib['H1 gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "9653002c",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_1_seqs = [i.split(';')[0] for i in synleth_lib['U6 Sequence;H1 Sequence']]\n",
    "guide_2_seqs = [i.split(';')[1] for i in synleth_lib['U6 Sequence;H1 Sequence']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "fa7341ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "guide_1_ids = [seq_to_guide[i] for i in guide_1_seqs]\n",
    "guide_2_ids = [seq_to_guide[i] for i in guide_2_seqs]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "d83b13d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16407139/ipykernel_195360/1527821993.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  synleth_lib['Guide 1'] = guide_1_ids\n",
      "/tmp/slurmtmp.16407139/ipykernel_195360/1527821993.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  synleth_lib['Guide 2'] = guide_2_ids\n"
     ]
    }
   ],
   "source": [
    "synleth_lib['Guide 1'] = guide_1_ids\n",
    "synleth_lib['Guide 2'] = guide_2_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "ed11badf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['U6 Sequence;H1 Sequence', 'U6 gene', 'U6 well', 'H1 gene', 'H1 well',\n",
       "       'Cell Line', 'Time Point', 'Rep A Reads', 'Rep B Reads', 'Rep C Reads',\n",
       "       'pDNA Reads', 'pDNA Log Norm', 'Rep A Log Norm', 'Rep B Log Norm',\n",
       "       'Rep C Log Norm', 'Rep A LFC', 'Rep B LFC', 'Rep C LFC', 'Avg LFC',\n",
       "       'U6 expected', 'H1 expected', 'Sum Expected', 'Measured - Expected',\n",
       "       'CL+TIME', 'Gene 1', 'Gene 2', 'Guide 1', 'Guide 2'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "b7a57318",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16407139/ipykernel_195360/973945180.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  synleth_lib[\"Count Replicates\"] = synleth_lib[['pDNA Reads', 'Rep A Reads', 'Rep B Reads', 'Rep C Reads']].apply(\n"
     ]
    }
   ],
   "source": [
    "synleth_lib[\"Count Replicates\"] = synleth_lib[['pDNA Reads', 'Rep A Reads', 'Rep B Reads', 'Rep C Reads']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "f9d03b59",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.16407139/ipykernel_195360/1828877387.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  synleth_lib[\"Condition\"] = ';'.join(['pDNA_Reads', 'Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads'])\n"
     ]
    }
   ],
   "source": [
    "synleth_lib[\"Condition\"] = ';'.join(['pDNA_Reads', 'Rep_A_Reads', 'Rep_B_Reads', 'Rep_C_Reads'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "619c6e8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>6T_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>6T</td>\n",
       "      <td>1137.0;2438.0;1997.0;2175.0</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>UBC_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>UBC</td>\n",
       "      <td>1112.0;2176.0;1385.0;1485.0</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>CD81_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>CD81</td>\n",
       "      <td>796.0;2294.0;1579.0;1844.0</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>CD81_2</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>CD81</td>\n",
       "      <td>839.0;2169.0;1229.0;1705.0</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>HSP90AA1_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>689.0;1771.0;984.0;1526.0</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55291</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>PARP2_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>PARP2</td>\n",
       "      <td>431.0;140.0;93.0;291.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55292</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>IMPDH1_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>399.0;185.0;270.0;418.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55293</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>HPRT intron_6</td>\n",
       "      <td>UBB</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>496.0;215.0;208.0;337.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55294</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>BCL2A1_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>548.0;178.0;186.0;486.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55295</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>CD81_10</td>\n",
       "      <td>UBB</td>\n",
       "      <td>CD81</td>\n",
       "      <td>317.0;145.0;96.0;244.0</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55296 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Guide 1        Guide 2       Gene 1       Gene 2  \\\n",
       "0      HPRT intron_1           6T_1  HPRT intron           6T   \n",
       "1      HPRT intron_1          UBC_1  HPRT intron          UBC   \n",
       "2      HPRT intron_1         CD81_1  HPRT intron         CD81   \n",
       "3      HPRT intron_1         CD81_2  HPRT intron         CD81   \n",
       "4      HPRT intron_1     HSP90AA1_1  HPRT intron     HSP90AA1   \n",
       "...              ...            ...          ...          ...   \n",
       "55291          UBB_6        PARP2_3          UBB        PARP2   \n",
       "55292          UBB_6       IMPDH1_3          UBB       IMPDH1   \n",
       "55293          UBB_6  HPRT intron_6          UBB  HPRT intron   \n",
       "55294          UBB_6       BCL2A1_3          UBB       BCL2A1   \n",
       "55295          UBB_6        CD81_10          UBB         CD81   \n",
       "\n",
       "                  Count Replicates Cell Line  \\\n",
       "0      1137.0;2438.0;1997.0;2175.0      786O   \n",
       "1      1112.0;2176.0;1385.0;1485.0      786O   \n",
       "2       796.0;2294.0;1579.0;1844.0      786O   \n",
       "3       839.0;2169.0;1229.0;1705.0      786O   \n",
       "4        689.0;1771.0;984.0;1526.0      786O   \n",
       "...                            ...       ...   \n",
       "55291       431.0;140.0;93.0;291.0    OVCAR8   \n",
       "55292      399.0;185.0;270.0;418.0    OVCAR8   \n",
       "55293      496.0;215.0;208.0;337.0    OVCAR8   \n",
       "55294      548.0;178.0;186.0;486.0    OVCAR8   \n",
       "55295       317.0;145.0;96.0;244.0    OVCAR8   \n",
       "\n",
       "                                            Condition  \n",
       "0      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "1      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "2      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "3      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "4      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "...                                               ...  \n",
       "55291  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "55292  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "55293  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "55294  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "55295  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  \n",
       "\n",
       "[55296 rows x 7 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synleth_lib[[\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Cell Line\", \"Condition\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "ac62c094",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.concat([counts_ref, synleth_lib[[\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Cell Line\", \"Condition\"]]])\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['najm_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "1501e898",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>6T_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>6T</td>\n",
       "      <td>1137.0;2438.0;1997.0;2175.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>UBC_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>UBC</td>\n",
       "      <td>1112.0;2176.0;1385.0;1485.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>CD81_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>CD81</td>\n",
       "      <td>796.0;2294.0;1579.0;1844.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>CD81_2</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>CD81</td>\n",
       "      <td>839.0;2169.0;1229.0;1705.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>HSP90AA1_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>689.0;1771.0;984.0;1526.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55291</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>PARP2_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>PARP2</td>\n",
       "      <td>431.0;140.0;93.0;291.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55292</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>IMPDH1_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>399.0;185.0;270.0;418.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55293</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>HPRT intron_6</td>\n",
       "      <td>UBB</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>496.0;215.0;208.0;337.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55294</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>BCL2A1_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>548.0;178.0;186.0;486.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55295</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>CD81_10</td>\n",
       "      <td>UBB</td>\n",
       "      <td>CD81</td>\n",
       "      <td>317.0;145.0;96.0;244.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55296 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Guide 1        Guide 2       Gene 1       Gene 2  \\\n",
       "0      HPRT intron_1           6T_1  HPRT intron           6T   \n",
       "1      HPRT intron_1          UBC_1  HPRT intron          UBC   \n",
       "2      HPRT intron_1         CD81_1  HPRT intron         CD81   \n",
       "3      HPRT intron_1         CD81_2  HPRT intron         CD81   \n",
       "4      HPRT intron_1     HSP90AA1_1  HPRT intron     HSP90AA1   \n",
       "...              ...            ...          ...          ...   \n",
       "55291          UBB_6        PARP2_3          UBB        PARP2   \n",
       "55292          UBB_6       IMPDH1_3          UBB       IMPDH1   \n",
       "55293          UBB_6  HPRT intron_6          UBB  HPRT intron   \n",
       "55294          UBB_6       BCL2A1_3          UBB       BCL2A1   \n",
       "55295          UBB_6        CD81_10          UBB         CD81   \n",
       "\n",
       "                  Count Replicates Type Sequencing Cell Line  \\\n",
       "0      1137.0;2438.0;1997.0;2175.0  NaN        NaN      786O   \n",
       "1      1112.0;2176.0;1385.0;1485.0  NaN        NaN      786O   \n",
       "2       796.0;2294.0;1579.0;1844.0  NaN        NaN      786O   \n",
       "3       839.0;2169.0;1229.0;1705.0  NaN        NaN      786O   \n",
       "4        689.0;1771.0;984.0;1526.0  NaN        NaN      786O   \n",
       "...                            ...  ...        ...       ...   \n",
       "55291       431.0;140.0;93.0;291.0  NaN        NaN    OVCAR8   \n",
       "55292      399.0;185.0;270.0;418.0  NaN        NaN    OVCAR8   \n",
       "55293      496.0;215.0;208.0;337.0  NaN        NaN    OVCAR8   \n",
       "55294      548.0;178.0;186.0;486.0  NaN        NaN    OVCAR8   \n",
       "55295       317.0;145.0;96.0;244.0  NaN        NaN    OVCAR8   \n",
       "\n",
       "                                            Condition     Study  \n",
       "0      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "1      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "2      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "3      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "4      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "...                                               ...       ...  \n",
       "55291  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55292  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55293  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55294  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55295  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "\n",
       "[55296 rows x 10 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "429f492f",
   "metadata": {},
   "source": [
    "## SL Score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "8ce7bc79",
   "metadata": {},
   "outputs": [],
   "source": [
    "SL_results = pd.read_csv(os.path.join(najm_loc, \"Supplementary Table 4 SynLet FDRs.txt\"), sep = '\\t')\n",
    "SL_results['Gene 1'] = [i.upper() for i in SL_results['Gene 1']]\n",
    "SL_results['Gene 2'] = [i.upper() for i in SL_results['Gene 2']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "b49808aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Group 1</th>\n",
       "      <th>Group 2</th>\n",
       "      <th>Source</th>\n",
       "      <th>SynLet q-value</th>\n",
       "      <th>Buffer q-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no 786O</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no A375</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no A549</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no HT29</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no Meljuso</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10928</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Apop</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.845612</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10929</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Mapk</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.857619</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10930</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Apop</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.867560</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10931</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>AKT</td>\n",
       "      <td>HT29</td>\n",
       "      <td>0.893805</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10932</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Predict</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.936601</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10933 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene 1  Gene 2 Group 1  Group 2      Source  SynLet q-value  \\\n",
       "0       AKT1    AKT1     AKT      AKT     no 786O        1.000000   \n",
       "1       AKT1    AKT1     AKT      AKT     no A375        1.000000   \n",
       "2       AKT1    AKT1     AKT      AKT     no A549        1.000000   \n",
       "3       AKT1    AKT1     AKT      AKT     no HT29        1.000000   \n",
       "4       AKT1    AKT1     AKT      AKT  no Meljuso        1.000000   \n",
       "...      ...     ...     ...      ...         ...             ...   \n",
       "10928   EEF2  BCL2A1    Ctrl     Apop     Meljuso        0.845612   \n",
       "10929   EEF2  MAP2K1    Ctrl     Mapk      OVCAR8        0.857619   \n",
       "10930   EEF2    BCL2    Ctrl     Apop     Meljuso        0.867560   \n",
       "10931   EEF2    AKT2    Ctrl      AKT        HT29        0.893805   \n",
       "10932   EEF2  IMPDH1    Ctrl  Predict      OVCAR8        0.936601   \n",
       "\n",
       "       Buffer q-value  \n",
       "0              0.0001  \n",
       "1              0.0001  \n",
       "2              0.0001  \n",
       "3              0.0001  \n",
       "4              0.0001  \n",
       "...               ...  \n",
       "10928          1.0000  \n",
       "10929          1.0000  \n",
       "10930          1.0000  \n",
       "10931          1.0000  \n",
       "10932          1.0000  \n",
       "\n",
       "[10933 rows x 7 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SL_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "c2b8b044",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_pair_sorted = []\n",
    "for i in range(SL_results.shape[0]):\n",
    "    gene1 = SL_results.loc[i, 'Gene 1']\n",
    "    gene2 = SL_results.loc[i, 'Gene 2']\n",
    "    \n",
    "    gene1, gene2 = sorted([gene1, gene2])\n",
    "    gene_pair_sorted.append('_'.join([gene1, gene2]))\n",
    "\n",
    "SL_results['gene pair sorted'] = gene_pair_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "6ec22465",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Group 1</th>\n",
       "      <th>Group 2</th>\n",
       "      <th>Source</th>\n",
       "      <th>SynLet q-value</th>\n",
       "      <th>Buffer q-value</th>\n",
       "      <th>gene pair sorted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no 786O</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>AKT1_AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no A375</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>AKT1_AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no A549</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>AKT1_AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no HT29</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>AKT1_AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT</td>\n",
       "      <td>AKT</td>\n",
       "      <td>no Meljuso</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>AKT1_AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10928</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Apop</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.845612</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BCL2A1_EEF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10929</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Mapk</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.857619</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EEF2_MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10930</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Apop</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.867560</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BCL2_EEF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10931</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>AKT</td>\n",
       "      <td>HT29</td>\n",
       "      <td>0.893805</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AKT2_EEF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10932</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>Ctrl</td>\n",
       "      <td>Predict</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.936601</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EEF2_IMPDH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10933 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene 1  Gene 2 Group 1  Group 2      Source  SynLet q-value  \\\n",
       "0       AKT1    AKT1     AKT      AKT     no 786O        1.000000   \n",
       "1       AKT1    AKT1     AKT      AKT     no A375        1.000000   \n",
       "2       AKT1    AKT1     AKT      AKT     no A549        1.000000   \n",
       "3       AKT1    AKT1     AKT      AKT     no HT29        1.000000   \n",
       "4       AKT1    AKT1     AKT      AKT  no Meljuso        1.000000   \n",
       "...      ...     ...     ...      ...         ...             ...   \n",
       "10928   EEF2  BCL2A1    Ctrl     Apop     Meljuso        0.845612   \n",
       "10929   EEF2  MAP2K1    Ctrl     Mapk      OVCAR8        0.857619   \n",
       "10930   EEF2    BCL2    Ctrl     Apop     Meljuso        0.867560   \n",
       "10931   EEF2    AKT2    Ctrl      AKT        HT29        0.893805   \n",
       "10932   EEF2  IMPDH1    Ctrl  Predict      OVCAR8        0.936601   \n",
       "\n",
       "       Buffer q-value gene pair sorted  \n",
       "0              0.0001        AKT1_AKT1  \n",
       "1              0.0001        AKT1_AKT1  \n",
       "2              0.0001        AKT1_AKT1  \n",
       "3              0.0001        AKT1_AKT1  \n",
       "4              0.0001        AKT1_AKT1  \n",
       "...               ...              ...  \n",
       "10928          1.0000      BCL2A1_EEF2  \n",
       "10929          1.0000      EEF2_MAP2K1  \n",
       "10930          1.0000        BCL2_EEF2  \n",
       "10931          1.0000        AKT2_EEF2  \n",
       "10932          1.0000      EEF2_IMPDH1  \n",
       "\n",
       "[10933 rows x 8 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SL_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "4d69a5c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_cell_lines = {'786O', 'A375', 'Meljuso', 'OVCAR8', 'A549', 'HT29'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "5fb97450",
   "metadata": {},
   "outputs": [],
   "source": [
    "single_cell_lines = np.array([True if i in all_cell_lines else False for i in SL_results['Source']])\n",
    "SL_results = SL_results.loc[single_cell_lines]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "034820f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "SL_results = SL_results.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "586a1226",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_GI = pd.DataFrame(columns = [\"Gene_A\", \"Gene_B\", \"Study_Source\", \"Cell_Line\", \"GI_Score\", \"GI_Cutoff\", \"Stat_Score\", \"Stat_Cutoff\"])\n",
    "all_GI[\"Gene_A\"] = SL_results['Gene 1']\n",
    "all_GI[\"Gene_B\"] = SL_results['Gene 2']\n",
    "all_GI[\"GI_Score\"] = [float(\"nan\")] * len(all_GI[\"Gene_A\"])\n",
    "all_GI[\"GI_Cutoff\"] = [float(\"nan\")] * len(all_GI[\"Gene_A\"])\n",
    "all_GI[\"Study_Source\"] = [study_name_to_pubmed_id['najm_data']] * len(all_GI[\"Gene_A\"])\n",
    "all_GI[\"Cell_Line\"] = SL_results['Source']\n",
    "all_GI[\"Stat_Score\"] = SL_results['SynLet q-value']\n",
    "all_GI[\"Stat_Cutoff\"] = [0.01] * len(all_GI[\"Gene_A\"])\n",
    "all_GI = all_GI.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "47d1cb72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>6T_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>6T</td>\n",
       "      <td>1137.0;2438.0;1997.0;2175.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>UBC_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>UBC</td>\n",
       "      <td>1112.0;2176.0;1385.0;1485.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>CD81_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>CD81</td>\n",
       "      <td>796.0;2294.0;1579.0;1844.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>CD81_2</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>CD81</td>\n",
       "      <td>839.0;2169.0;1229.0;1705.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HPRT intron_1</td>\n",
       "      <td>HSP90AA1_1</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>689.0;1771.0;984.0;1526.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>786O</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55291</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>PARP2_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>PARP2</td>\n",
       "      <td>431.0;140.0;93.0;291.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55292</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>IMPDH1_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>399.0;185.0;270.0;418.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55293</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>HPRT intron_6</td>\n",
       "      <td>UBB</td>\n",
       "      <td>HPRT intron</td>\n",
       "      <td>496.0;215.0;208.0;337.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55294</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>BCL2A1_3</td>\n",
       "      <td>UBB</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>548.0;178.0;186.0;486.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55295</th>\n",
       "      <td>UBB_6</td>\n",
       "      <td>CD81_10</td>\n",
       "      <td>UBB</td>\n",
       "      <td>CD81</td>\n",
       "      <td>317.0;145.0;96.0;244.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads</td>\n",
       "      <td>29251726</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55296 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Guide 1        Guide 2       Gene 1       Gene 2  \\\n",
       "0      HPRT intron_1           6T_1  HPRT intron           6T   \n",
       "1      HPRT intron_1          UBC_1  HPRT intron          UBC   \n",
       "2      HPRT intron_1         CD81_1  HPRT intron         CD81   \n",
       "3      HPRT intron_1         CD81_2  HPRT intron         CD81   \n",
       "4      HPRT intron_1     HSP90AA1_1  HPRT intron     HSP90AA1   \n",
       "...              ...            ...          ...          ...   \n",
       "55291          UBB_6        PARP2_3          UBB        PARP2   \n",
       "55292          UBB_6       IMPDH1_3          UBB       IMPDH1   \n",
       "55293          UBB_6  HPRT intron_6          UBB  HPRT intron   \n",
       "55294          UBB_6       BCL2A1_3          UBB       BCL2A1   \n",
       "55295          UBB_6        CD81_10          UBB         CD81   \n",
       "\n",
       "                  Count Replicates Type Sequencing Cell Line  \\\n",
       "0      1137.0;2438.0;1997.0;2175.0  NaN        NaN      786O   \n",
       "1      1112.0;2176.0;1385.0;1485.0  NaN        NaN      786O   \n",
       "2       796.0;2294.0;1579.0;1844.0  NaN        NaN      786O   \n",
       "3       839.0;2169.0;1229.0;1705.0  NaN        NaN      786O   \n",
       "4        689.0;1771.0;984.0;1526.0  NaN        NaN      786O   \n",
       "...                            ...  ...        ...       ...   \n",
       "55291       431.0;140.0;93.0;291.0  NaN        NaN    OVCAR8   \n",
       "55292      399.0;185.0;270.0;418.0  NaN        NaN    OVCAR8   \n",
       "55293      496.0;215.0;208.0;337.0  NaN        NaN    OVCAR8   \n",
       "55294      548.0;178.0;186.0;486.0  NaN        NaN    OVCAR8   \n",
       "55295       317.0;145.0;96.0;244.0  NaN        NaN    OVCAR8   \n",
       "\n",
       "                                            Condition     Study  \n",
       "0      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "1      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "2      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "3      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "4      pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "...                                               ...       ...  \n",
       "55291  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55292  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55293  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55294  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "55295  pDNA_Reads;Rep_A_Reads;Rep_B_Reads;Rep_C_Reads  29251726  \n",
       "\n",
       "[55296 rows x 10 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "356ad961",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_A</th>\n",
       "      <th>Gene_B</th>\n",
       "      <th>Study_Source</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>GI_Score</th>\n",
       "      <th>GI_Cutoff</th>\n",
       "      <th>Stat_Score</th>\n",
       "      <th>Stat_Cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>29251726</td>\n",
       "      <td>HT29</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001196</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT2</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>29251726</td>\n",
       "      <td>HT29</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001196</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT2</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>29251726</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.091691</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>29251726</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.091691</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>29251726</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.131109</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5041</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>BCL2A1</td>\n",
       "      <td>29251726</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.845612</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5042</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>29251726</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.857619</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5043</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>29251726</td>\n",
       "      <td>Meljuso</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.867560</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5044</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>29251726</td>\n",
       "      <td>HT29</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.893805</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5045</th>\n",
       "      <td>EEF2</td>\n",
       "      <td>IMPDH1</td>\n",
       "      <td>29251726</td>\n",
       "      <td>OVCAR8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.936601</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5046 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene_A  Gene_B Study_Source Cell_Line  GI_Score  GI_Cutoff  Stat_Score  \\\n",
       "0      AKT1    AKT2     29251726      HT29       0.0        0.0    0.001196   \n",
       "1      AKT2    AKT1     29251726      HT29       0.0        0.0    0.001196   \n",
       "2      AKT2    AKT3     29251726   Meljuso       0.0        0.0    0.091691   \n",
       "3      AKT3    AKT2     29251726   Meljuso       0.0        0.0    0.091691   \n",
       "4      AKT1    AKT3     29251726    OVCAR8       0.0        0.0    0.131109   \n",
       "...     ...     ...          ...       ...       ...        ...         ...   \n",
       "5041   EEF2  BCL2A1     29251726   Meljuso       0.0        0.0    0.845612   \n",
       "5042   EEF2  MAP2K1     29251726    OVCAR8       0.0        0.0    0.857619   \n",
       "5043   EEF2    BCL2     29251726   Meljuso       0.0        0.0    0.867560   \n",
       "5044   EEF2    AKT2     29251726      HT29       0.0        0.0    0.893805   \n",
       "5045   EEF2  IMPDH1     29251726    OVCAR8       0.0        0.0    0.936601   \n",
       "\n",
       "      Stat_Cutoff  \n",
       "0            0.01  \n",
       "1            0.01  \n",
       "2            0.01  \n",
       "3            0.01  \n",
       "4            0.01  \n",
       "...           ...  \n",
       "5041         0.01  \n",
       "5042         0.01  \n",
       "5043         0.01  \n",
       "5044         0.01  \n",
       "5045         0.01  \n",
       "\n",
       "[5046 rows x 8 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "d075f664",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          HT29\n",
       "1          HT29\n",
       "2       Meljuso\n",
       "3       Meljuso\n",
       "4        OVCAR8\n",
       "         ...   \n",
       "5041    Meljuso\n",
       "5042     OVCAR8\n",
       "5043    Meljuso\n",
       "5044       HT29\n",
       "5045     OVCAR8\n",
       "Name: Cell_Line, Length: 5046, dtype: object"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GI['Cell_Line']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "7dd0a150",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "1272\n",
      "---\n",
      "Only Stat cutoff is present...\n",
      "Counts reference...\n",
      "Number of double pairs: 32400\n",
      "Number of controls: 2646\n",
      "Number of singles: 20250\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = all_GI.copy(), study_controls = controls['najm_data'], study_conditions = study_conditions['najm_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "6cab373c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop duplicates (Najm Specific)\n",
    "db_inserts['score_ref'] = db_inserts['score_ref'].drop_duplicates(subset = ['gene_pair', 'cell_line_origin'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "1ade2b6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Not SL    1785\n",
       "SL          15\n",
       "Name: SL_or_not, dtype: int64"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 478,
   "id": "d1fccd80",
   "metadata": {},
   "outputs": [],
   "source": [
    "#db_inserts['score_ref']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 479,
   "id": "97bafe72",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Najm \n",
    "#db_inserts['score_ref'].loc[db_inserts['score_ref']['gene_1'] != db_inserts['score_ref']['gene_2'], 'SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 480,
   "id": "a1e0bb95",
   "metadata": {},
   "outputs": [],
   "source": [
    "#db_inserts['score_ref']['SL_or_not'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 482,
   "id": "7d9208ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 192\n",
      "Counts insert: 55296\n",
      "Score insert: 1800\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a611ac37",
   "metadata": {},
   "source": [
    "# Extra"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c06459d",
   "metadata": {},
   "source": [
    "## ShanTang Data (Saos2 and TT2 cell line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "7d90c4e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "shantang_loc = os.path.join(learning_goals_loc_general, \"ShanTang\", 'OS_DKOScreen_Final')\n",
    "lib_file = pd.read_csv(os.path.join(shantang_loc, \"lib_dko_os.txt\"), sep = '\\t')\n",
    "saos_counts = pd.read_csv(os.path.join(shantang_loc, \"OS_DKO_SaOS2_Miss1_20230209.txt\"), sep = ' ')\n",
    "tt2_counts = pd.read_csv(os.path.join(shantang_loc, \"OS_DKO_TT2_Miss1_20230209.txt\"), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "f5802b13",
   "metadata": {},
   "outputs": [],
   "source": [
    "for column in ['target_a_id', 'target_b_id']:\n",
    "    lib_file[column] = [i.split(':')[0] for i in lib_file[column]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 467,
   "id": "1a7a41a9",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'target_a_id'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/apps/python/3.9-2022.05/lib/python3.9/site-packages/pandas/core/indexes/base.py:3621\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3620\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3621\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3622\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/apps/python/3.9-2022.05/lib/python3.9/site-packages/pandas/_libs/index.pyx:136\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/apps/python/3.9-2022.05/lib/python3.9/site-packages/pandas/_libs/index.pyx:163\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:5198\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:5206\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'target_a_id'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Input \u001b[0;32mIn [467]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m idx \u001b[38;5;241m=\u001b[39m (\u001b[43mlib_file\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mtarget_a_id\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241m.\u001b[39misin([\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mluciferase\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mEGFP\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mLacZ\u001b[39m\u001b[38;5;124m'\u001b[39m])) \u001b[38;5;241m|\u001b[39m (lib_file[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtarget_b_id\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39misin([\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mluciferase\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mEGFP\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mLacZ\u001b[39m\u001b[38;5;124m'\u001b[39m]))\n",
      "File \u001b[0;32m/apps/python/3.9-2022.05/lib/python3.9/site-packages/pandas/core/frame.py:3505\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3503\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   3504\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 3505\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3506\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   3507\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m/apps/python/3.9-2022.05/lib/python3.9/site-packages/pandas/core/indexes/base.py:3623\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3621\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3622\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m-> 3623\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3624\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3625\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3626\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3627\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3628\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'target_a_id'"
     ]
    }
   ],
   "source": [
    "idx = (lib_file['target_a_id'].isin(['luciferase', 'EGFP', 'LacZ'])) | (lib_file['target_b_id'].isin(['luciferase', 'EGFP', 'LacZ']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "7db6dd61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5379"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "64394487",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove the 3 of them\n",
    "lib_file = lib_file[~idx]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "7330a436",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4_1__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRD4_2__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_2</td>\n",
       "      <td>ACAGGAGGAGGATTCGGCTG</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRD4_3__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_3</td>\n",
       "      <td>CACCAAACTCCTGAGCATCA</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRD4_4__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_4</td>\n",
       "      <td>TTCGACTGATGACTCTGAGG</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1_1__BRD4_1</td>\n",
       "      <td>CHEK1</td>\n",
       "      <td>CHEK1_1</td>\n",
       "      <td>TGGTATTGGAATAACTCACA</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4_1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59734</th>\n",
       "      <td>0Safe_safe_TMM_204550.337__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_TMM_204550.337</td>\n",
       "      <td>GAGAGTATACATTCAACC</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59735</th>\n",
       "      <td>0Safe_safe_U1_204550.3846__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U1_204550.3846</td>\n",
       "      <td>GGTTTAAACCCTTTAAAAT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59736</th>\n",
       "      <td>0Safe_safe_U1_204550.4037__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U1_204550.4037</td>\n",
       "      <td>GATGAAAGTCGAATCCTAT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59737</th>\n",
       "      <td>0Safe_safe_U2_204550.3016__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3016</td>\n",
       "      <td>GCACAAAAGTATTGGGGT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59738</th>\n",
       "      <td>0Safe_safe_U2_204550.3037__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe_safe_U2_204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57121 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id target_a_id  \\\n",
       "0                                         BRD4_1__BRD4_1        BRD4   \n",
       "1                                         BRD4_2__BRD4_1        BRD4   \n",
       "2                                         BRD4_3__BRD4_1        BRD4   \n",
       "3                                         BRD4_4__BRD4_1        BRD4   \n",
       "4                                        CHEK1_1__BRD4_1       CHEK1   \n",
       "...                                                  ...         ...   \n",
       "59734  0Safe_safe_TMM_204550.337__0Safe_safe_U2_20455...       0Safe   \n",
       "59735  0Safe_safe_U1_204550.3846__0Safe_safe_U2_20455...       0Safe   \n",
       "59736  0Safe_safe_U1_204550.4037__0Safe_safe_U2_20455...       0Safe   \n",
       "59737  0Safe_safe_U2_204550.3016__0Safe_safe_U2_20455...       0Safe   \n",
       "59738  0Safe_safe_U2_204550.3037__0Safe_safe_U2_20455...       0Safe   \n",
       "\n",
       "                      probe_a_id           probe_a_seq target_b_id  \\\n",
       "0                         BRD4_1  GGGAACAATAAAGAAGCGCT        BRD4   \n",
       "1                         BRD4_2  ACAGGAGGAGGATTCGGCTG        BRD4   \n",
       "2                         BRD4_3  CACCAAACTCCTGAGCATCA        BRD4   \n",
       "3                         BRD4_4  TTCGACTGATGACTCTGAGG        BRD4   \n",
       "4                        CHEK1_1  TGGTATTGGAATAACTCACA        BRD4   \n",
       "...                          ...                   ...         ...   \n",
       "59734  0Safe_safe_TMM_204550.337    GAGAGTATACATTCAACC       0Safe   \n",
       "59735  0Safe_safe_U1_204550.3846   GGTTTAAACCCTTTAAAAT       0Safe   \n",
       "59736  0Safe_safe_U1_204550.4037   GATGAAAGTCGAATCCTAT       0Safe   \n",
       "59737  0Safe_safe_U2_204550.3016    GCACAAAAGTATTGGGGT       0Safe   \n",
       "59738  0Safe_safe_U2_204550.3037     GTAGCTGACATTGCTAT       0Safe   \n",
       "\n",
       "                      probe_b_id           probe_b_seq  \n",
       "0                         BRD4_1  GGGAACAATAAAGAAGCGCT  \n",
       "1                         BRD4_1  GGGAACAATAAAGAAGCGCT  \n",
       "2                         BRD4_1  GGGAACAATAAAGAAGCGCT  \n",
       "3                         BRD4_1  GGGAACAATAAAGAAGCGCT  \n",
       "4                         BRD4_1  GGGAACAATAAAGAAGCGCT  \n",
       "...                          ...                   ...  \n",
       "59734  0Safe_safe_U2_204550.3037     GTAGCTGACATTGCTAT  \n",
       "59735  0Safe_safe_U2_204550.3037     GTAGCTGACATTGCTAT  \n",
       "59736  0Safe_safe_U2_204550.3037     GTAGCTGACATTGCTAT  \n",
       "59737  0Safe_safe_U2_204550.3037     GTAGCTGACATTGCTAT  \n",
       "59738  0Safe_safe_U2_204550.3037     GTAGCTGACATTGCTAT  \n",
       "\n",
       "[57121 rows x 7 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "8a2da01d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>S0A</th>\n",
       "      <th>S0B</th>\n",
       "      <th>S0C</th>\n",
       "      <th>SEA</th>\n",
       "      <th>SEB</th>\n",
       "      <th>SEC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>9</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>19</td>\n",
       "      <td>33</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>32</td>\n",
       "      <td>29</td>\n",
       "      <td>39</td>\n",
       "      <td>32</td>\n",
       "      <td>62</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>22</td>\n",
       "      <td>36</td>\n",
       "      <td>39</td>\n",
       "      <td>31</td>\n",
       "      <td>39</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>30</td>\n",
       "      <td>36</td>\n",
       "      <td>24</td>\n",
       "      <td>64</td>\n",
       "      <td>95</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_GE_204...</td>\n",
       "      <td>23</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>31</td>\n",
       "      <td>58</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62495</th>\n",
       "      <td>YAP1_4__WEE1_4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62496</th>\n",
       "      <td>YAP1_4__YAP1_1</td>\n",
       "      <td>35</td>\n",
       "      <td>59</td>\n",
       "      <td>62</td>\n",
       "      <td>56</td>\n",
       "      <td>124</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62497</th>\n",
       "      <td>YAP1_4__YAP1_2</td>\n",
       "      <td>33</td>\n",
       "      <td>44</td>\n",
       "      <td>60</td>\n",
       "      <td>49</td>\n",
       "      <td>54</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62498</th>\n",
       "      <td>YAP1_4__YAP1_3</td>\n",
       "      <td>40</td>\n",
       "      <td>50</td>\n",
       "      <td>48</td>\n",
       "      <td>20</td>\n",
       "      <td>17</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62499</th>\n",
       "      <td>YAP1_4__YAP1_4</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>38</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62500 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id  S0A  S0B  S0C  SEA  \\\n",
       "0      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...    9   12   12   19   \n",
       "1      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...   32   29   39   32   \n",
       "2      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...   22   36   39   31   \n",
       "3      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...   30   36   24   64   \n",
       "4      0Safe_safe_ACOC_204550.4522__0Safe_safe_GE_204...   23   30   43   31   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "62495                                     YAP1_4__WEE1_4    5    5   11    1   \n",
       "62496                                     YAP1_4__YAP1_1   35   59   62   56   \n",
       "62497                                     YAP1_4__YAP1_2   33   44   60   49   \n",
       "62498                                     YAP1_4__YAP1_3   40   50   48   20   \n",
       "62499                                     YAP1_4__YAP1_4    9   18   17   23   \n",
       "\n",
       "       SEB  SEC  \n",
       "0       33   30  \n",
       "1       62   90  \n",
       "2       39   26  \n",
       "3       95  101  \n",
       "4       58   61  \n",
       "...    ...  ...  \n",
       "62495    2    8  \n",
       "62496  124   72  \n",
       "62497   54   98  \n",
       "62498   17   42  \n",
       "62499   38   23  \n",
       "\n",
       "[62500 rows x 7 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "saos_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "00c8ebd1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>TEA</th>\n",
       "      <th>T0A</th>\n",
       "      <th>TEC</th>\n",
       "      <th>TEB</th>\n",
       "      <th>T0B</th>\n",
       "      <th>T0C</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>24</td>\n",
       "      <td>40</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...</td>\n",
       "      <td>36</td>\n",
       "      <td>32</td>\n",
       "      <td>49</td>\n",
       "      <td>36</td>\n",
       "      <td>51</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>26</td>\n",
       "      <td>23</td>\n",
       "      <td>34</td>\n",
       "      <td>36</td>\n",
       "      <td>22</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...</td>\n",
       "      <td>47</td>\n",
       "      <td>36</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4522__0Safe_safe_GE_204...</td>\n",
       "      <td>59</td>\n",
       "      <td>28</td>\n",
       "      <td>56</td>\n",
       "      <td>78</td>\n",
       "      <td>92</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62495</th>\n",
       "      <td>luciferase:1028-1047_luciferase_-__YAP1_3</td>\n",
       "      <td>40</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "      <td>28</td>\n",
       "      <td>85</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62496</th>\n",
       "      <td>luciferase:1028-1047_luciferase_-__YAP1_4</td>\n",
       "      <td>11</td>\n",
       "      <td>55</td>\n",
       "      <td>57</td>\n",
       "      <td>60</td>\n",
       "      <td>48</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62497</th>\n",
       "      <td>luciferase:1028-1047_luciferase_-__luciferase:...</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62498</th>\n",
       "      <td>luciferase:1028-1047_luciferase_-__luciferase:...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62499</th>\n",
       "      <td>luciferase:1028-1047_luciferase_-__luciferase:...</td>\n",
       "      <td>86</td>\n",
       "      <td>69</td>\n",
       "      <td>62</td>\n",
       "      <td>98</td>\n",
       "      <td>44</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62500 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id  TEA  T0A  TEC  TEB  \\\n",
       "0      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...   15   11   24   40   \n",
       "1      0Safe_safe_ACOC_204550.4522__0Safe_safe_ACOC_2...   36   32   49   36   \n",
       "2      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...   26   23   34   36   \n",
       "3      0Safe_safe_ACOC_204550.4522__0Safe_safe_DTKP_2...   47   36   64   64   \n",
       "4      0Safe_safe_ACOC_204550.4522__0Safe_safe_GE_204...   59   28   56   78   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "62495          luciferase:1028-1047_luciferase_-__YAP1_3   40   82   80   28   \n",
       "62496          luciferase:1028-1047_luciferase_-__YAP1_4   11   55   57   60   \n",
       "62497  luciferase:1028-1047_luciferase_-__luciferase:...   18   18   15   23   \n",
       "62498  luciferase:1028-1047_luciferase_-__luciferase:...    0    0    0    0   \n",
       "62499  luciferase:1028-1047_luciferase_-__luciferase:...   86   69   62   98   \n",
       "\n",
       "       T0B  T0C  \n",
       "0        9    9  \n",
       "1       51   39  \n",
       "2       22   40  \n",
       "3       38   53  \n",
       "4       92   71  \n",
       "...    ...  ...  \n",
       "62495   85   78  \n",
       "62496   48   49  \n",
       "62497   10   24  \n",
       "62498    1    0  \n",
       "62499   44   76  \n",
       "\n",
       "[62500 rows x 7 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tt2_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "7d054c1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "for column in ['target_a_id', 'probe_a_id', 'target_b_id', 'probe_b_id']:\n",
    "    lib_file[column] = ['-'.join(i.split('_')) for i in lib_file[column]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ef268a6",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "93aa2ade",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref = pd.DataFrame(columns = [\"Guide_ID\", \"Sequence\"])\n",
    "sgRNA_ref['Guide_ID'] = pd.concat([lib_file['probe_a_id'], lib_file['probe_b_id']])\n",
    "sgRNA_ref['Sequence'] = pd.concat([lib_file['probe_a_seq'], lib_file['probe_b_seq']])\n",
    "sgRNA_ref.index = sgRNA_ref['Guide_ID']\n",
    "sgRNA_ref = sgRNA_ref.drop_duplicates()\n",
    "sgRNA_ref = sgRNA_ref.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "d71be2db",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref['Guide_ID'] = [i.upper() for i in sgRNA_ref['Guide_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "accc9069",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide_ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRD4-2</td>\n",
       "      <td>ACAGGAGGAGGATTCGGCTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRD4-3</td>\n",
       "      <td>CACCAAACTCCTGAGCATCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRD4-4</td>\n",
       "      <td>TTCGACTGATGACTCTGAGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1-1</td>\n",
       "      <td>TGGTATTGGAATAACTCACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>0SAFE-SAFE-TMM-204550.337</td>\n",
       "      <td>GAGAGTATACATTCAACC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.3846</td>\n",
       "      <td>GGTTTAAACCCTTTAAAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.4037</td>\n",
       "      <td>GATGAAAGTCGAATCCTAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>0SAFE-SAFE-U2-204550.3016</td>\n",
       "      <td>GCACAAAAGTATTGGGGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>0SAFE-SAFE-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>239 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Guide_ID              Sequence\n",
       "0                       BRD4-1  GGGAACAATAAAGAAGCGCT\n",
       "1                       BRD4-2  ACAGGAGGAGGATTCGGCTG\n",
       "2                       BRD4-3  CACCAAACTCCTGAGCATCA\n",
       "3                       BRD4-4  TTCGACTGATGACTCTGAGG\n",
       "4                      CHEK1-1  TGGTATTGGAATAACTCACA\n",
       "..                         ...                   ...\n",
       "234  0SAFE-SAFE-TMM-204550.337    GAGAGTATACATTCAACC\n",
       "235  0SAFE-SAFE-U1-204550.3846   GGTTTAAACCCTTTAAAAT\n",
       "236  0SAFE-SAFE-U1-204550.4037   GATGAAAGTCGAATCCTAT\n",
       "237  0SAFE-SAFE-U2-204550.3016    GCACAAAAGTATTGGGGT\n",
       "238  0SAFE-SAFE-U2-204550.3037     GTAGCTGACATTGCTAT\n",
       "\n",
       "[239 rows x 2 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgRNA_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "bd73a4fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRD4-2</td>\n",
       "      <td>ACAGGAGGAGGATTCGGCTG</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRD4-3</td>\n",
       "      <td>CACCAAACTCCTGAGCATCA</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRD4-4</td>\n",
       "      <td>TTCGACTGATGACTCTGAGG</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1-1</td>\n",
       "      <td>TGGTATTGGAATAACTCACA</td>\n",
       "      <td>CHEK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>0Safe-safe-TMM-204550.337</td>\n",
       "      <td>GAGAGTATACATTCAACC</td>\n",
       "      <td>0Safe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>0Safe-safe-U1-204550.3846</td>\n",
       "      <td>GGTTTAAACCCTTTAAAAT</td>\n",
       "      <td>0Safe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>0Safe-safe-U1-204550.4037</td>\n",
       "      <td>GATGAAAGTCGAATCCTAT</td>\n",
       "      <td>0Safe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>0Safe-safe-U2-204550.3016</td>\n",
       "      <td>GCACAAAAGTATTGGGGT</td>\n",
       "      <td>0Safe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>0Safe</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>239 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0                       BRD4-1  GGGAACAATAAAGAAGCGCT              BRD4\n",
       "1                       BRD4-2  ACAGGAGGAGGATTCGGCTG              BRD4\n",
       "2                       BRD4-3  CACCAAACTCCTGAGCATCA              BRD4\n",
       "3                       BRD4-4  TTCGACTGATGACTCTGAGG              BRD4\n",
       "4                      CHEK1-1  TGGTATTGGAATAACTCACA             CHEK1\n",
       "..                         ...                   ...               ...\n",
       "234  0Safe-safe-TMM-204550.337    GAGAGTATACATTCAACC             0Safe\n",
       "235  0Safe-safe-U1-204550.3846   GGTTTAAACCCTTTAAAAT             0Safe\n",
       "236  0Safe-safe-U1-204550.4037   GATGAAAGTCGAATCCTAT             0Safe\n",
       "237  0Safe-safe-U2-204550.3016    GCACAAAAGTATTGGGGT             0Safe\n",
       "238  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT             0Safe\n",
       "\n",
       "[239 rows x 3 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "41845bdf",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sgRNA_ref.copy()\n",
    "sequence_ref['Target'] = [i.split('-')[0] for i in sequence_ref['Guide_ID']]\n",
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']\n",
    "sequence_ref.loc[sequence_ref['sgRNA_target_name'].isin(controls['shantang_2CL_data']), 'sgRNA_target_name'] = 'control'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "cd0fabfb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRD4-2</td>\n",
       "      <td>ACAGGAGGAGGATTCGGCTG</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRD4-3</td>\n",
       "      <td>CACCAAACTCCTGAGCATCA</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRD4-4</td>\n",
       "      <td>TTCGACTGATGACTCTGAGG</td>\n",
       "      <td>BRD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1-1</td>\n",
       "      <td>TGGTATTGGAATAACTCACA</td>\n",
       "      <td>CHEK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>0SAFE-SAFE-TMM-204550.337</td>\n",
       "      <td>GAGAGTATACATTCAACC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.3846</td>\n",
       "      <td>GGTTTAAACCCTTTAAAAT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.4037</td>\n",
       "      <td>GATGAAAGTCGAATCCTAT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>0SAFE-SAFE-U2-204550.3016</td>\n",
       "      <td>GCACAAAAGTATTGGGGT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>0SAFE-SAFE-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>239 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0                       BRD4-1  GGGAACAATAAAGAAGCGCT              BRD4\n",
       "1                       BRD4-2  ACAGGAGGAGGATTCGGCTG              BRD4\n",
       "2                       BRD4-3  CACCAAACTCCTGAGCATCA              BRD4\n",
       "3                       BRD4-4  TTCGACTGATGACTCTGAGG              BRD4\n",
       "4                      CHEK1-1  TGGTATTGGAATAACTCACA             CHEK1\n",
       "..                         ...                   ...               ...\n",
       "234  0SAFE-SAFE-TMM-204550.337    GAGAGTATACATTCAACC           control\n",
       "235  0SAFE-SAFE-U1-204550.3846   GGTTTAAACCCTTTAAAAT           control\n",
       "236  0SAFE-SAFE-U1-204550.4037   GATGAAAGTCGAATCCTAT           control\n",
       "237  0SAFE-SAFE-U2-204550.3016    GCACAAAAGTATTGGGGT           control\n",
       "238  0SAFE-SAFE-U2-204550.3037     GTAGCTGACATTGCTAT           control\n",
       "\n",
       "[239 rows x 3 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "6e4545ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(sequence_ref['sgRNA_target_name'] == 'control').sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a642d463",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ec170e0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "saos_counts = lib_file.merge(saos_counts, how = 'left', left_on = 'construct_id', right_on = 'construct_id')\n",
    "tt2_counts = lib_file.merge(tt2_counts, how = 'left', left_on = 'construct_id', right_on = 'construct_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "22949a11",
   "metadata": {},
   "outputs": [],
   "source": [
    "saos_counts['Count Replicates'] = saos_counts[['S0A', 'S0B', 'S0C', 'SEA', 'SEB', 'SEC']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "saos_counts['Cell Line'] = ['SAOS2'] * saos_counts.shape[0]\n",
    "saos_counts[\"Condition\"] = [';'.join(['S0A', 'S0B', 'S0C', 'SEA', 'SEB', 'SEC'])] * saos_counts.shape[0]\n",
    "\n",
    "tt2_counts['Count Replicates'] = tt2_counts[['T0A', 'T0B', 'T0C', 'TEA', 'TEB', 'TEC']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "\n",
    "tt2_counts['Cell Line'] = ['TT2'] * tt2_counts.shape[0]\n",
    "tt2_counts[\"Condition\"] = [';'.join(['T0A', 'T0B', 'T0C', 'TEA', 'TEB', 'TEC'])] * tt2_counts.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e6b1af77",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "      <th>S0A</th>\n",
       "      <th>S0B</th>\n",
       "      <th>S0C</th>\n",
       "      <th>SEA</th>\n",
       "      <th>SEB</th>\n",
       "      <th>SEC</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4_1__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>163</td>\n",
       "      <td>110</td>\n",
       "      <td>163</td>\n",
       "      <td>78</td>\n",
       "      <td>103</td>\n",
       "      <td>101</td>\n",
       "      <td>163;110;163;78;103;101</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRD4_2__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-2</td>\n",
       "      <td>ACAGGAGGAGGATTCGGCTG</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>719</td>\n",
       "      <td>525</td>\n",
       "      <td>845</td>\n",
       "      <td>308</td>\n",
       "      <td>449</td>\n",
       "      <td>434</td>\n",
       "      <td>719;525;845;308;449;434</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRD4_3__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-3</td>\n",
       "      <td>CACCAAACTCCTGAGCATCA</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>170</td>\n",
       "      <td>164</td>\n",
       "      <td>240</td>\n",
       "      <td>65</td>\n",
       "      <td>98</td>\n",
       "      <td>125</td>\n",
       "      <td>170;164;240;65;98;125</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRD4_4__BRD4_1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-4</td>\n",
       "      <td>TTCGACTGATGACTCTGAGG</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>194</td>\n",
       "      <td>179</td>\n",
       "      <td>231</td>\n",
       "      <td>107</td>\n",
       "      <td>134</td>\n",
       "      <td>177</td>\n",
       "      <td>194;179;231;107;134;177</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1_1__BRD4_1</td>\n",
       "      <td>CHEK1</td>\n",
       "      <td>CHEK1-1</td>\n",
       "      <td>TGGTATTGGAATAACTCACA</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>GGGAACAATAAAGAAGCGCT</td>\n",
       "      <td>24</td>\n",
       "      <td>21</td>\n",
       "      <td>35</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>24;21;35;14;20;12</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57116</th>\n",
       "      <td>0Safe_safe_TMM_204550.337__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-TMM-204550.337</td>\n",
       "      <td>GAGAGTATACATTCAACC</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>5;11;12;22;20;18</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57117</th>\n",
       "      <td>0Safe_safe_U1_204550.3846__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.3846</td>\n",
       "      <td>GGTTTAAACCCTTTAAAAT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>37</td>\n",
       "      <td>28</td>\n",
       "      <td>12;14;11;18;37;28</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57118</th>\n",
       "      <td>0Safe_safe_U1_204550.4037__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4037</td>\n",
       "      <td>GATGAAAGTCGAATCCTAT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>24</td>\n",
       "      <td>25</td>\n",
       "      <td>4;8;10;11;24;25</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57119</th>\n",
       "      <td>0Safe_safe_U2_204550.3016__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3016</td>\n",
       "      <td>GCACAAAAGTATTGGGGT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>56</td>\n",
       "      <td>31</td>\n",
       "      <td>86</td>\n",
       "      <td>70</td>\n",
       "      <td>76</td>\n",
       "      <td>106</td>\n",
       "      <td>56;31;86;70;76;106</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57120</th>\n",
       "      <td>0Safe_safe_U2_204550.3037__0Safe_safe_U2_20455...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>GTAGCTGACATTGCTAT</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>25</td>\n",
       "      <td>38</td>\n",
       "      <td>26</td>\n",
       "      <td>15;8;21;25;38;26</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57121 rows  16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id target_a_id  \\\n",
       "0                                         BRD4_1__BRD4_1        BRD4   \n",
       "1                                         BRD4_2__BRD4_1        BRD4   \n",
       "2                                         BRD4_3__BRD4_1        BRD4   \n",
       "3                                         BRD4_4__BRD4_1        BRD4   \n",
       "4                                        CHEK1_1__BRD4_1       CHEK1   \n",
       "...                                                  ...         ...   \n",
       "57116  0Safe_safe_TMM_204550.337__0Safe_safe_U2_20455...       0Safe   \n",
       "57117  0Safe_safe_U1_204550.3846__0Safe_safe_U2_20455...       0Safe   \n",
       "57118  0Safe_safe_U1_204550.4037__0Safe_safe_U2_20455...       0Safe   \n",
       "57119  0Safe_safe_U2_204550.3016__0Safe_safe_U2_20455...       0Safe   \n",
       "57120  0Safe_safe_U2_204550.3037__0Safe_safe_U2_20455...       0Safe   \n",
       "\n",
       "                      probe_a_id           probe_a_seq target_b_id  \\\n",
       "0                         BRD4-1  GGGAACAATAAAGAAGCGCT        BRD4   \n",
       "1                         BRD4-2  ACAGGAGGAGGATTCGGCTG        BRD4   \n",
       "2                         BRD4-3  CACCAAACTCCTGAGCATCA        BRD4   \n",
       "3                         BRD4-4  TTCGACTGATGACTCTGAGG        BRD4   \n",
       "4                        CHEK1-1  TGGTATTGGAATAACTCACA        BRD4   \n",
       "...                          ...                   ...         ...   \n",
       "57116  0Safe-safe-TMM-204550.337    GAGAGTATACATTCAACC       0Safe   \n",
       "57117  0Safe-safe-U1-204550.3846   GGTTTAAACCCTTTAAAAT       0Safe   \n",
       "57118  0Safe-safe-U1-204550.4037   GATGAAAGTCGAATCCTAT       0Safe   \n",
       "57119  0Safe-safe-U2-204550.3016    GCACAAAAGTATTGGGGT       0Safe   \n",
       "57120  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT       0Safe   \n",
       "\n",
       "                      probe_b_id           probe_b_seq  S0A  S0B  S0C  SEA  \\\n",
       "0                         BRD4-1  GGGAACAATAAAGAAGCGCT  163  110  163   78   \n",
       "1                         BRD4-1  GGGAACAATAAAGAAGCGCT  719  525  845  308   \n",
       "2                         BRD4-1  GGGAACAATAAAGAAGCGCT  170  164  240   65   \n",
       "3                         BRD4-1  GGGAACAATAAAGAAGCGCT  194  179  231  107   \n",
       "4                         BRD4-1  GGGAACAATAAAGAAGCGCT   24   21   35   14   \n",
       "...                          ...                   ...  ...  ...  ...  ...   \n",
       "57116  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT    5   11   12   22   \n",
       "57117  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT   12   14   11   18   \n",
       "57118  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT    4    8   10   11   \n",
       "57119  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT   56   31   86   70   \n",
       "57120  0Safe-safe-U2-204550.3037     GTAGCTGACATTGCTAT   15    8   21   25   \n",
       "\n",
       "       SEB  SEC         Count Replicates Cell Line                Condition  \n",
       "0      103  101   163;110;163;78;103;101     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "1      449  434  719;525;845;308;449;434     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "2       98  125    170;164;240;65;98;125     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "3      134  177  194;179;231;107;134;177     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "4       20   12        24;21;35;14;20;12     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "...    ...  ...                      ...       ...                      ...  \n",
       "57116   20   18         5;11;12;22;20;18     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "57117   37   28        12;14;11;18;37;28     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "57118   24   25          4;8;10;11;24;25     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "57119   76  106       56;31;86;70;76;106     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "57120   38   26         15;8;21;25;38;26     SAOS2  S0A;S0B;S0C;SEA;SEB;SEC  \n",
       "\n",
       "[57121 rows x 16 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "saos_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "4d61df19",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "4d330ebe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['diehl_data', 'han_data', 'horlbeck_data', 'ito_data', 'parrish_data', 'shen_data', 'thompson_data', 'wong_data', 'zhao_data', 'shantang_data', 'najm_data', 'shantang_2CL_data'])"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_name_to_pubmed_id.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "889cb334",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Guide 1'] = saos_counts['probe_a_id'].tolist() + tt2_counts['probe_a_id'].tolist()\n",
    "counts_ref['Guide 2'] = saos_counts['probe_b_id'].tolist() + tt2_counts['probe_b_id'].tolist()\n",
    "counts_ref['Gene 1'] = saos_counts['target_a_id'].tolist() + tt2_counts['target_a_id'].tolist()\n",
    "counts_ref['Gene 2'] = saos_counts['target_b_id'].tolist() + tt2_counts['target_b_id'].tolist()\n",
    "counts_ref['Count Replicates'] = saos_counts['Count Replicates'].tolist() + tt2_counts['Count Replicates'].tolist()\n",
    "counts_ref['Cell Line'] = saos_counts['Cell Line'].tolist() + tt2_counts['Cell Line'].tolist()\n",
    "counts_ref['Condition'] = saos_counts['Condition'].tolist() + tt2_counts['Condition'].tolist()\n",
    "counts_ref['Sequencing'] = ['Combinatorial CRISPR'] * counts_ref.shape[0]\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['shantang_2CL_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "d1920bc5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>163;110;163;78;103;101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRD4-2</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>719;525;845;308;449;434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRD4-3</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>170;164;240;65;98;125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRD4-4</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>194;179;231;107;134;177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHEK1-1</td>\n",
       "      <td>BRD4-1</td>\n",
       "      <td>CHEK1</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>24;21;35;14;20;12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>SAOS2</td>\n",
       "      <td>S0A;S0B;S0C;SEA;SEB;SEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114237</th>\n",
       "      <td>0Safe-safe-TMM-204550.337</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>22;12;15;6;20;14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>TT2</td>\n",
       "      <td>T0A;T0B;T0C;TEA;TEB;TEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114238</th>\n",
       "      <td>0Safe-safe-U1-204550.3846</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>21;37;25;16;24;14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>TT2</td>\n",
       "      <td>T0A;T0B;T0C;TEA;TEB;TEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114239</th>\n",
       "      <td>0Safe-safe-U1-204550.4037</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>10;3;8;12;81;14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>TT2</td>\n",
       "      <td>T0A;T0B;T0C;TEA;TEB;TEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114240</th>\n",
       "      <td>0Safe-safe-U2-204550.3016</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>50;50;64;37;59;73</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>TT2</td>\n",
       "      <td>T0A;T0B;T0C;TEA;TEB;TEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114241</th>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>0Safe-safe-U2-204550.3037</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>9;27;13;16;18;18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>TT2</td>\n",
       "      <td>T0A;T0B;T0C;TEA;TEB;TEC</td>\n",
       "      <td>Shan_2CL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114242 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Guide 1                    Guide 2 Gene 1 Gene 2  \\\n",
       "0                          BRD4-1                     BRD4-1   BRD4   BRD4   \n",
       "1                          BRD4-2                     BRD4-1   BRD4   BRD4   \n",
       "2                          BRD4-3                     BRD4-1   BRD4   BRD4   \n",
       "3                          BRD4-4                     BRD4-1   BRD4   BRD4   \n",
       "4                         CHEK1-1                     BRD4-1  CHEK1   BRD4   \n",
       "...                           ...                        ...    ...    ...   \n",
       "114237  0Safe-safe-TMM-204550.337  0Safe-safe-U2-204550.3037  0Safe  0Safe   \n",
       "114238  0Safe-safe-U1-204550.3846  0Safe-safe-U2-204550.3037  0Safe  0Safe   \n",
       "114239  0Safe-safe-U1-204550.4037  0Safe-safe-U2-204550.3037  0Safe  0Safe   \n",
       "114240  0Safe-safe-U2-204550.3016  0Safe-safe-U2-204550.3037  0Safe  0Safe   \n",
       "114241  0Safe-safe-U2-204550.3037  0Safe-safe-U2-204550.3037  0Safe  0Safe   \n",
       "\n",
       "               Count Replicates Type            Sequencing Cell Line  \\\n",
       "0        163;110;163;78;103;101  NaN  Combinatorial CRISPR     SAOS2   \n",
       "1       719;525;845;308;449;434  NaN  Combinatorial CRISPR     SAOS2   \n",
       "2         170;164;240;65;98;125  NaN  Combinatorial CRISPR     SAOS2   \n",
       "3       194;179;231;107;134;177  NaN  Combinatorial CRISPR     SAOS2   \n",
       "4             24;21;35;14;20;12  NaN  Combinatorial CRISPR     SAOS2   \n",
       "...                         ...  ...                   ...       ...   \n",
       "114237         22;12;15;6;20;14  NaN  Combinatorial CRISPR       TT2   \n",
       "114238        21;37;25;16;24;14  NaN  Combinatorial CRISPR       TT2   \n",
       "114239          10;3;8;12;81;14  NaN  Combinatorial CRISPR       TT2   \n",
       "114240        50;50;64;37;59;73  NaN  Combinatorial CRISPR       TT2   \n",
       "114241         9;27;13;16;18;18  NaN  Combinatorial CRISPR       TT2   \n",
       "\n",
       "                      Condition     Study  \n",
       "0       S0A;S0B;S0C;SEA;SEB;SEC  Shan_2CL  \n",
       "1       S0A;S0B;S0C;SEA;SEB;SEC  Shan_2CL  \n",
       "2       S0A;S0B;S0C;SEA;SEB;SEC  Shan_2CL  \n",
       "3       S0A;S0B;S0C;SEA;SEB;SEC  Shan_2CL  \n",
       "4       S0A;S0B;S0C;SEA;SEB;SEC  Shan_2CL  \n",
       "...                         ...       ...  \n",
       "114237  T0A;T0B;T0C;TEA;TEB;TEC  Shan_2CL  \n",
       "114238  T0A;T0B;T0C;TEA;TEB;TEC  Shan_2CL  \n",
       "114239  T0A;T0B;T0C;TEA;TEB;TEC  Shan_2CL  \n",
       "114240  T0A;T0B;T0C;TEA;TEB;TEC  Shan_2CL  \n",
       "114241  T0A;T0B;T0C;TEA;TEB;TEC  Shan_2CL  \n",
       "\n",
       "[114242 rows x 10 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7fe5860",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "79e025e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# there are no scores available, generate them instead"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "82851c9d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are no scores, but there are counts...Generating Placeholder...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurmtmp.14748441/ipykernel_173464/2751219690.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  curr_counts['sorted_genes'] = ['_'.join(sorted([curr_counts['Gene 1'].iloc[i], curr_counts['Gene 2'].iloc[i]])) for i in range(curr_counts.shape[0])]\n",
      "/tmp/slurmtmp.14748441/ipykernel_173464/2751219690.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  curr_counts.drop_duplicates(subset = ['sorted_genes', 'Cell Line'], keep = 'first', inplace = True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "0\n",
      "---\n",
      "No scores/stats cutoffs are available, possibly generated. Setting all to be NOT SL\n",
      "Counts reference...\n",
      "Number of double pairs: 98560\n",
      "Number of controls: 450\n",
      "Number of singles: 15232\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = None, study_controls = controls['shantang_2CL_data'], study_conditions = study_conditions['shantang_2CL_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "fd4803ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 239\n",
      "Counts insert: 114242\n",
      "Score insert: 3080\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c89060c",
   "metadata": {},
   "source": [
    "## Shuai Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "108e3a30",
   "metadata": {},
   "outputs": [],
   "source": [
    "shuai_loc = os.path.join(learning_goals_loc_general, \"Shuai\")\n",
    "lib_file = pd.read_csv(os.path.join(shuai_loc, \"cdko_lib.txt\"), sep = '\\t', skiprows = 3)\n",
    "counts_file = pd.read_csv(os.path.join(shuai_loc, \"CDCDKOLib_20230201003949_counts_combined.txt\"), sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "abc9f7f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "for column in ['target_a_id', 'probe_a_id', 'target_b_id', 'probe_b_id']:\n",
    "    lib_file[column] = ['-'.join(i.split('_')) for i in lib_file[column]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "075df166",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_file['target_a_id'] = [i.split('-')[0] for i in lib_file['probe_a_id']]\n",
    "lib_file['target_b_id'] = [i.split('-')[0] for i in lib_file['probe_b_id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "ec869bfe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FTH1_1__FTH1_1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FTH1_1__FTH1_2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-2</td>\n",
       "      <td>TGTTCACCTTGATATCCTGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTH1_1__FTH1_3</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-3</td>\n",
       "      <td>AATCTTCCTTCAGGATATCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FTH1_1__ANAPC10_1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>ANAPC10-1</td>\n",
       "      <td>GACATACCCGAATTTCTTGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FTH1_1__ANAPC10_2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>ANAPC10-2</td>\n",
       "      <td>ATTTAGTGAACATCCAATTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52895</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5816-1</td>\n",
       "      <td>GTGGAGTGGCGGAATCAGAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52896</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4097-1</td>\n",
       "      <td>GCTTCAATATGACAGAACTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52897</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4192-1</td>\n",
       "      <td>GGATGATCAAAAATACTGTA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52898</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4450-1</td>\n",
       "      <td>GCTTATTTAGTTTGGTTCAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52899</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52900 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id target_a_id  \\\n",
       "0                                         FTH1_1__FTH1_1        FTH1   \n",
       "1                                         FTH1_1__FTH1_2        FTH1   \n",
       "2                                         FTH1_1__FTH1_3        FTH1   \n",
       "3                                      FTH1_1__ANAPC10_1        FTH1   \n",
       "4                                      FTH1_1__ANAPC10_2        FTH1   \n",
       "...                                                  ...         ...   \n",
       "52895  0Safe_safe_U1_204550.4461_1__0Safe_safe_DTKP_2...       0Safe   \n",
       "52896  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52897  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52898  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52899  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "\n",
       "                        probe_a_id           probe_a_seq target_b_id  \\\n",
       "0                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "1                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "2                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "3                           FTH1-1  CACCATGGACAGGTAAACGT     ANAPC10   \n",
       "4                           FTH1-1  CACCATGGACAGGTAAACGT     ANAPC10   \n",
       "...                            ...                   ...         ...   \n",
       "52895  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52896  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52897  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52898  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52899  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "\n",
       "                          probe_b_id           probe_b_seq  \n",
       "0                             FTH1-1  CACCATGGACAGGTAAACGT  \n",
       "1                             FTH1-2  TGTTCACCTTGATATCCTGA  \n",
       "2                             FTH1-3  AATCTTCCTTCAGGATATCA  \n",
       "3                          ANAPC10-1  GACATACCCGAATTTCTTGA  \n",
       "4                          ANAPC10-2  ATTTAGTGAACATCCAATTC  \n",
       "...                              ...                   ...  \n",
       "52895  0Safe-safe-DTKP-204550.5816-1  GTGGAGTGGCGGAATCAGAT  \n",
       "52896    0Safe-safe-U1-204550.4097-1  GCTTCAATATGACAGAACTC  \n",
       "52897    0Safe-safe-U1-204550.4192-1  GGATGATCAAAAATACTGTA  \n",
       "52898    0Safe-safe-U1-204550.4450-1  GCTTATTTAGTTTGGTTCAA  \n",
       "52899    0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA  \n",
       "\n",
       "[52900 rows x 7 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "dde20409",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>F01_S4_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>F02_S5_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>F03_S6_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T01_S1_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T02_S2_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T03_S3_001_trimmed53_len_filtered_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...</td>\n",
       "      <td>344</td>\n",
       "      <td>572</td>\n",
       "      <td>563</td>\n",
       "      <td>524</td>\n",
       "      <td>638</td>\n",
       "      <td>647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...</td>\n",
       "      <td>57</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>72</td>\n",
       "      <td>95</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...</td>\n",
       "      <td>126</td>\n",
       "      <td>218</td>\n",
       "      <td>237</td>\n",
       "      <td>80</td>\n",
       "      <td>104</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...</td>\n",
       "      <td>75</td>\n",
       "      <td>161</td>\n",
       "      <td>113</td>\n",
       "      <td>49</td>\n",
       "      <td>63</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...</td>\n",
       "      <td>48</td>\n",
       "      <td>100</td>\n",
       "      <td>74</td>\n",
       "      <td>42</td>\n",
       "      <td>65</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52895</th>\n",
       "      <td>XIAP_3__WEE1_2</td>\n",
       "      <td>37</td>\n",
       "      <td>62</td>\n",
       "      <td>70</td>\n",
       "      <td>508</td>\n",
       "      <td>676</td>\n",
       "      <td>595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52896</th>\n",
       "      <td>XIAP_3__WEE1_3</td>\n",
       "      <td>54</td>\n",
       "      <td>109</td>\n",
       "      <td>75</td>\n",
       "      <td>185</td>\n",
       "      <td>230</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52897</th>\n",
       "      <td>XIAP_3__XIAP_1</td>\n",
       "      <td>411</td>\n",
       "      <td>538</td>\n",
       "      <td>585</td>\n",
       "      <td>429</td>\n",
       "      <td>560</td>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52898</th>\n",
       "      <td>XIAP_3__XIAP_2</td>\n",
       "      <td>285</td>\n",
       "      <td>498</td>\n",
       "      <td>504</td>\n",
       "      <td>371</td>\n",
       "      <td>497</td>\n",
       "      <td>482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52899</th>\n",
       "      <td>XIAP_3__XIAP_3</td>\n",
       "      <td>104</td>\n",
       "      <td>203</td>\n",
       "      <td>179</td>\n",
       "      <td>66</td>\n",
       "      <td>81</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52900 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id  \\\n",
       "0      0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...   \n",
       "1      0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...   \n",
       "2      0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...   \n",
       "3      0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...   \n",
       "4      0Safe_safe_ACOC_204550.4590_1__0Safe_safe_ACOC...   \n",
       "...                                                  ...   \n",
       "52895                                     XIAP_3__WEE1_2   \n",
       "52896                                     XIAP_3__WEE1_3   \n",
       "52897                                     XIAP_3__XIAP_1   \n",
       "52898                                     XIAP_3__XIAP_2   \n",
       "52899                                     XIAP_3__XIAP_3   \n",
       "\n",
       "       F01_S4_001_trimmed53_len_filtered_counts  \\\n",
       "0                                           344   \n",
       "1                                            57   \n",
       "2                                           126   \n",
       "3                                            75   \n",
       "4                                            48   \n",
       "...                                         ...   \n",
       "52895                                        37   \n",
       "52896                                        54   \n",
       "52897                                       411   \n",
       "52898                                       285   \n",
       "52899                                       104   \n",
       "\n",
       "       F02_S5_001_trimmed53_len_filtered_counts  \\\n",
       "0                                           572   \n",
       "1                                            68   \n",
       "2                                           218   \n",
       "3                                           161   \n",
       "4                                           100   \n",
       "...                                         ...   \n",
       "52895                                        62   \n",
       "52896                                       109   \n",
       "52897                                       538   \n",
       "52898                                       498   \n",
       "52899                                       203   \n",
       "\n",
       "       F03_S6_001_trimmed53_len_filtered_counts  \\\n",
       "0                                           563   \n",
       "1                                            68   \n",
       "2                                           237   \n",
       "3                                           113   \n",
       "4                                            74   \n",
       "...                                         ...   \n",
       "52895                                        70   \n",
       "52896                                        75   \n",
       "52897                                       585   \n",
       "52898                                       504   \n",
       "52899                                       179   \n",
       "\n",
       "       T01_S1_001_trimmed53_len_filtered_counts  \\\n",
       "0                                           524   \n",
       "1                                            72   \n",
       "2                                            80   \n",
       "3                                            49   \n",
       "4                                            42   \n",
       "...                                         ...   \n",
       "52895                                       508   \n",
       "52896                                       185   \n",
       "52897                                       429   \n",
       "52898                                       371   \n",
       "52899                                        66   \n",
       "\n",
       "       T02_S2_001_trimmed53_len_filtered_counts  \\\n",
       "0                                           638   \n",
       "1                                            95   \n",
       "2                                           104   \n",
       "3                                            63   \n",
       "4                                            65   \n",
       "...                                         ...   \n",
       "52895                                       676   \n",
       "52896                                       230   \n",
       "52897                                       560   \n",
       "52898                                       497   \n",
       "52899                                        81   \n",
       "\n",
       "       T03_S3_001_trimmed53_len_filtered_counts  \n",
       "0                                           647  \n",
       "1                                           178  \n",
       "2                                           131  \n",
       "3                                           106  \n",
       "4                                            70  \n",
       "...                                         ...  \n",
       "52895                                       595  \n",
       "52896                                       236  \n",
       "52897                                       572  \n",
       "52898                                       482  \n",
       "52899                                        90  \n",
       "\n",
       "[52900 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_file"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f09e69b",
   "metadata": {},
   "source": [
    "### Seq Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5200a50f",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref = pd.DataFrame(columns = [\"Guide_ID\", \"Sequence\"])\n",
    "sgRNA_ref['Guide_ID'] = pd.concat([lib_file['probe_a_id'], lib_file['probe_b_id']])\n",
    "sgRNA_ref['Sequence'] = pd.concat([lib_file['probe_a_seq'], lib_file['probe_b_seq']])\n",
    "sgRNA_ref.index = sgRNA_ref['Guide_ID']\n",
    "sgRNA_ref = sgRNA_ref.drop_duplicates()\n",
    "sgRNA_ref = sgRNA_ref.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2a23878d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgRNA_ref['Guide_ID'] = [i.upper() for i in sgRNA_ref['Guide_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e310cef7",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence_ref = sgRNA_ref.copy()\n",
    "sequence_ref['Target'] = [i.split('-')[0] for i in sequence_ref['Guide_ID']]\n",
    "sequence_ref.columns = ['sgRNA_guide_name', 'sgRNA_guide_seq', 'sgRNA_target_name']\n",
    "sequence_ref.loc[sequence_ref['sgRNA_target_name'].isin(controls['shuai_data']), 'sgRNA_target_name'] = 'control'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "638473d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA_guide_name</th>\n",
       "      <th>sgRNA_guide_seq</th>\n",
       "      <th>sgRNA_target_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FTH1-2</td>\n",
       "      <td>TGTTCACCTTGATATCCTGA</td>\n",
       "      <td>FTH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTH1-3</td>\n",
       "      <td>AATCTTCCTTCAGGATATCA</td>\n",
       "      <td>FTH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ANAPC10-1</td>\n",
       "      <td>GACATACCCGAATTTCTTGA</td>\n",
       "      <td>ANAPC10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ANAPC10-2</td>\n",
       "      <td>ATTTAGTGAACATCCAATTC</td>\n",
       "      <td>ANAPC10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>0SAFE-SAFE-DTKP-204550.5816-1</td>\n",
       "      <td>GTGGAGTGGCGGAATCAGAT</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.4097-1</td>\n",
       "      <td>GCTTCAATATGACAGAACTC</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.4192-1</td>\n",
       "      <td>GGATGATCAAAAATACTGTA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.4450-1</td>\n",
       "      <td>GCTTATTTAGTTTGGTTCAA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>0SAFE-SAFE-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sgRNA_guide_name       sgRNA_guide_seq sgRNA_target_name\n",
       "0                           FTH1-1  CACCATGGACAGGTAAACGT              FTH1\n",
       "1                           FTH1-2  TGTTCACCTTGATATCCTGA              FTH1\n",
       "2                           FTH1-3  AATCTTCCTTCAGGATATCA              FTH1\n",
       "3                        ANAPC10-1  GACATACCCGAATTTCTTGA           ANAPC10\n",
       "4                        ANAPC10-2  ATTTAGTGAACATCCAATTC           ANAPC10\n",
       "..                             ...                   ...               ...\n",
       "225  0SAFE-SAFE-DTKP-204550.5816-1  GTGGAGTGGCGGAATCAGAT           control\n",
       "226    0SAFE-SAFE-U1-204550.4097-1  GCTTCAATATGACAGAACTC           control\n",
       "227    0SAFE-SAFE-U1-204550.4192-1  GGATGATCAAAAATACTGTA           control\n",
       "228    0SAFE-SAFE-U1-204550.4450-1  GCTTATTTAGTTTGGTTCAA           control\n",
       "229    0SAFE-SAFE-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA           control\n",
       "\n",
       "[230 rows x 3 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "9b38ee37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(sequence_ref['sgRNA_target_name'] == 'control').sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a893302",
   "metadata": {},
   "source": [
    "### Counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "02dbdeea",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_file = lib_file.merge(counts_file, how = 'left', left_on = 'construct_id', right_on = 'construct_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "ac9910cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "      <th>F01_S4_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>F02_S5_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>F03_S6_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T01_S1_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T02_S2_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T03_S3_001_trimmed53_len_filtered_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FTH1_1__FTH1_1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>25</td>\n",
       "      <td>30</td>\n",
       "      <td>48</td>\n",
       "      <td>24</td>\n",
       "      <td>43</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FTH1_1__FTH1_2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-2</td>\n",
       "      <td>TGTTCACCTTGATATCCTGA</td>\n",
       "      <td>79</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "      <td>26</td>\n",
       "      <td>57</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTH1_1__FTH1_3</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-3</td>\n",
       "      <td>AATCTTCCTTCAGGATATCA</td>\n",
       "      <td>74</td>\n",
       "      <td>89</td>\n",
       "      <td>119</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FTH1_1__ANAPC10_1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>ANAPC10-1</td>\n",
       "      <td>GACATACCCGAATTTCTTGA</td>\n",
       "      <td>147</td>\n",
       "      <td>201</td>\n",
       "      <td>235</td>\n",
       "      <td>63</td>\n",
       "      <td>73</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FTH1_1__ANAPC10_2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>ANAPC10-2</td>\n",
       "      <td>ATTTAGTGAACATCCAATTC</td>\n",
       "      <td>70</td>\n",
       "      <td>127</td>\n",
       "      <td>119</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52895</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5816-1</td>\n",
       "      <td>GTGGAGTGGCGGAATCAGAT</td>\n",
       "      <td>65</td>\n",
       "      <td>104</td>\n",
       "      <td>114</td>\n",
       "      <td>83</td>\n",
       "      <td>97</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52896</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4097-1</td>\n",
       "      <td>GCTTCAATATGACAGAACTC</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52897</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4192-1</td>\n",
       "      <td>GGATGATCAAAAATACTGTA</td>\n",
       "      <td>28</td>\n",
       "      <td>46</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>34</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52898</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4450-1</td>\n",
       "      <td>GCTTATTTAGTTTGGTTCAA</td>\n",
       "      <td>20</td>\n",
       "      <td>43</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52899</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>17</td>\n",
       "      <td>24</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52900 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id target_a_id  \\\n",
       "0                                         FTH1_1__FTH1_1        FTH1   \n",
       "1                                         FTH1_1__FTH1_2        FTH1   \n",
       "2                                         FTH1_1__FTH1_3        FTH1   \n",
       "3                                      FTH1_1__ANAPC10_1        FTH1   \n",
       "4                                      FTH1_1__ANAPC10_2        FTH1   \n",
       "...                                                  ...         ...   \n",
       "52895  0Safe_safe_U1_204550.4461_1__0Safe_safe_DTKP_2...       0Safe   \n",
       "52896  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52897  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52898  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52899  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "\n",
       "                        probe_a_id           probe_a_seq target_b_id  \\\n",
       "0                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "1                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "2                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "3                           FTH1-1  CACCATGGACAGGTAAACGT     ANAPC10   \n",
       "4                           FTH1-1  CACCATGGACAGGTAAACGT     ANAPC10   \n",
       "...                            ...                   ...         ...   \n",
       "52895  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52896  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52897  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52898  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52899  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "\n",
       "                          probe_b_id           probe_b_seq  \\\n",
       "0                             FTH1-1  CACCATGGACAGGTAAACGT   \n",
       "1                             FTH1-2  TGTTCACCTTGATATCCTGA   \n",
       "2                             FTH1-3  AATCTTCCTTCAGGATATCA   \n",
       "3                          ANAPC10-1  GACATACCCGAATTTCTTGA   \n",
       "4                          ANAPC10-2  ATTTAGTGAACATCCAATTC   \n",
       "...                              ...                   ...   \n",
       "52895  0Safe-safe-DTKP-204550.5816-1  GTGGAGTGGCGGAATCAGAT   \n",
       "52896    0Safe-safe-U1-204550.4097-1  GCTTCAATATGACAGAACTC   \n",
       "52897    0Safe-safe-U1-204550.4192-1  GGATGATCAAAAATACTGTA   \n",
       "52898    0Safe-safe-U1-204550.4450-1  GCTTATTTAGTTTGGTTCAA   \n",
       "52899    0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA   \n",
       "\n",
       "       F01_S4_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            25   \n",
       "1                                            79   \n",
       "2                                            74   \n",
       "3                                           147   \n",
       "4                                            70   \n",
       "...                                         ...   \n",
       "52895                                        65   \n",
       "52896                                         2   \n",
       "52897                                        28   \n",
       "52898                                        20   \n",
       "52899                                        17   \n",
       "\n",
       "       F02_S5_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            30   \n",
       "1                                            77   \n",
       "2                                            89   \n",
       "3                                           201   \n",
       "4                                           127   \n",
       "...                                         ...   \n",
       "52895                                       104   \n",
       "52896                                        13   \n",
       "52897                                        46   \n",
       "52898                                        43   \n",
       "52899                                        24   \n",
       "\n",
       "       F03_S6_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            48   \n",
       "1                                            77   \n",
       "2                                           119   \n",
       "3                                           235   \n",
       "4                                           119   \n",
       "...                                         ...   \n",
       "52895                                       114   \n",
       "52896                                         3   \n",
       "52897                                        41   \n",
       "52898                                        42   \n",
       "52899                                        17   \n",
       "\n",
       "       T01_S1_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            24   \n",
       "1                                            26   \n",
       "2                                             3   \n",
       "3                                            63   \n",
       "4                                             6   \n",
       "...                                         ...   \n",
       "52895                                        83   \n",
       "52896                                         0   \n",
       "52897                                        23   \n",
       "52898                                         0   \n",
       "52899                                         4   \n",
       "\n",
       "       T02_S2_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            43   \n",
       "1                                            57   \n",
       "2                                             1   \n",
       "3                                            73   \n",
       "4                                             7   \n",
       "...                                         ...   \n",
       "52895                                        97   \n",
       "52896                                         0   \n",
       "52897                                        34   \n",
       "52898                                         0   \n",
       "52899                                        15   \n",
       "\n",
       "       T03_S3_001_trimmed53_len_filtered_counts  \n",
       "0                                            41  \n",
       "1                                            46  \n",
       "2                                             4  \n",
       "3                                            84  \n",
       "4                                            14  \n",
       "...                                         ...  \n",
       "52895                                       122  \n",
       "52896                                         1  \n",
       "52897                                        55  \n",
       "52898                                         1  \n",
       "52899                                        14  \n",
       "\n",
       "[52900 rows x 13 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "aff36aeb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['T1', 'T2', 'T3'], ['F1', 'F2', 'F3']]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_conditions['shuai_data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "cf003bb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_file['Count Replicates'] = counts_file[['T01_S1_001_trimmed53_len_filtered_counts', 'T02_S2_001_trimmed53_len_filtered_counts', 'T03_S3_001_trimmed53_len_filtered_counts',\n",
    "                                              'F01_S4_001_trimmed53_len_filtered_counts', 'F02_S5_001_trimmed53_len_filtered_counts', 'F03_S6_001_trimmed53_len_filtered_counts']].apply(    \n",
    "    lambda x: ';'.join(x.astype(str)),\n",
    "    axis=1\n",
    ")\n",
    "counts_file['Cell Line'] = ['Shuai_CL'] * counts_file.shape[0]\n",
    "counts_file[\"Condition\"] = [';'.join(['T1', 'T2', 'T3', 'F1', 'F2', 'F3'])] * counts_file.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ed4321bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>construct_id</th>\n",
       "      <th>target_a_id</th>\n",
       "      <th>probe_a_id</th>\n",
       "      <th>probe_a_seq</th>\n",
       "      <th>target_b_id</th>\n",
       "      <th>probe_b_id</th>\n",
       "      <th>probe_b_seq</th>\n",
       "      <th>F01_S4_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>F02_S5_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>F03_S6_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T01_S1_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T02_S2_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>T03_S3_001_trimmed53_len_filtered_counts</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FTH1_1__FTH1_1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>25</td>\n",
       "      <td>30</td>\n",
       "      <td>48</td>\n",
       "      <td>24</td>\n",
       "      <td>43</td>\n",
       "      <td>41</td>\n",
       "      <td>24;43;41;25;30;48</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FTH1_1__FTH1_2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-2</td>\n",
       "      <td>TGTTCACCTTGATATCCTGA</td>\n",
       "      <td>79</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "      <td>26</td>\n",
       "      <td>57</td>\n",
       "      <td>46</td>\n",
       "      <td>26;57;46;79;77;77</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTH1_1__FTH1_3</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-3</td>\n",
       "      <td>AATCTTCCTTCAGGATATCA</td>\n",
       "      <td>74</td>\n",
       "      <td>89</td>\n",
       "      <td>119</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>3;1;4;74;89;119</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FTH1_1__ANAPC10_1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>ANAPC10-1</td>\n",
       "      <td>GACATACCCGAATTTCTTGA</td>\n",
       "      <td>147</td>\n",
       "      <td>201</td>\n",
       "      <td>235</td>\n",
       "      <td>63</td>\n",
       "      <td>73</td>\n",
       "      <td>84</td>\n",
       "      <td>63;73;84;147;201;235</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FTH1_1__ANAPC10_2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>CACCATGGACAGGTAAACGT</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>ANAPC10-2</td>\n",
       "      <td>ATTTAGTGAACATCCAATTC</td>\n",
       "      <td>70</td>\n",
       "      <td>127</td>\n",
       "      <td>119</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>6;7;14;70;127;119</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52895</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_DTKP_2...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5816-1</td>\n",
       "      <td>GTGGAGTGGCGGAATCAGAT</td>\n",
       "      <td>65</td>\n",
       "      <td>104</td>\n",
       "      <td>114</td>\n",
       "      <td>83</td>\n",
       "      <td>97</td>\n",
       "      <td>122</td>\n",
       "      <td>83;97;122;65;104;114</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52896</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4097-1</td>\n",
       "      <td>GCTTCAATATGACAGAACTC</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0;0;1;2;13;3</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52897</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4192-1</td>\n",
       "      <td>GGATGATCAAAAATACTGTA</td>\n",
       "      <td>28</td>\n",
       "      <td>46</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>34</td>\n",
       "      <td>55</td>\n",
       "      <td>23;34;55;28;46;41</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52898</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4450-1</td>\n",
       "      <td>GCTTATTTAGTTTGGTTCAA</td>\n",
       "      <td>20</td>\n",
       "      <td>43</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0;0;1;20;43;42</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52899</th>\n",
       "      <td>0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>GTCTTAAAGAGTAAAGTACA</td>\n",
       "      <td>17</td>\n",
       "      <td>24</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "      <td>4;15;14;17;24;17</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52900 rows  16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            construct_id target_a_id  \\\n",
       "0                                         FTH1_1__FTH1_1        FTH1   \n",
       "1                                         FTH1_1__FTH1_2        FTH1   \n",
       "2                                         FTH1_1__FTH1_3        FTH1   \n",
       "3                                      FTH1_1__ANAPC10_1        FTH1   \n",
       "4                                      FTH1_1__ANAPC10_2        FTH1   \n",
       "...                                                  ...         ...   \n",
       "52895  0Safe_safe_U1_204550.4461_1__0Safe_safe_DTKP_2...       0Safe   \n",
       "52896  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52897  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52898  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "52899  0Safe_safe_U1_204550.4461_1__0Safe_safe_U1_204...       0Safe   \n",
       "\n",
       "                        probe_a_id           probe_a_seq target_b_id  \\\n",
       "0                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "1                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "2                           FTH1-1  CACCATGGACAGGTAAACGT        FTH1   \n",
       "3                           FTH1-1  CACCATGGACAGGTAAACGT     ANAPC10   \n",
       "4                           FTH1-1  CACCATGGACAGGTAAACGT     ANAPC10   \n",
       "...                            ...                   ...         ...   \n",
       "52895  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52896  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52897  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52898  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "52899  0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA       0Safe   \n",
       "\n",
       "                          probe_b_id           probe_b_seq  \\\n",
       "0                             FTH1-1  CACCATGGACAGGTAAACGT   \n",
       "1                             FTH1-2  TGTTCACCTTGATATCCTGA   \n",
       "2                             FTH1-3  AATCTTCCTTCAGGATATCA   \n",
       "3                          ANAPC10-1  GACATACCCGAATTTCTTGA   \n",
       "4                          ANAPC10-2  ATTTAGTGAACATCCAATTC   \n",
       "...                              ...                   ...   \n",
       "52895  0Safe-safe-DTKP-204550.5816-1  GTGGAGTGGCGGAATCAGAT   \n",
       "52896    0Safe-safe-U1-204550.4097-1  GCTTCAATATGACAGAACTC   \n",
       "52897    0Safe-safe-U1-204550.4192-1  GGATGATCAAAAATACTGTA   \n",
       "52898    0Safe-safe-U1-204550.4450-1  GCTTATTTAGTTTGGTTCAA   \n",
       "52899    0Safe-safe-U1-204550.4461-1  GTCTTAAAGAGTAAAGTACA   \n",
       "\n",
       "       F01_S4_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            25   \n",
       "1                                            79   \n",
       "2                                            74   \n",
       "3                                           147   \n",
       "4                                            70   \n",
       "...                                         ...   \n",
       "52895                                        65   \n",
       "52896                                         2   \n",
       "52897                                        28   \n",
       "52898                                        20   \n",
       "52899                                        17   \n",
       "\n",
       "       F02_S5_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            30   \n",
       "1                                            77   \n",
       "2                                            89   \n",
       "3                                           201   \n",
       "4                                           127   \n",
       "...                                         ...   \n",
       "52895                                       104   \n",
       "52896                                        13   \n",
       "52897                                        46   \n",
       "52898                                        43   \n",
       "52899                                        24   \n",
       "\n",
       "       F03_S6_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            48   \n",
       "1                                            77   \n",
       "2                                           119   \n",
       "3                                           235   \n",
       "4                                           119   \n",
       "...                                         ...   \n",
       "52895                                       114   \n",
       "52896                                         3   \n",
       "52897                                        41   \n",
       "52898                                        42   \n",
       "52899                                        17   \n",
       "\n",
       "       T01_S1_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            24   \n",
       "1                                            26   \n",
       "2                                             3   \n",
       "3                                            63   \n",
       "4                                             6   \n",
       "...                                         ...   \n",
       "52895                                        83   \n",
       "52896                                         0   \n",
       "52897                                        23   \n",
       "52898                                         0   \n",
       "52899                                         4   \n",
       "\n",
       "       T02_S2_001_trimmed53_len_filtered_counts  \\\n",
       "0                                            43   \n",
       "1                                            57   \n",
       "2                                             1   \n",
       "3                                            73   \n",
       "4                                             7   \n",
       "...                                         ...   \n",
       "52895                                        97   \n",
       "52896                                         0   \n",
       "52897                                        34   \n",
       "52898                                         0   \n",
       "52899                                        15   \n",
       "\n",
       "       T03_S3_001_trimmed53_len_filtered_counts      Count Replicates  \\\n",
       "0                                            41     24;43;41;25;30;48   \n",
       "1                                            46     26;57;46;79;77;77   \n",
       "2                                             4       3;1;4;74;89;119   \n",
       "3                                            84  63;73;84;147;201;235   \n",
       "4                                            14     6;7;14;70;127;119   \n",
       "...                                         ...                   ...   \n",
       "52895                                       122  83;97;122;65;104;114   \n",
       "52896                                         1          0;0;1;2;13;3   \n",
       "52897                                        55     23;34;55;28;46;41   \n",
       "52898                                         1        0;0;1;20;43;42   \n",
       "52899                                        14      4;15;14;17;24;17   \n",
       "\n",
       "      Cell Line          Condition  \n",
       "0      Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "1      Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "2      Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "3      Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "4      Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "...         ...                ...  \n",
       "52895  Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "52896  Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "52897  Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "52898  Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "52899  Shuai_CL  T1;T2;T3;F1;F2;F3  \n",
       "\n",
       "[52900 rows x 16 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c3fe954c",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref = pd.DataFrame(columns = [\"Guide 1\", \"Guide 2\", \"Gene 1\", \"Gene 2\", \"Count Replicates\", \"Type\", \"Sequencing\", \"Cell Line\", \"Condition\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6d66e133",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['diehl_data', 'han_data', 'horlbeck_data', 'ito_data', 'parrish_data', 'shen_data', 'thompson_data', 'wong_data', 'zhao_data', 'shantang_data', 'najm_data', 'shantang_2CL_data', 'shuai_data'])"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_name_to_pubmed_id.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "c091e4f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_ref['Guide 1'] = counts_file['probe_a_id'].tolist()\n",
    "counts_ref['Guide 2'] = counts_file['probe_b_id'].tolist()\n",
    "counts_ref['Gene 1'] = counts_file['target_a_id'].tolist()\n",
    "counts_ref['Gene 2'] = counts_file['target_b_id'].tolist()\n",
    "counts_ref['Count Replicates'] = counts_file['Count Replicates'].tolist()\n",
    "counts_ref['Cell Line'] = counts_file['Cell Line'].tolist()\n",
    "counts_ref['Condition'] = counts_file['Condition'].tolist()\n",
    "counts_ref['Sequencing'] = ['Combinatorial CRISPR'] * counts_ref.shape[0]\n",
    "counts_ref[\"Study\"] = [study_name_to_pubmed_id['shuai_data']] * counts_ref.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3668abca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guide 1</th>\n",
       "      <th>Guide 2</th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>Count Replicates</th>\n",
       "      <th>Type</th>\n",
       "      <th>Sequencing</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>24;43;41;25;30;48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>FTH1-2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>26;57;46;79;77;77</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>FTH1-3</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>3;1;4;74;89;119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>ANAPC10-1</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>63;73;84;147;201;235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FTH1-1</td>\n",
       "      <td>ANAPC10-2</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>6;7;14;70;127;119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52895</th>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>0Safe-safe-DTKP-204550.5816-1</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>83;97;122;65;104;114</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52896</th>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>0Safe-safe-U1-204550.4097-1</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0;0;1;2;13;3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52897</th>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>0Safe-safe-U1-204550.4192-1</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>23;34;55;28;46;41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52898</th>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>0Safe-safe-U1-204550.4450-1</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0;0;1;20;43;42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52899</th>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>0Safe-safe-U1-204550.4461-1</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>0Safe</td>\n",
       "      <td>4;15;14;17;24;17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Combinatorial CRISPR</td>\n",
       "      <td>Shuai_CL</td>\n",
       "      <td>T1;T2;T3;F1;F2;F3</td>\n",
       "      <td>Shuai</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52900 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Guide 1                        Guide 2 Gene 1  \\\n",
       "0                           FTH1-1                         FTH1-1   FTH1   \n",
       "1                           FTH1-1                         FTH1-2   FTH1   \n",
       "2                           FTH1-1                         FTH1-3   FTH1   \n",
       "3                           FTH1-1                      ANAPC10-1   FTH1   \n",
       "4                           FTH1-1                      ANAPC10-2   FTH1   \n",
       "...                            ...                            ...    ...   \n",
       "52895  0Safe-safe-U1-204550.4461-1  0Safe-safe-DTKP-204550.5816-1  0Safe   \n",
       "52896  0Safe-safe-U1-204550.4461-1    0Safe-safe-U1-204550.4097-1  0Safe   \n",
       "52897  0Safe-safe-U1-204550.4461-1    0Safe-safe-U1-204550.4192-1  0Safe   \n",
       "52898  0Safe-safe-U1-204550.4461-1    0Safe-safe-U1-204550.4450-1  0Safe   \n",
       "52899  0Safe-safe-U1-204550.4461-1    0Safe-safe-U1-204550.4461-1  0Safe   \n",
       "\n",
       "        Gene 2      Count Replicates Type            Sequencing Cell Line  \\\n",
       "0         FTH1     24;43;41;25;30;48  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "1         FTH1     26;57;46;79;77;77  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "2         FTH1       3;1;4;74;89;119  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "3      ANAPC10  63;73;84;147;201;235  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "4      ANAPC10     6;7;14;70;127;119  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "...        ...                   ...  ...                   ...       ...   \n",
       "52895    0Safe  83;97;122;65;104;114  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "52896    0Safe          0;0;1;2;13;3  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "52897    0Safe     23;34;55;28;46;41  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "52898    0Safe        0;0;1;20;43;42  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "52899    0Safe      4;15;14;17;24;17  NaN  Combinatorial CRISPR  Shuai_CL   \n",
       "\n",
       "               Condition  Study  \n",
       "0      T1;T2;T3;F1;F2;F3  Shuai  \n",
       "1      T1;T2;T3;F1;F2;F3  Shuai  \n",
       "2      T1;T2;T3;F1;F2;F3  Shuai  \n",
       "3      T1;T2;T3;F1;F2;F3  Shuai  \n",
       "4      T1;T2;T3;F1;F2;F3  Shuai  \n",
       "...                  ...    ...  \n",
       "52895  T1;T2;T3;F1;F2;F3  Shuai  \n",
       "52896  T1;T2;T3;F1;F2;F3  Shuai  \n",
       "52897  T1;T2;T3;F1;F2;F3  Shuai  \n",
       "52898  T1;T2;T3;F1;F2;F3  Shuai  \n",
       "52899  T1;T2;T3;F1;F2;F3  Shuai  \n",
       "\n",
       "[52900 rows x 10 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a9b5bbf",
   "metadata": {},
   "source": [
    "## SL Scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d869b741",
   "metadata": {},
   "outputs": [],
   "source": [
    "# there are no scores available, generate them instead"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "5c02350f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are no scores, but there are counts...Generating Placeholder...\n",
      "Starting processing...\n",
      "Score reference...\n",
      "Controls within SL score that are removed: \n",
      "71\n",
      "---\n",
      "No scores/stats cutoffs are available, possibly generated. Setting all to be NOT SL\n",
      "Counts reference...\n",
      "Number of double pairs: 44730\n",
      "Number of controls: 289\n",
      "Number of singles: 7881\n",
      "Sequence reference...\n",
      "Done! Returning...\n"
     ]
    }
   ],
   "source": [
    "# prepare the files for insertion\n",
    "db_inserts = prepare_study_for_export(sequence_ref = sequence_ref.copy(), counts_ref = counts_ref.copy(), score_ref = None, study_controls = controls['shuai_data'], study_conditions = study_conditions['shuai_data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "4a88bc34",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updating gene pairs with seperator |...\n",
      "Final QC...\n",
      "Beginning transaction...\n",
      "Done sequence\n",
      "Done counts\n",
      "Done score\n",
      "Successfully inserted!\n",
      "Added Record stats...\n",
      "Sequence insert: 230\n",
      "Counts insert: 52900\n",
      "Score insert: 2485\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "# insert to the database\n",
    "insert_study_to_db(engine_link = SLKB_engine, db_inserts = db_inserts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e4e3af9",
   "metadata": {},
   "source": [
    "## Done!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74ade34c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9 (Conda 2022.05) [python/3.9-2022.05]",
   "language": "python",
   "name": "python39_202205"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
